diff -Naur mythstream_mythtv-r21640-old/settings.pro mythstream_mythtv-r21640-new/settings.pro
--- mythstream_mythtv-r21640-old/settings.pro	2009-10-13 14:29:56.000000000 -0700
+++ mythstream_mythtv-r21640-new/settings.pro	2009-10-13 14:50:04.000000000 -0700
@@ -14,15 +14,15 @@
     RUNPREFIX = $$PREFIX
 }
 
-INCLUDEPATH += $${PREFIX}/include
-INCLUDEPATH += $${PREFIX}/include/mythtv
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythdb
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmyth
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythdb
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmyth
 
-INCLUDEPATH += $${PREFIX}/include/qt4/Qt
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/qt4/Qt
 
-#INCLUDEPATH *= /usr/local/include
+#INCLUDEPATH *= $${SYSROOT}/usr/local/include
 
 DEFINES += _GNU_SOURCE
 DEFINES += PREFIX=\\\"$${RUNPREFIX}\\\"
