diff -Naur openssh-5.6p1-old/configure.ac openssh-5.6p1-new/configure.ac
--- openssh-5.6p1-old/configure.ac	2010-08-15 20:15:23.000000000 -0700
+++ openssh-5.6p1-new/configure.ac	2010-08-27 10:27:22.000000000 -0700
@@ -1273,7 +1273,7 @@
 				LDFLAGS="-L${withval}/lib ${LDFLAGS}"
 			fi
 		fi
-		if test "x$use_pkgconfig_for_libedit" == "xyes"; then
+		if test "x$use_pkgconfig_for_libedit" = "xyes"; then
 			LIBEDIT=`$PKGCONFIG --libs-only-l libedit`
 			CPPFLAGS="$CPPFLAGS `$PKGCONFIG --cflags libedit`"
 		else
