diff -Naur linux-2.6.25-rc7-old/include/linux/time.h linux-2.6.25-rc7-new/include/linux/time.h
--- linux-2.6.25-rc7-old/include/linux/time.h	2008-03-25 18:38:14.000000000 -0700
+++ linux-2.6.25-rc7-new/include/linux/time.h	2008-03-26 09:19:58.000000000 -0700
@@ -170,7 +170,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
