diff -Naur linux-2.6.25-rc8-old/include/linux/time.h linux-2.6.25-rc8-new/include/linux/time.h
--- linux-2.6.25-rc8-old/include/linux/time.h	2008-04-01 12:44:26.000000000 -0700
+++ linux-2.6.25-rc8-new/include/linux/time.h	2008-04-01 19:16:20.000000000 -0700
@@ -170,7 +170,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
