diff -Naur PerlSharp-old/PerlSharp.cgi PerlSharp-new/PerlSharp.cgi
--- PerlSharp-old/PerlSharp.cgi	2008-02-20 23:16:19.000000000 -0800
+++ PerlSharp-new/PerlSharp.cgi	2008-02-20 23:17:54.000000000 -0800
@@ -135,7 +135,7 @@
 			@Files = sort @Files;
 			foreach $File (@Files) {
 					$CurFile = "$Dir/$File";
-					if (!-d $CurFile && $File ne "." && $File ne ".." && ($File =~ /\.pm$/i || $File =~ /\.pl$/i || $File =~ /\.cgi$/i)) {
+					if (!-d $CurFile && $File ne "." && $File ne ".." && ($File =~ /\.pm$/i || $File =~ /\.pl$/i || $File =~ /\.cgi$/i || $File =~ /\.al$/i || $File =~ /\.ix$/i)) {
 								&ProcessPerlFile($CurFile);
 					}
 			}
@@ -204,7 +204,7 @@
 							push @Directories, $CurFile;
 							next;
 					} else {
-							if ($File ne "." && $File ne ".." && ($File =~ /\.pm$/i || $File =~ /\.pl$/i || $File =~ /\.cgi$/i)) {
+							if ($File ne "." && $File ne ".." && ($File =~ /\.pm$/i || $File =~ /\.pl$/i || $File =~ /\.cgi$/i || $File =~ /\.al$/i || $File =~ /\.ix$/i)) {
 										&ProcessPerlFile($CurFile);
 							}
 					}
