diff -Naur xf86-input-evdev-2.1.1-old/configure.ac xf86-input-evdev-2.1.1-new/configure.ac
--- xf86-input-evdev-2.1.1-old/configure.ac	2009-01-12 17:36:14.000000000 -0800
+++ xf86-input-evdev-2.1.1-new/configure.ac	2009-01-12 18:00:54.000000000 -0800
@@ -57,7 +57,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, xorg-server xproto $REQUIRED_MODULES)
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 AC_SUBST(sdkdir)
 
 CFLAGS="$CFLAGS $XORG_CFLAGS "' -I$(top_srcdir)/src'
