diff -Naur alsa-driver-1.0.22.1-old/alsa-kernel/pci/hda/patch_analog.c alsa-driver-1.0.22.1-new/alsa-kernel/pci/hda/patch_analog.c
--- alsa-driver-1.0.22.1-old/alsa-kernel/pci/hda/patch_analog.c	2009-12-28 06:21:33.000000000 -0800
+++ alsa-driver-1.0.22.1-new/alsa-kernel/pci/hda/patch_analog.c	2009-12-28 21:52:46.000000000 -0800
@@ -3075,7 +3075,10 @@
 	SND_PCI_QUIRK(0x1043, 0x81ec, "Asus P5B-DLX", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x81f6, "Asus M2N-SLI", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8277, "Asus P5K-E/WIFI-AP", AD1988_6STACK_DIG),
+	SND_PCI_QUIRK(0x1043, 0x829f, "Asus M3N78-VM", AD1988_6STACK_DIG),
+	SND_PCI_QUIRK(0x1043, 0x82fe, "Asus M3N78-EM / P5N7A-VM", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8311, "Asus P5Q-Premium/Pro", AD1988_6STACK_DIG),
+	SND_PCI_QUIRK(0x1043, 0x8345, "Asus M3N78-VM", AD1988_6STACK_DIG),
 	{}
 };
 
diff -Naur alsa-driver-1.0.22.1-old/alsa-kernel/pci/hda/patch_realtek.c alsa-driver-1.0.22.1-new/alsa-kernel/pci/hda/patch_realtek.c
--- alsa-driver-1.0.22.1-old/alsa-kernel/pci/hda/patch_realtek.c	2009-12-28 06:21:33.000000000 -0800
+++ alsa-driver-1.0.22.1-new/alsa-kernel/pci/hda/patch_realtek.c	2009-12-28 21:52:46.000000000 -0800
@@ -9059,6 +9059,7 @@
 	SND_PCI_QUIRK(0x1043, 0x817f, "Asus P5LD2", ALC882_6ST_DIG),
 	SND_PCI_QUIRK(0x1043, 0x81d8, "Asus P5WD", ALC882_6ST_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8249, "Asus M2A-VM HDMI", ALC883_3ST_6ch_DIG),
+	SND_PCI_QUIRK(0x1043, 0x8290, "Asus M2N-VM DVI", ALC883_3ST_6ch_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8284, "Asus Z37E", ALC883_6ST_DIG),
 	SND_PCI_QUIRK(0x1043, 0x82fe, "Asus P5Q-EM HDMI", ALC1200_ASUS_P5Q),
 	SND_PCI_QUIRK(0x1043, 0x835f, "Asus Eee 1601", ALC888_ASUS_EEE1601),
