diff -Naur mythtv-0.21-19133-old/libs/libmythtv/videoout_xv.cpp mythtv-0.21-19133-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-0.21-19133-old/libs/libmythtv/videoout_xv.cpp	2008-11-19 19:14:16.000000000 -0800
+++ mythtv-0.21-19133-new/libs/libmythtv/videoout_xv.cpp	2008-11-19 19:14:20.000000000 -0800
@@ -1521,7 +1521,12 @@
         for (; it != renderers.end(); ++it)
             tmp += *it + ",";
 
-        renderer = renderers[0];
+        if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-blit") && (renderers.contains("xv-blit")))
+            renderer = "xv-blit";
+        else if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-opengl") && (renderers.contains("opengl")))
+            renderer = "opengl";
+        else
+            renderer = renderers[0];
         VERBOSE(VB_IMPORTANT, LOC + QString(
                     "Desired video renderer '%1' not available.\n\t\t\t"
                     "codec '%2' makes '%3' available, using '%4' instead.")
