diff -Naur mythtv-0.21-17404-old/libs/libmythtv/tv_play.cpp mythtv-0.21-17404-new/libs/libmythtv/tv_play.cpp
--- mythtv-0.21-17404-old/libs/libmythtv/tv_play.cpp	2008-05-26 07:33:01.000000000 -0700
+++ mythtv-0.21-17404-new/libs/libmythtv/tv_play.cpp	2008-05-26 07:33:02.000000000 -0700
@@ -1789,7 +1789,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
