diff -Naur mythtv-0.21-18566-old/configure mythtv-0.21-18566-new/configure
--- mythtv-0.21-18566-old/configure	2008-10-08 11:47:34.000000000 -0700
+++ mythtv-0.21-18566-new/configure	2008-10-08 11:50:29.000000000 -0700
@@ -2713,7 +2713,7 @@
     fi
 fi
 
-enabled libfftw3 && has_library libfftw3_threads && has_header fftw3.h ||
+enabled libfftw3 && has_library libfftw3f_threads && has_header fftw3.h ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
