diff -Naur mythtv-0.21-25121-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.21-25121-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.21-25121-old/libs/libmyth/mediamonitor-unix.cpp	2010-06-20 07:20:39.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmyth/mediamonitor-unix.cpp	2010-06-20 07:20:40.000000000 -0700
@@ -184,7 +184,8 @@
 {
 #ifdef linux
     QProcess udevinfo(this);
-    udevinfo.addArgument("udevinfo");
+    udevinfo.addArgument("udevadm");
+    udevinfo.addArgument("info");
     udevinfo.addArgument("-q");
     udevinfo.addArgument("name");
     udevinfo.addArgument("-rp");
