diff -Naur mythtv-0.21-19556-old/configure mythtv-0.21-19556-new/configure
--- mythtv-0.21-19556-old/configure	2009-01-03 19:02:09.000000000 -0800
+++ mythtv-0.21-19556-new/configure	2009-01-03 19:02:10.000000000 -0800
@@ -2715,7 +2715,7 @@
     fi
 fi
 
-enabled libfftw3 && has_library libfftw3_threads && has_header fftw3.h ||
+enabled libfftw3 && has_library libfftw3f_threads && has_header fftw3.h ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
