diff -Naur mythtv-0.21-25121-old/libs/libmyth/mythcdrom-linux.cpp mythtv-0.21-25121-new/libs/libmyth/mythcdrom-linux.cpp
--- mythtv-0.21-25121-old/libs/libmyth/mythcdrom-linux.cpp	2010-06-20 07:18:31.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmyth/mythcdrom-linux.cpp	2010-08-03 20:39:58.000000000 -0700
@@ -1,5 +1,6 @@
 #include "mythcdrom.h"
 #include "mythcdrom-linux.h"
+#include <sys/stat.h>
 #include <sys/ioctl.h>       // ioctls
 #include <linux/cdrom.h>     // old ioctls for cdrom
 #include <scsi/sg.h>
