diff -Naur mythtv-0.21-20088-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.21-20088-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.21-20088-old/libs/libmyth/mediamonitor-unix.cpp	2009-03-02 17:53:58.000000000 -0800
+++ mythtv-0.21-20088-new/libs/libmyth/mediamonitor-unix.cpp	2009-03-02 17:54:00.000000000 -0800
@@ -184,7 +184,8 @@
 {
 #ifdef linux
     QProcess udevinfo(this);
-    udevinfo.addArgument("udevinfo");
+    udevinfo.addArgument("udevadm");
+    udevinfo.addArgument("info");
     udevinfo.addArgument("-q");
     udevinfo.addArgument("name");
     udevinfo.addArgument("-rp");
