diff -Naur mythtv-0.21-20210-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.21-20210-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.21-20210-old/libs/libmyth/mediamonitor-unix.cpp	2009-03-14 10:21:34.000000000 -0700
+++ mythtv-0.21-20210-new/libs/libmyth/mediamonitor-unix.cpp	2009-03-14 10:21:35.000000000 -0700
@@ -184,7 +184,8 @@
 {
 #ifdef linux
     QProcess udevinfo(this);
-    udevinfo.addArgument("udevinfo");
+    udevinfo.addArgument("udevadm");
+    udevinfo.addArgument("info");
     udevinfo.addArgument("-q");
     udevinfo.addArgument("name");
     udevinfo.addArgument("-rp");
