diff -Naur mythtv-0.21-25121-old/libs/libmyth/mythmediamonitor.cpp mythtv-0.21-25121-new/libs/libmyth/mythmediamonitor.cpp
--- mythtv-0.21-25121-old/libs/libmyth/mythmediamonitor.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmyth/mythmediamonitor.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -281,7 +281,7 @@
     if (ignore.length())
         m_IgnoreList = QStringList::split(',', ignore);
     else
-        m_IgnoreList = QStringList::QStringList();  // Force empty list
+        m_IgnoreList = QStringList();  // Force empty list
 
     if (m_StartThread)
         VERBOSE(VB_MEDIA, "Creating MediaMonitor, SendEvents="
diff -Naur mythtv-0.21-25121-old/libs/libmythtv/datadirect.cpp mythtv-0.21-25121-new/libs/libmythtv/datadirect.cpp
--- mythtv-0.21-25121-old/libs/libmythtv/datadirect.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmythtv/datadirect.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -1242,7 +1242,7 @@
 bool DataDirectProcessor::GrabLineupsOnly(void)
 {
     const QDateTime start = QDateTime(QDate::currentDate().addDays(2),
-                                      QTime::QTime(23, 59, 0));
+                                      QTime(23, 59, 0));
     const QDateTime end   = start.addSecs(1);
 
     return GrabData(start, end);
diff -Naur mythtv-0.21-25121-old/libs/libmythtv/jobqueue.cpp mythtv-0.21-25121-new/libs/libmythtv/jobqueue.cpp
--- mythtv-0.21-25121-old/libs/libmythtv/jobqueue.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmythtv/jobqueue.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -594,7 +594,7 @@
             if (defer)
             {
                 schedruntime = schedruntime.addDays(defer);
-                schedruntime.setTime(QTime::QTime(0,0));
+                schedruntime.setTime(QTime(0,0));
             }
 
             QueueJob(JOB_TRANSCODE, chanid, starttime, args, comment, host,
@@ -1127,14 +1127,14 @@
     if (!queueStartTime.isValid())
     {
         VERBOSE(VB_IMPORTANT, "Invalid JobQueueWindowStart time, using 00:00");
-        queueStartTime = QTime::QTime(0, 0);
+        queueStartTime = QTime(0, 0);
     }
 
     queueEndTime = QTime::fromString(queueEndTimeStr);
     if (!queueEndTime.isValid())
     {
         VERBOSE(VB_IMPORTANT, "Invalid JobQueueWindowEnd time, using 23:59");
-        queueEndTime = QTime::QTime(23, 59);
+        queueEndTime = QTime(23, 59);
     }
     
     if ((queueStartTime <= curTime) && (curTime < queueEndTime))
diff -Naur mythtv-0.21-25121-old/libs/libmythtv/programinfo.cpp mythtv-0.21-25121-new/libs/libmythtv/programinfo.cpp
--- mythtv-0.21-25121-old/libs/libmythtv/programinfo.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmythtv/programinfo.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -105,7 +105,7 @@
     endts = startts;
     recstartts = startts;
     recendts = startts;
-    originalAirDate = QDate::QDate (0, 1, 1);
+    originalAirDate = QDate (0, 1, 1);
     lastmodified = startts;
     lastInUseTime = startts.addSecs(-4 * 60 * 60);
 
@@ -906,7 +906,7 @@
         
         if (query.value(17).isNull() || query.value(17).toString().isEmpty())
         {
-            proginfo->originalAirDate = QDate::QDate (0, 1, 1);
+            proginfo->originalAirDate = QDate (0, 1, 1);
             proginfo->hasAirDate = false;
         }
         else
@@ -3925,7 +3925,7 @@
     }
     if (findid > 0)
     {
-        QDate fdate = QDate::QDate (1970, 1, 1);
+        QDate fdate = QDate (1970, 1, 1);
         fdate = fdate.addDays(findid - 719528);
         ADD_PAR(QObject::tr("Find ID"), QString("%1 (%2)").arg(findid)
                 .arg(fdate.toString(fullDateFormat)), msg)
@@ -4678,7 +4678,7 @@
 
         if (query.value(17).isNull() || query.value(17).toString().isEmpty())
         {
-            p->originalAirDate = QDate::QDate (0, 1, 1);
+            p->originalAirDate = QDate (0, 1, 1);
             p->hasAirDate = false;
         }
         else
@@ -4857,7 +4857,7 @@
             if (query.value(26).isNull() || 
                 query.value(26).toString().isEmpty())
             {
-                proginfo->originalAirDate = QDate::QDate (0, 1, 1);
+                proginfo->originalAirDate = QDate (0, 1, 1);
                 proginfo->hasAirDate      = false;
             }
             else
diff -Naur mythtv-0.21-25121-old/libs/libmythtv/scheduledrecording.cpp mythtv-0.21-25121-new/libs/libmythtv/scheduledrecording.cpp
--- mythtv-0.21-25121-old/libs/libmythtv/scheduledrecording.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmythtv/scheduledrecording.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -213,7 +213,7 @@
         subtitle->setValue(from);
         description->setValue(forwhat);
         findday->setValue((startDate->dateValue().dayOfWeek() + 1) % 7);
-        QDate epoch = QDate::QDate (1970, 1, 1);
+        QDate epoch = QDate (1970, 1, 1);
         findid->setValue(epoch.daysTo(startDate->dateValue()) + 719528);
     } 
 }
@@ -812,7 +812,7 @@
 {
     QDate date = QDate::currentDate();
     QTime time = QTime::currentTime();
-    QDate epoch = QDate::QDate (1970, 1, 1);
+    QDate epoch = QDate (1970, 1, 1);
 
     id->setValue(0);
     title->setValue("");
@@ -899,7 +899,7 @@
             findday->setValue((proginfo->startts.date().dayOfWeek() + 1) % 7);
             findtime->setValue(proginfo->startts.time());
 
-            QDate epoch = QDate::QDate (1970, 1, 1);
+            QDate epoch = QDate (1970, 1, 1);
             findid->setValue(epoch.daysTo(proginfo->startts.date()) + 719528);
         }
         else
@@ -908,7 +908,7 @@
                 findid->setValue(proginfo->findid);
             else
             {
-                QDate epoch = QDate::QDate (1970, 1, 1);
+                QDate epoch = QDate (1970, 1, 1);
                 findid->setValue(epoch.daysTo(proginfo->startts.date()) +
                                               719528);
             }
diff -Naur mythtv-0.21-25121-old/programs/mythbackend/mainserver.cpp mythtv-0.21-25121-new/programs/mythbackend/mainserver.cpp
--- mythtv-0.21-25121-old/programs/mythbackend/mainserver.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/programs/mythbackend/mainserver.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -1200,7 +1200,7 @@
             if (query.value(26).isNull() || 
                 query.value(26).toString().isEmpty())
             {
-                proginfo->originalAirDate = QDate::QDate (0, 1, 1);
+                proginfo->originalAirDate = QDate (0, 1, 1);
                 proginfo->hasAirDate      = false;
             }
             else
diff -Naur mythtv-0.21-25121-old/programs/mythbackend/scheduler.cpp mythtv-0.21-25121-new/programs/mythbackend/scheduler.cpp
--- mythtv-0.21-25121-old/programs/mythbackend/scheduler.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/programs/mythbackend/scheduler.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -2371,7 +2371,7 @@
     }
     else if (query.size())
     {
-        QDate epoch = QDate::QDate (1970, 1, 1);
+        QDate epoch = QDate (1970, 1, 1);
         int findtoday =  epoch.daysTo(QDate::currentDate()) + 719528;
         query.prepare("UPDATE record set findid = :FINDID "
                       "WHERE type = :FINDONE AND findid <= 0;");
@@ -2855,7 +2855,7 @@
 
         if (result.value(32).isNull())
         {
-            p->originalAirDate = QDate::QDate (0, 1, 1);
+            p->originalAirDate = QDate (0, 1, 1);
             p->hasAirDate = false;
         }
         else
diff -Naur mythtv-0.21-25121-old/programs/mythfrontend/manualschedule.cpp mythtv-0.21-25121-new/programs/mythfrontend/manualschedule.cpp
--- mythtv-0.21-25121-old/programs/mythfrontend/manualschedule.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/programs/mythfrontend/manualschedule.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -123,7 +123,7 @@
     QTime thisTime = m_nowDateTime.time();
     thisTime = thisTime.addSecs((30 - thisTime.minute() % 30) * 60);
     
-    if (thisTime < QTime::QTime(0,30))
+    if (thisTime < QTime(0,30))
         m_startdate->setCurrentItem(m_startdate->currentItem() + 1);
 
     message = tr("Time:");
@@ -144,7 +144,7 @@
 
     for(int m_index = -1; m_index <= 24; m_index++)
     {
-        m_starthour->insertItem(QTime::QTime((m_index + 24) % 24, 0)
+        m_starthour->insertItem(QTime((m_index + 24) % 24, 0)
                                              .toString(hr_format));
         if (thisTime.hour() == m_index)
             m_starthour->setCurrentItem(m_starthour->count() - 1);
@@ -155,7 +155,7 @@
 
     for(int m_index = -5; m_index <= 60; m_index += 5)
     {
-        m_startminute->insertItem(QTime::QTime(0, (m_index + 60) % 60)
+        m_startminute->insertItem(QTime(0, (m_index + 60) % 60)
                                                .toString(":mm"));
         if (m_index == thisTime.minute())
             m_startminute->setCurrentItem(m_startminute->count() - 1);
diff -Naur mythtv-0.21-25121-old/programs/mythtranscode/transcode.cpp mythtv-0.21-25121-new/programs/mythtranscode/transcode.cpp
--- mythtv-0.21-25121-old/programs/mythtranscode/transcode.cpp	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/programs/mythtranscode/transcode.cpp	2010-06-20 07:20:38.000000000 -0700
@@ -662,7 +662,7 @@
         // framecontrol is true if we want to enforce fifo sync.
         if (framecontrol)
             VERBOSE(VB_GENERAL, "Enforcing sync on fifos");
-        fifow = new FIFOWriter::FIFOWriter(2, framecontrol);
+        fifow = new FIFOWriter(2, framecontrol);
 
         if (!fifow->FIFOInit(0, QString("video"), vidfifo, vidSize, 50) ||
             !fifow->FIFOInit(1, QString("audio"), audfifo, audio_size, 25))
diff -Naur mythtv-0.21-25121-old/programs/mythtranscode/transcode.h mythtv-0.21-25121-new/programs/mythtranscode/transcode.h
--- mythtv-0.21-25121-old/programs/mythtranscode/transcode.h	2010-06-20 07:20:38.000000000 -0700
+++ mythtv-0.21-25121-new/programs/mythtranscode/transcode.h	2010-06-20 07:20:38.000000000 -0700
@@ -30,7 +30,7 @@
     NuppelVideoPlayer *nvp;
     RingBuffer *inRingBuffer;
     RingBuffer *outRingBuffer;
-    FIFOWriter::FIFOWriter *fifow;
+    FIFOWriter *fifow;
     QPtrList<struct kfatable_entry> *kfa_table;
     bool showprogress;
 };
