diff -Naur mythtv-0.21-21384-old/libs/libmythtv/vsync.cpp mythtv-0.21-21384-new/libs/libmythtv/vsync.cpp
--- mythtv-0.21-21384-old/libs/libmythtv/vsync.cpp	2009-08-20 20:15:22.000000000 -0700
+++ mythtv-0.21-21384-new/libs/libmythtv/vsync.cpp	2009-08-20 20:15:23.000000000 -0700
@@ -96,7 +96,7 @@
 #endif    
 #ifndef _WIN32
     TESTVIDEOSYNC(nVidiaVideoSync);
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
