diff -Naur perl-5.8.8-old/ext/Errno/Errno_pm.PL perl-5.8.8-new/ext/Errno/Errno_pm.PL
--- perl-5.8.8-old/ext/Errno/Errno_pm.PL	2005-09-21 14:03:26.000000000 -0700
+++ perl-5.8.8-new/ext/Errno/Errno_pm.PL	2008-02-16 18:14:05.000000000 -0800
@@ -318,10 +318,6 @@
 use Config;
 use strict;
 
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$Config{'archname'}-$Config{'osvers'}" or
-	die "Errno architecture ($Config{'archname'}-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
-
 \$VERSION = "$VERSION";
 \$VERSION = eval \$VERSION;
 \@ISA = qw(Exporter);
