diff -Naur mythtv-trunk-16831-old/libs/libmyth/util.cpp mythtv-trunk-16831-new/libs/libmyth/util.cpp
--- mythtv-trunk-16831-old/libs/libmyth/util.cpp	2008-03-28 09:44:18.000000000 -0700
+++ mythtv-trunk-16831-new/libs/libmyth/util.cpp	2008-03-29 15:48:59.000000000 -0700
@@ -296,10 +296,11 @@
                     return GENERIC_EXIT_NOT_OK;
                 }
 
+                qApp->processEvents();
+
                 if (res > 0)
                     return WEXITSTATUS(status);
 
-                qApp->processEvents();
                 usleep(100000);
             }
         }
