diff -Naur mythtv-trunk-18336-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-18336-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-18336-old/libs/libmyth/mediamonitor-unix.cpp	2008-09-19 03:06:23.000000000 -0700
+++ mythtv-trunk-18336-new/libs/libmyth/mediamonitor-unix.cpp	2008-09-19 03:06:25.000000000 -0700
@@ -203,11 +203,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
