diff -Naur mythtv-trunk-16487-old/configure mythtv-trunk-16487-new/configure
--- mythtv-trunk-16487-old/configure	2008-03-09 23:18:39.000000000 -0700
+++ mythtv-trunk-16487-new/configure	2008-03-11 19:09:23.000000000 -0700
@@ -2706,12 +2706,11 @@
 check_header X11/extensions/XvMClib.h
 has_library libXvMCW || disable xvmcw
 check_header X11/extensions/vldXvMC.h
-has_library libviaXvMC || has_library libchromeXvMC || disable xvmc_vld
-has_library libviaXvMCPro || has_library libchromeXvMCPro || disable xvmc_pro
+has_library libviaXvMC || has_library libchromeXvMC || enabled xvmcw || disable xvmc_vld
+has_library libviaXvMCPro || has_library libchromeXvMCPro || enabled xvmcw || disable xvmc_pro
 
 if test x"$xvmc_lib" != x""; then
     VENDOR_XVMC_LIBS="-l"$xvmc_lib
-    disable xvmcw
 elif enabled xvmcw; then
     VENDOR_XVMC_LIBS="-lXvMCW"
 elif enabled xvmc_pro; then
@@ -2719,12 +2718,16 @@
         VENDOR_XVMC_LIBS="-lviaXvMCPro"
     elif has_library libchromeXvMCPro ; then
         VENDOR_XVMC_LIBS="-lchromeXvMCPro"
+    elif has_library libXvMCW ; then
+        VENDOR_XVMC_LIBS="-lXvMCW"
     fi
 elif enabled xvmc_vld; then
     if has_library libviaXvMC ; then
         VENDOR_XVMC_LIBS="-lviaXvMC"
     elif has_library libchromeXvMC ; then
         VENDOR_XVMC_LIBS="-lchromeXvMC"
+    elif has_library libXvMCW ; then
+        VENDOR_XVMC_LIBS="-lXvMCW"
     fi
 elif has_library libXvMCNVIDIA ; then
     VENDOR_XVMC_LIBS="-lXvMCNVIDIA"
@@ -2734,6 +2737,7 @@
     disable xvmc
 fi
 
+test x"$VENDOR_XVMC_LIBS" != x"-lXvMCW" && disable xvmcw
 test x"$VENDOR_XVMC_LIBS" != x"-lXvMCNVIDIA" && disable xvmc_opengl
 
 enabled xvmc_pro && enable xvmc_vld
