diff -Naur mythtv-trunk-19147-old/configure mythtv-trunk-19147-new/configure
--- mythtv-trunk-19147-old/configure	2008-11-22 10:08:24.000000000 -0800
+++ mythtv-trunk-19147-new/configure	2008-11-22 10:08:27.000000000 -0800
@@ -2963,7 +2963,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
