diff -Naur mythtv-trunk-25255-old/configure mythtv-trunk-25255-new/configure
--- mythtv-trunk-25255-old/configure	2010-07-02 12:40:35.000000000 -0700
+++ mythtv-trunk-25255-new/configure	2010-07-02 12:40:35.000000000 -0700
@@ -3804,7 +3804,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
