diff -Naur mythtv-trunk-21004-old/libs/libmythtv/vsync.cpp mythtv-trunk-21004-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-21004-old/libs/libmythtv/vsync.cpp	2009-07-23 07:55:20.000000000 -0700
+++ mythtv-trunk-21004-new/libs/libmythtv/vsync.cpp	2009-07-23 07:55:21.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
