diff -Naur mythtv-trunk-18943-old/configure mythtv-trunk-18943-new/configure
--- mythtv-trunk-18943-old/configure	2008-10-30 06:45:14.000000000 -0700
+++ mythtv-trunk-18943-new/configure	2008-10-30 06:49:23.000000000 -0700
@@ -1625,7 +1625,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
