diff -Naur mythtv-trunk-20959-old/configure mythtv-trunk-20959-new/configure
--- mythtv-trunk-20959-old/configure	2009-07-19 15:47:39.000000000 -0700
+++ mythtv-trunk-20959-new/configure	2009-07-19 15:47:40.000000000 -0700
@@ -3439,7 +3439,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
