diff -Naur mythtv-trunk-22998-old/libs/libmyth/mythcontext.cpp mythtv-trunk-22998-new/libs/libmyth/mythcontext.cpp
--- mythtv-trunk-22998-old/libs/libmyth/mythcontext.cpp	2009-12-22 09:27:42.000000000 -0800
+++ mythtv-trunk-22998-new/libs/libmyth/mythcontext.cpp	2009-12-22 10:16:22.000000000 -0800
@@ -1659,7 +1659,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
