diff -Naur mythtv-trunk-25319-old/bindings/python/python.pro mythtv-trunk-25319-new/bindings/python/python.pro
--- mythtv-trunk-25319-old/bindings/python/python.pro	2010-07-09 15:46:49.000000000 -0700
+++ mythtv-trunk-25319-new/bindings/python/python.pro	2010-07-10 09:52:29.000000000 -0700
@@ -9,13 +9,13 @@
 
 python_install.target = python_install
 python_install.CONFIG = no_path
-python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\"
+python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\" --prefix=\"$${PREFIX}\"
 
 INSTALLS += python_install
 
 python_uninstall.target = python_uninstall
 python_uninstall.depends = setup.py FORCE
-python_uninstall.commands = $$PYTHON setup.py uninstall
+python_uninstall.commands = $$PYTHON setup.py uninstall --prefix=\"$${PREFIX}\"
 
 QMAKE_EXTRA_TARGETS += python_uninstall
 
