diff -Naur mythtv-trunk-20830-old/libs/libmythtv/vsync.cpp mythtv-trunk-20830-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20830-old/libs/libmythtv/vsync.cpp	2009-07-07 18:58:59.000000000 -0700
+++ mythtv-trunk-20830-new/libs/libmythtv/vsync.cpp	2009-07-07 18:59:00.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
