diff -Naur mythtv-trunk-20261-old/configure mythtv-trunk-20261-new/configure
--- mythtv-trunk-20261-old/configure	2009-03-24 19:44:53.000000000 -0700
+++ mythtv-trunk-20261-new/configure	2009-03-24 19:44:54.000000000 -0700
@@ -1633,7 +1633,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
