diff -Naur mythtv-trunk-22548-old/configure mythtv-trunk-22548-new/configure
--- mythtv-trunk-22548-old/configure	2009-10-20 15:59:40.000000000 -0700
+++ mythtv-trunk-22548-new/configure	2009-10-20 15:59:41.000000000 -0700
@@ -3480,7 +3480,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
