diff -Naur mythtv-trunk-27338-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-27338-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-27338-old/libs/libmythtv/avformatdecoder.cpp	2010-11-24 16:43:06.000000000 -0800
+++ mythtv-trunk-27338-new/libs/libmythtv/avformatdecoder.cpp	2010-11-24 16:49:48.000000000 -0800
@@ -1283,6 +1283,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
