diff -Naur mythtv-trunk-17449-old/libs/libmythtv/tv_play.cpp mythtv-trunk-17449-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-17449-old/libs/libmythtv/tv_play.cpp	2008-06-06 12:33:19.000000000 -0700
+++ mythtv-trunk-17449-new/libs/libmythtv/tv_play.cpp	2008-06-06 12:33:20.000000000 -0700
@@ -1784,7 +1784,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
