diff -Naur mythtv-trunk-26841-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-26841-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-26841-old/libs/libmythtv/avformatdecoder.cpp	2010-10-17 06:30:44.000000000 -0700
+++ mythtv-trunk-26841-new/libs/libmythtv/avformatdecoder.cpp	2010-10-17 06:30:45.000000000 -0700
@@ -1256,6 +1256,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
