diff -Naur mythtv-trunk-20696-old/libs/libmythui/mythxdisplay.h mythtv-trunk-20696-new/libs/libmythui/mythxdisplay.h
--- mythtv-trunk-20696-old/libs/libmythui/mythxdisplay.h	2009-06-13 16:55:01.000000000 -0700
+++ mythtv-trunk-20696-new/libs/libmythui/mythxdisplay.h	2009-06-13 18:42:05.000000000 -0700
@@ -13,7 +13,7 @@
 
 #include "mythexp.h"
 
-class MythXDisplay
+class MPUBLIC MythXDisplay
 {
   public:
     MythXDisplay();
@@ -71,10 +71,10 @@
     MythXDisplay *disp;
 };
 
-MythXDisplay *GetMythXDisplay(Display*);
-MythXDisplay *OpenMythXDisplay(void);
-QSize         MythXGetDisplayDimensions(void);
-int           MythXGetRefreshRate(void);
+MythXDisplay         *GetMythXDisplay(Display*);
+MPUBLIC MythXDisplay *OpenMythXDisplay(void);
+QSize                 MythXGetDisplayDimensions(void);
+MPUBLIC int           MythXGetRefreshRate(void);
 #define XLOCK(dpy, arg) { dpy->Lock(); arg; dpy->Unlock(); }
 #endif // USING_X11
 
