diff -Naur mythtv-trunk-26739-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-26739-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-26739-old/libs/libmythtv/avformatdecoder.cpp	2010-10-10 12:59:03.000000000 -0700
+++ mythtv-trunk-26739-new/libs/libmythtv/avformatdecoder.cpp	2010-10-10 12:59:04.000000000 -0700
@@ -1248,6 +1248,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
