diff -Naur mythtv-trunk-19987-old/configure mythtv-trunk-19987-new/configure
--- mythtv-trunk-19987-old/configure	2009-02-12 16:52:50.000000000 -0800
+++ mythtv-trunk-19987-new/configure	2009-02-12 16:52:59.000000000 -0800
@@ -1633,7 +1633,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
