diff -Naur mythtv-trunk-17280-old/libs/libmythtv/videoout_xv.cpp mythtv-trunk-17280-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-trunk-17280-old/libs/libmythtv/videoout_xv.cpp	2008-05-06 17:18:04.000000000 -0700
+++ mythtv-trunk-17280-new/libs/libmythtv/videoout_xv.cpp	2008-05-06 17:21:57.000000000 -0700
@@ -1546,7 +1546,12 @@
         for (; it != renderers.end(); ++it)
             tmp += *it + ",";
 
-        renderer = renderers[0];
+        if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-blit") && (renderers.contains("xv-blit")))
+            renderer = "xv-blit";
+        else if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-opengl") && (renderers.contains("opengl")))
+            renderer = "opengl";
+        else
+            renderer = renderers[0];
         VERBOSE(VB_IMPORTANT, LOC + QString(
                     "Desired video renderer '%1' not available.\n\t\t\t"
                     "codec '%2' makes '%3' available, using '%4' instead.")
