diff -Naur mythtv-trunk-19997-old/configure mythtv-trunk-19997-new/configure
--- mythtv-trunk-19997-old/configure	2009-02-14 15:57:39.000000000 -0800
+++ mythtv-trunk-19997-new/configure	2009-02-14 15:57:55.000000000 -0800
@@ -2975,7 +2975,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
