diff -Naur mythtv-trunk-26007-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-26007-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-26007-old/libs/libmythtv/avformatdecoder.cpp	2010-08-30 13:59:56.000000000 -0700
+++ mythtv-trunk-26007-new/libs/libmythtv/avformatdecoder.cpp	2010-09-01 07:20:17.000000000 -0700
@@ -1242,6 +1242,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
