diff -Naur mythtv-trunk-21118-old/libs/libmythtv/vsync.cpp mythtv-trunk-21118-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-21118-old/libs/libmythtv/vsync.cpp	2009-08-03 18:04:24.000000000 -0700
+++ mythtv-trunk-21118-new/libs/libmythtv/vsync.cpp	2009-08-03 18:04:25.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
