diff -Naur mythtv-trunk-19922-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-19922-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-19922-old/libs/libmyth/mediamonitor-unix.cpp	2009-02-02 19:41:41.000000000 -0800
+++ mythtv-trunk-19922-new/libs/libmyth/mediamonitor-unix.cpp	2009-02-02 19:41:49.000000000 -0800
@@ -218,11 +218,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
