diff -Naur mythtv-trunk-25069-old/configure mythtv-trunk-25069-new/configure
--- mythtv-trunk-25069-old/configure	2010-06-09 19:06:45.000000000 -0700
+++ mythtv-trunk-25069-new/configure	2010-06-09 19:06:46.000000000 -0700
@@ -4582,8 +4582,8 @@
 #       -L from early in the link line, to ignore any old installed libmyth*
 # Note: Qt uses QMAKE_INCDIR_QT to build an absolute path for include
 #       directories, it cannot be modified.
-QMAKE_LIBDIR_QT -= $libdir
-LATE_LIBS            += -L$libdir
+QMAKE_LIBDIR_QT -= ${sysroot}${libdir}
+LATE_LIBS            += -L${sysroot}${libdir}
 EOF
 
 diff $TMPH $MYTH_CONFIG_H >/dev/null 2>&1
