diff -Naur mythtv-trunk-26433-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-26433-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-26433-old/libs/libmythtv/avformatdecoder.cpp	2010-09-21 11:05:16.000000000 -0700
+++ mythtv-trunk-26433-new/libs/libmythtv/avformatdecoder.cpp	2010-09-21 11:09:31.000000000 -0700
@@ -1273,6 +1273,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
