diff -Naur mythtv-trunk-20616-old/configure mythtv-trunk-20616-new/configure
--- mythtv-trunk-20616-old/configure	2009-05-24 09:33:05.000000000 -0700
+++ mythtv-trunk-20616-new/configure	2009-05-24 09:33:05.000000000 -0700
@@ -1638,7 +1638,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
