diff -Naur mythtv-trunk-19186-old/libs/libmythtv/tv_play.cpp mythtv-trunk-19186-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-19186-old/libs/libmythtv/tv_play.cpp	2008-11-30 11:16:36.000000000 -0800
+++ mythtv-trunk-19186-new/libs/libmythtv/tv_play.cpp	2008-11-30 11:16:41.000000000 -0800
@@ -1817,7 +1817,7 @@
     if (pthread_create(&decode, NULL, SpawnDecode, nvp))
         return false;
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
