diff -Naur mythtv-trunk-20451-old/configure mythtv-trunk-20451-new/configure
--- mythtv-trunk-20451-old/configure	2009-04-25 19:42:36.000000000 -0700
+++ mythtv-trunk-20451-new/configure	2009-04-25 19:42:37.000000000 -0700
@@ -1637,7 +1637,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
