diff -Naur mythtv-trunk-18866-old/configure mythtv-trunk-18866-new/configure
--- mythtv-trunk-18866-old/configure	2008-10-24 16:58:02.000000000 -0700
+++ mythtv-trunk-18866-new/configure	2008-10-24 17:00:59.000000000 -0700
@@ -2735,7 +2735,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
