diff -Naur mythtv-trunk-19726-old/libs/libmythdvdnav/libmythdvdnav.pro mythtv-trunk-19726-new/libs/libmythdvdnav/libmythdvdnav.pro
--- mythtv-trunk-19726-old/libs/libmythdvdnav/libmythdvdnav.pro	2009-01-17 20:53:58.000000000 -0800
+++ mythtv-trunk-19726-new/libs/libmythdvdnav/libmythdvdnav.pro	2009-01-17 20:58:20.000000000 -0800
@@ -3,7 +3,7 @@
 
 TEMPLATE = lib
 TARGET = mythdvdnav-$$LIBVERSION
-CONFIG += thread staticlib warn_off
+CONFIG += thread dll warn_off
 CONFIG -= qt
 target.path = $${LIBDIR}
 
diff -Naur mythtv-trunk-19726-old/libs/libmythtv/libmythtv.pro mythtv-trunk-19726-new/libs/libmythtv/libmythtv.pro
--- mythtv-trunk-19726-old/libs/libmythtv/libmythtv.pro	2009-01-17 20:53:50.000000000 -0800
+++ mythtv-trunk-19726-new/libs/libmythtv/libmythtv.pro	2009-01-17 20:58:20.000000000 -0800
@@ -62,7 +62,7 @@
 TARGETDEPS += ../libavcodec/libmythavcodec-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../libavformat/libmythavformat-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../libmythmpeg2/libmythmpeg2-$${MYTH_LIB_EXT}
-TARGETDEPS += ../libmythdvdnav/libmythdvdnav-$${MYTH_LIB_EXT}
+TARGETDEPS += ../libmythdvdnav/libmythdvdnav-$${MYTH_SHLIB_EXT}
 using_mheg: TARGETDEPS += ../libmythfreemheg/libmythfreemheg-$${MYTH_SHLIB_EXT}
 using_live: TARGETDEPS += ../libmythlivemedia/libmythlivemedia-$${MYTH_SHLIB_EXT}
 
diff -Naur mythtv-trunk-19726-old/programs/programs-libs.pro mythtv-trunk-19726-new/programs/programs-libs.pro
--- mythtv-trunk-19726-old/programs/programs-libs.pro	2009-01-17 20:54:01.000000000 -0800
+++ mythtv-trunk-19726-new/programs/programs-libs.pro	2009-01-17 20:58:20.000000000 -0800
@@ -8,10 +8,12 @@
 LIBS += -L../../libs/libmythdb
 LIBS += -L../../libs/libmythui
 LIBS += -L../../libs/libmythupnp
+LIBS += -L../../libs/libmythdvdnav
 
 LIBS += -lmythtv-$$LIBVERSION -lmythavformat-$$LIBVERSION
 LIBS += -lmythavutil-$$LIBVERSION -lmythavcodec-$$LIBVERSION 
 LIBS += -lmythupnp-$$LIBVERSION 
+LIBS += -lmythdvdnav-$$LIBVERSION
 LIBS += -lmyth-$$LIBVERSION -lmythui-$$LIBVERSION
 LIBS += -lmythdb-$$LIBVERSION
 
@@ -30,12 +32,14 @@
 TARGETDEPS += ../../libs/libavcodec/libmythavcodec-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../../libs/libavformat/libmythavformat-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../../libs/libmythupnp/libmythupnp-$${MYTH_SHLIB_EXT}
+TARGETDEPS += ../../libs/libmythdvdnav/libmythdvdnav-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../../libs/libmythdb/libmythdb-$${MYTH_SHLIB_EXT}
 using_live: TARGETDEPS += ../../libs/libmythlivemedia/libmythlivemedia-$${MYTH_SHLIB_EXT}
 
 DEPENDPATH += ../.. ../../libs ../../libs/libmyth ../../libs/libmythtv
 DEPENDPATH += ../../libs/libavutil ../../libs/libavformat ../../libs/libsavcodec
 DEPENDPATH += ../../libs/libmythupnp ../../libs/libmythui
+DEPENDPATH += ../../libs/libmythdvdnav
 DEPENDPATH += ../../libs/libmythlivemedia ../../libmythdb
 
 CONFIG += opengl
