diff -Naur mythtv-trunk-26882-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-26882-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-26882-old/libs/libmythtv/avformatdecoder.cpp	2010-10-18 10:11:03.000000000 -0700
+++ mythtv-trunk-26882-new/libs/libmythtv/avformatdecoder.cpp	2010-10-18 10:11:04.000000000 -0700
@@ -1256,6 +1256,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
