diff -Naur mythtv-trunk-24011-old/libs/libmythtv/util-xvmc.h mythtv-trunk-24011-new/libs/libmythtv/util-xvmc.h
--- mythtv-trunk-24011-old/libs/libmythtv/util-xvmc.h	2010-04-06 06:58:21.000000000 -0700
+++ mythtv-trunk-24011-new/libs/libmythtv/util-xvmc.h	2010-04-06 07:03:33.000000000 -0700
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(14),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
