diff -Naur mythtv-trunk-25488-old/libs/libmythtv/util-xvmc.h mythtv-trunk-25488-new/libs/libmythtv/util-xvmc.h
--- mythtv-trunk-25488-old/libs/libmythtv/util-xvmc.h	2010-07-29 17:01:25.000000000 -0700
+++ mythtv-trunk-25488-new/libs/libmythtv/util-xvmc.h	2010-07-29 17:01:26.000000000 -0700
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(14),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
