diff -Naur mythtv-trunk-21337-old/libs/libmythtv/vsync.cpp mythtv-trunk-21337-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-21337-old/libs/libmythtv/vsync.cpp	2009-08-17 13:38:58.000000000 -0700
+++ mythtv-trunk-21337-new/libs/libmythtv/vsync.cpp	2009-08-17 13:38:59.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
