diff -Naur mythtv-trunk-23009-old/programs/mythfrontend/programinfocache.h mythtv-trunk-23009-new/programs/mythfrontend/programinfocache.h
--- mythtv-trunk-23009-old/programs/mythfrontend/programinfocache.h	2009-12-27 09:06:51.000000000 -0800
+++ mythtv-trunk-23009-new/programs/mythfrontend/programinfocache.h	2009-12-27 10:11:22.000000000 -0800
@@ -1,5 +1,10 @@
 // -*- Mode: c++ -*-
 
+#ifndef PROGRAMINFOCACHE_H
+#define PROGRAMINFOCACHE_H
+
+#include <stdint.h>
+
 #include <vector>
 #include <map>
 using namespace std;
@@ -67,3 +72,5 @@
     uint                    m_loads_in_progress;
     mutable QWaitCondition  m_load_wait;
 };
+
+#endif
