diff -Naur mythtv-trunk-16873-old/libs/libmythtv/tv_play.cpp mythtv-trunk-16873-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-16873-old/libs/libmythtv/tv_play.cpp	2008-04-01 08:56:12.000000000 -0700
+++ mythtv-trunk-16873-new/libs/libmythtv/tv_play.cpp	2008-04-01 08:56:15.000000000 -0700
@@ -1818,7 +1818,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
