diff -Naur mythtv-trunk-19211-old/configure mythtv-trunk-19211-new/configure
--- mythtv-trunk-19211-old/configure	2008-12-02 04:54:14.000000000 -0800
+++ mythtv-trunk-19211-new/configure	2008-12-02 04:54:15.000000000 -0800
@@ -1632,7 +1632,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
