diff -Naur mythtv-trunk-24847-old/configure mythtv-trunk-24847-new/configure
--- mythtv-trunk-24847-old/configure	2010-05-24 20:26:05.000000000 -0700
+++ mythtv-trunk-24847-new/configure	2010-05-24 20:26:06.000000000 -0700
@@ -4582,8 +4582,8 @@
 #       -L from early in the link line, to ignore any old installed libmyth*
 # Note: Qt uses QMAKE_INCDIR_QT to build an absolute path for include
 #       directories, it cannot be modified.
-QMAKE_LIBDIR_QT -= $libdir
-LATE_LIBS            += -L$libdir
+QMAKE_LIBDIR_QT -= ${sysroot}${libdir}
+LATE_LIBS            += -L${sysroot}${libdir}
 EOF
 
 diff $TMPH $MYTH_CONFIG_H >/dev/null 2>&1
