diff -Naur mythtv-trunk-17177-old/libs/libmythtv/tv_play.cpp mythtv-trunk-17177-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-17177-old/libs/libmythtv/tv_play.cpp	2008-04-29 20:25:09.000000000 -0700
+++ mythtv-trunk-17177-new/libs/libmythtv/tv_play.cpp	2008-04-29 20:25:11.000000000 -0700
@@ -1832,7 +1832,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
