diff -Naur mythtv-trunk-20321-old/configure mythtv-trunk-20321-new/configure
--- mythtv-trunk-20321-old/configure	2009-04-08 05:03:28.000000000 -0700
+++ mythtv-trunk-20321-new/configure	2009-04-08 05:03:29.000000000 -0700
@@ -1634,7 +1634,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
