diff -Naur mythtv-trunk-18806-old/libs/libmythtv/tv_play.cpp mythtv-trunk-18806-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-18806-old/libs/libmythtv/tv_play.cpp	2008-10-20 20:40:11.000000000 -0700
+++ mythtv-trunk-18806-new/libs/libmythtv/tv_play.cpp	2008-10-20 20:40:14.000000000 -0700
@@ -1819,7 +1819,7 @@
     if (pthread_create(&decode, NULL, SpawnDecode, nvp))
         return false;
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
