diff -Naur mythtv-trunk-21509-old/libs/libmythtv/vsync.cpp mythtv-trunk-21509-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-21509-old/libs/libmythtv/vsync.cpp	2009-08-24 23:33:40.000000000 -0700
+++ mythtv-trunk-21509-new/libs/libmythtv/vsync.cpp	2009-08-24 23:33:41.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
