diff -Naur mythtv-trunk-20728-old/configure mythtv-trunk-20728-new/configure
--- mythtv-trunk-20728-old/configure	2009-06-20 09:21:10.000000000 -0700
+++ mythtv-trunk-20728-new/configure	2009-06-20 09:21:11.000000000 -0700
@@ -1635,7 +1635,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
