diff -Naur mythtv-trunk-19111-old/configure mythtv-trunk-19111-new/configure
--- mythtv-trunk-19111-old/configure	2008-11-16 17:44:39.000000000 -0800
+++ mythtv-trunk-19111-new/configure	2008-11-16 17:44:41.000000000 -0800
@@ -1625,7 +1625,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
