diff -Naur mythtv-trunk-17683-old/libs/libmythtv/tv_play.cpp mythtv-trunk-17683-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-17683-old/libs/libmythtv/tv_play.cpp	2008-07-01 09:40:07.000000000 -0700
+++ mythtv-trunk-17683-new/libs/libmythtv/tv_play.cpp	2008-07-01 09:40:08.000000000 -0700
@@ -1788,7 +1788,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
