diff -Naur mythtv-trunk-20635-old/libs/libmythtv/vsync.cpp mythtv-trunk-20635-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20635-old/libs/libmythtv/vsync.cpp	2009-05-25 21:14:15.000000000 -0700
+++ mythtv-trunk-20635-new/libs/libmythtv/vsync.cpp	2009-05-25 21:14:16.000000000 -0700
@@ -96,7 +96,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
