diff -Naur mythtv-trunk-20948-old/configure mythtv-trunk-20948-new/configure
--- mythtv-trunk-20948-old/configure	2009-07-18 06:12:22.000000000 -0700
+++ mythtv-trunk-20948-new/configure	2009-07-18 06:12:23.000000000 -0700
@@ -3431,7 +3431,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
