diff -Naur mythtv-trunk-18943-old/libs/libmythtv/tv_play.cpp mythtv-trunk-18943-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-18943-old/libs/libmythtv/tv_play.cpp	2008-10-30 06:49:59.000000000 -0700
+++ mythtv-trunk-18943-new/libs/libmythtv/tv_play.cpp	2008-10-30 06:50:13.000000000 -0700
@@ -1817,7 +1817,7 @@
     if (pthread_create(&decode, NULL, SpawnDecode, nvp))
         return false;
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
