diff -Naur mythtv-trunk-26176-old/configure mythtv-trunk-26176-new/configure
--- mythtv-trunk-26176-old/configure	2010-09-08 16:40:53.000000000 -0700
+++ mythtv-trunk-26176-new/configure	2010-09-08 16:40:53.000000000 -0700
@@ -4742,8 +4742,8 @@
 #       -L from early in the link line, to ignore any old installed libmyth*
 # Note: Qt uses QMAKE_INCDIR_QT to build an absolute path for include
 #       directories, it cannot be modified.
-QMAKE_LIBDIR_QT-=$libdir
-LATE_LIBS+=-L$libdir
+QMAKE_LIBDIR_QT-=${sysroot}$libdir
+LATE_LIBS+=-L${sysroot}$libdir
 EOF
 
 #echo "endif # FFMPEG_CONFIG_MAK" >> config.mak
