diff -Naur unichrome-20050421-old/unichrome/via_mode.h unichrome-20050421-new/unichrome/via_mode.h
--- unichrome-20050421-old/unichrome/via_mode.h	2005-04-11 21:28:18.000000000 -0700
+++ unichrome-20050421-new/unichrome/via_mode.h	2005-05-03 07:38:22.438710051 -0700
@@ -89,6 +89,7 @@
     {  65028, 0x866D,        0 },
     {  75000, 0x156E, 0x810017 },
     {  78800, 0x442C, 0xEB0029 },
+    {  81000, 0x1471, 0xC40021 },
     {  81135, 0x0622,        0 },
     {  81613, 0x4539,        0 },
     {  94500, 0x4542, 0xB7001A },
