diff -Naur unichrome-20050421-old/unichrome/via_driver.h unichrome-20050421-new/unichrome/via_driver.h
--- unichrome-20050421-old/unichrome/via_driver.h	2005-04-11 21:28:18.000000000 -0700
+++ unichrome-20050421-new/unichrome/via_driver.h	2005-05-05 07:13:17.748505181 -0700
@@ -26,7 +26,7 @@
 #ifndef _VIA_DRIVER_H_
 #define _VIA_DRIVER_H_ 1
 
-#define HAVE_DEBUG
+/* #define HAVE_DEBUG */
 
 #ifdef HAVE_DEBUG
 #define DEBUG(x) x
diff -Naur unichrome-20050421-old/unichrome/via_swov.c unichrome-20050421-new/unichrome/via_swov.c
--- unichrome-20050421-old/unichrome/via_swov.c	2005-04-20 11:28:54.000000000 -0700
+++ unichrome-20050421-new/unichrome/via_swov.c	2005-05-05 07:14:06.795944053 -0700
@@ -50,7 +50,7 @@
  *   - merge with CLEXF40040
  */
 
-#define   XV_DEBUG      1     /* write log msg to /var/log/XFree86.0.log */
+/* #define   XV_DEBUG      1 */    /* write log msg to /var/log/XFree86.0.log */
 
 #ifdef XV_DEBUG
 # define DBG_DD(x) (x)
