diff -Naur NVIDIA-Linux-x86_64-71.86.04-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-71.86.04-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-71.86.04-pkg2-old/usr/src/nv/conftest.sh	2008-01-21 11:13:08.000000000 -0800
+++ NVIDIA-Linux-x86_64-71.86.04-pkg2-new/usr/src/nv/conftest.sh	2008-02-01 06:57:20.000000000 -0800
@@ -852,6 +852,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
