diff -Naur libdvdread-0.9.5-old/dvdread/ifo_types.h libdvdread-0.9.5-new/dvdread/ifo_types.h
--- libdvdread-0.9.5-old/dvdread/ifo_types.h	2005-09-18 11:02:08.000000000 -0700
+++ libdvdread-0.9.5-new/dvdread/ifo_types.h	2006-02-15 11:08:14.000000000 -0800
@@ -23,15 +23,8 @@
 
 #include <dvdread/dvd_reader.h>
 
-#if defined(__BEOS__)
-#if !defined(_INTTYPES_H_) && !defined(_INTTYPES_H) && !defined(_STDINT_H_) && !defined(_STDINT_H)
-#error "Must include <inttypes.h> or <stdint.h> before any libdvdread header."
-#endif
-#else
-#if !defined(UINT8_MAX) || !defined(UINT16_MAX) || !defined(INT32_MAX)
-#error "Must include <inttypes.h> or <stdint.h> before any libdvdread header."
-#endif
-#endif
+@INCLUDE_INTTYPES_H@
+@INCLUDE_STDINT_H@
 
 #undef ATTRIBUTE_PACKED
 #undef PRAGMA_PACK_BEGIN 
