diff -Naur libdvdread-1169-old/misc/dvdread-config.in libdvdread-1169-new/misc/dvdread-config.in
--- libdvdread-1169-old/misc/dvdread-config.in	2009-04-22 21:05:36.000000000 -0700
+++ libdvdread-1169-new/misc/dvdread-config.in	2009-04-22 21:06:01.000000000 -0700
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
