diff -Naur libdvdread-0.9.7-old/dvdread/ifo_types.h libdvdread-0.9.7-new/dvdread/ifo_types.h
--- libdvdread-0.9.7-old/dvdread/ifo_types.h	2006-03-02 09:25:43.000000000 -0800
+++ libdvdread-0.9.7-new/dvdread/ifo_types.h	2006-10-06 15:42:46.000000000 -0700
@@ -23,15 +23,8 @@
 
 #include <dvdread/dvd_reader.h>
 
-#if defined(__BEOS__)
-#if !defined(_INTTYPES_H_) && !defined(_INTTYPES_H) && !defined(_STDINT_H_) && !defined(_STDINT_H)
-#error "Must include <inttypes.h> or <stdint.h> before any libdvdread header."
-#endif
-#else
-#if !defined(UINT8_MAX) || !defined(UINT16_MAX) || !defined(INT32_MAX)
-#error "Must include <inttypes.h> or <stdint.h> before any libdvdread header."
-#endif
-#endif
+@INCLUDE_INTTYPES_H@
+@INCLUDE_STDINT_H@
 
 #undef ATTRIBUTE_PACKED
 #undef PRAGMA_PACK_BEGIN 
