diff -Naur linux-2.6.17.14-old/arch/x86_64/boot/compressed/Makefile linux-2.6.17.14-new/arch/x86_64/boot/compressed/Makefile
--- linux-2.6.17.14-old/arch/x86_64/boot/compressed/Makefile	2006-10-13 11:55:04.000000000 -0700
+++ linux-2.6.17.14-new/arch/x86_64/boot/compressed/Makefile	2007-06-24 06:56:27.000000000 -0700
@@ -7,7 +7,8 @@
 #
 
 targets		:= vmlinux vmlinux.bin vmlinux.bin.gz head.o misc.o piggy.o
-EXTRA_AFLAGS	:= -traditional -m32
+EXTRA_AFLAGS	:= -traditional
+AFLAGS		:= $(subst -m64,-m32,$(AFLAGS))
 
 # cannot use EXTRA_CFLAGS because base CFLAGS contains -mkernel which conflicts with
 # -m32
