diff -Naur linux-2.6.17.14-old/drivers/ide/ide-dma.c linux-2.6.17.14-new/drivers/ide/ide-dma.c
--- linux-2.6.17.14-old/drivers/ide/ide-dma.c	2006-10-13 11:55:04.000000000 -0700
+++ linux-2.6.17.14-new/drivers/ide/ide-dma.c	2007-08-12 03:26:32.000000000 -0700
@@ -113,6 +113,8 @@
 	{ "CRD-8480B",			"ALL"		},
 	{ "CRD-8482B",			"ALL"		},
  	{ "CRD-84"		,	"ALL"		},
+	{ "SanDisk SDCFB-1024"	,	"ALL"		},
+	{ "SanDisk SDCFH-1024"	,	"ALL"		},
 	{ "SanDisk SDP3B"	,	"ALL"		},
 	{ "SanDisk SDP3B-64"	,	"ALL"		},
 	{ "SANYO CD-ROM CRD"	,	"ALL"		},
diff -Naur linux-2.6.17.14-old/drivers/scsi/libata-core.c linux-2.6.17.14-new/drivers/scsi/libata-core.c
--- linux-2.6.17.14-old/drivers/scsi/libata-core.c	2006-10-13 11:55:04.000000000 -0700
+++ linux-2.6.17.14-new/drivers/scsi/libata-core.c	2007-08-12 03:26:55.000000000 -0700
@@ -2587,6 +2587,8 @@
 	"CRD-8480B", NULL,
 	"CRD-8482B", NULL,
  	"CRD-84", NULL,
+	"SanDisk SDCFB-1024", NULL,
+	"SanDisk SDCFH-1024", NULL,
 	"SanDisk SDP3B", NULL,
 	"SanDisk SDP3B-64", NULL,
 	"SANYO CD-ROM CRD", NULL,
