diff -Naur linux-2.6.17.14-old/drivers/ide/ide-dma.c linux-2.6.17.14-new/drivers/ide/ide-dma.c
--- linux-2.6.17.14-old/drivers/ide/ide-dma.c	2006-10-13 11:55:04.000000000 -0700
+++ linux-2.6.17.14-new/drivers/ide/ide-dma.c	2007-08-12 09:09:27.000000000 -0700
@@ -666,6 +666,12 @@
 				    drive->name, id->model);
 		return blacklist;
 	}
+	if (id->config == 0x848a) {
+		printk(KERN_WARNING "%s: Disabling (U)DMA for %s (CFA)\n",
+				    drive->name, id->model);
+		return 1;
+	}
+
 	return 0;
 }
 
diff -Naur linux-2.6.17.14-old/drivers/scsi/libata-core.c linux-2.6.17.14-new/drivers/scsi/libata-core.c
--- linux-2.6.17.14-old/drivers/scsi/libata-core.c	2006-10-13 11:55:04.000000000 -0700
+++ linux-2.6.17.14-new/drivers/scsi/libata-core.c	2007-08-12 09:11:04.000000000 -0700
@@ -2680,6 +2680,8 @@
 		xfer_mask &= ata_id_xfermask(d->id);
 		if (ata_dma_blacklisted(d))
 			xfer_mask &= ~(ATA_MASK_MWDMA | ATA_MASK_UDMA);
+		if (ata_id_is_cfa(d->id))
+			xfer_mask &= ~(ATA_MASK_MWDMA | ATA_MASK_UDMA);
 		/* Apply cable rule here. Don't apply it early because when
 		   we handle hot plug the cable type can itself change */
 		if (ap->cbl == ATA_CBL_PATA40)
@@ -2689,6 +2691,9 @@
 	if (ata_dma_blacklisted(dev))
 		printk(KERN_WARNING "ata%u: dev %u is on DMA blacklist, "
 		       "disabling DMA\n", ap->id, dev->devno);
+	if (ata_id_is_cfa(dev->id))
+		printk(KERN_WARNING "ata%u: dev %u is a CFA, "
+		       "disabling DMA\n", ap->id, dev->devno);
 
 	if (hs->flags & ATA_HOST_SIMPLEX) {
 		if (hs->simplex_claimed)
