diff -Naur linux-3.0.4-old/fs/unionfs/unlink.c linux-3.0.4-new/fs/unionfs/unlink.c
--- linux-3.0.4-old/fs/unionfs/unlink.c	2011-09-02 20:41:02.000000000 -0700
+++ linux-3.0.4-new/fs/unionfs/unlink.c	2011-09-02 20:41:24.000000000 -0700
@@ -144,8 +144,6 @@
 	if (!err) {
 		unionfs_postcopyup_release(dentry);
 		unionfs_postcopyup_setmnt(parent);
-		if (inode->i_nlink == 0) /* drop lower inodes */
-			iput_lowers_all(inode, false);
 		d_drop(dentry);
 		/*
 		 * if unlink/whiteout succeeded, parent dir mtime has
