diff -Naur libxcb-1.0-old/configure libxcb-1.0-new/configure
--- libxcb-1.0-old/configure	2006-11-23 07:54:04.000000000 -0800
+++ libxcb-1.0-new/configure	2008-02-07 21:43:33.000000000 -0800
@@ -21211,7 +21211,7 @@
 # Find the xcb-proto protocol descriptions
 { echo "$as_me:$LINENO: checking XCBPROTO_XCBINCLUDEDIR" >&5
 echo $ECHO_N "checking XCBPROTO_XCBINCLUDEDIR... $ECHO_C" >&6; }
-XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
+XCBPROTO_XCBINCLUDEDIR=$DESTDIR`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
 { echo "$as_me:$LINENO: result: $XCBPROTO_XCBINCLUDEDIR" >&5
 echo "${ECHO_T}$XCBPROTO_XCBINCLUDEDIR" >&6; }
 
