diff -Naur vdpau-video-0.7.1-old/src/Makefile.am vdpau-video-0.7.1-new/src/Makefile.am
--- vdpau-video-0.7.1-old/src/Makefile.am	2010-10-05 23:00:39.000000000 -0700
+++ vdpau-video-0.7.1-new/src/Makefile.am	2010-10-06 15:01:51.000000000 -0700
@@ -74,6 +74,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 
diff -Naur vdpau-video-0.7.1-old/src/Makefile.in vdpau-video-0.7.1-new/src/Makefile.in
--- vdpau-video-0.7.1-old/src/Makefile.in	2010-10-05 23:00:52.000000000 -0700
+++ vdpau-video-0.7.1-new/src/Makefile.in	2010-10-06 15:01:51.000000000 -0700
@@ -597,6 +597,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
