diff -Naur cwiid-0.6.00-old/common/include/lib.mak.in cwiid-0.6.00-new/common/include/lib.mak.in
--- cwiid-0.6.00-old/common/include/lib.mak.in	2007-06-13 21:16:11.000000000 -0700
+++ cwiid-0.6.00-new/common/include/lib.mak.in	2008-01-29 19:54:02.000000000 -0800
@@ -25,8 +25,8 @@
 	ar rcs $(STATIC_LIB) $(OBJECTS)
 
 $(SHARED_LIB): $(OBJECTS)
-	$(CC) -shared -Wl,-soname,$(SO_NAME) $(LDFLAGS) $(LDLIBS) \
-	      -o $(SHARED_LIB) $(OBJECTS)
+	$(CC) -shared -Wl,-soname,$(SO_NAME) \
+	      -o $(SHARED_LIB) $(OBJECTS) $(LDFLAGS) $(LDLIBS)
 
 install: install_header install_static install_shared
 
diff -Naur cwiid-0.6.00-old/common/include/plugin.mak cwiid-0.6.00-new/common/include/plugin.mak
--- cwiid-0.6.00-old/common/include/plugin.mak	2007-06-13 21:16:11.000000000 -0700
+++ cwiid-0.6.00-new/common/include/plugin.mak	2008-01-29 19:53:44.000000000 -0800
@@ -16,7 +16,7 @@
 all: $(LIB_NAME)
 
 $(LIB_NAME): $(OBJECTS)
-	$(CC) -shared $(LDFLAGS) $(LDLIBS) -o $(LIB_NAME) $(OBJECTS)
+	$(CC) -shared -o $(LIB_NAME) $(OBJECTS) $(LDFLAGS) $(LDLIBS)
 
 install: $(LIB_NAME)
 	install -D $(LIB_NAME) $(DEST_INST_DIR)/$(LIB_NAME)
diff -Naur cwiid-0.6.00-old/configure.ac cwiid-0.6.00-new/configure.ac
--- cwiid-0.6.00-old/configure.ac	2007-08-26 17:27:58.000000000 -0700
+++ cwiid-0.6.00-new/configure.ac	2008-01-29 19:32:38.000000000 -0800
@@ -42,6 +42,28 @@
 fi
 
 AC_ARG_WITH(
+        [gtk],
+	[AS_HELP_STRING([--without-gtk],[compile without gtk support])],
+	[case $withval in 
+		yes)
+			REQUIRE_GTK=1
+		;;
+		no)
+			REQUIRE_GTK=
+		;;
+		*)
+			REQUIRE_GTK=1
+		;;
+	esac],
+	[REQUIRE_GTK=1])
+if test $REQUIRE_GTK; then
+	AC_DEFINE([HAVE_GTK],1,[Define to 1 if gtk support is enabled])
+	pkg_modules="gtk+-2.0 >= 2.0.0 gthread-2.0"
+	PKG_CHECK_MODULES([GTK], [$pkg_modules])
+	AC_SUBST(GTK_CFLAGS)
+	AC_SUBST(GTK_LIBS)
+fi
+AC_ARG_WITH(
 	[python],
 	[AS_HELP_STRING([--without-python],[compile without python support])],
 	[case $withval in 
@@ -124,11 +146,6 @@
 fi
 AC_SUBST(LDCONFIG)
 
-pkg_modules="gtk+-2.0 >= 2.0.0 gthread-2.0"
-PKG_CHECK_MODULES([GTK], [$pkg_modules])
-AC_SUBST(GTK_CFLAGS)
-AC_SUBST(GTK_LIBS)
-
 AC_OUTPUT(
 	[Makefile]
 	[defs.mak]
diff -Naur cwiid-0.6.00-old/Makefile.in cwiid-0.6.00-new/Makefile.in
--- cwiid-0.6.00-old/Makefile.in	2007-07-28 11:47:03.000000000 -0700
+++ cwiid-0.6.00-new/Makefile.in	2008-01-29 19:37:09.000000000 -0800
@@ -3,7 +3,10 @@
 include @top_builddir@/defs.mak
 
 LIB_DIRS = libcwiid
-BIN_DIRS = wmgui wminput lswm
+BIN_DIRS = wminput lswm
+ifdef HAVE_GTK
+BIN_DIRS += wmgui
+endif
 DOC_DIRS = man doc
 ifdef PYTHON
 BIND_DIRS = python
