diff -Naur cwiid-183-old/wminput/main.c cwiid-183-new/wminput/main.c
--- cwiid-183-old/wminput/main.c	2008-12-28 16:38:12.000000000 -0800
+++ cwiid-183-new/wminput/main.c	2008-12-28 16:38:48.000000000 -0800
@@ -289,7 +289,7 @@
 			}
 			/* TODO: avoid continuously calling cwiid_open */
 			cwiid_set_err(cwiid_err_connect);
-			while (!(wiimote = cwiid_open(&current_bdaddr, CWIID_FLAG_MESG_IFC)));
+			while (!(wiimote = cwiid_open(&current_bdaddr, CWIID_FLAG_MESG_IFC))) sleep(1);
 			cwiid_set_err(cwiid_err_default);
 		}
 		else {
