diff -Naur libX11-1.0.0-old/src/Makefile.am libX11-1.0.0-new/src/Makefile.am
--- libX11-1.0.0-old/src/Makefile.am	2005-12-02 20:41:48.000000000 -0800
+++ libX11-1.0.0-new/src/Makefile.am	2006-01-29 14:56:40.000000000 -0800
@@ -377,7 +377,7 @@
 KEYSYMDEF=@KEYSYMDEF@
 
 ks_tables.h: $(KEYSYMDEF) $(top_builddir)/src/util/makekeys$(EXEEXT)
-	$(top_builddir)/src/util/makekeys < $(KEYSYMDEF) > ks_tables_h
+	xorg-libX11-makekeys < $(KEYSYMDEF) > ks_tables_h
 	mv ks_tables_h $@
 
 $(top_builddir)/src/util/makekeys$(EXEEXT): force
diff -Naur libX11-1.0.0-old/src/Makefile.in libX11-1.0.0-new/src/Makefile.in
--- libX11-1.0.0-old/src/Makefile.in	2005-12-20 22:59:09.000000000 -0800
+++ libX11-1.0.0-new/src/Makefile.in	2006-01-29 14:55:50.000000000 -0800
@@ -1513,7 +1513,7 @@
 	$(COMPILE) -E -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 ks_tables.h: $(KEYSYMDEF) $(top_builddir)/src/util/makekeys$(EXEEXT)
-	$(top_builddir)/src/util/makekeys < $(KEYSYMDEF) > ks_tables_h
+	xorg-libX11-makekeys < $(KEYSYMDEF) > ks_tables_h
 	mv ks_tables_h $@
 
 $(top_builddir)/src/util/makekeys$(EXEEXT): force
