diff -Naur mythtv-0.25-20130112-fa3ebd4-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-0.25-20130112-fa3ebd4-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-0.25-20130112-fa3ebd4-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2013-01-15 18:20:14.000000000 -0800
+++ mythtv-0.25-20130112-fa3ebd4-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2013-01-15 18:20:15.000000000 -0800
@@ -2415,6 +2415,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
