diff -Naur busybox-1.00-old/networking/udhcp/options.c busybox-1.00-new/networking/udhcp/options.c
--- busybox-1.00-old/networking/udhcp/options.c	2004-03-15 00:29:01.000000000 -0800
+++ busybox-1.00-new/networking/udhcp/options.c	2005-07-04 06:36:56.000000000 -0700
@@ -32,7 +32,7 @@
 	{"ipttl",	OPTION_U8,				0x17},
 	{"mtu",		OPTION_U16,				0x1a},
 	{"broadcast",	OPTION_IP | OPTION_REQ,			0x1c},
-	{"ntpsrv",	OPTION_IP | OPTION_LIST,		0x2a},
+	{"ntpsrv",	OPTION_IP | OPTION_LIST | OPTION_REQ,	0x2a},
 	{"wins",	OPTION_IP | OPTION_LIST,		0x2c},
 	{"requestip",	OPTION_IP,				0x32},
 	{"lease",	OPTION_U32,				0x33},
