diff -Naur busybox-1.8.2-old/networking/tftp.c busybox-1.8.2-new/networking/tftp.c
--- busybox-1.8.2-old/networking/tftp.c	2007-11-09 17:40:47.000000000 -0800
+++ busybox-1.8.2-new/networking/tftp.c	2007-11-25 21:19:19.000000000 -0800
@@ -24,7 +24,7 @@
 #if ENABLE_FEATURE_TFTP_GET || ENABLE_FEATURE_TFTP_PUT
 
 #define TFTP_BLOCKSIZE_DEFAULT 512      /* according to RFC 1350, don't change */
-#define TFTP_TIMEOUT_MS         50
+#define TFTP_TIMEOUT_MS        250
 #define TFTP_MAXTIMEOUT_MS    2000
 #define TFTP_NUM_RETRIES        12      /* number of backed-off retries */
 
