diff -Naur busybox-1.11.3-old/networking/udhcp/options.c busybox-1.11.3-new/networking/udhcp/options.c
--- busybox-1.11.3-old/networking/udhcp/options.c	2008-11-14 06:36:24.000000000 -0800
+++ busybox-1.11.3-new/networking/udhcp/options.c	2008-11-14 06:37:08.000000000 -0800
@@ -43,6 +43,7 @@
 	{ OPTION_STRING                           , 0x42 }, /* tftp               */
 	{ OPTION_STRING                           , 0x43 }, /* bootfile           */
 	{ OPTION_STRING                           , 0x4D }, /* userclass          */
+	{ OPTION_STRING               | OPTION_REQ, 0x65 }, /* tcode              */
 #if ENABLE_FEATURE_RFC3397
 	{ OPTION_STR1035 | OPTION_LIST            , 0x77 }, /* search             */
 #endif
@@ -92,6 +93,7 @@
 	"tftp" "\0"
 	"bootfile" "\0"
 	"userclass" "\0"
+	"tcode" "\0"
 #if ENABLE_FEATURE_RFC3397
 	"search" "\0"
 #endif
