diff -Naur busybox-1.7.1-old/networking/udhcp/options.c busybox-1.7.1-new/networking/udhcp/options.c
--- busybox-1.7.1-old/networking/udhcp/options.c	2007-09-23 19:29:54.000000000 -0700
+++ busybox-1.7.1-new/networking/udhcp/options.c	2007-09-23 19:30:34.000000000 -0700
@@ -43,6 +43,7 @@
 	{"tftp",        OPTION_STRING,                          0x42},
 	{"bootfile",    OPTION_STRING,                          0x43},
 	{"userclass",   OPTION_STRING,                          0x4D},
+	{"tcode",       OPTION_STRING | OPTION_REQ,             0x65},
 #if ENABLE_FEATURE_RFC3397
 	{"search",      OPTION_STR1035 | OPTION_LIST | OPTION_REQ, 0x77},
 #endif
