diff -Naur busybox-1.4.2-old/networking/udhcp/options.c busybox-1.4.2-new/networking/udhcp/options.c
--- busybox-1.4.2-old/networking/udhcp/options.c	2007-03-18 11:36:17.000000000 -0700
+++ busybox-1.4.2-new/networking/udhcp/options.c	2007-03-18 11:37:11.000000000 -0700
@@ -43,6 +43,7 @@
 	{"tftp",        OPTION_STRING,                          0x42},
 	{"bootfile",    OPTION_STRING,                          0x43},
 	{"userclass",   OPTION_STRING,                          0x4D},
+	{"tcode",       OPTION_STRING | OPTION_REQ,             0x65},
 	/* MSIE's "Web Proxy Autodiscovery Protocol" support */
 	{"wpad",        OPTION_STRING,                          0xfc},
 	{"",            0x00,                                   0x00}
