diff -Naur busybox-1.4.1-old/networking/udhcp/options.c busybox-1.4.1-new/networking/udhcp/options.c
--- busybox-1.4.1-old/networking/udhcp/options.c	2007-03-10 11:37:36.000000000 -0800
+++ busybox-1.4.1-new/networking/udhcp/options.c	2007-03-10 11:40:49.000000000 -0800
@@ -43,6 +43,7 @@
 	{"tftp",        OPTION_STRING,                          0x42},
 	{"bootfile",    OPTION_STRING,                          0x43},
 	{"userclass",   OPTION_STRING,                          0x4D},
+	{"tcode",       OPTION_STRING | OPTION_REQ,             0x65},
 	/* MSIE's "Web Proxy Autodiscovery Protocol" support */
 	{"wpad",        OPTION_STRING,                          0xfc},
 	{"",            0x00,                                   0x00}
