diff -Naur busybox-1.7.2-old/networking/udhcp/options.c busybox-1.7.2-new/networking/udhcp/options.c
--- busybox-1.7.2-old/networking/udhcp/options.c	2007-09-03 04:48:26.000000000 -0700
+++ busybox-1.7.2-new/networking/udhcp/options.c	2007-10-05 19:04:55.000000000 -0700
@@ -18,7 +18,7 @@
 	{"timesvr",     OPTION_IP | OPTION_LIST,                0x04},
 	{"namesvr",     OPTION_IP | OPTION_LIST,                0x05},
 	{"dns",         OPTION_IP | OPTION_LIST | OPTION_REQ,   0x06},
-	{"logsvr",      OPTION_IP | OPTION_LIST,                0x07},
+	{"logsvr",      OPTION_IP | OPTION_LIST | OPTION_REQ,   0x07},
 	{"cookiesvr",   OPTION_IP | OPTION_LIST,                0x08},
 	{"lprsvr",      OPTION_IP | OPTION_LIST,                0x09},
 	{"hostname",    OPTION_STRING | OPTION_REQ,             0x0c},
