diff -Naur busybox-1.3.2-old/networking/udhcp/options.c busybox-1.3.2-new/networking/udhcp/options.c
--- busybox-1.3.2-old/networking/udhcp/options.c	2007-01-20 12:16:43.000000000 -0800
+++ busybox-1.3.2-new/networking/udhcp/options.c	2007-01-23 14:45:16.000000000 -0800
@@ -18,7 +18,7 @@
 	{"timesvr",     OPTION_IP | OPTION_LIST,                0x04},
 	{"namesvr",     OPTION_IP | OPTION_LIST,                0x05},
 	{"dns",         OPTION_IP | OPTION_LIST | OPTION_REQ,   0x06},
-	{"logsvr",      OPTION_IP | OPTION_LIST,                0x07},
+	{"logsvr",      OPTION_IP | OPTION_LIST | OPTION_REQ,   0x07},
 	{"cookiesvr",   OPTION_IP | OPTION_LIST,                0x08},
 	{"lprsvr",      OPTION_IP | OPTION_LIST,                0x09},
 	{"hostname",    OPTION_STRING | OPTION_REQ,             0x0c},
