diff -Naur Mesa-7.0.1-old/src/glx/x11/dri_glx.c Mesa-7.0.1-new/src/glx/x11/dri_glx.c
--- Mesa-7.0.1-old/src/glx/x11/dri_glx.c	2007-06-21 15:10:54.000000000 -0700
+++ Mesa-7.0.1-new/src/glx/x11/dri_glx.c	2008-03-21 06:08:36.000000000 -0700
@@ -197,7 +197,8 @@
    /* First, search Drivers list to see if we've already opened this driver */
    for (driver = Drivers; driver; driver = driver->next) {
       if (strcmp(driver->name, driverName) == 0) {
-         /* found it */
+         /* found it, increment library refcount & return */
+         dlopen(driver->libpath, RTLD_NOW | RTLD_GLOBAL);
          return driver;
       }
    }
@@ -241,7 +242,9 @@
             break; /* out of memory! */
          /* init the struct */
          driver->name = __glXstrdup(driverName);
-         if (!driver->name) {
+         driver->libpath = __glXstrdup(realDriverName);
+         if (!driver->name || !driver->libpath) {
+            if (driver->name) XFree(driver->name);
             Xfree(driver);
             driver = NULL;
             break; /* out of memory! */
@@ -404,6 +407,7 @@
 		       Drivers = driver->next;
 
 		    Xfree(driver->name);
+		    Xfree(driver->libpath);
 		    Xfree(driver);
 		    break;
 		 }
diff -Naur Mesa-7.0.1-old/src/glx/x11/glxclient.h Mesa-7.0.1-new/src/glx/x11/glxclient.h
--- Mesa-7.0.1-old/src/glx/x11/glxclient.h	2007-06-21 15:10:54.000000000 -0700
+++ Mesa-7.0.1-new/src/glx/x11/glxclient.h	2008-03-21 06:08:36.000000000 -0700
@@ -111,6 +111,7 @@
 */
 struct __DRIdriverRec {
    const char *name;
+   const char *libpath;
    void *handle;
    PFNCREATENEWSCREENFUNC createNewScreenFunc;
    struct __DRIdriverRec *next;
