diff -Naur Mesa-7.0.2-old/configs/linux-dri-x86 Mesa-7.0.2-new/configs/linux-dri-x86
--- Mesa-7.0.2-old/configs/linux-dri-x86	2007-11-01 12:09:01.000000000 -0700
+++ Mesa-7.0.2-new/configs/linux-dri-x86	2008-02-08 07:01:47.000000000 -0800
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86
 
diff -Naur Mesa-7.0.2-old/configs/linux-dri-x86-64 Mesa-7.0.2-new/configs/linux-dri-x86-64
--- Mesa-7.0.2-old/configs/linux-dri-x86-64	2007-06-21 15:10:53.000000000 -0700
+++ Mesa-7.0.2-new/configs/linux-dri-x86-64	2008-02-08 07:02:08.000000000 -0800
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86-64
 
