diff -Naur mythtv-0.18-old/configure mythtv-0.18-new/configure
--- mythtv-0.18-old/configure	2005-04-13 13:35:55.000000000 -0700
+++ mythtv-0.18-new/configure	2005-05-10 06:40:37.314553564 -0700
@@ -3,7 +3,7 @@
 # ffmpeg configure script (c) 2000, 2001, 2002 Fabrice Bellard
 #
 
-x86_mmx_cpus="pentium-mmx,i686,pentium2,pentium3,pentium3m"
+x86_mmx_cpus="c3,c3-2,pentium-mmx,i686,pentium2,pentium3,pentium3m"
 x86_mmx_cpus="$x86_mmx_cpus,i686,pentium-m,pentium4,pentium4m,prescott,athlon,athlon-xp"
 x86_cpus="i386,i486,i586,pentium,pentiumpro,${x86_mmx_cpus}"
 x86_cmov_cpus="pentium-m,pentium4,pentium4m,prescott,athlon,athlon-xp,pentiumpro,i686,c3-2"
