diff -Naur mythvideo-0.17-old/mythvideo/globalsettings.cpp mythvideo-0.17-new/mythvideo/globalsettings.cpp
--- mythvideo-0.17-old/mythvideo/globalsettings.cpp	2005-01-30 16:44:18.000000000 -0800
+++ mythvideo-0.17-new/mythvideo/globalsettings.cpp	2005-04-23 11:32:08.084710901 -0700
@@ -31,7 +31,7 @@
     gc->setLabel(QObject::tr("Default View"));
     gc->addSelection(QObject::tr("Gallery"), "1");
     gc->addSelection(QObject::tr("Browser"), "0");
-    gc->addSelection(QObject::tr("Listings"), "2"); 
+    gc->addSelection(QObject::tr("Listings"), "2", true); 
     gc->setHelpText(QObject::tr("The default view for MythVideo. "
                     "Other views can be reached via the popup menu available "
                     "via the MENU key."));
@@ -77,7 +77,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("VideoTreeNoDB");
     gc->setLabel(QObject::tr("Video List browses files"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If set, this will cause the Video List "
                     "screen to show all relevant files below "
                     "the MythVideo starting directory whether "
@@ -135,7 +135,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("VideoStartupDir");
     gc->setLabel(QObject::tr("Directory that holds videos"));
-    gc->setValue("/share/Movies/dvd");
+    gc->setValue("/mnt/video");
     gc->setHelpText(QObject::tr("This directory must exist, and the user "
                     "running MythVideo only needs to have read permission "
                     "to the directory."));
@@ -161,7 +161,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("VideoDefaultPlayer");
     gc->setLabel(QObject::tr("Default Player"));
-    gc->setValue("mplayer -fs -zoom -quiet -vo xv %s");
+    gc->setValue("xine -pfhq --no-splash --loop");
     gc->setHelpText(QObject::tr("This is the command used for any file "
                     "that the extension is not specifically defined. "
                     "You may also enter the name of one of the playback "
