diff -Naur lesstif-0.94.4-old/acinclude.m4 lesstif-0.94.4-new/acinclude.m4
--- lesstif-0.94.4-old/acinclude.m4	2005-03-31 11:19:11.000000000 -0800
+++ lesstif-0.94.4-new/acinclude.m4	2006-01-06 10:42:24.000000000 -0800
@@ -372,7 +372,6 @@
 AC_DEFUN([LT_HAVE_MAN2HTML],
 [
 AC_MSG_CHECKING(for suitable man2html)
-AC_PATH_PROG(man2html_cmd, man2html)
 if test -x "$man2html_cmd"
 then
   if test -z "`$man2html_cmd </dev/null 2>&1 | grep \"LessTif's man2html\"`"
@@ -403,12 +402,10 @@
 dnl
 AC_DEFUN([LT_HTML2TXT],
 [
-AC_PATH_PROG(LYNX_CMD,lynx)
 if test -x "$LYNX_CMD"
 then
   HTML2TEXT="$LYNX_CMD -dump -force_html "
 else
-  AC_PATH_PROG(LINKS_CMD,links)
   if test -x "$LINKS_CMD"
   then
      HTML2TEXT="$LINKS_CMD -dump "
