diff -Naur linux-2.6.32.2-old/drivers/staging/Kconfig linux-2.6.32.2-new/drivers/staging/Kconfig
--- linux-2.6.32.2-old/drivers/staging/Kconfig	2009-12-18 14:27:07.000000000 -0800
+++ linux-2.6.32.2-new/drivers/staging/Kconfig	2010-01-03 12:53:20.000000000 -0800
@@ -103,6 +103,8 @@
 
 source "drivers/gpu/drm/radeon/Kconfig"
 
+source "drivers/gpu/drm/nouveau/Kconfig"
+
 source "drivers/staging/octeon/Kconfig"
 
 source "drivers/staging/serqt_usb2/Kconfig"
