diff -Naur linux-2.6.32.3-old/drivers/staging/Kconfig linux-2.6.32.3-new/drivers/staging/Kconfig
--- linux-2.6.32.3-old/drivers/staging/Kconfig	2010-01-06 16:39:07.000000000 -0800
+++ linux-2.6.32.3-new/drivers/staging/Kconfig	2010-01-06 16:40:36.000000000 -0800
@@ -103,6 +103,8 @@
 
 source "drivers/gpu/drm/radeon/Kconfig"
 
+source "drivers/gpu/drm/nouveau/Kconfig"
+
 source "drivers/staging/octeon/Kconfig"
 
 source "drivers/staging/serqt_usb2/Kconfig"
