diff -Naur xf86-video-openchrome-737-old/src/via_video.c xf86-video-openchrome-737-new/src/via_video.c
--- xf86-video-openchrome-737-old/src/via_video.c	2009-03-17 06:03:04.000000000 -0700
+++ xf86-video-openchrome-737-new/src/via_video.c	2009-03-17 06:03:25.000000000 -0700
@@ -727,8 +727,13 @@
         BoxPtr pBox;
         int nBox;
 
+#ifdef COMPOSITE
         REGION_TRANSLATE(pScrn->pScreen, clipBoxes, - pPix->screen_x,
             - pPix->screen_y);
+#else
+        REGION_TRANSLATE(pScrn->pScreen, clipBoxes, 0,
+            0);
+#endif
 
         nBox = REGION_NUM_RECTS(clipBoxes);
         pBox = REGION_RECTS(clipBoxes);
