diff -Naur xf86-video-openchrome-0.2.903-old/libxvmc/viaLowLevelPro.c xf86-video-openchrome-0.2.903-new/libxvmc/viaLowLevelPro.c
--- xf86-video-openchrome-0.2.903-old/libxvmc/viaLowLevelPro.c	2008-06-07 10:28:37.000000000 -0700
+++ xf86-video-openchrome-0.2.903-new/libxvmc/viaLowLevelPro.c	2008-11-28 22:56:55.000000000 -0800
@@ -1460,13 +1460,13 @@
 
     if (size != mem->size) {
 	if (mem->size)
-	    drmCommandWrite(xl->fd, DRM_VIA_FREEMEM, mem, sizeof(mem));
+	    drmCommandWrite(xl->fd, DRM_VIA_FREEMEM, mem, sizeof(*mem));
 	mem->context = *(xl->drmcontext);
 	mem->size = size;
 	mem->type = VIA_MEM_VIDEO;
 
 	if (((ret = drmCommandWriteRead(xl->fd, DRM_VIA_ALLOCMEM, mem,
-			sizeof(mem))) < 0) || mem->size != size) {
+			sizeof(*mem))) < 0) || mem->size != size) {
 	    mem->size = 0;
 	    return -1;
 	}
@@ -1484,7 +1484,7 @@
     drm_via_mem_t *mem = &buf->mem;
 
     if (mem->size)
-	drmCommandWrite(xl->fd, DRM_VIA_FREEMEM, mem, sizeof(mem));
+	drmCommandWrite(xl->fd, DRM_VIA_FREEMEM, mem, sizeof(*mem));
     mem->size = 0;
 }
 
