diff -Naur mythtv-0.20-16082-old/libs/libmythtv/vsync.cpp mythtv-0.20-16082-new/libs/libmythtv/vsync.cpp
--- mythtv-0.20-16082-old/libs/libmythtv/vsync.cpp	2008-09-15 14:04:57.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmythtv/vsync.cpp	2009-10-31 08:18:32.000000000 -0700
@@ -76,6 +76,8 @@
 {
     VideoSync *trial = NULL;
     tryingVideoSync  = true;
+    bool tryDRM      = (gContext->GetNumSetting("UseDRMVSync", 1) &&
+                        (getenv("NO_DRM_VSYNC") == NULL));
     bool tryOpenGL   = (gContext->GetNumSetting("UseOpenGLVSync", 1) &&
                         (getenv("NO_OPENGL_VSYNC") == NULL));
 
@@ -92,7 +94,8 @@
     }
     
     TESTVIDEOSYNC(nVidiaVideoSync);
-    TESTVIDEOSYNC(DRMVideoSync);
+    if (tryDRM)
+        TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #ifdef __linux__
diff -Naur mythtv-0.20-16082-old/programs/mythfrontend/globalsettings.cpp mythtv-0.20-16082-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.20-16082-old/programs/mythfrontend/globalsettings.cpp	2008-09-15 14:05:07.000000000 -0700
+++ mythtv-0.20-16082-new/programs/mythfrontend/globalsettings.cpp	2009-10-31 08:19:16.000000000 -0700
@@ -2651,6 +2651,18 @@
     return gc;
 }
 
+static HostCheckBox *UseDRMVSync()
+{
+    HostCheckBox *gc = new HostCheckBox("UseDRMVSync");
+    gc->setLabel(QObject::tr("Enable DRM vertical sync for timing"));
+    gc->setValue(false);
+    gc->setHelpText(QObject::tr(
+                        "If it is supported by your hardware/drivers, "
+                        "MythTV will use DRM vertical syncing for "
+                        "video timing, reducing frame jitter."));
+    return gc;
+}
+
 #ifdef USING_OPENGL_VSYNC
 static HostCheckBox *UseOpenGLVSync()
 {
@@ -3402,6 +3414,7 @@
     general->addChild(new DeinterlaceSettings());
     general->addChild(CustomFilters());
     general->addChild(PreferredMPEG2Decoder());
+    general1->addChild(UseDRMVSync());
 #ifdef USING_OPENGL_VSYNC
     general->addChild(UseOpenGLVSync());
 #endif // USING_OPENGL_VSYNC
