diff -Naur mythtv-0.20-16082-old/libs/libavformat/libavformat.pro mythtv-0.20-16082-new/libs/libavformat/libavformat.pro
--- mythtv-0.20-16082-old/libs/libavformat/libavformat.pro	2008-09-15 14:04:58.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libavformat/libavformat.pro	2010-05-28 14:17:49.000000000 -0700
@@ -47,6 +47,7 @@
 INSTALLS += inc
 
 LIBS += -L../libavcodec -lmythavcodec-$$LIBVERSION -L../libavutil -lmythavutil-$$LIBVERSION
+LIBS += -lz
 
 macx {
     LIBS               += -lz
