diff -Naur mythtv-0.20-16082-old/libs/libmythtv/vsync.cpp mythtv-0.20-16082-new/libs/libmythtv/vsync.cpp
--- mythtv-0.20-16082-old/libs/libmythtv/vsync.cpp	2008-09-15 14:04:57.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmythtv/vsync.cpp	2009-05-18 14:16:19.000000000 -0700
@@ -92,7 +92,7 @@
     }
     
     TESTVIDEOSYNC(nVidiaVideoSync);
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #ifdef __linux__
