diff -Naur mythtv-0.20.2-old/configure mythtv-0.20.2-new/configure
--- mythtv-0.20.2-old/configure	2008-01-31 18:05:34.000000000 -0800
+++ mythtv-0.20.2-new/configure	2008-01-31 18:07:17.000000000 -0800
@@ -2529,7 +2529,7 @@
 fi
 
 lmsensors="no"
-if has_header "sensors/sensors.h" && has_library libsensors ; then
+if has_header "sensors/sensors.h" && has_header "sensors/chips.h" && has_library libsensors ; then
     lmsensors="yes"
 fi
 
