diff -Naur mythtv-0.20-16082-old/libs/libmythtv/util-xvmc.h mythtv-0.20-16082-new/libs/libmythtv/util-xvmc.h
--- mythtv-0.20-16082-old/libs/libmythtv/util-xvmc.h	2008-09-15 14:04:57.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmythtv/util-xvmc.h	2008-09-15 14:07:59.000000000 -0700
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(12),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
diff -Naur mythtv-0.20-16082-old/libs/libmythtv/videoout_xv.cpp mythtv-0.20-16082-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-0.20-16082-old/libs/libmythtv/videoout_xv.cpp	2008-09-15 14:04:57.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmythtv/videoout_xv.cpp	2008-09-15 14:07:59.000000000 -0700
@@ -696,7 +696,7 @@
         myth2av_codecid(myth_codec_id, vld, idct, mc);
 
         if (vld)
-            xvmc_buf_attr->SetNumSurf(16);
+            xvmc_buf_attr->SetNumSurf(12);
 
         vbuffers.Init(xvmc_buf_attr->GetNumSurf(),
                       false /* create an extra frame for pause? */,
