diff -Naur xtrans-1.2-old/Xtranssock.c xtrans-1.2-new/Xtranssock.c
--- xtrans-1.2-old/Xtranssock.c	2008-05-08 07:23:07.000000000 -0700
+++ xtrans-1.2-new/Xtranssock.c	2008-05-08 09:04:06.000000000 -0700
@@ -2576,7 +2576,7 @@
 Xtransport	TRANS(SocketUNIXFuncs) = {
 	/* Socket Interface */
 	"unix",
-#if !defined(LOCALCONN) && !defined(HAVE_ABSTRACT_SOCKETS)
+#if !defined(LOCALCONN)
         TRANS_ALIAS,
 #else
 	0,
