diff -Naur ncurses-5.4-old/ncurses/Makefile.in ncurses-5.4-new/ncurses/Makefile.in
--- ncurses-5.4-old/ncurses/Makefile.in	2004-12-29 07:49:47.679933046 -0800
+++ ncurses-5.4-new/ncurses/Makefile.in	2004-12-29 07:48:35.495244343 -0800
@@ -93,7 +93,7 @@
 CCFLAGS		= $(CPPFLAGS) $(CFLAGS)
 
 BUILD_CC	= @BUILD_CC@
-BUILD_CCFLAGS	= -DHAVE_CONFIG_H -I../ncurses -I$(srcdir) @BUILD_CPPFLAGS@ @BUILD_CFLAGS@
+BUILD_CCFLAGS	= -DHAVE_CONFIG_H -I../ncurses -I$(srcdir) -I$(INCDIR) @BUILD_CPPFLAGS@ @BUILD_CFLAGS@
 BUILD_LDFLAGS	= @BUILD_LDFLAGS@
 BUILD_LIBS	= @BUILD_LIBS@
 
