diff -Naur ncurses-5.6-old/aclocal.m4 ncurses-5.6-new/aclocal.m4
--- ncurses-5.6-old/aclocal.m4	2006-12-17 08:12:38.000000000 -0800
+++ ncurses-5.6-new/aclocal.m4	2006-12-23 17:19:42.000000000 -0800
@@ -73,14 +73,14 @@
 	ACPPFLAGS="-I\${srcdir}/../../include $ACPPFLAGS"
 fi
 if test "$GCC" != yes; then
-	ACPPFLAGS="$ACPPFLAGS -I\${includedir}"
+	ACPPFLAGS="$ACPPFLAGS -I\${DESTDIR}/\${includedir}"
 elif test "$includedir" != "/usr/include"; then
 	if test "$includedir" = '${prefix}/include' ; then
 		if test $prefix != /usr ; then
-			ACPPFLAGS="$ACPPFLAGS -I\${includedir}"
+			ACPPFLAGS="$ACPPFLAGS -I\${DESTDIR}/\${includedir}"
 		fi
 	else
-		ACPPFLAGS="$ACPPFLAGS -I\${includedir}"
+		ACPPFLAGS="$ACPPFLAGS -I\${DESTDIR}/\${includedir}"
 	fi
 fi
 AC_SUBST(ACPPFLAGS)
@@ -1713,14 +1713,14 @@
 	CPPFLAGS="-I\${srcdir}/../include $CPPFLAGS"
 fi
 if test "$GCC" != yes; then
-	CPPFLAGS="$CPPFLAGS -I\${includedir}"
+	CPPFLAGS="$CPPFLAGS -I\${DESTDIR}/\${includedir}"
 elif test "$includedir" != "/usr/include"; then
 	if test "$includedir" = '${prefix}/include' ; then
 		if test $prefix != /usr ; then
-			CPPFLAGS="$CPPFLAGS -I\${includedir}"
+			CPPFLAGS="$CPPFLAGS -I\${DESTDIR}/\${includedir}"
 		fi
 	else
-		CPPFLAGS="$CPPFLAGS -I\${includedir}"
+		CPPFLAGS="$CPPFLAGS -I\${DESTDIR}/\${includedir}"
 	fi
 fi
 AC_SUBST(CPPFLAGS)
diff -Naur ncurses-5.6-old/configure ncurses-5.6-new/configure
--- ncurses-5.6-old/configure	2006-12-17 08:33:38.000000000 -0800
+++ ncurses-5.6-new/configure	2006-12-23 17:20:53.000000000 -0800
@@ -15000,14 +15000,14 @@
 	CPPFLAGS="-I\${srcdir}/../include $CPPFLAGS"
 fi
 if test "$GCC" != yes; then
-	CPPFLAGS="$CPPFLAGS -I\${includedir}"
+	CPPFLAGS="$CPPFLAGS -I\${DESTDIR}/\${includedir}"
 elif test "$includedir" != "/usr/include"; then
 	if test "$includedir" = '${prefix}/include' ; then
 		if test $prefix != /usr ; then
-			CPPFLAGS="$CPPFLAGS -I\${includedir}"
+			CPPFLAGS="$CPPFLAGS -I\${DESTDIR}/\${includedir}"
 		fi
 	else
-		CPPFLAGS="$CPPFLAGS -I\${includedir}"
+		CPPFLAGS="$CPPFLAGS -I\${DESTDIR}/\${includedir}"
 	fi
 fi
 
@@ -15016,14 +15016,14 @@
 	ACPPFLAGS="-I\${srcdir}/../../include $ACPPFLAGS"
 fi
 if test "$GCC" != yes; then
-	ACPPFLAGS="$ACPPFLAGS -I\${includedir}"
+	ACPPFLAGS="$ACPPFLAGS -I\${DESTDIR}/\${includedir}"
 elif test "$includedir" != "/usr/include"; then
 	if test "$includedir" = '${prefix}/include' ; then
 		if test $prefix != /usr ; then
-			ACPPFLAGS="$ACPPFLAGS -I\${includedir}"
+			ACPPFLAGS="$ACPPFLAGS -I\${DESTDIR}/\${includedir}"
 		fi
 	else
-		ACPPFLAGS="$ACPPFLAGS -I\${includedir}"
+		ACPPFLAGS="$ACPPFLAGS -I\${DESTDIR}/\${includedir}"
 	fi
 fi
 
