diff -Naur NVIDIA-Linux-x86_64-169.12-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-169.12-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-169.12-pkg2-old/usr/src/nv/conftest.sh	2008-02-14 18:35:05.000000000 -0800
+++ NVIDIA-Linux-x86_64-169.12-pkg2-new/usr/src/nv/conftest.sh	2008-02-26 16:35:24.000000000 -0800
@@ -875,6 +875,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
