diff -Naur NVIDIA-Linux-x86-169.12-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-169.12-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-169.12-pkg1-old/usr/src/nv/conftest.sh	2008-03-19 11:23:22.000000000 -0700
+++ NVIDIA-Linux-x86-169.12-pkg1-new/usr/src/nv/conftest.sh	2008-03-19 11:23:42.000000000 -0700
@@ -905,6 +905,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
