diff -Naur NVIDIA-Linux-x86-169.12-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-169.12-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-169.12-pkg1-old/usr/src/nv/conftest.sh	2008-05-07 09:28:21.000000000 -0700
+++ NVIDIA-Linux-x86-169.12-pkg1-new/usr/src/nv/conftest.sh	2008-05-07 09:28:00.000000000 -0700
@@ -906,7 +906,6 @@
             #
             echo "#include <linux/autoconf.h>
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
