diff -Naur NVIDIA-Linux-x86-169.12-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-169.12-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-169.12-pkg1-old/usr/src/nv/conftest.sh	2008-02-14 18:47:02.000000000 -0800
+++ NVIDIA-Linux-x86-169.12-pkg1-new/usr/src/nv/conftest.sh	2008-02-26 16:36:26.000000000 -0800
@@ -875,6 +875,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
