diff -Naur NVIDIA-Linux-x86_64-169.12-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-169.12-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-169.12-pkg2-old/usr/src/nv/conftest.sh	2008-05-07 11:39:12.000000000 -0700
+++ NVIDIA-Linux-x86_64-169.12-pkg2-new/usr/src/nv/conftest.sh	2008-05-07 11:39:33.000000000 -0700
@@ -906,7 +906,6 @@
             #
             echo "#include <linux/autoconf.h>
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
