diff -Naur Mesa-7.11.1-old/configure.ac Mesa-7.11.1-new/configure.ac
--- Mesa-7.11.1-old/configure.ac	2011-10-14 17:43:58.000000000 -0700
+++ Mesa-7.11.1-new/configure.ac	2011-11-20 09:57:16.000000000 -0800
@@ -1740,7 +1740,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs` -lstdc++"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS llvmpipe"
