diff -Naur Mesa-7.11-rc4-old/configure.ac Mesa-7.11-rc4-new/configure.ac
--- Mesa-7.11-rc4-old/configure.ac	2011-07-28 11:47:44.000000000 -0700
+++ Mesa-7.11-rc4-new/configure.ac	2011-07-31 08:27:57.000000000 -0700
@@ -1736,7 +1736,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs` -lstdc++"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS llvmpipe"
