diff -Naur Mesa-8.0.4-old/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp Mesa-8.0.4-new/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
--- Mesa-8.0.4-old/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp	2012-07-16 12:27:17.000000000 -0700
+++ Mesa-8.0.4-new/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp	2012-07-16 12:28:32.000000000 -0700
@@ -235,7 +235,24 @@
    int AsmPrinterVariant = AsmInfo->getAssemblerDialect();
 #endif
 
-#if HAVE_LLVM >= 0x0300
+#if HAVE_LLVM >= 0x0301
+   OwningPtr<const MCRegisterInfo> MRI(T->createMCRegInfo(Triple));
+   if (!MRI) {
+      debug_printf("error: no register info for target %s\n", Triple.c_str());
+      return;
+   }
+
+   OwningPtr<const MCInstrInfo> MII(T->createMCInstrInfo());
+   if (!MII) {
+      debug_printf("error: no instruction info for target %s\n", Triple.c_str());
+      return;
+   }
+#endif
+
+#if HAVE_LLVM >= 0x0301
+   OwningPtr<MCInstPrinter> Printer(
+         T->createMCInstPrinter(AsmPrinterVariant, *AsmInfo, *MII, *MRI, *STI));
+#elif HAVE_LLVM == 0x0300
    OwningPtr<MCInstPrinter> Printer(
          T->createMCInstPrinter(AsmPrinterVariant, *AsmInfo, *STI));
 #elif HAVE_LLVM >= 0x0208
diff -Naur Mesa-8.0.4-old/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp Mesa-8.0.4-new/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
--- Mesa-8.0.4-old/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp	2012-07-16 12:27:17.000000000 -0700
+++ Mesa-8.0.4-new/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp	2012-07-16 12:28:32.000000000 -0700
@@ -62,7 +62,11 @@
 extern "C" void
 lp_register_oprofile_jit_event_listener(LLVMExecutionEngineRef EE)
 {
+#if HAVE_LLVM >= 0x0301
+   llvm::unwrap(EE)->RegisterJITEventListener(llvm::JITEventListener::createOProfileJITEventListener());
+#else
    llvm::unwrap(EE)->RegisterJITEventListener(llvm::createOProfileJITEventListener());
+#endif
 }
 
 
