diff -Naur Mesa-9.0.1-old/src/glsl/Makefile.am Mesa-9.0.1-new/src/glsl/Makefile.am
--- Mesa-9.0.1-old/src/glsl/Makefile.am	2012-10-05 13:22:23.000000000 -0700
+++ Mesa-9.0.1-new/src/glsl/Makefile.am	2012-11-16 17:55:43.000000000 -0800
@@ -93,7 +93,7 @@
 CLEANFILES = glsl_lexer.cc glsl_parser.cc $(BUILT_SOURCES)
 
 builtin_function.cpp: $(srcdir)/builtins/profiles/* $(srcdir)/builtins/ir/* $(srcdir)/builtins/glsl/* $(srcdir)/builtins/tools/generate_builtins.py $(srcdir)/builtins/tools/texture_builtins.py builtin_compiler$(EXEEXT)
-	$(AM_V_GEN) $(PYTHON2) $(PYTHON_FLAGS) $(srcdir)/builtins/tools/generate_builtins.py ./builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
+	$(AV_V_GEN) $(PYTHON2) $(PYTHON_FLAGS) $(srcdir)/builtins/tools/generate_builtins.py mesa-builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
 
 glcpp/libglcpp.la:
 	cd glcpp ; $(MAKE) $(AM_MAKEFLAGS)
