diff -Naur Mesa-8.0.4-old/src/glsl/Makefile Mesa-8.0.4-new/src/glsl/Makefile
--- Mesa-8.0.4-old/src/glsl/Makefile	2012-07-16 12:29:32.000000000 -0700
+++ Mesa-8.0.4-new/src/glsl/Makefile	2012-07-16 12:30:14.000000000 -0700
@@ -167,6 +167,6 @@
 
 builtin_function.cpp: builtins/profiles/* builtins/ir/* builtins/tools/generate_builtins.py builtins/tools/texture_builtins.py builtin_compiler
 	@echo Regenerating builtin_function.cpp...
-	$(PYTHON2) $(PYTHON_FLAGS) builtins/tools/generate_builtins.py ./builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
+	$(PYTHON2) $(PYTHON_FLAGS) builtins/tools/generate_builtins.py mesa-builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
 
 -include depend
