diff -Naur Mesa-7.9.1-old/src/gallium/drivers/nouveau/nouveau_screen.c Mesa-7.9.1-new/src/gallium/drivers/nouveau/nouveau_screen.c
--- Mesa-7.9.1-old/src/gallium/drivers/nouveau/nouveau_screen.c	2010-12-15 13:50:00.000000000 -0800
+++ Mesa-7.9.1-new/src/gallium/drivers/nouveau/nouveau_screen.c	2011-01-11 12:46:28.000000000 -0800
@@ -236,7 +236,7 @@
 	int ret;
 
 	ret = nouveau_channel_alloc(dev, 0xbeef0201, 0xbeef0202,
-				    &screen->channel);
+				    512*1024, &screen->channel);
 	if (ret)
 		return ret;
 	screen->device = dev;
diff -Naur Mesa-7.9.1-old/src/mesa/drivers/dri/nouveau/nouveau_context.c Mesa-7.9.1-new/src/mesa/drivers/dri/nouveau/nouveau_context.c
--- Mesa-7.9.1-old/src/mesa/drivers/dri/nouveau/nouveau_context.c	2010-12-15 13:50:00.000000000 -0800
+++ Mesa-7.9.1-new/src/mesa/drivers/dri/nouveau/nouveau_context.c	2011-01-11 12:46:28.000000000 -0800
@@ -129,7 +129,7 @@
 
 	/* Allocate a hardware channel. */
 	ret = nouveau_channel_alloc(context_dev(ctx), 0xbeef0201, 0xbeef0202,
-				    &nctx->hw.chan);
+				    512*1024, &nctx->hw.chan);
 	if (ret) {
 		nouveau_error("Error initializing the FIFO.\n");
 		return GL_FALSE;
diff -Naur Mesa-7.9.1-old/src/mesa/drivers/dri/nouveau/nouveau_vbo_t.c Mesa-7.9.1-new/src/mesa/drivers/dri/nouveau/nouveau_vbo_t.c
--- Mesa-7.9.1-old/src/mesa/drivers/dri/nouveau/nouveau_vbo_t.c	2010-12-15 13:50:00.000000000 -0800
+++ Mesa-7.9.1-new/src/mesa/drivers/dri/nouveau/nouveau_vbo_t.c	2011-01-11 12:46:28.000000000 -0800
@@ -32,7 +32,7 @@
 
 /* Arbitrary pushbuf length we can assume we can get with a single
  * WAIT_RING. */
-#define PUSHBUF_DWORDS 2048
+#define PUSHBUF_DWORDS 65536
 
 /* Functions to set up struct nouveau_array_state from something like
  * a GL array or index buffer. */
