diff -Naur Mesa-7.11.2-old/configure.ac Mesa-7.11.2-new/configure.ac
--- Mesa-7.11.2-old/configure.ac	2012-01-17 06:12:09.000000000 -0800
+++ Mesa-7.11.2-new/configure.ac	2012-01-17 06:13:14.000000000 -0800
@@ -1740,7 +1740,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs` -lstdc++"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS llvmpipe"
