diff -Naur libaacs-db866cd-old/configure.ac libaacs-db866cd-new/configure.ac
--- libaacs-db866cd-old/configure.ac	2011-10-01 06:39:25.000000000 -0700
+++ libaacs-db866cd-new/configure.ac	2011-10-01 06:39:58.000000000 -0700
@@ -101,6 +101,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
