diff -Naur libaacs-7801a71-old/configure.ac libaacs-7801a71-new/configure.ac
--- libaacs-7801a71-old/configure.ac	2011-12-01 11:14:08.000000000 -0800
+++ libaacs-7801a71-new/configure.ac	2011-12-01 11:14:32.000000000 -0800
@@ -123,6 +123,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
