diff -Naur libaacs-876f45a-old/configure.ac libaacs-876f45a-new/configure.ac
--- libaacs-876f45a-old/configure.ac	2011-11-02 17:29:46.000000000 -0700
+++ libaacs-876f45a-new/configure.ac	2011-11-02 17:31:35.000000000 -0700
@@ -103,6 +103,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
