diff -Naur libaacs-11542bc75ac203dc4e315cdf5cc8cbae6f22cc70-old/configure.ac libaacs-11542bc75ac203dc4e315cdf5cc8cbae6f22cc70-new/configure.ac
--- libaacs-11542bc75ac203dc4e315cdf5cc8cbae6f22cc70-old/configure.ac	2010-08-24 18:07:28.000000000 -0700
+++ libaacs-11542bc75ac203dc4e315cdf5cc8cbae6f22cc70-new/configure.ac	2010-08-24 18:09:58.000000000 -0700
@@ -66,6 +66,10 @@
 AC_CHECK_LIB([gcrypt], [gcry_check_version],,
   [AC_MSG_ERROR($library_not_found)])
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
