diff -Naur libaacs-964342f-old/configure.ac libaacs-964342f-new/configure.ac
--- libaacs-964342f-old/configure.ac	2011-06-24 13:35:05.000000000 -0700
+++ libaacs-964342f-new/configure.ac	2011-06-24 13:35:38.000000000 -0700
@@ -101,6 +101,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
