diff -Naur e2fsprogs-1.36-old/lib/Makefile.elf-lib e2fsprogs-1.36-new/lib/Makefile.elf-lib
--- e2fsprogs-1.36-old/lib/Makefile.elf-lib	2004-12-15 08:30:39.000000000 -0800
+++ e2fsprogs-1.36-new/lib/Makefile.elf-lib	2005-02-11 05:16:29.780412624 -0800
@@ -47,7 +47,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
