diff -Naur e2fsprogs-1.35-old/lib/Makefile.elf-lib e2fsprogs-1.35-new/lib/Makefile.elf-lib
--- e2fsprogs-1.35-old/lib/Makefile.elf-lib	2003-12-07 09:11:38.000000000 -0800
+++ e2fsprogs-1.35-new/lib/Makefile.elf-lib	2004-12-19 07:47:07.458974662 -0800
@@ -43,7 +43,7 @@
 	$(STRIP) --strip-debug \
 		$(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB)
 	$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
-	$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	-$(LDCONFIG)
 
