diff -Naur e2fsprogs-1.40.5-old/configure.in e2fsprogs-1.40.5-new/configure.in
--- e2fsprogs-1.40.5-old/configure.in	2008-01-27 17:03:07.000000000 -0800
+++ e2fsprogs-1.40.5-new/configure.in	2008-01-28 08:57:09.000000000 -0800
@@ -649,14 +649,6 @@
 dnl
 dnl Word sizes...
 dnl
-if test "$cross_compiling" = yes -a "$ac_cv_sizeof_long" = ""; then
-  # if cross-compiling, with no cached values, just assume something common. 
-  ac_cv_sizeof_short=2
-  ac_cv_sizeof_int=4
-  ac_cv_sizeof_long=4
-  ac_cv_sizeof_long_long=8
-  AC_MSG_WARN([Cross-compiling, so cannot check type sizes; assuming short=2, int=4, long=4, long long=8])
-fi
 AC_CHECK_SIZEOF(short)
 AC_CHECK_SIZEOF(int)
 AC_CHECK_SIZEOF(long)
