diff -Naur e2fsprogs-1.38-old/lib/Makefile.elf-lib e2fsprogs-1.38-new/lib/Makefile.elf-lib
--- e2fsprogs-1.38-old/lib/Makefile.elf-lib	2004-12-15 08:30:39.000000000 -0800
+++ e2fsprogs-1.38-new/lib/Makefile.elf-lib	2006-04-23 07:50:51.000000000 -0700
@@ -47,7 +47,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
diff -Naur e2fsprogs-1.38-old/util/Makefile.in e2fsprogs-1.38-new/util/Makefile.in
--- e2fsprogs-1.38-old/util/Makefile.in	2005-06-29 10:42:20.000000000 -0700
+++ e2fsprogs-1.38-new/util/Makefile.in	2006-04-23 07:55:38.000000000 -0700
@@ -15,7 +15,7 @@
 
 .c.o:
 	@echo "	CC $<"
-	@$(BUILD_CC) -c $(ALL_CFLAGS) $< -o $@
+	@$(BUILD_CC) -c $< -o $@
 
 PROGS=		subst
 
@@ -23,11 +23,11 @@
 
 subst: subst.o
 	@echo "	LD $@"
-	@$(BUILD_CC) $(ALL_LDFLAGS) -o subst subst.o
+	@$(BUILD_CC) -o subst subst.o
 
 copy_sparse: copy_sparse.o
 	@echo "	LD $@"
-	@$(BUILD_CC) $(ALL_LDFLAGS) -o copy_sparse copy_sparse.o
+	@$(BUILD_CC) -o copy_sparse copy_sparse.o
 
 gen-tarball: $(srcdir)/gen-tarball.in $(top_builddir)/config.status
 	@echo "	CONFIG.STATUS $@"
