diff -Naur e2fsprogs-1.40.4-old/lib/Makefile.elf-lib e2fsprogs-1.40.4-new/lib/Makefile.elf-lib
--- e2fsprogs-1.40.4-old/lib/Makefile.elf-lib	2007-06-30 05:58:34.000000000 -0700
+++ e2fsprogs-1.40.4-new/lib/Makefile.elf-lib	2008-01-01 14:48:44.000000000 -0800
@@ -45,7 +45,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
