diff -Naur libX11-1.2.2-old/configure.ac libX11-1.2.2-new/configure.ac
--- libX11-1.2.2-old/configure.ac	2009-07-09 21:07:33.000000000 -0700
+++ libX11-1.2.2-new/configure.ac	2009-07-10 06:41:36.000000000 -0700
@@ -93,6 +93,10 @@
 		# Unix domain sockets are faster now.
 		DEFAULT_LOCAL_TRANS="UNIX_TRANS,LOCAL_TRANS,TCP_TRANS"
 		;;
+	linux*)
+		# LOCAL_TRANS is used for abstract sockets.
+		DEFAULT_LOCAL_TRANS="LOCAL_TRANS,UNIX_TRANS,TCP_TRANS"
+		;;
 	*)
 		if test "$LOCALCONN" = "yes"; then
 			DEFAULT_LOCAL_TRANS="LOCAL_TRANS,UNIX_TRANS,TCP_TRANS"
diff -Naur libX11-1.2.2-old/src/ConnDis.c libX11-1.2.2-new/src/ConnDis.c
--- libX11-1.2.2-old/src/ConnDis.c	2009-04-07 02:28:44.000000000 -0700
+++ libX11-1.2.2-new/src/ConnDis.c	2009-07-10 06:41:36.000000000 -0700
@@ -93,7 +93,7 @@
 #ifdef UNIXCONN
 # define UNIX_TRANS	"unix"
 #endif
-#if defined(LOCALCONN) || defined(OS2PIPECONN)
+#if defined(LOCALCONN) || defined(OS2PIPECONN) || defined(UNIXCONN)
 # define LOCAL_TRANS	"local"
 #endif
 
