diff -Naur Mesa-7.4-old/configs/default Mesa-7.4-new/configs/default
--- Mesa-7.4-old/configs/default	2009-03-12 20:28:49.000000000 -0700
+++ Mesa-7.4-new/configs/default	2009-03-27 22:15:53.000000000 -0700
@@ -32,6 +32,7 @@
 MKDEP_OPTIONS = -fdepend
 MAKE = make
 INSTALL = $(SHELL) $(TOP)/bin/minstall
+PKG_CONFIG = pkg-config
 
 # Tools for regenerating glapi (generally only needed by the developers)
 PYTHON2 = python
diff -Naur Mesa-7.4-old/configs/freebsd-dri Mesa-7.4-new/configs/freebsd-dri
--- Mesa-7.4-old/configs/freebsd-dri	2009-03-12 20:28:49.000000000 -0700
+++ Mesa-7.4-new/configs/freebsd-dri	2009-03-27 22:15:53.000000000 -0700
@@ -29,8 +29,8 @@
 ASM_SOURCES = 
 
 # Library/program dependencies
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm`
-LIBDRM_LIB = `pkg-config --libs libdrm`
+LIBDRM_CFLAGS = `$(PKG_CONFIG) --cflags libdrm`
+LIBDRM_LIB = `$(PKG_CONFIG) --libs libdrm`
 DRI_LIB_DEPS = -L/usr/local/lib -lm -pthread -lexpat $(LIBDRM_LIB)
 GL_LIB_DEPS = -L/usr/local/lib -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 	-lm -pthread $(LIBDRM_LIB)
diff -Naur Mesa-7.4-old/configs/linux-dri Mesa-7.4-new/configs/linux-dri
--- Mesa-7.4-old/configs/linux-dri	2009-03-12 20:28:49.000000000 -0700
+++ Mesa-7.4-new/configs/linux-dri	2009-03-27 22:15:53.000000000 -0700
@@ -43,8 +43,8 @@
 # Library/program dependencies
 EXTRA_LIB_PATH=-L/usr/X11R6/lib
 
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm`
-LIBDRM_LIB = `pkg-config --libs libdrm`
+LIBDRM_CFLAGS = `$(PKG_CONFIG) --cflags libdrm`
+LIBDRM_LIB = `$(PKG_CONFIG) --libs libdrm`
 DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 		-lm -lpthread -ldl $(LIBDRM_LIB)
diff -Naur Mesa-7.4-old/configs/linux-dri-xcb Mesa-7.4-new/configs/linux-dri-xcb
--- Mesa-7.4-old/configs/linux-dri-xcb	2009-03-12 20:28:49.000000000 -0700
+++ Mesa-7.4-new/configs/linux-dri-xcb	2009-03-27 22:15:53.000000000 -0700
@@ -25,7 +25,7 @@
 	-DGLX_DIRECT_RENDERING -DGLX_INDIRECT_RENDERING \
         -DHAVE_ALIAS -DUSE_XCB -DHAVE_POSIX_MEMALIGN
 
-X11_INCLUDES = `pkg-config --cflags-only-I x11` `pkg-config --cflags-only-I xcb` `pkg-config --cflags-only-I x11-xcb` `pkg-config --cflags-only-I xcb-glx`
+X11_INCLUDES = `$(PKG_CONFIG) --cflags-only-I x11` `$(PKG_CONFIG) --cflags-only-I xcb` `$(PKG_CONFIG) --cflags-only-I x11-xcb` `$(PKG_CONFIG) --cflags-only-I xcb-glx`
 
 CFLAGS = -Wall -Wmissing-prototypes $(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) \
 	$(DEFINES) $(ASM_FLAGS) -std=c99 -ffast-math 
@@ -39,14 +39,14 @@
 ASM_SOURCES = 
 
 # Library/program dependencies
-EXTRA_LIB_PATH=`pkg-config --libs-only-L x11`
+EXTRA_LIB_PATH=`$(PKG_CONFIG) --libs-only-L x11`
 
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm`
-LIBDRM_LIB = `pkg-config --libs libdrm`
+LIBDRM_CFLAGS = `$(PKG_CONFIG) --cflags libdrm`
+LIBDRM_LIB = `$(PKG_CONFIG) --libs libdrm`
 DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
-		-lm -lpthread -ldl $(LIBDRM_LIB) `pkg-config --libs xcb` \
-		`pkg-config --libs x11-xcb` `pkg-config --libs xcb-glx`
+		-lm -lpthread -ldl $(LIBDRM_LIB) `$(PKG_CONFIG) --libs xcb` \
+		`$(PKG_CONFIG) --libs x11-xcb` `$(PKG_CONFIG) --libs xcb-glx`
 
 
 # This is now 0 by default since it seems to confuse the hell out of people
diff -Naur Mesa-7.4-old/configs/linux-solo Mesa-7.4-new/configs/linux-solo
--- Mesa-7.4-old/configs/linux-solo	2009-03-12 20:28:49.000000000 -0700
+++ Mesa-7.4-new/configs/linux-solo	2009-03-27 22:15:53.000000000 -0700
@@ -16,10 +16,10 @@
 ARCH_FLAGS ?=
 
 # DRM and pciaccess
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm`
-LIBDRM_LIB = `pkg-config --libs libdrm`
-PCIACCESS_CFLAGS = `pkg-config --cflags pciaccess`
-PCIACCESS_LIB = `pkg-config --libs pciaccess`
+LIBDRM_CFLAGS = `$(PKG_CONFIG) --cflags libdrm`
+LIBDRM_LIB = `$(PKG_CONFIG) --libs libdrm`
+PCIACCESS_CFLAGS = `$(PKG_CONFIG) --cflags pciaccess`
+PCIACCESS_LIB = `$(PKG_CONFIG) --libs pciaccess`
 
 
 DEFINES = -D_POSIX_SOURCE -D_POSIX_C_SOURCE=199309L -D_SVID_SOURCE \
diff -Naur Mesa-7.4-old/src/glut/directfb/Makefile Mesa-7.4-new/src/glut/directfb/Makefile
--- Mesa-7.4-old/src/glut/directfb/Makefile	2009-03-12 20:28:49.000000000 -0700
+++ Mesa-7.4-new/src/glut/directfb/Makefile	2009-03-27 22:15:53.000000000 -0700
@@ -9,9 +9,9 @@
 GLUT_MINOR = 7
 GLUT_TINY = 1
 
-INCLUDES = -I$(TOP)/include -I$(MARK) $(shell pkg-config --cflags directfb)
+INCLUDES = -I$(TOP)/include -I$(MARK) $(shell $(PKG_CONFIG) --cflags directfb)
 
-GLUT_LIB_DEPS += $(shell pkg-config --libs directfb)
+GLUT_LIB_DEPS += $(shell $(PKG_CONFIG) --libs directfb)
 
 CORE_SOURCES = \
 	callback.c \
diff -Naur Mesa-7.4-old/src/mesa/drivers/directfb/Makefile Mesa-7.4-new/src/mesa/drivers/directfb/Makefile
--- Mesa-7.4-old/src/mesa/drivers/directfb/Makefile	2008-08-25 07:46:42.000000000 -0700
+++ Mesa-7.4-new/src/mesa/drivers/directfb/Makefile	2009-03-27 22:15:53.000000000 -0700
@@ -16,8 +16,8 @@
 	-I$(TOP)/src/mesa/swrast \
 	-I$(TOP)/src/mesa/swrast_setup
 
-DFB_CFLAGS    = $(shell pkg-config --cflags directfb)
-DFB_MODULEDIR = $(shell pkg-config --variable=moduledir directfb-internal)
+DFB_CFLAGS    = $(shell $(PKG_CONFIG) --cflags directfb)
+DFB_MODULEDIR = $(shell $(PKG_CONFIG) --variable=moduledir directfb-internal)
 
 DIRECTFBGL_MESA_SOURCES = ../common/driverfuncs.c idirectfbgl_mesa.c
 
