diff -Naur Mesa-7.4.1-old/src/mesa/drivers/dri/i915/intel_context.c Mesa-7.4.1-new/src/mesa/drivers/dri/i915/intel_context.c
--- Mesa-7.4.1-old/src/mesa/drivers/dri/i915/intel_context.c	2009-04-18 09:51:25.000000000 -0700
+++ Mesa-7.4.1-new/src/mesa/drivers/dri/i915/intel_context.c	2009-04-25 06:49:37.000000000 -0700
@@ -534,7 +534,8 @@
    if ((ctx->DrawBuffer->Name == 0) && intel->front_buffer_dirty) {
       __DRIscreen *const screen = intel->intelScreen->driScrnPriv;
 
-      if ((screen->dri2.loader->base.version >= 2)
+      if (screen->dri2.loader
+          && (screen->dri2.loader->base.version >= 2)
 	  && (screen->dri2.loader->flushFrontBuffer != NULL)) {
 	 (*screen->dri2.loader->flushFrontBuffer)(intel->driDrawable,
 						  intel->driDrawable->loaderPrivate);
diff -Naur Mesa-7.4.1-old/src/mesa/drivers/dri/i965/intel_context.c Mesa-7.4.1-new/src/mesa/drivers/dri/i965/intel_context.c
--- Mesa-7.4.1-old/src/mesa/drivers/dri/i965/intel_context.c	2009-04-18 09:51:25.000000000 -0700
+++ Mesa-7.4.1-new/src/mesa/drivers/dri/i965/intel_context.c	2009-04-25 06:49:37.000000000 -0700
@@ -534,7 +534,8 @@
    if ((ctx->DrawBuffer->Name == 0) && intel->front_buffer_dirty) {
       __DRIscreen *const screen = intel->intelScreen->driScrnPriv;
 
-      if ((screen->dri2.loader->base.version >= 2)
+      if (screen->dri2.loader
+          && (screen->dri2.loader->base.version >= 2)
 	  && (screen->dri2.loader->flushFrontBuffer != NULL)) {
 	 (*screen->dri2.loader->flushFrontBuffer)(intel->driDrawable,
 						  intel->driDrawable->loaderPrivate);
diff -Naur Mesa-7.4.1-old/src/mesa/drivers/dri/intel/intel_context.c Mesa-7.4.1-new/src/mesa/drivers/dri/intel/intel_context.c
--- Mesa-7.4.1-old/src/mesa/drivers/dri/intel/intel_context.c	2009-04-18 09:51:25.000000000 -0700
+++ Mesa-7.4.1-new/src/mesa/drivers/dri/intel/intel_context.c	2009-04-25 06:49:37.000000000 -0700
@@ -534,7 +534,8 @@
    if ((ctx->DrawBuffer->Name == 0) && intel->front_buffer_dirty) {
       __DRIscreen *const screen = intel->intelScreen->driScrnPriv;
 
-      if ((screen->dri2.loader->base.version >= 2)
+      if (screen->dri2.loader
+          && (screen->dri2.loader->base.version >= 2)
 	  && (screen->dri2.loader->flushFrontBuffer != NULL)) {
 	 (*screen->dri2.loader->flushFrontBuffer)(intel->driDrawable,
 						  intel->driDrawable->loaderPrivate);
