diff -Naur Mesa-7.1-old/configure.ac Mesa-7.1-new/configure.ac
--- Mesa-7.1-old/configure.ac	2008-08-30 15:18:03.000000000 -0700
+++ Mesa-7.1-new/configure.ac	2008-08-30 15:24:56.000000000 -0700
@@ -4,10 +4,10 @@
 
 dnl Versioning - scrape the version from configs/default
 m4_define([mesa_version],
-    [m4_esyscmd([${MAKE-make} -s -f bin/version.mk version | tr -d '\n'])])
+    [7.1.0])
 m4_ifval(mesa_version,[],[
     m4_errprint([Error: Failed to get the Mesa version from the output of
-       running `make -f bin/version.mk version'
+       running `make -f bin/version.mk version-mk-version'
 ])
     m4_exit([1])
 ])
