diff -Naur Mesa-7.1-rc1-old/src/mesa/x86/Makefile Mesa-7.1-rc1-new/src/mesa/x86/Makefile
--- Mesa-7.1-rc1-old/src/mesa/x86/Makefile	2008-07-01 14:15:48.000000000 -0700
+++ Mesa-7.1-rc1-new/src/mesa/x86/Makefile	2008-07-01 14:20:14.000000000 -0700
@@ -21,7 +21,7 @@
 
 
 gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+	$(BUILD_CC) $(INCLUDE_DIRS) $(BUILD_CFLAGS) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
 matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
