diff -Naur Mesa-7.5.1-old/progs/glsl/Makefile Mesa-7.5.1-new/progs/glsl/Makefile
--- Mesa-7.5.1-old/progs/glsl/Makefile	2009-08-31 16:13:07.000000000 -0700
+++ Mesa-7.5.1-new/progs/glsl/Makefile	2009-09-06 04:34:05.000000000 -0700
@@ -190,7 +190,7 @@
 
 
 samplers_array.o: samplers.c $(UTIL_HEADERS)
-	$(APP_CC) $(CFLAGS) -DSAMPLERS_ARRAY $< -c -o $@
+	$(APP_CC) -c -I$(INCDIR) $(CFLAGS) -DSAMPLERS_ARRAY $< -o $@
 
 samplers_array: samplers_array.o $(UTIL_OBJS)
 
