diff -Naur Mesa-7.7.1-old/src/mesa/drivers/dri/r200/r200_cmdbuf.c Mesa-7.7.1-new/src/mesa/drivers/dri/r200/r200_cmdbuf.c
--- Mesa-7.7.1-old/src/mesa/drivers/dri/r200/r200_cmdbuf.c	2010-03-26 18:14:51.000000000 -0700
+++ Mesa-7.7.1-new/src/mesa/drivers/dri/r200/r200_cmdbuf.c	2010-10-27 11:30:35.000000000 -0700
@@ -51,9 +51,8 @@
  */
 #define insert_at_tail_if(atom_list, atom) \
    do { \
-      struct radeon_state_atom* __atom = (atom); \
-      if (__atom->check) \
-	 insert_at_tail((atom_list), __atom); \
+      if ((atom)->check) \
+	 insert_at_tail((atom_list), (atom)); \
    } while(0)
 
 void r200SetUpAtomList( r200ContextPtr rmesa )
