diff -Naur gawk-3.1.8-old/configure.ac gawk-3.1.8-new/configure.ac
--- gawk-3.1.8-old/configure.ac	2010-05-06 10:55:35.000000000 -0700
+++ gawk-3.1.8-new/configure.ac	2011-11-04 06:41:48.000000000 -0700
@@ -171,9 +171,9 @@
 AC_TYPE_SIGNAL
 AC_SIZE_T
 AC_TYPE_GETGROUPS
-gl_AC_TYPE_LONG_LONG
-gl_AC_TYPE_UNSIGNED_LONG_LONG
-gl_AC_TYPE_INTMAX_T
+AC_TYPE_LONG_LONG_INT
+AC_TYPE_UNSIGNED_LONG_LONG_INT
+gt_TYPE_INTMAX_T
 gl_AC_TYPE_UINTMAX_T
 AC_CHECK_TYPE(ssize_t, int)
 AC_CHECK_SIZEOF(unsigned int)
