diff -Naur alsa-driver-1.0.14-old/Makefile alsa-driver-1.0.14-new/Makefile
--- alsa-driver-1.0.14-old/Makefile	2007-06-04 00:28:51.000000000 -0700
+++ alsa-driver-1.0.14-new/Makefile	2007-06-06 12:38:50.000000000 -0700
@@ -201,7 +201,7 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	-depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
 	@echo "*** This is a \`staged' install using the prefix"
 	@echo "***	\"$(DESTDIR)\"."
