diff -Naur libaacs-ac44f760155953e6aa9e96d602109bc79ae14330-old/configure.ac libaacs-ac44f760155953e6aa9e96d602109bc79ae14330-new/configure.ac
--- libaacs-ac44f760155953e6aa9e96d602109bc79ae14330-old/configure.ac	2010-09-28 20:18:39.000000000 -0700
+++ libaacs-ac44f760155953e6aa9e96d602109bc79ae14330-new/configure.ac	2010-09-28 20:19:13.000000000 -0700
@@ -97,6 +97,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
