diff -Naur libaacs-c5f01a9-old/configure.ac libaacs-c5f01a9-new/configure.ac
--- libaacs-c5f01a9-old/configure.ac	2011-11-09 15:33:12.000000000 -0800
+++ libaacs-c5f01a9-new/configure.ac	2011-11-09 15:34:19.000000000 -0800
@@ -103,6 +103,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
