diff -Naur libaacs-old/configure.ac libaacs-new/configure.ac
--- libaacs-old/configure.ac	2010-08-10 09:18:36.000000000 -0700
+++ libaacs-new/configure.ac	2010-08-10 10:24:40.000000000 -0700
@@ -62,6 +62,10 @@
 AC_CHECK_LIB([gcrypt], [gcry_check_version],,
   [AC_MSG_ERROR($library_not_found)])
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
