diff -Naur libaacs-f5c5b88f843d5dcb6ebe8cd01044d04f276985ea-old/configure.ac libaacs-f5c5b88f843d5dcb6ebe8cd01044d04f276985ea-new/configure.ac
--- libaacs-f5c5b88f843d5dcb6ebe8cd01044d04f276985ea-old/configure.ac	2010-12-17 20:18:39.000000000 -0700
+++ libaacs-f5c5b88f843d5dcb6ebe8cd01044d04f276985ea-new/configure.ac	2010-12-17 20:19:13.000000000 -0700
@@ -97,6 +97,10 @@
 fi
 AC_FUNC_STRERROR_R
 
+# gpg-error check
+AC_CHECK_LIB([gpg-error], [gpg_strerror_r],,
+  [AC_MSG_ERROR($library_not_found)])
+
 # function testing for supported compiler options
 check_cc_options()
 {
