diff -Naur zsnes-4077-old/src/acinclude.m4 zsnes-4077-new/src/acinclude.m4
--- zsnes-4077-old/src/acinclude.m4	2006-12-15 20:44:58.000000000 -0800
+++ zsnes-4077-new/src/acinclude.m4	2006-12-15 20:57:28.000000000 -0800
@@ -90,7 +90,7 @@
   with_zlib=yes,
   with_zlib=no,
   [AC_MSG_RESULT(cross-compiling)
-  with_zlib=""
+  with_zlib=yes
   AC_MSG_WARN(Assuming zlib is available)])
 
 if test x$with_zlib != x; then
@@ -224,7 +224,7 @@
   with_libpng=yes,
   with_libpng=no,
   [AC_MSG_RESULT(cross-compiling)
-  with_libpng=""
+  with_libpng=yes
   AC_MSG_WARN(Assuming libpng is available)])
 
   if test x$with_libpng != x; then
diff -Naur zsnes-4077-old/src/configure.in zsnes-4077-new/src/configure.in
--- zsnes-4077-old/src/configure.in	2006-12-15 20:44:58.000000000 -0800
+++ zsnes-4077-new/src/configure.in	2006-12-15 20:47:32.000000000 -0800
@@ -7,12 +7,12 @@
 
 VERSION=1.50
 
-CFLAGS="$CFLAGS -pipe -I. -I/usr/local/include -I/usr/include"
+CFLAGS="$CFLAGS -pipe -I."
 dnl - You might wanna change -I/usr/include if you're trying to do this with
 dnl Mingw, because it doesnt have a standard file tree. Maybe if ming do this
 dnl else do -I/usr/include. Or some shit. You figure it out.
 
-LDFLAGS="$LDFLAGS -L/usr/local/lib -L/usr/lib"
+LDFLAGS="$LDFLAGS"
 dnl - Same thing here.
 
 dnl - Where are our compilers, and who are we compiling for?
