diff -Naur lm_sensors-3.1.2-old/CHANGES lm_sensors-3.1.2-new/CHANGES
--- lm_sensors-3.1.2-old/CHANGES	2010-06-01 06:51:15.000000000 -0700
+++ lm_sensors-3.1.2-new/CHANGES	2010-06-01 06:51:40.000000000 -0700
@@ -8,6 +8,7 @@
   fancontrol: Check that all referenced sysfs files exist
               Check that all devices match the configuration file
   pwmconfig: Exit immediately if not root
+                  Fix detection of RB-C3 AMD Family 10h CPU
              Save device paths and names in configuration file
   sensors.conf.default: Encourage user to not modify this file
                         Add a section for adt7473 and adt7475
diff -Naur lm_sensors-3.1.2-old/prog/detect/sensors-detect lm_sensors-3.1.2-new/prog/detect/sensors-detect
--- lm_sensors-3.1.2-old/prog/detect/sensors-detect	2010-06-01 06:51:13.000000000 -0700
+++ lm_sensors-3.1.2-new/prog/detect/sensors-detect	2010-06-01 06:51:40.000000000 -0700
@@ -5422,7 +5422,7 @@
 
 		next if $probecpu->{model} < 4;		# DR-B*
 		next if $probecpu->{model} == 8;	# HY-D0
-		if ($probecpu->{model} == 4) {		# RB-C*
+		if ($probecpu->{model} == 4 && $probecpu->{stepping} <= 2) {	# RB-C2
 			my @dram_cfg = split /\n/, `setpci -d 1022:1202 94.L 2>/dev/null`;
 			next unless @dram_cfg >= 1;
 			next unless hex($dram_cfg[0]) & 0x00000100;	# DDR3
