diff -Naur lm_sensors-3.1.2-old/CHANGES lm_sensors-3.1.2-new/CHANGES
--- lm_sensors-3.1.2-old/CHANGES	2010-08-22 13:36:50.000000000 -0700
+++ lm_sensors-3.1.2-new/CHANGES	2010-08-22 13:37:11.000000000 -0700
@@ -5,6 +5,7 @@
   libsensors: Support upcoming sysfs path to i2c adapters
   sensord: Fix support of multiple chips (#2377)
               Add support for HID devices
+  pwmconfig: Sample all fan speeds before returning to full speed (#2380)
   fancontrol: Check that all referenced sysfs files exist
               Check that all devices match the configuration file
   pwmconfig: Exit immediately if not root
diff -Naur lm_sensors-3.1.2-old/prog/pwm/pwmconfig lm_sensors-3.1.2-new/prog/pwm/pwmconfig
--- lm_sensors-3.1.2-old/prog/pwm/pwmconfig	2009-09-16 08:57:53.000000000 -0700
+++ lm_sensors-3.1.2-new/prog/pwm/pwmconfig	2010-08-22 13:37:11.000000000 -0700
@@ -437,15 +437,19 @@
 		echo '^C received, restoring PWM and aborting...'
 		exit 1
 	fi
+
+	# Sample all current fan speeds at once, so that we can quickly
+	# disable PWM and return all fans to full speed
+	CURRENT_SPEEDS="`cat $GOODFAN`"
+	pwmdisable $i
+
 	let pwmactivecount=0
 	let count=1
 	for j in $GOODFAN
 	do
 		OS=`echo $SPEEDS | cut -d' ' -f$count`
-		# this will return the first field if there's only one (sysfs)
-		S=`cat $j | cut -d' ' -f2`
+		S=`echo $CURRENT_SPEEDS | cut -d' ' -f$count`
 		echo "  $j ... speed was $OS now $S"
-		pwmdisable $i
 		let threshold=2*$OS/3
 		if [ $S -lt $threshold ]
 		then
