diff -Naur lm_sensors-3.1.2-old/prog/detect/sensors-detect lm_sensors-3.1.2-new/prog/detect/sensors-detect
--- lm_sensors-3.1.2-old/prog/detect/sensors-detect	2010-06-20 07:48:06.000000000 -0700
+++ lm_sensors-3.1.2-new/prog/detect/sensors-detect	2010-06-20 07:48:32.000000000 -0700
@@ -608,7 +608,7 @@
 		i2c_detect => sub { andigilog_detect(@_, 1); },
 	}, {
 		name => "Andigilog aSC7621",
-		driver => "to-be-written",
+		driver => "asc7621",
 		i2c_addrs => [0x2c..0x2e],
 		i2c_detect => sub { andigilog_detect(@_, 2); },
 	}, {
