diff -Naur lm_sensors-2.10.5-old/kernel/include/Module.mk lm_sensors-2.10.5-new/kernel/include/Module.mk
--- lm_sensors-2.10.5-old/kernel/include/Module.mk	2007-10-24 19:31:33.000000000 -0700
+++ lm_sensors-2.10.5-new/kernel/include/Module.mk	2007-10-27 15:24:25.000000000 -0700
@@ -22,7 +22,7 @@
 KERNELINCLUDEDIR := $(MODULE_DIR)
 KERNELCHIPSDIR := kernel/chips
 
-KERNELINCLUDEFILES := $(MODULE_DIR)/i2c-dev.h $(MODULE_DIR)/sensors.h
+KERNELINCLUDEFILES := $(MODULE_DIR)/sensors.h
 
 $(KERNELINCLUDEDIR)/sensors.h: $(KERNELINCLUDEDIR)/sensors.h.template
 	cat $@.template > $@
