diff -Naur lm_sensors-3.0.1-old/lib/conf-lex.l lm_sensors-3.0.1-new/lib/conf-lex.l
--- lm_sensors-3.0.1-old/lib/conf-lex.l	2007-09-16 08:47:16.000000000 -0700
+++ lm_sensors-3.0.1-new/lib/conf-lex.l	2008-02-24 14:54:15.000000000 -0800
@@ -48,29 +48,29 @@
 
 %}
 
- /* Scanner for configuration files */
+/* Scanner for configuration files */
 
 %option nodefault
 %option noyywrap
 %option nounput
 
- /* All states are exclusive */
+/* All states are exclusive */
 
 %x MIDDLE
 %x STRING
 %x ERR
 
- /* Any whitespace-like character */
+/* Any whitespace-like character */
 
 BLANK		[ \f\t\v]
 
 IDCHAR		[[:alnum:]_]
 
- /* Note: `10', `10.4' and `.4' are valid, `10.' is not */
+/* Note: `10', `10.4' and `.4' are valid, `10.' is not */
 
 FLOAT   [[:digit:]]*\.?[[:digit:]]+
 
- /* Only positive whole numbers are recognized here */
+/* Only positive whole numbers are recognized here */
 
 NUM	0|([1-9][[:digit:]]*)
 
