diff -Naur lm_sensors-3.0.3-old/CHANGES lm_sensors-3.0.3-new/CHANGES
--- lm_sensors-3.0.3-old/CHANGES	2008-10-28 22:06:19.000000000 -0700
+++ lm_sensors-3.0.3-new/CHANGES	2008-10-28 22:07:25.000000000 -0700
@@ -5,6 +5,7 @@
   libsensors: Avoid namespace pollution
   sensors-detect: Fix detection of ADT7463 and LM96000
   fancontrol: Don't use named pipes when we don't need them (#2319)
+                  Fix detection of Intel 5000 series FB-DIMM AMB
   pwmconfig: Tell the user about gnuplot if it isn't installed
              Fix MINSTOP and MINSTART test functions
              Test MINSTOP before MINSTART
diff -Naur lm_sensors-3.0.3-old/prog/detect/sensors-detect lm_sensors-3.0.3-new/prog/detect/sensors-detect
--- lm_sensors-3.0.3-old/prog/detect/sensors-detect	2008-10-28 22:06:19.000000000 -0700
+++ lm_sensors-3.0.3-new/prog/detect/sensors-detect	2008-10-28 22:07:25.000000000 -0700
@@ -2146,7 +2146,7 @@
   },
   {
     name => "Intel AMB FB-DIMM thermal sensor",
-    driver => "to-be-written",
+    driver => "i5k_amb",
     detect => sub { intel_amb_detect(); },
   },
   {
@@ -5644,12 +5644,10 @@
 }
 
 # Returns: undef if not detected, (9) if detected.
-# the device 0xa620 should not be visible on host PCI bus, gateway
-# must be used
 sub intel_amb_detect
 {
-	if ((exists $pci_list{'8086:a620'}) ||
-	   (exists $pci_list{'8086:25f0'})) {
+	if ((exists $pci_list{'8086:25f0'}) ||	# Intel 5000
+	    (exists $pci_list{'8086:4030'})) {	# Intel 5400
 		return 9;
 	}
 	return;
