diff -Naur lm_sensors-3.0.0-old/CHANGES lm_sensors-3.0.0-new/CHANGES
--- lm_sensors-3.0.0-old/CHANGES	2007-12-20 13:28:08.000000000 -0800
+++ lm_sensors-3.0.0-new/CHANGES	2007-12-20 13:30:17.000000000 -0800
@@ -11,6 +11,7 @@
   sensord: Fix rrd support (#2276)
                   Add SMSC LPC47B367-NC detection (no sensors)
                   Reduce w83781d/lm78 I2C address probing range
+  sensors.conf.eg: Fix voltage value references (g520sm, lm80, pc87366)
 
 3.0.0-rc3 (2007-10-28)
   libsensors: Add a default configuration file
diff -Naur lm_sensors-3.0.0-old/etc/sensors.conf.eg lm_sensors-3.0.0-new/etc/sensors.conf.eg
--- lm_sensors-3.0.0-old/etc/sensors.conf.eg	2007-12-20 13:26:31.000000000 -0800
+++ lm_sensors-3.0.0-new/etc/sensors.conf.eg	2007-12-20 13:30:17.000000000 -0800
@@ -937,7 +937,7 @@
 #
 
   compute in2 (197/47)*@  ,  @/(197/47)
-  compute in4 (5*@)-(4*in0) , (@+4*in0)/5
+  compute in4 (5*@)-(4*in0_input) , (@+4*in0_input)/5
 
 #  set in0_min 4.8
 #  set in0_max 5.2
@@ -987,8 +987,8 @@
     compute in2 (22.1/30 + 1) * @ ,       @ / (22.1/30 + 1)
     compute in3 (2.8/1.9) * @,            @ * 1.9/2.8
     compute in4 (160/30.1 + 1) * @,       @ / (160/30.1 + 1)
-    compute in5 (160/35.7)*(@ - in0) + @, (@ + in0 * 160/35.7)/ (1 + 160/35.7)
-    compute in6 (36/16.2)*(@ - in0) + @,  (@ + in0 * 36/16.2) / (1 + 36/16.2)
+    compute in5 (160/35.7)*(@ - in0_input) + @, (@ + in0_input * 160/35.7)/ (1 + 160/35.7)
+    compute in6 (36/16.2)*(@ - in0_input) + @,  (@ + in0_input * 36/16.2) / (1 + 36/16.2)
 
 #    set in0_min 5 * 0.95
 #    set in0_max 5 * 1.05
@@ -2064,12 +2064,12 @@
 # We also assume that you have a working in10 (which is forcibly enabled by
 # default). If not, use 3.3 instead, but you lose accuracy.
 
-   compute temp4  3435 / (` (1 / (1 - @ / in10) - 1) + 3435 / 298.15) - 273.15, \
-                  in10 * (1 - 1 / (1 + (^ (3435 / (273.15 + @) - 3435 / 298.15))))
-   compute temp5  3435 / (` (1 / (1 - @ / in10) - 1) + 3435 / 298.15) - 273.15, \
-                  in10 * (1 - 1 / (1 + (^ (3435 / (273.15 + @) - 3435 / 298.15))))
-   compute temp6  3435 / (` (1 / (1 - @ / in10) - 1) + 3435 / 298.15) - 273.15, \
-                  in10 * (1 - 1 / (1 + (^ (3435 / (273.15 + @) - 3435 / 298.15))))
+   compute temp4  3435 / (` (1 / (1 - @ / in10_input) - 1) + 3435 / 298.15) - 273.15, \
+                  in10_input * (1 - 1 / (1 + (^ (3435 / (273.15 + @) - 3435 / 298.15))))
+   compute temp5  3435 / (` (1 / (1 - @ / in10_input) - 1) + 3435 / 298.15) - 273.15, \
+                  in10_input * (1 - 1 / (1 + (^ (3435 / (273.15 + @) - 3435 / 298.15))))
+   compute temp6  3435 / (` (1 / (1 - @ / in10_input) - 1) + 3435 / 298.15) - 273.15, \
+                  in10_input * (1 - 1 / (1 + (^ (3435 / (273.15 + @) - 3435 / 298.15))))
 
 #   set temp4_min    0
 #   set temp4_max   70
