diff -Naur NVIDIA-Linux-x86_64-173.08-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-173.08-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-173.08-pkg2-old/usr/src/nv/conftest.sh	2008-04-02 09:25:20.000000000 -0700
+++ NVIDIA-Linux-x86_64-173.08-pkg2-new/usr/src/nv/conftest.sh	2008-05-07 11:41:38.000000000 -0700
@@ -963,7 +963,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
