diff -Naur NVIDIA-Linux-x86_64-180.08-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-180.08-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-180.08-pkg2-old/usr/src/nv/conftest.sh	2008-11-15 11:02:17.000000000 -0800
+++ NVIDIA-Linux-x86_64-180.08-pkg2-new/usr/src/nv/conftest.sh	2008-11-17 22:13:24.000000000 -0800
@@ -1143,7 +1143,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
