diff -Naur xine-lib-1.1.16.1-old/configure.ac xine-lib-1.1.16.1-new/configure.ac
--- xine-lib-1.1.16.1-old/configure.ac	2009-01-19 11:21:31.000000000 -0800
+++ xine-lib-1.1.16.1-new/configure.ac	2009-01-19 11:22:12.000000000 -0800
@@ -1795,7 +1795,7 @@
 ], bf_lsbf=0, AC_MSG_ERROR([unsupported bitfield ordering]))
   ],
   [case "$host" in
-     *-*-mingw32* | *-*-cygwin*)
+     i?86-* | x86_64-* | *-*-mingw32* | *-*-cygwin*)
        bf_lsbf=1
      ;;
      *)
