diff -Naur xine-lib-1.1.15-old/src/combined/ffmpeg/ff_audio_decoder.c xine-lib-1.1.15-new/src/combined/ffmpeg/ff_audio_decoder.c
--- xine-lib-1.1.15-old/src/combined/ffmpeg/ff_audio_decoder.c	2008-07-15 16:13:03.000000000 -0700
+++ xine-lib-1.1.15-new/src/combined/ffmpeg/ff_audio_decoder.c	2008-11-02 11:52:12.000000000 -0800
@@ -322,7 +322,7 @@
 
     if (!this->output_open) {
       if (!this->audio_bits || !this->audio_sample_rate || !this->audio_channels) {
-        avcodec_decode_audio (this->context,
+        avcodec_decode_audio2 (this->context,
                               (int16_t *)this->decode_buffer,
                               &decode_buffer_size,
                               &this->buf[0],
diff -Naur xine-lib-1.1.15-old/src/combined/ffmpeg/ffmpeg_decoder.h xine-lib-1.1.15-new/src/combined/ffmpeg/ffmpeg_decoder.h
--- xine-lib-1.1.15-old/src/combined/ffmpeg/ffmpeg_decoder.h	2008-05-19 07:15:42.000000000 -0700
+++ xine-lib-1.1.15-new/src/combined/ffmpeg/ffmpeg_decoder.h	2008-11-02 11:52:12.000000000 -0800
@@ -33,6 +33,10 @@
 #  include "../../libffmpeg/libavcodec/avcodec.h"
 #endif
 
+#if LIBAVCODEC_VERSION_MAJOR > 51
+#define bits_per_sample bits_per_coded_sample
+#endif
+
 typedef struct ff_codec_s {
   uint32_t          type;
   enum CodecID      id;
