diff -Naur xine-lib-1.1.1-old/src/xine-engine/alphablend.c xine-lib-1.1.1-new/src/xine-engine/alphablend.c
--- xine-lib-1.1.1-old/src/xine-engine/alphablend.c	2005-09-26 19:05:31.000000000 -0700
+++ xine-lib-1.1.1-new/src/xine-engine/alphablend.c	2006-08-14 10:25:09.000000000 -0700
@@ -2019,7 +2019,7 @@
     dst = dst_y;
 
     for (x = 0; x < src_width;) {
-      int len = (x + rle->len > hili_right) ? hili_right - x : rle->len;
+      int len = (x + rle->len > clip_right) ? clip_right - x : rle->len;
       int clipped = (y < clip_top);
 
       if (len > 0) {
