diff -Naur xine-lib-11552-old/src/combined/ffmpeg/Makefile.am xine-lib-11552-new/src/combined/ffmpeg/Makefile.am
--- xine-lib-11552-old/src/combined/ffmpeg/Makefile.am	2010-07-26 14:10:30.000000000 -0700
+++ xine-lib-11552-new/src/combined/ffmpeg/Makefile.am	2010-07-26 19:27:11.000000000 -0700
@@ -28,7 +28,7 @@
 
 # Generation of ffmpeg->xine codec mapping lists (see xine_*.list).
 
-AV_CPP = $(CPP) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS)
+AV_CPP = $(CPP) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(FFMPEG_CFLAGS)
 
 # Extract some CODEC_ID_* from avcodec.h. Requires some sed mangling.
 avcodec_audio.list: AV_CODECS:=/CODEC_ID_PCM_S16LE/,/CODEC_ID_DVD_SUBTITLE/
