diff -Naur xine-lib-1.1.8-old/src/input/vcd/libcdio/_cdio_linux.c xine-lib-1.1.8-new/src/input/vcd/libcdio/_cdio_linux.c
--- xine-lib-1.1.8-old/src/input/vcd/libcdio/_cdio_linux.c	2007-08-18 16:36:48.000000000 -0700
+++ xine-lib-1.1.8-new/src/input/vcd/libcdio/_cdio_linux.c	2007-12-29 21:51:54.000000000 -0800
@@ -59,6 +59,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <mntent.h>
+#include <limits.h>
 
 #include <linux/cdrom.h>
 #include <scsi/scsi.h>
