diff -Naur mythplugins-0.23-27077-old/mythvideo/mtd/dvdprobe.cpp mythplugins-0.23-27077-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-0.23-27077-old/mythvideo/mtd/dvdprobe.cpp	2010-11-03 11:15:57.000000000 -0700
+++ mythplugins-0.23-27077-new/mythvideo/mtd/dvdprobe.cpp	2010-11-03 11:15:58.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
