diff -Naur alsa-driver-1.0.19-old/alsa-kernel/pci/hda/patch_analog.c alsa-driver-1.0.19-new/alsa-kernel/pci/hda/patch_analog.c
--- alsa-driver-1.0.19-old/alsa-kernel/pci/hda/patch_analog.c	2009-01-19 03:08:58.000000000 -0800
+++ alsa-driver-1.0.19-new/alsa-kernel/pci/hda/patch_analog.c	2009-02-03 13:41:04.000000000 -0800
@@ -2925,6 +2925,7 @@
 	SND_PCI_QUIRK(0x1043, 0x81f6, "Asus M2N-SLI", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8277, "Asus P5K-E/WIFI-AP", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8311, "Asus P5Q-Premium/Pro", AD1988_6STACK_DIG),
+	SND_PCI_QUIRK(0x1043, 0x8345, "Asus M3N78-VM", AD1988_6STACK_DIG),
 	{}
 };
 
