diff -Naur myththemes-addons-0.21-17237-old/configure myththemes-addons-0.21-17237-new/configure
--- myththemes-addons-0.21-17237-old/configure	2008-05-04 09:46:26.000000000 -0700
+++ myththemes-addons-0.21-17237-new/configure	2008-05-04 09:47:13.000000000 -0700
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
