diff -Naur myththemes-addons-0.21-21384-old/configure myththemes-addons-0.21-21384-new/configure
--- myththemes-addons-0.21-21384-old/configure	2009-08-20 20:13:48.000000000 -0700
+++ myththemes-addons-0.21-21384-new/configure	2009-08-20 20:15:38.000000000 -0700
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
