diff -Naur myththemes-addons-0.21-19479-old/configure myththemes-addons-0.21-19479-new/configure
--- myththemes-addons-0.21-19479-old/configure	2008-12-28 22:44:45.000000000 -0800
+++ myththemes-addons-0.21-19479-new/configure	2008-12-28 22:45:43.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
