diff -Naur myththemes-addons-0.21-19900-old/configure myththemes-addons-0.21-19900-new/configure
--- myththemes-addons-0.21-19900-old/configure	2009-02-01 21:44:34.000000000 -0800
+++ myththemes-addons-0.21-19900-new/configure	2009-02-01 21:47:37.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
