diff -Naur fontconfig-2.6.0-old/fc-case/Makefile.am fontconfig-2.6.0-new/fc-case/Makefile.am
--- fontconfig-2.6.0-old/fc-case/Makefile.am	2008-01-02 08:44:39.000000000 -0800
+++ fontconfig-2.6.0-new/fc-case/Makefile.am	2008-07-24 20:50:45.000000000 -0700
@@ -36,6 +36,8 @@
 
 noinst_MANS=fc-case.man
 
+noinst_HEADERS=$(TARG)
+
 fc_case_SRCS= \
 	fc-case.c \
 	fccaseint.h \
@@ -52,7 +54,7 @@
 
 ALIAS_FILES = fcalias.h fcaliastail.h
 
-BUILT_SOURCES = $(ALIAS_FILES) $(TARG)
+BUILT_SOURCES = $(ALIAS_FILES)
 
 $(ALIAS_FILES): 
 	touch $(ALIAS_FILES)
