diff -Naur fontconfig-2.4.2-old/doc/Makefile.am fontconfig-2.4.2-new/doc/Makefile.am
--- fontconfig-2.4.2-old/doc/Makefile.am	2006-12-02 14:14:42.000000000 -0800
+++ fontconfig-2.4.2-new/doc/Makefile.am	2006-12-09 20:30:36.000000000 -0800
@@ -21,7 +21,15 @@
 # TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 # PERFORMANCE OF THIS SOFTWARE.
 
+CPP = $(CPP_FOR_BUILD)
 CC = @CC_FOR_BUILD@
+CXX = $(CXX_FOR_BUILD)
+AR = $(AR_FOR_BUILD)
+AS = $(AS_FOR_BUILD)
+CPPFLAGS = $(CXXFLAGS_FOR_BUILD)
+CFLAGS = $(CFLAGS_FOR_BUILD)
+CXXFLAGS = $(CXXFLAGS_FOR_BUILD)
+LDFLAGS = $(LDFLAGS_FOR_BUILD)
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.4.2-old/fc-arch/Makefile.am fontconfig-2.4.2-new/fc-arch/Makefile.am
--- fontconfig-2.4.2-old/fc-arch/Makefile.am	2006-09-05 02:05:07.000000000 -0700
+++ fontconfig-2.4.2-new/fc-arch/Makefile.am	2006-12-09 20:30:36.000000000 -0800
@@ -22,7 +22,15 @@
 # PERFORMANCE OF THIS SOFTWARE.
 #
 
+CPP = $(CPP_FOR_BUILD)
 CC = @CC_FOR_BUILD@
+CXX = $(CXX_FOR_BUILD)
+AR = $(AR_FOR_BUILD)
+AS = $(AS_FOR_BUILD)
+CPPFLAGS = $(CXXFLAGS_FOR_BUILD)
+CFLAGS = $(CFLAGS_FOR_BUILD)
+CXXFLAGS = $(CXXFLAGS_FOR_BUILD)
+LDFLAGS = $(LDFLAGS_FOR_BUILD)
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.4.2-old/fc-case/Makefile.am fontconfig-2.4.2-new/fc-case/Makefile.am
--- fontconfig-2.4.2-old/fc-case/Makefile.am	2006-09-05 02:04:31.000000000 -0700
+++ fontconfig-2.4.2-new/fc-case/Makefile.am	2006-12-09 20:30:36.000000000 -0800
@@ -22,7 +22,15 @@
 # PERFORMANCE OF THIS SOFTWARE.
 #
 
+CPP = $(CPP_FOR_BUILD)
 CC = @CC_FOR_BUILD@
+CXX = $(CXX_FOR_BUILD)
+AR = $(AR_FOR_BUILD)
+AS = $(AS_FOR_BUILD)
+CPPFLAGS = $(CXXFLAGS_FOR_BUILD)
+CFLAGS = $(CFLAGS_FOR_BUILD)
+CXXFLAGS = $(CXXFLAGS_FOR_BUILD)
+LDFLAGS = $(LDFLAGS_FOR_BUILD)
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.4.2-old/fc-glyphname/Makefile.am fontconfig-2.4.2-new/fc-glyphname/Makefile.am
--- fontconfig-2.4.2-old/fc-glyphname/Makefile.am	2006-09-05 02:04:47.000000000 -0700
+++ fontconfig-2.4.2-new/fc-glyphname/Makefile.am	2006-12-09 20:30:36.000000000 -0800
@@ -22,7 +22,15 @@
 # PERFORMANCE OF THIS SOFTWARE.
 #
 
+CPP = $(CPP_FOR_BUILD)
 CC = @CC_FOR_BUILD@
+CXX = $(CXX_FOR_BUILD)
+AR = $(AR_FOR_BUILD)
+AS = $(AS_FOR_BUILD)
+CPPFLAGS = $(CXXFLAGS_FOR_BUILD)
+CFLAGS = $(CFLAGS_FOR_BUILD)
+CXXFLAGS = $(CXXFLAGS_FOR_BUILD)
+LDFLAGS = $(LDFLAGS_FOR_BUILD)
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.4.2-old/fc-lang/Makefile.am fontconfig-2.4.2-new/fc-lang/Makefile.am
--- fontconfig-2.4.2-old/fc-lang/Makefile.am	2006-09-05 02:03:45.000000000 -0700
+++ fontconfig-2.4.2-new/fc-lang/Makefile.am	2006-12-09 20:30:36.000000000 -0800
@@ -22,7 +22,15 @@
 # PERFORMANCE OF THIS SOFTWARE.
 #
 
+CPP = $(CPP_FOR_BUILD)
 CC = @CC_FOR_BUILD@
+CXX = $(CXX_FOR_BUILD)
+AR = $(AR_FOR_BUILD)
+AS = $(AS_FOR_BUILD)
+CPPFLAGS = $(CXXFLAGS_FOR_BUILD)
+CFLAGS = $(CFLAGS_FOR_BUILD)
+CXXFLAGS = $(CXXFLAGS_FOR_BUILD)
+LDFLAGS = $(LDFLAGS_FOR_BUILD)
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
