diff -Naur gtk+-2.22.1-old/gdk/x11/gdkdnd-x11.c gtk+-2.22.1-new/gdk/x11/gdkdnd-x11.c
--- gtk+-2.22.1-old/gdk/x11/gdkdnd-x11.c	2010-11-15 04:13:09.000000000 -0800
+++ gtk+-2.22.1-new/gdk/x11/gdkdnd-x11.c	2010-11-15 21:18:30.000000000 -0800
@@ -30,7 +30,9 @@
 #include <X11/Xutil.h>
 #include <X11/Xatom.h>
 #include <X11/extensions/shape.h>
+#ifdef HAVE_XCOMPOSITE
 #include <X11/extensions/Xcomposite.h>
+#endif
 
 #include <string.h>
 
