diff -Naur linux-2.6.25-rc6-old/include/linux/time.h linux-2.6.25-rc6-new/include/linux/time.h
--- linux-2.6.25-rc6-old/include/linux/time.h	2008-03-16 16:32:14.000000000 -0700
+++ linux-2.6.25-rc6-new/include/linux/time.h	2008-03-19 15:48:35.000000000 -0700
@@ -170,7 +170,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
