diff -Naur zoneinfo-2008d-old/Makefile zoneinfo-2008d-new/Makefile
--- zoneinfo-2008d-old/Makefile	2008-03-05 08:58:23.000000000 -0800
+++ zoneinfo-2008d-new/Makefile	2008-07-08 18:57:31.000000000 -0700
@@ -344,16 +344,14 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 # We use the system's logwtmp in preference to ours if available.
 
 date:		$(DATEOBJS)
-		ar r ,lib.a logwtmp.o
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib ,lib.a ; fi
+		$(AR) r ,lib.a logwtmp.o
+		$(RANLIB) ,lib.a
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
 			$(LDLIBS) -lc ,lib.a -o $@
 		rm -f ,lib.a
