diff -Naur zoneinfo-2011d-old/Makefile zoneinfo-2011d-new/Makefile
--- zoneinfo-2011d-old/Makefile	2010-12-16 09:11:21.000000000 -0800
+++ zoneinfo-2011d-new/Makefile	2011-03-15 22:40:53.000000000 -0700
@@ -347,9 +347,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
