diff -Naur zoneinfo-2009j-old/Makefile zoneinfo-2009j-new/Makefile
--- zoneinfo-2009j-old/Makefile	2009-05-17 11:44:05.000000000 -0700
+++ zoneinfo-2009j-new/Makefile	2009-06-20 06:34:25.000000000 -0700
@@ -347,9 +347,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
