diff -Naur zoneinfo-2012e-old/Makefile zoneinfo-2012e-new/Makefile
--- zoneinfo-2012e-old/Makefile	2012-08-02 20:44:55.000000000 -0700
+++ zoneinfo-2012e-new/Makefile	2012-08-03 05:34:52.000000000 -0700
@@ -366,9 +366,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
