diff -Naur perl-5.10.0-old/ext/Errno/Errno_pm.PL perl-5.10.0-new/ext/Errno/Errno_pm.PL
--- perl-5.10.0-old/ext/Errno/Errno_pm.PL	2008-02-22 10:10:44.000000000 -0800
+++ perl-5.10.0-new/ext/Errno/Errno_pm.PL	2008-02-22 10:12:57.000000000 -0800
@@ -340,10 +340,6 @@
 use Config;
 use strict;
 
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$Config{'archname'}-$Config{'osvers'}" or
-	die "Errno architecture ($Config{'archname'}-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
-
 \$VERSION = "$VERSION";
 \$VERSION = eval \$VERSION;
 \@ISA = qw(Exporter);
