diff -Naur perl-5.8.8-old/ext/DynaLoader/DynaLoader_pm.PL perl-5.8.8-new/ext/DynaLoader/DynaLoader_pm.PL
--- perl-5.8.8-old/ext/DynaLoader/DynaLoader_pm.PL	2004-11-22 07:32:13.000000000 -0800
+++ perl-5.8.8-new/ext/DynaLoader/DynaLoader_pm.PL	2008-02-16 23:08:10.000000000 -0800
@@ -260,7 +260,7 @@
 		       "(auto/$modpname/$modfname.$dl_dlext)\n")
 	if $dl_debug;
 
-    foreach (@INC) {
+    foreach (reverse(@INC)) {
 	chop($_ = VMS::Filespec::unixpath($_)) if $Is_VMS;
 	my $dir;
 	if ($Is_MacOS) {
