diff -Naur perl-5.10.1-old/ext/Errno/Errno_pm.PL perl-5.10.1-new/ext/Errno/Errno_pm.PL
--- perl-5.10.1-old/ext/Errno/Errno_pm.PL	2009-08-24 10:10:03.000000000 -0700
+++ perl-5.10.1-new/ext/Errno/Errno_pm.PL	2009-08-24 10:11:33.000000000 -0700
@@ -344,10 +344,6 @@
 use Config;
 use strict;
 
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$Config{'archname'}-$Config{'osvers'}" or
-	die "Errno architecture ($Config{'archname'}-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
-
 \$VERSION = "$VERSION";
 \$VERSION = eval \$VERSION;
 \@ISA = qw(Exporter);
