diff -Naur perl-5.8.8-old/Configure perl-5.8.8-new/Configure
--- perl-5.8.8-old/Configure	2006-01-08 06:51:03.000000000 -0800
+++ perl-5.8.8-new/Configure	2008-02-14 19:58:12.000000000 -0800
@@ -1244,8 +1244,8 @@
 archname=''
 : Possible local include directories to search.
 : Set locincpth to "" in a hint file to defeat local include searches.
-locincpth="/usr/local/include /opt/local/include /usr/gnu/include"
-locincpth="$locincpth /opt/gnu/include /usr/GNU/include /opt/GNU/include"
+locincpth="$DESTDIR/usr/local/include $DESTDIR/opt/local/include $DESTDIR/usr/gnu/include"
+locincpth="$locincpth $DESTDIR/opt/gnu/include $DESTDIR/usr/GNU/include $DESTDIR/opt/GNU/include"
 :
 : no include file wanted by default
 inclwanted=''
@@ -1253,16 +1253,16 @@
 groupstype=''
 libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
-xlibpth='/usr/lib/386 /lib/386'
+xlibpth='$DESTDIR/usr/lib/386 $DESTDIR/lib/386'
 : Possible local library directories to search.
-loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
-loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
+loclibpth="$DESTDIR/usr/local/lib $DESTDIR/opt/local/lib $DESTDIR/usr/gnu/lib"
+loclibpth="$loclibpth $DESTDIR/opt/gnu/lib $DESTDIR/usr/GNU/lib $DESTDIR/opt/GNU/lib"
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
-test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
-test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
+glibpth="$DESTDIR/lib $DESTDIR/usr/lib $xlibpth"
+glibpth="$glibpth $DESTDIR/usr/ccs/lib $DESTDIR/usr/ucblib $DESTDIR/usr/local/lib"
+test -f $DESTDIR/usr/shlib/libc.so && glibpth="$DESTDIR/usr/shlib $glibpth"
+test -f $DESTDIR/shlib/libc.so     && glibpth="$DESTDIR/shlib $glibpth"
 
 : Private path used by Configure to find libraries.  Its value
 : is prepended to libpth. This variable takes care of special
