diff -Naur perl-5.12.1-old/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm perl-5.12.1-new/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
--- perl-5.12.1-old/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2010-04-26 01:08:34.000000000 -0700
+++ perl-5.12.1-new/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2010-07-19 21:40:56.000000000 -0700
@@ -295,7 +295,7 @@
 sub perl_inc {
   my $self = shift;
 
-  $self->perl_src() || File::Spec->catdir($self->{config}{archlibexp},"CORE");
+  $self->perl_src() || File::Spec->catdir($ENV{DESTDIR},$self->{config}{archlibexp},"CORE");
 }
 
 sub DESTROY {
