diff -Naur mythplugins-0.24-27113-old/configure mythplugins-0.24-27113-new/configure
--- mythplugins-0.24-27113-old/configure	2010-11-07 06:01:47.000000000 -0800
+++ mythplugins-0.24-27113-new/configure	2010-11-07 06:02:58.000000000 -0800
@@ -343,10 +343,15 @@
 
 check_py_lib(){
     log check_py_lib "$@"
+if test "x${PYTHONXCPREFIX}" = "x" ; then
+   py_prefix=${prefix}
+else
+   py_prefix=${PYTHONXCPREFIX}
+fi
     check_cmd $python <<EOF
 from distutils.sysconfig import get_python_lib
 import sys
-sys.path.append(get_python_lib(prefix='${prefix}'))
+sys.path.append(get_python_lib(prefix='${py_prefix}'))
 try:
     import ${1}
 except:
