diff -Naur busybox-1.12.2-old/networking/tftp.c busybox-1.12.2-new/networking/tftp.c
--- busybox-1.12.2-old/networking/tftp.c	2008-11-12 09:41:29.000000000 -0800
+++ busybox-1.12.2-new/networking/tftp.c	2008-11-12 09:42:06.000000000 -0800
@@ -27,7 +27,7 @@
 
 #define TFTP_BLKSIZE_DEFAULT       512  /* according to RFC 1350, don't change */
 #define TFTP_BLKSIZE_DEFAULT_STR "512"
-#define TFTP_TIMEOUT_MS             50
+#define TFTP_TIMEOUT_MS            250
 #define TFTP_MAXTIMEOUT_MS        2000
 #define TFTP_NUM_RETRIES            12  /* number of backed-off retries */
 
