diff -Naur busybox-1.1.2-old/networking/udhcp/options.c busybox-1.1.2-new/networking/udhcp/options.c
--- busybox-1.1.2-old/networking/udhcp/options.c	2006-03-22 13:16:19.000000000 -0800
+++ busybox-1.1.2-new/networking/udhcp/options.c	2006-04-20 17:24:52.000000000 -0700
@@ -21,7 +21,7 @@
 	{"timesvr",	OPTION_IP | OPTION_LIST,		0x04},
 	{"namesvr",	OPTION_IP | OPTION_LIST,		0x05},
 	{"dns",		OPTION_IP | OPTION_LIST | OPTION_REQ,	0x06},
-	{"logsvr",	OPTION_IP | OPTION_LIST,		0x07},
+	{"logsvr",	OPTION_IP | OPTION_LIST | OPTION_REQ,	0x07},
 	{"cookiesvr",	OPTION_IP | OPTION_LIST,		0x08},
 	{"lprsvr",	OPTION_IP | OPTION_LIST,		0x09},
 	{"hostname",	OPTION_STRING | OPTION_REQ,		0x0c},
