diff -Naur busybox-1.2.2.1-old/networking/udhcp/options.c busybox-1.2.2.1-new/networking/udhcp/options.c
--- busybox-1.2.2.1-old/networking/udhcp/options.c	2006-06-30 15:42:02.000000000 -0700
+++ busybox-1.2.2.1-new/networking/udhcp/options.c	2007-02-18 09:35:39.000000000 -0800
@@ -43,6 +43,7 @@
 	{"message",	OPTION_STRING,				0x38},
 	{"tftp",	OPTION_STRING,				0x42},
 	{"bootfile",	OPTION_STRING,				0x43},
+	{"tz_tzdb",	OPTION_STRING | OPTION_REQ,		0xFE},
 	{"",		0x00,				0x00}
 };
 
