diff -Naur busybox-1.01-old/networking/udhcp/options.c busybox-1.01-new/networking/udhcp/options.c
--- busybox-1.01-old/networking/udhcp/options.c	2005-08-16 18:29:10.000000000 -0700
+++ busybox-1.01-new/networking/udhcp/options.c	2005-09-04 07:03:44.000000000 -0700
@@ -21,7 +21,7 @@
 	{"timesvr",	OPTION_IP | OPTION_LIST,		0x04},
 	{"namesvr",	OPTION_IP | OPTION_LIST,		0x05},
 	{"dns",		OPTION_IP | OPTION_LIST | OPTION_REQ,	0x06},
-	{"logsvr",	OPTION_IP | OPTION_LIST,		0x07},
+	{"logsvr",	OPTION_IP | OPTION_LIST | OPTION_REQ,	0x07},
 	{"cookiesvr",	OPTION_IP | OPTION_LIST,		0x08},
 	{"lprsvr",	OPTION_IP | OPTION_LIST,		0x09},
 	{"hostname",	OPTION_STRING | OPTION_REQ,		0x0c},
@@ -32,7 +32,7 @@
 	{"ipttl",	OPTION_U8,				0x17},
 	{"mtu",		OPTION_U16,				0x1a},
 	{"broadcast",	OPTION_IP | OPTION_REQ,			0x1c},
-	{"ntpsrv",	OPTION_IP | OPTION_LIST,		0x2a},
+	{"ntpsrv",	OPTION_IP | OPTION_LIST | OPTION_REQ,	0x2a},
 	{"wins",	OPTION_IP | OPTION_LIST,		0x2c},
 	{"requestip",	OPTION_IP,				0x32},
 	{"lease",	OPTION_U32,				0x33},
