diff -Naur mythtv-0.20-old/bindings/perl/MythTV/Channel.pm mythtv-0.20-new/bindings/perl/MythTV/Channel.pm
--- mythtv-0.20-old/bindings/perl/MythTV/Channel.pm	2006-04-16 01:05:27.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV/Channel.pm	2006-09-12 06:19:18.000000000 -0700
@@ -3,8 +3,8 @@
 #
 # Object containing info about a particular MythTV channel.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV/Channel.pm $
-# @date      $Date: 2006-04-16 04:05:27 -0400 (Sun, 16 Apr 2006) $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV/Channel.pm $
+# @date      $Date: 2006-04-16 01:05:27 -0700 (Sun, 16 Apr 2006) $
 # @version   $Revision: 9725 $
 # @author    $Author: xris $
 #
diff -Naur mythtv-0.20-old/bindings/perl/MythTV/Program.pm mythtv-0.20-new/bindings/perl/MythTV/Program.pm
--- mythtv-0.20-old/bindings/perl/MythTV/Program.pm	2006-04-16 01:05:27.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV/Program.pm	2006-09-12 06:19:18.000000000 -0700
@@ -3,8 +3,8 @@
 #
 # Object containing info about a particular MythTV program.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV/Program.pm $
-# @date      $Date: 2006-04-16 04:05:27 -0400 (Sun, 16 Apr 2006) $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV/Program.pm $
+# @date      $Date: 2006-04-16 01:05:27 -0700 (Sun, 16 Apr 2006) $
 # @version   $Revision: 9725 $
 # @author    $Author: xris $
 #
diff -Naur mythtv-0.20-old/bindings/perl/MythTV/Recording.pm mythtv-0.20-new/bindings/perl/MythTV/Recording.pm
--- mythtv-0.20-old/bindings/perl/MythTV/Recording.pm	2006-07-19 19:20:17.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV/Recording.pm	2006-09-12 06:19:18.000000000 -0700
@@ -3,8 +3,8 @@
 #
 # Object containing info about a particular MythTV recording.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV/Recording.pm $
-# @date      $Date: 2006-07-19 22:20:17 -0400 (Wed, 19 Jul 2006) $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV/Recording.pm $
+# @date      $Date: 2006-07-19 19:20:17 -0700 (Wed, 19 Jul 2006) $
 # @version   $Revision: 10602 $
 # @author    $Author: xris $
 #
diff -Naur mythtv-0.20-old/bindings/perl/MythTV.pm mythtv-0.20-new/bindings/perl/MythTV.pm
--- mythtv-0.20-old/bindings/perl/MythTV.pm	2006-07-04 13:30:37.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV.pm	2006-09-12 06:19:18.000000000 -0700
@@ -1,8 +1,8 @@
 #
 # MythTV bindings for perl.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV.pm $
-# @date      $Date: 2006-07-04 16:30:37 -0400 (Tue, 04 Jul 2006) $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV.pm $
+# @date      $Date: 2006-07-04 13:30:37 -0700 (Tue, 04 Jul 2006) $
 # @version   $Revision: 10390 $
 # @author    $Author: xris $
 #
diff -Naur mythtv-0.20-old/configfiles/lircrc.native.example.mceusb2 mythtv-0.20-new/configfiles/lircrc.native.example.mceusb2
--- mythtv-0.20-old/configfiles/lircrc.native.example.mceusb2	2006-06-11 22:18:38.000000000 -0700
+++ mythtv-0.20-new/configfiles/lircrc.native.example.mceusb2	2006-09-12 06:18:09.000000000 -0700
@@ -2,7 +2,7 @@
 # lircrc config file for the Microsoft Media Center Edition Remote, model 1039
 #
 # @url       $URL$
-# @date      $Date: 2006-06-12 01:18:38 -0400 (Mon, 12 Jun 2006) $
+# @date      $Date: 2006-06-11 22:18:38 -0700 (Sun, 11 Jun 2006) $
 # @version   $Revision: 10187 $
 # @author    $Author: xris $
 #
diff -Naur mythtv-0.20-old/contrib/mythrename.pl mythtv-0.20-new/contrib/mythrename.pl
--- mythtv-0.20-old/contrib/mythrename.pl	2006-02-27 23:40:08.000000000 -0800
+++ mythtv-0.20-new/contrib/mythrename.pl	2006-09-12 06:19:21.000000000 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/perl -w
 #
-# $Date: 2006-02-28 02:40:08 -0500 (Tue, 28 Feb 2006) $
+# $Date: 2006-02-27 23:40:08 -0800 (Mon, 27 Feb 2006) $
 # $Revision: 9198 $
 # $Author: xris $
 #
diff -Naur mythtv-0.20-old/contrib/optimize_mythdb.pl mythtv-0.20-new/contrib/optimize_mythdb.pl
--- mythtv-0.20-old/contrib/optimize_mythdb.pl	2006-08-05 14:51:46.000000000 -0700
+++ mythtv-0.20-new/contrib/optimize_mythdb.pl	2006-09-12 06:19:21.000000000 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/perl -w
 #
-# $Date: 2006-08-05 17:51:46 -0400 (Sat, 05 Aug 2006) $
+# $Date: 2006-08-05 14:51:46 -0700 (Sat, 05 Aug 2006) $
 # $Revision: 10685 $
 # $Author: xris $
 #
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.cpp mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.cpp	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -12,7 +12,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: AAFilter.cpp 9996 2006-05-23 17:40:36Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.h mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.h
--- mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.h	2004-11-13 14:29:45.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.h	2006-09-12 06:18:33.000000000 -0700
@@ -13,7 +13,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2004-11-13 17:29:45 -0500 (Sat, 13 Nov 2004) $
+// Last changed  : $Date: 2004-11-13 14:29:45 -0800 (Sat, 13 Nov 2004) $
 // File revision : $Revision: 4714 $
 //
 // $Id: AAFilter.h 4714 2004-11-13 22:29:45Z ijr $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/BPMDetect.h mythtv-0.20-new/libs/libmythsoundtouch/BPMDetect.h
--- mythtv-0.20-old/libs/libmythsoundtouch/BPMDetect.h	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/BPMDetect.h	2006-09-12 06:18:33.000000000 -0700
@@ -26,7 +26,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: BPMDetect.h 9996 2006-05-23 17:40:36Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect.h mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect.h
--- mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect.h	2006-03-16 02:00:28.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect.h	2006-09-12 06:18:33.000000000 -0700
@@ -12,7 +12,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-03-16 05:00:28 -0500 (Thu, 16 Mar 2006) $
+// Last changed  : $Date: 2006-03-16 02:00:28 -0800 (Thu, 16 Mar 2006) $
 // File revision : $Revision: 9376 $
 //
 // $Id: cpu_detect.h 9376 2006-03-16 10:00:28Z ijr $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -12,7 +12,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: cpu_detect_x86_gcc.cpp 9996 2006-05-23 17:40:36Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.cpp mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.cpp	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -15,7 +15,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: FIFOSampleBuffer.cpp 9996 2006-05-23 17:40:36Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.h mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.h
--- mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.h	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.h	2006-09-12 06:18:33.000000000 -0700
@@ -15,7 +15,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: FIFOSampleBuffer.h 9996 2006-05-23 17:40:36Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/FIFOSamplePipe.h mythtv-0.20-new/libs/libmythsoundtouch/FIFOSamplePipe.h
--- mythtv-0.20-old/libs/libmythsoundtouch/FIFOSamplePipe.h	2005-05-23 13:39:12.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIFOSamplePipe.h	2006-09-12 06:18:33.000000000 -0700
@@ -17,7 +17,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2005-05-23 16:39:12 -0400 (Mon, 23 May 2005) $
+// Last changed  : $Date: 2005-05-23 13:39:12 -0700 (Mon, 23 May 2005) $
 // File revision : $Revision: 6468 $
 //
 // $Id: FIFOSamplePipe.h 6468 2005-05-23 20:39:12Z ijr $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.cpp mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.cpp	2006-03-31 22:19:12.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -11,7 +11,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-04-01 01:19:12 -0500 (Sat, 01 Apr 2006) $
+// Last changed  : $Date: 2006-03-31 22:19:12 -0800 (Fri, 31 Mar 2006) $
 // File revision : $Revision: 9588 $
 //
 // $Id: FIRFilter.cpp 9588 2006-04-01 06:19:12Z ijr $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.h mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.h
--- mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.h	2004-11-13 14:29:45.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.h	2006-09-12 06:18:33.000000000 -0700
@@ -11,7 +11,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2004-11-13 17:29:45 -0500 (Sat, 13 Nov 2004) $
+// Last changed  : $Date: 2004-11-13 14:29:45 -0800 (Sat, 13 Nov 2004) $
 // File revision : $Revision: 4714 $
 //
 // $Id: FIRFilter.h 4714 2004-11-13 22:29:45Z ijr $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/mmx_gcc.cpp mythtv-0.20-new/libs/libmythsoundtouch/mmx_gcc.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/mmx_gcc.cpp	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/mmx_gcc.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -15,7 +15,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: mmx_gcc.cpp 9998 2006-05-23 19:14:24Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.cpp mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.cpp	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -10,7 +10,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: RateTransposer.cpp 9998 2006-05-23 19:14:24Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.h mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.h
--- mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.h	2004-11-13 14:29:45.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.h	2006-09-12 06:18:33.000000000 -0700
@@ -14,7 +14,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2004-11-13 17:29:45 -0500 (Sat, 13 Nov 2004) $
+// Last changed  : $Date: 2004-11-13 14:29:45 -0800 (Sat, 13 Nov 2004) $
 // File revision : $Revision: 4714 $
 //
 // $Id: RateTransposer.h 4714 2004-11-13 22:29:45Z ijr $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.cpp mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.cpp	2006-05-24 03:48:15.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -41,7 +41,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-24 06:48:15 -0400 (Wed, 24 May 2006) $
+// Last changed  : $Date: 2006-05-24 03:48:15 -0700 (Wed, 24 May 2006) $
 // File revision : $Revision: 10003 $
 //
 // $Id: SoundTouch.cpp 10003 2006-05-24 10:48:15Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.h mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.h
--- mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.h	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.h	2006-09-12 06:18:33.000000000 -0700
@@ -41,7 +41,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: SoundTouch.h 9996 2006-05-23 17:40:36Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/STTypes.h mythtv-0.20-new/libs/libmythsoundtouch/STTypes.h
--- mythtv-0.20-old/libs/libmythsoundtouch/STTypes.h	2006-05-24 07:56:37.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/STTypes.h	2006-09-12 06:18:33.000000000 -0700
@@ -8,7 +8,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-24 10:56:37 -0400 (Wed, 24 May 2006) $
+// Last changed  : $Date: 2006-05-24 07:56:37 -0700 (Wed, 24 May 2006) $
 // File revision : $Revision: 10005 $
 //
 // $Id: STTypes.h 10005 2006-05-24 14:56:37Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.cpp mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.cpp	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -13,7 +13,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: TDStretch.cpp 9998 2006-05-23 19:14:24Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.h mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.h
--- mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.h	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.h	2006-09-12 06:18:33.000000000 -0700
@@ -13,7 +13,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: TDStretch.h 9998 2006-05-23 19:14:24Z danielk $
diff -Naur mythtv-0.20-old/libs/libmythtv/dbcheck.cpp mythtv-0.20-new/libs/libmythtv/dbcheck.cpp
--- mythtv-0.20-old/libs/libmythtv/dbcheck.cpp	2006-09-10 21:35:19.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/dbcheck.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -10,7 +10,7 @@
 #include "mythdbcon.h"
 
 /// This is the DB schema version expected by the running MythTV instance.
-const QString currentDatabaseVersion = "1158";
+const QString currentDatabaseVersion = "1160";
 
 static bool UpdateDBVersionNumber(const QString &newnumber);
 static bool performActualUpdate(const QString updates[], QString version,
@@ -2510,6 +2510,41 @@
             return false;
     }
 
+    if (dbver == "1158")
+    {
+        const QString updates[] = {
+"ALTER TABLE recorded ADD COLUMN watched TINYINT NOT NULL DEFAULT '0';",
+""
+};
+
+        if (!performActualUpdate(updates, "1159", dbver))
+            return false;
+    }
+
+    if (dbver == "1159")
+    {
+        MSqlQuery query(MSqlQuery::InitCon());
+        query.prepare("SELECT DISTINCT chanid, starttime FROM recordedmarkup "
+                      "WHERE type = 1;");
+        if (query.exec() && query.isActive() && query.size() > 0)
+        {
+            MSqlQuery fixup(MSqlQuery::InitCon());
+            while (query.next())
+            {
+                fixup.prepare(
+                       "UPDATE recorded SET cutlist = 1 "
+                       "WHERE chanid = :CHANID AND starttime =  :STARTTIME;");
+                fixup.bindValue(":CHANID", query.value(0).toString());
+                fixup.bindValue(":STARTTIME", query.value(1).toDateTime());
+
+                fixup.exec();
+            }
+        }
+
+        const QString updates[] = { "" };
+        if (!performActualUpdate(updates, "1160", dbver))
+            return false;
+    }
 
 //"ALTER TABLE capturecard DROP COLUMN dvb_recordts;" in 0.21
 //"ALTER TABLE capturecard DROP COLUMN dvb_hw_decoder;" in 0.21
diff -Naur mythtv-0.20-old/libs/libmythtv/programinfo.cpp mythtv-0.20-new/libs/libmythtv/programinfo.cpp
--- mythtv-0.20-old/libs/libmythtv/programinfo.cpp	2006-09-10 21:35:19.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/programinfo.cpp	2006-09-12 06:18:33.000000000 -0700
@@ -1662,17 +1662,28 @@
     query.bindValue(":PROFILE",     schd->getProfileName());
 
     bool ok = query.exec() && (query.numRowsAffected() > 0);
-    if (!ok && !query.isActive())
+    bool active = query.isActive();
+
+    query.prepare("UNLOCK TABLES");
+    query.exec();
+
+    if (!ok && !active)
         MythContext::DBError("insert_program -- insert", query);
     else
     {
         query.prepare("UPDATE record SET last_record = NOW() "
                       "WHERE recordid = :RECORDID");
-        query.bindValue(":RECORDID",    pg->recordid);
+        query.bindValue(":RECORDID", pg->recordid);
         query.exec();
+
+        if (pg->rectype == kOverrideRecord && pg->parentid > 0)
+        {
+            query.prepare("UPDATE record SET last_record = NOW() "
+                          "WHERE recordid = :PARENTID");
+            query.bindValue(":PARENTID", pg->parentid);
+            query.exec();
+        }
     }
-    query.prepare("UNLOCK TABLES");
-    query.exec();
 
     return ok;
 }
