diff -Naur acpid-2.0.7-old/libnetlink.c acpid-2.0.7-new/libnetlink.c
--- acpid-2.0.7-old/libnetlink.c	2009-04-29 07:37:13.000000000 -0700
+++ acpid-2.0.7-new/libnetlink.c	2011-01-07 11:20:11.000000000 -0800
@@ -518,7 +518,7 @@
 
 int addraw_l(struct nlmsghdr *n, int maxlen, const void *data, int len)
 {
-	if ((int)NLMSG_ALIGN(n->nlmsg_len) + NLMSG_ALIGN(len) > maxlen) {
+	if ((int)NLMSG_ALIGN(n->nlmsg_len) + (int)NLMSG_ALIGN(len) > maxlen) {
 		fprintf(stderr, "addraw_l ERROR: message exceeded bound of %d\n",maxlen);
 		return -1;
 	}
