diff -Naur acpid-2.0.2-old/Makefile acpid-2.0.2-new/Makefile
--- acpid-2.0.2-old/Makefile	2010-01-26 15:36:23.000000000 -0800
+++ acpid-2.0.2-new/Makefile	2010-02-17 07:29:53.000000000 -0800
@@ -57,7 +57,7 @@
 	mkdir -p $(DESTDIR)/$(MANDIR)/man8
 	install -m 0644 $(MAN8GZ) $(DESTDIR)/$(MANDIR)/man8
 	# If DESTDIR is empty and mandb exists, run it
-	[ -z "$(DESTDIR)" ] && which mandb 1>/dev/null 2>&1 && mandb -q
+	( [ -z "$(DESTDIR)" ] && which mandb 1>/dev/null 2>&1 && mandb -q ) || true
 
 DISTTMP=/tmp
 DISTNAME=acpid-$(VERSION)
