diff -Naur lirc-20071217Z-old/configure.ac lirc-20071217Z-new/configure.ac
--- lirc-20071217Z-old/configure.ac	2007-12-18 08:11:41.000000000 -0800
+++ lirc-20071217Z-new/configure.ac	2007-12-18 08:12:39.000000000 -0800
@@ -163,7 +163,6 @@
 	(lirc_dev lirc_it87) \
 	(lirc_dev lirc_mceusb) \
 	(lirc_dev lirc_mceusb2) \
-	(lirc_dev lirc_parallel) \
 	(lirc_dev lirc_sasem) \
 	(lirc_dev lirc_serial) \
 	(lirc_dev lirc_sir) \
@@ -290,9 +289,6 @@
 AH_TEMPLATE([LIRC_SYSLOG],
 	[syslog facility to use])
 
-AH_TEMPLATE([LIRC_TIMER],
-	[Set the timer for the parallel port driver])
-
 AH_TEMPLATE([LOCALSTATEDIR],
 	[modifiable single-machine data])
 
@@ -430,7 +426,7 @@
                           livedrive_seq, logitech, lptX, macmini,
                           mceusb, mceusb2, mediafocusI, mouseremote,
                           mouseremote_ps2, mp3anywhere,
-                          nslu2,packard_bell, parallel, pcmak,
+                          nslu2,packard_bell, pcmak,
                           pcmak_usb, pctv, pixelview_bt878,
                           pixelview_pak, pixelview_pro, provideo,
                           realmagic, remotemaster, sa1100, sasem,
@@ -569,8 +565,6 @@
 		;;
 	lirc_dev-lirc_mceusb2)
 		;;
-	lirc_dev-lirc_parallel)
-		;;
 	lirc_dev-lirc_sasem)
 		;;
 	lirc_dev-lirc_serial)
@@ -1045,18 +1039,6 @@
   lircmd_conf="logitech/lircmd.conf.logitech"
 fi
 
-if test "$driver" = "lpt1"; then
-  lirc_driver="lirc_dev lirc_parallel"
-  port=0x378
-  irq=7
-fi
-
-if test "$driver" = "lpt2"; then
-  lirc_driver="lirc_dev lirc_parallel"
-  port=0x278
-  irq=5
-fi
-
 if test "$driver" = "macmini"; then
   lirc_driver="$driver"
   hw_module="hw_hiddev.o"
@@ -1113,10 +1095,6 @@
   lircd_conf="packard_bell/lircd.conf.packard_bell"
 fi
 
-if test "$driver" = "parallel"; then
-  lirc_driver="lirc_dev lirc_parallel"
-fi
-
 if test "$driver" = "pcmak" -o "$driver" = "pcmak_usb"; then
   lirc_driver="none"
   hw_module="hw_pcmak.o serial.o"
@@ -1385,7 +1363,6 @@
 	lirc_it87 \
 	lirc_mceusb \
 	lirc_mceusb2 \
-	lirc_parallel \
 	lirc_sasem \
 	lirc_serial \
 	lirc_sir \
@@ -1455,10 +1432,6 @@
 irq=${withval}
 )
 
-AC_ARG_WITH(timer,
-[  --with-timer=value      specify the timer value for the parallel driver],
-timer=${withval})
-
 dnl I dont think libirman  will accept this at the moment
 AC_ARG_WITH(tty,
 [  --with-tty=file         specify the tty to use (Irman, RemoteMaster, etc.)],
@@ -1661,7 +1634,6 @@
   drivers/lirc_it87/Makefile
   drivers/lirc_mceusb/Makefile
   drivers/lirc_mceusb2/Makefile
-  drivers/lirc_parallel/Makefile
   drivers/lirc_sasem/Makefile
   drivers/lirc_serial/Makefile
   drivers/lirc_sir/Makefile
