diff -Naur lirc-0.8.4a-old/configure.ac lirc-0.8.4a-new/configure.ac
--- lirc-0.8.4a-old/configure.ac	2008-10-26 07:11:30.000000000 -0700
+++ lirc-0.8.4a-new/configure.ac	2008-10-26 14:16:00.000000000 -0700
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
