diff -Naur lirc-0.8.4pre1-old/configure.ac lirc-0.8.4pre1-new/configure.ac
--- lirc-0.8.4pre1-old/configure.ac	2008-09-17 11:34:08.000000000 -0700
+++ lirc-0.8.4pre1-new/configure.ac	2008-09-21 11:09:15.000000000 -0700
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
