diff -Naur lirc-6b440d65b50f3d2fab0a618e7c35ee6a34cdc1e8-old/configure.ac lirc-6b440d65b50f3d2fab0a618e7c35ee6a34cdc1e8-new/configure.ac
--- lirc-6b440d65b50f3d2fab0a618e7c35ee6a34cdc1e8-old/configure.ac	2010-12-20 16:33:32.000000000 -0800
+++ lirc-6b440d65b50f3d2fab0a618e7c35ee6a34cdc1e8-new/configure.ac	2010-12-20 16:34:21.000000000 -0800
@@ -1553,6 +1553,7 @@
 	lirc_atiusb \
 	lirc_bt829 \
 	lirc_ene0100 \
+	lirc_gpio \
 	lirc_i2c \
 	lirc_igorplugusb \
 	lirc_imon \
