diff -Naur lirc-0.8.6-old/daemons/lircd.c lirc-0.8.6-new/daemons/lircd.c
--- lirc-0.8.6-old/daemons/lircd.c	2009-08-29 00:46:44.000000000 -0700
+++ lirc-0.8.6-new/daemons/lircd.c	2009-11-29 14:01:21.000000000 -0800
@@ -445,7 +445,7 @@
 		goto setup_error;
 	}
 
-        for(key = KEY_RESERVED; key <= KEY_UNKNOWN; key++)
+        for(key = 0; key <= KEY_MAX; key++)
 	{
                 if(ioctl(fd, UI_SET_KEYBIT, key) != 0)
 		{
