diff -Naur lirc-20090201Z-old/configure.ac lirc-20090201Z-new/configure.ac
--- lirc-20090201Z-old/configure.ac	2009-02-02 16:59:51.000000000 -0800
+++ lirc-20090201Z-new/configure.ac	2009-02-02 17:05:51.000000000 -0800
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
