diff -Naur lirc-20080212Z-old/daemons/config_file.c lirc-20080212Z-new/daemons/config_file.c
--- lirc-20080212Z-old/daemons/config_file.c	2007-07-29 11:20:06.000000000 -0700
+++ lirc-20080212Z-new/daemons/config_file.c	2008-02-12 11:58:05.000000000 -0800
@@ -690,7 +690,26 @@
 		if(val!=NULL){
 			val2=strtok(NULL, " \t");
 			LOGPRINTF(3,"\"%s\" \"%s\"",key,val);
-                        if (strcasecmp("begin",key)==0){
+			if (strcasecmp("include",key)==0){
+                                FILE* childFile = fopen(val, "r");
+				if (childFile == NULL){
+					logprintf(LOG_ERR,"error opening child file defined at line %d:",line);
+					logprintf(LOG_ERR,"ignoring this child file for now.");
+				}
+				else{
+					if (!top_rem){
+						/* create first remote */
+						LOGPRINTF(2,"creating first remote");
+						top_rem = rem = read_config(childFile);
+					}else{
+						/* create new remote */
+						LOGPRINTF(2,"creating next remote");
+						rem->next=read_config(childFile);
+						rem=rem->next;
+					}
+					fclose(childFile);
+				}
+			}else if (strcasecmp("begin",key)==0){
 				if (strcasecmp("codes", val)==0){
                                         /* init codes mode */
 					LOGPRINTF(2,"    begin codes");
