diff -Naur lirc-0.8.4pre2-old/configure.ac lirc-0.8.4pre2-new/configure.ac
--- lirc-0.8.4pre2-old/configure.ac	2008-10-05 10:48:04.000000000 -0700
+++ lirc-0.8.4pre2-new/configure.ac	2008-10-08 10:45:27.000000000 -0700
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
