diff -Naur lirc-20080109Z-old/drivers/lirc_it87/lirc_it87.c lirc-20080109Z-new/drivers/lirc_it87/lirc_it87.c
--- lirc-20080109Z-old/drivers/lirc_it87/lirc_it87.c	2007-12-15 09:28:01.000000000 -0800
+++ lirc-20080109Z-new/drivers/lirc_it87/lirc_it87.c	2008-01-10 08:58:03.000000000 -0800
@@ -115,7 +115,11 @@
 static int it87_enable_demodulator;
 
 static int timer_enabled;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
 static spinlock_t timer_lock = SPIN_LOCK_UNLOCKED;
+#else
+static DEFINE_SPINLOCK(timer_lock);
+#endif
 static struct timer_list timerlist;
 /* time of last signal change detected */
 static struct timeval last_tv = {0, 0};
@@ -125,8 +129,13 @@
 
 static DECLARE_WAIT_QUEUE_HEAD(lirc_read_queue);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
 static spinlock_t hardware_lock = SPIN_LOCK_UNLOCKED;
 static spinlock_t dev_lock = SPIN_LOCK_UNLOCKED;
+#else
+static DEFINE_SPINLOCK(hardware_lock);
+static DEFINE_SPINLOCK(dev_lock);
+#endif
 
 static lirc_t rx_buf[RBUF_LEN];
 unsigned int rx_tail, rx_head;
diff -Naur lirc-20080109Z-old/drivers/lirc_sir/lirc_sir.c lirc-20080109Z-new/drivers/lirc_sir/lirc_sir.c
--- lirc-20080109Z-old/drivers/lirc_sir/lirc_sir.c	2007-12-15 09:28:01.000000000 -0800
+++ lirc-20080109Z-new/drivers/lirc_sir/lirc_sir.c	2008-01-10 08:58:03.000000000 -0800
@@ -179,7 +179,11 @@
 static int threshold = 3;
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
 static spinlock_t timer_lock = SPIN_LOCK_UNLOCKED;
+#else
+static DEFINE_SPINLOCK(timer_lock);
+#endif
 static struct timer_list timerlist;
 /* time of last signal change detected */
 static struct timeval last_tv = {0, 0};
@@ -189,8 +193,13 @@
 
 static DECLARE_WAIT_QUEUE_HEAD(lirc_read_queue);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
 static spinlock_t hardware_lock = SPIN_LOCK_UNLOCKED;
 static spinlock_t dev_lock = SPIN_LOCK_UNLOCKED;
+#else
+static DEFINE_SPINLOCK(hardware_lock);
+static DEFINE_SPINLOCK(dev_lock);
+#endif
 
 static lirc_t rx_buf[RBUF_LEN];
 static unsigned int rx_tail, rx_head;
