diff -Naur lirc-20070801-old/configure.ac lirc-20070801-new/configure.ac
--- lirc-20070801-old/configure.ac	2007-08-04 07:51:48.000000000 -0700
+++ lirc-20070801-new/configure.ac	2007-08-04 07:52:06.000000000 -0700
@@ -386,11 +386,6 @@
   possible_drivers="${possible_drivers} (dsp)"
   ]
 )
-dnl i2cuser driver requires the Linux I2C API
-AC_CHECK_HEADERS(linux/i2c-dev.h,[
-  possible_drivers="${possible_drivers} (i2cuser)"
-  ]
-)
 
 dnl here we see what driver the user wants.
 
@@ -406,7 +401,7 @@
                           digimatrix, dsp, dvico, ea65, exaudio,
                           flyvideo, gvbctv5pci, hauppauge,
                           hauppauge_dvb, hercules_smarttv_stereo,
-                          i2cuser, igorplugusb, iguanaIR, imon,
+                          igorplugusb, iguanaIR, imon,
                           imon_24g, imon_knob, imon_pad, imon_rsc,
                           irdeo, irdeo_remote, irman, irreal, it87,
                           knc_one, kworld, leadtek_0007, leadtek_0010,
@@ -509,9 +504,6 @@
 		;;
 	exaudio)
 		;;
-	i2cuser)
-		hw_module="${hw_module} hw_i2cuser.o receive.o"
-		;;
 	iguanaIR)
 		hw_module="${hw_module} hw_iguanaIR.o transmit.o receive.o"
 		iguanaIR_lib=-liguanaIR
@@ -897,12 +889,6 @@
   lircd_conf="hercules/lircd.conf.smarttv_stereo"
 fi
 
-if test "$driver" = "i2cuser"; then
-  lirc_driver="none"
-  hw_module="hw_i2cuser.o receive.o"
-  HW_DEFAULT="hw_i2cuser"
-fi
-
 if test "$driver" = "igorplugusb"; then
   lirc_driver="lirc_dev lirc_igorplugusb"
 fi
diff -Naur lirc-20070801-old/daemons/hw-types.c lirc-20070801-new/daemons/hw-types.c
--- lirc-20070801-old/daemons/hw-types.c	2007-07-15 00:26:39.000000000 -0700
+++ lirc-20070801-new/daemons/hw-types.c	2007-08-04 07:53:44.000000000 -0700
@@ -21,7 +21,6 @@
 extern struct hardware hw_dsp;
 extern struct hardware hw_dvico;
 extern struct hardware hw_ea65;
-extern struct hardware hw_i2cuser;
 extern struct hardware hw_iguanaIR;
 extern struct hardware hw_irman;
 extern struct hardware hw_livedrive_midi;
@@ -101,7 +100,6 @@
 #endif
 	&hw_dvico,
 	&hw_ea65,
-	&hw_i2cuser,
 #ifdef HAVE_IGUANAIR
 	&hw_iguanaIR,
 #endif
diff -Naur lirc-20070801-old/daemons/Makefile.am lirc-20070801-new/daemons/Makefile.am
--- lirc-20070801-old/daemons/Makefile.am	2007-07-15 00:26:39.000000000 -0700
+++ lirc-20070801-new/daemons/Makefile.am	2007-08-04 07:52:06.000000000 -0700
@@ -30,7 +30,6 @@
 			hw_dsp.c \
 			hw_ea65.c hw_ea65.h \
 			hw_hiddev.c \
-			hw_i2cuser.c \
 			hw_iguanaIR.c \
 			hw_irman.c hw_irman.h \
 			hw_livedrive_common.c hw_livedrive_common.h \
@@ -79,7 +78,6 @@
 			hw_dsp.c \
 			hw_ea65.c hw_ea65.h \
 			hw_hiddev.c \
-			hw_i2cuser.c \
 			hw_iguanaIR.c \
 			hw_irman.c hw_irman.h \
 			hw_livedrive_common.c hw_livedrive_common.h \
