diff -Naur lirc-0.8.2-old/configure.in lirc-0.8.2-new/configure.in
--- lirc-0.8.2-old/configure.in	2007-06-09 11:54:09.000000000 -0700
+++ lirc-0.8.2-new/configure.in	2007-06-25 18:14:41.000000000 -0700
@@ -175,7 +175,6 @@
 	(lirc_dev lirc_it87) \
 	(lirc_dev lirc_mceusb) \
 	(lirc_dev lirc_mceusb2) \
-	(lirc_dev lirc_parallel) \
 	(lirc_dev lirc_sasem) \
 	(lirc_dev lirc_serial) \
 	(lirc_dev lirc_sir) \
@@ -275,7 +274,7 @@
                           livedrive_midi, livedrive_seq, logitech,
                           lptX, macmini, mceusb, mceusb2, mediafocusI,
                           mouseremote, mouseremote_ps2, mp3anywhere,
-                          nslu2,packard_bell, parallel, pcmak,
+                          nslu2,packard_bell,  pcmak,
                           pcmak_usb, pctv, pixelview_bt878,
                           pixelview_pak, pixelview_pro, provideo,
                           realmagic, remotemaster, sa1100, sasem,
@@ -411,8 +410,6 @@
 		;;
 	lirc_dev-lirc_mceusb2)
 		;;
-	lirc_dev-lirc_parallel)
-		;;
 	lirc_dev-lirc_sasem)
 		;;
 	lirc_dev-lirc_serial)
@@ -885,18 +882,6 @@
   lircmd_conf="logitech/lircmd.conf.logitech"
 fi
 
-if test "$driver" = "lpt1"; then
-  lirc_driver="lirc_dev lirc_parallel"
-  port=0x378
-  irq=7
-fi
-
-if test "$driver" = "lpt2"; then
-  lirc_driver="lirc_dev lirc_parallel"
-  port=0x278
-  irq=5
-fi
-
 if test "$driver" = "macmini"; then
   lirc_driver="none"
   hw_module="hw_hiddev.o"
@@ -953,10 +938,6 @@
   lircd_conf="packard_bell/lircd.conf.packard_bell"
 fi
 
-if test "$driver" = "parallel"; then
-  lirc_driver="lirc_dev lirc_parallel"
-fi
-
 if test "$driver" = "pcmak" -o "$driver" = "pcmak_usb"; then
   lirc_driver="none"
   hw_module="hw_pcmak.o serial.o"
@@ -1217,7 +1198,6 @@
 	lirc_it87 \
 	lirc_mceusb \
 	lirc_mceusb2 \
-	lirc_parallel \
 	lirc_sasem \
 	lirc_serial \
 	lirc_sir \
@@ -1286,10 +1266,6 @@
 irq=${withval}
 )
 
-AC_ARG_WITH(timer,
-[  --with-timer=value      specify the timer value for the parallel driver],
-timer=${withval})
-
 dnl I dont think libirman  will accept this at the moment
 AC_ARG_WITH(tty,
 [  --with-tty=file         specify the tty to use (Irman, RemoteMaster, etc.)],
@@ -1403,9 +1379,6 @@
 if test "$irq" != ""; then
   AC_DEFINE_UNQUOTED(LIRC_IRQ, $irq)
 fi
-if test "$timer" != ""; then
-  AC_DEFINE_UNQUOTED(LIRC_TIMER, $timer)
-fi
 AC_DEFINE_UNQUOTED(LIRC_IRTTY, "$irtty")
 AC_DEFINE_UNQUOTED(LIRC_SYSLOG, $facility)
 
@@ -1428,7 +1401,6 @@
   drivers/lirc_it87/Makefile
   drivers/lirc_mceusb/Makefile
   drivers/lirc_mceusb2/Makefile
-  drivers/lirc_parallel/Makefile
   drivers/lirc_sasem/Makefile
   drivers/lirc_serial/Makefile
   drivers/lirc_sir/Makefile
