diff -Naur mythplugins-0.20.2-old/mythweb/includes/mythbackend.php mythplugins-0.20.2-new/mythweb/includes/mythbackend.php
--- mythplugins-0.20.2-old/mythweb/includes/mythbackend.php	2007-08-25 09:47:51.000000000 -0700
+++ mythplugins-0.20.2-new/mythweb/includes/mythbackend.php	2007-08-25 09:48:45.000000000 -0700
@@ -22,7 +22,7 @@
 // NUMPROGRAMLINES is defined in mythtv/libs/libmythtv/programinfo.h and is
 // the number of items in a ProgramInfo QStringList group used by
 // ProgramInfo::ToSringList and ProgramInfo::FromStringList.
-    $NUMPROGRAMLINES = 42;
+    $NUMPROGRAMLINES = 44;
 
 // Keep track of the master backend port/ip
     $Master_Host = get_backend_setting('MasterServerIP');
