diff -Naur SDL-1.2.8-old/configure SDL-1.2.8-new/configure
--- SDL-1.2.8-old/configure	2005-02-06 12:00:59.998198208 -0800
+++ SDL-1.2.8-new/configure	2005-02-06 12:01:25.215295142 -0800
@@ -31456,10 +31456,10 @@
 # Set runtime shared library paths as needed
 
 if test $ARCH = linux -o $ARCH = freebsd -o $ARCH = bsdi; then
-  SDL_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
+  SDL_RLD_FLAGS=""
 fi
 if test $ARCH = solaris; then
-  SDL_RLD_FLAGS="-R\${exec_prefix}/lib"
+  SDL_RLD_FLAGS=""
 fi
 
 case "$ARCH" in
