diff -Naur mythtv-0.20.2-old/libs/libmyth/lcddevice.cpp mythtv-0.20.2-new/libs/libmyth/lcddevice.cpp
--- mythtv-0.20.2-old/libs/libmyth/lcddevice.cpp	2006-08-18 08:23:03.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmyth/lcddevice.cpp	2008-01-20 08:30:17.000000000 -0800
@@ -15,6 +15,7 @@
 
 #include <unistd.h>
 #include <sys/wait.h>	// For WIFEXITED on Mac OS X
+#include <cstdlib>
 #include <cmath>
 
 #include <qapplication.h>
diff -Naur mythtv-0.20.2-old/libs/libmyth/lirc.cpp mythtv-0.20.2-new/libs/libmyth/lirc.cpp
--- mythtv-0.20.2-old/libs/libmyth/lirc.cpp	2005-10-30 03:41:09.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmyth/lirc.cpp	2008-01-20 08:31:06.000000000 -0800
@@ -9,6 +9,7 @@
 
 #include "mythcontext.h"
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/libs/libmyth/managedlist.cpp mythtv-0.20.2-new/libs/libmyth/managedlist.cpp
--- mythtv-0.20.2-old/libs/libmyth/managedlist.cpp	2005-11-05 08:46:23.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmyth/managedlist.cpp	2008-01-20 08:32:10.000000000 -0800
@@ -1,5 +1,7 @@
 using namespace std;
 
+#include <cstdlib>
+
 #include "managedlist.h"
 #include "mythcontext.h"
 #include "xmlparse.h"
diff -Naur mythtv-0.20.2-old/libs/libmyth/mythdbcon.cpp mythtv-0.20.2-new/libs/libmyth/mythdbcon.cpp
--- mythtv-0.20.2-old/libs/libmyth/mythdbcon.cpp	2006-04-03 01:58:08.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmyth/mythdbcon.cpp	2008-01-20 08:32:48.000000000 -0800
@@ -1,3 +1,5 @@
+#include <cstdlib>
+
 #include "unistd.h"
 
 #include "mythdbcon.h"
diff -Naur mythtv-0.20.2-old/libs/libmyth/mythsocket.cpp mythtv-0.20.2-new/libs/libmyth/mythsocket.cpp
--- mythtv-0.20.2-old/libs/libmyth/mythsocket.cpp	2008-01-20 08:29:16.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmyth/mythsocket.cpp	2008-01-20 08:33:16.000000000 -0800
@@ -5,6 +5,7 @@
 #include "util.h"
 #include "mythsocket.h"
 #include <sys/select.h>
+#include <cstdlib>
 #include <cassert>
 
 #define SLOC(a) QString("MythSocket(%1:%2): ").arg((unsigned long)a, 0, 16)\
diff -Naur mythtv-0.20.2-old/libs/libmyth/uitypes.cpp mythtv-0.20.2-new/libs/libmyth/uitypes.cpp
--- mythtv-0.20.2-old/libs/libmyth/uitypes.cpp	2006-08-28 12:26:59.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmyth/uitypes.cpp	2008-01-20 08:33:33.000000000 -0800
@@ -1,4 +1,5 @@
 
+#include <cstdlib>
 #include <iostream>
 #include <qapplication.h>
 #include <qsimplerichtext.h>
diff -Naur mythtv-0.20.2-old/libs/libmyth/util.cpp mythtv-0.20.2-new/libs/libmyth/util.cpp
--- mythtv-0.20.2-old/libs/libmyth/util.cpp	2007-08-21 11:30:38.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmyth/util.cpp	2008-01-20 08:33:53.000000000 -0800
@@ -3,6 +3,7 @@
 using namespace std;
 
 // C headers
+#include <cstdlib>
 #include <cerrno>
 #include <unistd.h>
 #include <fcntl.h>
diff -Naur mythtv-0.20.2-old/libs/libmyth/xmlparse.cpp mythtv-0.20.2-new/libs/libmyth/xmlparse.cpp
--- mythtv-0.20.2-old/libs/libmyth/xmlparse.cpp	2006-08-28 12:26:59.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmyth/xmlparse.cpp	2008-01-20 08:34:16.000000000 -0800
@@ -1,6 +1,7 @@
 #include <iostream>
 using namespace std;
 
+#include <cstdlib>
 #include <cmath>
 
 #include <qapplication.h>
diff -Naur mythtv-0.20.2-old/libs/libmythtv/cc708decoder.cpp mythtv-0.20.2-new/libs/libmythtv/cc708decoder.cpp
--- mythtv-0.20.2-old/libs/libmythtv/cc708decoder.cpp	2006-08-21 21:06:45.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythtv/cc708decoder.cpp	2008-01-20 08:34:40.000000000 -0800
@@ -1,6 +1,7 @@
 // -*- Mode: c++ -*-
 // Copyright (c) 2003-2005, Daniel Kristjansson 
 
+#include <cstdlib>
 #include <cassert>
 
 #include "mythcontext.h"
diff -Naur mythtv-0.20.2-old/libs/libmythtv/eitscanner.cpp mythtv-0.20.2-new/libs/libmythtv/eitscanner.cpp
--- mythtv-0.20.2-old/libs/libmythtv/eitscanner.cpp	2006-07-25 01:32:20.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythtv/eitscanner.cpp	2008-01-20 09:47:57.000000000 -0800
@@ -1,5 +1,7 @@
 // -*- Mode: c++ -*-
 
+#include <cstdlib>
+
 // POSIX headers
 #include <sys/time.h>
 #include <sys/resource.h>
diff -Naur mythtv-0.20.2-old/libs/libmythtv/jobqueue.cpp mythtv-0.20.2-new/libs/libmythtv/jobqueue.cpp
--- mythtv-0.20.2-old/libs/libmythtv/jobqueue.cpp	2006-07-30 12:53:33.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythtv/jobqueue.cpp	2008-01-20 09:34:17.000000000 -0800
@@ -11,6 +11,7 @@
 #include <sys/wait.h>
 #include <sys/resource.h>
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/libs/libmythtv/osdsurface.cpp mythtv-0.20.2-new/libs/libmythtv/osdsurface.cpp
--- mythtv-0.20.2-old/libs/libmythtv/osdsurface.cpp	2007-02-19 13:16:11.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/osdsurface.cpp	2008-01-20 08:35:00.000000000 -0800
@@ -2,6 +2,7 @@
 #include "dithertable.h"
 #include "mythcontext.h"
 
+#include <cstdlib>
 #include <algorithm>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/libs/libmythtv/osdsurface.h mythtv-0.20.2-new/libs/libmythtv/osdsurface.h
--- mythtv-0.20.2-old/libs/libmythtv/osdsurface.h	2007-02-19 13:16:11.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/osdsurface.h	2008-01-20 08:35:29.000000000 -0800
@@ -41,7 +41,7 @@
     void BlendToYV12(unsigned char *yptr,
                      unsigned char *uptr,
                      unsigned char *vptr,
-                     int ystride, int ystride, int vstride) const;
+                     int ystride, int ustride, int vstride) const;
     void BlendToARGB(unsigned char *argbptr,
                      uint stride, uint height, bool blendtoblack=false,
                      uint threshold = 0) const;
diff -Naur mythtv-0.20.2-old/libs/libmythtv/programinfo.cpp mythtv-0.20.2-new/libs/libmythtv/programinfo.cpp
--- mythtv-0.20.2-old/libs/libmythtv/programinfo.cpp	2008-01-20 08:29:16.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/programinfo.cpp	2008-01-20 08:36:15.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <iostream>
 #include <qsocket.h>
 #include <qregexp.h>
diff -Naur mythtv-0.20.2-old/libs/libmythtv/util-xv.cpp mythtv-0.20.2-new/libs/libmythtv/util-xv.cpp
--- mythtv-0.20.2-old/libs/libmythtv/util-xv.cpp	2006-08-16 05:35:16.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythtv/util-xv.cpp	2008-01-20 08:36:44.000000000 -0800
@@ -2,6 +2,7 @@
 #include <signal.h>
 
 // C++ headers
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/libs/libmythtv/videosource.cpp mythtv-0.20.2-new/libs/libmythtv/videosource.cpp
--- mythtv-0.20.2-old/libs/libmythtv/videosource.cpp	2008-01-20 08:29:16.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/videosource.cpp	2008-01-20 08:37:14.000000000 -0800
@@ -8,6 +8,7 @@
 #include <sys/stat.h>
 
 // C++ headers
+#include <cstdlib>
 #include <iostream>
 
 // Qt headers
diff -Naur mythtv-0.20.2-old/libs/libmythtv/viewschdiff.cpp mythtv-0.20.2-new/libs/libmythtv/viewschdiff.cpp
--- mythtv-0.20.2-old/libs/libmythtv/viewschdiff.cpp	2006-02-20 14:25:27.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/viewschdiff.cpp	2008-01-20 08:37:42.000000000 -0800
@@ -9,6 +9,7 @@
 #include <qregexp.h>
 #include <qheader.h>
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/libs/libmythtv/xbox.cpp mythtv-0.20.2-new/libs/libmythtv/xbox.cpp
--- mythtv-0.20.2-old/libs/libmythtv/xbox.cpp	2003-12-01 12:17:55.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/xbox.cpp	2008-01-20 08:38:13.000000000 -0800
@@ -1,3 +1,5 @@
+#include <cstdlib>
+
 #include "xbox.h"
 
 #include "mythcontext.h"
diff -Naur mythtv-0.20.2-old/libs/libmythtv/yuv2rgb.cpp mythtv-0.20.2-new/libs/libmythtv/yuv2rgb.cpp
--- mythtv-0.20.2-old/libs/libmythtv/yuv2rgb.cpp	2006-09-09 04:40:46.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythtv/yuv2rgb.cpp	2008-01-20 08:38:56.000000000 -0800
@@ -23,6 +23,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <climits>
 #include <cstdio>
 #include <cstdlib>
 #include <algorithm>
diff -Naur mythtv-0.20.2-old/libs/libmythui/myththemedmenu.cpp mythtv-0.20.2-new/libs/libmythui/myththemedmenu.cpp
--- mythtv-0.20.2-old/libs/libmythui/myththemedmenu.cpp	2007-06-02 07:35:10.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythui/myththemedmenu.cpp	2008-01-20 08:39:46.000000000 -0800
@@ -3,6 +3,7 @@
 #include <qdir.h>
 
 #include <iostream>
+#include <cstdlib>
 #include <cmath>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/libs/libmythui/xmlparsebase.cpp mythtv-0.20.2-new/libs/libmythui/xmlparsebase.cpp
--- mythtv-0.20.2-old/libs/libmythui/xmlparsebase.cpp	2006-05-27 16:10:40.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythui/xmlparsebase.cpp	2008-01-20 08:40:45.000000000 -0800
@@ -1,3 +1,6 @@
+#include <cstdlib>
+#include <typeinfo>
+
 #include <qfile.h>
 
 #include "xmlparsebase.h"
diff -Naur mythtv-0.20.2-old/libs/libmythupnp/httprequest.cpp mythtv-0.20.2-new/libs/libmythupnp/httprequest.cpp
--- mythtv-0.20.2-old/libs/libmythupnp/httprequest.cpp	2007-08-23 13:59:44.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythupnp/httprequest.cpp	2008-01-20 08:42:20.000000000 -0800
@@ -8,6 +8,8 @@
 //                                                                            
 //////////////////////////////////////////////////////////////////////////////
 
+#include <cstdlib>
+
 #include "httprequest.h"
 
 #include <qregexp.h>
diff -Naur mythtv-0.20.2-old/libs/libmythupnp/ssdp.cpp mythtv-0.20.2-new/libs/libmythupnp/ssdp.cpp
--- mythtv-0.20.2-old/libs/libmythupnp/ssdp.cpp	2007-08-23 09:09:51.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythupnp/ssdp.cpp	2008-01-20 08:43:20.000000000 -0800
@@ -8,6 +8,8 @@
 //                                                                            
 //////////////////////////////////////////////////////////////////////////////
 
+#include <cstdlib>
+
 #include "upnp.h"
 
 #include "upnptasksearch.h"
diff -Naur mythtv-0.20.2-old/libs/libmythupnp/upnptasknotify.cpp mythtv-0.20.2-new/libs/libmythupnp/upnptasknotify.cpp
--- mythtv-0.20.2-old/libs/libmythupnp/upnptasknotify.cpp	2007-08-23 09:09:51.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythupnp/upnptasknotify.cpp	2008-01-20 08:43:37.000000000 -0800
@@ -8,6 +8,8 @@
 //                                                                            
 //////////////////////////////////////////////////////////////////////////////
 
+#include <cstdlib>
+
 #include "upnp.h"
 #include "mythcontext.h"
 #include "multicast.h"
diff -Naur mythtv-0.20.2-old/libs/libmythupnp/upnptasksearch.cpp mythtv-0.20.2-new/libs/libmythupnp/upnptasksearch.cpp
--- mythtv-0.20.2-old/libs/libmythupnp/upnptasksearch.cpp	2007-08-23 09:09:51.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythupnp/upnptasksearch.cpp	2008-01-20 08:43:53.000000000 -0800
@@ -8,6 +8,8 @@
 //                                                                            
 //////////////////////////////////////////////////////////////////////////////
 
+#include <cstdlib>
+
 #include "upnp.h"
 #include "mythcontext.h"
 #include "upnptasksearch.h"
diff -Naur mythtv-0.20.2-old/libs/libmythupnp/upnputil.h mythtv-0.20.2-new/libs/libmythupnp/upnputil.h
--- mythtv-0.20.2-old/libs/libmythupnp/upnputil.h	2007-08-23 09:09:51.000000000 -0700
+++ mythtv-0.20.2-new/libs/libmythupnp/upnputil.h	2008-01-20 08:44:12.000000000 -0800
@@ -17,7 +17,7 @@
 
 // __suseconds_t doesn't exist on some older Unixes. e.g. Darwin/Mac OS X
 
-#if defined(__FreeBSD__) 
+#if defined(__FreeBSD__) || defined(__GLIBC__)
 #define __suseconds_t_defined  // It exists on FreeBSD, but doesn't define this
 #endif
 
diff -Naur mythtv-0.20.2-old/programs/mythbackend/httpstatus.cpp mythtv-0.20.2-new/programs/mythbackend/httpstatus.cpp
--- mythtv-0.20.2-old/programs/mythbackend/httpstatus.cpp	2007-08-23 09:09:51.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythbackend/httpstatus.cpp	2008-01-20 08:44:45.000000000 -0800
@@ -8,6 +8,8 @@
 //                                                                            
 //////////////////////////////////////////////////////////////////////////////
 
+#include <cstdlib>
+
 #include "httpstatus.h"
 #include "mythxml.h"
 
diff -Naur mythtv-0.20.2-old/programs/mythbackend/main.cpp mythtv-0.20.2-new/programs/mythbackend/main.cpp
--- mythtv-0.20.2-old/programs/mythbackend/main.cpp	2007-08-23 09:09:51.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythbackend/main.cpp	2008-01-20 08:45:09.000000000 -0800
@@ -10,6 +10,7 @@
 #include <libgen.h>
 #include <signal.h>
 #include <cerrno>
+#include <cstdlib>
 
 #include "mythconfig.h"
 #ifdef CONFIG_DARWIN
diff -Naur mythtv-0.20.2-old/programs/mythcommflag/BlankFrameDetector.cpp mythtv-0.20.2-new/programs/mythcommflag/BlankFrameDetector.cpp
--- mythtv-0.20.2-old/programs/mythcommflag/BlankFrameDetector.cpp	2006-08-17 23:37:43.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythcommflag/BlankFrameDetector.cpp	2008-01-20 08:45:44.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <stdio.h>
 #include <math.h>
 
diff -Naur mythtv-0.20.2-old/programs/mythcommflag/ClassicLogoDetector.cpp mythtv-0.20.2-new/programs/mythcommflag/ClassicLogoDetector.cpp
--- mythtv-0.20.2-old/programs/mythcommflag/ClassicLogoDetector.cpp	2006-07-15 23:47:08.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythcommflag/ClassicLogoDetector.cpp	2008-01-20 08:46:12.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <unistd.h>
 
 #include "ClassicLogoDetector.h"
diff -Naur mythtv-0.20.2-old/programs/mythcommflag/EdgeDetector.cpp mythtv-0.20.2-new/programs/mythcommflag/EdgeDetector.cpp
--- mythtv-0.20.2-old/programs/mythcommflag/EdgeDetector.cpp	2006-08-17 06:50:34.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythcommflag/EdgeDetector.cpp	2008-01-20 08:46:33.000000000 -0800
@@ -1,3 +1,5 @@
+#include <cstdlib>
+
 #include "NuppelVideoPlayer.h"
 #include "avcodec.h"        // AVPicture
 #include "frame.h"          // VideoFrame
diff -Naur mythtv-0.20.2-old/programs/mythcommflag/Histogram.cpp mythtv-0.20.2-new/programs/mythcommflag/Histogram.cpp
--- mythtv-0.20.2-old/programs/mythcommflag/Histogram.cpp	2006-07-15 23:47:08.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythcommflag/Histogram.cpp	2008-01-20 09:54:51.000000000 -0800
@@ -1,6 +1,7 @@
 #include "Histogram.h"
 #include <string>
 #include <cmath>
+#include <cstring>
 
 Histogram::Histogram()
 {
diff -Naur mythtv-0.20.2-old/programs/mythcommflag/SceneChangeDetector.cpp mythtv-0.20.2-new/programs/mythcommflag/SceneChangeDetector.cpp
--- mythtv-0.20.2-old/programs/mythcommflag/SceneChangeDetector.cpp	2006-08-17 06:52:25.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythcommflag/SceneChangeDetector.cpp	2008-01-20 08:48:05.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <stdio.h>
 #include <math.h>
 
diff -Naur mythtv-0.20.2-old/programs/mythcommflag/TemplateFinder.cpp mythtv-0.20.2-new/programs/mythcommflag/TemplateFinder.cpp
--- mythtv-0.20.2-old/programs/mythcommflag/TemplateFinder.cpp	2006-08-17 06:52:25.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythcommflag/TemplateFinder.cpp	2008-01-20 08:48:39.000000000 -0800
@@ -1,3 +1,5 @@
+#include <cstdlib>
+
 #include <qfile.h>
 #include <qfileinfo.h>
 #include <math.h>
diff -Naur mythtv-0.20.2-old/programs/mythfrontend/main.cpp mythtv-0.20.2-new/programs/mythfrontend/main.cpp
--- mythtv-0.20.2-old/programs/mythfrontend/main.cpp	2007-08-21 11:53:55.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythfrontend/main.cpp	2008-01-20 08:50:02.000000000 -0800
@@ -11,6 +11,8 @@
 #include <signal.h>
 #include <pthread.h>
 
+#include <cstdlib>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/programs/mythfrontend/playbackbox.cpp mythtv-0.20.2-new/programs/mythfrontend/playbackbox.cpp
--- mythtv-0.20.2-old/programs/mythfrontend/playbackbox.cpp	2007-07-13 21:40:55.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythfrontend/playbackbox.cpp	2008-01-20 08:49:30.000000000 -0800
@@ -17,6 +17,7 @@
 #include <qsqldatabase.h>
 #include <qmap.h>
 
+#include <cstdlib>
 #include <cmath>
 #include <unistd.h>
 
diff -Naur mythtv-0.20.2-old/programs/mythfrontend/statusbox.cpp mythtv-0.20.2-new/programs/mythfrontend/statusbox.cpp
--- mythtv-0.20.2-old/programs/mythfrontend/statusbox.cpp	2006-09-08 21:45:30.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythfrontend/statusbox.cpp	2008-01-20 08:50:20.000000000 -0800
@@ -11,6 +11,7 @@
 
 #include <iostream>
 #include <cerrno>
+#include <cstdlib>
 using namespace std;
 
 #include "config.h"
diff -Naur mythtv-0.20.2-old/programs/mythfrontend/viewscheduled.cpp mythtv-0.20.2-new/programs/mythfrontend/viewscheduled.cpp
--- mythtv-0.20.2-old/programs/mythfrontend/viewscheduled.cpp	2006-08-11 15:22:12.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythfrontend/viewscheduled.cpp	2008-01-20 08:50:31.000000000 -0800
@@ -9,6 +9,7 @@
 #include <qregexp.h>
 #include <qheader.h>
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/programs/mythlcdserver/lcdprocclient.cpp mythtv-0.20.2-new/programs/mythlcdserver/lcdprocclient.cpp
--- mythtv-0.20.2-old/programs/mythlcdserver/lcdprocclient.cpp	2006-06-27 07:48:40.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythlcdserver/lcdprocclient.cpp	2008-01-20 08:50:56.000000000 -0800
@@ -10,6 +10,7 @@
 #include <qregexp.h>
 #include <unistd.h>
 #include <cmath>
+#include <cstdlib>
 
 #include "lcdprocclient.h"
 #include "mythcontext.h"
diff -Naur mythtv-0.20.2-old/programs/mythlcdserver/lcdserver.cpp mythtv-0.20.2-new/programs/mythlcdserver/lcdserver.cpp
--- mythtv-0.20.2-old/programs/mythlcdserver/lcdserver.cpp	2006-06-27 07:48:40.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythlcdserver/lcdserver.cpp	2008-01-20 08:51:31.000000000 -0800
@@ -55,6 +55,8 @@
 
 */
 
+#include <cstdlib>
+
 #include <qstringlist.h>
 #include <qregexp.h>
 #include <qdir.h>
diff -Naur mythtv-0.20.2-old/programs/mythlcdserver/main.cpp mythtv-0.20.2-new/programs/mythlcdserver/main.cpp
--- mythtv-0.20.2-old/programs/mythlcdserver/main.cpp	2006-05-19 18:40:11.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythlcdserver/main.cpp	2008-01-20 08:51:44.000000000 -0800
@@ -10,6 +10,7 @@
 #include <qfile.h>
 #include <qtextstream.h>
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 #include <unistd.h>
diff -Naur mythtv-0.20.2-old/programs/mythshutdown/main.cpp mythtv-0.20.2-new/programs/mythshutdown/main.cpp
--- mythtv-0.20.2-old/programs/mythshutdown/main.cpp	2006-02-25 03:20:50.000000000 -0800
+++ mythtv-0.20.2-new/programs/mythshutdown/main.cpp	2008-01-20 10:08:08.000000000 -0800
@@ -2,6 +2,7 @@
 #include <qfile.h>
 #include <qtextstream.h>
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 #include <unistd.h>
diff -Naur mythtv-0.20.2-old/programs/mythuitest/main.cpp mythtv-0.20.2-new/programs/mythuitest/main.cpp
--- mythtv-0.20.2-old/programs/mythuitest/main.cpp	2006-05-21 21:00:27.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythuitest/main.cpp	2008-01-20 10:05:37.000000000 -0800
@@ -1,6 +1,7 @@
 #include <qapplication.h>
 #include <qsqldatabase.h>
 
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.20.2-old/programs/mythwelcome/welcomedialog.cpp mythtv-0.20.2-new/programs/mythwelcome/welcomedialog.cpp
--- mythtv-0.20.2-old/programs/mythwelcome/welcomedialog.cpp	2006-10-01 04:51:07.000000000 -0700
+++ mythtv-0.20.2-new/programs/mythwelcome/welcomedialog.cpp	2008-01-20 08:52:03.000000000 -0800
@@ -1,3 +1,5 @@
+#include <cstdlib>
+
 #include <qapplication.h>
 #include <unistd.h>
 
