diff -Naur mythtv-0.20.2-old/libs/libmythtv/tv_play.cpp mythtv-0.20.2-new/libs/libmythtv/tv_play.cpp
--- mythtv-0.20.2-old/libs/libmythtv/tv_play.cpp	2007-11-29 06:24:41.000000000 -0800
+++ mythtv-0.20.2-new/libs/libmythtv/tv_play.cpp	2007-11-29 06:26:19.000000000 -0800
@@ -1151,7 +1151,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
