diff -Naur xf86-input-evtouch-0.8.7-old/calibrate.sh xf86-input-evtouch-0.8.7-new/calibrate.sh
--- xf86-input-evtouch-0.8.7-old/calibrate.sh	2007-03-01 06:45:47.000000000 -0800
+++ xf86-input-evtouch-0.8.7-new/calibrate.sh	2007-09-10 11:38:29.000000000 -0700
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 XINIT=`which xinit`
 XSERVER=`which X`
 EVCALIBRATE=`which ev_calibrate || echo ./ev_calibrate`
diff -Naur xf86-input-evtouch-0.8.7-old/ev_calibrate.c xf86-input-evtouch-0.8.7-new/ev_calibrate.c
--- xf86-input-evtouch-0.8.7-old/ev_calibrate.c	2007-03-30 08:31:14.000000000 -0700
+++ xf86-input-evtouch-0.8.7-new/ev_calibrate.c	2007-09-10 11:38:29.000000000 -0700
@@ -198,7 +198,7 @@
         Pixmap cursor_pic;
         XColor cursor_fg, cursor_bg;
         XFontStruct* font_info;
-        char* font_name = "*-helvetica-*-12-*";
+        char* font_name = "fixed";
         char str[256];   
         unsigned int cursor_width, cursor_height;
         int hotspot_x, hotspot_y;
