diff -Naur sdlmame0114-old/makefile sdlmame0114-new/makefile
--- sdlmame0114-old/makefile	2007-03-30 00:03:29.000000000 -0700
+++ sdlmame0114-new/makefile	2007-04-08 03:02:09.000000000 -0700
@@ -78,7 +78,7 @@
 # ATHLON = 1
 # I686 = 1
 # P4 = 1
-PM = 1
+# PM = 1
 # AMD64 = 1
 # G4 = 1
 # G5 = 1
@@ -88,10 +88,10 @@
 # PTR64 = 1
 
 # uncomment next line to build expat as part of MAME build
-BUILD_EXPAT = 1
+# BUILD_EXPAT = 1
 
 # uncomment next line to build zlib as part of MAME build
-BUILD_ZLIB = 1
+# BUILD_ZLIB = 1
 
 # uncomment next line to include the symbols
 # SYMBOLS = 1
@@ -173,11 +173,11 @@
 
 
 # compiler, linker and utilities
-AR = @ar
-CC = @gcc
-LD = @gcc
-MD = -mkdir
-RM = @rm -f
+AR ?= ar
+CC ?= gcc
+LD = $(CC)
+MD = mkdir -p
+RM = rm -f
 
 
 
