diff -Naur mythtv-0.19-old/libs/libmythtv/videobuffers.cpp mythtv-0.19-new/libs/libmythtv/videobuffers.cpp
--- mythtv-0.19-old/libs/libmythtv/videobuffers.cpp	2006-06-07 18:18:56.000000000 -0700
+++ mythtv-0.19-new/libs/libmythtv/videobuffers.cpp	2006-06-07 18:19:13.000000000 -0700
@@ -1139,16 +1139,18 @@
     if (surfs.size()>allocSize())
     {
         VERBOSE(VB_PLAYBACK,
-                "Woo Hoo! We have more XvMC Surfaces than we need");
+                QString("Allocated %1 XvMC surfaces, minimum was %2 surfaces")
+                .arg(surfs.size()).arg(allocSize()));
+
         Reset();
-        bool extra_for_pause = allocSize() > numbuffers;
-        bool normal = (needprebufferframes_normal == needprebufferframes);
         uint increment = surfs.size() - allocSize();
-        
-        Init(numbuffers, extra_for_pause, 
-             needfreeframes, needprebufferframes_normal+increment-1,
-             needprebufferframes_small, keepprebufferframes+1);
-        SetPrebuffering(normal);
+
+        Init(surfs.size(),
+             false /* create an extra frame for pause? */,
+             needfreeframes,
+             needprebufferframes_normal + increment - 1,
+             needprebufferframes_small,
+             keepprebufferframes + 1);
     }
 
     for (uint i = 0; i < allocSize(); i++)
diff -Naur mythtv-0.19-old/libs/libmythtv/videoout_xv.cpp mythtv-0.19-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-0.19-old/libs/libmythtv/videoout_xv.cpp	2006-06-07 18:18:56.000000000 -0700
+++ mythtv-0.19-new/libs/libmythtv/videoout_xv.cpp	2006-06-07 18:19:13.000000000 -0700
@@ -793,9 +793,6 @@
         bool vld, idct, mc;
         myth2av_codecid(myth_codec_id, vld, idct, mc);
 
-        if (vld)
-            xvmc_buf_attr->SetNumSurf(16);
-
         vbuffers.Init(xvmc_buf_attr->GetNumSurf(),
                       false /* create an extra frame for pause? */,
                       xvmc_buf_attr->GetFrameReserve(),
