diff -Naur mythstream-0.16_5-old/libs/configelements.h mythstream-0.16_5-new/libs/configelements.h
--- mythstream-0.16_5-old/libs/configelements.h	2005-05-14 06:22:51.000000000 -0700
+++ mythstream-0.16_5-new/libs/configelements.h	2006-03-28 09:04:21.000000000 -0800
@@ -176,8 +176,8 @@
 
     void buildGroup(int boxes);
     void setStatus(Status stat);
-    QLabel* EditGroup::getLabel(QString name);
-    QLineEdit* EditGroup::getLineEdit(QString name);
+    QLabel* getLabel(QString name);
+    QLineEdit* getLineEdit(QString name);
 
     int boxCount;
 
diff -Naur mythstream-0.16_5-old/libs/streamconfig.h mythstream-0.16_5-new/libs/streamconfig.h
--- mythstream-0.16_5-old/libs/streamconfig.h	2005-05-14 06:22:51.000000000 -0700
+++ mythstream-0.16_5-new/libs/streamconfig.h	2006-03-28 09:04:21.000000000 -0800
@@ -92,7 +92,7 @@
     ConfigButton *overwriteButton;
 
     // Helpers
-    bool StorageGroup::getStorageValuesByName(ValueList& values, const QString& storageName);
+    bool getStorageValuesByName(ValueList& values, const QString& storageName);
 
     // GUI
     void buildGroup();
