diff -Naur MPlayer-1.0rc2-old/configure MPlayer-1.0rc2-new/configure
--- MPlayer-1.0rc2-old/configure	2015-06-30 17:45:33.649000373 -0700
+++ MPlayer-1.0rc2-new/configure	2015-06-30 17:47:22.786001254 -0700
@@ -2942,12 +2942,7 @@
 
 echocheck "word size"
 _mp_wordsize="#undef MP_WORDSIZE"
-cat > $TMPC << EOF
-#include <stdio.h>
-#include <sys/types.h>
-int main(void) { printf("%d\n", sizeof(size_t)*8); return 0; }
-EOF
-cc_check && _wordsize=`$TMPEXE` && _mp_wordsize="#define MP_WORDSIZE $_wordsize"
+_wordsize=64 && _mp_wordsize="#define MP_WORDSIZE $_wordsize"
 echores "$_wordsize"
 
 
