diff -Naur libXdamage-1.1.3-old/configure.ac libXdamage-1.1.3-new/configure.ac
--- libXdamage-1.1.3-old/configure.ac	2010-06-08 18:50:35.000000000 -0700
+++ libXdamage-1.1.3-new/configure.ac	2012-12-29 05:12:11.000000000 -0800
@@ -42,7 +42,7 @@
           [m4_fatal([must install xorg-macros 1.3 or later before running autoconf/autogen])])
 XORG_MACROS_VERSION(1.3)
 XORG_DEFAULT_OPTIONS
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Check for progs
 AC_PROG_CC
