diff -Naur uvc-127-old/Makefile uvc-127-new/Makefile
--- uvc-127-old/Makefile	2007-09-28 10:25:35.000000000 -0700
+++ uvc-127-new/Makefile	2007-09-28 10:26:05.000000000 -0700
@@ -16,7 +16,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Modules.symvers
