diff -Naur uvc-195-old/Makefile uvc-195-new/Makefile
--- uvc-195-old/Makefile	2008-03-15 17:40:15.000000000 -0700
+++ uvc-195-new/Makefile	2008-03-15 17:40:45.000000000 -0700
@@ -17,7 +17,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae $(KERNEL_VERSION)
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Module.symvers version.h
