diff -Naur uvc-200-old/Makefile uvc-200-new/Makefile
--- uvc-200-old/Makefile	2008-04-15 20:28:13.000000000 -0700
+++ uvc-200-new/Makefile	2008-04-15 20:28:40.000000000 -0700
@@ -17,7 +17,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae $(KERNEL_VERSION)
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Module.symvers version.h
