diff -Naur uvc-147-old/Makefile uvc-147-new/Makefile
--- uvc-147-old/Makefile	2007-11-24 10:01:03.000000000 -0800
+++ uvc-147-new/Makefile	2007-11-24 10:01:41.000000000 -0800
@@ -16,7 +16,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Modules.symvers
