diff -Naur linux-2.6.24-rc5-old/drivers/net/via-rhine.c linux-2.6.24-rc5-new/drivers/net/via-rhine.c
--- linux-2.6.24-rc5-old/drivers/net/via-rhine.c	2007-12-17 21:31:53.000000000 -0800
+++ linux-2.6.24-rc5-new/drivers/net/via-rhine.c	2007-12-17 21:33:06.000000000 -0800
@@ -131,7 +131,7 @@
 module_param(max_interrupt_work, int, 0);
 module_param(debug, int, 0);
 module_param(rx_copybreak, int, 0);
-module_param(avoid_D3, bool, 0);
+module_param(avoid_D3, bool, 1);
 MODULE_PARM_DESC(max_interrupt_work, "VIA Rhine maximum events handled per interrupt");
 MODULE_PARM_DESC(debug, "VIA Rhine debug level (0-7)");
 MODULE_PARM_DESC(rx_copybreak, "VIA Rhine copy breakpoint for copy-only-tiny-frames");
