diff -Naur linux-2.6.24.3-old/include/linux/Kbuild linux-2.6.24.3-new/include/linux/Kbuild
--- linux-2.6.24.3-old/include/linux/Kbuild	2008-02-25 16:20:20.000000000 -0800
+++ linux-2.6.24.3-new/include/linux/Kbuild	2008-02-26 11:43:59.000000000 -0800
@@ -217,7 +217,6 @@
 unifdef-y += icmp.h
 unifdef-y += icmpv6.h
 unifdef-y += if_addr.h
-unifdef-y += if_addrlabel.h
 unifdef-y += if_arp.h
 unifdef-y += if_bridge.h
 unifdef-y += if_ec.h
