diff -Naur linux-2.6.24.4-old/include/linux/time.h linux-2.6.24.4-new/include/linux/time.h
--- linux-2.6.24.4-old/include/linux/time.h	2008-03-24 11:49:18.000000000 -0700
+++ linux-2.6.24.4-new/include/linux/time.h	2008-03-24 17:35:53.000000000 -0700
@@ -169,7 +169,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
