diff -Naur linux-2.6.24.4-old/arch/x86/kernel/quirks.c linux-2.6.24.4-new/arch/x86/kernel/quirks.c
--- linux-2.6.24.4-old/arch/x86/kernel/quirks.c	2008-03-24 11:49:18.000000000 -0700
+++ linux-2.6.24.4-new/arch/x86/kernel/quirks.c	2008-03-26 03:03:13.000000000 -0700
@@ -355,6 +355,8 @@
 			nvidia_force_enable_hpet);
 
 /* LPC bridges */
+DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_NVIDIA, 0x0260,
+			nvidia_force_enable_hpet);
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_NVIDIA, 0x0360,
 			nvidia_force_enable_hpet);
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_NVIDIA, 0x0361,
