diff -Naur module-init-tools-3.4.1-old/Makefile.in module-init-tools-3.4.1-new/Makefile.in
--- module-init-tools-3.4.1-old/Makefile.in	2008-09-09 09:31:43.000000000 -0700
+++ module-init-tools-3.4.1-new/Makefile.in	2008-09-11 13:30:04.000000000 -0700
@@ -239,8 +239,8 @@
 EXTRA_insmod_static_SOURCES = backwards_compat.c
 EXTRA_depmod_SOURCES = moduleops_core.c
 EXTRA_modinfo_SOURCES = backwards_compat.c
-MAN5 = modprobe.conf.5 modules.dep.5 depmod.conf.5
-MAN8 = depmod.8 insmod.8 lsmod.8 rmmod.8 modprobe.8 modinfo.8
+MAN5 =
+MAN8 =
 SGML = $(addprefix doc/,  $(MAN5:%.5=%.sgml) $(MAN8:%.8=%.sgml))
 man_MANS = $(MAN5) $(MAN8)
 TESTSUITE := $(shell find tests -type f ! -name '*~') tests/vg-suppressions
