diff -Naur mythtv-0.19-old/libs/libmythtv/videoout_xv.cpp mythtv-0.19-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-0.19-old/libs/libmythtv/videoout_xv.cpp	2006-02-08 08:38:17.000000000 -0800
+++ mythtv-0.19-new/libs/libmythtv/videoout_xv.cpp	2006-05-21 07:20:31.000000000 -0700
@@ -794,7 +794,7 @@
         myth2av_codecid(myth_codec_id, vld, idct, mc);
 
         if (vld)
-            xvmc_buf_attr->SetNumSurf(16);
+            xvmc_buf_attr->SetNumSurf(8);
 
         vbuffers.Init(xvmc_buf_attr->GetNumSurf(),
                       false /* create an extra frame for pause? */,
