diff -Naur NVIDIA-Linux-x86_64-1.0-7184-pkg2-old/usr/src/nv/nv-linux.h NVIDIA-Linux-x86_64-1.0-7184-pkg2-new/usr/src/nv/nv-linux.h
--- NVIDIA-Linux-x86_64-1.0-7184-pkg2-old/usr/src/nv/nv-linux.h	2006-08-01 20:01:08.000000000 -0700
+++ NVIDIA-Linux-x86_64-1.0-7184-pkg2-new/usr/src/nv/nv-linux.h	2007-04-07 18:30:27.000000000 -0700
@@ -14,8 +14,10 @@
 
 #include "nv.h"
 
-#include <linux/config.h>
 #include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19)
+#  include <linux/config.h>
+#endif
 #include <linux/utsname.h>
 
 
