diff -Naur coreutils-7.2-old/Makefile.am coreutils-7.2-new/Makefile.am
--- coreutils-7.2-old/Makefile.am	2009-04-02 21:29:53.000000000 -0700
+++ coreutils-7.2-new/Makefile.am	2009-04-02 21:30:36.000000000 -0700
@@ -15,7 +15,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = cfg.mk maint.mk \
   .prev-version THANKS-to-translators THANKStt.in \
   .version \
diff -Naur coreutils-7.2-old/Makefile.in coreutils-7.2-new/Makefile.in
--- coreutils-7.2-old/Makefile.in	2009-04-02 21:29:53.000000000 -0700
+++ coreutils-7.2-new/Makefile.in	2009-04-02 21:30:36.000000000 -0700
@@ -931,7 +931,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = cfg.mk maint.mk \
   .prev-version THANKS-to-translators THANKStt.in \
   .version \
