diff -Naur coreutils-8.3-old/Makefile.am coreutils-8.3-new/Makefile.am
--- coreutils-8.3-old/Makefile.am	2010-01-07 18:13:13.000000000 -0800
+++ coreutils-8.3-new/Makefile.am	2010-01-07 18:13:48.000000000 -0800
@@ -17,7 +17,7 @@
 
 ALL_RECURSIVE_TARGETS =
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 
 changelog_etc =				\
   ChangeLog-2005			\
diff -Naur coreutils-8.3-old/Makefile.in coreutils-8.3-new/Makefile.in
--- coreutils-8.3-old/Makefile.in	2010-01-07 18:13:13.000000000 -0800
+++ coreutils-8.3-new/Makefile.in	2010-01-07 18:13:48.000000000 -0800
@@ -1132,7 +1132,7 @@
 
 # Some tests always need root privileges, others need them only sometimes.
 ALL_RECURSIVE_TARGETS = install-root check-root distcheck-hook
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 changelog_etc = \
   ChangeLog-2005			\
   ChangeLog-2006			\
