diff -Naur coreutils-6.10-old/Makefile.am coreutils-6.10-new/Makefile.am
--- coreutils-6.10-old/Makefile.am	2008-01-07 09:18:10.000000000 -0800
+++ coreutils-6.10-new/Makefile.am	2008-01-24 18:41:16.000000000 -0800
@@ -15,7 +15,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = Makefile.cfg Makefile.maint GNUmakefile \
   .prev-version THANKS-to-translators THANKStt.in \
   .vg-suppressions \
diff -Naur coreutils-6.10-old/Makefile.in coreutils-6.10-new/Makefile.in
--- coreutils-6.10-old/Makefile.in	2008-01-21 15:27:00.000000000 -0800
+++ coreutils-6.10-new/Makefile.in	2008-01-24 18:41:33.000000000 -0800
@@ -678,7 +678,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = Makefile.cfg Makefile.maint GNUmakefile \
   .prev-version THANKS-to-translators THANKStt.in \
   .vg-suppressions \
