diff -Naur gnash-0.8.6-old/libbase/GnashImagePng.cpp gnash-0.8.6-new/libbase/GnashImagePng.cpp
--- gnash-0.8.6-old/libbase/GnashImagePng.cpp	2009-09-13 16:22:16.000000000 -0700
+++ gnash-0.8.6-new/libbase/GnashImagePng.cpp	2010-01-08 20:06:48.000000000 -0800
@@ -165,7 +165,7 @@
     if (type == PNG_COLOR_TYPE_GRAY && bitDepth < 8)
     {
         log_debug("Setting grey bit depth(%d) to 8", bitDepth);
-        png_set_gray_1_2_4_to_8(_pngPtr);
+        png_set_expand_gray_1_2_4_to_8(_pngPtr);
     }
 
     // Apply the transparency block if it exists.
