diff -Naur xc-old/config/cf/Imake.rules xc-new/config/cf/Imake.rules
--- xc-old/config/cf/Imake.rules	2005-02-11 05:15:20.768265508 -0800
+++ xc-new/config/cf/Imake.rules	2005-02-11 05:17:54.645756218 -0800
@@ -1733,7 +1733,7 @@
 	@MakeFlagsToShellFlags(i,set +e); \				@@\
 	if [ -d Concat($(DESTDIR),opath/cdir) ]; then \			@@\
 	  RELPATH=`echo opath/cdir | \					@@\
-		sed -e 's:^/::' -e 's:[^/.][^/]*:..:'g -e 's:/\.$$::'`; \ @@\
+		sed -e 's:^/*::' -e 's:[^/.][^/]*:..:'g -e 's:/\.$$::'`; \ @@\
 	  cd Concat($(DESTDIR),opath/cdir); \				@@\
 	  if [ -d rdir -a ! -h rdir ]; then \				@@\
 	    (cd rdir; tar cf - . | \					@@\
@@ -1780,7 +1780,7 @@
 	@MakeFlagsToShellFlags(i,set +e); \				@@\
 	if [ -d Concat($(DESTDIR),opath) ]; then \			@@\
 	  RELPATH=`echo opath | \					@@\
-		sed -e 's:^.::' -e 's:[^/.][^/]*:..:'g`; \		@@\
+		sed -e 's:^/*::' -e 's:[^/.][^/]*:..:'g`; \		@@\
 	  cd Concat($(DESTDIR),opath); \				@@\
 	  if [ -f lfile -a ! -h lfile ]; then \				@@\
 	    cp -p lfile Concat($(DESTDIR),npath/cfile); \		@@\
