diff -Naur vlc-1.1.0-git-old/configure.ac vlc-1.1.0-git-new/configure.ac
--- vlc-1.1.0-git-old/configure.ac	2009-09-01 15:04:36.000000000 -0700
+++ vlc-1.1.0-git-new/configure.ac	2009-09-02 22:51:17.000000000 -0700
@@ -2852,8 +2852,8 @@
       AC_CHECK_HEADERS(libavcodec/avcodec.h ffmpeg/avcodec.h)
       AC_CHECK_HEADERS(libavutil/avutil.h ffmpeg/avutil.h)
       VLC_ADD_PLUGIN([avcodec])
-      VLC_ADD_LIBS([avcodec],[$AVCODEC_LIBS])
-      VLC_ADD_CFLAGS([avcodec],[$AVCODEC_CFLAGS])
+      VLC_ADD_LIBS([avcodec],[$AVCODEC_LIBS $X_LIBS $X_PRE_LIBS -lX11 -lva-x11])
+      VLC_ADD_CFLAGS([avcodec],[$AVCODEC_CFLAGS $X_CFLAGS])
       VLC_RESTORE_FLAGS
       have_avcodec="yes"
     ],[
