diff -Naur coreutils-6.11-old/Makefile.am coreutils-6.11-new/Makefile.am
--- coreutils-6.11-old/Makefile.am	2008-04-19 22:33:32.000000000 -0700
+++ coreutils-6.11-new/Makefile.am	2008-04-19 22:34:51.000000000 -0700
@@ -15,7 +15,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = cfg.mk maint.mk \
   .prev-version THANKS-to-translators THANKStt.in \
   .version \
diff -Naur coreutils-6.11-old/Makefile.in coreutils-6.11-new/Makefile.in
--- coreutils-6.11-old/Makefile.in	2008-04-19 22:33:32.000000000 -0700
+++ coreutils-6.11-new/Makefile.in	2008-04-19 22:35:04.000000000 -0700
@@ -714,7 +714,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = cfg.mk maint.mk \
   .prev-version THANKS-to-translators THANKStt.in \
   .version \
