diff -Naur coreutils-6.9-old/Makefile.am coreutils-6.9-new/Makefile.am
--- coreutils-6.9-old/Makefile.am	2007-03-18 14:36:43.000000000 -0700
+++ coreutils-6.9-new/Makefile.am	2007-10-05 15:29:57.000000000 -0700
@@ -17,7 +17,7 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 # 02110-1301, USA.
 
-SUBDIRS = lib src doc man po tests
+SUBDIRS = lib src doc po tests
 EXTRA_DIST = Makefile.cfg Makefile.maint GNUmakefile \
   .gitignore \
   .hgignore \
diff -Naur coreutils-6.9-old/Makefile.in coreutils-6.9-new/Makefile.in
--- coreutils-6.9-old/Makefile.in	2007-03-22 14:20:26.000000000 -0700
+++ coreutils-6.9-new/Makefile.in	2007-10-05 15:30:06.000000000 -0700
@@ -520,7 +520,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = lib src doc man po tests
+SUBDIRS = lib src doc po tests
 EXTRA_DIST = Makefile.cfg Makefile.maint GNUmakefile \
   .gitignore \
   .hgignore \
