diff -Naur libpciaccess-0.10.3-old/configure.ac libpciaccess-0.10.3-new/configure.ac
--- libpciaccess-0.10.3-old/configure.ac	2008-06-10 12:24:44.000000000 -0700
+++ libpciaccess-0.10.3-new/configure.ac	2008-09-23 07:49:41.000000000 -0700
@@ -91,7 +91,7 @@
 AM_CONDITIONAL(OPENBSD, [test "x$openbsd" = xyes])
 AM_CONDITIONAL(SOLARIS, [test "x$solaris" = xyes])
 
-AC_CHECK_FILE([/usr/include/asm/mtrr.h],
+AC_CHECK_HEADER([asm/mtrr.h],
               [have_mtrr_h="yes"], [have_mtrr_h="no"])
 if test "x$have_mtrr_h" = xyes; then
     AC_DEFINE(HAVE_MTRR, 1, [Use MTRRs on mappings])
