diff -Naur linux-libc-headers-2.6.12.0-old/include/linux/agpgart.h linux-libc-headers-2.6.12.0-new/include/linux/agpgart.h
--- linux-libc-headers-2.6.12.0-old/include/linux/agpgart.h	2005-03-26 10:58:04.000000000 -0800
+++ linux-libc-headers-2.6.12.0-new/include/linux/agpgart.h	2005-10-27 07:09:37.000000000 -0700
@@ -50,6 +50,7 @@
 #endif
 
 #include <linux/types.h>
+#include <asm/types.h>
 
 struct agp_version {
 	__u16 major;
