diff -Naur mysql-4.1.22-old/configure.in mysql-4.1.22-new/configure.in
--- mysql-4.1.22-old/configure.in	2006-11-28 20:16:56.000000000 -0800
+++ mysql-4.1.22-new/configure.in	2006-11-28 20:17:40.000000000 -0800
@@ -51,7 +51,7 @@
       ;;
     *)
   echo "$i/errmsg.sys: $i/errmsg.txt
-	\$(top_builddir)/extra/comp_err -C\$(srcdir)/charsets/ $i/errmsg.txt $i/errmsg.sys" \
+	mysql-comp_err -C\$(srcdir)/charsets/ $i/errmsg.txt $i/errmsg.sys" \
     >> $AVAILABLE_LANGUAGES_ERRORS_RULES
       ;;
   esac
diff -Naur mysql-4.1.22-old/sql/share/Makefile.am mysql-4.1.22-new/sql/share/Makefile.am
--- mysql-4.1.22-old/sql/share/Makefile.am	2006-11-28 20:16:34.000000000 -0800
+++ mysql-4.1.22-new/sql/share/Makefile.am	2006-11-28 20:17:40.000000000 -0800
@@ -53,7 +53,7 @@
 fix_errors:
 	for lang in @AVAILABLE_LANGUAGES@; \
 	do \
-	../../extra/comp_err -C$(srcdir)/charsets/ $(srcdir)/$$lang/errmsg.txt $(srcdir)/$$lang/errmsg.sys; \
+	mysql-comp_err -C$(srcdir)/charsets/ $(srcdir)/$$lang/errmsg.txt $(srcdir)/$$lang/errmsg.sys; \
         done
 
 # Don't update the files from bitkeeper
diff -Naur mysql-4.1.22-old/sql/share/Makefile.in mysql-4.1.22-new/sql/share/Makefile.in
--- mysql-4.1.22-old/sql/share/Makefile.in	2006-11-28 20:16:34.000000000 -0800
+++ mysql-4.1.22-new/sql/share/Makefile.in	2006-11-28 20:17:40.000000000 -0800
@@ -523,7 +523,7 @@
 fix_errors:
 	for lang in @AVAILABLE_LANGUAGES@; \
 	do \
-	../../extra/comp_err -C$(srcdir)/charsets/ $(srcdir)/$$lang/errmsg.txt $(srcdir)/$$lang/errmsg.sys; \
+	mysql-comp_err -C$(srcdir)/charsets/ $(srcdir)/$$lang/errmsg.txt $(srcdir)/$$lang/errmsg.sys; \
         done
 
 # Don't update the files from bitkeeper
