diff -Naur mysql-5.0.41-old/extra/Makefile.am mysql-5.0.41-new/extra/Makefile.am
--- mysql-5.0.41-old/extra/Makefile.am	2007-06-15 21:33:10.000000000 -0700
+++ mysql-5.0.41-new/extra/Makefile.am	2007-06-15 21:34:33.000000000 -0700
@@ -34,7 +34,7 @@
 $(top_builddir)/include/mysqld_error.h: comp_err.c \
 			$(top_srcdir)/sql/share/errmsg.txt
 	$(MAKE) $(AM_MAKEFLAGS) comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.41-old/extra/Makefile.in mysql-5.0.41-new/extra/Makefile.in
--- mysql-5.0.41-old/extra/Makefile.in	2007-06-15 21:33:10.000000000 -0700
+++ mysql-5.0.41-new/extra/Makefile.in	2007-06-15 21:34:33.000000000 -0700
@@ -898,7 +898,7 @@
 $(top_builddir)/include/mysqld_error.h: comp_err.c \
 			$(top_srcdir)/sql/share/errmsg.txt
 	$(MAKE) $(AM_MAKEFLAGS) comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err$(EXEEXT) \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.41-old/sql/Makefile.am mysql-5.0.41-new/sql/Makefile.am
--- mysql-5.0.41-old/sql/Makefile.am	2007-06-15 21:33:09.000000000 -0700
+++ mysql-5.0.41-new/sql/Makefile.am	2007-06-15 21:36:27.000000000 -0700
@@ -165,7 +165,7 @@
 # this avoid the rebuild of the built files in a source dist
 lex_hash.h:	gen_lex_hash.cc lex.h
 		$(MAKE) $(AM_MAKEFLAGS) gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@-t
+		mysql-gen_lex_hash > $@-t
 		$(MV) $@-t $@
 
 # For testing of udf_example.so
diff -Naur mysql-5.0.41-old/sql/Makefile.in mysql-5.0.41-new/sql/Makefile.in
--- mysql-5.0.41-old/sql/Makefile.in	2007-06-15 21:33:09.000000000 -0700
+++ mysql-5.0.41-new/sql/Makefile.in	2007-06-15 21:37:13.000000000 -0700
@@ -1222,7 +1222,7 @@
 # this avoid the rebuild of the built files in a source dist
 lex_hash.h:	gen_lex_hash.cc lex.h
 		$(MAKE) $(AM_MAKEFLAGS) gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@-t
+		mysql-gen_lex_hash > $@-t
 		$(MV) $@-t $@
 
 # Don't update the files from bitkeeper
