diff -Naur mysql-5.0.51a-old/extra/Makefile.am mysql-5.0.51a-new/extra/Makefile.am
--- mysql-5.0.51a-old/extra/Makefile.am	2008-02-26 11:31:42.000000000 -0800
+++ mysql-5.0.51a-new/extra/Makefile.am	2008-02-26 11:32:18.000000000 -0800
@@ -34,7 +34,7 @@
 $(top_builddir)/include/mysqld_error.h: comp_err.c \
 			$(top_srcdir)/sql/share/errmsg.txt
 	$(MAKE) $(AM_MAKEFLAGS) comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.51a-old/extra/Makefile.in mysql-5.0.51a-new/extra/Makefile.in
--- mysql-5.0.51a-old/extra/Makefile.in	2008-02-26 11:31:42.000000000 -0800
+++ mysql-5.0.51a-new/extra/Makefile.in	2008-02-26 11:32:18.000000000 -0800
@@ -900,7 +900,7 @@
 $(top_builddir)/include/mysqld_error.h: comp_err.c \
 			$(top_srcdir)/sql/share/errmsg.txt
 	$(MAKE) $(AM_MAKEFLAGS) comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err$(EXEEXT) \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.51a-old/sql/Makefile.am mysql-5.0.51a-new/sql/Makefile.am
--- mysql-5.0.51a-old/sql/Makefile.am	2008-02-26 11:31:42.000000000 -0800
+++ mysql-5.0.51a-new/sql/Makefile.am	2008-02-26 11:32:18.000000000 -0800
@@ -166,7 +166,7 @@
 # this avoid the rebuild of the built files in a source dist
 lex_hash.h:	gen_lex_hash.cc lex.h
 		$(MAKE) $(AM_MAKEFLAGS) gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@-t
+		mysql-gen_lex_hash > $@-t
 		$(MV) $@-t $@
 
 # For testing of udf_example.so
diff -Naur mysql-5.0.51a-old/sql/Makefile.in mysql-5.0.51a-new/sql/Makefile.in
--- mysql-5.0.51a-old/sql/Makefile.in	2008-02-26 11:31:42.000000000 -0800
+++ mysql-5.0.51a-new/sql/Makefile.in	2008-02-26 11:32:18.000000000 -0800
@@ -1227,7 +1227,7 @@
 # this avoid the rebuild of the built files in a source dist
 lex_hash.h:	gen_lex_hash.cc lex.h
 		$(MAKE) $(AM_MAKEFLAGS) gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@-t
+		mysql-gen_lex_hash > $@-t
 		$(MV) $@-t $@
 
 # Don't update the files from bitkeeper
