diff -Naur mysql-5.0.17-old/configure mysql-5.0.17-new/configure
--- mysql-5.0.17-old/configure	2005-12-14 15:35:25.000000000 -0800
+++ mysql-5.0.17-new/configure	2005-12-30 06:06:46.000000000 -0800
@@ -20838,7 +20838,7 @@
 echo "${ECHO_T}\"int\"" >&6
     ;;
   *)
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -20887,7 +20887,7 @@
 sed 's/^/| /' conftest.$ac_ext >&5
 
 ( exit $ac_status )
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -25575,7 +25575,7 @@
 echo $ECHO_N "checking for atomic operations... $ECHO_C" >&6
 
   atom_ops=
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -25625,7 +25625,7 @@
 fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -26054,7 +26054,7 @@
 echo "${ECHO_T}no" >&6
     ;;
   *)
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -27700,7 +27700,7 @@
 if test "${ac_cv_sys_restartable_syscalls+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -32493,7 +32493,7 @@
 if test "${ac_cv_conv_longlong_to_float+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = gar; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
diff -Naur mysql-5.0.17-old/extra/Makefile.in mysql-5.0.17-new/extra/Makefile.in
--- mysql-5.0.17-old/extra/Makefile.in	2005-12-14 15:34:50.000000000 -0800
+++ mysql-5.0.17-new/extra/Makefile.in	2005-12-30 06:03:22.000000000 -0800
@@ -871,7 +871,7 @@
 
 # This will build mysqld_error.h and sql_state.h
 $(top_builddir)/include/mysqld_error.h: comp_err
-	$(top_builddir)/extra/comp_err \
+	mysql-comp_err
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.17-old/sql/Makefile.in mysql-5.0.17-new/sql/Makefile.in
--- mysql-5.0.17-old/sql/Makefile.in	2005-12-14 15:35:20.000000000 -0800
+++ mysql-5.0.17-new/sql/Makefile.in	2005-12-30 06:04:38.000000000 -0800
@@ -1254,7 +1254,7 @@
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
 lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+		mysql-gen_lex_hash$(EXEEXT) > $@
 
 # For testing of udf_example.so;  Works on platforms with gcc
 # (This is not part of our build process but only provided as an example)
