diff -Naur mysql-5.0.19-old/extra/Makefile.am mysql-5.0.19-new/extra/Makefile.am
--- mysql-5.0.19-old/extra/Makefile.am	2006-03-04 12:35:13.000000000 -0800
+++ mysql-5.0.19-new/extra/Makefile.am	2006-03-16 19:36:21.000000000 -0800
@@ -30,7 +30,7 @@
 
 # This will build mysqld_error.h and sql_state.h
 $(top_builddir)/include/mysqld_error.h: comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.19-old/extra/Makefile.in mysql-5.0.19-new/extra/Makefile.in
--- mysql-5.0.19-old/extra/Makefile.in	2006-03-04 12:36:35.000000000 -0800
+++ mysql-5.0.19-new/extra/Makefile.in	2006-03-16 19:36:21.000000000 -0800
@@ -874,7 +874,7 @@
 
 # This will build mysqld_error.h and sql_state.h
 $(top_builddir)/include/mysqld_error.h: comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.19-old/sql/Makefile.am mysql-5.0.19-new/sql/Makefile.am
--- mysql-5.0.19-old/sql/Makefile.am	2006-03-04 12:34:51.000000000 -0800
+++ mysql-5.0.19-new/sql/Makefile.am	2006-03-16 19:36:21.000000000 -0800
@@ -151,7 +151,7 @@
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
 lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+		mysql-gen_lex_hash > $@
 
 # For testing of udf_example.so;  Works on platforms with gcc
 # (This is not part of our build process but only provided as an example)
diff -Naur mysql-5.0.19-old/sql/Makefile.in mysql-5.0.19-new/sql/Makefile.in
--- mysql-5.0.19-old/sql/Makefile.in	2006-03-04 12:37:00.000000000 -0800
+++ mysql-5.0.19-new/sql/Makefile.in	2006-03-16 19:36:21.000000000 -0800
@@ -1260,7 +1260,7 @@
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
 lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+		mysql-gen_lex_hash > $@
 
 # For testing of udf_example.so;  Works on platforms with gcc
 # (This is not part of our build process but only provided as an example)
