diff -Naur webfs-1.21-old/ls.c webfs-1.21-new/ls.c
--- webfs-1.21-old/ls.c	2004-06-10 02:45:50.000000000 -0700
+++ webfs-1.21-new/ls.c	2007-04-23 14:18:53.000000000 -0700
@@ -265,7 +265,10 @@
     len  = 0;
 
     len += sprintf(buf+len,
-		   "<head><title>%s:%d%s</title></head>\n"
+		   "<head>\n"
+		   "  <title>%s:%d%s</title>\n"
+		   "  <meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />"
+		   "</head>\n"
 		   "<body bgcolor=white text=black link=darkblue vlink=firebrick alink=red>\n"
 		   "<h1>listing: \n",
 		   hostname,tcp_port,path);
