diff -Naur alsa-driver-1.0.16-old/alsa-kernel/pci/hda/patch_realtek.c alsa-driver-1.0.16-new/alsa-kernel/pci/hda/patch_realtek.c
--- alsa-driver-1.0.16-old/alsa-kernel/pci/hda/patch_realtek.c	2008-02-05 01:23:24.000000000 -0800
+++ alsa-driver-1.0.16-new/alsa-kernel/pci/hda/patch_realtek.c	2008-04-11 05:32:14.000000000 -0700
@@ -11619,7 +11619,7 @@
 	SND_PCI_QUIRK(0x1043, 0x1338, "ASUS F2/3", ALC861_ASUS_LAPTOP),
 	SND_PCI_QUIRK(0x1043, 0x1393, "ASUS", ALC861_ASUS),
 	SND_PCI_QUIRK(0x1043, 0x13d7, "ASUS A9rp", ALC861_ASUS_LAPTOP),
-	SND_PCI_QUIRK(0x1043, 0x81cb, "ASUS P1-AH2", ALC861_3ST_DIG),
+	/* SND_PCI_QUIRK(0x1043, 0x81cb, "ASUS P1-AH2", ALC861_3ST_DIG), */
 	SND_PCI_QUIRK(0x1179, 0xff00, "Toshiba", ALC861_TOSHIBA),
 	/* FIXME: the entry below breaks Toshiba A100 (model=auto works!)
 	 *        Any other models that need this preset?
