diff -Naur ffmpeg-25753.32628-old/libavutil/common.h ffmpeg-25753.32628-new/libavutil/common.h
--- ffmpeg-25753.32628-old/libavutil/common.h	2010-11-14 13:24:36.000000000 -0800
+++ ffmpeg-25753.32628-new/libavutil/common.h	2010-11-14 13:29:15.000000000 -0800
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
