diff -Naur ffmpeg-24030.31628-old/libavutil/common.h ffmpeg-24030.31628-new/libavutil/common.h
--- ffmpeg-24030.31628-old/libavutil/common.h	2010-07-03 13:11:02.000000000 -0700
+++ ffmpeg-24030.31628-new/libavutil/common.h	2010-07-03 13:11:45.000000000 -0700
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
