diff -Naur ffmpeg-26400.32793-old/libavutil/common.h ffmpeg-26400.32793-new/libavutil/common.h
--- ffmpeg-26400.32793-old/libavutil/common.h	2011-01-17 21:26:45.000000000 -0800
+++ ffmpeg-26400.32793-new/libavutil/common.h	2011-01-17 21:28:36.000000000 -0800
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
