diff -Naur ffmpeg-16117.28147-old/configure ffmpeg-16117.28147-new/configure
--- ffmpeg-16117.28147-old/configure	2008-12-13 17:13:59.000000000 -0800
+++ ffmpeg-16117.28147-new/configure	2008-12-13 17:14:57.000000000 -0800
@@ -148,6 +148,7 @@
   echo "  --disable-neon           disable neon optimizations"
   echo "  --disable-vis            disable VIS optimizations"
   echo "  --disable-vdpau          disable VDPAU support"
+  echo "  --disable-xvmc           disable XvMC support"
   echo "  --disable-network        disable network support [no]"
   echo "  --disable-ipv6           disable IPv6 support [no]"
   echo "  --disable-mpegaudio-hp   faster (but less accurate)"
@@ -817,6 +818,7 @@
     ssse3
     vis
     vdpau
+    xvmc
 '
 
 HAVE_LIST="
@@ -1899,6 +1901,7 @@
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
 enabled vdpau      && require  libvdpau vdpau/vdpau_x11.h vdp_device_create_x11 -lvdpau
+enabled xvmc       && check_header X11/extensions/XvMClib.h
 
 # libdc1394 check
 if enabled libdc1394; then
