diff -Naur ffmpeg-e781c4e-old/libavcodec/vaapi_vc1.c ffmpeg-e781c4e-new/libavcodec/vaapi_vc1.c
--- ffmpeg-e781c4e-old/libavcodec/vaapi_vc1.c	2011-01-25 09:05:07.000000000 -0800
+++ ffmpeg-e781c4e-new/libavcodec/vaapi_vc1.c	2011-01-27 11:52:52.000000000 -0800
@@ -160,6 +160,7 @@
     pic_param->sequence_fields.bits.syncmarker                      = s->resync_marker;
     pic_param->sequence_fields.bits.rangered                        = v->rangered;
     pic_param->sequence_fields.bits.max_b_frames                    = s->avctx->max_b_frames;
+    pic_param->sequence_fields.bits.profile                         = v->profile;
     pic_param->coded_width                                          = s->avctx->coded_width;
     pic_param->coded_height                                         = s->avctx->coded_height;
     pic_param->entrypoint_fields.value                              = 0; /* reset all bits */
