diff -Naur ffmpeg-12d2765-old/configure ffmpeg-12d2765-new/configure
--- ffmpeg-12d2765-old/configure	2011-11-25 10:44:54.000000000 -0800
+++ ffmpeg-12d2765-new/configure	2011-11-25 10:45:01.000000000 -0800
@@ -1980,7 +1980,7 @@
 cc_default="${cross_prefix}${cc_default}"
 cxx_default="${cross_prefix}${cxx_default}"
 nm_default="${cross_prefix}${nm_default}"
-pkg_config_default="${cross_prefix}${pkg_config_default}"
+#pkg_config_default="${cross_prefix}${pkg_config_default}"
 ranlib="${cross_prefix}${ranlib}"
 strip_default="${cross_prefix}${strip_default}"
 
