diff -Naur ffmpeg-16117.28147-old/configure ffmpeg-16117.28147-new/configure
--- ffmpeg-16117.28147-old/configure	2008-12-13 17:12:51.000000000 -0800
+++ ffmpeg-16117.28147-new/configure	2008-12-13 17:13:59.000000000 -0800
@@ -147,6 +147,7 @@
   echo "  --disable-mmi            disable MMI optimizations"
   echo "  --disable-neon           disable neon optimizations"
   echo "  --disable-vis            disable VIS optimizations"
+  echo "  --disable-vdpau          disable VDPAU support"
   echo "  --disable-network        disable network support [no]"
   echo "  --disable-ipv6           disable IPv6 support [no]"
   echo "  --disable-mpegaudio-hp   faster (but less accurate)"
@@ -815,6 +816,7 @@
     neon
     ssse3
     vis
+    vdpau
 '
 
 HAVE_LIST="
@@ -1014,6 +1016,7 @@
 zmbv_encoder_select="zlib"
 
 # external libraries
+h264_vdpau_decoder_deps="vdpau"
 libamr_nb_decoder_deps="libamr_nb"
 libamr_nb_encoder_deps="libamr_nb"
 libamr_wb_decoder_deps="libamr_wb"
@@ -1036,6 +1039,9 @@
 libx264_encoder_deps="libx264"
 libxvid_encoder_deps="libxvid"
 mpeg4aac_decoder_deps="libfaad"
+mpeg_vdpau_decoder_deps="vdpau"
+wmv3_vdpau_decoder_deps="vdpau"
+vc1_vdpau_decoder_deps="vdpau"
 
 # demuxers / muxers
 ac3_demuxer_deps="ac3_parser"
@@ -1892,6 +1898,7 @@
                         die "ERROR: libx264 version must be >= 0.65."; }
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
+enabled vdpau      && require  libvdpau vdpau/vdpau_x11.h vdp_device_create_x11 -lvdpau
 
 # libdc1394 check
 if enabled libdc1394; then
