diff -Naur ffmpeg-22445.30881-old/libswscale/libswscale.v ffmpeg-22445.30881-new/libswscale/libswscale.v
--- ffmpeg-22445.30881-old/libswscale/libswscale.v	2010-03-10 17:35:58.000000000 -0800
+++ ffmpeg-22445.30881-new/libswscale/libswscale.v	2010-03-10 17:36:04.000000000 -0800
@@ -1,4 +1,4 @@
 LIBSWSCALE_$MAJOR {
-        global: swscale_*; sws_*; ff_*;
+        global: swscale_*; sws_*; ff_*; palette8tobgr15; palette8tobgr16; palette8torgb15; palette8torgb16; palette8topacked24; palette8topacked32; rgb24tobgr24; rgb32tobgr32; yuv422ptoyuy2; yv12toyuy2;
         local: *;
 };
