diff -Naur ffmpeg-7ac1674-old/libavutil/common.h ffmpeg-7ac1674-new/libavutil/common.h
--- ffmpeg-7ac1674-old/libavutil/common.h	2011-03-31 21:03:02.000000000 -0700
+++ ffmpeg-7ac1674-new/libavutil/common.h	2011-03-31 21:03:03.000000000 -0700
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
