diff -Naur ffmpeg-8cf9a09-old/libavutil/common.h ffmpeg-8cf9a09-new/libavutil/common.h
--- ffmpeg-8cf9a09-old/libavutil/common.h	2011-03-03 21:02:53.000000000 -0800
+++ ffmpeg-8cf9a09-new/libavutil/common.h	2011-03-03 21:07:31.000000000 -0800
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
