diff -Naur ffmpeg-d4a50a2-old/configure ffmpeg-d4a50a2-new/configure
--- ffmpeg-d4a50a2-old/configure	2011-03-20 21:07:22.000000000 -0700
+++ ffmpeg-d4a50a2-new/configure	2011-03-20 21:07:30.000000000 -0700
@@ -1830,7 +1830,7 @@
 ar_default="${cross_prefix}${ar_default}"
 cc_default="${cross_prefix}${cc_default}"
 nm_default="${cross_prefix}${nm_default}"
-pkg_config_default="${cross_prefix}${pkg_config_default}"
+#pkg_config_default="${cross_prefix}${pkg_config_default}"
 ranlib="${cross_prefix}${ranlib}"
 strip_default="${cross_prefix}${strip_default}"
 
