diff -Naur NVIDIA-Linux-x86_64-180.37-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-180.37-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-180.37-pkg2-old/usr/src/nv/conftest.sh	2009-03-06 01:08:25.000000000 -0800
+++ NVIDIA-Linux-x86_64-180.37-pkg2-new/usr/src/nv/conftest.sh	2009-03-06 13:27:00.000000000 -0800
@@ -1166,7 +1166,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
