diff -Naur linux-2.6.12.3-old/drivers/net/via-rhine.c linux-2.6.12.3-new/drivers/net/via-rhine.c
--- linux-2.6.12.3-old/drivers/net/via-rhine.c	2005-07-15 14:18:57.000000000 -0700
+++ linux-2.6.12.3-new/drivers/net/via-rhine.c	2005-07-15 18:45:55.000000000 -0700
@@ -1934,9 +1934,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
