diff -Naur linux-2.6.14.7-old/drivers/net/via-rhine.c linux-2.6.14.7-new/drivers/net/via-rhine.c
--- linux-2.6.14.7-old/drivers/net/via-rhine.c	2006-01-30 22:52:07.000000000 -0800
+++ linux-2.6.14.7-new/drivers/net/via-rhine.c	2006-02-04 03:35:14.000000000 -0800
@@ -1934,9 +1934,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
