diff -Naur linux-2.6.10-old/drivers/usb/Makefile linux-2.6.10-new/drivers/usb/Makefile
--- linux-2.6.10-old/drivers/usb/Makefile	2004-12-24 13:35:28.000000000 -0800
+++ linux-2.6.10-new/drivers/usb/Makefile	2004-12-28 07:31:02.136964890 -0800
@@ -27,6 +27,7 @@
 obj-$(CONFIG_USB_KBTAB)		+= input/
 obj-$(CONFIG_USB_MOUSE)		+= input/
 obj-$(CONFIG_USB_MTOUCH)	+= input/
+obj-$(CONFIG_USB_EGALAX)	+= input/
 obj-$(CONFIG_USB_POWERMATE)	+= input/
 obj-$(CONFIG_USB_WACOM)		+= input/
 obj-$(CONFIG_USB_XPAD)		+= input/
