diff -Naur mythtv-0.21-19443-old/libs/libmythdvdnav/dvd_reader.c mythtv-0.21-19443-new/libs/libmythdvdnav/dvd_reader.c
--- mythtv-0.21-19443-old/libs/libmythdvdnav/dvd_reader.c	2008-12-23 19:49:24.000000000 -0800
+++ mythtv-0.21-19443-new/libs/libmythdvdnav/dvd_reader.c	2008-12-23 19:49:28.000000000 -0800
@@ -1021,6 +1021,28 @@
     return offset;
 }
 
+int DVDFileSeekForce( dvd_file_t *dvd_file, int offset, int force_size )
+{
+  /* Check arguments. */
+  if( dvd_file == NULL || offset < 0 )
+    return -1;
+
+  if( dvd_file->dvd->isImageFile ) {
+    if( force_size < 0 )
+      force_size = (offset - 1) / DVD_VIDEO_LB_LEN + 1;
+    if( dvd_file->filesize < force_size ) {
+      dvd_file->filesize = force_size;
+      fprintf(stderr, "libdvdread: Ignored UDF provided size of file.\n");
+    }
+  }
+
+  if( offset > dvd_file->filesize * DVD_VIDEO_LB_LEN ) {
+    return -1;
+  }
+  dvd_file->seek_pos = (uint32_t) offset;
+  return offset;
+}
+
 ssize_t DVDReadBytes( dvd_file_t *dvd_file, void *data, size_t byte_size )
 {
     unsigned char *secbuf_base, *secbuf;
@@ -1061,7 +1083,7 @@
     memcpy( data, &(secbuf[ seek_byte ]), byte_size );
     free( secbuf_base );
 
-    dvd_file->seek_pos += byte_size;
+    DVDFileSeekForce(dvd_file, dvd_file->seek_pos + byte_size, -1);
     return byte_size;
 }
 
diff -Naur mythtv-0.21-19443-old/libs/libmythdvdnav/dvd_reader.h mythtv-0.21-19443-new/libs/libmythdvdnav/dvd_reader.h
--- mythtv-0.21-19443-old/libs/libmythdvdnav/dvd_reader.h	2008-12-23 19:49:24.000000000 -0800
+++ mythtv-0.21-19443-new/libs/libmythdvdnav/dvd_reader.h	2008-12-23 19:49:28.000000000 -0800
@@ -229,6 +229,8 @@
 int DVDUDFVolumeInfo( dvd_reader_t *, char *, unsigned int,
 		      unsigned char *, unsigned int );
 
+int DVDFileSeekForce( dvd_file_t *, int, int );
+
 /**
  * Get the ISO9660 VolumeIdentifier and VolumeSetIdentifier
  *
diff -Naur mythtv-0.21-19443-old/libs/libmythdvdnav/ifo_read.c mythtv-0.21-19443-new/libs/libmythdvdnav/ifo_read.c
--- mythtv-0.21-19443-old/libs/libmythdvdnav/ifo_read.c	2008-12-23 19:49:24.000000000 -0800
+++ mythtv-0.21-19443-new/libs/libmythdvdnav/ifo_read.c	2008-12-23 19:49:28.000000000 -0800
@@ -93,6 +93,10 @@
   return (DVDFileSeek(dvd_file, (int)offset) == (int)offset);
 }
 
+static inline int DVDFileSeekForce_( dvd_file_t *dvd_file, uint32_t offset, int force_size ) {
+  return (DVDFileSeekForce(dvd_file, (int)offset, force_size) == (int)offset);
+}
+
 
 ifo_handle_t *ifoOpen(dvd_reader_t *dvd, int title) {
   ifo_handle_t *ifofile;
@@ -1503,7 +1507,7 @@
   unsigned int i;
   int info_length;
 
-  if(!DVDFileSeek_(ifofile->file, sector * DVD_BLOCK_LEN))
+  if(!DVDFileSeekForce_(ifofile->file, sector * DVD_BLOCK_LEN, sector))
     return 0;
 
   if(!(DVDReadBytes(ifofile->file, vobu_admap, VOBU_ADMAP_SIZE)))
