diff -Naur mythtv-0.21-20671-old/libs/libmythtv/vsync.cpp mythtv-0.21-20671-new/libs/libmythtv/vsync.cpp
--- mythtv-0.21-20671-old/libs/libmythtv/vsync.cpp	2009-06-04 08:59:34.000000000 -0700
+++ mythtv-0.21-20671-new/libs/libmythtv/vsync.cpp	2009-06-04 08:59:35.000000000 -0700
@@ -96,7 +96,7 @@
 #endif    
 #ifndef _WIN32
     TESTVIDEOSYNC(nVidiaVideoSync);
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
