diff -Naur mythtv-0.21-20312-old/configure mythtv-0.21-20312-new/configure
--- mythtv-0.21-20312-old/configure	2009-04-06 14:02:37.000000000 -0700
+++ mythtv-0.21-20312-new/configure	2009-04-06 14:02:38.000000000 -0700
@@ -2715,7 +2715,7 @@
     fi
 fi
 
-enabled libfftw3 && has_library libfftw3_threads && has_header fftw3.h ||
+enabled libfftw3 && has_library libfftw3f_threads && has_header fftw3.h ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
