diff -Naur mythtv-0.21-20769-old/libs/libmythtv/vsync.cpp mythtv-0.21-20769-new/libs/libmythtv/vsync.cpp
--- mythtv-0.21-20769-old/libs/libmythtv/vsync.cpp	2009-06-30 16:13:15.000000000 -0700
+++ mythtv-0.21-20769-new/libs/libmythtv/vsync.cpp	2009-06-30 16:13:16.000000000 -0700
@@ -96,7 +96,7 @@
 #endif    
 #ifndef _WIN32
     TESTVIDEOSYNC(nVidiaVideoSync);
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
