diff -Naur mythtv-0.21-19344-old/libs/libmythtv/tv_play.cpp mythtv-0.21-19344-new/libs/libmythtv/tv_play.cpp
--- mythtv-0.21-19344-old/libs/libmythtv/tv_play.cpp	2008-12-11 19:52:00.000000000 -0800
+++ mythtv-0.21-19344-new/libs/libmythtv/tv_play.cpp	2008-12-11 19:52:00.000000000 -0800
@@ -1789,7 +1789,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
