diff -Naur mythtv-0.21-21185-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.21-21185-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.21-21185-old/libs/libmyth/mediamonitor-unix.cpp	2009-08-09 10:53:40.000000000 -0700
+++ mythtv-0.21-21185-new/libs/libmyth/mediamonitor-unix.cpp	2009-08-09 10:53:41.000000000 -0700
@@ -184,7 +184,8 @@
 {
 #ifdef linux
     QProcess udevinfo(this);
-    udevinfo.addArgument("udevinfo");
+    udevinfo.addArgument("udevadm");
+    udevinfo.addArgument("info");
     udevinfo.addArgument("-q");
     udevinfo.addArgument("name");
     udevinfo.addArgument("-rp");
