diff -Naur mythtv-0.21-20622-old/themes/cpsvndir mythtv-0.21-20622-new/themes/cpsvndir
--- mythtv-0.21-20622-old/themes/cpsvndir	2009-05-25 21:11:01.000000000 -0700
+++ mythtv-0.21-20622-new/themes/cpsvndir	2009-05-25 21:11:02.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
