diff -Naur mythtv-0.21-17684-old/libs/libmyth/mythcontext.cpp mythtv-0.21-17684-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.21-17684-old/libs/libmyth/mythcontext.cpp	2008-07-01 09:17:27.000000000 -0700
+++ mythtv-0.21-17684-new/libs/libmyth/mythcontext.cpp	2008-07-01 09:17:35.000000000 -0700
@@ -1804,7 +1804,7 @@
 #if defined(CONFIG_DARWIN) || (__FreeBSD__) || defined(__OpenBSD__)
     char *command = "ps -ax | grep -i mythbackend | grep -v grep > /dev/null";
 #else
-    char *command = "ps -ae | grep mythbackend > /dev/null";
+    char *command = "ps | grep mythbackend | grep -v grep > /dev/null";
 #endif
     bool res = myth_system(command,
                            MYTH_SYSTEM_DONT_BLOCK_LIRC |
