diff -Naur mythtv-0.21-18228-old/libs/libmyth/mythmediamonitor.cpp mythtv-0.21-18228-new/libs/libmyth/mythmediamonitor.cpp
--- mythtv-0.21-18228-old/libs/libmyth/mythmediamonitor.cpp	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/libs/libmyth/mythmediamonitor.cpp	2008-09-01 09:25:39.000000000 -0700
@@ -598,6 +598,12 @@
         VERBOSE(VB_MEDIA, "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getDevicePath() == "/dev/disk-minimyth")
+    {
+        VERBOSE(VB_MEDIA, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
diff -Naur mythtv-0.21-18228-old/libs/libmythtv/videodisplayprofile.cpp mythtv-0.21-18228-new/libs/libmythtv/videodisplayprofile.cpp
--- mythtv-0.21-18228-old/libs/libmythtv/videodisplayprofile.cpp	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/libs/libmythtv/videodisplayprofile.cpp	2008-09-01 09:25:43.000000000 -0700
@@ -1054,6 +1054,16 @@
                   "linearblend", "linearblend", "");
 }
 
+void VideoDisplayProfile::CreateMiniMythProfiles(const QString &hostname)
+{
+    (void) QObject::tr("MiniMyth", "MiniMyth");
+    DeleteProfileGroup("MiniMyth", hostname);
+    uint groupid = CreateProfileGroup("MiniMyth", hostname);
+    CreateProfile(groupid, 1, ">", 0, 0, "", 0, 0,
+                  "ffmpeg", 1, "xv-blit", "softblend", false,
+                  "bobdeint", "onefield", "");
+}
+
 void VideoDisplayProfile::CreateProfiles(const QString &hostname)
 {
     CreateOldProfiles(hostname);
diff -Naur mythtv-0.21-18228-old/libs/libmythtv/videodisplayprofile.h mythtv-0.21-18228-new/libs/libmythtv/videodisplayprofile.h
--- mythtv-0.21-18228-old/libs/libmythtv/videodisplayprofile.h	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/libs/libmythtv/videodisplayprofile.h	2008-09-01 09:25:43.000000000 -0700
@@ -133,6 +133,7 @@
     static void        CreateProfiles(const QString &hostname);
     static void        CreateOldProfiles(const QString &hostname);
     static void        CreateNewProfiles(const QString &hostname);
+    static void        CreateMiniMythProfiles(const QString &hostname);
 
     static QStringList GetVideoRenderers(const QString &decoder);
     static QString     GetVideoRendererHelp(const QString &renderer);
diff -Naur mythtv-0.21-18228-old/programs/mythfrontend/globalsettings.cpp mythtv-0.21-18228-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.21-18228-old/programs/mythfrontend/globalsettings.cpp	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/programs/mythfrontend/globalsettings.cpp	2008-09-01 09:25:43.000000000 -0700
@@ -1361,6 +1361,11 @@
     if (profiles.empty())
         return;
 
+    if (!profiles.contains("MiniMyth"))
+    {
+        VideoDisplayProfile::CreateMiniMythProfiles(host);
+        profiles = VideoDisplayProfile::GetProfiles(host);
+    }
     if (!profiles.contains("Normal") &&
         !profiles.contains("High Quality") &&
         !profiles.contains("Slim"))
diff -Naur mythtv-0.21-18228-old/programs/mythfrontend/minimyth_tools.xml mythtv-0.21-18228-new/programs/mythfrontend/minimyth_tools.xml
--- mythtv-0.21-18228-old/programs/mythfrontend/minimyth_tools.xml	1969-12-31 16:00:00.000000000 -0800
+++ mythtv-0.21-18228-new/programs/mythfrontend/minimyth_tools.xml	2008-09-01 09:25:43.000000000 -0700
@@ -0,0 +1,27 @@
+<mythmenu name="MINIMYTH_TOOLS">
+
+   <button>
+      <type>MINIMYTH_GAME_SAVE</type>
+      <text>Save Game Settings</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command game_save</action>
+   </button>
+
+   <button>
+      <type>MINIMYTH_GAME_RESTORE</type>
+      <text>Restore Game Settings</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command game_restore</action>
+   </button>
+
+   <button>
+      <type>MINIMYTH_THEMECACHE_SAVE</type>
+      <text>Save Themecache</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command themecache_save</action>
+   </button>
+
+   <button>
+      <type>MINIMYTH_CODECS_SAVE</type>
+      <text>Fetch and Save Codecs</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command codecs_fetch_and_save</action>
+   </button>
+
+</mythmenu>
diff -Naur mythtv-0.21-18228-old/programs/mythfrontend/mythfrontend.pro mythtv-0.21-18228-new/programs/mythfrontend/mythfrontend.pro
--- mythtv-0.21-18228-old/programs/mythfrontend/mythfrontend.pro	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/programs/mythfrontend/mythfrontend.pro	2008-09-01 09:25:43.000000000 -0700
@@ -16,6 +16,7 @@
 setting.files += recpriorities_settings.xml tv_search.xml tv_lists.xml
 setting.files += library.xml manage_recordings.xml optical_menu.xml tvmenu.xml
 setting.files += tv_settings.xml
+setting.files += minimyth_tools.xml
 setting.extra = -ldconfig
 
 INSTALLS += setting
diff -Naur mythtv-0.21-18228-old/programs/mythfrontend/util_menu.xml mythtv-0.21-18228-new/programs/mythfrontend/util_menu.xml
--- mythtv-0.21-18228-old/programs/mythfrontend/util_menu.xml	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/programs/mythfrontend/util_menu.xml	2008-09-01 09:25:43.000000000 -0700
@@ -1,6 +1,12 @@
 <mythmenu name="UTIL">
 
    <button>
+      <type>MINIMYTH_TOOLS</type>
+      <text>MiniMyth Tools</text>
+      <action>MENU minimyth_tools.xml</action>
+   </button>
+
+   <button>
       <type>MUSIC_PLAYLIST</type>
       <text>Music Tools</text>
       <text lang="IT">Opzioni Musica</text>
diff -Naur mythtv-0.21-18228-old/themes/classic/mainmenu.xml mythtv-0.21-18228-new/themes/classic/mainmenu.xml
--- mythtv-0.21-18228-old/themes/classic/mainmenu.xml	2008-09-01 09:25:32.000000000 -0700
+++ mythtv-0.21-18228-new/themes/classic/mainmenu.xml	2008-09-01 09:25:43.000000000 -0700
@@ -288,6 +288,12 @@
    </button>
 
    <button>
+      <type>MINIMYTH_TOOLS</type>
+      <text>MiniMyth Tools</text>
+      <action>MENU minimyth_tools.xml</action>
+   </button>
+
+   <button>
      <type>SETUP</type>
      <text>Setup</text>
      <text lang="IT">Impostazioni</text>
diff -Naur mythtv-0.21-18228-old/themes/classic/minimyth_tools.xml mythtv-0.21-18228-new/themes/classic/minimyth_tools.xml
--- mythtv-0.21-18228-old/themes/classic/minimyth_tools.xml	1969-12-31 16:00:00.000000000 -0800
+++ mythtv-0.21-18228-new/themes/classic/minimyth_tools.xml	2008-09-01 09:25:43.000000000 -0700
@@ -0,0 +1,27 @@
+<mythmenu name="MINIMYTH_TOOLS">
+
+   <button>
+      <type>MINIMYTH_GAME_SAVE</type>
+      <text>Save Game Settings</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command game_save</action>
+   </button>
+
+   <button>
+      <type>MINIMYTH_GAME_RESTORE</type>
+      <text>Restore Game Settings</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command game_restore</action>
+   </button>
+
+   <button>
+      <type>MINIMYTH_THEMECACHE_SAVE</type>
+      <text>Save Themecache</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command themecache_save</action>
+   </button>
+
+   <button>
+      <type>MINIMYTH_CODECS_SAVE</type>
+      <text>Fetch and Save Codecs</text>
+      <action>EXEC /usr/bin/mm_term_start /usr/bin/mm_command codecs_fetch_and_save</action>
+   </button>
+
+</mythmenu>
