diff -Naur mythtv-0.21-21276-old/configure mythtv-0.21-21276-new/configure
--- mythtv-0.21-21276-old/configure	2009-08-15 13:15:16.000000000 -0700
+++ mythtv-0.21-21276-new/configure	2009-08-15 13:15:16.000000000 -0700
@@ -2726,7 +2726,7 @@
     fi
 fi
 
-enabled libfftw3 && has_library libfftw3_threads && has_header fftw3.h ||
+enabled libfftw3 && has_library libfftw3f_threads && has_header fftw3.h ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
