diff -Naur mythtv-0.21-21074-old/libs/libmythtv/vsync.cpp mythtv-0.21-21074-new/libs/libmythtv/vsync.cpp
--- mythtv-0.21-21074-old/libs/libmythtv/vsync.cpp	2009-08-01 10:11:31.000000000 -0700
+++ mythtv-0.21-21074-new/libs/libmythtv/vsync.cpp	2009-08-01 10:11:32.000000000 -0700
@@ -96,7 +96,7 @@
 #endif    
 #ifndef _WIN32
     TESTVIDEOSYNC(nVidiaVideoSync);
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
