diff -Naur mythtv-0.21-18831-old/libs/libmythtv/tv_play.cpp mythtv-0.21-18831-new/libs/libmythtv/tv_play.cpp
--- mythtv-0.21-18831-old/libs/libmythtv/tv_play.cpp	2008-10-23 15:52:14.000000000 -0700
+++ mythtv-0.21-18831-new/libs/libmythtv/tv_play.cpp	2008-10-23 15:52:32.000000000 -0700
@@ -1789,7 +1789,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
