diff -Naur mythtv-0.21-19046-old/configure mythtv-0.21-19046-new/configure
--- mythtv-0.21-19046-old/configure	2008-11-13 18:28:06.000000000 -0800
+++ mythtv-0.21-19046-new/configure	2008-11-13 18:28:11.000000000 -0800
@@ -2714,7 +2714,7 @@
     fi
 fi
 
-enabled libfftw3 && has_library libfftw3_threads && has_header fftw3.h ||
+enabled libfftw3 && has_library libfftw3f_threads && has_header fftw3.h ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
