diff -Naur mythtv-0.21-25121-old/libs/libmythtv/avformatdecoder.cpp mythtv-0.21-25121-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.21-25121-old/libs/libmythtv/avformatdecoder.cpp	2010-09-01 07:01:32.000000000 -0700
+++ mythtv-0.21-25121-new/libs/libmythtv/avformatdecoder.cpp	2010-09-01 07:02:32.000000000 -0700
@@ -1624,6 +1624,9 @@
                 if (video_codec_id > kCodec_NORMAL_END)
                     thread_count = 1;
 
+                if (enc->codec_id == CODEC_ID_SVQ3)
+                    thread_count = 1;
+
                 VERBOSE(VB_PLAYBACK, QString("Using %1 CPUs for decoding")
                         .arg(ENABLE_THREADS ? thread_count : 1));
 
