diff -Naur mythtv-0.21-20592-old/configure mythtv-0.21-20592-new/configure
--- mythtv-0.21-20592-old/configure	2009-05-19 08:07:55.000000000 -0700
+++ mythtv-0.21-20592-new/configure	2009-05-19 08:07:58.000000000 -0700
@@ -2724,7 +2724,7 @@
     fi
 fi
 
-enabled libfftw3 && has_library libfftw3_threads && has_header fftw3.h ||
+enabled libfftw3 && has_library libfftw3f_threads && has_header fftw3.h ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
