diff -Naur mythtv-0.21-16838-old/libs/libmyth/util.cpp mythtv-0.21-16838-new/libs/libmyth/util.cpp
--- mythtv-0.21-16838-old/libs/libmyth/util.cpp	2008-03-29 22:00:30.000000000 -0700
+++ mythtv-0.21-16838-new/libs/libmyth/util.cpp	2008-03-29 22:00:30.000000000 -0700
@@ -293,10 +293,11 @@
                     return GENERIC_EXIT_NOT_OK;
                 }
 
+                qApp->processEvents();
+
                 if (res > 0)
                     return WEXITSTATUS(status);
 
-                qApp->processEvents();
                 usleep(100000);
             }
         }
