diff -Naur xorg-server-ff4006bd5a71d39cc5655679447c5c47a99a2751-old/hw/xfree86/common/xf86Config.h xorg-server-ff4006bd5a71d39cc5655679447c5c47a99a2751-new/hw/xfree86/common/xf86Config.h
--- xorg-server-ff4006bd5a71d39cc5655679447c5c47a99a2751-old/hw/xfree86/common/xf86Config.h	2008-04-30 09:44:48.000000000 -0700
+++ xorg-server-ff4006bd5a71d39cc5655679447c5c47a99a2751-new/hw/xfree86/common/xf86Config.h	2008-04-30 09:45:57.000000000 -0700
@@ -55,13 +55,6 @@
 } ModuleDefault;
 
 static ModuleDefault ModuleDefaults[] = {
-    {.name = "extmod",   .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "dbe",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "glx",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "freetype", .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "type1",    .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "record",   .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "dri",      .toLoad = TRUE,    .load_opt=NULL},
     {.name = NULL,       .toLoad = FALSE,   .load_opt=NULL}
 };
 
