diff -Naur xorg-server-1.4.2-old/hw/xfree86/common/xf86Config.h xorg-server-1.4.2-new/hw/xfree86/common/xf86Config.h
--- xorg-server-1.4.2-old/hw/xfree86/common/xf86Config.h	2008-06-11 12:03:50.000000000 -0700
+++ xorg-server-1.4.2-new/hw/xfree86/common/xf86Config.h	2008-06-11 12:04:31.000000000 -0700
@@ -55,13 +55,6 @@
 } ModuleDefault;
 
 static ModuleDefault ModuleDefaults[] = {
-    {.name = "extmod",   .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "dbe",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "glx",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "freetype", .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "type1",    .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "record",   .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "dri",      .toLoad = TRUE,    .load_opt=NULL},
     {.name = NULL,       .toLoad = FALSE,   .load_opt=NULL}
 };
 
