diff -Naur util-linux-ng-2.15-old/libs/blkid/src/Makefile.am util-linux-ng-2.15-new/libs/blkid/src/Makefile.am
--- util-linux-ng-2.15-old/libs/blkid/src/Makefile.am	2009-05-05 12:16:46.000000000 -0700
+++ util-linux-ng-2.15-new/libs/blkid/src/Makefile.am	2009-05-14 11:12:06.000000000 -0700
@@ -26,7 +26,6 @@
 		     $(top_srcdir)/lib/md5.c
 
 libblkid_la_LIBADD = probers/libblkid_probers.la $(common_ldadd)
-libblkid_la_DEPENDENCIES = $(libblkid_la_LIBADD) blkid.sym
 
 # TODO: for unknown reason "make distcheck" requires
 #       full path to the version script
