diff -Naur udev-052-old/Makefile udev-052-new/Makefile
--- udev-052-old/Makefile	2005-02-08 14:54:29.000000000 -0800
+++ udev-052-new/Makefile	2005-02-19 10:41:23.000000000 -0800
@@ -425,8 +425,8 @@
 	$(INSTALL_PROGRAM) -D $(SENDER) $(DESTDIR)$(sbindir)/$(SENDER)
 	$(INSTALL_PROGRAM) -D $(INFO) $(DESTDIR)$(usrbindir)/$(INFO)
 	$(INSTALL_PROGRAM) -D $(TESTER) $(DESTDIR)$(usrbindir)/$(TESTER)
-	- ln -f -s $(sbindir)/udev $(DESTDIR)$(sbindir)/$(STARTER)
-	- ln -f -s $(sbindir)/$(SENDER) $(DESTDIR)$(hotplugdir)/10-udev.hotplug
+	- ln -f -s $(DESTDIR)$(sbindir)/udev $(DESTDIR)$(sbindir)/$(STARTER)
+	- ln -f -s $(DESTDIR)$(sbindir)/$(SENDER) $(DESTDIR)$(hotplugdir)/10-udev.hotplug
 ifndef DESTDIR
 	- killall $(DAEMON)
 	- rm -rf $(udevdb)
