diff -Naur udev-154-old/configure.ac udev-154-new/configure.ac
--- udev-154-old/configure.ac	2010-04-24 12:34:48.000000000 -0700
+++ udev-154-new/configure.ac	2010-05-11 18:34:20.000000000 -0700
@@ -87,17 +87,24 @@
 	PKG_CHECK_MODULES(USBUTILS, usbutils >= 0.82)
 	AC_SUBST([USB_DATABASE], [$($PKG_CONFIG --variable=usbids usbutils)])
 
-	AC_CHECK_FILES([/usr/share/pci.ids], [pciids=/usr/share/pci.ids])
-	AC_CHECK_FILES([/usr/share/hwdata/pci.ids], [pciids=/usr/share/hwdata/pci.ids])
-	AC_CHECK_FILES([/usr/share/misc/pci.ids], [pciids=/usr/share/misc/pci.ids])
+	if test "x$cross_compiling" = "xno" ; then
+		AC_CHECK_FILES([/usr/share/pci.ids], [pciids=/usr/share/pci.ids])
+		AC_CHECK_FILES([/usr/share/hwdata/pci.ids], [pciids=/usr/share/hwdata/pci.ids])
+		AC_CHECK_FILES([/usr/share/misc/pci.ids], [pciids=/usr/share/misc/pci.ids])
+	else
+ 		# just assume location
+		pciids=/usr/share/pci.ids
+	fi
 	AC_ARG_WITH(pci-ids-path,
-		AS_HELP_STRING([--with-pci-ids-path=DIR], [Path to pci.ids file]),
+		[AS_HELP_STRING([--with-pci-ids-path=DIR], [Path to pci.ids file])],
 		[PCI_DATABASE=${withval}],
 		[if test -n "$pciids" ; then
 			PCI_DATABASE="$pciids"
 		else
 			AC_MSG_ERROR([pci.ids not found, try --with-pci-ids-path=])
 		fi])
+	AC_MSG_CHECKING([for PCI database location])
+	AC_MSG_RESULT([$PCI_DATABASE])
 	AC_SUBST(PCI_DATABASE)
 
 	AC_CHECK_HEADER([linux/input.h], [:], AC_MSG_ERROR([kernel headers not found]))
