diff -Naur udev-169-old/configure.ac udev-169-new/configure.ac
--- udev-169-old/configure.ac	2011-05-17 18:00:35.000000000 -0700
+++ udev-169-new/configure.ac	2011-05-19 10:28:53.000000000 -0700
@@ -75,7 +75,7 @@
 # ------------------------------------------------------------------------------
 # rule_generator - persistent network and optical device rule generator
 # ------------------------------------------------------------------------------
-AC_ARG_ENABLE([hwdb],
+AC_ARG_ENABLE([rule_generator],
 	AS_HELP_STRING([--disable-rule_generator], [disable persistent network, cdrom support]),
 	[], [enable_rule_generator=yes])
 AM_CONDITIONAL([ENABLE_RULE_GENERATOR], [test "x$enable_rule_genarator" = xyes])
@@ -190,7 +190,7 @@
 # ------------------------------------------------------------------------------
 # mobile-action-modeswitch - switch Mobile Action cables into serial mode
 # ------------------------------------------------------------------------------
-AC_ARG_ENABLE([ACTION_MODESWITCH],
+AC_ARG_ENABLE([action_modeswitch],
 	AS_HELP_STRING([--enable-action_modeswitch], [enable action modeswitch support]),
 	[], [enable_action_modeswitch=no])
 if test "x$enable_action_modeswitch" = xyes; then
