diff -Naur udev-069-old/Makefile udev-069-new/Makefile
--- udev-069-old/Makefile	2005-09-13 10:24:51.000000000 -0700
+++ udev-069-new/Makefile	2005-09-13 18:16:39.000000000 -0700
@@ -374,7 +374,7 @@
 endif
 	@extras="$(EXTRAS)"; for target in $$extras; do \
 		echo $$target; \
-		$(MAKE) prefix=$(prefix) -C $$target $@; \
+		$(MAKE) prefix=$(prefix) -C $$target install; \
 	done;
 .PHONY: install-bin
 
@@ -393,7 +393,7 @@
 	- killall udevd
 	@extras="$(EXTRAS)"; for target in $$extras; do \
 		echo $$target; \
-		$(MAKE) prefix=$(prefix) -C $$target $@; \
+		$(MAKE) prefix=$(prefix) -C $$target uninstall; \
 	done;
 .PHONY: uninstall-bin
 
