diff -Naur xorg-server-1.3.0.0-old/os/osdep.h xorg-server-1.3.0.0-new/os/osdep.h
--- xorg-server-1.3.0.0-old/os/osdep.h	2007-02-16 08:03:13.000000000 -0800
+++ xorg-server-1.3.0.0-new/os/osdep.h	2008-04-25 16:54:21.000000000 -0700
@@ -59,7 +59,9 @@
 #define MAXBUFSIZE (1 << 22)
 #endif
 
+#if defined(XDMCP) || defined(HASXDMAUTH)
 #include <X11/Xdmcp.h>
+#endif
 
 #ifndef sgi	    /* SGI defines OPEN_MAX in a useless way */
 #ifndef X_NOT_POSIX
@@ -125,9 +127,11 @@
 
 #include <stddef.h>
 
+#if defined(XDMCP) || defined(HASXDMAUTH)
 typedef Bool (*ValidatorFunc)(ARRAY8Ptr Auth, ARRAY8Ptr Data, int packet_type);
 typedef Bool (*GeneratorFunc)(ARRAY8Ptr Auth, ARRAY8Ptr Data, int packet_type);
 typedef Bool (*AddAuthorFunc)(unsigned name_length, char *name, unsigned data_length, char *data);
+#endif
 
 typedef struct _connectionInput {
     struct _connectionInput *next;
@@ -296,6 +300,7 @@
 /* in secauth.c */
 extern XID AuthSecurityCheck (AuthCheckArgs);
 
+#ifdef XDMCP
 /* in xdmcp.c */
 extern void XdmcpUseMsg (void);
 extern int XdmcpOptions(int argc, char **argv, int i);
@@ -324,6 +329,7 @@
 
 struct sockaddr_in;
 extern void XdmcpRegisterBroadcastAddress (struct sockaddr_in *addr);
+#endif
 
 #ifdef HASXDMAUTH
 extern void XdmAuthenticationInit (char *cookie, int cookie_length);
