diff -Naur iputils-s20121114-old/Makefile iputils-s20121114-new/Makefile
--- iputils-s20121114-old/Makefile	2012-11-14 05:10:12.000000000 -0800
+++ iputils-s20121114-new/Makefile	2012-11-14 11:43:31.000000000 -0800
@@ -49,7 +49,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 LASTTAG:=`git describe HEAD | sed -e 's/-.*//'`
