diff -Naur iputils-s20121125-old/Makefile iputils-s20121125-new/Makefile
--- iputils-s20121125-old/Makefile	2012-11-24 23:42:40.000000000 -0800
+++ iputils-s20121125-new/Makefile	2012-11-25 07:32:03.000000000 -0800
@@ -79,7 +79,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 LASTTAG:=`git describe HEAD | sed -e 's/-.*//'`
