diff -Naur linux-2.6.34.5-old/drivers/usb/core/driver.c linux-2.6.34.5-new/drivers/usb/core/driver.c
--- linux-2.6.34.5-old/drivers/usb/core/driver.c	2010-08-20 18:54:57.000000000 -0700
+++ linux-2.6.34.5-new/drivers/usb/core/driver.c	2010-08-20 18:56:46.000000000 -0700
@@ -1266,8 +1266,6 @@
 
 static void choose_wakeup(struct usb_device *udev, pm_message_t msg)
 {
-	int	w;
-
 	/* Remote wakeup is needed only when we actually go to sleep.
 	 * For things like FREEZE and QUIESCE, if the device is already
 	 * autosuspended then its current wakeup setting is okay.
@@ -1278,17 +1276,11 @@
 		return;
 	}
 
-	/* Enable remote wakeup if it is allowed, even if no interface drivers
+	/* Allow remote wakeup if it is enabled, even if no interface drivers
 	 * actually want it.
 	 */
-	w = device_may_wakeup(&udev->dev);
 
-	/* If the device is autosuspended with the wrong wakeup setting,
-	 * autoresume now so the setting can be changed.
-	 */
-	if (udev->state == USB_STATE_SUSPENDED && w != udev->do_remote_wakeup)
-		pm_runtime_resume(&udev->dev);
-	udev->do_remote_wakeup = w;
+        udev->do_remote_wakeup = device_may_wakeup(&udev->dev);
 }
 
 /* The device lock is held by the PM core */
