diff -Naur iguanaIR-0.99-old/Makefile.in iguanaIR-0.99-new/Makefile.in
--- iguanaIR-0.99-old/Makefile.in	2010-08-13 10:35:09.000000000 -0700
+++ iguanaIR-0.99-new/Makefile.in	2010-08-13 10:49:08.000000000 -0700
@@ -52,10 +52,10 @@
 
 igdaemon: daemon.o client-interface.o device-interface.o usbclient.o \
           list.o protocol-versions.o support.o pipes.o dataPackets.o compat.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lpopt -lusb -lpthread $(LIBNAME) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -fwhole-program $^ -lpopt -lusb -lpthread $(LIBNAME) -o $@
 
 igclient: client.o list.o support.o compat.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lpopt $(LIBNAME) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -fwhole-program $^ -lpopt $(LIBNAME) -o $@
 
 %.o: %.c Makefile iguanaIR.h
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
