diff -Naur iguanaIR-0.99-old/autoconf/ac_python_devel.m4 iguanaIR-0.99-new/autoconf/ac_python_devel.m4
--- iguanaIR-0.99-old/autoconf/ac_python_devel.m4	2008-11-18 17:44:32.000000000 -0800
+++ iguanaIR-0.99-new/autoconf/ac_python_devel.m4	2010-08-26 17:02:20.000000000 -0700
@@ -175,7 +175,7 @@
 		py_version=`$PYTHON -c "from distutils.sysconfig import *; \
 			from string import join; \
 			print join(get_config_vars('VERSION'))"`
-		if test "$py_version" == "[None]"; then
+		if test "$py_version" = "[None]"; then
 			if test -n "$PYTHON_VERSION"; then
 				py_version=$PYTHON_VERSION
 			else
diff -Naur iguanaIR-0.99-old/autoconf/configure.ac iguanaIR-0.99-new/autoconf/configure.ac
--- iguanaIR-0.99-old/autoconf/configure.ac	2008-11-18 18:28:01.000000000 -0800
+++ iguanaIR-0.99-new/autoconf/configure.ac	2010-08-26 16:58:48.000000000 -0700
@@ -134,7 +134,7 @@
     else
       AC_ARG_ENABLE(no-threads,
           [  --disable-no-threads enable the "no threads" option ],
-          [ if test "x$enableval" == "xno"; then
+          [ if test "x$enableval" = "xno"; then
                 LIBUSB_THREADS_OPTIONAL=no
             fi
           ]
@@ -155,7 +155,7 @@
 # Allow the user to override python detection
 AC_ARG_ENABLE(python,
   [  --disable-python do build the Python interface.],
-  [ if test x$enableval == xno; then
+  [ if test "x$enableval" = "xno"; then
         USE_PYTHON=no
     fi ],
 )
