diff -Naur linux-2.6.17.14-old/drivers/net/via-rhine.c linux-2.6.17.14-new/drivers/net/via-rhine.c
--- linux-2.6.17.14-old/drivers/net/via-rhine.c	2006-10-20 13:50:11.000000000 -0700
+++ linux-2.6.17.14-new/drivers/net/via-rhine.c	2006-10-20 13:52:20.000000000 -0700
@@ -1964,9 +1964,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
