diff -Naur mythtv-0.26-20141023-7e9a205-old/mythtv/programs/mythfrontend/main.cpp mythtv-0.26-20141023-7e9a205-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-0.26-20141023-7e9a205-old/mythtv/programs/mythfrontend/main.cpp	2015-06-08 21:10:58.699000363 -0700
+++ mythtv-0.26-20141023-7e9a205-new/mythtv/programs/mythfrontend/main.cpp	2015-06-08 21:11:00.209000363 -0700
@@ -1675,6 +1675,18 @@
                 QString("NetworkControl failed to bind to port %1.")
                    .arg(port));
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
 #ifdef __linux__
 #ifdef CONFIG_BINDINGS_PYTHON
