diff -Naur xorg-server-1.1.0-old/hw/xfree86/os-support/linux/Makefile.am xorg-server-1.1.0-new/hw/xfree86/os-support/linux/Makefile.am
--- xorg-server-1.1.0-old/hw/xfree86/os-support/linux/Makefile.am	2006-05-09 11:04:29.000000000 -0700
+++ xorg-server-1.1.0-new/hw/xfree86/os-support/linux/Makefile.am	2006-11-12 15:09:15.000000000 -0800
@@ -36,7 +36,7 @@
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS) @SERVER_DEFINES@ @LOADER_DEFINES@
 
-INCLUDES = $(XORG_INCS) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
diff -Naur xorg-server-1.1.0-old/hw/xfree86/os-support/linux/Makefile.in xorg-server-1.1.0-new/hw/xfree86/os-support/linux/Makefile.in
--- xorg-server-1.1.0-old/hw/xfree86/os-support/linux/Makefile.in	2006-05-22 10:21:09.000000000 -0700
+++ xorg-server-1.1.0-new/hw/xfree86/os-support/linux/Makefile.in	2006-11-12 15:09:15.000000000 -0800
@@ -568,7 +568,7 @@
                      $(PLATFORM_PCI_SUPPORT)
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS) @SERVER_DEFINES@ @LOADER_DEFINES@
-INCLUDES = $(XORG_INCS) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
