diff -Naur mythplugins-0.23-24879-old/mythvideo/mtd/dvdprobe.cpp mythplugins-0.23-24879-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-0.23-24879-old/mythvideo/mtd/dvdprobe.cpp	2010-06-02 19:56:11.000000000 -0700
+++ mythplugins-0.23-24879-new/mythvideo/mtd/dvdprobe.cpp	2010-06-02 19:56:11.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
