diff -Naur qt-everywhere-opensource-src-4.8.0-old/src/3rdparty/webkit/Source/WebKit.pri qt-everywhere-opensource-src-4.8.0-new/src/3rdparty/webkit/Source/WebKit.pri
--- qt-everywhere-opensource-src-4.8.0-old/src/3rdparty/webkit/Source/WebKit.pri	2011-12-07 21:06:03.000000000 -0800
+++ qt-everywhere-opensource-src-4.8.0-new/src/3rdparty/webkit/Source/WebKit.pri	2012-03-23 18:06:27.000000000 -0700
@@ -102,7 +102,7 @@
 
 # Treat warnings as errors on x86/Linux/GCC
 linux-g++* {
-    isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386): QMAKE_CXXFLAGS += -Werror
+    #isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386): QMAKE_CXXFLAGS += -Werror
 
     greaterThan(QT_GCC_MAJOR_VERSION, 3):greaterThan(QT_GCC_MINOR_VERSION, 5) {
         if (!contains(QMAKE_CXXFLAGS, -std=c++0x) && !contains(QMAKE_CXXFLAGS, -std=gnu++0x)) {
