diff -Naur qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/ChangeLog qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/ChangeLog
--- qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/ChangeLog	2010-09-10 02:05:21.000000000 -0700
+++ qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/ChangeLog	2010-09-21 16:09:36.000000000 -0700
@@ -8,6 +8,28 @@
 
         * plugins/mac/PluginViewMac.mm:
 
+2010-06-30  Andreas Kling  <andreas.kling@nokia.com>
+
+        Reviewed by Kenneth Rohde Christiansen.
+
+        [Qt] Make "disallow multiple calls to SetWindow" on windowed plugins a quirk
+        https://bugs.webkit.org/show_bug.cgi?id=41407
+
+        Only disallow calling SetWindow more than once if we're using Flash 9 or older.
+        This fixes resizing of YouTube videos via the player's own button.
+
+        Removed copy-pasted logic in PluginViewGtk since the issue only exists
+        when running Flash with a non-Gtk toolkit.
+
+        * plugins/PluginPackage.cpp:
+        (WebCore::PluginPackage::determineQuirks):
+        * plugins/PluginQuirkSet.h:
+        (WebCore::):
+        * plugins/gtk/PluginViewGtk.cpp:
+        (WebCore::PluginView::setNPWindowIfNeeded): Remove unnecessary logic copy-pasted from PluginViewQt.
+        * plugins/qt/PluginViewQt.cpp:
+        (WebCore::PluginView::setNPWindowIfNeeded):
+
 2010-06-14  Andreas Kling  <andreas.kling@nokia.com>
 
         Reviewed by Tor Arne Vestbø.
diff -Naur qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/plugins/PluginPackage.cpp qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/plugins/PluginPackage.cpp
--- qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/plugins/PluginPackage.cpp	2010-09-10 02:05:21.000000000 -0700
+++ qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/plugins/PluginPackage.cpp	2010-09-21 16:08:18.000000000 -0700
@@ -189,6 +189,10 @@
         } else {
             // Flash 9 and older requests windowless plugins if we return a mozilla user agent
             m_quirks.add(PluginQuirkWantsMozillaUserAgent);
+#if PLATFORM(QT) && 0
+            // Flash 9 and older would crash on repeated calls to SetWindow in windowed mode
+            m_quirks.add(PluginQuirkDontCallSetWindowMoreThanOnce);
+#endif
         }
 
         m_quirks.add(PluginQuirkThrottleInvalidate);
diff -Naur qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/plugins/PluginQuirkSet.h qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/plugins/PluginQuirkSet.h
--- qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/plugins/PluginQuirkSet.h	2010-09-10 02:05:21.000000000 -0700
+++ qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/plugins/PluginQuirkSet.h	2010-09-21 16:08:18.000000000 -0700
@@ -46,7 +46,8 @@
         PluginQuirkDontSetNullWindowHandleOnDestroy = 1 << 10,
         PluginQuirkDontAllowMultipleInstances = 1 << 11,
         PluginQuirkRequiresGtkToolKit = 1 << 12,
-        PluginQuirkRequiresDefaultScreenDepth = 1 << 13
+        PluginQuirkRequiresDefaultScreenDepth = 1 << 13,
+        PluginQuirkDontCallSetWindowMoreThanOnce = 1 << 14
     };
 
     class PluginQuirkSet {
diff -Naur qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp
--- qt-everywhere-opensource-src-4.7.0-old/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp	2010-09-10 02:05:21.000000000 -0700
+++ qt-everywhere-opensource-src-4.7.0-new/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp	2010-09-21 16:08:18.000000000 -0700
@@ -501,9 +501,14 @@
         m_npWindow.clipRect.bottom = 0;
     }
 
-    // FLASH WORKAROUND: Only set initially. Multiple calls to
-    // setNPWindow() cause the plugin to crash in windowed mode.
-    if (!m_isWindowed || m_npWindow.width == -1 || m_npWindow.height == -1) {
+    if (m_plugin->quirks().contains(PluginQuirkDontCallSetWindowMoreThanOnce)) {
+        // FLASH WORKAROUND: Only set initially. Multiple calls to
+        // setNPWindow() cause the plugin to crash in windowed mode.
+        if (!m_isWindowed || m_npWindow.width == -1 || m_npWindow.height == -1) {
+            m_npWindow.width = m_windowRect.width();
+            m_npWindow.height = m_windowRect.height();
+        }
+    } else {
         m_npWindow.width = m_windowRect.width();
         m_npWindow.height = m_windowRect.height();
     }
