diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/ChangeLog qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/ChangeLog
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/ChangeLog	2010-08-22 10:50:31.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/ChangeLog	2010-08-22 10:51:36.000000000 -0700
@@ -1,3 +1,20 @@
+2010-06-20  Robert Hogan  <robert@webkit.org>
+
+        Reviewed by Simon Hausmann.
+
+        [Qt] NPP_SetWindow seems to not be called when TestNetscapePlugin is moved
+
+        https://bugs.webkit.org/show_bug.cgi?id=36702
+
+        setNPWindowIfNeeded() is called on paint() in PluginViewQt, which doesn't
+        work for DRT. So call it if we are in DRT mode and the window geometry
+        has changed.
+
+        Unskips plugins/reentrant-update-widget-positions.html
+
+        * plugins/qt/PluginViewQt.cpp:
+        (WebCore::PluginView::updatePluginWidget):
+
 2010-06-16  Dawit Alemayehu  <adawit@kde.org>
 
         Reviewed by Simon Hausmann.
diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp	2010-06-01 19:03:11.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp	2010-08-22 10:51:36.000000000 -0700
@@ -60,6 +60,7 @@
 #include "runtime.h"
 #include "runtime_root.h"
 #include "QWebPageClient.h"
+#include "qwebpage_p.h"
 #include <QApplication>
 #include <QDesktopWidget>
 #include <QKeyEvent>
@@ -119,12 +120,15 @@
     // do not call setNPWindowIfNeeded immediately, will be called on paint()
     m_hasPendingGeometryChange = true;
 
-    // in order to move/resize the plugin window at the same time as the
+    // (i) in order to move/resize the plugin window at the same time as the
     // rest of frame during e.g. scrolling, we set the window geometry
     // in the paint() function, but as paint() isn't called when the
     // plugin window is outside the frame which can be caused by a
     // scroll, we need to move/resize immediately.
-    if (!m_windowRect.intersects(frameView->frameRect()))
+    // (ii) if we are running layout tests from DRT, paint() won't ever get called
+    // so we need to call setNPWindowIfNeeded() if window geometry has changed
+    if (!m_windowRect.intersects(frameView->frameRect())
+        || (QWebPagePrivate::drtRun && platformPluginWidget() && (m_windowRect != oldWindowRect || m_clipRect != oldClipRect)))
         setNPWindowIfNeeded();
 
     // Make sure we get repainted afterwards. This is necessary for downward
