diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/ChangeLog qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/ChangeLog
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/ChangeLog	2010-06-01 19:03:12.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/ChangeLog	2010-08-22 13:29:27.000000000 -0700
@@ -1,3 +1,17 @@
+2010-01-07  Brent Fulgham  <bfulgham@webkit.org>
+
+        Reviewed by Jon Honeycutt.
+
+        Account for scale and current context position (as well as page
+        position) in the device context world transform passed to
+        plugins.  This is especially important for print surface DPI.
+        http://bugs.webkit.org/show_bug.cgi?id=32909.
+
+        * plugins/win/PluginViewWin.cpp:
+        (WebCore::PluginView::paintWindowedPluginIntoContext): Revise
+          world context handling to include scaling and translation
+          already set for the CTM.
+
 2010-05-04  Tucker Jay  <jay.tucker@nokia.com>
 
         Reviewed by Holger Freyther.
diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/win/PluginViewWin.cpp qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/win/PluginViewWin.cpp
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/win/PluginViewWin.cpp	2010-06-01 19:03:11.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/win/PluginViewWin.cpp	2010-08-22 13:29:27.000000000 -0700
@@ -561,9 +561,9 @@
 
     // The plugin expects the DC to be in client coordinates, so we translate
     // the DC to make that so.
-    XFORM transform = originalTransform;
-    transform.eDx = locationInWindow.x();
-    transform.eDy = locationInWindow.y();
+    TransformationMatrix ctm = context->getCTM();
+    ctm.translate(locationInWindow.x(), locationInWindow.y());
+    XFORM transform = static_cast<XFORM>(ctm);
 
     SetWorldTransform(hdc, &transform);
 
