diff -Naur qt-everywhere-opensource-src-4.8.1-old/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h qt-everywhere-opensource-src-4.8.1-new/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h
--- qt-everywhere-opensource-src-4.8.1-old/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-28 19:52:42.000000000 -0700
+++ qt-everywhere-opensource-src-4.8.1-new/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-28 19:55:18.000000000 -0700
@@ -57,9 +57,7 @@
     
     class JSGlobalObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
-
-        struct JSGlobalObjectData : public JSVariableObjectData {
+        struct JSGlobalObjectData : public JSVariableObject::JSVariableObjectData {
             // We use an explicit destructor function pointer instead of a
             // virtual destructor because we want to avoid adding a vtable
             // pointer to this struct. Adding a vtable pointer would force the
diff -Naur qt-everywhere-opensource-src-4.8.1-old/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h qt-everywhere-opensource-src-4.8.1-new/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h
--- qt-everywhere-opensource-src-4.8.1-old/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h	2012-03-28 19:52:42.000000000 -0700
+++ qt-everywhere-opensource-src-4.8.1-new/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h	2012-03-28 19:55:18.000000000 -0700
@@ -32,8 +32,7 @@
     
     class JSStaticScopeObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
-        struct JSStaticScopeObjectData : public JSVariableObjectData {
+        struct JSStaticScopeObjectData : public JSVariableObject::JSVariableObjectData {
             JSStaticScopeObjectData()
                 : JSVariableObjectData(&symbolTable, &registerStore + 1)
             {
