diff -Naur xf86-input-evtouch-0.8.7-old/evtouch.c xf86-input-evtouch-0.8.7-new/evtouch.c
--- xf86-input-evtouch-0.8.7-old/evtouch.c	2007-09-10 11:38:21.000000000 -0700
+++ xf86-input-evtouch-0.8.7-new/evtouch.c	2007-09-10 11:39:04.000000000 -0700
@@ -106,7 +106,12 @@
 static void
 EVTouchPtrCtrl(DeviceIntPtr device, PtrCtrl *ctrl);
 
+#ifdef DBG
+#undef DBG
+#endif
+
 static int debug_level = 0;
+#define DBG(lvl, f) {if ((lvl) <= debug_level) f;}
 
 InputDriverRec EVTOUCH = {
         1,
