diff -Naur xf86-input-evtouch-0.8.6-old/evtouch.c xf86-input-evtouch-0.8.6-new/evtouch.c
--- xf86-input-evtouch-0.8.6-old/evtouch.c	2007-06-09 08:30:37.000000000 -0700
+++ xf86-input-evtouch-0.8.6-new/evtouch.c	2007-09-08 14:51:30.000000000 -0700
@@ -100,7 +100,12 @@
 static void
 EVTouchPtrCtrl(DeviceIntPtr device, PtrCtrl *ctrl);
 
+#ifdef DBG
+#undef DBG
+#endif
+
 static int debug_level = 0;
+#define DBG(lvl, f) {if ((lvl) <= debug_level) f;}
 
 InputDriverRec EVTOUCH = {
         1,
