diff -Naur taglib-1.4-old/bindings/c/tag_c.cpp taglib-1.4-new/bindings/c/tag_c.cpp
--- taglib-1.4-old/bindings/c/tag_c.cpp	2005-05-17 13:17:26.000000000 -0700
+++ taglib-1.4-new/bindings/c/tag_c.cpp	2008-01-16 21:27:22.000000000 -0800
@@ -28,6 +28,7 @@
 #include <flacfile.h>
 #include <mpcfile.h>
 #include <tag.h>
+#include <string.h>
 
 namespace TagLib
 {
diff -Naur taglib-1.4-old/taglib/toolkit/tbytevector.cpp taglib-1.4-new/taglib/toolkit/tbytevector.cpp
--- taglib-1.4-old/taglib/toolkit/tbytevector.cpp	2005-05-17 13:44:14.000000000 -0700
+++ taglib-1.4-new/taglib/toolkit/tbytevector.cpp	2008-01-16 21:23:56.000000000 -0800
@@ -26,6 +26,8 @@
 
 #include "tbytevector.h"
 
+#include <string.h>
+
 // This is a bit ugly to keep writing over and over again.
 
 // A rather obscure feature of the C++ spec that I hadn't thought of that makes
diff -Naur taglib-1.4-old/taglib/toolkit/tfile.cpp taglib-1.4-new/taglib/toolkit/tfile.cpp
--- taglib-1.4-old/taglib/toolkit/tfile.cpp	2005-07-25 14:16:32.000000000 -0700
+++ taglib-1.4-new/taglib/toolkit/tfile.cpp	2008-01-16 21:25:40.000000000 -0800
@@ -25,6 +25,8 @@
 
 #include <stdio.h>
 #include <sys/stat.h>
+#include <stdlib.h>
+#include <string.h>
 #include <unistd.h>
 
 using namespace TagLib;
diff -Naur taglib-1.4-old/taglib/toolkit/tstring.cpp taglib-1.4-new/taglib/toolkit/tstring.cpp
--- taglib-1.4-old/taglib/toolkit/tstring.cpp	2005-07-25 14:31:15.000000000 -0700
+++ taglib-1.4-new/taglib/toolkit/tstring.cpp	2008-01-16 21:22:20.000000000 -0800
@@ -25,6 +25,8 @@
 
 #include <iostream>
 
+#include <string.h>
+
 namespace TagLib {
 
   inline unsigned short byteSwap(unsigned short x)
