diff -Naur iputils-s20100418-old/Makefile iputils-s20100418-new/Makefile
--- iputils-s20100418-old/Makefile	2010-04-17 21:45:45.000000000 -0700
+++ iputils-s20100418-new/Makefile	2010-04-19 09:05:53.000000000 -0700
@@ -14,7 +14,7 @@
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 #CCOPT=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g -Werror
 CCOPT=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g
-CFLAGS=$(CCOPT) $(GLIBCFIX) $(DEFINES) 
+CFLAGS+=$(CCOPT) $(GLIBCFIX) $(DEFINES) 
 
 IPV4_TARGETS=tracepath ping clockdiff rdisc arping tftpd rarpd
 IPV6_TARGETS=tracepath6 traceroute6 ping6
