diff -Naur iputils-s20121126-old/Makefile iputils-s20121126-new/Makefile
--- iputils-s20121126-old/Makefile	2012-11-26 02:28:05.000000000 -0800
+++ iputils-s20121126-new/Makefile	2012-11-26 17:03:03.000000000 -0800
@@ -79,7 +79,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 LASTTAG:=`git describe HEAD | sed -e 's/-.*//'`
