diff -Naur mythplugins-0.24-27089-old/configure mythplugins-0.24-27089-new/configure
--- mythplugins-0.24-27089-old/configure	2010-11-03 11:36:51.000000000 -0700
+++ mythplugins-0.24-27089-new/configure	2010-11-03 11:36:52.000000000 -0700
@@ -396,11 +396,27 @@
 
 if ! disabled fftw; then
     disable fftw_lib3
-    if has_library libfftw3f_threads ; then
-        if has_library libfftw3_threads ; then
-            if has_header fftw3.h ; then
-                enable fftw_lib3
-            fi
+    if has_library libfftw3  ; then
+        if has_header fftw3.h ; then
+            enable fftw_lib3
+        fi
+    fi
+fi
+
+if ! disabled fftw; then
+    disable fftw_lib3f
+    if has_library libfftw3f ; then
+        if has_header fftw3.h ; then
+            enable fftw_lib3f
+        fi
+    fi
+fi
+
+if ! disabled fftw; then
+    disable fftw_lib3l
+    if has_library libfftw3l ; then
+        if has_header fftw3.h ; then
+            enable fftw_lib3l
         fi
     fi
 fi
@@ -847,11 +863,21 @@
         echo "        libvisual      support will not be included in MythMusic"
     fi
 
-    if enabled fftw_lib3; then
+    if enabled fftw_lib3f; then
+        echo "        FFTW v.3       support will be included in MythMusic"
+        echo "#define FFTW3_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
+        echo "#define FFTW3_SUPPORT_FFTWF 1" >> ./mythmusic/mythmusic/config.h
+        echo "LIBS += -lfftw3f -lm" >> ./mythmusic/mythmusic/config.pro
+    elif enabled fftw_lib3; then
+        echo "        FFTW v.3       support will be included in MythMusic"
+        echo "#define FFTW3_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
+        echo "#define FFTW3_SUPPORT_FFTW 1" >> ./mythmusic/mythmusic/config.h
+        echo "LIBS += -lfftw3 -lm" >> ./mythmusic/mythmusic/config.pro
+    elif enabled fftw_lib3l; then
         echo "        FFTW v.3       support will be included in MythMusic"
         echo "#define FFTW3_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
-        echo "LIBS += -lfftw3 -lm" >> \
-             ./mythmusic/mythmusic/config.pro
+        echo "#define FFTW3_SUPPORT_FFTWL 1" >> ./mythmusic/mythmusic/config.h
+        echo "LIBS += -lfftw3d -lm" >> ./mythmusic/mythmusic/config.pro
     elif enabled fftw_lib; then
         echo "        FFTW v.2       support will be included in MythMusic"
         echo "#define FFTW2_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
diff -Naur mythplugins-0.24-27089-old/mythmusic/mythmusic/visualize.cpp mythplugins-0.24-27089-new/mythmusic/mythmusic/visualize.cpp
--- mythplugins-0.24-27089-old/mythmusic/mythmusic/visualize.cpp	2010-11-03 11:36:51.000000000 -0700
+++ mythplugins-0.24-27089-new/mythmusic/mythmusic/visualize.cpp	2010-11-03 11:36:52.000000000 -0700
@@ -59,8 +59,8 @@
     rout = (myth_fftw_complex*)
         av_malloc(sizeof(myth_fftw_complex)*(FFTW_N/2+1));
 
-    lplan = fftw_plan_dft_r2c_1d(FFTW_N, lin, (myth_fftw_complex_cast*)lout, FFTW_MEASURE);
-    rplan = fftw_plan_dft_r2c_1d(FFTW_N, rin, (myth_fftw_complex_cast*)rout, FFTW_MEASURE);
+    lplan = myth_fftw_plan_dft_r2c_1d(FFTW_N, lin, (myth_fftw_complex_cast*)lout, FFTW_MEASURE);
+    rplan = myth_fftw_plan_dft_r2c_1d(FFTW_N, rin, (myth_fftw_complex_cast*)rout, FFTW_MEASURE);
 #elif FFTW2_SUPPORT
     lin = (fftw_real*) av_malloc(sizeof(fftw_real)*FFTW_N);
     rin = (fftw_real*) av_malloc(sizeof(fftw_real)*FFTW_N);
@@ -87,8 +87,8 @@
         av_free(rout);
 #endif
 #ifdef FFTW3_SUPPORT
-    fftw_destroy_plan(lplan);
-    fftw_destroy_plan(rplan);
+    myth_fftw_destroy_plan(lplan);
+    myth_fftw_destroy_plan(rplan);
 #elif FFTW2_SUPPORT
     rfftw_destroy_plan(plan);
 #endif
@@ -153,8 +153,8 @@
     fast_reals_set(lin + i, rin + i, 0, FFTW_N - i);
 
 #ifdef FFTW3_SUPPORT
-    fftw_execute(lplan);
-    fftw_execute(rplan);
+    myth_fftw_execute(lplan);
+    myth_fftw_execute(rplan);
 #elif FFTW2_SUPPORT
     rfftw_one(plan, lin, lout);
     rfftw_one(plan, rin, rout);
@@ -761,8 +761,8 @@
     rout = (myth_fftw_complex*)
         av_malloc(sizeof(myth_fftw_complex)*(FFTW_N/2+1));
 
-    lplan = fftw_plan_dft_r2c_1d(FFTW_N, lin, (myth_fftw_complex_cast*) lout, FFTW_MEASURE);
-    rplan = fftw_plan_dft_r2c_1d(FFTW_N, rin, (myth_fftw_complex_cast*) rout, FFTW_MEASURE);
+    lplan = myth_fftw_plan_dft_r2c_1d(FFTW_N, lin, (myth_fftw_complex_cast*) lout, FFTW_MEASURE);
+    rplan = myth_fftw_plan_dft_r2c_1d(FFTW_N, rin, (myth_fftw_complex_cast*) rout, FFTW_MEASURE);
 #elif FFTW2_SUPPORT
     lin = (fftw_real*) av_malloc(sizeof(fftw_real)*FFTW_N);
     rin = (fftw_real*) av_malloc(sizeof(fftw_real)*FFTW_N);
@@ -792,8 +792,8 @@
         av_free(rout);
 #endif
 #ifdef FFTW3_SUPPORT
-    fftw_destroy_plan(lplan);
-    fftw_destroy_plan(rplan);
+    myth_fftw_destroy_plan(lplan);
+    myth_fftw_destroy_plan(rplan);
 #elif FFTW2_SUPPORT
     rfftw_destroy_plan(plan);
 #endif
@@ -850,8 +850,8 @@
 
     fast_reals_set(lin + i, rin + i, 0, FFTW_N - i);
 #ifdef FFTW3_SUPPORT
-    fftw_execute(lplan);
-    fftw_execute(rplan);
+    myth_fftw_execute(lplan);
+    myth_fftw_execute(rplan);
 #elif FFTW2_SUPPORT
     rfftw_one(plan, lin, lout);
     rfftw_one(plan, rin, rout);
diff -Naur mythplugins-0.24-27089-old/mythmusic/mythmusic/visualize.h mythplugins-0.24-27089-new/mythmusic/mythmusic/visualize.h
--- mythplugins-0.24-27089-old/mythmusic/mythmusic/visualize.h	2010-11-03 11:36:51.000000000 -0700
+++ mythplugins-0.24-27089-new/mythmusic/mythmusic/visualize.h	2010-11-03 11:36:52.000000000 -0700
@@ -29,14 +29,35 @@
 extern "C" {
 #ifdef FFTW3_SUPPORT
 #include <fftw3.h>
-#define myth_fftw_float double /* need to use different plan function to change */
-#define fftw_real myth_fftw_float
+#if ! defined (FFTW3_SUPPORT_FFTW) && ! defined (FFTW3_SUPPORT_FFTWF) && ! defined (FFTW3_SUPPORT_FFTWL)
+#define FFTW3_SUPPORT_FFTW
+#endif
+#if defined (FFTW3_SUPPORT_FFTW)
+#define myth_fftw_float double
 #define myth_fftw_complex std::complex<myth_fftw_float>
-#if (myth_fftw_float == double)
 #define myth_fftw_complex_cast fftw_complex
-#elif (myth_fftw_float == float)
+#define myth_fftw_plan fftw_plan
+#define myth_fftw_destroy_plan(a) fftw_destroy_plan(a)
+#define myth_fftw_execute(a) fftw_execute(a)
+#define myth_fftw_plan_dft_r2c_1d(a,b,c,d) fftw_plan_dft_r2c_1d(a,b,c,d)
+#elif defined (FFTW3_SUPPORT_FFTWF)
+#define myth_fftw_float float
+#define myth_fftw_complex std::complex<myth_fftw_float>
 #define myth_fftw_complex_cast fftwf_complex
+#define myth_fftw_plan fftwf_plan
+#define myth_fftw_destroy_plan(a) fftwf_destroy_plan(a)
+#define myth_fftw_execute(a) fftwf_execute(a)
+#define myth_fftw_plan_dft_r2c_1d(a,b,c,d) fftwf_plan_dft_r2c_1d(a,b,c,d)
+#elif defined (FFTW3_SUPPORT_FFTWL)
+#define myth_fftw_float long double
+#define myth_fftw_complex std::complex<myth_fftw_float>
+#define myth_fftw_complex_cast fftwl_complex
+#define myth_fftw_plan fftwl_plan
+#define myth_fftw_destroy_plan(a) fftwl_destroy_plan(a)
+#define myth_fftw_execute(a) fftwl_execute(a)
+#define myth_fftw_plan_dft_r2c_1d(a,b,c,d) fftwl_plan_dft_r2c_1d(a,b,c,d)
 #endif
+#define fftw_real myth_fftw_float
 #elif	FFTW2_SUPPORT
 #include <rfftw.h>
 #include <fftw.h>
@@ -74,7 +95,7 @@
     int analyzerBarWidth;
 
 #ifdef FFTW3_SUPPORT
-    fftw_plan lplan, rplan;
+    myth_fftw_plan lplan, rplan;
     myth_fftw_float *lin, *rin;
     myth_fftw_complex *lout, *rout;
 #elif FFTW2_SUPPORT
@@ -170,7 +191,7 @@
     GLfloat angle, view_roty;
 
 #ifdef FFTW3_SUPPORT
-    fftw_plan lplan, rplan;
+    myth_fftw_plan lplan, rplan;
     myth_fftw_float *lin, *rin;
     myth_fftw_complex *lout, *rout;
 #elif FFTW2_SUPPORT
