diff -Naur mythtv-0.27-20150616-ad97d24-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-0.27-20150616-ad97d24-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-0.27-20150616-ad97d24-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2015-06-20 21:41:38.872000115 -0700
+++ mythtv-0.27-20150616-ad97d24-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2015-06-20 21:41:39.957000116 -0700
@@ -2483,6 +2483,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
