diff -Naur mythtv-0.27-20150616-ad97d24-old/mythtv/external/zeromq/configure.in mythtv-0.27-20150616-ad97d24-new/mythtv/external/zeromq/configure.in
--- mythtv-0.27-20150616-ad97d24-old/mythtv/external/zeromq/configure.in	2015-06-20 21:41:06.000000000 -0700
+++ mythtv-0.27-20150616-ad97d24-new/mythtv/external/zeromq/configure.in	2015-06-20 21:41:38.368000117 -0700
@@ -10,7 +10,7 @@
 
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
-AM_CONFIG_HEADER(src/platform.hpp)
+AC_CONFIG_HEADERS(src/platform.hpp)
 AM_INIT_AUTOMAKE(tar-ustar dist-zip foreign)
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
