diff -Naur xf86-video-ati-6.14.0-old/src/drmmode_display.c xf86-video-ati-6.14.0-new/src/drmmode_display.c
--- xf86-video-ati-6.14.0-old/src/drmmode_display.c	2011-02-01 16:23:58.000000000 -0800
+++ xf86-video-ati-6.14.0-new/src/drmmode_display.c	2011-02-04 07:39:16.000000000 -0800
@@ -657,7 +657,7 @@
 		mon = xf86InterpretEDID(output->scrn->scrnIndex,
 					drmmode_output->edid_blob->data);
 		if (mon && drmmode_output->edid_blob->length > 128)
-			mon->flags |= MONITOR_EDID_COMPLETE_RAWDATA;
+			mon->flags |= EDID_COMPLETE_RAWDATA;
 	}
 	xf86OutputSetEDID(output, mon);
 
