diff -Naur mythtv-0.22-22840-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.22-22840-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.22-22840-old/libs/libmyth/mediamonitor-unix.cpp	2009-11-16 18:17:32.000000000 -0800
+++ mythtv-0.22-22840-new/libs/libmyth/mediamonitor-unix.cpp	2009-11-16 18:17:33.000000000 -0800
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
