diff -Naur mythtv-0.22-23084-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23084-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23084-old/libs/libmyth/mythcontext.cpp	2010-01-09 07:21:44.000000000 -0800
+++ mythtv-0.22-23084-new/libs/libmyth/mythcontext.cpp	2010-01-09 07:24:10.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
