diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythvideo/mtd/dvdprobe.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythvideo/mtd/dvdprobe.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-01-15 08:08:25.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-01-15 08:08:26.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
