diff -Naur mythtv-0.22-27412-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-0.22-27412-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.22-27412-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2010-12-02 14:12:15.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2010-12-02 14:12:17.000000000 -0800
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
