diff -Naur mythtv-0.22-25328-old/libs/libmyth/mythcontext.cpp mythtv-0.22-25328-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-25328-old/libs/libmyth/mythcontext.cpp	2010-07-11 17:30:20.000000000 -0700
+++ mythtv-0.22-25328-new/libs/libmyth/mythcontext.cpp	2010-07-11 17:30:21.000000000 -0700
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
