diff -Naur mythtv-0.22-27412-old/mythtv/libs/libmythdvdnav/libmythdvdnav.pro mythtv-0.22-27412-new/mythtv/libs/libmythdvdnav/libmythdvdnav.pro
--- mythtv-0.22-27412-old/mythtv/libs/libmythdvdnav/libmythdvdnav.pro	2010-12-02 14:12:05.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/libs/libmythdvdnav/libmythdvdnav.pro	2010-12-02 14:12:06.000000000 -0800
@@ -13,6 +13,9 @@
 #build position independent code since the library is linked into a shared library
 QMAKE_CFLAGS += -fPIC -DPIC
 
+# for -ldl
+LIBS += -ldl
+
 DEFINES += HAVE_AV_CONFIG_H _LARGEFILE_SOURCE
 
 # DEFINES += LOG_DEBUG TRACE
