diff -Naur mythtv-0.22-24184-old/libs/libmyth/mythcontext.cpp mythtv-0.22-24184-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-24184-old/libs/libmyth/mythcontext.cpp	2010-04-23 11:37:15.000000000 -0700
+++ mythtv-0.22-24184-new/libs/libmyth/mythcontext.cpp	2010-04-23 11:39:45.000000000 -0700
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
