diff -Naur mythtv-0.22-23298-old/configure mythtv-0.22-23298-new/configure
--- mythtv-0.22-23298-old/configure	2010-01-25 18:19:56.000000000 -0800
+++ mythtv-0.22-23298-new/configure	2010-01-25 18:19:56.000000000 -0800
@@ -3448,7 +3448,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
