diff -Naur mythtv-0.22-23710-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.22-23710-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.22-23710-old/libs/libmyth/mediamonitor-unix.cpp	2010-03-10 07:47:00.000000000 -0800
+++ mythtv-0.22-23710-new/libs/libmyth/mediamonitor-unix.cpp	2010-03-10 07:47:01.000000000 -0800
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
