diff -Naur mythtv-0.22-27412-old/mythtv/configure mythtv-0.22-27412-new/mythtv/configure
--- mythtv-0.22-27412-old/mythtv/configure	2010-12-02 14:11:57.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/configure	2010-12-02 14:11:58.000000000 -0800
@@ -2622,13 +2622,13 @@
         enabled shared || check_cflags -mdynamic-no-pic
     fi
     # Apple GCC doesn't support march=core2 yet:
-    if test $cpu == "core2"; then
+    if test $cpu = "core2"; then
         cpu="nocona"
     fi
     # Core Duo might be identified as march=pentium3, which errors:
     # ...MachineExceptions.h:... error: '__m128i' does not name a type
     # ...MachineExceptions.h:... error: '__m128i' does not name a type
-    if test $cpu == "pentium3"; then
+    if test $cpu = "pentium3"; then
         cpu=prescott
     fi
 fi
