diff -Naur mythtv-0.22-27412-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-0.22-27412-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-0.22-27412-old/mythtv/programs/mythtranscode/transcode.cpp	2010-12-02 14:12:09.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/programs/mythtranscode/transcode.cpp	2010-12-02 14:12:10.000000000 -0800
@@ -748,7 +748,7 @@
         // framecontrol is true if we want to enforce fifo sync.
         if (framecontrol)
             VERBOSE(VB_GENERAL, "Enforcing sync on fifos");
-        fifow = new FIFOWriter::FIFOWriter(2, framecontrol);
+        fifow = new FIFOWriter(2, framecontrol);
 
         if (!fifow->FIFOInit(0, QString("video"), vidfifo, vidSize, 50) ||
             !fifow->FIFOInit(1, QString("audio"), audfifo, audio_size, 25))
diff -Naur mythtv-0.22-27412-old/mythtv/programs/mythtranscode/transcode.h mythtv-0.22-27412-new/mythtv/programs/mythtranscode/transcode.h
--- mythtv-0.22-27412-old/mythtv/programs/mythtranscode/transcode.h	2010-12-02 14:12:09.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/programs/mythtranscode/transcode.h	2010-12-02 14:12:10.000000000 -0800
@@ -38,7 +38,7 @@
     PlayerContext          *player_ctx;
     RingBuffer             *inRingBuffer;
     RingBuffer             *outRingBuffer;
-    FIFOWriter::FIFOWriter *fifow;
+    FIFOWriter             *fifow;
     KFATable               *kfa_table;
     bool                    showprogress;
     QString                 recorderOptions;
