diff -Naur mythtv-0.22-21860-old/libs/libmythtv/vsync.cpp mythtv-0.22-21860-new/libs/libmythtv/vsync.cpp
--- mythtv-0.22-21860-old/libs/libmythtv/vsync.cpp	2009-09-14 16:33:40.000000000 -0700
+++ mythtv-0.22-21860-new/libs/libmythtv/vsync.cpp	2009-09-14 16:33:41.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
