diff -Naur mythtv-0.22-27412-old/mythtv/libs/libmyth/mythcdrom-linux.cpp mythtv-0.22-27412-new/mythtv/libs/libmyth/mythcdrom-linux.cpp
--- mythtv-0.22-27412-old/mythtv/libs/libmyth/mythcdrom-linux.cpp	2010-12-02 14:11:59.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/libs/libmyth/mythcdrom-linux.cpp	2010-12-02 14:12:01.000000000 -0800
@@ -1,4 +1,5 @@
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <sys/ioctl.h>       // ioctls
 #include <linux/cdrom.h>     // old ioctls for cdrom
 #include <scsi/sg.h>
