diff -Naur mythtv-0.22-25120-old/libs/libmythdvdnav/libmythdvdnav.pro mythtv-0.22-25120-new/libs/libmythdvdnav/libmythdvdnav.pro
--- mythtv-0.22-25120-old/libs/libmythdvdnav/libmythdvdnav.pro	2010-06-20 07:25:11.000000000 -0700
+++ mythtv-0.22-25120-new/libs/libmythdvdnav/libmythdvdnav.pro	2010-06-20 07:25:12.000000000 -0700
@@ -13,6 +13,9 @@
 #build position independent code since the library is linked into a shared library
 QMAKE_CFLAGS += -fPIC -DPIC
 
+# for -ldl
+LIBS += -ldl
+
 DEFINES += HAVE_AV_CONFIG_H _LARGEFILE_SOURCE
 
 # DEFINES += LOG_DEBUG TRACE
