diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythmusic/mythmusic/importmusic.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythmusic/mythmusic/importmusic.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythmusic/mythmusic/importmusic.cpp	2011-10-07 15:27:31.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythmusic/mythmusic/importmusic.cpp	2012-04-12 04:54:13.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // qt
 #include <QDir>
 #include <QFontMetrics>
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythdb/compat.h mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythdb/compat.h
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythdb/compat.h	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythdb/compat.h	2012-04-12 04:54:13.000000000 -0700
@@ -23,6 +23,8 @@
 #include <sys/resource.h> // for setpriority
 #include <sys/socket.h>
 #include <sys/wait.h>     // For WIFEXITED on Mac OS X
+#include <stdio.h>        // for snprintf(), used by inline dlerror()
+#include <unistd.h>       // for usleep()
 #endif
 
 #ifdef _WIN32
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythfreemheg/BaseClasses.h mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythfreemheg/BaseClasses.h
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythfreemheg/BaseClasses.h	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythfreemheg/BaseClasses.h	2012-04-12 04:54:13.000000000 -0700
@@ -89,7 +89,7 @@
             return MHSequence<BASE>::m_Values[--MHSequence<BASE>::m_VecSize];
         }
         // Push an element on the stack.
-        void Push(BASE b) { Append(b); }
+        void Push(BASE b) { this->Append(b); }
         // Return the top of the stack.
         BASE Top() { 
             MHASSERT(MHSequence<BASE>::m_VecSize > 0);
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/audioinputoss.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/audioinputoss.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/audioinputoss.cpp	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-12 04:54:13.000000000 -0700
@@ -20,6 +20,8 @@
  * 02110-1301, USA.
  */
 
+#include <unistd.h>
+
 #include "mythconfig.h"
 #if HAVE_SYS_SOUNDCARD_H
     #include <sys/soundcard.h>
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/firewiredevice.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/firewiredevice.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/firewiredevice.cpp	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/firewiredevice.cpp	2012-04-12 04:54:13.000000000 -0700
@@ -4,6 +4,8 @@
  *  Distributed as part of MythTV under GPL v2 and later.
  */
 
+#include <unistd.h>
+
 // C++ headers
 #include <algorithm>
 
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/util-vdpau.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/util-vdpau.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/util-vdpau.cpp	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/util-vdpau.cpp	2012-04-12 05:07:32.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <cstdio>
 #include <cstdlib>
 
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/programs/mythtranscode/transcode.cpp	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/programs/mythtranscode/transcode.cpp	2012-04-12 05:13:39.000000000 -0700
@@ -748,7 +748,7 @@
         // framecontrol is true if we want to enforce fifo sync.
         if (framecontrol)
             VERBOSE(VB_GENERAL, "Enforcing sync on fifos");
-        fifow = new FIFOWriter::FIFOWriter(2, framecontrol);
+        fifow = new FIFOWriter(2, framecontrol);
 
         if (!fifow->FIFOInit(0, QString("video"), vidfifo, vidSize, 50) ||
             !fifow->FIFOInit(1, QString("audio"), audfifo, audio_size, 25))
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/programs/mythtranscode/transcode.h mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/programs/mythtranscode/transcode.h
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/programs/mythtranscode/transcode.h	2011-10-07 15:27:32.000000000 -0700
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/programs/mythtranscode/transcode.h	2012-04-12 05:15:33.000000000 -0700
@@ -38,7 +38,7 @@
     PlayerContext          *player_ctx;
     RingBuffer             *inRingBuffer;
     RingBuffer             *outRingBuffer;
-    FIFOWriter::FIFOWriter *fifow;
+    FIFOWriter             *fifow;
     KFATable               *kfa_table;
     bool                    showprogress;
     QString                 recorderOptions;
