diff -Naur mythtv-0.22-23097-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23097-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23097-old/libs/libmyth/mythcontext.cpp	2010-01-09 19:36:32.000000000 -0800
+++ mythtv-0.22-23097-new/libs/libmyth/mythcontext.cpp	2010-01-09 19:38:59.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
