diff -Naur mythtv-0.24-20120422-e12af86-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-20120422-e12af86-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-20120422-e12af86-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2012-04-28 06:31:47.000000000 -0700
+++ mythtv-0.24-20120422-e12af86-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2012-04-28 06:35:36.000000000 -0700
@@ -1312,6 +1312,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
