diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmyth/programinfoupdater.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmyth/programinfoupdater.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmyth/programinfoupdater.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmyth/programinfoupdater.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "programinfoupdater.h"
 #include "remoteutil.h"
 #include "compat.h"
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/httpcomms.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/httpcomms.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/httpcomms.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/httpcomms.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/lcddevice.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/lcddevice.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/lcddevice.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/lcddevice.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -7,6 +7,8 @@
     (c) 2002, 2003 Thor Sigvaldason, Dan Morphis and Isaac Richards
 */
 
+#include <unistd.h>
+
 // ANSI C headers
 #include <cstdlib>
 #include <cmath>
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythdbcon.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythdbcon.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythdbcon.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythdbcon.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythdownloadmanager.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythdownloadmanager.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythdownloadmanager.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythdownloadmanager.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // qt
 #include <QCoreApplication>
 #include <QString>
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythsocket.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythsocket.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythsocket.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythsocket.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 #include <cassert>
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythsocketthread.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythsocketthread.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/mythsocketthread.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/mythsocketthread.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/remotefile.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/remotefile.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythdb/remotefile.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythdb/remotefile.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythtv/audioinputoss.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythtv/audioinputoss.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -20,6 +20,8 @@
  * 02110-1301, USA.
  */
 
+#include <unistd.h>
+
 #include "mythconfig.h"
 #if HAVE_SYS_SOUNDCARD_H
     #include <sys/soundcard.h>
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythui/mythrender_vdpau.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythui/mythrender_vdpau.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythui/mythrender_vdpau.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythui/mythrender_vdpau.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "math.h"
 
 #include <QSize>
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythupnp/upnpdevice.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythupnp/upnpdevice.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythupnp/upnpdevice.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythupnp/upnpdevice.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -21,6 +21,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include "upnp.h"
 #include "upnpdevice.h"
 #include "httpcomms.h"
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythupnp/upnptasknotify.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythupnp/upnptasknotify.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythupnp/upnptasknotify.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythupnp/upnptasknotify.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -21,6 +21,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 // ANSI C headers
 #include <cstdlib>
 
diff -Naur mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythupnp/upnptasksearch.cpp mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythupnp/upnptasksearch.cpp
--- mythtv-0.24-20120315-322de47-931028f-old/mythtv/libs/libmythupnp/upnptasksearch.cpp	2012-04-01 08:43:34.000000000 -0700
+++ mythtv-0.24-20120315-322de47-931028f-new/mythtv/libs/libmythupnp/upnptasksearch.cpp	2012-04-01 08:50:29.000000000 -0700
@@ -21,6 +21,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include <compat.h>
 #include <stdlib.h>
 
