diff -Naur mythtv-0.24-20120422-e12af86-931028f-old/mythtv/libs/libmythtv/vsync.cpp mythtv-0.24-20120422-e12af86-931028f-new/mythtv/libs/libmythtv/vsync.cpp
--- mythtv-0.24-20120422-e12af86-931028f-old/mythtv/libs/libmythtv/vsync.cpp	2012-04-28 06:35:47.000000000 -0700
+++ mythtv-0.24-20120422-e12af86-931028f-new/mythtv/libs/libmythtv/vsync.cpp	2012-04-28 06:35:49.000000000 -0700
@@ -100,7 +100,11 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+    if (gCoreContext->GetNumSetting("UseDRMVSync", 1) &&
+       (getenv("NO_DRM_VSYNC") == NULL))
+    {
+        TESTVIDEOSYNC(DRMVideoSync);
+    }
 #ifdef USING_OPENGL_VSYNC
 /*
     if (gCoreContext->GetNumSetting("UseOpenGLVSync", 1) &&
diff -Naur mythtv-0.24-20120422-e12af86-931028f-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-0.24-20120422-e12af86-931028f-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.24-20120422-e12af86-931028f-old/mythtv/programs/mythfrontend/globalsettings.cpp	2012-04-28 06:35:47.000000000 -0700
+++ mythtv-0.24-20120422-e12af86-931028f-new/mythtv/programs/mythfrontend/globalsettings.cpp	2012-04-28 06:35:49.000000000 -0700
@@ -3512,6 +3512,18 @@
     };
 };
 
+static HostCheckBox *UseDRMVSync()
+{
+    HostCheckBox *gc = new HostCheckBox("UseDRMVSync");
+    gc->setLabel(QObject::tr("Enable DRM vertical sync for timing"));
+    gc->setValue(false);
+    gc->setHelpText(QObject::tr(
+                        "If it is supported by your hardware/drivers, "
+                        "MythTV will use DRM vertical syncing for "
+                        "video timing, reducing frame jitter."));
+    return gc;
+}
+
 #ifdef USING_OPENGL_VSYNC
 /*
 static HostCheckBox *UseOpenGLVSync()
@@ -4039,6 +4051,7 @@
 
     general1->addChild(columns);
     general1->addChild(LiveTVIdleTimeout());
+    general1->addChild(UseDRMVSync());
 #ifdef USING_OPENGL_VSYNC
     //general1->addChild(UseOpenGLVSync());
 #endif // USING_OPENGL_VSYNC
