diff -Naur mythtv-0.24-20120221-c29d36f-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-20120221-c29d36f-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-20120221-c29d36f-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2012-02-29 17:59:04.000000000 -0800
+++ mythtv-0.24-20120221-c29d36f-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2012-02-29 18:01:35.000000000 -0800
@@ -1308,6 +1308,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
