diff -Naur mythtv-0.24-20110617-958dd8c-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-20110617-958dd8c-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-20110617-958dd8c-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-06-18 10:29:38.000000000 -0700
+++ mythtv-0.24-20110617-958dd8c-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-06-18 13:42:18.000000000 -0700
@@ -1305,6 +1305,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
