diff -Naur libX11-1.1.4-old/configure.ac libX11-1.1.4-new/configure.ac
--- libX11-1.1.4-old/configure.ac	2008-03-06 12:48:26.000000000 -0800
+++ libX11-1.1.4-new/configure.ac	2008-04-22 20:45:40.000000000 -0700
@@ -38,6 +38,7 @@
 case "$ac_cv_use_xcb" in
 no)
 	X11_REQUIRES="xau xcmiscproto bigreqsproto"
+	X11_EXTRA_DEPS="xau xdmcp"
 	PKG_CHECK_MODULES(XDMCP, xdmcp)
 	AC_CHECK_LIB(Xdmcp, XdmcpWrap, [xdmauth="yes"], [xdmauth="no"], [$XDMCP_LIBS])
 	AC_DEFINE(USE_XCB, 0, [Use XCB for low-level protocol implementation])
diff -Naur libX11-1.1.4-old/x11.pc.in libX11-1.1.4-new/x11.pc.in
--- libX11-1.1.4-old/x11.pc.in	2008-03-06 12:45:06.000000000 -0800
+++ libX11-1.1.4-new/x11.pc.in	2008-04-22 20:44:51.000000000 -0700
@@ -9,7 +9,7 @@
 Description: X Library
 Version: @PACKAGE_VERSION@
 Requires: xproto @XKBPROTO_REQUIRES@
-Requires.private: xau xdmcp @X11_EXTRA_DEPS@
+Requires.private: @X11_EXTRA_DEPS@
 Cflags: -I${includedir} @XTHREAD_CFLAGS@
 Libs: -L${libdir} -lX11
 Libs.private: @XTHREADLIB@
