diff -Naur Mesa-7.8.2-old/progs/egl/Makefile Mesa-7.8.2-new/progs/egl/Makefile
--- Mesa-7.8.2-old/progs/egl/Makefile	2010-06-15 10:43:41.000000000 -0700
+++ Mesa-7.8.2-new/progs/egl/Makefile	2010-06-21 14:56:31.000000000 -0700
@@ -60,7 +60,7 @@
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LIBS) -lm $(X11_LIBS)
 
 xeglthreads: xeglthreads.o $(HEADERS) $(LIB_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LIBS) -lm $(X11_LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LIBS) -lpthread -lm $(X11_LIBS)
 
 xegl_tri: xegl_tri.o $(HEADERS) $(LIB_DEP)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LIBS) -lm $(X11_LIBS)
