diff -Naur dbus-1.2.10-old/bus/Makefile.am dbus-1.2.10-new/bus/Makefile.am
--- dbus-1.2.10-old/bus/Makefile.am	2008-08-07 11:44:35.000000000 -0700
+++ dbus-1.2.10-new/bus/Makefile.am	2008-12-24 11:18:57.000000000 -0800
@@ -1,7 +1,7 @@
 
 configdir=$(sysconfdir)/dbus-1
 
-INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDAEMON_NAME=\"dbus-daemon\"  -DDBUS_COMPILATION
 
@@ -81,7 +81,7 @@
 	$(DBUS_BUS_LIBS)				\
 	$(top_builddir)/dbus/libdbus-convenience.la
 
-dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 
 LAUNCH_HELPER_SOURCES=				\
 	$(XML_SOURCES)				\
diff -Naur dbus-1.2.10-old/bus/Makefile.in dbus-1.2.10-new/bus/Makefile.in
--- dbus-1.2.10-old/bus/Makefile.in	2008-12-19 11:36:16.000000000 -0800
+++ dbus-1.2.10-new/bus/Makefile.in	2008-12-24 11:18:51.000000000 -0800
@@ -416,7 +416,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 configdir = $(sysconfdir)/dbus-1
-INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDAEMON_NAME=\"dbus-daemon\"  -DDBUS_COMPILATION
 
@@ -480,7 +480,7 @@
 	$(DBUS_BUS_LIBS)				\
 	$(top_builddir)/dbus/libdbus-convenience.la
 
-dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 LAUNCH_HELPER_SOURCES = \
 	$(XML_SOURCES)				\
 	config-parser-common.c			\
