diff -Naur dbus-1.4.1-old/bus/Makefile.am dbus-1.4.1-new/bus/Makefile.am
--- dbus-1.4.1-old/bus/Makefile.am	2010-12-13 15:24:54.000000000 -0800
+++ dbus-1.4.1-new/bus/Makefile.am	2010-12-20 17:12:14.000000000 -0800
@@ -1,7 +1,7 @@
 
 configdir=$(sysconfdir)/dbus-1
 
-INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDBUS_COMPILATION
 
@@ -88,7 +88,7 @@
 	$(EFENCE)					\
 	$(DBUS_BUS_LIBS)
 
-dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 
 LAUNCH_HELPER_SOURCES=				\
 	$(XML_SOURCES)				\
diff -Naur dbus-1.4.1-old/bus/Makefile.in dbus-1.4.1-new/bus/Makefile.in
--- dbus-1.4.1-old/bus/Makefile.in	2010-12-20 13:39:49.000000000 -0800
+++ dbus-1.4.1-new/bus/Makefile.in	2010-12-20 17:12:14.000000000 -0800
@@ -513,7 +513,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 configdir = $(sysconfdir)/dbus-1
-INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDBUS_COMPILATION
 
@@ -581,7 +581,7 @@
 	$(EFENCE)					\
 	$(DBUS_BUS_LIBS)
 
-dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 LAUNCH_HELPER_SOURCES = \
 	$(XML_SOURCES)				\
 	config-parser-common.c			\
