diff -Naur myththemes-addons-0.21-19162-old/configure myththemes-addons-0.21-19162-new/configure
--- myththemes-addons-0.21-19162-old/configure	2008-11-26 17:35:12.000000000 -0800
+++ myththemes-addons-0.21-19162-new/configure	2008-11-26 17:36:07.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
