diff -Naur myththemes-addons-0.21-19540-old/cpsvndir myththemes-addons-0.21-19540-new/cpsvndir
--- myththemes-addons-0.21-19540-old/cpsvndir	2009-01-02 14:20:40.000000000 -0800
+++ myththemes-addons-0.21-19540-new/cpsvndir	2009-01-02 14:20:41.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
