diff -Naur myththemes-addons-0.21-17237-old/cpsvndir myththemes-addons-0.21-17237-new/cpsvndir
--- myththemes-addons-0.21-17237-old/cpsvndir	2008-05-04 09:47:15.000000000 -0700
+++ myththemes-addons-0.21-17237-new/cpsvndir	2008-05-04 09:47:16.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
