diff -Naur myththemes-addons-0.21-19961-old/configure myththemes-addons-0.21-19961-new/configure
--- myththemes-addons-0.21-19961-old/configure	2009-02-10 06:22:11.000000000 -0800
+++ myththemes-addons-0.21-19961-new/configure	2009-02-10 06:22:58.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
