diff -Naur myththemes-addons-0.21-19706-old/configure myththemes-addons-0.21-19706-new/configure
--- myththemes-addons-0.21-19706-old/configure	2009-01-17 19:39:38.000000000 -0800
+++ myththemes-addons-0.21-19706-new/configure	2009-01-17 19:40:34.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
