diff -Naur myththemes-addons-0.21-20278-old/configure myththemes-addons-0.21-20278-new/configure
--- myththemes-addons-0.21-20278-old/configure	2009-03-28 08:18:19.000000000 -0700
+++ myththemes-addons-0.21-20278-new/configure	2009-03-28 09:53:18.000000000 -0700
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
