diff -Naur myththemes-addons-0.21-19133-old/configure myththemes-addons-0.21-19133-new/configure
--- myththemes-addons-0.21-19133-old/configure	2008-11-19 19:13:58.000000000 -0800
+++ myththemes-addons-0.21-19133-new/configure	2008-11-19 19:14:57.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
