diff -Naur myththemes-addons-0.21-19258-old/configure myththemes-addons-0.21-19258-new/configure
--- myththemes-addons-0.21-19258-old/configure	2008-12-08 19:28:03.000000000 -0800
+++ myththemes-addons-0.21-19258-new/configure	2008-12-08 19:28:59.000000000 -0800
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
