diff -Naur myththemes-addons-0.21-17445-old/configure myththemes-addons-0.21-17445-new/configure
--- myththemes-addons-0.21-17445-old/configure	2008-06-04 19:43:33.000000000 -0700
+++ myththemes-addons-0.21-17445-new/configure	2008-06-04 19:44:31.000000000 -0700
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
