diff -Naur myththemes-addons-0.21-17595-old/configure myththemes-addons-0.21-17595-new/configure
--- myththemes-addons-0.21-17595-old/configure	2008-06-23 16:45:06.000000000 -0700
+++ myththemes-addons-0.21-17595-new/configure	2008-06-23 16:46:03.000000000 -0700
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
