diff -Naur mythplugins-0.20-16082-old/mythbrowser/mythbrowser/mythbrowser.pro mythplugins-0.20-16082-new/mythbrowser/mythbrowser/mythbrowser.pro
--- mythplugins-0.20-16082-old/mythbrowser/mythbrowser/mythbrowser.pro	2008-09-15 14:05:48.000000000 -0700
+++ mythplugins-0.20-16082-new/mythbrowser/mythbrowser/mythbrowser.pro	2010-05-28 15:22:13.000000000 -0700
@@ -16,7 +16,7 @@
 INSTALLS += installimages
 
 LIBS += -lmyth-$$LIBVERSION -lmythui-$$LIBVERSION $$EXTRA_LIBS
-LIBS += -L/opt/kde3/lib -lkhtml
+LIBS += -L/opt/kde3/lib -lkhtml -lkdecore
 
 # Input
 HEADERS += webpage.h tabview.h
