diff -Naur mythvideo-0.17-old/mythvideo/metadata.h mythvideo-0.17-new/mythvideo/metadata.h
--- mythvideo-0.17-old/mythvideo/metadata.h	2005-02-09 13:35:13.000000000 -0800
+++ mythvideo-0.17-new/mythvideo/metadata.h	2005-03-23 06:12:34.524273217 -0800
@@ -78,17 +78,17 @@
         filename = "";
         coverfile = "";
         title = "";
-        year = 1895;
+        year = 0;
         inetref = "";
         director = "";
         plot = "";
-        userrating = 0;
+        userrating = 0.0;
         rating = "";
         length = 0;
         showlevel = 1;
         id = 0;
-        childID = 1;
-        browse = 1;
+        childID = -1;
+        browse = true;
         playcommand = "";
         category = "";
         genres = QStringList();
