diff -Naur which-2.16-old/tilde/tilde.c which-2.16-new/tilde/tilde.c
--- which-2.16-old/tilde/tilde.c	1999-05-27 17:15:55.000000000 -0700
+++ which-2.16-new/tilde/tilde.c	2006-12-20 05:59:04.000000000 -0800
@@ -182,7 +182,10 @@
   int result_size, result_index;
 
   result_index = result_size = 0;
-  if (result = strchr (string, '~'))
+
+  result = strchr (string, '~');
+  
+  if (!result)
     result = xmalloc (result_size = (strlen (string) + 16));
   else
     result = xmalloc (result_size = (strlen (string) + 1));
diff -Naur which-2.16-old/which.c which-2.16-new/which.c
--- which-2.16-old/which.c	2003-09-11 17:58:49.000000000 -0700
+++ which-2.16-new/which.c	2006-12-20 05:59:04.000000000 -0800
@@ -22,6 +22,7 @@
 #include "getopt.h"
 #include "tilde/tilde.h"
 #include "bash.h"
+#include <ctype.h>
 
 static const char *progname;
 
@@ -538,7 +539,7 @@
     while (fgets(buf, sizeof(buf), stdin))
     {
       int looks_like_function_start = 0;
-      int function_start_version_205b;
+      int function_start_version_205b = 0;
       if (read_functions)
       {
 	// bash version 2.0.5a and older output a pattern for `str' like
