diff -Naur rxvt-2.6.4-old/autoconf/Make.common.in rxvt-2.6.4-new/autoconf/Make.common.in
--- rxvt-2.6.4-old/autoconf/Make.common.in	1999-03-27 22:15:42.000000000 -0800
+++ rxvt-2.6.4-new/autoconf/Make.common.in	2008-05-27 17:58:36.000000000 -0700
@@ -44,7 +44,7 @@
 CMP = @CMP@
 TBL = @TBL@
 INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL@ -s -m 755
+INSTALL_PROGRAM = @INSTALL@ -m 755
 INSTALL_DATA = @INSTALL@ -m 644
 
 # Flags & libs
