diff -Naur MPlayer-30512-old/libvo/vo_corevideo.m MPlayer-30512-new/libvo/vo_corevideo.m
--- MPlayer-30512-old/libvo/vo_corevideo.m	2010-02-05 06:27:03.000000000 -0800
+++ MPlayer-30512-new/libvo/vo_corevideo.m	2010-02-05 06:37:30.000000000 -0800
@@ -500,6 +500,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -538,6 +540,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
