diff -Naur MPlayer-31918-old/libvo/vo_corevideo.m MPlayer-31918-new/libvo/vo_corevideo.m
--- MPlayer-31918-old/libvo/vo_corevideo.m	2010-08-03 21:12:34.000000000 -0700
+++ MPlayer-31918-new/libvo/vo_corevideo.m	2010-08-03 21:13:52.000000000 -0700
@@ -502,6 +502,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -538,6 +540,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
