diff -Naur MPlayer-30881-old/libvo/vo_corevideo.m MPlayer-30881-new/libvo/vo_corevideo.m
--- MPlayer-30881-old/libvo/vo_corevideo.m	2010-03-10 17:36:56.000000000 -0800
+++ MPlayer-30881-new/libvo/vo_corevideo.m	2010-03-10 18:05:54.000000000 -0800
@@ -500,6 +500,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -538,6 +540,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
