diff -Naur MPlayer-28461.16984-old/libvo/vo_vdpau.c MPlayer-28461.16984-new/libvo/vo_vdpau.c
--- MPlayer-28461.16984-old/libvo/vo_vdpau.c	2009-02-03 20:16:12.000000000 -0800
+++ MPlayer-28461.16984-new/libvo/vo_vdpau.c	2009-02-03 20:20:11.000000000 -0800
@@ -701,7 +701,7 @@
             uint32_t round_width = (vid_width + 15) & ~15;
             uint32_t round_height = (vid_height + 15) & ~15;
             uint32_t surf_size = (round_width * round_height * 3) / 2;
-            max_references = (12 * 1024 * 1024) / surf_size;
+            max_references = (12 * 1024 * 1024) / surf_size + 11;
             if (max_references > 16) {
                 max_references = 16;
             }
