diff -Naur MPlayer-32037-old/libvo/vo_corevideo.m MPlayer-32037-new/libvo/vo_corevideo.m
--- MPlayer-32037-old/libvo/vo_corevideo.m	2010-09-01 09:39:08.000000000 -0700
+++ MPlayer-32037-new/libvo/vo_corevideo.m	2010-09-01 09:40:26.000000000 -0700
@@ -502,6 +502,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -538,6 +540,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
