diff -Naur faad2-old/Makefile.am faad2-new/Makefile.am
--- faad2-old/Makefile.am	2004-02-06 08:04:39.000000000 -0800
+++ faad2-new/Makefile.am	2005-07-21 05:57:44.000000000 -0700
@@ -2,7 +2,7 @@
 
 EXTRA_DIST = faad2.spec
 
-rpm:    Makefile
-       make dist
-       $(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
-       rm $(PACKAGE)-$(VERSION).tar.gz
+rpm: Makefile
+	make dist
+	$(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
+	rm $(PACKAGE)-$(VERSION).tar.gz
