diff -Naur linux-2.6.28.1-old/include/linux/serial.h linux-2.6.28.1-new/include/linux/serial.h
--- linux-2.6.28.1-old/include/linux/serial.h	2009-01-18 11:05:24.000000000 -0800
+++ linux-2.6.28.1-new/include/linux/serial.h	2009-01-18 11:06:45.000000000 -0800
@@ -10,8 +10,9 @@
 #ifndef _LINUX_SERIAL_H
 #define _LINUX_SERIAL_H
 
-#ifdef __KERNEL__
 #include <linux/types.h>
+
+#ifdef __KERNEL__
 #include <asm/page.h>
 
 /*
