diff -Naur xine-lib-1.1.4-old/m4/optimizations.m4 xine-lib-1.1.4-new/m4/optimizations.m4
--- xine-lib-1.1.4-old/m4/optimizations.m4	2006-06-17 08:20:56.000000000 -0700
+++ xine-lib-1.1.4-new/m4/optimizations.m4	2007-02-17 20:21:40.000000000 -0800
@@ -20,8 +20,8 @@
         dnl eg: -mpreferred-stack-boundary=2 and 2.91.66,
         dnl and gcc-2.7.2.3 support a bit less options
         dnl
-        AC_TRY_CFLAGS("-mpreferred-stack-boundary=2",
-            m_psb="-mpreferred-stack-boundary=2", m_psb="")
+dnl        AC_TRY_CFLAGS("-mpreferred-stack-boundary=2",
+dnl            m_psb="-mpreferred-stack-boundary=2", m_psb="")
         AC_TRY_CFLAGS("-fschedule-insns2", f_si="-fschedule-insns2", f_si="")
         AC_TRY_CFLAGS("-mwide-multiply", m_wm="-mwide-multiply", m_wm="")
         dnl
