diff -Naur ratpoison-1.4.4-old/configure.in ratpoison-1.4.4-new/configure.in
--- ratpoison-1.4.4-old/configure.in	2009-04-26 18:55:10.000000000 -0700
+++ ratpoison-1.4.4-new/configure.in	2009-06-23 06:28:54.000000000 -0700
@@ -146,7 +146,7 @@
 dnl Checks for typedefs, structures, and compiler characteristics.
 
 dnl Checks for library functions.
-AC_CHECK_FUNCS(getopt getopt_long setsid setpgid setpgrp putenv vsnprintf usleep)
+AC_CHECK_FUNCS(getline getopt getopt_long setsid setpgid setpgrp putenv vsnprintf usleep)
 
 AC_TYPE_SIGNAL
 
