diff -Naur ratpoison-1.4.1-old/src/actions.c ratpoison-1.4.1-new/src/actions.c
--- ratpoison-1.4.1-old/src/actions.c	2006-12-14 07:01:15.000000000 -0800
+++ ratpoison-1.4.1-new/src/actions.c	2007-01-25 20:04:27.000000000 -0800
@@ -1799,6 +1799,8 @@
 {
   int ret = -1;
 
+  if (!strcasecmp (data, "static") || !strcasecmp (data, "static") || !strcmp (data, "10"))
+    ret = StaticGravity;
   if (!strcasecmp (data, "northwest") || !strcasecmp (data, "nw") || !strcmp (data, "7"))
     ret = NorthWestGravity;
   if (!strcasecmp (data, "north") || !strcasecmp (data, "n") || !strcmp (data, "8"))
diff -Naur ratpoison-1.4.1-old/src/manage.c ratpoison-1.4.1-new/src/manage.c
--- ratpoison-1.4.1-old/src/manage.c	2006-12-19 16:55:21.000000000 -0800
+++ ratpoison-1.4.1-new/src/manage.c	2007-01-25 20:07:15.000000000 -0800
@@ -567,6 +567,9 @@
     case SouthEastGravity:
       win->x = frame->x + frame->width - win->width - win->border;
       break;
+    case StaticGravity:
+      win->x = frame->x + win->x;
+      break;
     }
 
   /* Y coord. */
@@ -587,6 +590,9 @@
     case SouthWestGravity:
       win->y = frame->y + frame->height - win->height - win->border;
       break;
+    case StaticGravity:
+      win->y = frame->y + win->y;
+      break;
     }
 }
 
