diff -Naur alsa-driver-1.0.10-old/Makefile alsa-driver-1.0.10-new/Makefile
--- alsa-driver-1.0.10-old/Makefile	2005-08-12 09:11:57.000000000 -0700
+++ alsa-driver-1.0.10-new/Makefile	2005-12-05 20:55:11.000000000 -0800
@@ -197,9 +197,9 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
-	-/sbin/depmod -a -b $(DESTDIR)/ $(SYSTEM_MAP_OPT) $(kaversion)
+	depmod -a -b $(DESTDIR)/ $(SYSTEM_MAP_OPT) $(kaversion)
 endif
 
 .PHONY: install-scripts
