diff -Naur alsa-driver-1.0.11rc4-old/Makefile alsa-driver-1.0.11rc4-new/Makefile
--- alsa-driver-1.0.11rc4-old/Makefile	2005-11-17 03:15:20.000000000 -0800
+++ alsa-driver-1.0.11rc4-new/Makefile	2006-03-23 18:58:36.000000000 -0800
@@ -194,9 +194,9 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
-	-/sbin/depmod -a -b $(DESTDIR)/ $(SYSTEM_MAP_OPT) $(kaversion)
+	depmod -a -b $(DESTDIR)/ $(SYSTEM_MAP_OPT) $(kaversion)
 endif
 
 .PHONY: install-scripts
