diff -Naur alsa-driver-1.0.8-old/configure alsa-driver-1.0.8-new/configure
--- alsa-driver-1.0.8-old/configure	2005-01-13 02:02:09.000000000 -0800
+++ alsa-driver-1.0.8-new/configure	2005-01-25 15:52:20.740586049 -0800
@@ -5342,6 +5342,8 @@
 	fprintf(file, "mwinchip");
 #elif defined(CONFIG_MCYRIXIII)
 	fprintf(file, "mcyrixiii");
+#elif defined(CONFIG_MVIAC3_2)
+	fprintf(file, "mviac3_2");
 #elif defined(CONFIG_IA64)
 	fprintf(file, "ia64");
 #elif defined(CONFIG_ALPHA_GENERIC)
@@ -5478,6 +5480,19 @@
       c_opts="$c_opts -malign-functions=0 -malign-jumps=0 -malign-loops=0"
     fi
     ;;
+  mviac3_2)
+    ARCH=i386
+    if $KCC -march=c3-2 -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then
+      c_opts="-march=c3-2"
+    else
+      c_opts="-march=i486"
+    fi
+    if $KCC -falign-functions=0 -falign-jumps=0 -falign-loops=0 -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then
+      c_opts="$c_opts -falign-functions=0 -falign-jumps=0 -falign-loops=0"
+    else
+      c_opts="$c_opts -malign-functions=0 -malign-jumps=0 -malign-loops=0"
+    fi
+    ;;
   ia64)
     ARCH=ia64
     c_opts="-Wa,-x -ffixed-r13 -mfixed-range=f10-f15,f32-f127 -funwind-tables -falign-functions=32"
@@ -5635,7 +5650,7 @@
 esac
 
 case "$processor" in
-  i?86*|k?|crusoe|mcyrixiii|mwinchip)
+  i?86*|k?|crusoe|mcyrixiii|mviac3_2|mwinchip)
     if $KCC -mpreferred-stack-boundary=2 -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then
       c_opts="-mpreferred-stack-boundary=2 $c_opts"
       CONFIG_X86=y
