diff -Naur alsa-driver-1.0.23-old/alsa-kernel/pci/hda/patch_hdmi.c alsa-driver-1.0.23-new/alsa-kernel/pci/hda/patch_hdmi.c
--- alsa-driver-1.0.23-old/alsa-kernel/pci/hda/patch_hdmi.c	2010-08-04 08:08:13.000000000 -0700
+++ alsa-driver-1.0.23-new/alsa-kernel/pci/hda/patch_hdmi.c	2010-08-04 08:12:09.000000000 -0700
@@ -52,6 +52,10 @@
 	 */
 	struct hda_multi_out multiout;
 	unsigned int codec_type;
+
+	/* misc flags */
+	/* PD bit indicates only the update, not the current state */
+	unsigned int old_pin_detect:1;
 };
 
 
@@ -616,6 +620,9 @@
  * Unsolicited events
  */
 
+static void hdmi_present_sense(struct hda_codec *codec, hda_nid_t pin_nid,
+			       struct hdmi_eld *eld);
+
 static void hdmi_intrinsic_event(struct hda_codec *codec, unsigned int res)
 {
 	struct hdmi_spec *spec = codec->spec;
@@ -632,6 +639,12 @@
 	if (index < 0)
 		return;
 
+	if (spec->old_pin_detect) {
+		if (pind)
+			hdmi_present_sense(codec, tag, &spec->sink_eld[index]);
+		pind = spec->sink_eld[index].monitor_present;
+	}
+
 	spec->sink_eld[index].monitor_present = pind;
 	spec->sink_eld[index].eld_valid = eldv;
 
diff -Naur alsa-driver-1.0.23-old/alsa-kernel/pci/hda/patch_nvhdmi.c alsa-driver-1.0.23-new/alsa-kernel/pci/hda/patch_nvhdmi.c
--- alsa-driver-1.0.23-old/alsa-kernel/pci/hda/patch_nvhdmi.c	2010-08-04 08:08:13.000000000 -0700
+++ alsa-driver-1.0.23-new/alsa-kernel/pci/hda/patch_nvhdmi.c	2010-08-04 08:12:09.000000000 -0700
@@ -478,6 +478,7 @@
 
 	codec->spec = spec;
 	spec->codec_type = HDA_CODEC_NVIDIA_MCP89;
+	spec->old_pin_detect = 1;
 
 	if (hdmi_parse_codec(codec) < 0) {
 		codec->spec = NULL;
@@ -508,6 +509,7 @@
 	spec->multiout.max_channels = 8;
 	spec->multiout.dig_out_nid = nvhdmi_master_con_nid_7x;
 	spec->codec_type = HDA_CODEC_NVIDIA_MCP7X;
+	spec->old_pin_detect = 1;
 
 	codec->patch_ops = nvhdmi_patch_ops_8ch_7x;
 
@@ -528,6 +530,7 @@
 	spec->multiout.max_channels = 2;
 	spec->multiout.dig_out_nid = nvhdmi_master_con_nid_7x;
 	spec->codec_type = HDA_CODEC_NVIDIA_MCP7X;
+	spec->old_pin_detect = 1;
 
 	codec->patch_ops = nvhdmi_patch_ops_2ch;
 
