diff -Naur libbluray-24f5fa1e65fefb5423fa7c275256b5c0b1d93410-old/configure.ac libbluray-24f5fa1e65fefb5423fa7c275256b5c0b1d93410-new/configure.ac
--- libbluray-24f5fa1e65fefb5423fa7c275256b5c0b1d93410-old/configure.ac	2010-11-12 13:28:34.000000000 -0800
+++ libbluray-24f5fa1e65fefb5423fa7c275256b5c0b1d93410-new/configure.ac	2010-11-12 13:29:15.000000000 -0800
@@ -33,7 +33,7 @@
 using_normal_linking="Using libaacs and libbdplus via normal linking."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
