diff -Naur libbluray-b2fb25b-old/configure.ac libbluray-b2fb25b-new/configure.ac
--- libbluray-b2fb25b-old/configure.ac	2011-02-09 15:39:37.000000000 -0800
+++ libbluray-b2fb25b-new/configure.ac	2011-02-09 15:42:57.000000000 -0800
@@ -35,7 +35,7 @@
 pkg_config_missing="Could not find pkg-config.  Assuming..."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
