diff -Naur libbluray-496f127a1cdb5a957d6be977b0cb786683a96415-old/configure.ac libbluray-496f127a1cdb5a957d6be977b0cb786683a96415-new/configure.ac
--- libbluray-496f127a1cdb5a957d6be977b0cb786683a96415-old/configure.ac	2010-11-15 22:19:23.000000000 -0800
+++ libbluray-496f127a1cdb5a957d6be977b0cb786683a96415-new/configure.ac	2010-11-15 22:20:10.000000000 -0800
@@ -33,7 +33,7 @@
 using_normal_linking="Using libaacs and libbdplus via normal linking."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
