diff -Naur libbluray-4ba0af806cd244304a27378f6814876c78ebb049-old/configure.ac libbluray-4ba0af806cd244304a27378f6814876c78ebb049-new/configure.ac
--- libbluray-4ba0af806cd244304a27378f6814876c78ebb049-old/configure.ac	2010-12-08 13:03:28.000000000 -0800
+++ libbluray-4ba0af806cd244304a27378f6814876c78ebb049-new/configure.ac	2010-12-08 13:03:55.000000000 -0800
@@ -33,7 +33,7 @@
 using_normal_linking="Using libaacs and libbdplus via normal linking."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
