diff -Naur libbluray-230290238060eafc3786a67d0d46687d86e0194c-old/configure.ac libbluray-230290238060eafc3786a67d0d46687d86e0194c-new/configure.ac
--- libbluray-230290238060eafc3786a67d0d46687d86e0194c-old/configure.ac	2010-10-18 13:52:36.000000000 -0700
+++ libbluray-230290238060eafc3786a67d0d46687d86e0194c-new/configure.ac	2010-10-18 16:17:23.000000000 -0700
@@ -33,7 +33,7 @@
 using_normal_linking="Using libaacs and libbdplus via normal linking."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
