diff -Naur libbluray-e4f0877177eff975f0a6ba1dcf12c84892dfd25d-old/configure.ac libbluray-e4f0877177eff975f0a6ba1dcf12c84892dfd25d-new/configure.ac
--- libbluray-e4f0877177eff975f0a6ba1dcf12c84892dfd25d-old/configure.ac	2010-10-17 05:44:26.000000000 -0700
+++ libbluray-e4f0877177eff975f0a6ba1dcf12c84892dfd25d-new/configure.ac	2010-10-17 06:06:43.000000000 -0700
@@ -33,7 +33,7 @@
 using_normal_linking="Using libaacs and libbdplus via normal linking."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
