diff -Naur mythstream-0.18_1-old/parsers/youtube/vid.pl mythstream-0.18_1-new/parsers/youtube/vid.pl
--- mythstream-0.18_1-old/parsers/youtube/vid.pl	2008-02-11 13:22:11.000000000 -0800
+++ mythstream-0.18_1-new/parsers/youtube/vid.pl	2008-04-06 15:03:45.000000000 -0700
@@ -59,7 +59,7 @@
 # BASE_YT_URL:'http://youtube.com/',video_id:'iYWrMWH6rGA',l:'83',t:'OEgsToPDskL33Dsy3NGyvOx2YJG_MEtK',sk:
 my $a, $b;
 
-if ( ($a, $b) = $data =~ /"video\_id":\s*"([^"]+)"[^t]+t":\s*"([^"]+)"/ )
+if ( ($a, $b) = $data =~ /,\s*"video\_id":\s*"([^"]+)".*,\s*"t":\s*"([^"]+)"/)
 {
       $url = "http://www.youtube.com/get_video?video_id=$a&t=$b";
 
diff -Naur mythstream-0.18_1-old/parsers/youtube/vid_p.pl mythstream-0.18_1-new/parsers/youtube/vid_p.pl
--- mythstream-0.18_1-old/parsers/youtube/vid_p.pl	2008-01-13 08:22:55.000000000 -0800
+++ mythstream-0.18_1-new/parsers/youtube/vid_p.pl	2008-04-06 15:02:26.000000000 -0700
@@ -11,4 +11,4 @@
 
 $playlist_enable = 1;
 
-require "youtube/youvid.pl";
+require "youtube/vid.pl";
