diff -Naur mythstream-0.16_5-old/mythstream/main.cpp mythstream-0.16_5-new/mythstream/main.cpp
--- mythstream-0.16_5-old/mythstream/main.cpp	2005-05-14 06:22:51.000000000 -0700
+++ mythstream-0.16_5-new/mythstream/main.cpp	2006-05-22 19:24:30.000000000 -0700
@@ -35,7 +35,7 @@
 #include <mythtv/mythcontext.h>
 #include <mythtv/mythdialogs.h>
 #include <mythtv/mythplugin.h>
-#include <mythtv/themedmenu.h>
+#include <mythtv/libmythui/myththemedmenu.h>
 
 using namespace std;
 
@@ -73,8 +73,8 @@
 {
     QString themedir = gContext->GetThemeDir();
 
-    ThemedMenu *diag = new ThemedMenu(themedir.ascii(), which_menu,
-                                      gContext->GetMainWindow(), "stream menu");
+    MythThemedMenu *diag = new MythThemedMenu(themedir.ascii(), which_menu,
+                                      GetMythMainWindow()->GetMainStack(), "stream menu");
 
     QString data;
 
@@ -83,11 +83,13 @@
 
     if (diag->foundTheme())
     {
-        diag->exec();
-        delete diag;
+        GetMythMainWindow()->GetMainStack()->AddScreen(diag);
     }
     else
+    {
         cerr << "Couldn't find theme " << themedir << endl;
+        delete diag;
+    }
 
 }
 
