diff -Naur xf86-video-intel-2.8.1-old/configure.ac xf86-video-intel-2.8.1-new/configure.ac
--- xf86-video-intel-2.8.1-old/configure.ac	2009-08-25 16:56:32.000000000 -0700
+++ xf86-video-intel-2.8.1-new/configure.ac	2009-08-25 18:40:55.000000000 -0700
@@ -45,6 +45,7 @@
 DOLT
 AC_PROG_CC
 AM_PROG_CC_C_O
+PKG_PROG_PKG_CONFIG
 
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
 
@@ -102,8 +103,9 @@
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.6 xproto fontsproto $REQUIRED_MODULES])
 
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
-drm_cflags=$(pkg-config --cflags libdrm)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
+drm_cflags=$($PKG_CONFIG --cflags libdrm)
 
 # Checks for libraries.
 
