diff -Naur xf86-video-intel-2.7.1-old/configure.ac xf86-video-intel-2.7.1-new/configure.ac
--- xf86-video-intel-2.7.1-old/configure.ac	2009-05-12 17:54:57.000000000 -0700
+++ xf86-video-intel-2.7.1-new/configure.ac	2009-05-12 17:55:41.000000000 -0700
@@ -40,6 +40,7 @@
 DOLT
 AC_PROG_CC
 AM_PROG_CC_C_O
+PKG_PROG_PKG_CONFIG
 
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
 AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
@@ -92,8 +93,9 @@
    AC_DEFINE(SERVER_1_5, 1, [Building against server 1.5])
 fi
 
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
-drm_cflags=$(pkg-config --cflags libdrm)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
+drm_cflags=$($PKG_CONFIG --cflags libdrm)
 
 # Checks for libraries.
 
