diff -Naur xf86-video-ati-6.6.3-old/configure.ac xf86-video-ati-6.6.3-new/configure.ac
--- xf86-video-ati-6.6.3-old/configure.ac	2006-10-03 08:01:32.000000000 -0700
+++ xf86-video-ati-6.6.3-new/configure.ac	2008-01-22 17:21:56.000000000 -0800
@@ -47,6 +47,7 @@
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+PKG_PROG_PKG_CONFIG
 
 AH_TOP([#include "xorg-server.h"])
 
@@ -77,7 +78,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server xproto fontsproto $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 
