diff -Naur xf86-input-evdev-2.1.3-old/src/evdev.h xf86-input-evdev-2.1.3-new/src/evdev.h
--- xf86-input-evdev-2.1.3-old/src/evdev.h	2009-02-16 22:13:39.000000000 -0800
+++ xf86-input-evdev-2.1.3-new/src/evdev.h	2009-02-26 08:55:22.000000000 -0800
@@ -37,7 +37,7 @@
 #include <xf86_OSproc.h>
 
 #ifdef XKB
-#include <xkbstr.h>
+#include <X11/extensions/XKBstr.h>
 #endif
 
 #ifndef EV_CNT /* linux 2.4 kernels and earlier lack _CNT defines */
