diff -Naur xf86-input-evdev-2.2.0-old/src/evdev.c xf86-input-evdev-2.2.0-new/src/evdev.c
--- xf86-input-evdev-2.2.0-old/src/evdev.c	2009-03-08 16:11:15.000000000 -0700
+++ xf86-input-evdev-2.2.0-new/src/evdev.c	2009-03-09 21:30:36.000000000 -0700
@@ -263,8 +263,17 @@
     static char warned[KEY_CNT];
 
     /* Filter all repeated events from device.
-       We'll do softrepeat in the server */
-    if (value == 2)
+       We'll do softrepeat in the server, but only since 1.6 */
+    if (value == 2
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) <= 2
+        && (ev->code == KEY_LEFTCTRL || ev->code == KEY_RIGHTCTRL ||
+            ev->code == KEY_LEFTSHIFT || ev->code == KEY_RIGHTSHIFT ||
+            ev->code == KEY_LEFTALT || ev->code == KEY_RIGHTALT ||
+            ev->code == KEY_LEFTMETA || ev->code == KEY_RIGHTMETA ||
+            ev->code == KEY_CAPSLOCK || ev->code == KEY_NUMLOCK ||
+            ev->code == KEY_SCROLLLOCK) /* XXX windows keys? */
+#endif
+            )
 	return;
 
     if (code > 255)
