diff -Naur xf86-input-evdev-2.0.8-old/configure.ac xf86-input-evdev-2.0.8-new/configure.ac
--- xf86-input-evdev-2.0.8-old/configure.ac	2008-11-18 16:28:49.000000000 -0800
+++ xf86-input-evdev-2.0.8-new/configure.ac	2008-12-07 21:31:17.000000000 -0800
@@ -76,7 +76,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, xorg-server xproto $REQUIRED_MODULES)
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 CFLAGS="$CFLAGS $XORG_CFLAGS "' -I$(top_srcdir)/src'
 AC_SUBST([CFLAGS])
