diff -Naur xf86-input-evdev-2.2.2-old/src/evdev.h xf86-input-evdev-2.2.2-new/src/evdev.h
--- xf86-input-evdev-2.2.2-old/src/evdev.h	2009-04-07 03:55:25.000000000 -0700
+++ xf86-input-evdev-2.2.2-new/src/evdev.h	2009-05-01 04:23:06.000000000 -0700
@@ -35,7 +35,7 @@
 
 #include <xf86Xinput.h>
 #include <xf86_OSproc.h>
-#include <xkbstr.h>
+#include <X11/extensions/XKBstr.h>
 
 #ifndef EV_CNT /* linux 2.4 kernels and earlier lack _CNT defines */
 #define EV_CNT (EV_MAX+1)
