diff -Naur MPlayer-1.0pre7try2-old/configure MPlayer-1.0pre7try2-new/configure
--- MPlayer-1.0pre7try2-old/configure	2005-08-26 21:33:25.000000000 -0700
+++ MPlayer-1.0pre7try2-new/configure	2005-08-26 21:33:51.000000000 -0700
@@ -76,7 +76,7 @@
 # x86/x86pc is used by QNX
 x86() {
   case "$host_arch" in
-    i[3-9]86|x86|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*|i586-i686) return 0 ;;
+    i[3-9]86|x86|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*|i586-i686|c3|c3-2) return 0 ;;
     *) return 1 ;;
   esac
 }
@@ -471,7 +471,7 @@
 
       # x86/x86pc is used by QNX
       case "`( uname -m ) 2>&1`" in
-      i[3-9]86*|x86|x86pc|k5|k6|k6_2|k6_3|k6-2|k6-3|pentium*|athlon*|i586_i686|i586-i686|BePC) host_arch=i386 ;;
+      i[3-9]86*|x86|x86pc|k5|k6|k6_2|k6_3|k6-2|k6-3|pentium*|athlon*|i586_i686|i586-i686|BePC|c3|c3-2) host_arch=i386 ;;
       ia64) host_arch=ia64 ;;
       x86_64|amd64)
         if [ "`$_cc -dumpmachine | grep x86_64 | cut -d- -f1`" = "x86_64" -a \
@@ -739,7 +739,7 @@
 }
 
 case "$host_arch" in
-  i[3-9]86|x86|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*|i586-i686)
+  i[3-9]86|x86|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*|i586-i686|c3|c3-2)
   _def_arch="#define ARCH_X86 1"
   _target_arch="TARGET_ARCH_X86 = yes"
 
