diff -Naur lighttpd-1.4.22-old/configure lighttpd-1.4.22-new/configure
--- lighttpd-1.4.22-old/configure	2009-02-17 00:58:51.000000000 -0800
+++ lighttpd-1.4.22-new/configure	2009-05-31 14:12:46.000000000 -0700
@@ -24573,7 +24573,7 @@
 { echo "$as_me:$LINENO: result: $WITH_PCRE" >&5
 echo "${ECHO_T}$WITH_PCRE" >&6; }
 
-if test "x$cross_compiling" = xno -a "$WITH_PCRE" != "no"; then
+if test "$WITH_PCRE" != "no"; then
   # Extract the first word of "pcre-config", so it can be a program name with args.
 set dummy pcre-config; ac_word=$2
 { echo "$as_me:$LINENO: checking for $ac_word" >&5
diff -Naur lighttpd-1.4.22-old/configure.in lighttpd-1.4.22-new/configure.in
--- lighttpd-1.4.22-old/configure.in	2009-02-17 00:55:16.000000000 -0800
+++ lighttpd-1.4.22-new/configure.in	2009-05-31 14:13:30.000000000 -0700
@@ -252,7 +252,7 @@
     [WITH_PCRE=$withval],[WITH_PCRE=yes])
 AC_MSG_RESULT([$WITH_PCRE])
 
-if test "x$cross_compiling" = xno -a "$WITH_PCRE" != "no"; then
+if test "$WITH_PCRE" != "no"; then
   AC_PATH_PROG(PCRECONFIG, pcre-config)
 
   if test x"$PCRECONFIG" != x; then
