diff -Naur libdvdnav-4.1.2-old/Makefile libdvdnav-4.1.2-new/Makefile
--- libdvdnav-4.1.2-old/Makefile	2008-04-10 08:03:03.000000000 -0700
+++ libdvdnav-4.1.2-new/Makefile	2008-04-19 11:40:11.000000000 -0700
@@ -176,6 +176,7 @@
 	@echo '#!/bin/sh' > $(.OBJDIR)/dvdnav-config
 	@echo 'prefix='$(PREFIX) >> $(.OBJDIR)/dvdnav-config
 	@echo 'libdir='$(shlibdir) >> $(.OBJDIR)/dvdnav-config
+	@echo 'includedir='$(incdir) >> $(.OBJDIR)/dvdnav-config
 	@echo 'version='$(SHLIB_VERSION) >> $(.OBJDIR)/dvdnav-config
 	@echo 'dvdread='$(DVDREAD) >> $(.OBJDIR)/dvdnav-config
 	@echo 'dvdreaddir='$(DVDREAD_DIR) >> $(.OBJDIR)/dvdnav-config
diff -Naur libdvdnav-4.1.2-old/misc/dvdnav-config2.sh libdvdnav-4.1.2-new/misc/dvdnav-config2.sh
--- libdvdnav-4.1.2-old/misc/dvdnav-config2.sh	2008-01-01 09:31:41.000000000 -0800
+++ libdvdnav-4.1.2-new/misc/dvdnav-config2.sh	2008-04-19 11:40:47.000000000 -0700
@@ -65,17 +65,17 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $dvdreadcflags $extracflags $threadcflags
+      echo $dvdreadcflags $extracflags $threadcflags
 fi
 
 if test "$echo_minicflags" = "yes"; then
-      echo -I$prefix/include -I$prefix/include/dvdnav $extracflags $threadcflags
+      echo -I$DESTDIR$includedir $extracflags $threadcflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir -ldvdnav $dvdreadlib $threadlib
+      echo -L$DESTDIR$libdir -ldvdnav $dvdreadlib $threadlib
 fi      
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L$libdir -ldvdnavmini $threadlib
+      echo -L$DESTDIR$libdir -ldvdnavmini $threadlib
 fi
