diff -Naur libdvdnav-4.1.3-old/misc/dvdnav-config.in libdvdnav-4.1.3-new/misc/dvdnav-config.in
--- libdvdnav-4.1.3-old/misc/dvdnav-config.in	2008-05-31 05:34:25.000000000 -0700
+++ libdvdnav-4.1.3-new/misc/dvdnav-config.in	2008-09-10 21:29:18.000000000 -0700
@@ -76,13 +76,13 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@ -I@includedir@/libdvdread @THREAD_CFLAGS@
+      echo -I$DESTDIR@includedir@ -I$DESTDIR@includedir@/libdvdread @THREAD_CFLAGS@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdnav -ldvdread @THREAD_LIBS@
+      echo -L$DESTDIR@libdir@ -ldvdnav -ldvdread @THREAD_LIBS@
 fi
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L@libdir@ -ldvdnavmini @THREAD_LIBS@
+      echo -L$DESTDIR@libdir@ -ldvdnavmini @THREAD_LIBS@
 fi      
