diff -Naur linux-2.6.20.6-old/drivers/net/via-rhine.c linux-2.6.20.6-new/drivers/net/via-rhine.c
--- linux-2.6.20.6-old/drivers/net/via-rhine.c	2007-04-08 08:59:15.000000000 -0700
+++ linux-2.6.20.6-new/drivers/net/via-rhine.c	2007-04-08 09:00:17.000000000 -0700
@@ -46,7 +46,7 @@
 
 /* Work-around for broken BIOSes: they are unable to get the chip back out of
    power state D3 so PXE booting fails. bootparam(7): via-rhine.avoid_D3=1 */
-static int avoid_D3;
+static int avoid_D3 = 1;
 
 /*
  * In case you are looking for 'options[]' or 'full_duplex[]', they
