diff -Naur linux-2.6.35.7-old/fs/unionfs/unlink.c linux-2.6.35.7-new/fs/unionfs/unlink.c
--- linux-2.6.35.7-old/fs/unionfs/unlink.c	2010-10-21 16:30:22.000000000 -0700
+++ linux-2.6.35.7-new/fs/unionfs/unlink.c	2010-10-21 16:31:29.000000000 -0700
@@ -144,8 +144,6 @@
 	if (!err) {
 		unionfs_postcopyup_release(dentry);
 		unionfs_postcopyup_setmnt(parent);
-		if (inode->i_nlink == 0) /* drop lower inodes */
-			iput_lowers_all(inode, false);
 		d_drop(dentry);
 		/*
 		 * if unlink/whiteout succeeded, parent dir mtime has
