diff -Naur xine-ui-0.99.5-old/src/xitk/Imlib-light/load.c xine-ui-0.99.5-new/src/xitk/Imlib-light/load.c
--- xine-ui-0.99.5-old/src/xitk/Imlib-light/load.c	2003-07-14 15:44:10.000000000 -0700
+++ xine-ui-0.99.5-new/src/xitk/Imlib-light/load.c	2010-01-08 21:14:55.000000000 -0800
@@ -193,7 +193,7 @@
     return 0;
   fread(buf, 1, 8, f);
   rewind(f);
-  return (int)png_check_sig(buf, 8);
+  return (int)(!png_sig_cmp(buf, (png_size_t)0, (png_size_t)8));
 }
 
 ImlibImage * Imlib_load_image(ImlibData * id, char *file) {
