diff -Naur xine-ui-0.99.9-old/misc/Makefile.am xine-ui-0.99.9-new/misc/Makefile.am
--- xine-ui-0.99.9-old/misc/Makefile.am	2015-06-28 15:44:35.057999742 -0700
+++ xine-ui-0.99.9-new/misc/Makefile.am	2015-06-28 15:45:33.919005566 -0700
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.quiet
 
-SUBDIRS = desktops visuals
+SUBDIRS = visuals
 
 # splash-foo/manifest contains names of extra source files, distributed only.
 
