diff -Naur xine-ui-3082-old/src/xitk/Makefile.am xine-ui-3082-new/src/xitk/Makefile.am
--- xine-ui-3082-old/src/xitk/Makefile.am	2011-02-03 06:13:11.000000000 -0800
+++ xine-ui-3082-new/src/xitk/Makefile.am	2011-02-03 06:13:15.000000000 -0800
@@ -70,7 +70,7 @@
 	     $(top_builddir)/src/xitk/oxine/liboxine.a \
 	     $(X_PRE_LIBS) $(LIRC_LIBS) $(XFT_LIBS) \
 	     $(X_LIBS) $(XTEST_LIBS) $(XSSAVER_LIBS) $(CURL_LIBS) $(NVTVSIMPLE_LIBS) \
-	     -lX11 -lXext -lm $(DBG_LIBS) $(XV_LIB) $(LIBPNG_LIBS) $(JPEG_LIBS) \
+	     -lX11 -lXext -lm -lrt $(DBG_LIBS) $(XV_LIB) $(LIBPNG_LIBS) $(JPEG_LIBS) \
 	     $(INTLLIBS) $(THREAD_LIBS)
 
 xine_remote_CFLAGS = $(AM_CFLAGS) -DNETWORK_CLIENT
