diff -Naur xine-ui-3119-old/misc/Makefile.am xine-ui-3119-new/misc/Makefile.am
--- xine-ui-3119-old/misc/Makefile.am	2012-04-26 16:34:14.000000000 -0700
+++ xine-ui-3119-new/misc/Makefile.am	2012-04-26 16:34:14.000000000 -0700
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.quiet
 
-SUBDIRS = desktops visuals
+SUBDIRS = visuals
 
 # splash-foo/manifest contains names of extra source files, distributed only.
 
