diff -Naur mythstream-0.18_1-old/parsers/apple.pl mythstream-0.18_1-new/parsers/apple.pl
--- mythstream-0.18_1-old/parsers/apple.pl	2007-09-25 14:21:02.000000000 -0700
+++ mythstream-0.18_1-new/parsers/apple.pl	2009-07-07 14:29:29.000000000 -0700
@@ -108,7 +108,7 @@
 {
    # en met bot geweld...
 	($url) = $response->content =~ /<key>previewURL<\/key><string>(.*?)<\/string>/;
-	($name) = $response->content =~ /<key>podcastName<\/key><string>(.*?)<\/string>/;
+	($name) = $response->content =~ /<key>playlistName<\/key><string>(.*?)<\/string>/;
 
    newItem(	name => $name,
 				url => $url,
