diff -Naur mythstream-0.18_1-old/mythstream/mythstream.cpp mythstream-0.18_1-new/mythstream/mythstream.cpp
--- mythstream-0.18_1-old/mythstream/mythstream.cpp	2009-08-14 18:05:47.000000000 -0700
+++ mythstream-0.18_1-new/mythstream/mythstream.cpp	2009-08-14 22:12:36.000000000 -0700
@@ -318,6 +318,8 @@
     
     clearVideo = false; // was video screen drawn?
 
+    clearOnly = false;  // should the next paint be clear only.
+
     setNoErase();
     updateBackground();
 
@@ -754,7 +756,8 @@
 
         if (clearVideo)
         {
-          updateBotView(true); // oops, gui is mess... (clear video window)
+          clearOnly = true; // oops, gui is mess... (clear video window)
+          update(m_BotRect);
           clearVideo = false;
         }
 
@@ -1100,7 +1103,8 @@
           else
           {
             streamBrowser->hideVideo();      // hide video window
-            updateBotView(true); // oops, gui is mess... (clear video window)
+            clearOnly = true; // oops, gui is mess... (clear video window)
+            update(m_BotRect);
           }
 
         }
@@ -1327,15 +1331,30 @@
 {
     QRect r = e->rect();
 
+    // the video panel is not clean so clean it
+    if (clearOnly)
+    {
+        updateBotView();
+        clearOnly = false;
+        // if we are playing a video, then there is nothing left to update,
+        // although I do not know why we would clearing the video panel were a video playing
+        if ( playerState.getDisplayStatus() == PlayerState::video )
+        {
+            return;
+        }
+    }
+
     if (r.intersects(m_TopRect))
+    {
         updateTopView();
+    }
     if (r.intersects(m_MidRect))
     {
         updateMidView();
     }
     if (r.intersects(m_BotRect))
     {
-        updateBotView(false);
+        updateBotView();
     }
 }
 
@@ -1545,7 +1564,6 @@
 
 void MythStream::updateTopView()
 {
-    
    // unconditional update
     QPixmap pix(m_TopRect.size());
     pix.fill(this, m_TopRect.topLeft());
@@ -1601,7 +1619,7 @@
 }
 
 
-void MythStream::updateBotView(bool clearOnly)
+void MythStream::updateBotView()
 {
     
     LayerSet* container = 0;
diff -Naur mythstream-0.18_1-old/mythstream/mythstream.h mythstream-0.18_1-new/mythstream/mythstream.h
--- mythstream-0.18_1-old/mythstream/mythstream.h	2007-09-25 15:54:46.000000000 -0700
+++ mythstream-0.18_1-new/mythstream/mythstream.h	2009-08-14 18:07:39.000000000 -0700
@@ -159,7 +159,7 @@
 
     void updateTopView();
     void updateMidView();
-    void updateBotView(bool clearOnly);
+    void updateBotView();
 
     void keyPressEvent( QKeyEvent *k );
 
@@ -192,6 +192,7 @@
     unsigned int m_FolderDisplaySize;
     bool clearVideo;
     bool harvesting;
+    bool clearOnly;
 
     QRect m_fftRect;
     FFTBox *fft;
