diff -Naur libcdio-0.79-old/lib/driver/utf8.c libcdio-0.79-new/lib/driver/utf8.c
--- libcdio-0.79-old/lib/driver/utf8.c	2006-12-13 16:22:30.000000000 -0800
+++ libcdio-0.79-new/lib/driver/utf8.c	2007-11-11 10:27:35.000000000 -0800
@@ -18,11 +18,11 @@
 */
 /* UTF-8 support */
 
-#ifdef HAVE_JOLIET
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
 
+#ifdef HAVE_JOLIET
 #ifdef HAVE_STRING_H
 # include <string.h>
 #endif
