diff -Naur mythplugins-trunk-19997-old/mythzoneminder/mythzoneminder/zmconsole.cpp mythplugins-trunk-19997-new/mythzoneminder/mythzoneminder/zmconsole.cpp
--- mythplugins-trunk-19997-old/mythzoneminder/mythzoneminder/zmconsole.cpp	2009-02-14 15:48:59.000000000 -0800
+++ mythplugins-trunk-19997-new/mythzoneminder/mythzoneminder/zmconsole.cpp	2009-02-14 15:59:34.000000000 -0800
@@ -299,7 +299,7 @@
         item->SetText(monitor->zmcStatus, "zmcstatus");
         item->SetText(monitor->zmaStatus, "zmastatus");
         item->SetText(QString("%1").arg(monitor->events), "eventcount");
-        item->SetData((void*) monitor);
+        item->SetData(qVariantFromValue((void*) monitor));
     }
 
     m_monitor_list->SetItemCurrent(pos);
