diff -Naur mythplugins-trunk-21133-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-21133-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-21133-old/mythvideo/mtd/dvdprobe.cpp	2009-08-05 15:41:53.000000000 -0700
+++ mythplugins-trunk-21133-new/mythvideo/mtd/dvdprobe.cpp	2009-08-05 15:44:03.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
