diff -Naur mythplugins-trunk-17324-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17324-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17324-old/mythgallery/mythgallery/mythgallery.pro	2008-05-14 06:50:29.000000000 -0700
+++ mythplugins-trunk-17324-new/mythgallery/mythgallery/mythgallery.pro	2008-05-14 06:50:30.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
