diff -Naur mythplugins-trunk-19970-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19970-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19970-old/mythmusic/mythmusic/aacdecoder.cpp	2009-02-11 05:55:58.000000000 -0800
+++ mythplugins-trunk-19970-new/mythmusic/mythmusic/aacdecoder.cpp	2009-02-11 05:58:24.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
