diff -Naur mythplugins-trunk-17081-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17081-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17081-old/mythgallery/mythgallery/mythgallery.pro	2008-04-17 18:30:02.000000000 -0700
+++ mythplugins-trunk-17081-new/mythgallery/mythgallery/mythgallery.pro	2008-04-17 18:30:03.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
