diff -Naur mythplugins-trunk-18515-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-18515-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-18515-old/mythvideo/mtd/dvdprobe.cpp	2008-09-29 16:35:31.000000000 -0700
+++ mythplugins-trunk-18515-new/mythvideo/mtd/dvdprobe.cpp	2008-09-29 16:35:31.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
