diff -Naur mythplugins-trunk-18336-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-18336-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-18336-old/mythmusic/mythmusic/aacdecoder.cpp	2008-09-19 03:03:39.000000000 -0700
+++ mythplugins-trunk-18336-new/mythmusic/mythmusic/aacdecoder.cpp	2008-09-19 03:06:35.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
