diff -Naur mythplugins-trunk-18885-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-18885-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-18885-old/mythmusic/mythmusic/aacdecoder.cpp	2008-10-26 17:03:55.000000000 -0700
+++ mythplugins-trunk-18885-new/mythmusic/mythmusic/aacdecoder.cpp	2008-10-26 17:06:59.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
