diff -Naur mythplugins-trunk-21118-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-21118-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-21118-old/mythvideo/mtd/dvdprobe.cpp	2009-08-03 18:02:46.000000000 -0700
+++ mythplugins-trunk-21118-new/mythvideo/mtd/dvdprobe.cpp	2009-08-03 18:04:31.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
