diff -Naur mythplugins-trunk-16716-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-16716-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-16716-old/mythgallery/mythgallery/mythgallery.pro	2008-03-20 02:01:37.000000000 -0700
+++ mythplugins-trunk-16716-new/mythgallery/mythgallery/mythgallery.pro	2008-03-20 02:04:10.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
