diff -Naur mythplugins-trunk-21184-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-21184-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-21184-old/mythvideo/mtd/dvdprobe.cpp	2009-08-09 10:59:31.000000000 -0700
+++ mythplugins-trunk-21184-new/mythvideo/mtd/dvdprobe.cpp	2009-08-09 11:11:48.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
