diff -Naur mythplugins-trunk-17350-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17350-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17350-old/mythgallery/mythgallery/mythgallery.pro	2008-05-18 11:35:23.000000000 -0700
+++ mythplugins-trunk-17350-new/mythgallery/mythgallery/mythgallery.pro	2008-05-18 11:35:23.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
