diff -Naur mythplugins-trunk-16873-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-16873-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-16873-old/mythgallery/mythgallery/mythgallery.pro	2008-04-01 08:56:18.000000000 -0700
+++ mythplugins-trunk-16873-new/mythgallery/mythgallery/mythgallery.pro	2008-04-01 08:56:19.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
