diff -Naur mythplugins-trunk-24011-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-24011-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-24011-old/mythvideo/mtd/dvdprobe.cpp	2010-04-06 07:03:47.000000000 -0700
+++ mythplugins-trunk-24011-new/mythvideo/mtd/dvdprobe.cpp	2010-04-06 07:03:48.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
