diff -Naur mythplugins-trunk-19090-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19090-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19090-old/mythmusic/mythmusic/aacdecoder.cpp	2008-11-13 18:19:05.000000000 -0800
+++ mythplugins-trunk-19090-new/mythmusic/mythmusic/aacdecoder.cpp	2008-11-13 18:21:52.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
