diff -Naur mythplugins-trunk-21679-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-21679-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-21679-old/mythvideo/mtd/dvdprobe.cpp	2009-09-05 10:22:36.000000000 -0700
+++ mythplugins-trunk-21679-new/mythvideo/mtd/dvdprobe.cpp	2009-09-05 10:24:15.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
