diff -Naur mythplugins-trunk-16840-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-16840-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-16840-old/mythgallery/mythgallery/mythgallery.pro	2008-03-29 22:05:53.000000000 -0700
+++ mythplugins-trunk-16840-new/mythgallery/mythgallery/mythgallery.pro	2008-03-29 22:05:53.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
