diff -Naur mythplugins-trunk-18866-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-18866-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-18866-old/mythmusic/mythmusic/aacdecoder.cpp	2008-10-24 16:58:22.000000000 -0700
+++ mythplugins-trunk-18866-new/mythmusic/mythmusic/aacdecoder.cpp	2008-10-24 17:01:23.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
