diff -Naur mythplugins-trunk-19622-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-19622-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-19622-old/mythvideo/mtd/dvdprobe.cpp	2009-01-08 18:32:34.000000000 -0800
+++ mythplugins-trunk-19622-new/mythvideo/mtd/dvdprobe.cpp	2009-01-08 18:32:35.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
