diff -Naur mythplugins-trunk-16956-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-16956-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-16956-old/mythvideo/mtd/dvdprobe.cpp	2008-04-04 19:36:36.000000000 -0700
+++ mythplugins-trunk-16956-new/mythvideo/mtd/dvdprobe.cpp	2008-04-04 19:39:03.000000000 -0700
@@ -18,6 +18,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 
