diff -Naur mythplugins-trunk-16733-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-16733-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-16733-old/mythgallery/mythgallery/mythgallery.pro	2008-03-20 17:22:01.000000000 -0700
+++ mythplugins-trunk-16733-new/mythgallery/mythgallery/mythgallery.pro	2008-03-20 17:24:26.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
