diff -Naur mythplugins-trunk-19487-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-19487-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-19487-old/mythvideo/mtd/dvdprobe.cpp	2008-12-28 23:02:25.000000000 -0800
+++ mythplugins-trunk-19487-new/mythvideo/mtd/dvdprobe.cpp	2008-12-28 23:02:26.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
