diff -Naur mythplugins-trunk-21808-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-21808-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-21808-old/mythvideo/mtd/dvdprobe.cpp	2009-09-12 06:36:03.000000000 -0700
+++ mythplugins-trunk-21808-new/mythvideo/mtd/dvdprobe.cpp	2009-09-12 06:37:44.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
