diff -Naur mythplugins-trunk-17449-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17449-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17449-old/mythgallery/mythgallery/mythgallery.pro	2008-06-06 12:33:28.000000000 -0700
+++ mythplugins-trunk-17449-new/mythgallery/mythgallery/mythgallery.pro	2008-06-06 12:33:29.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
