diff -Naur mythplugins-trunk-16978-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-16978-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-16978-old/mythgallery/mythgallery/mythgallery.pro	2008-04-06 16:07:09.000000000 -0700
+++ mythplugins-trunk-16978-new/mythgallery/mythgallery/mythgallery.pro	2008-04-06 16:07:10.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
