diff -Naur mythplugins-trunk-17283-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-17283-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-17283-old/mythvideo/mtd/dvdprobe.cpp	2008-05-07 22:35:10.000000000 -0700
+++ mythplugins-trunk-17283-new/mythvideo/mtd/dvdprobe.cpp	2008-05-07 22:37:42.000000000 -0700
@@ -18,6 +18,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 
