diff -Naur mythplugins-trunk-20963-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-20963-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-20963-old/mythvideo/mtd/dvdprobe.cpp	2009-07-19 17:13:41.000000000 -0700
+++ mythplugins-trunk-20963-new/mythvideo/mtd/dvdprobe.cpp	2009-07-19 17:15:22.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
