diff -Naur mythplugins-trunk-17598-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17598-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17598-old/mythgallery/mythgallery/mythgallery.pro	2008-06-23 16:21:25.000000000 -0700
+++ mythplugins-trunk-17598-new/mythgallery/mythgallery/mythgallery.pro	2008-06-23 16:21:27.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
