diff -Naur mythplugins-trunk-19951-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-19951-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-19951-old/mythvideo/mtd/dvdprobe.cpp	2009-02-07 09:12:20.000000000 -0800
+++ mythplugins-trunk-19951-new/mythvideo/mtd/dvdprobe.cpp	2009-02-07 09:12:20.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
