diff -Naur pcre-8.21-old/Makefile.in pcre-8.21-new/Makefile.in
--- pcre-8.21-old/Makefile.in	2012-01-09 21:46:59.000000000 -0800
+++ pcre-8.21-new/Makefile.in	2012-01-09 21:55:40.000000000 -0800
@@ -1600,7 +1600,7 @@
 	cp -p pcre.h $@
 
 @WITH_REBUILD_CHARTABLES_TRUE@pcre_chartables.c: dftables$(EXEEXT)
-@WITH_REBUILD_CHARTABLES_TRUE@	./dftables$(EXEEXT) $@
+@WITH_REBUILD_CHARTABLES_TRUE@	pcre-dftables$(EXEEXT) $@
 
 @WITH_REBUILD_CHARTABLES_FALSE@pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
 @WITH_REBUILD_CHARTABLES_FALSE@	rm -f $@
