diff -Naur pcre-7.4-old/Makefile.in pcre-7.4-new/Makefile.in
--- pcre-7.4-old/Makefile.in	2007-09-23 15:58:25.000000000 -0700
+++ pcre-7.4-new/Makefile.in	2007-09-23 15:59:14.000000000 -0700
@@ -1397,7 +1397,7 @@
 	cp -p pcre.h $@
 
 @WITH_REBUILD_CHARTABLES_TRUE@pcre_chartables.c: dftables$(EXEEXT)
-@WITH_REBUILD_CHARTABLES_TRUE@	./dftables$(EXEEXT) $@
+@WITH_REBUILD_CHARTABLES_TRUE@	pcre-dftables$(EXEEXT) $@
 
 @WITH_REBUILD_CHARTABLES_FALSE@pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
 @WITH_REBUILD_CHARTABLES_FALSE@	rm -f $@
