diff -Naur xorg-server-1.12.1-old/hw/xfree86/common/xf86pciBus.c xorg-server-1.12.1-new/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.12.1-old/hw/xfree86/common/xf86pciBus.c	2012-04-13 21:03:13.000000000 -0700
+++ xorg-server-1.12.1-new/hw/xfree86/common/xf86pciBus.c	2012-04-13 21:14:50.000000000 -0700
@@ -1107,7 +1107,7 @@
         driverList[0] = "ast";
         break;
     case 0x1002:
-        driverList[0] = "ati";
+        driverList[0] = "radeon";
         break;
     case 0x102c:
         driverList[0] = "chips";
