diff -Naur xorg-server-1.13.1-old/hw/xfree86/common/xf86pciBus.c xorg-server-1.13.1-new/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.13.1-old/hw/xfree86/common/xf86pciBus.c	2012-12-13 22:14:51.000000000 -0800
+++ xorg-server-1.13.1-new/hw/xfree86/common/xf86pciBus.c	2012-12-13 22:15:44.000000000 -0800
@@ -1132,7 +1132,7 @@
         driverList[0] = "ast";
         break;
     case 0x1002:
-        driverList[0] = "ati";
+        driverList[0] = "radeon";
         break;
     case 0x102c:
         driverList[0] = "chips";
