diff -Naur xorg-server-1.12.4-old/hw/xfree86/common/xf86pciBus.c xorg-server-1.12.4-new/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.12.4-old/hw/xfree86/common/xf86pciBus.c	2012-08-27 16:25:40.000000000 -0700
+++ xorg-server-1.12.4-new/hw/xfree86/common/xf86pciBus.c	2012-08-27 16:26:29.000000000 -0700
@@ -1107,7 +1107,7 @@
         driverList[0] = "ast";
         break;
     case 0x1002:
-        driverList[0] = "ati";
+        driverList[0] = "radeon";
         break;
     case 0x102c:
         driverList[0] = "chips";
