diff -Naur linux-2.6.24.2-old/sound/pci/hda/patch_atihdmi.c linux-2.6.24.2-new/sound/pci/hda/patch_atihdmi.c
--- linux-2.6.24.2-old/sound/pci/hda/patch_atihdmi.c	2008-02-11 10:28:03.000000000 -0800
+++ linux-2.6.24.2-new/sound/pci/hda/patch_atihdmi.c	2008-02-11 10:28:44.000000000 -0800
@@ -163,5 +163,6 @@
 	{ .id = 0x1002791a, .name = "ATI RS690/780 HDMI", .patch = patch_atihdmi },
 	{ .id = 0x1002aa01, .name = "ATI R6xx HDMI", .patch = patch_atihdmi },
 	{ .id = 0x10951392, .name = "SI HDMI", .patch = patch_atihdmi },
+	{ .id = 0x17e80047, .name = "Unknown HDMI",  .patch = patch_atihdmi },
 	{} /* terminator */
 };
