diff -Naur linux-2.6.24.6-old/include/linux/time.h linux-2.6.24.6-new/include/linux/time.h
--- linux-2.6.24.6-old/include/linux/time.h	2008-05-01 14:50:00.000000000 -0700
+++ linux-2.6.24.6-new/include/linux/time.h	2008-05-04 10:09:29.000000000 -0700
@@ -169,7 +169,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
