diff -Naur linux-2.6.24.3-old/include/linux/time.h linux-2.6.24.3-new/include/linux/time.h
--- linux-2.6.24.3-old/include/linux/time.h	2008-02-25 16:20:20.000000000 -0800
+++ linux-2.6.24.3-new/include/linux/time.h	2008-02-26 10:29:01.000000000 -0800
@@ -169,7 +169,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
