diff -Naur libXt-X11R7.0-1.0.0-old/configure.ac libXt-X11R7.0-1.0.0-new/configure.ac
--- libXt-X11R7.0-1.0.0-old/configure.ac	2005-12-14 16:24:34.000000000 -0800
+++ libXt-X11R7.0-1.0.0-new/configure.ac	2006-11-09 14:51:14.000000000 -0800
@@ -28,6 +28,9 @@
 dnl
 dnl
 AC_INIT(libXt, 1.0.0, [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg], libXt)
+AC_CANONICAL_BUILD
+AC_CANONICAL_HOST
+
 AM_INIT_AUTOMAKE([dist-bzip2])
 AM_MAINTAINER_MODE
 
@@ -37,6 +40,15 @@
 AC_PROG_CC
 AC_PROG_LIBTOOL
 
+AC_MSG_CHECKING([for CC_FOR_BUILD])
+if test x$host != x$build ; then
+    CC_FOR_BUILD=${CC_FOR_BUILD-gcc}
+else
+    CC_FOR_BUILD=${CC}
+fi
+AC_MSG_RESULT([$CC_FOR_BUILD])
+AC_SUBST(CC_FOR_BUILD)
+
 PKG_CHECK_MODULES(XT, sm x11 xproto kbproto)
 
 # Needed for including Xalloca.h
diff -Naur libXt-X11R7.0-1.0.0-old/util/Makefile.am libXt-X11R7.0-1.0.0-new/util/Makefile.am
--- libXt-X11R7.0-1.0.0-old/util/Makefile.am	2005-12-10 08:29:07.000000000 -0800
+++ libXt-X11R7.0-1.0.0-new/util/Makefile.am	2006-11-09 14:50:45.000000000 -0800
@@ -13,6 +13,12 @@
 makestrs_CFLAGS = $(XT_CFLAGS)
 makestrs_SOURCES = makestrs.c
 
+COMPILE = $(CC_FOR_BUILD) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
+	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+LINK = $(CC_FOR_BUILD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+
+$(srcdir)/makestrs-makestrs.o:
+	$(COMPILE) -c $< -o $@
 
 # Man page
 appmandir = $(APP_MAN_DIR)
