diff -Naur libexif-0.6.13-old/doc/Makefile.in libexif-0.6.13-new/doc/Makefile.in
--- libexif-0.6.13-old/doc/Makefile.in	2005-12-27 13:18:36.000000000 -0800
+++ libexif-0.6.13-new/doc/Makefile.in	2006-02-13 16:14:59.000000000 -0800
@@ -444,13 +444,17 @@
 
 all-local: $(DOXYGEN_STAMPS)
 
-install-data-local: install-apidocs install-apidocs-internals
+@HAVE_DOXYGEN_TRUE@install-data-local: install-apidocs install-apidocs-internals
+@HAVE_DOXYGEN_FALSE@install-data-local:
 
-uninstall-local: uninstall-apidocs uninstall-apidocs-internals
+@HAVE_DOXYGEN_TRUE@uninstall-local: uninstall-apidocs uninstall-apidocs-internals
+@HAVE_DOXYGEN_FALSE@uninstall-local:
 
-clean-local: clean-apidocs
+@HAVE_DOXYGEN_TRUE@clean-local: clean-apidocs
+@HAVE_DOXYGEN_FALSE@clean-local:
 
-upload: $(DOXYGEN_UPLOAD)
+@HAVE_DOXYGEN_TRUE@upload: $(DOXYGEN_UPLOAD)
+@HAVE_DOXYGEN_FALSE@upload:
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
