diff -Naur aewm-1.3.3-old/clients/Makefile aewm-1.3.3-new/clients/Makefile
--- aewm-1.3.3-old/clients/Makefile	2006-02-11 11:20:31.000000000 -0800
+++ aewm-1.3.3-new/clients/Makefile	2006-02-13 20:01:21.000000000 -0800
@@ -5,17 +5,19 @@
 CC       = gcc
 CFLAGS   = -g -O2 -Wall
 
-AELIBDIR = ../lib
-MYLIBDIR = lib
-XROOT    = /usr/X11R6
-BINDIR   = $(DESTDIR)$(XROOT)/bin
-MANDIR   = $(DESTDIR)$(XROOT)/man/man1
-CFGDIR   = $(DESTDIR)/etc/X11/aewm
+AELIBDIR   = ../lib
+MYLIBDIR   = lib
+XROOT      = /usr/X11R6
+BINDIR     = $(DESTDIR)$(XROOT)/bin
+INCLUDEDIR = $(DESTDIR)$(XROOT)/include
+LIBDIR     = $(DESTDIR)$(XROOT)/lib
+MANDIR     = $(DESTDIR)$(XROOT)/man/man1
+CFGDIR     = $(DESTDIR)/etc/X11/aewm
 
 INCLUDES = -I$(AELIBDIR) -I$(MYLIBDIR)
-X11PATH  = -I$(XROOT)/include
-X11LIBS  = -L$(XROOT)/lib -lX11
-XAWLIBS  = -L$(XROOT)/lib -lX11 -lXt -lXaw
+X11PATH  = -I$(INCLUDEDIR)
+X11LIBS  = -L$(LIBDIR) -lX11
+XAWLIBS  = -L$(LIBDIR) -lX11 -lXt -lXaw
 GTKPATH  = `pkg-config --cflags gtk+-2.0`
 GTKLIBS  = `pkg-config --libs gtk+-2.0`
 
@@ -74,7 +76,7 @@
 # --------------------------------------------------------------------
 
 install: all
-	install -s $(ALLPROGS) $(BINDIR)
+	install $(ALLPROGS) $(BINDIR)
 	install -m 644 $(MANPAGE) $(MANDIR)
 	gzip -9vf $(MANDIR)/$(MANPAGE)
 	mkdir -p $(CFGDIR) && cp clientsrc.sample $(CFGDIR)/clientsrc
diff -Naur aewm-1.3.3-old/src/Makefile aewm-1.3.3-new/src/Makefile
--- aewm-1.3.3-old/src/Makefile	2006-02-11 11:25:21.000000000 -0800
+++ aewm-1.3.3-new/src/Makefile	2006-02-13 20:01:21.000000000 -0800
@@ -12,13 +12,13 @@
 
 # Uncomment to add MWM hints support. This requires Lesstif headers at
 # build time, but does not link against Lesstif.
-#DEFINES += -DMWM_HINTS
+DEFINES += -DMWM_HINTS
 
 # Uncomment to add Xft support (requires X.Org or XFree86 >= 4.0.2).
 # This depends on -lXext above, even if you have disabled shape support.
-#DEFINES += -DXFT
-#EXTRA_INC += `pkg-config --cflags xft`
-#EXTRA_LIBS += `pkg-config --libs xft`
+DEFINES += -DXFT
+EXTRA_INC += `pkg-config --cflags xft`
+EXTRA_LIBS += `pkg-config --libs xft`
 
 # Uncomment for debugging info (abandon all hope, ye who enter here).
 #DEFINES += -DDEBUG
@@ -28,13 +28,15 @@
 CC       = gcc
 CFLAGS   = -g -O2 -Wall
 
-MYLIBDIR = ../lib
-BINDIR   = $(DESTDIR)$(XROOT)/bin
-MANDIR   = $(DESTDIR)$(XROOT)/man/man1
-CFGDIR   = $(DESTDIR)/etc/X11/aewm
-INCLUDES = -I$(MYLIBDIR) -I$(XROOT)/include $(EXTRA_INC)
-LDPATH   = -L$(XROOT)/lib
-LIBS     = -lX11 $(EXTRA_LIBS)
+MYLIBDIR   = ../lib
+BINDIR     = $(DESTDIR)$(XROOT)/bin
+INCLUDEDIR = $(DESTDIR)$(XROOT)/include
+LIBDIR     = $(DESTDIR)$(XROOT)/lib
+MANDIR     = $(DESTDIR)$(XROOT)/man/man1
+CFGDIR     = $(DESTDIR)/etc/X11/aewm
+INCLUDES   = -I$(MYLIBDIR) -I$(INCLUDEDIR) $(EXTRA_INC)
+LDPATH     = -L$(LIBDIR)
+LIBS       = -lX11 $(EXTRA_LIBS)
 
 PROG     = aewm
 MANPAGE  = aewm.1x
@@ -50,7 +52,7 @@
 	$(CC) $(CFLAGS) $(DEFINES) $(INCLUDES) -c $< -o $@
 
 install: all
-	install -s $(PROG) $(BINDIR)
+	install $(PROG) $(BINDIR)
 	install -m 644 $(MANPAGE) $(MANDIR)
 	gzip -9vf $(MANDIR)/$(MANPAGE)
 	mkdir -p $(CFGDIR) && cp aewmrc.sample $(CFGDIR)/aewmrc
