diff -Naur NVIDIA-Linux-x86-173.14.12-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-173.14.12-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-173.14.12-pkg1-old/usr/src/nv/conftest.sh	2008-07-17 18:42:51.000000000 -0700
+++ NVIDIA-Linux-x86-173.14.12-pkg1-new/usr/src/nv/conftest.sh	2008-08-04 15:09:57.000000000 -0700
@@ -987,7 +987,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
