diff -Naur NVIDIA-Linux-x86_64-180.51-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-180.51-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-180.51-pkg2-old/usr/src/nv/conftest.sh	2009-04-17 00:44:56.000000000 -0700
+++ NVIDIA-Linux-x86_64-180.51-pkg2-new/usr/src/nv/conftest.sh	2009-04-21 18:57:29.000000000 -0700
@@ -1190,7 +1190,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
