diff -Naur libdrm-2.4.34-old/intel/Makefile.am libdrm-2.4.34-new/intel/Makefile.am
--- libdrm-2.4.34-old/intel/Makefile.am	2012-03-09 17:20:54.000000000 -0800
+++ libdrm-2.4.34-new/intel/Makefile.am	2012-05-14 22:04:16.000000000 -0700
@@ -57,7 +57,7 @@
 			      intel_debug.h
 
 # This may be interesting even outside of "make check", due to the -dump option.
-noinst_PROGRAMS = test_decode
+#noinst_PROGRAMS = test_decode
 
 BATCHES = \
 	tests/gen4-3d.batch \
