diff -Naur mythtv-0.26-20141023-7e9a205-old/mythtv/libs/libmythtv/vaapicontext.h mythtv-0.26-20141023-7e9a205-new/mythtv/libs/libmythtv/vaapicontext.h
--- mythtv-0.26-20141023-7e9a205-old/mythtv/libs/libmythtv/vaapicontext.h	2015-06-14 12:46:30.000000000 -0700
+++ mythtv-0.26-20141023-7e9a205-new/mythtv/libs/libmythtv/vaapicontext.h	2015-08-02 14:16:41.457000902 -0700
@@ -4,6 +4,10 @@
 extern "C" {
 #include "libavcodec/vaapi.h"
 }
+#include "va/va_version.h"
+#if VA_CHECK_VERSION(0,34,0)
+#include "va/va_compat.h"
+#endif
 #include "va/va_x11.h"
 #include "va/va_glx.h"
 #include "videocolourspace.h"
