diff -Naur xorg-server-1.0.1-old/hw/xfree86/os-support/linux/Makefile.am xorg-server-1.0.1-new/hw/xfree86/os-support/linux/Makefile.am
--- xorg-server-1.0.1-old/hw/xfree86/os-support/linux/Makefile.am	2005-12-01 22:02:42.000000000 -0800
+++ xorg-server-1.0.1-new/hw/xfree86/os-support/linux/Makefile.am	2006-02-07 21:06:21.000000000 -0800
@@ -24,7 +24,7 @@
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS) @SERVER_DEFINES@ @LOADER_DEFINES@
 
-INCLUDES = $(XORG_INCS) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
diff -Naur xorg-server-1.0.1-old/hw/xfree86/os-support/linux/Makefile.in xorg-server-1.0.1-new/hw/xfree86/os-support/linux/Makefile.in
--- xorg-server-1.0.1-old/hw/xfree86/os-support/linux/Makefile.in	2005-12-20 23:36:29.000000000 -0800
+++ xorg-server-1.0.1-new/hw/xfree86/os-support/linux/Makefile.in	2006-02-07 21:07:05.000000000 -0800
@@ -444,7 +444,7 @@
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS) @SERVER_DEFINES@ @LOADER_DEFINES@
 
-INCLUDES = $(XORG_INCS) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
