diff -Naur xc-old/programs/Xserver/hw/xfree86/etc/gtf.c xc-new/programs/Xserver/hw/xfree86/etc/gtf.c
--- xc-old/programs/Xserver/hw/xfree86/etc/gtf.c	2004-04-23 12:53:57.000000000 -0700
+++ xc-new/programs/Xserver/hw/xfree86/etc/gtf.c	2005-07-18 19:46:54.000000000 -0700
@@ -117,7 +117,7 @@
 #endif
 
 #define MARGIN_PERCENT    1.8   /* % of active vertical image                */
-#define CELL_GRAN         8.0   /* assumed character cell granularity        */
+#define CELL_GRAN         1.0   /* assumed character cell granularity        */
 #define MIN_PORCH         1     /* minimum front porch                       */
 #define V_SYNC_RQD        3     /* width of vsync in lines                   */
 #define H_SYNC_PERCENT    8.0   /* width of hsync as % of total line         */
