diff -Naur xc-old/programs/Xserver/fb/Imakefile xc-new/programs/Xserver/fb/Imakefile
--- xc-old/programs/Xserver/fb/Imakefile	2005-07-12 03:02:10.000000000 -0700
+++ xc-new/programs/Xserver/fb/Imakefile	2005-10-28 11:05:26.000000000 -0700
@@ -3,19 +3,6 @@
 XCOMM 
 XCOMM Id: Imakefile,v 1.1 1999/11/02 03:54:44 keithp Exp $
 
-#if defined(HasGcc34) && HasGcc34
-MMXOPTIONS= -mmmx -msse -Winline --param inline-unit-growth=10000 \
-	--param large-function-growth=10000 -DUSE_MMX
-
-#if defined(i386Architecture) || defined(AMD64Architecture)
-SpecialCObjectRule(fbmmx,fbmmx.c,$(MMXOPTIONS))
-SpecialCObjectRule(fbpict,fbpict.c,$(MMXOPTIONS))
-SpecialCObjectRule(fbfill,fbfill.c,$(MMXOPTIONS))
-SpecialCObjectRule(fbcopy,fbcopy.c,$(MMXOPTIONS))
-#endif
-
-#endif
-
 #if DoLoadableServer
 #if !BuildModuleInSubdir
 #define IHaveModules
