diff -Naur xc-old/lib/GL/mesa/drivers/dri/unichrome/Imakefile.inc xc-new/lib/GL/mesa/drivers/dri/unichrome/Imakefile.inc
--- xc-old/lib/GL/mesa/drivers/dri/unichrome/Imakefile.inc	2005-08-12 09:31:00.000000000 -0700
+++ xc-new/lib/GL/mesa/drivers/dri/unichrome/Imakefile.inc	2005-10-26 21:15:37.000000000 -0700
@@ -104,7 +104,7 @@
 LinkSourceFile(via_common.h, $(MESADRVSRCDIR)/unichrome)
 LinkSourceFile(via_context.c, $(MESADRVSRCDIR)/unichrome)
 LinkSourceFile(via_context.h, $(MESADRVSRCDIR)/unichrome)
-LinkSourceFile(via_dri.h, $(MESADRVSRCDIR)/unichrome)
+LinkSourceFile(via_dri.h, $(MESADRVSRCDIR)/unichrome/server)
 LinkSourceFile(via_fb.c, $(MESADRVSRCDIR)/unichrome)
 LinkSourceFile(via_fb.h, $(MESADRVSRCDIR)/unichrome)
 LinkSourceFile(via_ioctl.c, $(MESADRVSRCDIR)/unichrome)
