diff -Naur NVIDIA-Linux-x86-180.18-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-180.18-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-180.18-pkg1-old/usr/src/nv/conftest.sh	2008-12-18 15:07:31.000000000 -0800
+++ NVIDIA-Linux-x86-180.18-pkg1-new/usr/src/nv/conftest.sh	2008-12-22 19:49:27.000000000 -0800
@@ -1143,7 +1143,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
