diff -Naur xtrans-1.3.5-old/Xtranssock.c xtrans-1.3.5-new/Xtranssock.c
--- xtrans-1.3.5-old/Xtranssock.c	2014-09-12 11:32:47.000000000 -0700
+++ xtrans-1.3.5-new/Xtranssock.c	2015-07-14 13:52:11.317004816 -0700
@@ -2697,7 +2697,7 @@
 Xtransport	TRANS(SocketUNIXFuncs) = {
 	/* Socket Interface */
 	"unix",
-#if !defined(LOCALCONN) && !defined(HAVE_ABSTRACT_SOCKETS)
+#if !defined(LOCALCONN)
         TRANS_ALIAS,
 #else
 	0,
