diff -Naur mythtv-0.20-16082-old/libs/libmyth/mythcdrom-linux.cpp mythtv-0.20-16082-new/libs/libmyth/mythcdrom-linux.cpp
--- mythtv-0.20-16082-old/libs/libmyth/mythcdrom-linux.cpp	2008-09-15 14:05:01.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmyth/mythcdrom-linux.cpp	2010-08-03 20:52:07.000000000 -0700
@@ -1,4 +1,5 @@
 #include "mythcdrom.h"
+#include <sys/stat.h>
 #include <sys/ioctl.h>                // ioctls
 #include <linux/cdrom.h>        // old ioctls for cdrom
 #include <errno.h>
