diff -Naur mythtv-0.20-16082-old/libs/libmythfreemheg/Text.cpp mythtv-0.20-16082-new/libs/libmythfreemheg/Text.cpp
--- mythtv-0.20-16082-old/libs/libmythfreemheg/Text.cpp	2008-09-15 14:05:06.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmythfreemheg/Text.cpp	2009-06-23 19:12:58.000000000 -0700
@@ -244,7 +244,7 @@
     else { // Textual form.
         const unsigned char *str = attrs.Bytes();
         ASSERT(str != NULL);
-        const char *p = (const char *)str;
+        char *p = (char *)str;
         char *q = strchr(p, '.'); // Find the terminating dot
         if (q != NULL) { // plain, italic etc.
             if (q-p == 6 && strncmp(p, "italic", 6) == 0) style = 1;
