diff -Naur mythtv-0.20-16082-old/configure mythtv-0.20-16082-new/configure
--- mythtv-0.20-16082-old/configure	2008-09-15 14:09:23.000000000 -0700
+++ mythtv-0.20-16082-new/configure	2008-09-15 14:09:23.000000000 -0700
@@ -2529,7 +2529,7 @@
 fi
 
 lmsensors="no"
-if has_header "sensors/sensors.h" && has_library libsensors ; then
+if has_header "sensors/sensors.h" && has_header "sensors/chips.h" && has_library libsensors ; then
     lmsensors="yes"
 fi
 
