diff -Naur lsdvd-0.15-old/configure.in lsdvd-0.15-new/configure.in
--- lsdvd-0.15-old/configure.in	2004-12-24 02:37:57.000000000 -0800
+++ lsdvd-0.15-new/configure.in	2005-12-29 19:57:41.000000000 -0800
@@ -1,7 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(lsdvd.c)
 AM_INIT_AUTOMAKE(lsdvd, 0.15)
-CFLAGS=-g
 AC_PROG_CC
 AC_CHECK_LIB(dvdread, DVDOpen, , AC_MSG_ERROR([libdvdread not found!]))
 AC_MSG_CHECKING([for dvdread/ifo_read.h])
