diff -Naur ntp-4.2.0-old/ntpdc/Makefile.in ntp-4.2.0-new/ntpdc/Makefile.in
--- ntp-4.2.0-old/ntpdc/Makefile.in	2003-10-15 02:33:08.000000000 -0700
+++ ntp-4.2.0-new/ntpdc/Makefile.in	2005-12-30 13:01:07.000000000 -0800
@@ -485,7 +485,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
