diff -Naur ntp-4.2.0-old/include/ntp_control.h ntp-4.2.0-new/include/ntp_control.h
--- ntp-4.2.0-old/include/ntp_control.h	2003-08-14 00:31:47.000000000 -0700
+++ ntp-4.2.0-new/include/ntp_control.h	2005-12-15 20:33:30.000000000 -0800
@@ -266,6 +266,7 @@
 	u_char tr_flags;		/* trap flags */
 	u_char tr_version;		/* version number of trapper */
 };
+extern struct ctl_trap ctl_trap[];
 
 /*
  * Flag bits
diff -Naur ntp-4.2.0-old/include/ntpd.h ntp-4.2.0-new/include/ntpd.h
--- ntp-4.2.0-old/include/ntpd.h	2003-09-12 20:08:04.000000000 -0700
+++ ntp-4.2.0-new/include/ntpd.h	2005-12-15 20:35:38.000000000 -0800
@@ -226,8 +226,6 @@
 #endif
 
 /* ntp_control.c */
-struct ctl_trap;
-extern struct ctl_trap ctl_trap[];
 extern int	num_ctl_traps;
 extern keyid_t	ctl_auth_keyid;		/* keyid used for authenticating write requests */
 
diff -Naur ntp-4.2.0-old/include/ntp_refclock.h ntp-4.2.0-new/include/ntp_refclock.h
--- ntp-4.2.0-old/include/ntp_refclock.h	2003-07-17 03:27:16.000000000 -0700
+++ ntp-4.2.0-new/include/ntp_refclock.h	2005-12-15 20:33:30.000000000 -0800
@@ -65,6 +65,7 @@
 	const char *clocktype;	/* long description */
 	const char *abbrev;	/* short description */
 };
+extern struct clktype clktypes[];
 
 /*
  * Configuration flag values
diff -Naur ntp-4.2.0-old/include/ntp_stdlib.h ntp-4.2.0-new/include/ntp_stdlib.h
--- ntp-4.2.0-old/include/ntp_stdlib.h	2003-07-17 03:27:16.000000000 -0700
+++ ntp-4.2.0-new/include/ntp_stdlib.h	2005-12-15 20:33:30.000000000 -0800
@@ -133,10 +133,6 @@
 extern u_char *	cache_key;		/* key pointer */
 extern u_int	cache_keylen;		/* key length */
 
-/* clocktypes.c */
-struct clktype;
-extern struct clktype clktypes[];
-
 /* getopt.c */
 extern char *	ntp_optarg;		/* global argument pointer */
 extern int	ntp_optind;		/* global argv index */
