diff -Naur ntp-4.2.0-old/ntpd/ntp_config.c ntp-4.2.0-new/ntpd/ntp_config.c
--- ntp-4.2.0-old/ntpd/ntp_config.c	2003-10-07 01:35:14.000000000 -0700
+++ ntp-4.2.0-new/ntpd/ntp_config.c	2006-01-05 17:48:36.000000000 -0800
@@ -15,8 +15,6 @@
 #include "ntp_refclock.h"
 #include "ntp_filegen.h"
 #include "ntp_stdlib.h"
-#include "ntp_config.h"
-#include "ntp_cmdargs.h"
 
 #include <stdio.h>
 #include <ctype.h>
@@ -33,6 +31,13 @@
 # endif
 #endif /* VMS */
 
+/*
+ * Include these after the system headers since some Linux
+ * systems have '#undef CONFIG_PHONE' in them.
+ */
+#include "ntp_config.h"
+#include "ntp_cmdargs.h"
+
 #ifdef SYS_WINNT
 # include <io.h>
 extern HANDLE ResolverThreadHandle;
