diff -Naur ntp-4.2.6p1-old/ntpdc/Makefile.am ntp-4.2.6p1-new/ntpdc/Makefile.am
--- ntp-4.2.6p1-old/ntpdc/Makefile.am	2010-04-12 13:28:09.000000000 -0700
+++ ntp-4.2.6p1-new/ntpdc/Makefile.am	2010-04-12 13:28:51.000000000 -0700
@@ -54,7 +54,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout $(srcdir)/layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
diff -Naur ntp-4.2.6p1-old/ntpdc/Makefile.in ntp-4.2.6p1-new/ntpdc/Makefile.in
--- ntp-4.2.6p1-old/ntpdc/Makefile.in	2010-04-12 13:28:09.000000000 -0700
+++ ntp-4.2.6p1-new/ntpdc/Makefile.in	2010-04-12 13:28:51.000000000 -0700
@@ -741,7 +741,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout $(srcdir)/layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
