diff -Naur mythtv-0.19-old/libs/libmyth/mythmedia.cpp mythtv-0.19-new/libs/libmyth/mythmedia.cpp
--- mythtv-0.19-old/libs/libmyth/mythmedia.cpp	2006-01-15 22:45:15.000000000 -0800
+++ mythtv-0.19-new/libs/libmyth/mythmedia.cpp	2007-03-29 10:41:54.000000000 -0700
@@ -17,8 +17,8 @@
 
 // end for testing
 
-static const QString PATHTO_MOUNT("/bin/mount");
-static const QString PATHTO_UNMOUNT("/bin/umount");
+static const QString PATHTO_MOUNT("/usr/bin/pmount");
+static const QString PATHTO_UNMOUNT("/usr/bin/pumount");
 static const QString PATHTO_MOUNTS("/proc/mounts");
 
 const char* MythMediaDevice::MediaStatusStrings[] =
