diff -Naur linux-2.6.37.3-old/fs/unionfs/unlink.c linux-2.6.37.3-new/fs/unionfs/unlink.c
--- linux-2.6.37.3-old/fs/unionfs/unlink.c	2011-03-08 12:08:07.000000000 -0800
+++ linux-2.6.37.3-new/fs/unionfs/unlink.c	2011-03-08 12:08:34.000000000 -0800
@@ -144,8 +144,6 @@
 	if (!err) {
 		unionfs_postcopyup_release(dentry);
 		unionfs_postcopyup_setmnt(parent);
-		if (inode->i_nlink == 0) /* drop lower inodes */
-			iput_lowers_all(inode, false);
 		d_drop(dentry);
 		/*
 		 * if unlink/whiteout succeeded, parent dir mtime has
