diff -Naur mythtv-0.23-20110720-226ba71-de0d6b3-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.23-20110720-226ba71-de0d6b3-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.23-20110720-226ba71-de0d6b3-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-07-23 04:19:30.000000000 -0700
+++ mythtv-0.23-20110720-226ba71-de0d6b3-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-07-23 04:19:31.000000000 -0700
@@ -4717,6 +4717,14 @@
 
     GetNVP()->ReinitAudio();
 
+    if (using_passthru != DoPassThrough(codec_ctx))
+    {
+        // detected audio capabilities changed half-way
+        // recheck audio stream
+        VERBOSE(VB_AUDIO, LOC + "Audio config changed. Rescanning");
+        return SetupAudioStream();
+    }
+
     if (LCD *lcd = LCD::Get())
     {
         LCDAudioFormatSet audio_format;
