diff -Naur mythtv-0.23-27077-old/bindings/python/python.pro mythtv-0.23-27077-new/bindings/python/python.pro
--- mythtv-0.23-27077-old/bindings/python/python.pro	2010-11-03 11:15:42.000000000 -0700
+++ mythtv-0.23-27077-new/bindings/python/python.pro	2010-11-03 11:15:43.000000000 -0700
@@ -2,11 +2,11 @@
 
 python_build.target = python_build
 python_build.depends = setup.py FORCE
-python_build.commands = $$PYTHON setup.py build
+python_build.commands = $$PYTHON setup.py build --cross-compile
 
 python_install.target = python_install
 python_install.CONFIG = no_path
-python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\"
+python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\" --prefix=\"$${PREFIX}\"
 
 # This is done so the default /usr/local prefix installs to the "normal"
 # /usr/lib/python{VER}/site-packages directory.
diff -Naur mythtv-0.23-27077-old/bindings/python/setup.py mythtv-0.23-27077-new/bindings/python/setup.py
--- mythtv-0.23-27077-old/bindings/python/setup.py	2010-11-03 11:15:42.000000000 -0700
+++ mythtv-0.23-27077-new/bindings/python/setup.py	2010-11-03 11:15:43.000000000 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
+from setuptools import setup
 
 setup(
         name='MythTV',
