diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp	2012-04-09 20:50:33.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp	2012-04-09 20:50:35.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
