diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythplugins/mythmusic/mythmusic/importmusic.cpp mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythplugins/mythmusic/mythmusic/importmusic.cpp
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythplugins/mythmusic/mythmusic/importmusic.cpp	2012-04-09 20:47:47.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythplugins/mythmusic/mythmusic/importmusic.cpp	2012-04-12 04:26:09.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // qt
 #include <QDir>
 #include <QFontMetrics>
diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythdb/compat.h mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythdb/compat.h
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythdb/compat.h	2012-04-09 20:47:48.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythdb/compat.h	2012-04-12 03:28:47.000000000 -0700
@@ -53,6 +53,8 @@
 #include <sys/resource.h> // for setpriority
 #include <sys/socket.h>
 #include <sys/wait.h>     // For WIFEXITED on Mac OS X
+#include <stdio.h>        // for snprintf(), used by inline dlerror()
+#include <unistd.h>       // for usleep()
 #endif
 
 #ifdef USING_MINGW
diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythfreemheg/BaseClasses.h mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythfreemheg/BaseClasses.h
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythfreemheg/BaseClasses.h	2012-04-09 20:47:48.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythfreemheg/BaseClasses.h	2012-04-12 04:00:32.000000000 -0700
@@ -89,7 +89,7 @@
             return MHSequence<BASE>::m_Values[--MHSequence<BASE>::m_VecSize];
         }
         // Push an element on the stack.
-        void Push(BASE b) { Append(b); }
+        void Push(BASE b) { this->Append(b); }
         // Return the top of the stack.
         BASE Top() { 
             MHASSERT(MHSequence<BASE>::m_VecSize > 0);
diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythtv/audioinputoss.cpp mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythtv/audioinputoss.cpp
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-09 20:47:48.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-12 04:09:54.000000000 -0700
@@ -20,6 +20,8 @@
  * 02110-1301, USA.
  */
 
+#include <unistd.h>
+
 #include "mythconfig.h"
 #if HAVE_SYS_SOUNDCARD_H
     #include <sys/soundcard.h>
diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythtv/firewiredevice.cpp mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythtv/firewiredevice.cpp
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/libs/libmythtv/firewiredevice.cpp	2012-04-09 20:47:48.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/libs/libmythtv/firewiredevice.cpp	2012-04-12 04:13:10.000000000 -0700
@@ -4,6 +4,8 @@
  *  Distributed as part of MythTV under GPL v2 and later.
  */
 
+#include <unistd.h>
+
 // C++ headers
 #include <algorithm>
 
diff -Naur mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/programs/mythfrontend/backendconnectionmanager.cpp mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/programs/mythfrontend/backendconnectionmanager.cpp
--- mythtv-0.23-20120402-4f31db5-de0d6b3-old/mythtv/programs/mythfrontend/backendconnectionmanager.cpp	2012-04-09 20:47:48.000000000 -0700
+++ mythtv-0.23-20120402-4f31db5-de0d6b3-new/mythtv/programs/mythfrontend/backendconnectionmanager.cpp	2012-04-12 04:22:59.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QThreadPool>
 #include <QRunnable>
