diff -Naur mythtv-0.23-20110720-226ba71-de0d6b3-old/mythtv/libs/libmythtv/util-xvmc.h mythtv-0.23-20110720-226ba71-de0d6b3-new/mythtv/libs/libmythtv/util-xvmc.h
--- mythtv-0.23-20110720-226ba71-de0d6b3-old/mythtv/libs/libmythtv/util-xvmc.h	2011-07-23 04:19:32.000000000 -0700
+++ mythtv-0.23-20110720-226ba71-de0d6b3-new/mythtv/libs/libmythtv/util-xvmc.h	2011-07-23 04:19:33.000000000 -0700
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(14),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
