diff -Naur mythtv-0.23-20110222-4fe0b09-de0d6b3-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.23-20110222-4fe0b09-de0d6b3-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.23-20110222-4fe0b09-de0d6b3-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-03-03 06:52:08.000000000 -0800
+++ mythtv-0.23-20110222-4fe0b09-de0d6b3-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-03-03 06:54:48.000000000 -0800
@@ -1974,6 +1974,9 @@
                 if (!codec_is_std(video_codec_id))
                     thread_count = 1;
 
+                if (enc->codec_id == CODEC_ID_SVQ3)
+                    thread_count = 1;
+
                 VERBOSE(VB_PLAYBACK, QString("Using %1 CPUs for decoding")
                         .arg(HAVE_THREADS ? thread_count : 1));
 
