diff -Naur mythtv-0.23-24220-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.23-24220-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.23-24220-old/libs/libmyth/mediamonitor-unix.cpp	2010-04-23 11:45:03.000000000 -0700
+++ mythtv-0.23-24220-new/libs/libmyth/mediamonitor-unix.cpp	2010-04-23 11:45:04.000000000 -0700
@@ -222,11 +222,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
