diff -Naur mythtv-0.23-20110413-6802d58-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp mythtv-0.23-20110413-6802d58-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp
--- mythtv-0.23-20110413-6802d58-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-05-11 16:12:15.000000000 -0700
+++ mythtv-0.23-20110413-6802d58-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-05-11 16:12:17.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
