diff -Naur mythtv-0.23-25328-old/bindings/python/python.pro mythtv-0.23-25328-new/bindings/python/python.pro
--- mythtv-0.23-25328-old/bindings/python/python.pro	2010-07-11 14:47:01.000000000 -0700
+++ mythtv-0.23-25328-new/bindings/python/python.pro	2010-07-11 14:48:53.000000000 -0700
@@ -6,7 +6,7 @@
 
 python_install.target = python_install
 python_install.CONFIG = no_path
-python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\"
+python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\" --prefix=\"$${PREFIX}\"
 
 # This is done so the default /usr/local prefix installs to the "normal"
 # /usr/lib/python{VER}/site-packages directory.
