diff -Naur mythtv-0.23-20110413-6802d58-de0d6b3-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-0.23-20110413-6802d58-de0d6b3-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.23-20110413-6802d58-de0d6b3-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2011-05-11 16:12:09.000000000 -0700
+++ mythtv-0.23-20110413-6802d58-de0d6b3-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2011-05-11 16:12:10.000000000 -0700
@@ -222,11 +222,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
