diff -Naur pixman-0.20.0-old/configure.ac pixman-0.20.0-new/configure.ac
--- pixman-0.20.0-old/configure.ac	2010-10-27 13:51:30.000000000 -0700
+++ pixman-0.20.0-new/configure.ac	2010-10-27 21:33:13.000000000 -0700
@@ -150,7 +150,7 @@
 # if we're using Sun Studio and neither the user nor a config.site
 # has set CFLAGS.
 if test $SUNCC = yes &&			\
-   test "$test_CFLAGS" == "" &&		\
+   test "$test_CFLAGS" = "" &&		\
    test "$CFLAGS" = "-g"
 then
   CFLAGS="-O -g"
