diff -Naur pixman-0.19.2-old/configure.ac pixman-0.19.2-new/configure.ac
--- pixman-0.19.2-old/configure.ac	2010-08-21 03:32:58.000000000 -0700
+++ pixman-0.19.2-new/configure.ac	2010-08-27 09:58:41.000000000 -0700
@@ -150,7 +150,7 @@
 # if we're using Sun Studio and neither the user nor a config.site
 # has set CFLAGS.
 if test $SUNCC = yes &&			\
-   test "$test_CFLAGS" == "" &&		\
+   test "$test_CFLAGS" = "" &&		\
    test "$CFLAGS" = "-g"
 then
   CFLAGS="-O -g"
