diff -Naur pixman-0.21.8-old/configure.ac pixman-0.21.8-new/configure.ac
--- pixman-0.21.8-old/configure.ac	2011-04-18 20:59:52.000000000 -0700
+++ pixman-0.21.8-new/configure.ac	2011-04-19 20:01:15.000000000 -0700
@@ -150,7 +150,7 @@
 # if we're using Sun Studio and neither the user nor a config.site
 # has set CFLAGS.
 if test $SUNCC = yes &&			\
-   test "$test_CFLAGS" == "" &&		\
+   test "$test_CFLAGS" = "" &&		\
    test "$CFLAGS" = "-g"
 then
   CFLAGS="-O -g"
