diff -Naur mythstream-0.18-old/libs/streamstorage.cpp mythstream-0.18-new/libs/streamstorage.cpp
--- mythstream-0.18-old/libs/streamstorage.cpp	2007-09-13 15:49:22.000000000 -0700
+++ mythstream-0.18-new/libs/streamstorage.cpp	2007-09-23 16:51:32.000000000 -0700
@@ -144,7 +144,7 @@
           }
         }
         else
-          if ( fi->fileName() != "list.xml" ) // shouldn't be there
+          if ( fi->fileName() != "list.xml" && fi->fileName().right(3) == ".pl" ) // PERL only, leave binaries 
             copyFile(sourceDir + "/" + fi->fileName(), destDir + "/" + fi->fileName() );
         ++it;
     }
