diff -Naur mythstream-0.18_1-old/libs/harvester.cpp mythstream-0.18_1-new/libs/harvester.cpp
--- mythstream-0.18_1-old/libs/harvester.cpp	2007-09-25 15:58:20.000000000 -0700
+++ mythstream-0.18_1-new/libs/harvester.cpp	2007-11-16 17:23:47.000000000 -0800
@@ -1673,7 +1673,7 @@
     
     eParser->addArgument( "sh" );
     eParser->addArgument( "-c" );
-    eParser->addArgument( parser + " list.xml \"" + sourceUrl + "\"" + srcName + "\"" );
+    eParser->addArgument( parser + " list.xml \"" + sourceUrl + "\" \"" + srcName + "\"" );
     //cout << "SHELL PARSER" << endl;
     
 #endif
