diff -Naur splashutils-1.5.3.3-old/src/Makefile.am splashutils-1.5.3.3-new/src/Makefile.am
--- splashutils-1.5.3.3-old/src/Makefile.am	2007-12-31 00:43:42.000000000 -0800
+++ splashutils-1.5.3.3-new/src/Makefile.am	2008-01-07 07:03:57.000000000 -0800
@@ -120,7 +120,7 @@
 	fbsplash.h
 libfbsplashrender_la_CFLAGS   = $(AM_CFLAGS)
 libfbsplashrender_la_LDFLAGS  = $(AM_LDFLAGS) -version-info $(libfbsplashrender_version)
-libfbsplashrender_la_LIBADD   = -lfbsplash
+libfbsplashrender_la_LIBADD   = libfbsplash.la
 
 libfbsplashrender_la_CFLAGS  += $(JPEG_CFLAGS)
 libfbsplashrender_la_LIBADD  += $(JPEG_LIBS)
