commit e1e6851c761e368fee6be02c6bf3f9a507aa6198
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Sep 10 13:40:22 2009 +0000

    Add VASliceParameterBufferMPEG2.slice_horizontal_position for i965 driver.

diff --git a/src/va.h b/src/va.h
index 9080817..6cb7063 100755
--- a/src/va.h
+++ b/src/va.h
@@ -585,6 +585,7 @@ typedef struct _VASliceParameterBufferMPEG2
     unsigned int slice_data_offset;/* the offset to the first byte of slice data */
     unsigned int slice_data_flag; /* see VA_SLICE_DATA_FLAG_XXX defintions */
     unsigned int macroblock_offset;/* the offset to the first bit of MB from the first byte of slice data */
+    unsigned int slice_horizontal_position;
     unsigned int slice_vertical_position;
     int quantiser_scale_code;
     int intra_slice_flag;
