commit b6e043b47d0bf52000809e335c7493a9d6463ab6
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Fri Feb 12 11:56:02 2010 +0000

    Add missing libs (pthreads).

diff --git a/test/putsurface/Makefile.am b/test/putsurface/Makefile.am
index c3aa2d7..b0e20ba 100644
--- a/test/putsurface/Makefile.am
+++ b/test/putsurface/Makefile.am
@@ -24,7 +24,7 @@ bin_PROGRAMS = putsurface
 
 AM_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/src/x11 -DIN_LIBVA
 
-TEST_LIBS = $(top_srcdir)/src/$(libvabackendlib)
+TEST_LIBS = $(top_srcdir)/src/$(libvabackendlib) -lpthread
 
 putsurface_LDADD = $(TEST_LIBS)
 putsurface_SOURCES = putsurface.c
