commit 78d6366a0cd4859f660ab41d6ac2da928a389c01
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Mon Jul 6 10:41:05 2009 +0000

    Fix build after X11 -> x11 move.

diff --git a/src/va_backend.h b/src/va_backend.h
index a167395..9d787a9 100644
--- a/src/va_backend.h
+++ b/src/va_backend.h
@@ -31,7 +31,7 @@
 
 #ifdef IN_LIBVA
 #include "va.h"
-#include "X11/va_x11.h"
+#include "x11/va_x11.h"
 #else
 #include <va/va.h>
 #include <va/va_x11.h>
