commit 2ee4b9353562dda501a130e1426b9f8a6dbaeb61
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Tue Sep 22 12:29:56 2009 +0000

    Fix vainfo dependencies (Konstantin Pavlov).

diff --git a/test/Makefile.am b/test/Makefile.am
index 5734226..2c61ba8 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -25,9 +25,8 @@ bin_PROGRAMS = vainfo
 
 AM_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/test/basic -I$(top_srcdir)/src/x11 -DIN_LIBVA
 
-
-vainfo_LDADD = $(top_srcdir)/src/$(libvabackendlib)
-vainfo_DEPENDENCIES = $(top_srcdir)/src/$(libvabackendlib)
+vainfo_LDADD = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
+vainfo_DEPENDENCIES = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
 vainfo_SOURCES = vainfo.c
 
 SUBDIRS = basic decode encode putsurface
