commit 41cc6f33ec8e7a6d10ab0208685989b6b7d27431
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Jun 24 07:52:36 2009 +0000

    Drop explicit CFLAGS override.

diff --git a/src/x11/Makefile.am b/dri/Makefile.am
index 8ce9bd3..c0f2d01 100644
--- a/src/x11/Makefile.am
+++ b/src/x11/Makefile.am
@@ -26,6 +26,5 @@ libva_x11includedir =  ${includedir}/src
 libva_x11include_HEADERS = va_x11.h va_dri.h va_dri2.h va_dricommon.h
 
 libva_x11_la_SOURCES = va_x11.c va_dri.c va_dri2.c va_dricommon.c dri2_util.c dri1_util.c
-CFLAGS = -ansi -O2 
 
 EXTRA_DIST = va_dristr.h va_dri2str.h va_dri2tokens.h
diff --git a/src/Makefile.am b/src/Makefile.am
index d098ed2..f7ed55e 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -28,7 +28,6 @@ libva_la_LTLIBRARIES = libva.la
 libva_ladir = $(libdir)
 libva_la_LDFLAGS = -version-number 0:30:4 -no-undefined
 libva_la_LIBADD = $(LIBVA_LIBS) -ldl -lX11 -lXext x11/libva_x11.la -ldrm -lXfixes
-CFLAGS = -ansi -O2 
 
 nodist_libva_la_SOURCES = va_version.h
 BUILT_SOURCES = va_version.h
