commit 1b986000375deba926fa8526de13fc86d54e856b
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jun 25 14:22:07 2009 +0000

    Fix dummy_drv_video.so init function name.

diff --git a/dummy_drv_video/dummy_drv_video.c b/dummy_drv_video/dummy_drv_video.c
index f3b8a7b..1eb51a0 100644
--- a/dummy_drv_video/dummy_drv_video.c
+++ b/dummy_drv_video/dummy_drv_video.c
@@ -1211,7 +1211,7 @@ VAStatus dummy_Terminate( VADriverContextP ctx )
     return VA_STATUS_SUCCESS;
 }
 
-VAStatus __vaDriverInit_0_29(  VADriverContextP ctx )
+VAStatus __vaDriverInit_0_30(  VADriverContextP ctx )
 {
     object_base_p obj;
     int result;
