commit 5ad7cc0b0397486ecf3974ee0b9f5963efa413ff
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jul 2 09:21:36 2009 +0000

    Add check for libXfixes.

diff --git a/configure.ac b/configure.ac
index 63d08aa..4dd8d11 100644
--- a/configure.ac
+++ b/configure.ac
@@ -94,6 +94,7 @@ AC_SYS_LARGEFILE
 
 PKG_CHECK_MODULES([X11], [x11])
 PKG_CHECK_MODULES([XEXT],[xext])
+PKG_CHECK_MODULES([XFIXES], [xfixes])
 PKG_CHECK_MODULES([DRM], [libdrm])
 
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
