commit d80b668d9261b8df8dd8859320689dcefcc87324
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jul 2 12:35:57 2009 +0000

    Fix make install (<va/va_x11.h>).

diff --git a/src/x11/Makefile.am b/dri/Makefile.am
index bb11d8d..58bf13c 100644
--- a/src/x11/Makefile.am
+++ b/src/x11/Makefile.am
@@ -22,7 +22,7 @@ AM_CFLAGS = -DLINUX -DIN_LIBVA -I$(top_srcdir)/src $(DRM_CFLAGS)
 
 noinst_LTLIBRARIES = libva_x11.la	
 
-libva_x11includedir =  ${includedir}/va
+libva_x11includedir = ${includedir}/va
 libva_x11include_HEADERS = va_x11.h va_dri.h va_dri2.h va_dricommon.h
 
 libva_x11_la_SOURCES = va_x11.c va_dri.c va_dri2.c va_dricommon.c dri2_util.c dri1_util.c va_nvctrl.c
