commit 1dda31442525eec4881903f916e4dc33575f86f7
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jul 2 13:33:42 2009 +0000

    Add VA_INVALID_ID helper macro.

diff --git a/src/va.h b/src/va.h
index b2dbccd..0519f6f 100755
--- a/src/va.h
+++ b/src/va.h
@@ -370,7 +370,8 @@ typedef VAGenericID VAContextID;
 
 typedef VAGenericID VASurfaceID;
 
-#define VA_INVALID_SURFACE	-1
+#define VA_INVALID_ID		0xffffffff
+#define VA_INVALID_SURFACE	VA_INVALID_ID
 
 /* 
  * vaCreateSurfaces - Create an array of surfaces used for decode and display  
