commit 7e38d7fed5135380ceadcc73f2a127fc6bba2a6c
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Fri Mar 20 13:12:45 2009 +0000

    Add new bit to signal VC-1 skipped P-frames.

diff --git a/src/va.h b/src/va.h
index 886e21b..4a4aa25 100755
--- a/src/va.h
+++ b/src/va.h
@@ -744,7 +744,7 @@ typedef struct _VAPictureParameterBufferVC1
     unsigned char luma_shift;		/* PICTURE_LAYER::LUMSHIFT */
     union {
         struct {
-            unsigned int picture_type		: 2; /* PICTURE_LAYER::PTYPE */
+            unsigned int picture_type		: 3; /* PICTURE_LAYER::PTYPE */
             unsigned int frame_coding_mode	: 3; /* PICTURE_LAYER::FCM */
             unsigned int top_field_first	: 1; /* PICTURE_LAYER::TFF */
             unsigned int is_first_field		: 1; /* set to 1 if it is the first field */
