diff -Naur lirc-0.7.0-old/configure lirc-0.7.0-new/configure
--- lirc-0.7.0-old/configure	2004-11-07 05:38:20.000000000 -0800
+++ lirc-0.7.0-new/configure	2004-12-30 09:43:56.072928878 -0800
@@ -8369,8 +8369,8 @@
   case ${any_driver} in
 	atilibusb)
 		hw_module="${hw_module} hw_atilibusb.o receive.o"
-		atilibusb_lib=`libusb-config --libs`
-		CFLAGS="$CFLAGS `libusb-config --cflags`"
+		atilibusb_lib=`${DESTDIR}${bindir}/libusb-config --libs`
+		CFLAGS="$CFLAGS `${DESTDIR}${bindir}/libusb-config --cflags`"
 		;;
 	audio)
 		hw_module="${hw_module} hw_audio.o receive.o"
