diff -Naur lirc-20061209Z-old/daemons/receive.c lirc-20061209Z-new/daemons/receive.c
--- lirc-20061209Z-old/daemons/receive.c	2006-07-16 01:31:14.000000000 -0700
+++ lirc-20061209Z-new/daemons/receive.c	2006-12-09 17:23:19.000000000 -0800
@@ -59,7 +59,7 @@
 		{
 			lirc_t data;
 			
-			data=hw.readdata(2*maxusec<50000 ? 50000:2*maxusec);
+			data=hw.readdata(2*maxusec<100000 ? 100000:2*maxusec);
 			if(!data)
 			{
 				LOGPRINTF(3,"timeout: %u", maxusec);
