diff -Naur lirc-0.7.2pre3-old/configure lirc-0.7.2pre3-new/configure
--- lirc-0.7.2pre3-old/configure	2005-08-06 11:29:59.000000000 -0700
+++ lirc-0.7.2pre3-new/configure	2005-08-06 12:39:01.000000000 -0700
@@ -8586,8 +8586,8 @@
 		;;
 	atilibusb)
 		hw_module="${hw_module} hw_atilibusb.o receive.o"
-		atilibusb_lib=`libusb-config --libs`
-		CFLAGS="$CFLAGS `libusb-config --cflags`"
+		atilibusb_lib=`${DESTDIR}${bindir}/libusb-config --libs`
+		CFLAGS="$CFLAGS `${DESTDIR}${bindir}/libusb-config --cflags`"
 		;;
 	audio)
 		hw_module="${hw_module} hw_audio.o receive.o"
