diff -Naur lirc-0.8.0-old/drivers/lirc_mceusb2/lirc_mceusb2.c lirc-0.8.0-new/drivers/lirc_mceusb2/lirc_mceusb2.c
--- lirc-0.8.0-old/drivers/lirc_mceusb2/lirc_mceusb2.c	2005-10-29 07:18:53.000000000 -0700
+++ lirc-0.8.0-new/drivers/lirc_mceusb2/lirc_mceusb2.c	2006-08-08 13:54:42.000000000 -0700
@@ -351,10 +351,6 @@
 		if ((ir->buf_in[0]==MCE_PACKET_HEADER)) {
 
 		    int i,keycode,pulse;
-			
-			/* buffer exhausted? */
-			if (ir->lirccnt>(LIRCBUF_SIZE-MCE_CODE_LENGTH))
-				send_packet_to_lirc(ir);
 
 		    for(i=0;i<MCE_PACKET_SIZE;i++) {
 			pulse = 0;
@@ -382,6 +378,7 @@
 			    ir->last_space += keycode;
 			}
 		    }
+                    send_packet_to_lirc(ir);
 		}
 	    }
 	    else {
