diff -Naur nss_ldap-258-old/ldap-pwd.c nss_ldap-258-new/ldap-pwd.c
--- nss_ldap-258-old/ldap-pwd.c	2007-10-12 15:10:10.000000000 -0700
+++ nss_ldap-258-new/ldap-pwd.c	2007-10-14 14:04:49.000000000 -0700
@@ -34,6 +34,7 @@
 #endif
 
 #include <stdlib.h>
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <string.h>
@@ -146,17 +147,70 @@
 	return stat;
     }
 
+  tmp = buffer;
+  tmplen = buflen;
   stat =
     _nss_ldap_assign_attrval (e, AT (homeDirectory), &pw->pw_dir, &buffer,
 			      &buflen);
   if (stat != NSS_SUCCESS)
     (void) _nss_ldap_assign_emptystring (&pw->pw_dir, &buffer, &buflen);
+  else 
+    {
+      /* if home directory does not exist, then override it. */
+      if (access (pw->pw_dir, F_OK))
+        {
+          pw->pw_dir = NULL;
+          buffer = tmp;
+          buflen = tmplen;
+          tmp = "/";
+          tmplen = strlen ("/");
+          if (buflen >= (tmplen + 1))
+            {
+               pw->pw_dir = buffer;
+               strncpy (pw->pw_dir, tmp, tmplen + 1);
+               pw->pw_dir[tmplen] = '\0';
+               buffer += tmplen + 1;
+               buflen -= tmplen + 1;
+            }
+          else
+            (void) _nss_ldap_assign_emptystring (&pw->pw_dir, &buffer, &buflen);
+        }
+    }
 
+  tmp = buffer;
+  tmplen = buflen;
   stat =
     _nss_ldap_assign_attrval (e, AT (loginShell), &pw->pw_shell, &buffer,
 			      &buflen);
   if (stat != NSS_SUCCESS)
     (void) _nss_ldap_assign_emptystring (&pw->pw_shell, &buffer, &buflen);
+  else 
+    {
+      /* if shell does not exist, then override it. */
+      if (access(pw->pw_shell, X_OK))
+        {
+          pw->pw_shell = NULL;
+          buffer = tmp;
+          buflen = tmplen;
+          tmp = "/bin/ash";
+          tmplen = strlen ("/bin/ash");
+          if (!access (tmp, X_OK))
+            {
+              if (buflen >= (tmplen + 1))
+                {
+                   pw->pw_shell = buffer;
+                   strncpy (pw->pw_shell, tmp, tmplen + 1);
+                   pw->pw_shell[tmplen] = '\0';
+                   buffer += tmplen + 1;
+                   buflen -= tmplen + 1;
+                }
+              else
+                (void) _nss_ldap_assign_emptystring (&pw->pw_shell, &buffer, &buflen);
+            }
+          else
+            (void) _nss_ldap_assign_emptystring (&pw->pw_shell, &buffer, &buflen);
+        }
+    }
 
 #ifdef HAVE_NSSWITCH_H
   stat =
