diff -Naur mythtv-0.20-16082-old/libs/libmyth/audiooutputalsa.cpp mythtv-0.20-16082-new/libs/libmyth/audiooutputalsa.cpp
--- mythtv-0.20-16082-old/libs/libmyth/audiooutputalsa.cpp	2008-09-15 14:09:20.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmyth/audiooutputalsa.cpp	2008-09-15 14:09:20.000000000 -0700
@@ -9,6 +9,9 @@
 #include "mythcontext.h"
 #include "audiooutputalsa.h"
     
+#define LOC QString("ALSA: ")
+#define LOC_WARN QString("ALSA, Warning: ")
+#define LOC_ERR QString("ALSA, Error: ")
 
 AudioOutputALSA::AudioOutputALSA(
     QString laudio_main_device, QString           laudio_passthru_device,
@@ -113,7 +116,7 @@
         return false;
     }
 
-    err = SetParameters(pcm_handle, SND_PCM_ACCESS_MMAP_INTERLEAVED,
+    err = SetParameters(pcm_handle,
                         format, audio_channels, audio_samplerate, buffer_time,
                         period_time);
     if (err < 0) 
@@ -164,7 +167,7 @@
     
     while (frames > 0) 
     {
-        lw = snd_pcm_mmap_writei(pcm_handle, tmpbuf, frames);
+        lw = pcm_write_func(pcm_handle, tmpbuf, frames);
         
         if (lw >= 0)
         {
@@ -221,7 +224,7 @@
         }
         else
         {
-            VERBOSE(VB_IMPORTANT, QString("snd_pcm_mmap_writei: %1 (%2)")
+            VERBOSE(VB_IMPORTANT, QString("pcm_write_func: %1 (%2)")
                     .arg(snd_strerror(lw)).arg(lw));
             VERBOSE(VB_IMPORTANT, QString("WriteAudio: snd_pcm_state == %1")
                     .arg(snd_pcm_state(pcm_handle)));
@@ -292,7 +295,7 @@
 }
 
 
-int AudioOutputALSA::SetParameters(snd_pcm_t *handle, snd_pcm_access_t access,
+int AudioOutputALSA::SetParameters(snd_pcm_t *handle,
                                    snd_pcm_format_t format, unsigned int channels,
                                    unsigned int rate, unsigned int buffer_time,
                                    unsigned int period_time)
@@ -324,12 +327,26 @@
         return err;
     }
 
-    /* set the interleaved read/write format */
-    if ((err = snd_pcm_hw_params_set_access(handle, params, access)) < 0)
+    /* set the interleaved read/write format, use mmap if available */
+    pcm_write_func = &snd_pcm_mmap_writei;
+    err = snd_pcm_hw_params_set_access(
+        handle, params, SND_PCM_ACCESS_MMAP_INTERLEAVED);
+    if (err < 0)
     {
-        Error(QString("Access type not available: %1")
-              .arg(snd_strerror(err)));
-        return err;
+        VERBOSE(VB_GENERAL, LOC_WARN +
+                "mmap not available, attempting to fall back to slow writes.");
+        QString old_err = snd_strerror(err);
+        pcm_write_func = &snd_pcm_writei;
+        err = snd_pcm_hw_params_set_access(
+            handle, params, SND_PCM_ACCESS_RW_INTERLEAVED);
+        if (err < 0)
+        {
+            Error("Interleaved sound types MMAP & RW are not available");
+            VERBOSE(VB_IMPORTANT,
+                    QString("MMAP Error: %1\n\t\t\tRW Error: %2")
+                    .arg(old_err).arg(snd_strerror(err)));
+            return err;
+        }
     }
 
     /* set the sample format */
diff -Naur mythtv-0.20-16082-old/libs/libmyth/audiooutputalsa.h mythtv-0.20-16082-new/libs/libmyth/audiooutputalsa.h
--- mythtv-0.20-16082-old/libs/libmyth/audiooutputalsa.h	2008-09-15 14:09:20.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmyth/audiooutputalsa.h	2008-09-15 14:09:20.000000000 -0700
@@ -37,7 +37,7 @@
     virtual inline int getBufferedOnSoundcard(void);
 
   private:
-    inline int SetParameters(snd_pcm_t *handle, snd_pcm_access_t access,
+    inline int SetParameters(snd_pcm_t *handle,
                              snd_pcm_format_t format, unsigned int channels,
                              unsigned int rate, unsigned int buffer_time,
                              unsigned int period_time);
@@ -59,6 +59,8 @@
     float        volume_range_multiplier;
     long         playback_vol_min;
     long         playback_vol_max;
+    snd_pcm_sframes_t (*pcm_write_func)(
+        snd_pcm_t*, const void*, snd_pcm_uframes_t);
 };
 
 #endif
