diff -Naur mythtv-0.20-16082-old/libs/libmythtv/tv_play.cpp mythtv-0.20-16082-new/libs/libmythtv/tv_play.cpp
--- mythtv-0.20-16082-old/libs/libmythtv/tv_play.cpp	2008-09-15 14:09:29.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmythtv/tv_play.cpp	2008-09-15 14:09:30.000000000 -0700
@@ -1151,7 +1151,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
