diff -Naur splashutils-1.5.3.4-old/src/Makefile.am splashutils-1.5.3.4-new/src/Makefile.am
--- splashutils-1.5.3.4-old/src/Makefile.am	2008-01-27 03:46:21.000000000 -0800
+++ splashutils-1.5.3.4-new/src/Makefile.am	2008-01-28 09:46:25.000000000 -0800
@@ -120,7 +120,7 @@
 	fbsplash.h
 libfbsplashrender_la_CFLAGS   = $(AM_CFLAGS)
 libfbsplashrender_la_LDFLAGS  = $(AM_LDFLAGS) -version-info $(libfbsplashrender_version)
-libfbsplashrender_la_LIBADD   = -lfbsplash
+libfbsplashrender_la_LIBADD   = libfbsplash.la
 
 libfbsplashrender_la_CFLAGS  += $(JPEG_CFLAGS)
 libfbsplashrender_la_LIBADD  += $(JPEG_LIBS)
