diff -Naur splashutils-1.4.3-old/Makefile splashutils-1.4.3-new/Makefile
--- splashutils-1.4.3-old/Makefile	2007-07-07 07:07:34.000000000 -0700
+++ splashutils-1.4.3-new/Makefile	2007-07-12 21:02:23.000000000 -0700
@@ -26,15 +26,15 @@
 LIB		?= lib
 
 prefix     =
-sbindir    = ${prefix}/sbin
-libdir     = ${prefix}/${LIB}
-usrbindir  = ${prefix}/usr/bin
-mandir 	   = ${prefix}/usr/share/man
+sbindir    = $(prefix)/sbin
+libdir     = $(prefix)/$(LIB)
+usrbindir  = $(prefix)/usr/bin
+mandir 	   = $(prefix)/usr/share/man
 
 INSTALL 		= /usr/bin/install -c
-INSTALL_PROG 	= ${INSTALL}
-INSTALL_DATA 	= ${INSTALL} -m 644
-INSTALL_SCRIPT 	= ${INSTALL_PROG}
+INSTALL_PROG 	= $(INSTALL)
+INSTALL_DATA 	= $(INSTALL) -m 644
+INSTALL_SCRIPT 	= $(INSTALL_PROG)
 
 ifeq ($(strip $(K_SHARED)),true)
 	K_LDFLAGS = -shared
@@ -117,7 +117,7 @@
 OBJDIR	= objs
 KOUT	= $(OBJDIR)/kernel
 dotg    = \e[32;01m*\e[0m
-infmsg	= if [ "${QUIET}" = "true" ] ; then printf "  %-7s %s\n" $(1) $(2) ; fi
+infmsg	= if [ "$(QUIET)" = "true" ] ; then printf "  %-7s %s\n" $(1) $(2) ; fi
 install_script = $(call infmsg,INSTALL,$(1)) ; $(INSTALL_PROG) -D scripts/$(1) $(DESTDIR)$(usrbindir)/$(1) ; \
 		 sed -e 's/%PKG_VERSION%/$(PKG_VERSION)/g' -i $(DESTDIR)$(usrbindir)/$(1)
 
