diff -Naur mythstream-0.17_2-old/libs/streamstorage.cpp mythstream-0.17_2-new/libs/streamstorage.cpp
--- mythstream-0.17_2-old/libs/streamstorage.cpp	2006-10-14 13:04:31.000000000 -0700
+++ mythstream-0.17_2-new/libs/streamstorage.cpp	2006-11-30 14:01:22.000000000 -0800
@@ -144,7 +144,7 @@
           }
         }
         else
-          if ( fi->fileName() != "list.xml" ) // shouldn't be there
+          if ( fi->fileName() != "list.xml" && fi->fileName().right(3) == ".pl" ) // PERL only, leave binaries 
             copyFile(sourceDir + "/" + fi->fileName(), destDir + "/" + fi->fileName() );
         ++it;
     }
