diff -Naur xf86-video-radeonhd-1.2.5-old/configure.ac xf86-video-radeonhd-1.2.5-new/configure.ac
--- xf86-video-radeonhd-1.2.5-old/configure.ac	2009-04-09 06:11:34.000000000 -0700
+++ xf86-video-radeonhd-1.2.5-new/configure.ac	2009-04-09 06:33:43.000000000 -0700
@@ -76,7 +76,8 @@
 # @@@@ need to check for the right server version once RandR 1.2 is in.
 # @@@ do we really need fontsproto here?
 PKG_CHECK_MODULES(XORG, [xorg-server xproto fontsproto $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 
 # Header files checks.
