diff -Naur mythtv-0.27-20150529-b7afc9b-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-0.27-20150529-b7afc9b-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-0.27-20150529-b7afc9b-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2015-06-09 00:06:44.047000363 -0700
+++ mythtv-0.27-20150529-b7afc9b-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2015-06-09 00:06:46.579000362 -0700
@@ -2483,6 +2483,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
