diff -Naur libdrm-2.4.17-old/include/drm/drm.h libdrm-2.4.17-new/include/drm/drm.h
--- libdrm-2.4.17-old/include/drm/drm.h	2009-12-08 20:34:56.000000000 -0800
+++ libdrm-2.4.17-new/include/drm/drm.h	2010-01-03 21:42:08.000000000 -0800
@@ -44,6 +44,7 @@
 
 #else /* One of the BSDs */
 
+#include <stdint.h>
 #include <sys/ioccom.h>
 #include <sys/types.h>
 typedef int8_t   __s8;
diff -Naur libdrm-2.4.17-old/include/drm/drm_mode.h libdrm-2.4.17-new/include/drm/drm_mode.h
--- libdrm-2.4.17-old/include/drm/drm_mode.h	2009-12-08 20:34:56.000000000 -0800
+++ libdrm-2.4.17-new/include/drm/drm_mode.h	2010-01-03 21:42:48.000000000 -0800
@@ -27,6 +27,8 @@
 #ifndef _DRM_MODE_H
 #define _DRM_MODE_H
 
+#include <stdint.h>
+
 #define DRM_DISPLAY_INFO_LEN	32
 #define DRM_CONNECTOR_NAME_LEN	32
 #define DRM_DISPLAY_MODE_LEN	32
diff -Naur libdrm-2.4.17-old/include/drm/nouveau_drm.h libdrm-2.4.17-new/include/drm/nouveau_drm.h
--- libdrm-2.4.17-old/include/drm/nouveau_drm.h	2009-11-30 15:17:08.000000000 -0800
+++ libdrm-2.4.17-new/include/drm/nouveau_drm.h	2010-01-03 21:42:53.000000000 -0800
@@ -25,6 +25,8 @@
 #ifndef __NOUVEAU_DRM_H__
 #define __NOUVEAU_DRM_H__
 
+#include <stdint.h>
+
 #define NOUVEAU_DRM_HEADER_PATCHLEVEL 15
 
 struct drm_nouveau_channel_alloc {
diff -Naur libdrm-2.4.17-old/include/drm/radeon_drm.h libdrm-2.4.17-new/include/drm/radeon_drm.h
--- libdrm-2.4.17-old/include/drm/radeon_drm.h	2009-11-30 15:17:08.000000000 -0800
+++ libdrm-2.4.17-new/include/drm/radeon_drm.h	2010-01-03 21:43:10.000000000 -0800
@@ -33,6 +33,8 @@
 #ifndef __RADEON_DRM_H__
 #define __RADEON_DRM_H__
 
+#include <stdint.h>
+
 #include "drm.h"
 
 /* WARNING: If you change any of these defines, make sure to change the
