diff -Naur linux-2.6.22.1-old/drivers/ata/libata-core.c linux-2.6.22.1-new/drivers/ata/libata-core.c
--- linux-2.6.22.1-old/drivers/ata/libata-core.c	2007-07-10 11:56:30.000000000 -0700
+++ linux-2.6.22.1-new/drivers/ata/libata-core.c	2007-08-12 03:35:11.000000000 -0700
@@ -3755,6 +3755,8 @@
 	{ "CRD-8480B",		NULL,		ATA_HORKAGE_NODMA },
 	{ "CRD-8482B",		NULL,		ATA_HORKAGE_NODMA },
 	{ "CRD-84",		NULL,		ATA_HORKAGE_NODMA },
+	{ "SanDisk SDCFB-1024",	NULL,		ATA_HORKAGE_NODMA },
+	{ "SanDisk SDCFH-1024",	NULL,		ATA_HORKAGE_NODMA },
 	{ "SanDisk SDP3B",	NULL,		ATA_HORKAGE_NODMA },
 	{ "SanDisk SDP3B-64",	NULL,		ATA_HORKAGE_NODMA },
 	{ "SANYO CD-ROM CRD",	NULL,		ATA_HORKAGE_NODMA },
diff -Naur linux-2.6.22.1-old/drivers/ide/ide-dma.c linux-2.6.22.1-new/drivers/ide/ide-dma.c
--- linux-2.6.22.1-old/drivers/ide/ide-dma.c	2007-07-10 11:56:30.000000000 -0700
+++ linux-2.6.22.1-new/drivers/ide/ide-dma.c	2007-08-12 03:35:32.000000000 -0700
@@ -112,6 +112,8 @@
 	{ "CRD-8480B",			"ALL"		},
 	{ "CRD-8482B",			"ALL"		},
  	{ "CRD-84"		,	"ALL"		},
+	{ "SanDisk SDCFB-1024"	,	"ALL"		},
+	{ "SanDisk SDCFH-1024"	,	"ALL"		},
 	{ "SanDisk SDP3B"	,	"ALL"		},
 	{ "SanDisk SDP3B-64"	,	"ALL"		},
 	{ "SANYO CD-ROM CRD"	,	"ALL"		},
