diff -Naur xf86-video-openchrome-292-old/unichrome/via_vt162x.h xf86-video-openchrome-292-new/unichrome/via_vt162x.h
--- xf86-video-openchrome-292-old/unichrome/via_vt162x.h	2007-02-06 20:19:25.000000000 -0800
+++ xf86-video-openchrome-292-new/unichrome/via_vt162x.h	2007-02-06 20:20:47.000000000 -0800
@@ -755,9 +755,9 @@
     { MODEPREFIX("1024x768Over"), ... ,   MODESUFFIXPAL },
     { MODEPREFIX("720x576Over"), ... ,   MODESUFFIXPAL },*/
 
-    { MODEPREFIX("640x480"),  30000,  640,  680,  808, 1000, 0,  480,  520,  523,  600, 0, V_NHSYNC | V_NVSYNC, MODESUFFIXPAL  },
-    { MODEPREFIX("800x600"),  34500,  800,  816,  880,  920, 0,  600,  604,  620,  750, 0, V_PHSYNC | V_PVSYNC, MODESUFFIXPAL  },
-    { MODEPREFIX("1024x768"), 57000, 1024, 1040, 1112, 1200, 0,  768,  829,  840,  950, 0, V_NHSYNC | V_NVSYNC, MODESUFFIXPAL  },
+    { MODEPREFIX("640x480"),  30000,  640,  704,  760, 1000, 0,  480,  500,  506,  600, 0, V_NHSYNC | V_NVSYNC, MODESUFFIXPAL  },
+    { MODEPREFIX("800x600"),  34500,  800,  816,  880,  920, 0,  600,  606,  622,  750, 0, V_PHSYNC | V_PVSYNC, MODESUFFIXPAL  },
+    { MODEPREFIX("1024x768"), 57000, 1024, 1048, 1064, 1200, 0,  768,  800,  806,  975, 0, V_NHSYNC | V_NVSYNC, MODESUFFIXPAL  },
 
     { MODEPREFIX("720x576"),  28500,  720,  728,  744,  760, 0,  576,  635,  643,  750, 0, V_NHSYNC | V_PVSYNC, MODESUFFIXPAL  },
     { MODEPREFIX("720x576Over"),  27000,  720,  768,  800,  864, 0,  576,  577,  579,  625, 0, V_NHSYNC | V_PVSYNC, MODESUFFIXPAL  },
@@ -912,6 +912,54 @@
       0x1E555518, 0x1E554CC3,
     },
 */
+    { "640x480", 640, 480, TVTYPE_PAL, 0, 0,
+      /*  00                                                                                         0F */
+      { 0x03,    0, 0x10, 0x1F, 0x03,    0,    0, 0xF2,  0x42, 0x0A, 0x88,    0, 0x55, 0x5E, 0x07, 0xB0,
+	   0, 0x80, 0x0A, 0x09, 0x19, 0x28, 0xCB, 0x8A,  0x09, 0x2A, 0x06, 0x50, 0x01, 0x80,    0, 0x10,
+	0x17, 0x1C, 0x35, 0x7D, 0x02, 0x5F, 0x34, 0x8C,  0x4F, 0x5E, 0x15, 0xA2, 0x22, 0x80, 0xD3, 0x10 },
+      /*  4A                            4F    50                                                     59 */
+      { 0xC5, 0x0F,    0, 0x01, 0x10, 0x51, 0xE7, 0x81,  0x23, 0x57, 0x22, 0x5F, 0x6F, 0x7F, 0x23, 0x94,
+      /*  5A                            5F    60                       64 */
+	0xD0, 0x1C, 0x8F, 0x16, 0xE1, 0x38, 0x7A, 0x28,  0xFF, 0xBF, 0x03 },
+      /* RBG 65,66,67,27,2b,2c */
+      { 0x51, 0x00, 0x00, 0x8C, 0xA2, 0x22 },
+      /* Y-Cb-Cr 65,66,67 */
+      { 0x51, 0x00, 0x00 },
+      /* Subcarrier 19,18,17,16, DotCrawl Subcarrier (set bit 3 of reg 11 then subcarrier) */
+      0x2A0984CB, 0x2A0984CB,
+    },
+    { "800x600", 800, 600, TVTYPE_PAL, 0, 0,
+      /*  00                                                                                         0F */
+      { 0x03,    0, 0x10, 0x90, 0x03,    0,    0, 0x28,  0x41, 0x0F, 0x88,    0, 0x55, 0x5E, 0x07, 0xB0,
+	   0, 0x80, 0x12, 0x04, 0x17, 0x48, 0xCB, 0x8A,  0x09, 0x2A, 0x06, 0x50, 0x01, 0x80,    0, 0x10,
+	0x17, 0x1C, 0x31, 0x7D, 0x02, 0x5F, 0x34, 0x8C,  0x4F, 0x5E, 0x15, 0xA2, 0x22, 0x80, 0xD3, 0x10 },
+      /*  4A                            4F    50                                                     59 */
+      { 0xC5, 0x0F,    0, 0x01, 0x10, 0x51, 0x97, 0x22,  0x33, 0xED, 0x22, 0x5F, 0x73, 0x7F, 0x23, 0x94,
+      /*  5A                            5F    60                       64 */
+	0xD0, 0x27, 0x8F, 0x16, 0x99, 0x38, 0x99, 0xAA,  0xFF, 0x4F, 0x04 },
+      /* RBG 65,66,67,27,2b,2c */
+      { 0x51, 0x00, 0x00,    0,    0,    0 },
+      /* Y-Cb-Cr 65,66,67 */
+      { 0x51, 0x00, 0x00 },
+      /* Subcarrier 19,18,17,16, DotCrawl Subcarrier (set bit 3 of reg 11 then subcarrier) */
+      0x2A098ACB, 0x2A098ACB,
+    },
+    { "1024x768", 1024, 768, TVTYPE_PAL, 0, 0,
+      /*  00                                                                                         0F */
+      { 0x03,    0, 0x10, 0x9A, 0x03,    0,    0, 0x89,  0x64, 0x0D, 0x88,    0, 0x55, 0x5E, 0x07, 0xB0,
+	   0, 0x80, 0x2D, 0x08, 0x27, 0x24, 0x87, 0x5C,  0x06, 0x1C, 0x83, 0x50, 0x01, 0x80,    0,    0,
+	0x1D, 0x1C, 0x89, 0x7D, 0x02, 0x5F, 0x47, 0x8C,  0x74, 0x5E, 0x15, 0xA2, 0x22, 0xC6, 0xF6, 0x10 },
+      /*  4A                            4F    50                                                     59 */
+      { 0xC5, 0x0F,    0, 0x01, 0x10, 0x51, 0xAF, 0x02,  0x44, 0xCE, 0x03, 0x0F, 0xA4, 0xBE, 0x35, 0xDC,
+      /*  5A                            5F    60                       64 */
+	0x3E, 0xB5, 0xDB, 0x59, 0xAE, 0xCD, 0xC7, 0x28,  0xFF, 0x4F, 0x07 },
+      /* RBG 65,66,67,27,2b,2c */
+      { 0x51, 0x00, 0x00, 0x8C, 0xA2, 0x22 },
+      /* Y-Cb-Cr 65,66,67 */
+      { 0x51, 0x00, 0x00 },
+      /* Subcarrier 19,18,17,16, DotCrawl Subcarrier (set bit 3 of reg 11 then subcarrier) */
+      0x1C065C87, 0x1C065C87,
+    },
     { NULL, 0, 0, 0, 0, 0,
       { 0, 0, 0, 0, 0, 0, 0, 0,  0, 0, 0, 0, 0, 0, 0, 0,
 	0, 0, 0, 0, 0, 0, 0, 0,  0, 0, 0, 0, 0, 0, 0, 0,
