diff -Naur xorg-server-1.1.99.903-old/GL/mesa/main/Makefile.am xorg-server-1.1.99.903-new/GL/mesa/main/Makefile.am
--- xorg-server-1.1.99.903-old/GL/mesa/main/Makefile.am	2006-12-01 14:48:41.000000000 -0800
+++ xorg-server-1.1.99.903-new/GL/mesa/main/Makefile.am	2006-12-05 15:20:10.000000000 -0800
@@ -60,6 +60,7 @@
                     light.c \
                     lines.c \
                     matrix.c \
+                    mipmap.c \
                     mm.c \
                     occlude.c \
                     pixel.c \
diff -Naur xorg-server-1.1.99.903-old/GL/mesa/main/Makefile.in xorg-server-1.1.99.903-new/GL/mesa/main/Makefile.in
--- xorg-server-1.1.99.903-old/GL/mesa/main/Makefile.in	2006-12-01 15:47:13.000000000 -0800
+++ xorg-server-1.1.99.903-new/GL/mesa/main/Makefile.in	2006-12-05 15:20:24.000000000 -0800
@@ -598,6 +598,7 @@
                     light.c \
                     lines.c \
                     matrix.c \
+                    mipmap.c \
                     mm.c \
                     occlude.c \
                     pixel.c \
