diff -Naur xorg-server-1.1.99.903-old/hw/xfree86/os-support/linux/Makefile.am xorg-server-1.1.99.903-new/hw/xfree86/os-support/linux/Makefile.am
--- xorg-server-1.1.99.903-old/hw/xfree86/os-support/linux/Makefile.am	2006-12-01 14:48:41.000000000 -0800
+++ xorg-server-1.1.99.903-new/hw/xfree86/os-support/linux/Makefile.am	2006-12-05 13:33:07.000000000 -0800
@@ -38,7 +38,7 @@
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS) $(PLATFORM_DEFINES)
 
-INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES)
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
diff -Naur xorg-server-1.1.99.903-old/hw/xfree86/os-support/linux/Makefile.in xorg-server-1.1.99.903-new/hw/xfree86/os-support/linux/Makefile.in
--- xorg-server-1.1.99.903-old/hw/xfree86/os-support/linux/Makefile.in	2006-12-01 15:47:41.000000000 -0800
+++ xorg-server-1.1.99.903-new/hw/xfree86/os-support/linux/Makefile.in	2006-12-05 13:33:07.000000000 -0800
@@ -574,7 +574,7 @@
                      $(PLATFORM_PCI_SUPPORT)
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS) $(PLATFORM_DEFINES)
-INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES)
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
