diff -Naur xorg-server-1.1.99.901-old/hw/xfree86/os-support/linux/Makefile.am xorg-server-1.1.99.901-new/hw/xfree86/os-support/linux/Makefile.am
--- xorg-server-1.1.99.901-old/hw/xfree86/os-support/linux/Makefile.am	2006-10-11 15:42:19.000000000 -0700
+++ xorg-server-1.1.99.901-new/hw/xfree86/os-support/linux/Makefile.am	2006-11-06 18:47:54.000000000 -0800
@@ -36,7 +36,7 @@
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS)
 
-INCLUDES = $(XORG_INCS) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
diff -Naur xorg-server-1.1.99.901-old/hw/xfree86/os-support/linux/Makefile.in xorg-server-1.1.99.901-new/hw/xfree86/os-support/linux/Makefile.in
--- xorg-server-1.1.99.901-old/hw/xfree86/os-support/linux/Makefile.in	2006-10-13 15:53:55.000000000 -0700
+++ xorg-server-1.1.99.901-new/hw/xfree86/os-support/linux/Makefile.in	2006-11-06 18:49:06.000000000 -0800
@@ -569,7 +569,7 @@
                      $(PLATFORM_PCI_SUPPORT)
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(XORG_CFLAGS)
-INCLUDES = $(XORG_INCS) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SOURCES = \
