diff -Naur alsa-driver-1.0.23-old/alsa-kernel/usb/urb.c alsa-driver-1.0.23-new/alsa-kernel/usb/urb.c
--- alsa-driver-1.0.23-old/alsa-kernel/usb/urb.c	2010-04-16 04:10:10.000000000 -0700
+++ alsa-driver-1.0.23-new/alsa-kernel/usb/urb.c	2010-05-18 07:13:04.000000000 -0700
@@ -17,6 +17,7 @@
 
 #include <linux/gfp.h>
 #include <linux/init.h>
+#include <linux/slab.h>
 #include <linux/usb.h>
 #include <linux/usb/audio.h>
 
