diff -Naur bluez-utils-3.29-old/audio/device.c bluez-utils-3.29-new/audio/device.c
--- bluez-utils-3.29-old/audio/device.c	2008-02-01 15:16:30.000000000 -0800
+++ bluez-utils-3.29-new/audio/device.c	2008-03-21 23:07:45.000000000 -0700
@@ -26,6 +26,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
diff -Naur bluez-utils-3.29-old/hcid/dbus-security.c bluez-utils-3.29-new/hcid/dbus-security.c
--- bluez-utils-3.29-old/hcid/dbus-security.c	2008-03-21 15:19:25.000000000 -0700
+++ bluez-utils-3.29-new/hcid/dbus-security.c	2008-03-21 23:07:45.000000000 -0700
@@ -27,6 +27,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <errno.h>
 #include <stdlib.h>
diff -Naur bluez-utils-3.29-old/hcid/dbus-test.c bluez-utils-3.29-new/hcid/dbus-test.c
--- bluez-utils-3.29-old/hcid/dbus-test.c	2008-02-01 15:16:32.000000000 -0800
+++ bluez-utils-3.29-new/hcid/dbus-test.c	2008-03-21 23:07:45.000000000 -0700
@@ -26,6 +26,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <errno.h>
 #include <stdlib.h>
diff -Naur bluez-utils-3.29-old/network/common.c bluez-utils-3.29-new/network/common.c
--- bluez-utils-3.29-old/network/common.c	2008-02-01 15:16:33.000000000 -0800
+++ bluez-utils-3.29-new/network/common.c	2008-03-21 23:07:45.000000000 -0700
@@ -25,6 +25,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
diff -Naur bluez-utils-3.29-old/network/connection.c bluez-utils-3.29-new/network/connection.c
--- bluez-utils-3.29-old/network/connection.c	2008-02-01 15:16:33.000000000 -0800
+++ bluez-utils-3.29-new/network/connection.c	2008-03-21 23:07:45.000000000 -0700
@@ -25,6 +25,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
diff -Naur bluez-utils-3.29-old/network/server.c bluez-utils-3.29-new/network/server.c
--- bluez-utils-3.29-old/network/server.c	2008-02-10 11:38:30.000000000 -0800
+++ bluez-utils-3.29-new/network/server.c	2008-03-21 23:07:45.000000000 -0700
@@ -25,6 +25,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
diff -Naur bluez-utils-3.29-old/serial/manager.c bluez-utils-3.29-new/serial/manager.c
--- bluez-utils-3.29-old/serial/manager.c	2008-02-01 15:16:33.000000000 -0800
+++ bluez-utils-3.29-new/serial/manager.c	2008-03-21 23:07:45.000000000 -0700
@@ -25,6 +25,7 @@
 #include <config.h>
 #endif
 
+#include <limits.h>
 #include <ctype.h>
 #include <dirent.h>
 #include <errno.h>
