diff -Naur transcode-1.1.5-old/configure.in transcode-1.1.5-new/configure.in
--- transcode-1.1.5-old/configure.in	2009-10-31 09:39:02.000000000 -0700
+++ transcode-1.1.5-new/configure.in	2010-08-27 10:34:57.000000000 -0700
@@ -1291,7 +1291,7 @@
     yes) if test yes != "$have_lzo"; then
            AC_MSG_ERROR([liblzo2 is required for NuppelVideo support, try --enable-lzo])
          fi
-    	 if test true == "$have_x86_64cpu"; then
+    	 if test true = "$have_x86_64cpu"; then
            AC_MSG_WARN([NuppelVideo support on x86_64 platform is ***EXPERIMENTAL***])
          elif test true != "$have_x86cpu"; then
            AC_MSG_ERROR([NuppelVideo is supported only on x86 and x86_64 platforms yet])
