diff -Naur transcode-1.0.5-old/import/nuv/RTjpegN.h transcode-1.0.5-new/import/nuv/RTjpegN.h
--- transcode-1.0.5-old/import/nuv/RTjpegN.h	2003-03-08 05:42:48.000000000 -0800
+++ transcode-1.0.5-new/import/nuv/RTjpegN.h	2008-01-24 19:45:30.000000000 -0800
@@ -22,7 +22,10 @@
     
 */
 
+#include <asm/types.h>
+
 #ifndef _I386_TYPES_H
+#ifndef _ASM_X86_TYPES_H
 typedef unsigned char __u8;
 typedef unsigned short __u16;
 typedef unsigned long __u32;
@@ -31,10 +34,11 @@
 typedef signed short __s16;
 typedef signed long __s32;
 #endif
+#endif
 
 extern void RTjpeg_init_Q(__u8 Q);
-extern void RTjpeg_init_compress(long unsigned int *buf, int width, int height, __u8 Q);
-extern void RTjpeg_init_decompress(long unsigned int *buf, int width, int height);
+extern void RTjpeg_init_compress(__u32 *buf, int width, int height, __u8 Q);
+extern void RTjpeg_init_decompress(__u32 *buf, int width, int height);
 extern int RTjpeg_compressYUV420(__s8 *sp, unsigned char *bp);
 extern int RTjpeg_compressYUV422(__s8 *sp, unsigned char *bp);
 extern void RTjpeg_decompressYUV420(__s8 *sp, __u8 *bp);
