diff -Naur transcode-1.0.2-old/configure.in transcode-1.0.2-new/configure.in
--- transcode-1.0.2-old/configure.in	2005-10-29 21:57:45.000000000 -0700
+++ transcode-1.0.2-new/configure.in	2006-02-19 08:53:09.000000000 -0800
@@ -865,8 +865,9 @@
 ],
     [lame_version="`./conftest$ac_exeext`"],
     [TC_PKG_ERROR(lame, lame.h, yes, lame, [http://www.mp3dev.org/],
-      [cannot compile and run a test program])],,
-    [echo $ac_n "cross compiling; assumed OK... $ac_c"])
+      [cannot compile and run a test program])],
+    [echo $ac_n "cross compiling; assumed OK... $ac_c"
+    AC_DEFINE_UNQUOTED([TC_LAME_VERSION], 396, [lame version])])
   CFLAGS="$ac_save_CFLAGS"
   LIBS="$ac_save_LIBS"
 
