diff -Naur cwiid-183-old/wminput/plugins/ir_ptr/Makefile.in cwiid-183-new/wminput/plugins/ir_ptr/Makefile.in
--- cwiid-183-old/wminput/plugins/ir_ptr/Makefile.in	2008-12-28 16:30:17.000000000 -0800
+++ cwiid-183-new/wminput/plugins/ir_ptr/Makefile.in	2008-12-28 17:30:14.000000000 -0800
@@ -5,6 +5,7 @@
 PLUGIN_NAME = ir_ptr
 SOURCES = ir_ptr.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
+LDLIBS += -lm
 INST_DIR = $(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
diff -Naur cwiid-183-old/wminput/plugins/led/Makefile.in cwiid-183-new/wminput/plugins/led/Makefile.in
--- cwiid-183-old/wminput/plugins/led/Makefile.in	2008-12-28 16:30:18.000000000 -0800
+++ cwiid-183-new/wminput/plugins/led/Makefile.in	2008-12-28 17:30:17.000000000 -0800
@@ -5,6 +5,7 @@
 PLUGIN_NAME = led
 SOURCES = led.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
+LDLIBS += -lm
 INST_DIR = $(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
