diff -Naur xf86-input-evdev-2.2.0-old/src/evdev.h xf86-input-evdev-2.2.0-new/src/evdev.h
--- xf86-input-evdev-2.2.0-old/src/evdev.h	2009-03-08 17:00:05.000000000 -0700
+++ xf86-input-evdev-2.2.0-new/src/evdev.h	2009-03-09 09:07:56.000000000 -0700
@@ -35,7 +35,7 @@
 
 #include <xf86Xinput.h>
 #include <xf86_OSproc.h>
-#include <xkbstr.h>
+#include <X11/extensions/XKBstr.h>
 
 #ifndef EV_CNT /* linux 2.4 kernels and earlier lack _CNT defines */
 #define EV_CNT (EV_MAX+1)
