diff -Naur xf86-input-evdev-2.2.1-old/configure.ac xf86-input-evdev-2.2.1-new/configure.ac
--- xf86-input-evdev-2.2.1-old/configure.ac	2009-03-24 02:44:49.000000000 -0700
+++ xf86-input-evdev-2.2.1-new/configure.ac	2009-03-24 02:45:36.000000000 -0700
@@ -75,9 +75,10 @@
 # to satisfy silly distcheck requirements.
 PKG_CHECK_MODULES(XORG, xorg-server xproto $REQUIRED_MODULES)
 XORG_CFLAGS="$CWARNFLAGS $XORG_CFLAGS"
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
 AC_ARG_WITH([sdkdir], [],
     [sdkdir="$withval"],
-    [sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`])
+    [sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"])
 AC_SUBST([sdkdir])
 
 # Checks for libraries.
