diff -Naur ntp-4.2.2-old/ntpdc/Makefile.in ntp-4.2.2-new/ntpdc/Makefile.in
--- ntp-4.2.2-old/ntpdc/Makefile.in	2006-06-06 13:24:16.000000000 -0700
+++ ntp-4.2.2-new/ntpdc/Makefile.in	2006-06-19 18:16:05.000000000 -0700
@@ -489,7 +489,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
