diff -Naur ntp-4.2.4-old/ntpdc/Makefile.am ntp-4.2.4-new/ntpdc/Makefile.am
--- ntp-4.2.4-old/ntpdc/Makefile.am	2006-12-28 04:03:51.000000000 -0800
+++ ntp-4.2.4-new/ntpdc/Makefile.am	2006-12-30 11:19:08.000000000 -0800
@@ -40,7 +40,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
diff -Naur ntp-4.2.4-old/ntpdc/Makefile.in ntp-4.2.4-new/ntpdc/Makefile.in
--- ntp-4.2.4-old/ntpdc/Makefile.in	2006-12-28 15:59:02.000000000 -0800
+++ ntp-4.2.4-new/ntpdc/Makefile.in	2006-12-30 11:19:08.000000000 -0800
@@ -616,7 +616,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
