diff -Naur mythstream-0.18_1-old/mythstream.pro mythstream-0.18_1-new/mythstream.pro
--- mythstream-0.18_1-old/mythstream.pro	2007-09-25 16:07:30.000000000 -0700
+++ mythstream-0.18_1-new/mythstream.pro	2007-11-16 17:27:38.000000000 -0800
@@ -1,6 +1,6 @@
 TEMPLATE = subdirs
 
 # Directories
-SUBDIRS = mythstream
-#SUBDIRS = mythstream alt_parsers
+#SUBDIRS = mythstream
+SUBDIRS = mythstream alt_parsers
 
diff -Naur mythstream-0.18_1-old/settings.pro mythstream-0.18_1-new/settings.pro
--- mythstream-0.18_1-old/settings.pro	2007-11-16 17:27:21.000000000 -0800
+++ mythstream-0.18_1-new/settings.pro	2007-11-16 17:27:38.000000000 -0800
@@ -13,9 +13,9 @@
 DEFINES += SHELL=2
 DEFINES += EXEC=3
 
-DEFINES += PARSER=PERL
+#DEFINES += PARSER=PERL
 #DEFINES += PARSER=SHELL
-#DEFINES += PARSER=EXEC
+DEFINES += PARSER=EXEC
 
 DEFINES += _GNU_SOURCE
 DEFINES += PREFIX=\"$${PREFIX}\"
