diff -Naur Mesa-8.0.2-old/configure.ac Mesa-8.0.2-new/configure.ac
--- Mesa-8.0.2-old/configure.ac	2012-03-21 08:54:27.000000000 -0700
+++ Mesa-8.0.2-new/configure.ac	2012-03-22 18:10:36.000000000 -0700
@@ -1770,7 +1770,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version | sed 's/svn.*//g'`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags|sed -e 's/-DNDEBUG\>//g' -e 's/-pedantic//g'`
-	LLVM_LIBS="`$LLVM_CONFIG --libs`"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	DEFINES="$DEFINES -D__STDC_CONSTANT_MACROS"
