diff -Naur Mesa-7.10.3-old/configure.ac Mesa-7.10.3-new/configure.ac
--- Mesa-7.10.3-old/configure.ac	2011-06-14 17:25:53.000000000 -0700
+++ Mesa-7.10.3-new/configure.ac	2011-06-14 17:27:02.000000000 -0700
@@ -1557,7 +1557,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs jit interpreter nativecodegen bitwriter` -lstdc++"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	if test "x$HAS_UDIS86" != xno; then
 	    LLVM_LIBS="$LLVM_LIBS -ludis86"
