diff -Naur Mesa-7.11.2-old/src/glsl/Makefile Mesa-7.11.2-new/src/glsl/Makefile
--- Mesa-7.11.2-old/src/glsl/Makefile	2012-01-17 06:14:01.000000000 -0800
+++ Mesa-7.11.2-new/src/glsl/Makefile	2012-01-17 06:14:32.000000000 -0800
@@ -209,6 +209,6 @@
 
 builtin_function.cpp: builtins/profiles/* builtins/ir/* builtins/tools/generate_builtins.py builtins/tools/texture_builtins.py builtin_compiler
 	@echo Regenerating builtin_function.cpp...
-	$(PYTHON2) $(PYTHON_FLAGS) builtins/tools/generate_builtins.py ./builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
+	$(PYTHON2) $(PYTHON_FLAGS) builtins/tools/generate_builtins.py mesa-builtin_compiler > builtin_function.cpp || rm -f builtin_function.cpp
 
 -include depend
