diff -Naur mythplugins-0.20.2-old/mythdvd/mtd/logging.cpp mythplugins-0.20.2-new/mythdvd/mtd/logging.cpp
--- mythplugins-0.20.2-old/mythdvd/mtd/logging.cpp	2006-02-16 22:55:46.000000000 -0800
+++ mythplugins-0.20.2-new/mythdvd/mtd/logging.cpp	2008-01-16 06:53:26.000000000 -0800
@@ -9,6 +9,7 @@
 */
 #include "logging.h"
 
+#include <stdlib.h>
 #include <unistd.h>
 #include <qdatetime.h>
 
diff -Naur mythplugins-0.20.2-old/mythdvd/mtd/mtd.cpp mythplugins-0.20.2-new/mythdvd/mtd/mtd.cpp
--- mythplugins-0.20.2-old/mythdvd/mtd/mtd.cpp	2006-03-05 18:56:29.000000000 -0800
+++ mythplugins-0.20.2-new/mythdvd/mtd/mtd.cpp	2008-01-16 06:56:19.000000000 -0800
@@ -8,6 +8,7 @@
 
 */
 
+#include <stdlib.h>
 #include <unistd.h>
 #include <qstringlist.h>
 #include <qregexp.h>
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/aacdecoder.cpp	2006-02-13 08:58:55.000000000 -0800
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/aacdecoder.cpp	2008-01-16 09:23:39.000000000 -0800
@@ -16,6 +16,7 @@
 
 #include <iostream>
 #include <string>
+#include <stdlib.h>
 #include <qobject.h>
 #include <qiodevice.h>
 #include <qfile.h>
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/databasebox.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/databasebox.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/databasebox.cpp	2006-02-15 23:13:21.000000000 -0800
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/databasebox.cpp	2008-01-16 12:30:46.000000000 -0800
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <iostream>
 using namespace std;
 
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/flacdecoder.h mythplugins-0.20.2-new/mythmusic/mythmusic/flacdecoder.h
--- mythplugins-0.20.2-old/mythmusic/mythmusic/flacdecoder.h	2007-07-30 10:41:27.000000000 -0700
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/flacdecoder.h	2008-01-16 11:31:48.000000000 -0800
@@ -1,7 +1,7 @@
 #ifndef FLACDECODER_H_
 #define FLACDECODER_H_
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/flacencoder.h mythplugins-0.20.2-new/mythmusic/mythmusic/flacencoder.h
--- mythplugins-0.20.2-old/mythmusic/mythmusic/flacencoder.h	2007-02-18 21:51:08.000000000 -0800
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/flacencoder.h	2008-01-16 09:26:27.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include <qstring.h>
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/main.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/main.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/main.cpp	2006-08-21 19:06:03.000000000 -0700
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/main.cpp	2008-01-16 11:33:14.000000000 -0800
@@ -8,6 +8,7 @@
 #include <qregexp.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <stdlib.h>
 #include <unistd.h>
 
 #include <cdaudio.h>
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/metadata.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/metadata.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/metadata.cpp	2006-08-29 13:13:35.000000000 -0700
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/metadata.cpp	2008-01-16 11:35:24.000000000 -0800
@@ -1,4 +1,5 @@
 #include <iostream> 
+#include <stdlib.h>
 #include <qapplication.h>
 #include <qregexp.h> 
 #include <qdatetime.h>
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/metaioflacvorbiscomment.h mythplugins-0.20.2-new/mythmusic/mythmusic/metaioflacvorbiscomment.h
--- mythplugins-0.20.2-old/mythmusic/mythmusic/metaioflacvorbiscomment.h	2005-02-06 10:27:45.000000000 -0800
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-01-16 11:29:25.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include "metaio.h"
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 
 class MetaIOFLACVorbisComment : public MetaIO
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/metaiomp4.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/metaiomp4.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/metaiomp4.cpp	2006-08-21 19:06:03.000000000 -0700
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/metaiomp4.cpp	2008-01-16 09:23:17.000000000 -0800
@@ -1,6 +1,7 @@
 #include <iostream>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <sys/types.h>
 
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/metaiooggvorbiscomment.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/metaiooggvorbiscomment.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/metaiooggvorbiscomment.cpp	2005-02-06 10:27:45.000000000 -0800
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/metaiooggvorbiscomment.cpp	2008-01-16 12:34:54.000000000 -0800
@@ -1,4 +1,5 @@
 #include <iostream>
+#include <stdlib.h>
 #include <unistd.h>
 #include <sys/stat.h>
 #include <math.h>
diff -Naur mythplugins-0.20.2-old/mythmusic/mythmusic/playlist.cpp mythplugins-0.20.2-new/mythmusic/mythmusic/playlist.cpp
--- mythplugins-0.20.2-old/mythmusic/mythmusic/playlist.cpp	2006-10-22 06:10:37.000000000 -0700
+++ mythplugins-0.20.2-new/mythmusic/mythmusic/playlist.cpp	2008-01-16 12:28:12.000000000 -0800
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <unistd.h>
 #include <inttypes.h>
 #include <iostream>
diff -Naur mythplugins-0.20.2-old/mythnews/mythnews/mythnewsconfig.cpp mythplugins-0.20.2-new/mythnews/mythnews/mythnewsconfig.cpp
--- mythplugins-0.20.2-old/mythnews/mythnews/mythnewsconfig.cpp	2005-02-24 15:57:35.000000000 -0800
+++ mythplugins-0.20.2-new/mythnews/mythnews/mythnewsconfig.cpp	2008-01-16 12:39:46.000000000 -0800
@@ -21,6 +21,7 @@
 
 #include <qapplication.h>
 #include <iostream>
+#include <stdlib.h>
 
 #include <qptrlist.h>
 #include <qstring.h>
diff -Naur mythplugins-0.20.2-old/mythnews/mythnews/mythnews.cpp mythplugins-0.20.2-new/mythnews/mythnews/mythnews.cpp
--- mythplugins-0.20.2-old/mythnews/mythnews/mythnews.cpp	2006-05-16 02:21:34.000000000 -0700
+++ mythplugins-0.20.2-new/mythnews/mythnews/mythnews.cpp	2008-01-16 12:39:37.000000000 -0800
@@ -20,6 +20,7 @@
  * ============================================================ */
 
 #include <iostream>
+#include <stdlib.h>
 
 #include <qnetwork.h>
 #include <qdatetime.h>
diff -Naur mythplugins-0.20.2-old/mythphone/mythphone/h263.cpp mythplugins-0.20.2-new/mythphone/mythphone/h263.cpp
--- mythplugins-0.20.2-old/mythphone/mythphone/h263.cpp	2005-06-02 16:26:50.000000000 -0700
+++ mythplugins-0.20.2-new/mythphone/mythphone/h263.cpp	2008-01-16 12:44:31.000000000 -0800
@@ -8,6 +8,7 @@
   TODO:-
     These are non reentrant so need a mutex implemented, if they are to be used elsewhere in the Myth frontend
 */
+#include <stdlib.h>
 #include <qapplication.h>
 #include <qimage.h>
 
diff -Naur mythplugins-0.20.2-old/mythphone/mythphone/main.cpp mythplugins-0.20.2-new/mythphone/mythphone/main.cpp
--- mythplugins-0.20.2-old/mythphone/mythphone/main.cpp	2006-03-02 22:05:46.000000000 -0800
+++ mythplugins-0.20.2-new/mythphone/mythphone/main.cpp	2008-01-16 12:41:16.000000000 -0800
@@ -14,6 +14,7 @@
 #include <qdir.h>
 #include <qmutex.h>
 #include <qregexp.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <netdb.h>
 #include <netinet/in.h>
diff -Naur mythplugins-0.20.2-old/mythphone/mythphone/vxml.cpp mythplugins-0.20.2-new/mythphone/mythphone/vxml.cpp
--- mythplugins-0.20.2-old/mythphone/mythphone/vxml.cpp	2005-02-20 13:22:01.000000000 -0800
+++ mythplugins-0.20.2-new/mythphone/mythphone/vxml.cpp	2008-01-16 12:42:56.000000000 -0800
@@ -104,6 +104,7 @@
 #include <qdom.h>
 #include <qimage.h>
 #include <pthread.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <math.h>
 #include <iostream>
diff -Naur mythplugins-0.20.2-old/mythvideo/mythvideo/dbaccess.cpp mythplugins-0.20.2-new/mythvideo/mythvideo/dbaccess.cpp
--- mythplugins-0.20.2-old/mythvideo/mythvideo/dbaccess.cpp	2007-02-25 17:46:11.000000000 -0800
+++ mythplugins-0.20.2-new/mythvideo/mythvideo/dbaccess.cpp	2008-01-16 16:43:33.000000000 -0800
@@ -4,6 +4,7 @@
 #include "dbaccess.h"
 #include "cleanup.h"
 
+#include <algorithm>
 #include <vector>
 #include <map>
 
diff -Naur mythplugins-0.20.2-old/mythvideo/mythvideo/editmetadata.cpp mythplugins-0.20.2-new/mythvideo/mythvideo/editmetadata.cpp
--- mythplugins-0.20.2-old/mythvideo/mythvideo/editmetadata.cpp	2006-08-18 20:00:36.000000000 -0700
+++ mythplugins-0.20.2-new/mythvideo/mythvideo/editmetadata.cpp	2008-01-16 16:40:23.000000000 -0800
@@ -7,6 +7,8 @@
 
 */
 
+#include <algorithm>
+
 #include <mythtv/mythcontext.h>
 #include <mythtv/uitypes.h>
 
diff -Naur mythplugins-0.20.2-old/mythvideo/mythvideo/videodlg.cpp mythplugins-0.20.2-new/mythvideo/mythvideo/videodlg.cpp
--- mythplugins-0.20.2-old/mythvideo/mythvideo/videodlg.cpp	2006-09-25 11:40:51.000000000 -0700
+++ mythplugins-0.20.2-new/mythvideo/mythvideo/videodlg.cpp	2008-01-16 16:42:32.000000000 -0800
@@ -1,3 +1,5 @@
+#include <stdlib.h>
+
 #include <qapplication.h>
 
 #include <mythtv/xmlparse.h>
diff -Naur mythplugins-0.20.2-old/mythvideo/mythvideo/videogallery.cpp mythplugins-0.20.2-new/mythvideo/mythvideo/videogallery.cpp
--- mythplugins-0.20.2-old/mythvideo/mythvideo/videogallery.cpp	2006-08-20 01:08:23.000000000 -0700
+++ mythplugins-0.20.2-new/mythvideo/mythvideo/videogallery.cpp	2008-01-16 16:33:44.000000000 -0800
@@ -14,6 +14,7 @@
 #include <qapplication.h>
 #include <qstringlist.h>
 #include <qpixmap.h>
+#include <stdlib.h>
 #include <unistd.h>
 
 #include <cmath>
diff -Naur mythplugins-0.20.2-old/mythvideo/mythvideo/videomanager.cpp mythplugins-0.20.2-new/mythvideo/mythvideo/videomanager.cpp
--- mythplugins-0.20.2-old/mythvideo/mythvideo/videomanager.cpp	2006-09-10 00:41:22.000000000 -0700
+++ mythplugins-0.20.2-new/mythvideo/mythvideo/videomanager.cpp	2008-01-16 16:30:08.000000000 -0800
@@ -6,6 +6,7 @@
 #include <qprocess.h>
 #include <qpainter.h>
 
+#include <stdlib.h>
 #include <unistd.h>
 
 #include <mythtv/mythcontext.h>
diff -Naur mythplugins-0.20.2-old/mythvideo/mythvideo/videoselected.cpp mythplugins-0.20.2-new/mythvideo/mythvideo/videoselected.cpp
--- mythplugins-0.20.2-old/mythvideo/mythvideo/videoselected.cpp	2006-09-25 11:40:51.000000000 -0700
+++ mythplugins-0.20.2-new/mythvideo/mythvideo/videoselected.cpp	2008-01-16 16:41:31.000000000 -0800
@@ -15,6 +15,7 @@
 #include <qstringlist.h>
 #include <qpixmap.h>
 
+#include <stdlib.h>
 #include <unistd.h>
 
 #include <mythtv/mythcontext.h>
diff -Naur mythplugins-0.20.2-old/mythweather/mythweather/weather.cpp mythplugins-0.20.2-new/mythweather/mythweather/weather.cpp
--- mythplugins-0.20.2-old/mythweather/mythweather/weather.cpp	2006-03-31 22:42:32.000000000 -0800
+++ mythplugins-0.20.2-new/mythweather/mythweather/weather.cpp	2008-01-16 16:47:31.000000000 -0800
@@ -24,6 +24,7 @@
 #include <qstringlist.h>
 #include <qheader.h>
 #include <qpixmap.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <qurl.h>
 #include <qdir.h>
