diff -Naur libdvdread-1187-old/misc/dvdread-config.in libdvdread-1187-new/misc/dvdread-config.in
--- libdvdread-1187-old/misc/dvdread-config.in	2009-11-30 20:57:27.000000000 -0800
+++ libdvdread-1187-new/misc/dvdread-config.in	2009-11-30 20:57:50.000000000 -0800
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
