diff -Naur libdvdread-1168-old/misc/dvdread-config.in libdvdread-1168-new/misc/dvdread-config.in
--- libdvdread-1168-old/misc/dvdread-config.in	2009-03-02 22:47:58.000000000 -0800
+++ libdvdread-1168-new/misc/dvdread-config.in	2009-03-02 22:49:55.000000000 -0800
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
