diff -Naur libdvdread-1233-old/misc/dvdread-config.in libdvdread-1233-new/misc/dvdread-config.in
--- libdvdread-1233-old/misc/dvdread-config.in	2011-08-17 21:07:51.000000000 -0700
+++ libdvdread-1233-new/misc/dvdread-config.in	2011-08-17 21:08:43.000000000 -0700
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
