diff -Naur libdvdread-1181-old/misc/dvdread-config.in libdvdread-1181-new/misc/dvdread-config.in
--- libdvdread-1181-old/misc/dvdread-config.in	2009-09-12 17:04:39.000000000 -0700
+++ libdvdread-1181-new/misc/dvdread-config.in	2009-09-12 17:05:17.000000000 -0700
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
