diff -Naur libdvdread-4.1.3-old/misc/dvdread-config.in libdvdread-4.1.3-new/misc/dvdread-config.in
--- libdvdread-4.1.3-old/misc/dvdread-config.in	2008-06-02 02:24:35.000000000 -0700
+++ libdvdread-4.1.3-new/misc/dvdread-config.in	2008-09-10 21:24:32.000000000 -0700
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
