diff -Naur nfs-utils-1.2.4-old/configure.ac nfs-utils-1.2.4-new/configure.ac
--- nfs-utils-1.2.4-old/configure.ac	2011-06-30 06:00:42.000000000 -0700
+++ nfs-utils-1.2.4-new/configure.ac	2011-07-29 11:51:13.000000000 -0700
@@ -256,9 +256,6 @@
   dnl check for nfsidmap libraries and headers
   AC_LIBNFSIDMAP
 
-  dnl enable nfsidmap when its support by libnfsidmap
-  AM_CONDITIONAL(CONFIG_NFSIDMAP, [test "$enable_nfsidmap" = "yes"])
-
   dnl check for the keyutils libraries and headers
   AC_KEYUTILS
 
@@ -268,6 +265,9 @@
     AC_RPCSEC_VERSION
   fi
 fi
+dnl enable nfsidmap when its support by libnfsidmap
+AM_CONDITIONAL(CONFIG_NFSIDMAP, [test "$enable_nfsidmap" = "yes"])
+
 
 if test "$knfsd_cv_glibc2" = no; then
     AC_CHECK_LIB(bsd, daemon, [LIBBSD="-lbsd"])
