diff -Naur nfs-utils-1.1.0-old/utils/mount/mount.c nfs-utils-1.1.0-new/utils/mount/mount.c
--- nfs-utils-1.1.0-old/utils/mount/mount.c	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0-new/utils/mount/mount.c	2007-08-25 03:39:57.000000000 -0700
@@ -520,7 +520,7 @@
 				fprintf(stderr,
 					"%s: rpc.statd is not running but is "
 					"required for remote locking\n"
-					"   Either use \"-o nolocks\" to keep "
+					"   Either use \"-o nolock\" to keep "
 					"locks local, or start statd.\n",
 					progname);
 				exit(1);
