diff -Naur mythstream-0.17-old/libs/configelements.h mythstream-0.17-new/libs/configelements.h
--- mythstream-0.17-old/libs/configelements.h	2006-08-07 17:54:30.000000000 -0700
+++ mythstream-0.17-new/libs/configelements.h	2006-08-07 17:59:44.000000000 -0700
@@ -183,8 +183,8 @@
 
     void buildGroup(int boxes);
     void setStatus(Status stat);
-    QLabel* EditGroup::getLabel(QString name);
-    MythLineEdit* EditGroup::getLineEdit(QString name);
+    QLabel* getLabel(QString name);
+    MythLineEdit* getLineEdit(QString name);
 
     int boxCount;
 
diff -Naur mythstream-0.17-old/libs/streamconfig.h mythstream-0.17-new/libs/streamconfig.h
--- mythstream-0.17-old/libs/streamconfig.h	2006-08-07 17:54:30.000000000 -0700
+++ mythstream-0.17-new/libs/streamconfig.h	2006-08-07 17:59:44.000000000 -0700
@@ -93,7 +93,7 @@
     ConfigButton *overwriteButton;
 
     // Helpers
-    bool StorageGroup::getStorageValuesByName(ValueList& values, const QString& storageName);
+    bool getStorageValuesByName(ValueList& values, const QString& storageName);
 
     // GUI
     void buildGroup();
