diff -Naur alsa-driver-1.0.11-old/alsa-kernel/pci/ac97/ac97_patch.c alsa-driver-1.0.11-new/alsa-kernel/pci/ac97/ac97_patch.c
--- alsa-driver-1.0.11-old/alsa-kernel/pci/ac97/ac97_patch.c	2006-04-11 13:46:47.000000000 -0700
+++ alsa-driver-1.0.11-new/alsa-kernel/pci/ac97/ac97_patch.c	2006-08-19 06:09:51.000000000 -0700
@@ -2742,6 +2742,7 @@
  */
 int patch_vt1617a(struct snd_ac97 * ac97)
 {
+	snd_ac97_write_cache(ac97, 0x5c, 0x20);	/* bring analog power consumption to normal, like WinXP driver for EPIA SP */
 	ac97->ext_id |= AC97_EI_SPDIF;	/* force the detection of spdif */
 	ac97->rates[AC97_RATES_SPDIF] = SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000;
 	return 0;
