diff -Naur alsa-driver-1.0.9b-old/Makefile alsa-driver-1.0.9b-new/Makefile
--- alsa-driver-1.0.9b-old/Makefile	2005-03-22 07:37:25.000000000 -0800
+++ alsa-driver-1.0.9b-new/Makefile	2005-06-08 14:13:30.538625802 -0700
@@ -176,9 +176,9 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
-	-/sbin/depmod -a -b $(DESTDIR)/ $(SYSTEM_MAP_OPT) $(kaversion)
+	depmod -a -b $(DESTDIR)/ $(SYSTEM_MAP_OPT) $(kaversion)
 endif
 
 .PHONY: install-scripts
