diff -Naur xorg-server-1.7.4-old/hw/xfree86/common/xf86Config.c xorg-server-1.7.4-new/hw/xfree86/common/xf86Config.c
--- xorg-server-1.7.4-old/hw/xfree86/common/xf86Config.c	2010-01-07 18:32:04.000000000 -0800
+++ xorg-server-1.7.4-new/hw/xfree86/common/xf86Config.c	2010-01-07 18:32:46.000000000 -0800
@@ -100,22 +100,6 @@
 #endif
 
 static ModuleDefault ModuleDefaults[] = {
-    {.name = "extmod",   .toLoad = TRUE,    .load_opt=NULL},
-#ifdef DBE
-    {.name = "dbe",      .toLoad = TRUE,    .load_opt=NULL},
-#endif
-#ifdef GLXEXT
-    {.name = "glx",      .toLoad = TRUE,    .load_opt=NULL},
-#endif
-#ifdef XRECORD
-    {.name = "record",   .toLoad = TRUE,    .load_opt=NULL},
-#endif
-#ifdef XF86DRI
-    {.name = "dri",      .toLoad = TRUE,    .load_opt=NULL},
-#endif
-#ifdef DRI2
-    {.name = "dri2",     .toLoad = TRUE,    .load_opt=NULL},
-#endif
     {.name = NULL,       .toLoad = FALSE,   .load_opt=NULL}
 };
 
