diff -Naur mythplugins-9970-old/mythmusic/mythmusic/playlist.cpp mythplugins-9970-new/mythmusic/mythmusic/playlist.cpp
--- mythplugins-9970-old/mythmusic/mythmusic/playlist.cpp	2006-05-22 17:14:23.000000000 -0700
+++ mythplugins-9970-new/mythmusic/mythmusic/playlist.cpp	2006-05-22 17:15:40.000000000 -0700
@@ -329,10 +329,11 @@
     query.prepare("SELECT playlistid FROM musicplaylist "
                   "WHERE name != :DEFAULT  "
                   "AND name != :BACKUP  "
-                  "AND hostname = :HOST ORDER BY playlistid ;");
+                  "AND (hostname = :HOST OR hostname = :EMPTY) ORDER BY playlistid ;");
     query.bindValue(":DEFAULT", "default_playlist_storage");
     query.bindValue(":BACKUP", "backup_playlist_storage"); 
     query.bindValue(":HOST", my_host);
+    query.bindValue(":EMPTY", "");
 
     if (query.exec() && query.isActive() && query.size() > 0)
     {
@@ -470,9 +471,10 @@
     MSqlQuery query(MSqlQuery::InitCon());
     query.prepare("SELECT playlistid, name, songlist FROM "
                   "musicplaylist WHERE name = :NAME AND "
-                  "hostname = :HOST ;");
+                  "(hostname = :HOST OR hostname = :EMPTY) ;");
     query.bindValue(":NAME", a_name);
     query.bindValue(":HOST", a_host);
+    query.bindValue(":EMPTY", "");
 
     if (query.exec() && query.size() > 0)
     {
@@ -500,9 +502,10 @@
     MSqlQuery query(MSqlQuery::InitCon());
     query.prepare("SELECT playlistid, name, songlist FROM "
                   "musicplaylist WHERE playlistid = :ID AND "
-                  "hostname = :HOST ;");
+                  "(hostname = :HOST OR hostname = :EMPTY) ;");
     query.bindValue(":ID", id);
     query.bindValue(":HOST", a_host);
+    query.bindValue(":EMPTY", "");
 
     query.exec();
 
@@ -1234,13 +1237,13 @@
         Playlist *a_list;
         for(a_list = all_other_playlists->last(); a_list; a_list = all_other_playlists->prev())
         {
-            if (a_list->getID() * -1 == index)
+            if (a_list->getID() == index)
             {
                 return a_list->getName();   
             }
         }
     }
-    cerr << "playlist.o: Asked to getPlaylistName() with an index number I couldn't find" << endl ;
+    cerr << "playlist.o: Asked to getPlaylistName() with an index number (" << index << ") I couldn't find" << endl ;
     reference = true;
     return QObject::tr("Something is Wrong");
 }
