diff -Naur mythplugins-12782-old/mythmusic/mythmusic/flacdecoder.cpp mythplugins-12782-new/mythmusic/mythmusic/flacdecoder.cpp
--- mythplugins-12782-old/mythmusic/mythmusic/flacdecoder.cpp	2007-02-19 04:39:26.000000000 -0800
+++ mythplugins-12782-new/mythmusic/mythmusic/flacdecoder.cpp	2007-02-19 04:39:44.000000000 -0800
@@ -264,7 +264,7 @@
 
     decoder = decoder_new();
     decoder_set_md5_checking(decoder, false);
-    decoder_setup(decoder, flacread, flacseek, flactell, flaclength, \
+    decoder_setup(decoder, flacread, flacseek, flactell, flaclength,
                         flaceof, flacwrite, flacmetadata, flacerror, this);
 
     freq = 0;
diff -Naur mythplugins-12782-old/mythmusic/mythmusic/flacencoder.cpp mythplugins-12782-new/mythmusic/mythmusic/flacencoder.cpp
--- mythplugins-12782-old/mythmusic/mythmusic/flacencoder.cpp	2007-02-19 04:39:26.000000000 -0800
+++ mythplugins-12782-new/mythmusic/mythmusic/flacencoder.cpp	2007-02-19 04:39:44.000000000 -0800
@@ -43,15 +43,15 @@
     int rice_parameter_search_dist = 0;
 
     encoder = encoder_new();
-    encoder_setup(encoder, streamable_subset, \
-                  do_mid_side, loose_mid_side, \
-                  NUM_CHANNELS, bits_per_sample, \
-                  sample_rate, blocksize, \
-                  max_lpc_order, qlp_coeff_precision, \
-                  qlp_coeff_prec_search, do_escape_coding, \
-                  do_exhaustive_model_search, \
-                  min_residual_partition_order, \
-                  max_residual_partition_order, \
+    encoder_setup(encoder, streamable_subset,
+                  do_mid_side, loose_mid_side,
+                  NUM_CHANNELS, bits_per_sample,
+                  sample_rate, blocksize,
+                  max_lpc_order, qlp_coeff_precision,
+                  qlp_coeff_prec_search, do_escape_coding,
+                  do_exhaustive_model_search,
+                  min_residual_partition_order,
+                  max_residual_partition_order,
                   rice_parameter_search_dist);
 
 #if !defined(NEWFLAC)
diff -Naur mythplugins-12782-old/mythmusic/mythmusic/flacencoder.h mythplugins-12782-new/mythmusic/mythmusic/flacencoder.h
--- mythplugins-12782-old/mythmusic/mythmusic/flacencoder.h	2007-02-19 04:39:26.000000000 -0800
+++ mythplugins-12782-new/mythmusic/mythmusic/flacencoder.h	2007-02-19 04:39:44.000000000 -0800
@@ -18,21 +18,21 @@
             min_residual_partition_order, max_residual_partition_order, \
             rice_parameter_search_dist) \
             { \
-              FLAC__file_encoder_set_streamable_subset(enc, streamable_subset) \
-              FLAC__file_encoder_set_do_mid_side_stereo(enc, do_mid_side_stereo) \
-              FLAC__file_encoder_set_loose_mid_side_stereo(enc, loose_mid_side_stereo) \
-              FLAC__file_encoder_set_channels(enc, channels) \
-              FLAC__file_encoder_set_bits_per_sample(enc, bits_per_sample) \
-              FLAC__file_encoder_set_sample_rate(enc, sample_rate) \
-              FLAC__file_encoder_set_blocksize(enc, blocksize) \
-              FLAC__file_encoder_set_max_lpc_order(enc, max_lpc_order) \
-              FLAC__file_encoder_set_qlp_coeff_precision(enc, qlp_coeff_precision) \
-              FLAC__file_encoder_set_do_qlp_coeff_prec_search(enc, do_qlp_coeff_prec_search) \
-              FLAC__file_encoder_set_do_escape_coding(enc, do_escape_coding) \
-              FLAC__file_encoder_set_do_exhaustive_model_search(enc, do_exhaustive_model_search) \
-              FLAC__file_encoder_set_min_residual_partition_order(enc, min_residual_partition_order) \
-              FLAC__file_encoder_set_max_residual_partition_order(enc, max_residual_partition_order) \
-              FLAC__file_encoder_set_rice_parameter_search_dist(enc, rice_parameter_search_dist) \
+              FLAC__file_encoder_set_streamable_subset(enc, streamable_subset); \
+              FLAC__file_encoder_set_do_mid_side_stereo(enc, do_mid_side_stereo); \
+              FLAC__file_encoder_set_loose_mid_side_stereo(enc, loose_mid_side_stereo); \
+              FLAC__file_encoder_set_channels(enc, channels); \
+              FLAC__file_encoder_set_bits_per_sample(enc, bits_per_sample); \
+              FLAC__file_encoder_set_sample_rate(enc, sample_rate); \
+              FLAC__file_encoder_set_blocksize(enc, blocksize); \
+              FLAC__file_encoder_set_max_lpc_order(enc, max_lpc_order); \
+              FLAC__file_encoder_set_qlp_coeff_precision(enc, qlp_coeff_precision); \
+              FLAC__file_encoder_set_do_qlp_coeff_prec_search(enc, do_qlp_coeff_prec_search); \
+              FLAC__file_encoder_set_do_escape_coding(enc, do_escape_coding); \
+              FLAC__file_encoder_set_do_exhaustive_model_search(enc, do_exhaustive_model_search); \
+              FLAC__file_encoder_set_min_residual_partition_order(enc, min_residual_partition_order); \
+              FLAC__file_encoder_set_max_residual_partition_order(enc, max_residual_partition_order); \
+              FLAC__file_encoder_set_rice_parameter_search_dist(enc, rice_parameter_search_dist); \
             }
   #define encoder_finish(enc) FLAC__file_encoder_finish(enc)
   #define encoder_delete(enc) FLAC__file_encoder_delete(enc)
