diff -Naur libdrm-2.4.60-old/include/drm/drm.h libdrm-2.4.60-new/include/drm/drm.h
--- libdrm-2.4.60-old/include/drm/drm.h	2014-06-10 06:14:57.000000000 -0700
+++ libdrm-2.4.60-new/include/drm/drm.h	2015-07-17 15:11:54.006999517 -0700
@@ -641,7 +641,11 @@
 	__u64 value;
 };
 
+#ifdef O_CLOEXEC
 #define DRM_CLOEXEC O_CLOEXEC
+#else
+#define DRM_CLOEXEC 0
+#endif
 struct drm_prime_handle {
 	__u32 handle;
 
