diff -Naur mysql-5.0.18-old/extra/Makefile.am mysql-5.0.18-new/extra/Makefile.am
--- mysql-5.0.18-old/extra/Makefile.am	2005-12-21 11:39:55.000000000 -0800
+++ mysql-5.0.18-new/extra/Makefile.am	2006-01-04 17:23:06.000000000 -0800
@@ -30,7 +30,7 @@
 
 # This will build mysqld_error.h and sql_state.h
 $(top_builddir)/include/mysqld_error.h: comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.18-old/extra/Makefile.in mysql-5.0.18-new/extra/Makefile.in
--- mysql-5.0.18-old/extra/Makefile.in	2005-12-21 11:41:16.000000000 -0800
+++ mysql-5.0.18-new/extra/Makefile.in	2006-01-04 17:22:14.000000000 -0800
@@ -871,7 +871,7 @@
 
 # This will build mysqld_error.h and sql_state.h
 $(top_builddir)/include/mysqld_error.h: comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.0.18-old/sql/Makefile.am mysql-5.0.18-new/sql/Makefile.am
--- mysql-5.0.18-old/sql/Makefile.am	2005-12-21 11:39:32.000000000 -0800
+++ mysql-5.0.18-new/sql/Makefile.am	2006-01-04 17:24:01.000000000 -0800
@@ -149,7 +149,7 @@
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
 lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+		mysql-gen_lex_hash > $@
 
 # For testing of udf_example.so;  Works on platforms with gcc
 # (This is not part of our build process but only provided as an example)
diff -Naur mysql-5.0.18-old/sql/Makefile.in mysql-5.0.18-new/sql/Makefile.in
--- mysql-5.0.18-old/sql/Makefile.in	2005-12-21 11:41:41.000000000 -0800
+++ mysql-5.0.18-new/sql/Makefile.in	2006-01-04 17:23:48.000000000 -0800
@@ -1254,7 +1254,7 @@
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
 lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+		mysql-gen_lex_hash > $@
 
 # For testing of udf_example.so;  Works on platforms with gcc
 # (This is not part of our build process but only provided as an example)
