diff -Naur xc-old/programs/Xserver/render/mitri.c xc-new/programs/Xserver/render/mitri.c
--- xc-old/programs/Xserver/render/mitri.c	2004-08-06 16:42:10.000000000 -0700
+++ xc-new/programs/Xserver/render/mitri.c	2006-05-03 05:32:53.000000000 -0700
@@ -141,7 +141,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     for (tri = tris; npoint >= 3; npoint--, points++, tri++)
@@ -173,7 +173,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     first = points++;
