diff -Naur font-misc-misc-1.0.0-old/configure.ac font-misc-misc-1.0.0-new/configure.ac
--- font-misc-misc-1.0.0-old/configure.ac	2005-12-14 16:24:25.000000000 -0800
+++ font-misc-misc-1.0.0-new/configure.ac	2008-01-22 18:27:52.000000000 -0800
@@ -27,6 +27,7 @@
 AM_MAINTAINER_MODE
 
 AC_PROG_INSTALL
+PKG_PROG_PKG_CONFIG
 
 AC_PATH_PROGS(PERL, perl perl5)
 AC_PATH_PROG(BDFTOPCF, bdftopcf)
@@ -45,7 +46,8 @@
 AC_SUBST(FONTDIR)
 
 PKG_CHECK_MODULES(MAPS, [fontutil])
-MAPFILES_PATH=$(pkg-config --variable=mapdir fontutil)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+MAPFILES_PATH="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=mapdir fontutil)"
 AC_SUBST(MAPFILES_PATH)
 
 XORG_FONT_CHECK_ISO8859_1
