diff -Naur util-linux-2.12r-old/disk-utils/mkfs.cramfs.c util-linux-2.12r-new/disk-utils/mkfs.cramfs.c
--- util-linux-2.12r-old/disk-utils/mkfs.cramfs.c	2004-12-11 06:56:01.000000000 -0800
+++ util-linux-2.12r-new/disk-utils/mkfs.cramfs.c	2006-08-18 13:54:34.000000000 -0700
@@ -472,7 +472,7 @@
  * entries, using a stack to remember the directories
  * we've seen.
  */
-#define MAXENTRIES (100)
+#define MAXENTRIES (255)
 static unsigned int write_directory_structure(struct entry *entry, char *base, unsigned int offset)
 {
 	int stack_entries = 0;
