diff -Naur MPlayer-1.0pre7try2-old/configure MPlayer-1.0pre7try2-new/configure
--- MPlayer-1.0pre7try2-old/configure	2005-12-30 11:09:05.000000000 -0800
+++ MPlayer-1.0pre7try2-new/configure	2005-12-30 11:19:05.000000000 -0800
@@ -508,8 +508,7 @@
     morphos) system_name=MorphOS ;;
     mingw32msvc) system_name=MINGW32 ;;
   esac
-  # We need to convert underscores so that values like k6-2 and pentium-mmx can be passed
-  host_arch=`echo $_target | cut -d '-' -f 1 | tr '_' '-'`
+  host_arch=`echo $_target | cut -d '-' -f 1`
 fi
 
 echo "Detected operating system: $system_name"
@@ -2558,12 +2557,7 @@
 
 echocheck "word size"
 _mp_wordsize="#undef MP_WORDSIZE"
-cat > $TMPC << EOF
-#include <stdio.h>
-#include <sys/types.h>
-int main(void) { printf("%d\n", sizeof(size_t)*8); return 0; }
-EOF
-cc_check && _wordsize=`$TMPO` && _mp_wordsize="#define MP_WORDSIZE $_wordsize"
+_wordsize=32 &&  _mp_wordsize="#define MP_WORDSIZE $_wordsize"
 echores "$_wordsize"
 
 
@@ -2708,37 +2702,7 @@
 _def_threads='#undef HAVE_THREADS'
 
 echocheck "pthread"
-cat > $TMPC << EOF
-#include <pthread.h>
-void* func(void *arg) { return arg; }
-int main(void) { pthread_t tid; return pthread_create (&tid, 0, func, 0) == 0 ? 0 : 1; }
-EOF
-if mingw32 ; then
-  _ld_pthread=''
-elif hpux ; then
-  _ld_pthread=''
-elif ( cc_check && $TMPO ) ; then              # QNX
-  _ld_pthread=' ' # _ld_pthread='' would disable pthreads, but the test worked
-elif ( cc_check -lpthread && $TMPO ) ; then
-  _ld_pthread='-lpthread'
-elif ( cc_check -pthread && $TMPO ) ; then
-  _ld_pthread='-pthread'
-else
-  if test "$_ld_static" ; then
-    # for crosscompilation, we cannot execute the program, be happy if we can link statically
-    if ( cc_check -lpthread ) ; then
-      _ld_pthread='-lpthread'
-    elif ( cc_check -pthread ) ; then
-     _ld_pthread='-pthread'
-    else
-      echores "no static pthread found (v4l, vo_gl, ao_alsa, ao_nas, ao_macosx, Win32 loader disabled)"
-      _ld_pthread='' ; _nas=no ; _tv_v4l=no ; _macosx=no ; _win32=no
-    fi
-  else
-    echores "no (v4l, vo_gl, ao_alsa, ao_nas, ao_macosx, win32 loader disabled)"
-    _ld_pthread='' ; _nas=no ; _tv_v4l=no ; _macosx=no ; _win32=no
-  fi
-fi
+_ld_pthread='-lpthread'
 if test "$_ld_pthread" != '' ; then
   echores "yes (using $_ld_pthread)"
   _pthreads='yes'
@@ -4632,41 +4596,12 @@
 
 
 echocheck "ALSA audio"
-if test "$_alsa" != no ; then
-  _alsa=no
-  cat > $TMPC << EOF
-#include <sys/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==5)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.5.x'
-
-  cat > $TMPC << EOF
-#include <sys/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==9)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.9.x-sys'
-  cat > $TMPC << EOF
-#include <alsa/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==9)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.9.x-alsa'
-
-  cat > $TMPC << EOF
-#include <sys/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==1 && SND_LIB_MINOR==0)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='1.0.x-sys'
-  cat > $TMPC << EOF
-#include <alsa/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==1 && SND_LIB_MINOR==0)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='1.0.x-alsa'
-fi
 _def_alsa5='#undef HAVE_ALSA5'
 _def_alsa9='#undef HAVE_ALSA9'
 _def_alsa1x='#undef HAVE_ALSA1X'
 _def_sys_asoundlib_h='#undef HAVE_SYS_ASOUNDLIB_H'
 _def_alsa_asoundlib_h='#undef HAVE_ALSA_ASOUNDLIB_H'
+_alsaver='1.0.x-alsa'
 if test "$_alsaver" ; then
   if test "$_alsaver" = '0.5.x' ; then
     _aosrc="$_aosrc ao_alsa5.c"
