diff -Naur MPlayer-1.0pre7try2-old/vobsub.c MPlayer-1.0pre7try2-new/vobsub.c
--- MPlayer-1.0pre7try2-old/vobsub.c	2004-12-25 04:08:33.000000000 -0800
+++ MPlayer-1.0pre7try2-new/vobsub.c	2006-06-20 07:48:54.000000000 -0700
@@ -227,7 +227,7 @@
 /**********************************************************************/
 
 static ssize_t
-getline(char **lineptr, size_t *n, rar_stream_t *stream)
+vobsub_getline(char **lineptr, size_t *n, rar_stream_t *stream)
 {
     size_t res = 0;
     int c;
@@ -949,7 +949,7 @@
 	size_t line_reserve = 0;
 	char *line = NULL;
     do {
-	line_size = getline(&line, &line_reserve, fd);
+	line_size = vobsub_getline(&line, &line_reserve, fd);
 	if (line_size < 0) {
 	    break;
 	}
