diff -Naur gcc-4.1.0-old/gcc/config/i386/linux64.h gcc-4.1.0-new/gcc/config/i386/linux64.h
--- gcc-4.1.0-old/gcc/config/i386/linux64.h	2005-08-10 10:53:01.000000000 -0700
+++ gcc-4.1.0-new/gcc/config/i386/linux64.h	2006-03-27 21:32:15.000000000 -0800
@@ -61,7 +61,7 @@
     %{!static: \
       %{rdynamic:-export-dynamic} \
       %{m32:%{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}} \
-      %{!m32:%{!dynamic-linker:-dynamic-linker /lib64/ld-linux-x86-64.so.2}}} \
+      %{!m32:%{!dynamic-linker:-dynamic-linker /lib/ld-linux-x86-64.so.2}}} \
     %{static:-static}}"
 
 /* Similar to standard Linux, but adding -ffast-math support.  */
diff -Naur gcc-4.1.0-old/gcc/config/mips/linux64.h gcc-4.1.0-new/gcc/config/mips/linux64.h
--- gcc-4.1.0-old/gcc/config/mips/linux64.h	2005-06-24 18:22:41.000000000 -0700
+++ gcc-4.1.0-new/gcc/config/mips/linux64.h	2006-03-27 21:32:15.000000000 -0800
@@ -47,8 +47,8 @@
       %{!static: \
         %{rdynamic:-export-dynamic} \
         %{!dynamic-linker: \
-	  %{mabi=n32: -dynamic-linker /lib32/ld.so.1} \
-	  %{mabi=64: -dynamic-linker /lib64/ld.so.1} \
+	  %{mabi=n32: -dynamic-linker /lib/ld.so.1} \
+	  %{mabi=64: -dynamic-linker /lib/ld.so.1} \
 	  %{mabi=32: -dynamic-linker /lib/ld.so.1}}} \
       %{static:-static}}} \
 %{mabi=n32:-melf32%{EB:b}%{EL:l}tsmipn32} \
diff -Naur gcc-4.1.0-old/gcc/config/rs6000/linux64.h gcc-4.1.0-new/gcc/config/rs6000/linux64.h
--- gcc-4.1.0-old/gcc/config/rs6000/linux64.h	2006-02-10 12:58:33.000000000 -0800
+++ gcc-4.1.0-new/gcc/config/rs6000/linux64.h	2006-03-27 21:32:15.000000000 -0800
@@ -343,7 +343,7 @@
 
 #define LINK_OS_LINUX_SPEC64 "-m elf64ppc %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
-  %{!dynamic-linker:-dynamic-linker /lib64/ld64.so.1}}}"
+  %{!dynamic-linker:-dynamic-linker /lib/ld64.so.1}}}"
 
 #undef  TOC_SECTION_ASM_OP
 #define TOC_SECTION_ASM_OP \
diff -Naur gcc-4.1.0-old/gcc/config/sparc/linux64.h gcc-4.1.0-new/gcc/config/sparc/linux64.h
--- gcc-4.1.0-old/gcc/config/sparc/linux64.h	2006-02-11 00:38:51.000000000 -0800
+++ gcc-4.1.0-new/gcc/config/sparc/linux64.h	2006-03-27 21:32:15.000000000 -0800
@@ -176,7 +176,7 @@
     %{!ibcs: \
       %{!static: \
         %{rdynamic:-export-dynamic} \
-        %{!dynamic-linker:-dynamic-linker /lib64/ld-linux.so.2}} \
+        %{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}} \
         %{static:-static}}} \
 "
 
@@ -257,7 +257,7 @@
     %{!ibcs: \
       %{!static: \
         %{rdynamic:-export-dynamic} \
-        %{!dynamic-linker:-dynamic-linker /lib64/ld-linux.so.2}} \
+        %{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}} \
         %{static:-static}}} \
 %{mlittle-endian:-EL} \
 %{!mno-relax:%{!r:-relax}} \
