diff -Naur NVIDIA-Linux-x86-1.0-8756-pkg1-old/usr/src/nv/nv.c NVIDIA-Linux-x86-1.0-8756-pkg1-new/usr/src/nv/nv.c
--- NVIDIA-Linux-x86-1.0-8756-pkg1-old/usr/src/nv/nv.c	2006-03-29 15:02:20.000000000 -0800
+++ NVIDIA-Linux-x86-1.0-8756-pkg1-new/usr/src/nv/nv.c	2006-05-25 07:27:26.000000000 -0700
@@ -1794,7 +1794,7 @@
         }
 
         status = request_irq(nv->interrupt_line, nv_kern_isr,
-                             SA_INTERRUPT | SA_SHIRQ, "nvidia",
+                             SA_SHIRQ, "nvidia",
                              (void *) nvl);
         if (status != 0)
         {
