diff -Naur mythplugins-15447-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-15447-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-15447-old/mythmusic/mythmusic/aacdecoder.cpp	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/aacdecoder.cpp	2008-01-16 21:32:46.000000000 -0800
@@ -16,6 +16,7 @@
 
 #include <iostream>
 #include <string>
+#include <stdlib.h>
 #include <qobject.h>
 #include <qiodevice.h>
 #include <qfile.h>
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/databasebox.cpp mythplugins-15447-new/mythmusic/mythmusic/databasebox.cpp
--- mythplugins-15447-old/mythmusic/mythmusic/databasebox.cpp	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/databasebox.cpp	2008-01-16 21:32:46.000000000 -0800
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <iostream>
 using namespace std;
 
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/flacdecoder.h mythplugins-15447-new/mythmusic/mythmusic/flacdecoder.h
--- mythplugins-15447-old/mythmusic/mythmusic/flacdecoder.h	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/flacdecoder.h	2008-01-16 21:32:46.000000000 -0800
@@ -1,7 +1,7 @@
 #ifndef FLACDECODER_H_
 #define FLACDECODER_H_
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/flacencoder.h mythplugins-15447-new/mythmusic/mythmusic/flacencoder.h
--- mythplugins-15447-old/mythmusic/mythmusic/flacencoder.h	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/flacencoder.h	2008-01-16 21:32:46.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include <qstring.h>
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/metaioflacvorbiscomment.h mythplugins-15447-new/mythmusic/mythmusic/metaioflacvorbiscomment.h
--- mythplugins-15447-old/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-01-16 21:32:57.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include "metaio.h"
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 
 class MetaIOFLACVorbisComment : public MetaIO
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/metaiomp4.cpp mythplugins-15447-new/mythmusic/mythmusic/metaiomp4.cpp
--- mythplugins-15447-old/mythmusic/mythmusic/metaiomp4.cpp	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/metaiomp4.cpp	2008-01-16 21:32:57.000000000 -0800
@@ -1,6 +1,7 @@
 #include <iostream>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <sys/types.h>
 
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/metaiooggvorbiscomment.cpp mythplugins-15447-new/mythmusic/mythmusic/metaiooggvorbiscomment.cpp
--- mythplugins-15447-old/mythmusic/mythmusic/metaiooggvorbiscomment.cpp	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/metaiooggvorbiscomment.cpp	2008-01-16 21:39:09.000000000 -0800
@@ -1,4 +1,5 @@
 // POSIX headers
+#include <stdlib.h>
 #include <unistd.h>
 #include <sys/stat.h>
 
diff -Naur mythplugins-15447-old/mythmusic/mythmusic/playlist.cpp mythplugins-15447-new/mythmusic/mythmusic/playlist.cpp
--- mythplugins-15447-old/mythmusic/mythmusic/playlist.cpp	2008-01-16 20:03:36.000000000 -0800
+++ mythplugins-15447-new/mythmusic/mythmusic/playlist.cpp	2008-01-16 21:32:57.000000000 -0800
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <unistd.h>
 #include <inttypes.h>
 #include <iostream>
diff -Naur mythplugins-15447-old/mythphone/mythphone/h263.cpp mythplugins-15447-new/mythphone/mythphone/h263.cpp
--- mythplugins-15447-old/mythphone/mythphone/h263.cpp	2008-01-16 20:04:05.000000000 -0800
+++ mythplugins-15447-new/mythphone/mythphone/h263.cpp	2008-01-16 21:32:57.000000000 -0800
@@ -8,6 +8,7 @@
   TODO:-
     These are non reentrant so need a mutex implemented, if they are to be used elsewhere in the Myth frontend
 */
+#include <stdlib.h>
 #include <qapplication.h>
 #include <qimage.h>
 
diff -Naur mythplugins-15447-old/mythphone/mythphone/main.cpp mythplugins-15447-new/mythphone/mythphone/main.cpp
--- mythplugins-15447-old/mythphone/mythphone/main.cpp	2008-01-16 20:04:05.000000000 -0800
+++ mythplugins-15447-new/mythphone/mythphone/main.cpp	2008-01-16 21:32:57.000000000 -0800
@@ -15,6 +15,7 @@
 #include <qdir.h>
 #include <qmutex.h>
 #include <qregexp.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <netdb.h>
 #include <netinet/in.h>
diff -Naur mythplugins-15447-old/mythphone/mythphone/vxml.cpp mythplugins-15447-new/mythphone/mythphone/vxml.cpp
--- mythplugins-15447-old/mythphone/mythphone/vxml.cpp	2008-01-16 20:04:05.000000000 -0800
+++ mythplugins-15447-new/mythphone/mythphone/vxml.cpp	2008-01-16 21:32:57.000000000 -0800
@@ -104,6 +104,7 @@
 #include <qdom.h>
 #include <qimage.h>
 #include <pthread.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <math.h>
 #include <iostream>
diff -Naur mythplugins-15447-old/mythvideo/mythvideo/editmetadata.cpp mythplugins-15447-new/mythvideo/mythvideo/editmetadata.cpp
--- mythplugins-15447-old/mythvideo/mythvideo/editmetadata.cpp	2008-01-16 20:04:04.000000000 -0800
+++ mythplugins-15447-new/mythvideo/mythvideo/editmetadata.cpp	2008-01-16 21:33:04.000000000 -0800
@@ -7,6 +7,8 @@
 
 */
 
+#include <algorithm>
+
 #include <mythtv/mythcontext.h>
 #include <mythtv/uitypes.h>
 
diff -Naur mythplugins-15447-old/mythvideo/mythvideo/videogallery.cpp mythplugins-15447-new/mythvideo/mythvideo/videogallery.cpp
--- mythplugins-15447-old/mythvideo/mythvideo/videogallery.cpp	2008-01-16 20:04:04.000000000 -0800
+++ mythplugins-15447-new/mythvideo/mythvideo/videogallery.cpp	2008-01-16 21:33:04.000000000 -0800
@@ -14,6 +14,7 @@
 #include <qapplication.h>
 #include <qstringlist.h>
 #include <qpixmap.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <cstdlib>
 
diff -Naur mythplugins-15447-old/mythvideo/mythvideo/videomanager.cpp mythplugins-15447-new/mythvideo/mythvideo/videomanager.cpp
--- mythplugins-15447-old/mythvideo/mythvideo/videomanager.cpp	2008-01-16 20:04:04.000000000 -0800
+++ mythplugins-15447-new/mythvideo/mythvideo/videomanager.cpp	2008-01-16 21:33:04.000000000 -0800
@@ -7,6 +7,7 @@
 #include <qprocess.h>
 #include <qpainter.h>
 
+#include <stdlib.h>
 #include <unistd.h>
 #include <cstdlib>
 
diff -Naur mythplugins-15447-old/mythvideo/mythvideo/videoselected.cpp mythplugins-15447-new/mythvideo/mythvideo/videoselected.cpp
--- mythplugins-15447-old/mythvideo/mythvideo/videoselected.cpp	2008-01-16 20:04:04.000000000 -0800
+++ mythplugins-15447-new/mythvideo/mythvideo/videoselected.cpp	2008-01-16 21:33:04.000000000 -0800
@@ -15,6 +15,7 @@
 #include <qstringlist.h>
 #include <qpixmap.h>
 
+#include <stdlib.h>
 #include <unistd.h>
 #include <cstdlib>
 
