diff -Naur mythplugins-16133-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-16133-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-16133-old/mythmusic/mythmusic/aacdecoder.cpp	2008-02-18 08:14:27.000000000 -0800
+++ mythplugins-16133-new/mythmusic/mythmusic/aacdecoder.cpp	2008-02-18 08:23:56.000000000 -0800
@@ -14,6 +14,7 @@
 #define FAAD_MODIFIED
 #endif
 
+#include <cstdlib>
 #include <iostream>
 #include <string>
 #include <qobject.h>
diff -Naur mythplugins-16133-old/mythmusic/mythmusic/flacdecoder.h mythplugins-16133-new/mythmusic/mythmusic/flacdecoder.h
--- mythplugins-16133-old/mythmusic/mythmusic/flacdecoder.h	2008-02-18 08:14:27.000000000 -0800
+++ mythplugins-16133-new/mythmusic/mythmusic/flacdecoder.h	2008-02-18 08:19:27.000000000 -0800
@@ -1,7 +1,7 @@
 #ifndef FLACDECODER_H_
 #define FLACDECODER_H_
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-16133-old/mythmusic/mythmusic/flacencoder.h mythplugins-16133-new/mythmusic/mythmusic/flacencoder.h
--- mythplugins-16133-old/mythmusic/mythmusic/flacencoder.h	2008-02-18 08:14:27.000000000 -0800
+++ mythplugins-16133-new/mythmusic/mythmusic/flacencoder.h	2008-02-18 08:19:27.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include <qstring.h>
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-16133-old/mythmusic/mythmusic/metaioflacvorbiscomment.h mythplugins-16133-new/mythmusic/mythmusic/metaioflacvorbiscomment.h
--- mythplugins-16133-old/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-02-18 08:14:27.000000000 -0800
+++ mythplugins-16133-new/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-02-18 08:19:27.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include "metaio.h"
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 
 class MetaIOFLACVorbisComment : public MetaIO
diff -Naur mythplugins-16133-old/mythmusic/mythmusic/metaiomp4.cpp mythplugins-16133-new/mythmusic/mythmusic/metaiomp4.cpp
--- mythplugins-16133-old/mythmusic/mythmusic/metaiomp4.cpp	2008-02-18 08:14:27.000000000 -0800
+++ mythplugins-16133-new/mythmusic/mythmusic/metaiomp4.cpp	2008-02-18 08:23:19.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <iostream>
 #include <sys/stat.h>
 #include <fcntl.h>
