diff -Naur NVIDIA-Linux-x86-177.80-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-177.80-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-177.80-pkg1-old/usr/src/nv/conftest.sh	2008-10-01 15:13:04.000000000 -0700
+++ NVIDIA-Linux-x86-177.80-pkg1-new/usr/src/nv/conftest.sh	2008-10-08 14:02:37.000000000 -0700
@@ -1110,7 +1110,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
