diff -Naur myththemes-12156-old/cpsvndir myththemes-12156-new/cpsvndir
--- myththemes-12156-old/cpsvndir	2006-11-30 09:11:33.000000000 -0800
+++ myththemes-12156-new/cpsvndir	2006-11-30 09:12:03.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
