diff -Naur mythplugins-0.21-21625-old/mythbrowser/mythbrowser/mythbrowser.pro mythplugins-0.21-21625-new/mythbrowser/mythbrowser/mythbrowser.pro
--- mythplugins-0.21-21625-old/mythbrowser/mythbrowser/mythbrowser.pro	2009-09-04 19:52:32.000000000 -0700
+++ mythplugins-0.21-21625-new/mythbrowser/mythbrowser/mythbrowser.pro	2010-05-28 15:56:10.000000000 -0700
@@ -17,7 +17,7 @@
 
 INSTALLS += installimages
 
-LIBS += -L$$SYSTEM(kde-config --expandvars --install lib) -lkhtml
+LIBS += -L$$SYSTEM(kde-config --expandvars --install lib) -lkhtml -lkdecore
 
 # Input
 HEADERS += webpage.h tabview.h
