diff -Naur mythplugins-0.21-25121-old/mythbrowser/mythbrowser/mythbrowser.pro mythplugins-0.21-25121-new/mythbrowser/mythbrowser/mythbrowser.pro
--- mythplugins-0.21-25121-old/mythbrowser/mythbrowser/mythbrowser.pro	2010-06-20 07:19:01.000000000 -0700
+++ mythplugins-0.21-25121-new/mythbrowser/mythbrowser/mythbrowser.pro	2010-06-20 07:20:47.000000000 -0700
@@ -17,7 +17,7 @@
 
 INSTALLS += installimages
 
-LIBS += -L$$SYSTEM(kde-config --expandvars --install lib) -lkhtml
+LIBS += -L$$SYSTEM(kde-config --expandvars --install lib) -lkhtml -lkdecore
 
 # Input
 HEADERS += webpage.h tabview.h
