diff -Naur mythplugins-0.21-17722-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-0.21-17722-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-0.21-17722-old/mythmusic/mythmusic/aacdecoder.cpp	2008-07-06 08:47:36.000000000 -0700
+++ mythplugins-0.21-17722-new/mythmusic/mythmusic/aacdecoder.cpp	2008-07-06 08:50:27.000000000 -0700
@@ -16,6 +16,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstdlib>
 #include <qobject.h>
 #include <qiodevice.h>
 #include <qfile.h>
diff -Naur mythplugins-0.21-17722-old/mythmusic/mythmusic/metaiomp4.cpp mythplugins-0.21-17722-new/mythmusic/mythmusic/metaiomp4.cpp
--- mythplugins-0.21-17722-old/mythmusic/mythmusic/metaiomp4.cpp	2008-07-06 08:47:36.000000000 -0700
+++ mythplugins-0.21-17722-new/mythmusic/mythmusic/metaiomp4.cpp	2008-07-06 08:50:27.000000000 -0700
@@ -1,4 +1,5 @@
 #include <iostream>
+#include <cstdlib>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
