diff -Naur lirc-20090116Z-old/daemons/hw_livedrive_midi.c lirc-20090116Z-new/daemons/hw_livedrive_midi.c
--- lirc-20090116Z-old/daemons/hw_livedrive_midi.c	2009-01-18 16:45:45.000000000 -0800
+++ lirc-20090116Z-new/daemons/hw_livedrive_midi.c	2009-01-18 19:06:31.000000000 -0800
@@ -19,6 +19,8 @@
  *     USA. 
  */
 
+#include <sys/types.h>
+
 #include "hardware.h"
 #include "ir_remote.h"
 #include "hw_livedrive_common.h"
diff -Naur lirc-20090116Z-old/daemons/hw_livedrive_seq.c lirc-20090116Z-new/daemons/hw_livedrive_seq.c
--- lirc-20090116Z-old/daemons/hw_livedrive_seq.c	2009-01-18 16:45:45.000000000 -0800
+++ lirc-20090116Z-new/daemons/hw_livedrive_seq.c	2009-01-18 19:06:36.000000000 -0800
@@ -19,6 +19,8 @@
  *     USA. 
  */
 
+#include <sys/types.h>
+
 #include "hardware.h"
 #include "ir_remote.h"
 #include "hw_livedrive_common.h"
