diff -Naur lirc-0.8.5-old/configure.ac lirc-0.8.5-new/configure.ac
--- lirc-0.8.5-old/configure.ac	2009-05-21 16:47:30.000000000 -0700
+++ lirc-0.8.5-new/configure.ac	2009-05-21 16:48:30.000000000 -0700
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
