diff -Naur lirc-0.8.2-old/drivers/lirc_i2c/lirc_i2c.c lirc-0.8.2-new/drivers/lirc_i2c/lirc_i2c.c
--- lirc-0.8.2-old/drivers/lirc_i2c/lirc_i2c.c	2007-02-12 22:45:15.000000000 -0800
+++ lirc-0.8.2-new/drivers/lirc_i2c/lirc_i2c.c	2007-07-25 14:36:37.000000000 -0700
@@ -406,12 +406,12 @@
 	switch(addr)
 	{
 	case 0x64:
-		strcpy(ir->c.name,"Pixelview IR");
+		snprintf(ir->c.name, I2C_NAME_SIZE, "Pixelview IR");
 		ir->l.code_length = 8;
 		ir->l.add_to_buf=add_to_buf_pixelview;
 		break;
 	case 0x4b:
-		strcpy(ir->c.name,"PV951 IR");
+		snprintf(ir->c.name, I2C_NAME_SIZE, "PV951 IR");
 		ir->l.code_length = 32;
 		ir->l.add_to_buf=add_to_buf_pv951;
 		break;
@@ -420,12 +420,12 @@
 		   Hauppauge cards, but the data flow is different, so we need
 		   to deal with it by its own.
 		 */
-		strcpy(ir->c.name,"Hauppauge IR (PVR150)");
+		snprintf(ir->c.name, I2C_NAME_SIZE, "Hauppauge IR PVR150");
 		ir->l.code_length = 13;
 		ir->l.add_to_buf=add_to_buf_haup_pvr150;
 		break;
 	case 0x6b:
-		strcpy(ir->c.name,"Adaptec IR");
+		snprintf(ir->c.name, I2C_NAME_SIZE, "Adaptec IR");
 		ir->l.code_length = 32;
 		ir->l.add_to_buf=add_to_buf_adap;
 		break;
@@ -438,25 +438,25 @@
 		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_BT848))
 #endif
 		{
-			strcpy(ir->c.name,"Hauppauge IR");
+			snprintf(ir->c.name, I2C_NAME_SIZE, "Hauppauge IR");
 			ir->l.code_length = 13;
 			ir->l.add_to_buf=add_to_buf_haup;
 		}
 		else /* I2C_HW_B_CX2388x */
 		{
-			strcpy(ir->c.name,"Leadtek IR");
+			snprintf(ir->c.name, I2C_NAME_SIZE, "Leadtek IR");
 			ir->l.code_length = 8;
 			ir->l.add_to_buf=add_to_buf_pvr2000;
 		}
 		break;
 	case 0x30:
-		strcpy(ir->c.name,"KNC ONE IR");
+		snprintf(ir->c.name, I2C_NAME_SIZE, "KNC ONE IR");
 		ir->l.code_length = 8;
 		ir->l.add_to_buf=add_to_buf_knc1;
 		break;
 	case 0x21:
 	case 0x23:
-		strcpy(ir->c.name,"TV-Box IR");
+		snprintf(ir->c.name, I2C_NAME_SIZE, "TV-Box IR");
 		ir->l.code_length = 8;
 		ir->l.add_to_buf=add_to_buf_pcf8574;
 		ir->bits = flags & 0xff;
