diff -Naur lirc-0.8.4pre1-old/drivers/lirc_imon/lirc_imon.c lirc-0.8.4pre1-new/drivers/lirc_imon/lirc_imon.c
--- lirc-0.8.4pre1-old/drivers/lirc_imon/lirc_imon.c	2008-09-21 10:35:41.000000000 -0700
+++ lirc-0.8.4pre1-new/drivers/lirc_imon/lirc_imon.c	2008-09-21 10:36:38.000000000 -0700
@@ -123,8 +123,12 @@
 static ssize_t vfd_write(struct file *file, const char *buf,
 				size_t n_bytes, loff_t *pos);
 
-/* LCD file_operations override function prototypes */
-static ssize_t lcd_write(struct file *file, const char *buf,
+/* VFD display */
+static ssize_t vfd_write_vfd(struct file *file, const char *buf,
+				size_t n_bytes, loff_t *pos);
+
+/* LCD display */
+static ssize_t vfd_write_lcd(struct file *file, const char *buf,
 				size_t n_bytes, loff_t *pos);
 
 /* LIRC plugin function prototypes */
@@ -618,6 +622,23 @@
 #endif
 
 
+/**
+ * Writes data to the VFD.
+ */
+static ssize_t vfd_write(struct file *file, const char *buf,
+			 size_t n_bytes, loff_t *pos)
+{
+	/**
+	 * We SHOULD detect this now, rather than just doing whatever our module
+	 * parameters tells us...
+	 */
+	if (is_lcd) {
+		return vfd_write_lcd(file, buf, n_bytes, pos);
+	} else {
+		return vfd_write_vfd(file, buf, n_bytes, pos);
+	}
+}
+ 
 
 /**
  * Writes data to the VFD.  The IMON VFD is 2x16 characters
@@ -630,7 +651,7 @@
  * than 32 bytes are provided spaces will be appended to
  * generate a full screen.
  */
-static ssize_t vfd_write(struct file *file, const char *buf,
+static ssize_t vfd_write_vfd(struct file *file, const char *buf,
 			 size_t n_bytes, loff_t *pos)
 {
 	int i;
@@ -717,7 +738,7 @@
  * display whatever diacritics you need, and so on), but it's also
  * a lot more complicated than most LCDs...
  */
-static ssize_t lcd_write(struct file *file, const char *buf,
+static ssize_t vfd_write_lcd(struct file *file, const char *buf,
 			 size_t n_bytes, loff_t *pos)
 {
 	int retval = SUCCESS;
@@ -1090,13 +1111,6 @@
 
 	info("%s: found IMON device", __FUNCTION__);
 
-	/*
-	 * If it's the LCD, as opposed to the VFD, we just need to replace
-	 * the "write" file op.
-	 */
-	if (is_lcd)
-		vfd_fops.write = &lcd_write;
-
 #if !defined(KERNEL_2_5)
 	for (subminor = 0; subminor < MAX_DEVICES; ++subminor) {
 		if (minor_table[subminor] == NULL)
