diff -Naur lirc-0.8.6-old/drivers/lirc_mceusb/lirc_mceusb.c lirc-0.8.6-new/drivers/lirc_mceusb/lirc_mceusb.c
--- lirc-0.8.6-old/drivers/lirc_mceusb/lirc_mceusb.c	2009-09-02 07:04:02.000000000 -0700
+++ lirc-0.8.6-new/drivers/lirc_mceusb/lirc_mceusb.c	2009-12-28 10:11:02.000000000 -0800
@@ -142,6 +142,7 @@
 #define VENDOR_WISTRON		0x0fb8
 #define VENDOR_COMPRO		0x185b
 #define VENDOR_NORTHSTAR	0x04eb
+#define VENDOR_REALTEK		0x0bda
 
 static struct usb_device_id mceusb_dev_table[] = {
 	/* Original Microsoft MCE IR Transceiver (often HP-branded) */
@@ -158,6 +159,8 @@
 	{ USB_DEVICE(VENDOR_PHILIPS, 0x0613) },
 	/* Philips eHome Infrared Transceiver */
 	{ USB_DEVICE(VENDOR_PHILIPS, 0x0815) },
+	/* Realtek MCE IR Receiver */
+	{ USB_DEVICE(VENDOR_REALTEK, 0x0161) },
 	/* SMK/Toshiba G83C0004D410 */
 	{ USB_DEVICE(VENDOR_SMK, 0x031d) },
 	/* SMK eHome Infrared Transceiver (Sony VAIO) */
