diff -Naur lirc-0.8.6-old/drivers/lirc_mceusb/lirc_mceusb.c lirc-0.8.6-new/drivers/lirc_mceusb/lirc_mceusb.c
--- lirc-0.8.6-old/drivers/lirc_mceusb/lirc_mceusb.c	2009-12-28 10:11:02.000000000 -0800
+++ lirc-0.8.6-new/drivers/lirc_mceusb/lirc_mceusb.c	2009-12-28 10:13:46.000000000 -0800
@@ -143,6 +143,7 @@
 #define VENDOR_COMPRO		0x185b
 #define VENDOR_NORTHSTAR	0x04eb
 #define VENDOR_REALTEK		0x0bda
+#define VENDOR_TIVO		0x105a
 
 static struct usb_device_id mceusb_dev_table[] = {
 	/* Original Microsoft MCE IR Transceiver (often HP-branded) */
@@ -225,6 +226,8 @@
 	{ USB_DEVICE(VENDOR_COMPRO, 0x3082) },
 	/* Northstar Systems eHome Infrared Transceiver */
 	{ USB_DEVICE(VENDOR_NORTHSTAR, 0xe004) },
+	/* TiVo PC IR Receiver */
+	{ USB_DEVICE(VENDOR_TIVO, 0x2000) },
 	/* Terminating entry */
 	{ }
 };
