diff -Naur lirc-20090120Z-old/drivers/lirc_i2c/lirc_i2c.c lirc-20090120Z-new/drivers/lirc_i2c/lirc_i2c.c
--- lirc-20090120Z-old/drivers/lirc_i2c/lirc_i2c.c	2009-01-19 13:12:21.000000000 -0800
+++ lirc-20090120Z-new/drivers/lirc_i2c/lirc_i2c.c	2009-01-19 20:41:58.000000000 -0800
@@ -652,10 +652,6 @@
 
 int init_module(void)
 {
-	request_module("bttv");
-	request_module("rivatv");
-	request_module("ivtv");
-	request_module("cx8800");
 	i2c_add_driver(&driver);
 	return 0;
 }
