diff -Naur lirc-20090618PST-old/configure.ac lirc-20090618PST-new/configure.ac
--- lirc-20090618PST-old/configure.ac	2009-06-18 08:00:32.000000000 -0700
+++ lirc-20090618PST-new/configure.ac	2009-06-18 08:02:32.000000000 -0700
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
