diff -Naur lirc-20071217Z-old/drivers/lirc_imon/lirc_imon.c lirc-20071217Z-new/drivers/lirc_imon/lirc_imon.c
--- lirc-20071217Z-old/drivers/lirc_imon/lirc_imon.c	2007-09-30 02:58:45.000000000 -0700
+++ lirc-20071217Z-new/drivers/lirc_imon/lirc_imon.c	2007-12-18 08:11:14.000000000 -0800
@@ -233,7 +233,7 @@
 /* to prevent races between open() and disconnect() */
 static DECLARE_MUTEX(disconnect_sem);
 
-static int debug;
+static int debug = 0;
 
 #if !defined(KERNEL_2_5)
 
@@ -260,7 +260,7 @@
 MODULE_DESCRIPTION(MOD_DESC);
 MODULE_LICENSE("GPL");
 MODULE_DEVICE_TABLE(usb, imon_usb_id_table);
-module_param(debug, int, 0);
+module_param(debug, int, 0644);
 MODULE_PARM_DESC(debug, "Debug messages: 0=no, 1=yes(default: no)");
 
 static inline void delete_context(struct imon_context *context)
