diff -Naur lirc-0.8.2pre3-old/drivers/lirc_imon/lirc_imon.c lirc-0.8.2pre3-new/drivers/lirc_imon/lirc_imon.c
--- lirc-0.8.2pre3-old/drivers/lirc_imon/lirc_imon.c	2007-04-29 07:23:04.000000000 -0700
+++ lirc-0.8.2pre3-new/drivers/lirc_imon/lirc_imon.c	2007-05-30 09:29:50.000000000 -0700
@@ -260,6 +260,7 @@
 MODULE_DEVICE_TABLE (usb, imon_usb_id_table);
 module_param (debug, int, 0);
 MODULE_PARM_DESC (debug, "Debug messages: 0=no, 1=yes (default: no)");
+MODULE_DEVICE_TABLE(usb, imon_usb_id_table);
 
 static inline void delete_context (struct imon_context *context) {
 
