diff -Naur NVIDIA-Linux-x86_64-185.18.14-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-185.18.14-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-185.18.14-pkg2-old/usr/src/nv/conftest.sh	2009-05-27 02:02:40.000000000 -0700
+++ NVIDIA-Linux-x86_64-185.18.14-pkg2-new/usr/src/nv/conftest.sh	2009-05-28 08:59:32.000000000 -0700
@@ -1142,7 +1142,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
