diff -Naur Mesa-7.4.2-old/src/mesa/x86/Makefile Mesa-7.4.2-new/src/mesa/x86/Makefile
--- Mesa-7.4.2-old/src/mesa/x86/Makefile	2009-05-17 07:00:58.000000000 -0700
+++ Mesa-7.4.2-new/src/mesa/x86/Makefile	2009-05-17 07:02:06.000000000 -0700
@@ -21,7 +21,7 @@
 
 
 gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+	$(BUILD_CC) $(INCLUDE_DIRS) $(BUILD_CFLAGS) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
 matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
