diff -Naur stella-2.6-old/src/emucore/CartAR.cxx stella-2.6-new/src/emucore/CartAR.cxx
--- stella-2.6-old/src/emucore/CartAR.cxx	2008-03-28 16:29:13.000000000 -0700
+++ stella-2.6-new/src/emucore/CartAR.cxx	2008-05-18 16:44:17.000000000 -0700
@@ -17,6 +17,7 @@
 //============================================================================
 
 #include <cassert>
+#include <cstring>
 
 #include "M6502Hi.hxx"
 #include "Random.hxx"
diff -Naur stella-2.6-old/src/emucore/Cart.cxx stella-2.6-new/src/emucore/Cart.cxx
--- stella-2.6-old/src/emucore/Cart.cxx	2008-02-27 12:13:55.000000000 -0800
+++ stella-2.6-new/src/emucore/Cart.cxx	2008-05-18 16:44:17.000000000 -0700
@@ -18,6 +18,7 @@
 
 #include <cassert>
 #include <sstream>
+#include <cstring>
 
 #include "bspf.hxx"
 #include "Cart.hxx"
diff -Naur stella-2.6-old/src/emucore/CartDPC.cxx stella-2.6-new/src/emucore/CartDPC.cxx
--- stella-2.6-old/src/emucore/CartDPC.cxx	2008-03-28 16:29:13.000000000 -0700
+++ stella-2.6-new/src/emucore/CartDPC.cxx	2008-05-18 16:44:17.000000000 -0700
@@ -17,6 +17,7 @@
 //============================================================================
 
 #include <cassert>
+#include <cstring>
 #include <iostream>
 
 #include "System.hxx"
diff -Naur stella-2.6-old/src/emucore/TrackBall.cxx stella-2.6-new/src/emucore/TrackBall.cxx
--- stella-2.6-old/src/emucore/TrackBall.cxx	2008-05-12 15:40:25.000000000 -0700
+++ stella-2.6-new/src/emucore/TrackBall.cxx	2008-05-18 16:48:33.000000000 -0700
@@ -16,6 +16,7 @@
 // $Id: TrackBall.cxx,v 1.1 2008/05/12 22:40:25 stephena Exp $
 //============================================================================
 
+#include <cstdlib>
 #include "Event.hxx"
 #include "System.hxx"
 #include "TIA.hxx"
diff -Naur stella-2.6-old/src/unix/SerialPortUNIX.cxx stella-2.6-new/src/unix/SerialPortUNIX.cxx
--- stella-2.6-old/src/unix/SerialPortUNIX.cxx	2008-04-11 10:56:34.000000000 -0700
+++ stella-2.6-new/src/unix/SerialPortUNIX.cxx	2008-05-18 16:45:32.000000000 -0700
@@ -20,6 +20,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <string.h>
 #include <sys/termios.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
