diff -Naur pcre-7.9-old/Makefile.in pcre-7.9-new/Makefile.in
--- pcre-7.9-old/Makefile.in	2009-04-11 16:54:02.000000000 -0700
+++ pcre-7.9-new/Makefile.in	2009-04-11 16:55:05.000000000 -0700
@@ -1433,7 +1433,7 @@
 	cp -p pcre.h $@
 
 @WITH_REBUILD_CHARTABLES_TRUE@pcre_chartables.c: dftables$(EXEEXT)
-@WITH_REBUILD_CHARTABLES_TRUE@	./dftables$(EXEEXT) $@
+@WITH_REBUILD_CHARTABLES_TRUE@	pcre-dftables$(EXEEXT) $@
 
 @WITH_REBUILD_CHARTABLES_FALSE@pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
 @WITH_REBUILD_CHARTABLES_FALSE@	rm -f $@
