diff -Naur pcre-8.02-old/Makefile.in pcre-8.02-new/Makefile.in
--- pcre-8.02-old/Makefile.in	2010-03-20 18:36:16.000000000 -0700
+++ pcre-8.02-new/Makefile.in	2010-03-20 18:37:17.000000000 -0700
@@ -1476,7 +1476,7 @@
 	cp -p pcre.h $@
 
 @WITH_REBUILD_CHARTABLES_TRUE@pcre_chartables.c: dftables$(EXEEXT)
-@WITH_REBUILD_CHARTABLES_TRUE@	./dftables$(EXEEXT) $@
+@WITH_REBUILD_CHARTABLES_TRUE@	pcre-dftables$(EXEEXT) $@
 
 @WITH_REBUILD_CHARTABLES_FALSE@pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
 @WITH_REBUILD_CHARTABLES_FALSE@	rm -f $@
