diff -Naur lighttpd-1.4.31-old/configure.ac lighttpd-1.4.31-new/configure.ac
--- lighttpd-1.4.31-old/configure.ac	2012-05-31 07:49:18.000000000 -0700
+++ lighttpd-1.4.31-new/configure.ac	2012-07-16 16:56:15.000000000 -0700
@@ -16,6 +16,7 @@
 dnl Checks for programs.
 AC_PROG_CC
 AM_PROG_CC_C_O
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 AC_PROG_LD
 AC_PROG_INSTALL
 AC_PROG_AWK
@@ -38,9 +39,6 @@
 
 AC_EXEEXT
 
-dnl more automake stuff
-AM_C_PROTOTYPES
-
 dnl libtool
 AC_DISABLE_STATIC
 AC_ENABLE_SHARED
