diff -Naur xf86-video-openchrome-0.2.902-old/src/via_video.c xf86-video-openchrome-0.2.902-new/src/via_video.c
--- xf86-video-openchrome-0.2.902-old/src/via_video.c	2008-04-09 12:42:55.000000000 -0700
+++ xf86-video-openchrome-0.2.902-new/src/via_video.c	2008-04-21 07:13:41.000000000 -0700
@@ -723,8 +723,13 @@
         BoxPtr pBox;
         int nBox;
 
+#ifdef COMPOSITE
         REGION_TRANSLATE(pScrn->pScreen, clipBoxes, - pPix->screen_x,
             - pPix->screen_y);
+#else
+        REGION_TRANSLATE(pScrn->pScreen, clipBoxes, 0,
+            0);
+#endif
 
         nBox = REGION_NUM_RECTS(clipBoxes);
         pBox = REGION_RECTS(clipBoxes);
