diff -Naur lame-3.99.3-old/configure.in lame-3.99.3-new/configure.in
--- lame-3.99.3-old/configure.in	2011-11-26 11:49:21.000000000 -0800
+++ lame-3.99.3-new/configure.in	2011-11-28 20:55:04.000000000 -0800
@@ -388,7 +388,7 @@
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+AM_PATH_GTK_2_0(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
