diff -Naur lame-3.99.1-old/configure.in lame-3.99.1-new/configure.in
--- lame-3.99.1-old/configure.in	2011-11-05 10:09:20.000000000 -0700
+++ lame-3.99.1-new/configure.in	2011-11-10 15:18:37.000000000 -0800
@@ -388,7 +388,7 @@
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+AM_PATH_GTK_2_0(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
