diff -Naur nfs-utils-1.1.0-old/utils/exportfs/exportfs.c nfs-utils-1.1.0-new/utils/exportfs/exportfs.c
--- nfs-utils-1.1.0-old/utils/exportfs/exportfs.c	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0-new/utils/exportfs/exportfs.c	2010-08-03 16:47:12.000000000 -0700
@@ -12,6 +12,7 @@
 #include <config.h>
 #endif
 
+#include <sys/stat.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
diff -Naur nfs-utils-1.1.0-old/utils/mount/mount.c nfs-utils-1.1.0-new/utils/mount/mount.c
--- nfs-utils-1.1.0-old/utils/mount/mount.c	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0-new/utils/mount/mount.c	2010-08-03 16:50:52.000000000 -0700
@@ -21,6 +21,7 @@
 #include "config.h"
 #include <unistd.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <stdio.h>
 #include <string.h>
 #include <errno.h>
