diff -Naur myththemes-addons-trunk-16487-old/cpsvndir myththemes-addons-trunk-16487-new/cpsvndir
--- myththemes-addons-trunk-16487-old/cpsvndir	2008-03-09 23:24:48.000000000 -0700
+++ myththemes-addons-trunk-16487-new/cpsvndir	2008-03-09 23:24:48.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
