diff -Naur myththemes-addons-trunk-21969-old/configure myththemes-addons-trunk-21969-new/configure
--- myththemes-addons-trunk-21969-old/configure	2009-09-21 05:55:26.000000000 -0700
+++ myththemes-addons-trunk-21969-new/configure	2009-09-21 05:56:20.000000000 -0700
@@ -4,6 +4,7 @@
 #
 
 prefix="/usr/local"
+sysroot=""
 
 qmake="qmake"
 
@@ -11,6 +12,8 @@
   case "$opt" in
   --prefix=*) prefix=`echo $opt | cut -d '=' -f 2`
   ;;
+  --sysroot=*) sysroot=`echo $opt | cut -d '=' -f 2`
+  ;;
   --qmake=*) qmake=`echo $opt | cut -d '=' -f 2`
   ;;
   esac
@@ -22,7 +25,8 @@
 Usage:   configure [options]
 Options: [defaults in brackets after descriptions]
 Generic options:
-  --prefix=PREFIX          MythTV install location PREFIX [$prefix]"
+  --prefix=PREFIX          MythTV install location PREFIX [$prefix]
+  --sysroot=SYSROOT        sysroot location SYSROOT [$sysroot]
   --qmake=QMAKE            use specified qmake [$qmake]
   --help                   print this message
 EOF
@@ -30,13 +34,13 @@
 fi
 
 # bring in mythtv config
-if [ -e $prefix/include/mythtv/mythconfig.mak ] ; then
+if [ -e $sysroot$prefix/include/mythtv/mythconfig.mak ] ; then
   rm mythconfig.mak 2> /dev/null
-  ln -s $prefix/include/mythtv/mythconfig.mak mythconfig.mak
+  ln -s $sysroot$prefix/include/mythtv/mythconfig.mak mythconfig.mak
 else
-  echo "ERROR: mythconfig.mak not found at $prefix/include/mythtv/mythconfig.mak"
+  echo "ERROR: mythconfig.mak not found at $sysroot$prefix/include/mythtv/mythconfig.mak"
   echo "Did you make AND install MythTV first?"
-  echo "Are you using the correct prefix ($prefix)?"
+  echo "Are you using the correct prefix ($prefix) and sysroot ($sysroot)?"
   echo "Bailing out!!"
   exit
 fi
@@ -45,7 +49,7 @@
 {
     $1 --version 2>&1 | grep -q "Qt version 4"
 }
-
+ 
 if ! is_qmake4 $qmake; then
     if is_qmake4 qmake-qt4; then
         qmake=qmake-qt4
@@ -56,5 +60,3 @@
 fi
 
 ${qmake} QMAKE=$(which ${qmake}) themes.pro
-
-
diff -Naur myththemes-addons-trunk-21969-old/configure.orig myththemes-addons-trunk-21969-new/configure.orig
--- myththemes-addons-trunk-21969-old/configure.orig	1969-12-31 16:00:00.000000000 -0800
+++ myththemes-addons-trunk-21969-new/configure.orig	2009-09-21 05:56:20.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/sh
+#
+# Configure script for myththemes
+#
+
+prefix="/usr/local"
+
+qmake="qmake"
+
+for opt do
+  case "$opt" in
+  --prefix=*) prefix=`echo $opt | cut -d '=' -f 2`
+  ;;
+  --qmake=*) qmake=`echo $opt | cut -d '=' -f 2`
+  ;;
+  esac
+done
+
+if test x"$1" = x"-h" -o x"$1" = x"--help" ; then
+cat << EOF
+
+Usage:   configure [options]
+Options: [defaults in brackets after descriptions]
+Generic options:
+  --prefix=PREFIX          MythTV install location PREFIX [$prefix]"
+  --qmake=QMAKE            use specified qmake [$qmake]
+  --help                   print this message
+EOF
+exit 1
+fi
+
+# bring in mythtv config
+if [ -e $prefix/include/mythtv/mythconfig.mak ] ; then
+  rm mythconfig.mak 2> /dev/null
+  ln -s $prefix/include/mythtv/mythconfig.mak mythconfig.mak
+else
+  echo "ERROR: mythconfig.mak not found at $prefix/include/mythtv/mythconfig.mak"
+  echo "Did you make AND install MythTV first?"
+  echo "Are you using the correct prefix ($prefix)?"
+  echo "Bailing out!!"
+  exit
+fi
+
+is_qmake4()
+{
+    $1 --version 2>&1 | grep -q "Qt version 4"
+}
+
+if ! is_qmake4 $qmake; then
+    if is_qmake4 qmake-qt4; then
+        qmake=qmake-qt4
+    else
+        echo "qmake for Qt4 not found. Please specify the correct qmake with --qmake="
+        exit 1
+    fi
+fi
+
+${qmake} QMAKE=$(which ${qmake}) themes.pro
+
+
