diff -Naur xtrans-1.1-old/Xtranssock.c xtrans-1.1-new/Xtranssock.c
--- xtrans-1.1-old/Xtranssock.c	2008-03-05 17:48:42.000000000 -0800
+++ xtrans-1.1-new/Xtranssock.c	2008-04-25 04:09:14.000000000 -0700
@@ -1180,7 +1180,7 @@
 
     if (abstract) {
 	sockname.sun_path[0] = '\0';
-	namelen = sizeof(sockname);
+	namelen = offsetof(struct sockaddr_un, sun_path) + 1 + strlen(&sockname.sun_path[1]);
     }
     else
 	unlink (sockname.sun_path);
