diff -Naur mythtv-7426-old/programs/programs.pro mythtv-7426-new/programs/programs.pro
--- mythtv-7426-old/programs/programs.pro	2005-10-25 09:18:36.000000000 -0700
+++ mythtv-7426-new/programs/programs.pro	2005-10-25 11:00:20.000000000 -0700
@@ -6,7 +6,7 @@
 # Directories
 using_frontend {
     SUBDIRS += mythepg mythprogfind mythtv mythfrontend mythcommflag mythlcd 
-    SUBDIRS += mythtvosd mythjobqueue mythuitest mythlcdserver
+    SUBDIRS += mythtvosd mythjobqueue mythuitest
 }
 
 using_backend {
