diff -Naur mythtv-7426-old/programs/mythfrontend/globalsettings.cpp mythtv-7426-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-7426-old/programs/mythfrontend/globalsettings.cpp	2005-10-25 09:36:17.000000000 -0700
+++ mythtv-7426-new/programs/mythfrontend/globalsettings.cpp	2005-10-25 09:44:22.000000000 -0700
@@ -22,6 +22,10 @@
     HostComboBox *gc = new HostComboBox("AudioOutputDevice", true);
 
     gc->setLabel(QObject::tr("Audio output device"));
+
+    gc->addSelection(QObject::tr("ALSA:default"), "ALSA:default", true);
+    gc->addSelection(QObject::tr("ALSA:iec958"), "ALSA:iec958");
+
     QDir dev("/dev", "dsp*", QDir::Name, QDir::System);
     gc->fillSelectionsFromDir(dev);
     dev.setNameFilter("adsp*");
@@ -43,7 +47,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("MythControlsVolume");
     gc->setLabel(QObject::tr("Use internal volume controls"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("MythTV can control the PCM and master "
                     "mixer volume.  If you prefer to use an external mixer "
                     "program, then disable this option."));
@@ -55,6 +59,9 @@
     HostComboBox *gc = new HostComboBox("MixerDevice", true);
 
     gc->setLabel(QObject::tr("Mixer Device"));
+
+    gc->addSelection(QObject::tr("default"), "default", true);
+
     QDir dev("/dev", "mixer*", QDir::Name, QDir::System);
     gc->fillSelectionsFromDir(dev);
 
@@ -148,7 +155,7 @@
     gc->setLabel(QObject::tr("Algorithm"));
     gc->addSelection(QObject::tr("Linear blend"), "linearblend");
     gc->addSelection(QObject::tr("Kernel (less motion blur)"), "kerneldeint");
-    gc->addSelection(QObject::tr("Bob (2x framerate)"), "bobdeint");
+    gc->addSelection(QObject::tr("Bob (2x framerate)"), "bobdeint", true);
     gc->addSelection(QObject::tr("One field"), "onefield");
     gc->setHelpText(QObject::tr("Deinterlace algorithm. "
                                 "'Kernel' require SSE. 'Bob' requires "
@@ -188,7 +195,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("DecodeExtraAudio");
     gc->setLabel(QObject::tr("Extra audio buffering"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("Enable this setting if MythTV is playing "
                     "\"crackly\" audio and you are using hardware encoding. "
                     "This setting will have no effect "
@@ -641,7 +648,7 @@
     gc->setLabel(QObject::tr("Sort Episodes"));
     gc->addSelection(QObject::tr("Record date"), "Date");
     gc->addSelection(QObject::tr("Original Air date"), "OrigAirDate");
-    gc->addSelection(QObject::tr("Program ID"), "Id");
+    gc->addSelection(QObject::tr("Program ID"), "Id", true);
     gc->setHelpText(QObject::tr("Selects how to sort a shows episodes"));
     return gc;
 }
@@ -650,7 +657,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("StickyKeys");
     gc->setLabel(QObject::tr("Sticky keys"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If enabled, fast forward and rewind "
                     "continue after the key is released.  Pressing the key "
                     "again increases the fast forward or rewind speed.  The "
@@ -719,6 +726,8 @@
             gc->addSelection(theme->fileName());
     }
 
+    gc->setValue(gc->getValueIndex("G.A.N.T."));
+
     return gc;
 }
 
@@ -746,6 +755,8 @@
             gc->addSelection(theme->fileName());
     }
 
+    gc->setValue(gc->getValueIndex("Titivillus-OSD"));
+
     return gc;
 }
 
@@ -756,6 +767,8 @@
     QDir ttf(gContext->GetFontsDir(), gContext->GetFontsNameFilter());
     gc->fillSelectionsFromDir(ttf, false);
 
+    gc->setValue(gc->getValueIndex("FreeSans.ttf"));
+
     return gc;
 }
 
@@ -766,6 +779,8 @@
     QDir ttf(gContext->GetFontsDir(), gContext->GetFontsNameFilter());
     gc->fillSelectionsFromDir(ttf, false);
 
+    gc->setValue(gc->getValueIndex("FreeSans.ttf"));
+
     return gc;
 }
 
@@ -846,7 +861,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("UseVideoTimebase");
     gc->setLabel(QObject::tr("Use video as timebase"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("Use the video as the timebase and warp "
                     "the audio to keep it in sync. (Experimental)"));
     return gc;
@@ -975,7 +990,7 @@
     HostComboBox *gc = new HostComboBox("PlaybackExitPrompt");
     gc->setLabel(QObject::tr("Action on playback exit"));
     gc->addSelection(QObject::tr("Just exit"), "0");
-    gc->addSelection(QObject::tr("Save position and exit"), "2");
+    gc->addSelection(QObject::tr("Save position and exit"), "2", true);
     gc->addSelection(QObject::tr("Always prompt"), "1");
     gc->setHelpText(QObject::tr("If set to prompt, a menu will be displayed "
                     "when you exit playback mode.  The options available will "
@@ -988,7 +1003,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("EndOfRecordingExitPrompt");
     gc->setLabel(QObject::tr("Prompt at end of recording"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If set, a menu will be displayed allowing "
                     "you to delete the recording when it has finished "
                     "playing."));
@@ -1036,7 +1051,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("PlaybackPreview");
     gc->setLabel(QObject::tr("Display live preview of recordings"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("When enabled, a preview of the recording "
                     "will play in a small window on the \"Watch a "
                     "Recording\" menu."));
@@ -1084,7 +1099,7 @@
     HostComboBox *gc = new HostComboBox("AllowQuitShutdown");
     gc->setLabel(QObject::tr("System Exit key"));
     gc->addSelection(QObject::tr("ESC"), "4");
-    gc->addSelection(QObject::tr("No exit key"), "0");
+    gc->addSelection(QObject::tr("No exit key"), "0", true);
     gc->addSelection(QObject::tr("Control-ESC"), "1");
     gc->addSelection(QObject::tr("Meta-ESC"), "2");
     gc->addSelection(QObject::tr("Alt-ESC"), "3");
@@ -1110,7 +1125,7 @@
 {
     HostLineEdit *ge = new HostLineEdit("HaltCommand");
     ge->setLabel(QObject::tr("Halt command"));
-    ge->setValue("halt");
+    ge->setValue("poweroff");
     ge->setHelpText(QObject::tr("If you have configured an exit key using the "
                     "System Shutdown option, you will be given the opportunity "
                     "to exit MythTV or halt the system completely. "
@@ -1722,7 +1737,7 @@
     GlobalCheckBox *bc = new GlobalCheckBox("LastFreeCard");
     bc->setLabel(QObject::tr("Avoid conflicts between live TV and "
                  "scheduled shows."));
-    bc->setValue(false);
+    bc->setValue(true);
     bc->setHelpText(QObject::tr("If enabled, live TV will choose a tuner card "
                     "that is less likely to have scheduled recordings "
                     "rather than the best card available."));
@@ -1777,7 +1792,8 @@
     gc->addSelection(QObject::tr("Blender - Transparent (CPU Usage - Middle)"),
                      QString::number((int)UIGuideType::Dense));
     gc->addSelection(QObject::tr("Eco - Transparent (CPU Usage - Low)"),
-                     QString::number((int)UIGuideType::Eco));
+                     QString::number((int)UIGuideType::Eco),
+                     true);
     gc->addSelection(QObject::tr("Solid (CPU Usage - Middle)"),
                      QString::number((int)UIGuideType::Solid));
     return gc;
@@ -2224,7 +2240,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("UseXvMcVld");
     gc->setLabel(QObject::tr("Use hardware XVMC VLD Decoding (incompatible with libmpeg2)"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("Enables the use of Via XvMC hardware MPEG "
                                 "decoding for the Via Unichrome Chipset.  ")+
                     QObject::tr(
