diff -Naur pciutils-2.2.10-old/Makefile pciutils-2.2.10-new/Makefile
--- pciutils-2.2.10-old/Makefile	2008-02-12 03:31:02.000000000 -0800
+++ pciutils-2.2.10-new/Makefile	2008-02-12 11:31:36.000000000 -0800
@@ -68,7 +68,7 @@
 install: all
 # -c is ignored on Linux, but required on FreeBSD
 	$(DIRINSTALL) -m 755 $(DESTDIR)$(SBINDIR) $(DESTDIR)$(IDSDIR) $(DESTDIR)$(MANDIR)/man8
-	$(INSTALL) -c -m 755 -s lspci setpci $(DESTDIR)$(SBINDIR)
+	$(INSTALL) -c -m 755 lspci setpci $(DESTDIR)$(SBINDIR)
 	$(INSTALL) -c -m 755 update-pciids $(DESTDIR)$(SBINDIR)
 	$(INSTALL) -c -m 644 $(PCI_IDS) $(DESTDIR)$(IDSDIR)
 	$(INSTALL) -c -m 644 lspci.8 setpci.8 update-pciids.8 $(DESTDIR)$(MANDIR)/man8
