diff -Naur linux-2.6.16.28-old/drivers/net/via-rhine.c linux-2.6.16.28-new/drivers/net/via-rhine.c
--- linux-2.6.16.28-old/drivers/net/via-rhine.c	2006-08-26 08:01:36.000000000 -0700
+++ linux-2.6.16.28-new/drivers/net/via-rhine.c	2006-08-26 08:03:37.000000000 -0700
@@ -1970,9 +1970,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
