diff -Naur linux-2.6.19-old/drivers/net/via-rhine.c linux-2.6.19-new/drivers/net/via-rhine.c
--- linux-2.6.19-old/drivers/net/via-rhine.c	2006-12-03 08:23:49.000000000 -0800
+++ linux-2.6.19-new/drivers/net/via-rhine.c	2006-12-03 08:27:40.000000000 -0800
@@ -46,7 +46,7 @@
 
 /* Work-around for broken BIOSes: they are unable to get the chip back out of
    power state D3 so PXE booting fails. bootparam(7): via-rhine.avoid_D3=1 */
-static int avoid_D3;
+static int avoid_D3 = 1;
 
 /*
  * In case you are looking for 'options[]' or 'full_duplex[]', they
