diff -Naur linux-2.6.15.7-old/drivers/acpi/utilities/utmisc.c linux-2.6.15.7-new/drivers/acpi/utilities/utmisc.c
--- linux-2.6.15.7-old/drivers/acpi/utilities/utmisc.c	2006-03-27 23:06:28.000000000 -0800
+++ linux-2.6.15.7-new/drivers/acpi/utilities/utmisc.c	2006-04-09 19:24:26.000000000 -0700
@@ -436,6 +436,7 @@
 	char *name_ptr = (char *)&name;
 	char character;
 	acpi_native_uint i;
+	int badcnt = 0
 
 	ACPI_FUNCTION_ENTRY();
 
@@ -446,9 +447,11 @@
 		if (!((character == '_') ||
 		      (character >= 'A' && character <= 'Z') ||
 		      (character >= '0' && character <= '9'))) {
-			return (FALSE);
+			badcnt++;
 		}
 	}
+	if (badcnt > 1)
+		return (FALSE);
 
 	return (TRUE);
 }
