diff -Naur linux-2.6.15-old/drivers/net/via-rhine.c linux-2.6.15-new/drivers/net/via-rhine.c
--- linux-2.6.15-old/drivers/net/via-rhine.c	2006-01-02 19:21:10.000000000 -0800
+++ linux-2.6.15-new/drivers/net/via-rhine.c	2006-01-03 17:14:48.000000000 -0800
@@ -1964,9 +1964,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
