diff -Naur linux-2.6.17.14-old/arch/i386/kernel/cpu/cpufreq/longhaul.c linux-2.6.17.14-new/arch/i386/kernel/cpu/cpufreq/longhaul.c
--- linux-2.6.17.14-old/arch/i386/kernel/cpu/cpufreq/longhaul.c	2006-10-20 13:46:41.000000000 -0700
+++ linux-2.6.17.14-new/arch/i386/kernel/cpu/cpufreq/longhaul.c	2006-10-20 13:49:39.000000000 -0700
@@ -30,6 +30,8 @@
 #include <linux/slab.h>
 #include <linux/string.h>
 #include <linux/pci.h>
+#include <linux/ide.h>
+#include <linux/delay.h>
 
 #include <asm/msr.h>
 #include <asm/timex.h>
@@ -89,6 +91,24 @@
 }
 #endif
 
+static void ide_idle(void)
+{
+	int i;
+	ide_hwif_t *hwif = ide_hwifs;
+	ide_drive_t *drive;
+
+	i = 0;
+	do {
+		drive = &hwif->drives[i];
+		i++;
+		if (strncmp(drive->name,"hd",2) == 0) {
+			while (drive->waiting_for_dma)
+			udelay(10);
+		} else {
+			i = 0;
+		}
+	} while (i != 0);
+}
 
 static unsigned int calc_speed(int mult)
 {
@@ -144,6 +164,7 @@
 	longhaul->bits.RevisionKey = 0;
 
 	preempt_disable();
+	ide_idle(); /* avoid ide timeouts when bus master off */
 	local_irq_save(flags);
 
 	/*
