diff -Naur linux-2.6.10-old/drivers/net/via-rhine.c linux-2.6.10-new/drivers/net/via-rhine.c
--- linux-2.6.10-old/drivers/net/via-rhine.c	2004-12-24 13:35:39.000000000 -0800
+++ linux-2.6.10-new/drivers/net/via-rhine.c	2005-01-22 21:16:47.512485134 -0800
@@ -1933,9 +1933,6 @@
 	writeb(0x01, ioaddr + PwcfgSet);
 	writeb(readb(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 
-	/* Hit power state D3 (sleep) */
-	writeb(readb(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
