diff -Naur linux-2.6.8.1-old/drivers/usb/Makefile linux-2.6.8.1-new/drivers/usb/Makefile
--- linux-2.6.8.1-old/drivers/usb/Makefile	2004-06-15 22:20:04.000000000 -0700
+++ linux-2.6.8.1-new/drivers/usb/Makefile	2004-09-11 17:01:34.000000000 -0700
@@ -26,6 +26,7 @@
 obj-$(CONFIG_USB_KBTAB)		+= input/
 obj-$(CONFIG_USB_MOUSE)		+= input/
 obj-$(CONFIG_USB_MTOUCH)	+= input/
+obj-$(CONFIG_USB_EGALAX)	+= input/
 obj-$(CONFIG_USB_POWERMATE)	+= input/
 obj-$(CONFIG_USB_WACOM)		+= input/
 obj-$(CONFIG_USB_XPAD)		+= input/
