diff -Naur libXrender-0.9.3-old/include/X11/extensions/Xrender.h libXrender-0.9.3-new/include/X11/extensions/Xrender.h
--- libXrender-0.9.3-old/include/X11/extensions/Xrender.h	2007-01-02 19:05:07.000000000 -0800
+++ libXrender-0.9.3-new/include/X11/extensions/Xrender.h	2007-09-04 18:35:08.000000000 -0700
@@ -25,13 +25,13 @@
 #ifndef _XRENDER_H_
 #define _XRENDER_H_
 
-#include <X11/extensions/render.h>
-
 #include <X11/Xlib.h>
 #include <X11/Xfuncproto.h>
 #include <X11/Xosdefs.h>
 #include <X11/Xutil.h>
 
+#include <X11/extensions/render.h>
+
 typedef struct {
     short   red;
     short   redMask;
