diff -Naur dbus-1.2.10-old/dbus/dbus-sysdeps-util-unix.c dbus-1.2.10-new/dbus/dbus-sysdeps-util-unix.c
--- dbus-1.2.10-old/dbus/dbus-sysdeps-util-unix.c	2008-12-19 11:35:23.000000000 -0800
+++ dbus-1.2.10-new/dbus/dbus-sysdeps-util-unix.c	2008-12-20 10:34:43.000000000 -0800
@@ -36,6 +36,7 @@
 #include <signal.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <syslog.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <sys/stat.h>
@@ -1227,4 +1228,4 @@
   _dbus_string_free (&cmdline);
   _dbus_string_free (&path);
   return FALSE;
-}
\ No newline at end of file
+}
