diff -Naur dbus-1.4.8-old/bus/Makefile.am dbus-1.4.8-new/bus/Makefile.am
--- dbus-1.4.8-old/bus/Makefile.am	2011-03-28 05:36:11.000000000 -0700
+++ dbus-1.4.8-new/bus/Makefile.am	2011-04-08 14:22:29.000000000 -0700
@@ -1,7 +1,7 @@
 
 configdir=$(sysconfdir)/dbus-1
 
-INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDBUS_COMPILATION
 
@@ -88,7 +88,7 @@
 	$(EFENCE)					\
 	$(DBUS_BUS_LIBS)
 
-dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 
 LAUNCH_HELPER_SOURCES=				\
 	$(XML_SOURCES)				\
diff -Naur dbus-1.4.8-old/bus/Makefile.in dbus-1.4.8-new/bus/Makefile.in
--- dbus-1.4.8-old/bus/Makefile.in	2011-04-08 06:30:42.000000000 -0700
+++ dbus-1.4.8-new/bus/Makefile.in	2011-04-08 14:22:29.000000000 -0700
@@ -514,7 +514,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 configdir = $(sysconfdir)/dbus-1
-INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDBUS_COMPILATION
 
@@ -582,7 +582,7 @@
 	$(EFENCE)					\
 	$(DBUS_BUS_LIBS)
 
-dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 LAUNCH_HELPER_SOURCES = \
 	$(XML_SOURCES)				\
 	config-parser-common.c			\
