diff -Naur glibc-2.4-old/elf/ldd.bash.in glibc-2.4-new/elf/ldd.bash.in
--- glibc-2.4-old/elf/ldd.bash.in	2006-01-01 11:15:56.000000000 -0800
+++ glibc-2.4-new/elf/ldd.bash.in	2006-05-25 06:19:58.000000000 -0700
@@ -110,7 +110,7 @@
 # environments where the executed program might not have permissions
 # to write to the console/tty.  But only bash 3.x supports the pipefail
 # option, and we don't bother to handle the case for older bash versions.
-if set -o pipefail 2> /dev/null; then
+if `set -o pipefail 2> /dev/null ; echo $?`; then
   try_trace() {
     eval $add_env '"$@"' | cat
   }
