diff -Naur glibc-2.7-old/elf/ldd.bash.in glibc-2.7-new/elf/ldd.bash.in
--- glibc-2.7-old/elf/ldd.bash.in	2007-11-14 19:31:39.000000000 -0800
+++ glibc-2.7-new/elf/ldd.bash.in	2007-11-14 19:32:48.000000000 -0800
@@ -110,7 +110,7 @@
 # environments where the executed program might not have permissions
 # to write to the console/tty.  But only bash 3.x supports the pipefail
 # option, and we don't bother to handle the case for older bash versions.
-if set -o pipefail 2> /dev/null; then
+if `set -o pipefail 2> /dev/null ; echo $?`; then
   try_trace() {
     eval $add_env '"$@"' | cat
   }
