diff -Naur glibc-2.6.1-old/configure glibc-2.6.1-new/configure
--- glibc-2.6.1-old/configure	2007-07-31 06:46:12.000000000 -0700
+++ glibc-2.6.1-new/configure	2007-12-29 17:29:41.000000000 -0800
@@ -4563,7 +4563,7 @@
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  SYSINCLUDES="-nostdinc -isystem $ccheaders -isystem $ccheaders-fixed \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&5` &&
diff -Naur glibc-2.6.1-old/configure.in glibc-2.6.1-new/configure.in
--- glibc-2.6.1-old/configure.in	2007-03-20 05:11:23.000000000 -0700
+++ glibc-2.6.1-new/configure.in	2007-12-29 17:28:24.000000000 -0800
@@ -912,7 +912,7 @@
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  SYSINCLUDES="-nostdinc -isystem $ccheaders -isystem $ccheaders-fixed \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&AS_MESSAGE_LOG_FD` &&
