diff -Naur glibc-2.7-old/sysdeps/i386/i586/memcpy_chk.S glibc-2.7-new/sysdeps/i386/i586/memcpy_chk.S
--- glibc-2.7-old/sysdeps/i386/i586/memcpy_chk.S	1969-12-31 16:00:00.000000000 -0800
+++ glibc-2.7-new/sysdeps/i386/i586/memcpy_chk.S	2007-11-14 19:48:28.000000000 -0800
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/memcpy_chk.S>
diff -Naur glibc-2.7-old/sysdeps/i386/i586/mempcpy_chk.S glibc-2.7-new/sysdeps/i386/i586/mempcpy_chk.S
--- glibc-2.7-old/sysdeps/i386/i586/mempcpy_chk.S	1969-12-31 16:00:00.000000000 -0800
+++ glibc-2.7-new/sysdeps/i386/i586/mempcpy_chk.S	2007-11-14 19:48:28.000000000 -0800
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/mempcpy_chk.S>
diff -Naur glibc-2.7-old/sysdeps/i386/i586/memset_chk.S glibc-2.7-new/sysdeps/i386/i586/memset_chk.S
--- glibc-2.7-old/sysdeps/i386/i586/memset_chk.S	1969-12-31 16:00:00.000000000 -0800
+++ glibc-2.7-new/sysdeps/i386/i586/memset_chk.S	2007-11-14 19:48:28.000000000 -0800
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/memset_chk.S>
