diff -Naur glibc-2.10.1-old/elf/ldd.bash.in glibc-2.10.1-new/elf/ldd.bash.in
--- glibc-2.10.1-old/elf/ldd.bash.in	2009-05-23 08:24:23.000000000 -0700
+++ glibc-2.10.1-new/elf/ldd.bash.in	2009-05-23 08:26:00.000000000 -0700
@@ -112,7 +112,7 @@
 # environments where the executed program might not have permissions
 # to write to the console/tty.  But only bash 3.x supports the pipefail
 # option, and we don't bother to handle the case for older bash versions.
-if set -o pipefail 2> /dev/null; then
+if `set -o pipefail 2> /dev/null ; echo $?`; then
   try_trace() {
     eval $add_env '"$@"' | cat
   }
