diff -Naur ffmpeg-23418.31303-old/libswscale/libswscale.v ffmpeg-23418.31303-new/libswscale/libswscale.v
--- ffmpeg-23418.31303-old/libswscale/libswscale.v	2010-06-01 15:24:47.000000000 -0700
+++ ffmpeg-23418.31303-new/libswscale/libswscale.v	2010-06-01 15:24:47.000000000 -0700
@@ -1,4 +1,4 @@
 LIBSWSCALE_$MAJOR {
-        global: swscale_*; sws_*; ff_*;
+        global: swscale_*; sws_*; ff_*; palette8tobgr15; palette8tobgr16; palette8torgb15; palette8torgb16; palette8topacked24; palette8topacked32; rgb24tobgr24; rgb32tobgr32; yuv422ptoyuy2; yv12toyuy2;
         local: *;
 };
