diff -Naur ffmpeg-8cf9a09-old/libswscale/libswscale.v ffmpeg-8cf9a09-new/libswscale/libswscale.v
--- ffmpeg-8cf9a09-old/libswscale/libswscale.v	2011-03-03 21:07:31.000000000 -0800
+++ ffmpeg-8cf9a09-new/libswscale/libswscale.v	2011-03-03 21:07:32.000000000 -0800
@@ -1,4 +1,4 @@
 LIBSWSCALE_$MAJOR {
-        global: swscale_*; sws_*; ff_*;
+        global: swscale_*; sws_*; ff_*; palette8tobgr15; palette8tobgr16; palette8torgb15; palette8torgb16; palette8topacked24; palette8topacked32; rgb24tobgr24; rgb32tobgr32; yuv422ptoyuy2; yv12toyuy2;
         local: *;
 };
