diff -Naur ffmpeg-16182.28158-old/configure ffmpeg-16182.28158-new/configure
--- ffmpeg-16182.28158-old/configure	2008-12-16 18:38:07.000000000 -0800
+++ ffmpeg-16182.28158-new/configure	2008-12-16 18:38:51.000000000 -0800
@@ -86,6 +86,7 @@
   echo "  --enable-pthreads        use pthreads [no]"
   echo "  --enable-w32threads      use Win32 threads [no]"
   echo "  --enable-x11grab         enable X11 grabbing [no]"
+  echo "  --enable-vdpau           disable VDPAU support [no]"
   echo "  --enable-xvmc            enable XvMC support [no]"
   echo
   echo "External library support:"
@@ -776,6 +777,7 @@
     swscale
     vhook
     x11grab
+    vdpau
     xvmc
     zlib
 "
@@ -1016,6 +1018,7 @@
 zmbv_encoder_select="zlib"
 
 # external libraries
+h264_vdpau_decoder_deps="vdpau"
 libamr_nb_decoder_deps="libamr_nb"
 libamr_nb_encoder_deps="libamr_nb"
 libamr_wb_decoder_deps="libamr_wb"
@@ -1038,6 +1041,9 @@
 libx264_encoder_deps="libx264"
 libxvid_encoder_deps="libxvid"
 mpeg4aac_decoder_deps="libfaad"
+mpeg_vdpau_decoder_deps="vdpau"
+wmv3_vdpau_decoder_deps="vdpau"
+vc1_vdpau_decoder_deps="vdpau"
 
 # demuxers / muxers
 ac3_demuxer_deps="ac3_parser"
@@ -1894,6 +1900,7 @@
                         die "ERROR: libx264 version must be >= 0.65."; }
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
+enabled vdpau      && require  libvdpau vdpau/vdpau_x11.h vdp_device_create_x11 -lvdpau
 
 # libdc1394 check
 if enabled libdc1394; then
