diff -Naur ffmpeg-16006.28089-old/configure ffmpeg-16006.28089-new/configure
--- ffmpeg-16006.28089-old/configure	2008-12-13 07:30:06.000000000 -0800
+++ ffmpeg-16006.28089-new/configure	2008-12-13 07:31:42.000000000 -0800
@@ -146,6 +146,7 @@
   echo "  --disable-mmi            disable MMI optimizations"
   echo "  --disable-neon           disable neon optimizations"
   echo "  --disable-vis            disable VIS optimizations"
+  echo "  --disable-vdpau          disable VDPAU support"
   echo "  --disable-network        disable network support [no]"
   echo "  --disable-ipv6           disable IPv6 support [no]"
   echo "  --disable-mpegaudio-hp   faster (but less accurate)"
@@ -814,6 +815,7 @@
     neon
     ssse3
     vis
+    vdpau
 '
 
 HAVE_LIST="
@@ -1012,6 +1014,7 @@
 zmbv_encoder_select="zlib"
 
 # external libraries
+h264_vdpau_decoder_deps="vdpau"
 libamr_nb_decoder_deps="libamr_nb"
 libamr_nb_encoder_deps="libamr_nb"
 libamr_wb_decoder_deps="libamr_wb"
@@ -1034,6 +1037,9 @@
 libx264_encoder_deps="libx264"
 libxvid_encoder_deps="libxvid"
 mpeg4aac_decoder_deps="libfaad"
+mpeg_vdpau_decoder_deps="vdpau"
+wmv3_vdpau_decoder_deps="vdpau"
+vc1_vdpau_decoder_deps="vdpau"
 
 # demuxers / muxers
 ac3_demuxer_deps="ac3_parser"
@@ -1890,6 +1896,7 @@
                         die "ERROR: libx264 version must be >= 0.65."; }
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
+enabled vdpau      && require  libvdpau vdpau/vdpau_x11.h vdp_device_create_x11 -lvdpau
 
 # libdc1394 check
 if enabled libdc1394; then
