diff -Naur ffmpeg-15948.28048-old/configure ffmpeg-15948.28048-new/configure
--- ffmpeg-15948.28048-old/configure	2008-11-28 11:21:06.000000000 -0800
+++ ffmpeg-15948.28048-new/configure	2008-11-28 11:21:48.000000000 -0800
@@ -814,6 +814,7 @@
     mmx2
     neon
     ssse3
+    xvmc
     vis
 '
 
@@ -1896,6 +1897,7 @@
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
 enabled vdpau      && require  libvdpau vdpau/vdpau_x11.h vdp_device_create_x11 -lvdpau
+enabled xvmc       && check_header X11/extensions/XvMClib.h
 
 # libdc1394 check
 if enabled libdc1394; then
