diff -Naur ffmpeg-1.0.1-old/libavutil/common.h ffmpeg-1.0.1-new/libavutil/common.h
--- ffmpeg-1.0.1-old/libavutil/common.h	2012-12-05 18:20:27.000000000 -0800
+++ ffmpeg-1.0.1-new/libavutil/common.h	2012-12-05 18:21:37.000000000 -0800
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
