diff -Naur ffmpeg-24684.31912-old/libswscale/libswscale.v ffmpeg-24684.31912-new/libswscale/libswscale.v
--- ffmpeg-24684.31912-old/libswscale/libswscale.v	2010-08-03 07:49:26.000000000 -0700
+++ ffmpeg-24684.31912-new/libswscale/libswscale.v	2010-08-03 07:49:27.000000000 -0700
@@ -1,4 +1,4 @@
 LIBSWSCALE_$MAJOR {
-        global: swscale_*; sws_*; ff_*;
+        global: swscale_*; sws_*; ff_*; palette8tobgr15; palette8tobgr16; palette8torgb15; palette8torgb16; palette8topacked24; palette8topacked32; rgb24tobgr24; rgb32tobgr32; yuv422ptoyuy2; yv12toyuy2;
         local: *;
 };
