diff -Naur ffmpeg-1.0.1-old/libswscale/libswscale.v ffmpeg-1.0.1-new/libswscale/libswscale.v
--- ffmpeg-1.0.1-old/libswscale/libswscale.v	2012-12-05 18:21:25.000000000 -0800
+++ ffmpeg-1.0.1-new/libswscale/libswscale.v	2012-12-05 18:21:57.000000000 -0800
@@ -1,4 +1,4 @@
 LIBSWSCALE_$MAJOR {
-        global: swscale_*; sws_*; ff_*;
+        global: swscale_*; sws_*; ff_*; palette8tobgr15; palette8tobgr16; palette8torgb15; palette8torgb16; palette8topacked24; palette8topacked32; rgb24tobgr24; rgb32tobgr32; yuv422ptoyuy2; yv12toyuy2;
         local: *;
 };
