diff -Naur ffmpeg-797ef67-old/libavutil/common.h ffmpeg-797ef67-new/libavutil/common.h
--- ffmpeg-797ef67-old/libavutil/common.h	2011-03-24 20:22:58.000000000 -0700
+++ ffmpeg-797ef67-new/libavutil/common.h	2011-03-24 20:23:01.000000000 -0700
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
