diff -Naur strace-4.5.19-old/configure strace-4.5.19-new/configure
--- strace-4.5.19-old/configure	2009-10-21 10:41:11.000000000 -0700
+++ strace-4.5.19-new/configure	2009-12-05 05:51:48.000000000 -0800
@@ -5880,7 +5880,7 @@
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_compile "$LINENO" "$ac_header" "$as_ac_Header" "#include <stddef.h>
-#include <linux/socket.h>
+#include <sys/socket.h>
 "
 eval as_val=\$$as_ac_Header
    if test "x$as_val" = x""yes; then :
diff -Naur strace-4.5.19-old/configure.ac strace-4.5.19-new/configure.ac
--- strace-4.5.19-old/configure.ac	2009-10-12 12:54:43.000000000 -0700
+++ strace-4.5.19-new/configure.ac	2009-12-05 05:51:37.000000000 -0800
@@ -258,7 +258,7 @@
 ], [], [])
 AC_CHECK_HEADERS([linux/icmp.h linux/in6.h linux/netlink.h linux/if_packet.h],
                  [], [], [#include <stddef.h>
-#include <linux/socket.h>])
+#include <sys/socket.h>])
 AC_CHECK_HEADERS([asm/sigcontext.h], [], [], [#include <signal.h>])
 AC_CHECK_TYPES([struct sigcontext_struct,
 		struct sigcontext],,, [#include <signal.h>])
