diff -Naur mythstream_mythtv-r21640-old/mythstream/main.cpp mythstream_mythtv-r21640-new/mythstream/main.cpp
--- mythstream_mythtv-r21640-old/mythstream/main.cpp	2009-09-02 09:40:49.000000000 -0700
+++ mythstream_mythtv-r21640-new/mythstream/main.cpp	2011-02-03 17:10:56.000000000 -0800
@@ -36,7 +36,7 @@
 #include <mythtv/mythmedia.h>
 #include <mythtv/mythpluginapi.h>
 #include <mythtv/libmythui/myththemedmenu.h>
-#include <mythtv/libmythdb/mythversion.h>
+#include <mythtv/libmythbase/mythversion.h>
 #include <mythtv/libmythui/mythuihelper.h>
 
 using namespace std;
