diff -Naur xorg-server-1.0.1-old/render/mitri.c xorg-server-1.0.1-new/render/mitri.c
--- xorg-server-1.0.1-old/render/mitri.c	2005-07-03 00:02:08.000000000 -0700
+++ xorg-server-1.0.1-new/render/mitri.c	2006-05-03 05:41:16.000000000 -0700
@@ -145,7 +145,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     for (tri = tris; npoint >= 3; npoint--, points++, tri++)
@@ -177,7 +177,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     first = points++;
