diff -Naur xine-ui-0.99.9-old/configure.ac xine-ui-0.99.9-new/configure.ac
--- xine-ui-0.99.9-old/configure.ac	2014-06-17 14:13:56.000000000 -0700
+++ xine-ui-0.99.9-new/configure.ac	2015-06-28 15:40:59.385992816 -0700
@@ -25,7 +25,7 @@
 dnl evilness
 
 AC_CONFIG_SRCDIR([src/xitk/main.c])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_LIBOBJ_DIR([src/common])
 
 AM_INIT_AUTOMAKE([dist-xz no-dist-gzip])
