diff -Naur xine-ui-0.99.5-old/configure.ac xine-ui-0.99.5-new/configure.ac
--- xine-ui-0.99.5-old/configure.ac	2007-09-04 15:30:57.000000000 -0700
+++ xine-ui-0.99.5-new/configure.ac	2007-09-04 15:39:12.000000000 -0700
@@ -386,11 +386,13 @@
   CFLAGS="$CFLAGS $XFT_CFLAGS"
   LIBS="$LIBS $XFT_LIBS"
   AC_TRY_LINK([
+#include <X11/Xlib.h>
 #include <X11/Xft/Xft.h>], [ XftFontClose(0, 0); return 1; ], 
     [
       AC_DEFINE(WITH_XFT, 1, [font antialiasing support])
       AC_MSG_CHECKING([Xft UTF-8 support])
       AC_TRY_LINK([
+#include <X11/Xlib.h>
 #include <X11/Xft/Xft.h>
         ], [ 
 XftDrawStringUtf8(0, 0, 0, 0, 0, 0, 0); return 0; 
