diff -Naur NVIDIA-Linux-x86-180.06-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-180.06-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-180.06-pkg1-old/usr/src/nv/conftest.sh	2008-11-08 12:53:12.000000000 -0800
+++ NVIDIA-Linux-x86-180.06-pkg1-new/usr/src/nv/conftest.sh	2008-11-16 15:03:22.000000000 -0800
@@ -1143,7 +1143,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
