diff -Naur libsndfile-1.0.22-old/configure.ac libsndfile-1.0.22-new/configure.ac
--- libsndfile-1.0.22-old/configure.ac	2010-10-03 17:39:04.000000000 -0700
+++ libsndfile-1.0.22-new/configure.ac	2010-10-04 09:25:00.000000000 -0700
@@ -17,9 +17,6 @@
 
 AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
 
-dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
-
 AC_LANG([C])
 
 #------------------------------------------------------------------------------------
diff -Naur libsndfile-1.0.22-old/Makefile.am libsndfile-1.0.22-new/Makefile.am
--- libsndfile-1.0.22-old/Makefile.am	2010-08-27 19:28:06.000000000 -0700
+++ libsndfile-1.0.22-new/Makefile.am	2010-10-04 09:25:00.000000000 -0700
@@ -2,6 +2,8 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
+ACLOCAL_AMFLAGS = -I M4
+
 if BUILD_OCTAVE_MOD
 octave_dir = Octave
 endif
