diff -Naur libsndfile-1.0.21-old/configure.ac libsndfile-1.0.21-new/configure.ac
--- libsndfile-1.0.21-old/configure.ac	2009-12-12 22:46:01.000000000 -0800
+++ libsndfile-1.0.21-new/configure.ac	2010-08-17 16:03:56.000000000 -0700
@@ -16,9 +16,6 @@
 
 AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
 
-dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
-
 AC_LANG([C])
 
 #------------------------------------------------------------------------------------
diff -Naur libsndfile-1.0.21-old/Makefile.am libsndfile-1.0.21-new/Makefile.am
--- libsndfile-1.0.21-old/Makefile.am	2009-09-27 04:40:52.000000000 -0700
+++ libsndfile-1.0.21-new/Makefile.am	2010-08-17 16:03:48.000000000 -0700
@@ -2,6 +2,8 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
+ACLOCAL_AMFLAGS = -I M4
+
 if BUILD_OCTAVE_MOD
 octave_dir = Octave
 endif
