diff -Naur lesstif-0.94.4-old/configure.in lesstif-0.94.4-new/configure.in
--- lesstif-0.94.4-old/configure.in	2005-03-31 11:19:11.000000000 -0800
+++ lesstif-0.94.4-new/configure.in	2006-01-06 10:40:19.000000000 -0800
@@ -79,7 +79,7 @@
 AH_TEMPLATE([HAVE_GETHOSTNAME], [Define if gethostname() is available (may be in non-standard libs])
 
 AC_MSG_CHECKING(where to store autoconf macro files)
-ACLOCALDIR=`aclocal --print-ac-dir`
+ACLOCALDIR=${datadir}/aclocal
 AM_CONDITIONAL(Aclocal, test x$ACLOCALDIR != x )
 AC_MSG_RESULT($ACLOCALDIR)
 
