diff -Naur lesstif-0.94.4-old/configure lesstif-0.94.4-new/configure
--- lesstif-0.94.4-old/configure	2005-03-31 11:44:45.000000000 -0800
+++ lesstif-0.94.4-new/configure	2005-11-14 11:04:04.000000000 -0800
@@ -2150,16 +2150,11 @@
 
 echo "$as_me:$LINENO: checking where to store autoconf macro files" >&5
 echo $ECHO_N "checking where to store autoconf macro files... $ECHO_C" >&6
-ACLOCALDIR=`aclocal --print-ac-dir`
+ACLOCALDIR=${datadir}/aclocal
 
 
-if test x$ACLOCALDIR != x ; then
-  Aclocal_TRUE=
-  Aclocal_FALSE='#'
-else
-  Aclocal_TRUE='#'
-  Aclocal_FALSE=
-fi
+Aclocal_TRUE=
+Aclocal_FALSE='#'
 
 echo "$as_me:$LINENO: result: $ACLOCALDIR" >&5
 echo "${ECHO_T}$ACLOCALDIR" >&6
