diff -Naur xtrans-1.2.4-old/Xtranssock.c xtrans-1.2.4-new/Xtranssock.c
--- xtrans-1.2.4-old/Xtranssock.c	2009-07-30 16:06:14.000000000 -0700
+++ xtrans-1.2.4-new/Xtranssock.c	2009-07-31 18:10:24.000000000 -0700
@@ -2470,7 +2470,7 @@
 Xtransport	TRANS(SocketUNIXFuncs) = {
 	/* Socket Interface */
 	"unix",
-#if !defined(LOCALCONN) && !defined(HAVE_ABSTRACT_SOCKETS)
+#if !defined(LOCALCONN)
         TRANS_ALIAS,
 #else
 	0,
