diff -Naur vdpau-video-0.4.1-old/src/Makefile.am vdpau-video-0.4.1-new/src/Makefile.am
--- vdpau-video-0.4.1-old/src/Makefile.am	2009-09-10 08:24:10.000000000 -0700
+++ vdpau-video-0.4.1-new/src/Makefile.am	2009-09-10 21:19:10.000000000 -0700
@@ -64,6 +64,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 
diff -Naur vdpau-video-0.4.1-old/src/Makefile.in vdpau-video-0.4.1-new/src/Makefile.in
--- vdpau-video-0.4.1-old/src/Makefile.in	2009-09-10 08:27:09.000000000 -0700
+++ vdpau-video-0.4.1-new/src/Makefile.in	2009-09-10 21:19:36.000000000 -0700
@@ -578,6 +578,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
