diff -Naur xf86-video-intel-2.17.0-old/configure.ac xf86-video-intel-2.17.0-new/configure.ac
--- xf86-video-intel-2.17.0-old/configure.ac	2011-11-16 15:14:23.000000000 -0800
+++ xf86-video-intel-2.17.0-new/configure.ac	2011-11-18 02:34:05.000000000 -0800
@@ -58,6 +58,7 @@
 # Initialize libtool
 LT_PREREQ([2.2])
 LT_INIT([disable-static])
+PKG_PROG_PKG_CONFIG
 
 # Are we in a git checkout?
 dot_git=no
@@ -178,7 +179,8 @@
 PKG_CHECK_MODULES(DRI2, [dri2proto >= 2.6],, DRI2=no)
 PKG_CHECK_MODULES(PCIACCESS, [pciaccess >= 0.10])
 
-sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 if test "x$enable_dri" != "xno"; then
         save_CFLAGS="$CFLAGS"
