diff -Naur linux-2.6.36.3-old/drivers/usb/core/hcd.c linux-2.6.36.3-new/drivers/usb/core/hcd.c
--- linux-2.6.36.3-old/drivers/usb/core/hcd.c	2011-01-26 14:57:40.000000000 -0800
+++ linux-2.6.36.3-new/drivers/usb/core/hcd.c	2011-01-26 14:59:05.000000000 -0800
@@ -1982,6 +1982,8 @@
 
 	usb_lock_device(udev);
 	usb_remote_wakeup(udev);
+	if (hcd->state == HC_STATE_RUNNING)
+		clear_bit(HCD_FLAG_WAKEUP_PENDING, &hcd->flags);
 	usb_unlock_device(udev);
 }
 
