diff -Naur libXvMC-1.0.1-old/src/Makefile.am libXvMC-1.0.1-new/src/Makefile.am
--- libXvMC-1.0.1-old/src/Makefile.am	2005-12-02 20:41:51.000000000 -0800
+++ libXvMC-1.0.1-new/src/Makefile.am	2006-04-07 19:17:49.000000000 -0700
@@ -15,7 +15,7 @@
 libXvMCW_la_LIBADD = @XVMC_LIBS@
 libXvMCW_la_CFLAGS =				\
 	$(AM_CFLAGS)				\
-	-DXVMC_CONFIGDIR=XConfigDir		\
+	-DXVMC_CONFIGDIR=@sysconfdir@/X11	\
 	-DXVMC_SOVERSION=\".1.0\"		\
 	-DXV_SOVERSION=\".1.0\"
 
diff -Naur libXvMC-1.0.1-old/src/Makefile.in libXvMC-1.0.1-new/src/Makefile.in
--- libXvMC-1.0.1-old/src/Makefile.in	2005-12-20 23:18:14.000000000 -0800
+++ libXvMC-1.0.1-new/src/Makefile.in	2006-04-07 19:18:35.000000000 -0700
@@ -158,7 +158,7 @@
 libXvMCW_la_LIBADD = @XVMC_LIBS@
 libXvMCW_la_CFLAGS = \
 	$(AM_CFLAGS)				\
-	-DXVMC_CONFIGDIR=XConfigDir		\
+	-DXVMC_CONFIGDIR=@sysconfdir@/X11	\
 	-DXVMC_SOVERSION=\".1.0\"		\
 	-DXV_SOVERSION=\".1.0\"
 
