diff -Naur ifplugd-0.28-old/src/interface.c ifplugd-0.28-new/src/interface.c
--- ifplugd-0.28-old/src/interface.c	2005-01-04 16:25:31.000000000 -0800
+++ ifplugd-0.28-new/src/interface.c	2007-12-23 13:34:01.000000000 -0800
@@ -23,11 +23,11 @@
 #endif
 
 #include <linux/sockios.h>
-#include <linux/if_ether.h>
+#include <netinet/if_ether.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
+// #include <linux/if.h>
 #include <syslog.h>
 #include <string.h>
 #include <errno.h>
