diff -Naur mythplugins-0.19-old/mythmusic/mythmusic/flacdecoder.cpp mythplugins-0.19-new/mythmusic/mythmusic/flacdecoder.cpp
--- mythplugins-0.19-old/mythmusic/mythmusic/flacdecoder.cpp	2007-02-23 21:30:27.000000000 -0800
+++ mythplugins-0.19-new/mythmusic/mythmusic/flacdecoder.cpp	2007-02-23 21:31:17.000000000 -0800
@@ -17,7 +17,7 @@
 
 #include <qtimer.h>
 
-static StreamDecoderReadStatus flacread(const StreamDecoder *decoder, FLAC__byte bufferp[], unsigned *bytes, void *client_data)
+static StreamDecoderReadStatus flacread(const StreamDecoder *decoder, FLAC__byte bufferp[], bytesSize *bytes, void *client_data)
 {
     decoder = decoder;
 
@@ -263,7 +263,7 @@
 
     decoder = decoder_new();
     decoder_set_md5_checking(decoder, false);
-    decoder_setup(decoder, flacread, flacseek, flactell, flaclength, \
+    decoder_setup(decoder, flacread, flacseek, flactell, flaclength,
                         flaceof, flacwrite, flacmetadata, flacerror, this);
 
     freq = 0;
diff -Naur mythplugins-0.19-old/mythmusic/mythmusic/flacdecoder.h mythplugins-0.19-new/mythmusic/mythmusic/flacdecoder.h
--- mythplugins-0.19-old/mythmusic/mythmusic/flacdecoder.h	2007-02-23 21:30:27.000000000 -0800
+++ mythplugins-0.19-new/mythmusic/mythmusic/flacdecoder.h	2007-02-23 21:31:17.000000000 -0800
@@ -41,6 +41,7 @@
   #define decoder_seek_absolute FLAC__seekable_stream_decoder_seek_absolute
   #define decoder_process_single FLAC__seekable_stream_decoder_process_single
   #define decoder_get_state FLAC__seekable_stream_decoder_get_state
+  #define bytesSize unsigned
 #else
   /* FLAC 1.1.3 and up */
   #define NEWFLAC
@@ -69,6 +70,7 @@
   #define decoder_seek_absolute FLAC__stream_decoder_seek_absolute
   #define decoder_process_single FLAC__stream_decoder_process_single
   #define decoder_get_state FLAC__stream_decoder_get_state
+  #define bytesSize size_t
 #endif
 
 #include "decoder.h"
