diff -Naur mythtv-0.25-20130811-b5adf03-old/mythtv/libs/libmythtv/vaapicontext.h mythtv-0.25-20130811-b5adf03-new/mythtv/libs/libmythtv/vaapicontext.h
--- mythtv-0.25-20130811-b5adf03-old/mythtv/libs/libmythtv/vaapicontext.h	2015-06-14 12:45:06.000000000 -0700
+++ mythtv-0.25-20130811-b5adf03-new/mythtv/libs/libmythtv/vaapicontext.h	2015-08-02 14:34:17.908004106 -0700
@@ -4,6 +4,10 @@
 extern "C" {
 #include "libavcodec/vaapi.h"
 }
+#include "va/va_version.h"
+#if VA_CHECK_VERSION(0,34,0)
+#include "va/va_compat.h"
+#endif
 #include "va/va_x11.h"
 #include "va/va_glx.h"
 #include "videocolourspace.h"
