diff -Naur mythtv-0.25-20120716-4e44650-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-0.25-20120716-4e44650-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-0.25-20120716-4e44650-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-07-18 19:51:46.000000000 -0700
+++ mythtv-0.25-20120716-4e44650-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-07-18 19:51:47.000000000 -0700
@@ -2415,6 +2415,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
