diff -Naur mythtv-0.25-20120508-f22e556-old/mythplugins/configure mythtv-0.25-20120508-f22e556-new/mythplugins/configure
--- mythtv-0.25-20120508-f22e556-old/mythplugins/configure	2012-05-09 04:06:49.000000000 -0700
+++ mythtv-0.25-20120508-f22e556-new/mythplugins/configure	2012-05-09 04:06:51.000000000 -0700
@@ -574,7 +574,7 @@
 fi
 
 #Test if fftw_lib3 is available
-if enabled fftw &&! check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3f -lfftw3 ; then
+if enabled fftw &&! check_lib2 fftw3.h fftwf_init_threads -lfftw3f_threads -lfftw3f ; then
     disable fftw
 fi
 
@@ -898,7 +898,7 @@
     if enabled fftw; then
         echo "        FFTW v.3       support will be included in MythMusic"
         echo "#define FFTW3_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
-        echo "LIBS += -lfftw3 -lm" >> \
+        echo "LIBS += -lfftw3f -lm" >> \
              ./mythmusic/mythmusic/config.pro
     else
         echo "        FFTW           support will not be included in MythMusic"
diff -Naur mythtv-0.25-20120508-f22e556-old/mythtv/configure mythtv-0.25-20120508-f22e556-new/mythtv/configure
--- mythtv-0.25-20120508-f22e556-old/mythtv/configure	2012-05-09 04:06:49.000000000 -0700
+++ mythtv-0.25-20120508-f22e556-new/mythtv/configure	2012-05-09 04:06:51.000000000 -0700
@@ -4137,7 +4137,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3f -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftwf_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 # ass_flush_events was added in 0.9.10
