diff -Naur mythtv-0.25-20121123-76d9c3e-old/mythtv/programs/mythfrontend/main.cpp mythtv-0.25-20121123-76d9c3e-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-0.25-20121123-76d9c3e-old/mythtv/programs/mythfrontend/main.cpp	2012-11-25 08:24:06.000000000 -0800
+++ mythtv-0.25-20121123-76d9c3e-new/mythtv/programs/mythfrontend/main.cpp	2012-11-25 08:24:07.000000000 -0800
@@ -1690,6 +1690,18 @@
                 QString("NetworkControl failed to bind to port %1.")
                    .arg(port));
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
 #ifdef __linux__
 #ifdef CONFIG_BINDINGS_PYTHON
