diff -Naur libX11-1.3.3-old/configure.ac libX11-1.3.3-new/configure.ac
--- libX11-1.3.3-old/configure.ac	2010-01-16 11:41:21.000000000 -0800
+++ libX11-1.3.3-new/configure.ac	2010-01-16 17:47:13.000000000 -0800
@@ -114,6 +114,10 @@
 		# Unix domain sockets are faster now.
 		DEFAULT_LOCAL_TRANS="UNIX_TRANS,LOCAL_TRANS,TCP_TRANS"
 		;;
+	linux*)
+		# LOCAL_TRANS is used for abstract sockets.
+		DEFAULT_LOCAL_TRANS="LOCAL_TRANS,UNIX_TRANS,TCP_TRANS"
+		;;
 	*)
 		if test "$LOCALCONN" = "yes"; then
 			DEFAULT_LOCAL_TRANS="LOCAL_TRANS,UNIX_TRANS,TCP_TRANS"
diff -Naur libX11-1.3.3-old/src/ConnDis.c libX11-1.3.3-new/src/ConnDis.c
--- libX11-1.3.3-old/src/ConnDis.c	2010-01-14 17:11:36.000000000 -0800
+++ libX11-1.3.3-new/src/ConnDis.c	2010-01-16 17:47:13.000000000 -0800
@@ -92,7 +92,7 @@
 #ifdef UNIXCONN
 # define UNIX_TRANS	"unix"
 #endif
-#if defined(LOCALCONN) || defined(OS2PIPECONN)
+#if defined(LOCALCONN) || defined(OS2PIPECONN) || defined(UNIXCONN)
 # define LOCAL_TRANS	"local"
 #endif
 
