diff -Naur xf86-video-ati-c72b732a2f5597cd72bd4ebe207799ae07f2681e-old/configure.ac xf86-video-ati-c72b732a2f5597cd72bd4ebe207799ae07f2681e-new/configure.ac
--- xf86-video-ati-c72b732a2f5597cd72bd4ebe207799ae07f2681e-old/configure.ac	2008-12-08 11:57:33.000000000 -0800
+++ xf86-video-ati-c72b732a2f5597cd72bd4ebe207799ae07f2681e-new/configure.ac	2008-12-08 11:59:13.000000000 -0800
@@ -38,6 +38,7 @@
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+PKG_PROG_PKG_CONFIG
 
 if test "x$GCC" = "xyes"; then
 	CPPFLAGS="$CPPFLAGS -Wall"
@@ -77,7 +78,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.2 xproto fontsproto $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 
