diff -Naur glib-2.28.2-old/configure.ac glib-2.28.2-new/configure.ac
--- glib-2.28.2-old/configure.ac	2011-03-04 17:20:22.000000000 -0800
+++ glib-2.28.2-new/configure.ac	2011-03-08 11:51:43.000000000 -0800
@@ -858,6 +858,7 @@
 AC_CHECK_HEADERS([sys/vfs.h sys/mount.h sys/vmount.h sys/statfs.h sys/statvfs.h])
 AC_CHECK_HEADERS([mntent.h sys/mnttab.h sys/vfstab.h sys/mntctl.h sys/sysctl.h fstab.h])
 AC_CHECK_HEADERS([sys/uio.h sys/mkdev.h])
+AC_CHECK_HEADERS([linux/magic.h])
 
 # check for structure fields
 AC_CHECK_MEMBERS([struct stat.st_mtimensec, struct stat.st_mtim.tv_nsec, struct stat.st_atimensec, struct stat.st_atim.tv_nsec, struct stat.st_ctimensec, struct stat.st_ctim.tv_nsec])
diff -Naur glib-2.28.2-old/glib/gfileutils.c glib-2.28.2-new/glib/gfileutils.c
--- glib-2.28.2-old/glib/gfileutils.c	2011-01-05 13:44:06.000000000 -0800
+++ glib-2.28.2-new/glib/gfileutils.c	2011-03-08 11:51:43.000000000 -0800
@@ -54,7 +54,9 @@
 #include "glibintl.h"
 
 #ifdef __linux__ /* for btrfs check */
+#ifdef HAVE_LINUX_MAGIC_H
 #include <linux/magic.h>
+#endif
 #include <sys/vfs.h>
 #endif
 
