diff -Naur mythstream-0.18_1-old/libs/harvester.cpp mythstream-0.18_1-new/libs/harvester.cpp
--- mythstream-0.18_1-old/libs/harvester.cpp	2009-08-13 22:56:54.000000000 -0700
+++ mythstream-0.18_1-new/libs/harvester.cpp	2009-08-13 22:57:55.000000000 -0700
@@ -33,6 +33,7 @@
    http://lists.trolltech.com/qt-interest/2004-05/thread00280-0.html
 */
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qnetwork.h>
diff -Naur mythstream-0.18_1-old/libs/recorder.cpp mythstream-0.18_1-new/libs/recorder.cpp
--- mythstream-0.18_1-old/libs/recorder.cpp	2009-08-13 22:56:54.000000000 -0700
+++ mythstream-0.18_1-new/libs/recorder.cpp	2009-08-13 22:57:55.000000000 -0700
@@ -21,6 +21,7 @@
 
 // recorder experimental
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qnetwork.h>
diff -Naur mythstream-0.18_1-old/libs/storageconfig.cpp mythstream-0.18_1-new/libs/storageconfig.cpp
--- mythstream-0.18_1-old/libs/storageconfig.cpp	2009-08-13 22:57:07.000000000 -0700
+++ mythstream-0.18_1-new/libs/storageconfig.cpp	2009-08-13 22:57:55.000000000 -0700
@@ -22,6 +22,7 @@
 
 // development snapshot, many todo's
 
+#include <cstdlib>
 #include <iostream>
 #include <qlayout.h>
 #include <qcursor.h>
diff -Naur mythstream-0.18_1-old/libs/streamconfig.cpp mythstream-0.18_1-new/libs/streamconfig.cpp
--- mythstream-0.18_1-old/libs/streamconfig.cpp	2009-08-13 22:57:07.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamconfig.cpp	2009-08-13 22:57:55.000000000 -0700
@@ -22,6 +22,7 @@
 
 // development snapshot, many todo's
 
+#include <cstdlib>
 #include <iostream>
 #include <qlayout.h>
 #include <qcursor.h>
diff -Naur mythstream-0.18_1-old/libs/streamstatus.cpp mythstream-0.18_1-new/libs/streamstatus.cpp
--- mythstream-0.18_1-old/libs/streamstatus.cpp	2009-08-13 22:57:07.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstatus.cpp	2009-08-13 22:57:55.000000000 -0700
@@ -21,6 +21,7 @@
 
 // development snapshot, many todo's
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qapplication.h>
diff -Naur mythstream-0.18_1-old/libs/streamstorage.cpp mythstream-0.18_1-new/libs/streamstorage.cpp
--- mythstream-0.18_1-old/libs/streamstorage.cpp	2009-08-13 22:56:54.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstorage.cpp	2009-08-13 22:57:55.000000000 -0700
@@ -19,6 +19,7 @@
  *
  * ============================================================ */
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qnetwork.h>
