diff -Naur mythmusic-0.17-old/mythmusic/main.cpp mythmusic-0.17-new/mythmusic/main.cpp
--- mythmusic-0.17-old/mythmusic/main.cpp	2005-02-09 13:35:13.000000000 -0800
+++ mythmusic-0.17-new/mythmusic/main.cpp	2005-04-10 17:47:46.879173101 -0700
@@ -385,8 +385,6 @@
     REG_KEY("Music", "DELETE", "Delete track from playlist", "D");
     REG_KEY("Music", "NEXTTRACK", "Move to the next track", ">,.,Z,End");
     REG_KEY("Music", "PREVTRACK", "Move to the previous track", ",,<,Q,Home");
-    REG_KEY("Music", "FFWD", "Fast forward", "PgDown");
-    REG_KEY("Music", "RWND", "Rewind", "PgUp");
     REG_KEY("Music", "PAUSE", "Pause/Start playback", "P");
     REG_KEY("Music", "STOP", "Stop playback", "O");
     REG_KEY("Music", "VOLUMEDOWN", "Volume down", "[,{,F10");
diff -Naur mythmusic-0.17-old/mythmusic/playbackbox.cpp mythmusic-0.17-new/mythmusic/playbackbox.cpp
--- mythmusic-0.17-old/mythmusic/playbackbox.cpp	2005-02-07 11:52:23.000000000 -0800
+++ mythmusic-0.17-new/mythmusic/playbackbox.cpp	2005-04-10 17:54:05.175186331 -0700
@@ -383,6 +383,10 @@
                 music_tree_list->forceLastBin();
                 music_tree_list->refresh();
             }
+            else if (action == "PAGEUP")
+                music_tree_list->pageUp();
+            else if (action == "PAGEDOWN")
+                music_tree_list->pageDown();
             else
                 handled = false;
         }
