diff -Naur mythplugins-0.22-25328-old/mythvideo/mythvideo/videodlg.h mythplugins-0.22-25328-new/mythvideo/mythvideo/videodlg.h
--- mythplugins-0.22-25328-old/mythvideo/mythvideo/videodlg.h	2010-08-17 07:10:46.000000000 -0700
+++ mythplugins-0.22-25328-new/mythvideo/mythvideo/videodlg.h	2010-09-21 15:05:01.000000000 -0700
@@ -146,7 +146,7 @@
     QString RemoteImageCheck(QString host, QString filename);
     QString GetCoverImage(MythGenericTree *node);
     QString GetFirstImage(MythGenericTree *node, QString type,
-                          QString gpnode = NULL, int levels = 0);
+                          QString gpnode = QString(), int levels = 0);
     QString GetImageFromFolder(Metadata *metadata);
     QString GetScreenshot(MythGenericTree *node);
     QString GetBanner(MythGenericTree *node);
