diff -Naur gtk+-2.16.0-old/Makefile.am gtk+-2.16.0-new/Makefile.am
--- gtk+-2.16.0-old/Makefile.am	2009-03-13 06:41:13.000000000 -0700
+++ gtk+-2.16.0-new/Makefile.am	2009-03-16 21:17:27.000000000 -0700
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
diff -Naur gtk+-2.16.0-old/Makefile.in gtk+-2.16.0-new/Makefile.in
--- gtk+-2.16.0-old/Makefile.in	2009-03-13 07:28:58.000000000 -0700
+++ gtk+-2.16.0-new/Makefile.in	2009-03-16 21:17:27.000000000 -0700
@@ -520,7 +520,7 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
