diff -Naur gtk+-2.14.6-old/Makefile.am gtk+-2.14.6-new/Makefile.am
--- gtk+-2.14.6-old/Makefile.am	2008-12-15 10:39:04.000000000 -0800
+++ gtk+-2.14.6-new/Makefile.am	2008-12-16 19:03:08.000000000 -0800
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
diff -Naur gtk+-2.14.6-old/Makefile.in gtk+-2.14.6-new/Makefile.in
--- gtk+-2.14.6-old/Makefile.in	2008-12-15 10:41:42.000000000 -0800
+++ gtk+-2.14.6-new/Makefile.in	2008-12-16 19:03:08.000000000 -0800
@@ -520,7 +520,7 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
