diff -Naur mythplugins-0.21-16317-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-0.21-16317-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-0.21-16317-old/mythmusic/mythmusic/aacdecoder.cpp	2008-02-29 13:21:46.000000000 -0800
+++ mythplugins-0.21-16317-new/mythmusic/mythmusic/aacdecoder.cpp	2008-02-29 13:40:38.000000000 -0800
@@ -16,6 +16,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstdlib>
 #include <qobject.h>
 #include <qiodevice.h>
 #include <qfile.h>
diff -Naur mythplugins-0.21-16317-old/mythmusic/mythmusic/flacdecoder.h mythplugins-0.21-16317-new/mythmusic/mythmusic/flacdecoder.h
--- mythplugins-0.21-16317-old/mythmusic/mythmusic/flacdecoder.h	2008-02-29 13:21:46.000000000 -0800
+++ mythplugins-0.21-16317-new/mythmusic/mythmusic/flacdecoder.h	2008-02-29 13:40:38.000000000 -0800
@@ -1,7 +1,7 @@
 #ifndef FLACDECODER_H_
 #define FLACDECODER_H_
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-0.21-16317-old/mythmusic/mythmusic/flacencoder.h mythplugins-0.21-16317-new/mythmusic/mythmusic/flacencoder.h
--- mythplugins-0.21-16317-old/mythmusic/mythmusic/flacencoder.h	2008-02-29 13:21:46.000000000 -0800
+++ mythplugins-0.21-16317-new/mythmusic/mythmusic/flacencoder.h	2008-02-29 13:40:38.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include <qstring.h>
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 
 #include <FLAC/export.h>
 #if !defined(FLAC_API_VERSION_CURRENT) || FLAC_API_VERSION_CURRENT <= 7
diff -Naur mythplugins-0.21-16317-old/mythmusic/mythmusic/metaioflacvorbiscomment.h mythplugins-0.21-16317-new/mythmusic/mythmusic/metaioflacvorbiscomment.h
--- mythplugins-0.21-16317-old/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-02-29 13:21:46.000000000 -0800
+++ mythplugins-0.21-16317-new/mythmusic/mythmusic/metaioflacvorbiscomment.h	2008-02-29 13:40:38.000000000 -0800
@@ -3,7 +3,7 @@
 
 #include "metaio.h"
 
-#define HAVE_INTTYPES_H
+#define HAVE_INTTYPES_H 1
 #include <FLAC/all.h>
 
 class MetaIOFLACVorbisComment : public MetaIO
diff -Naur mythplugins-0.21-16317-old/mythmusic/mythmusic/metaiomp4.cpp mythplugins-0.21-16317-new/mythmusic/mythmusic/metaiomp4.cpp
--- mythplugins-0.21-16317-old/mythmusic/mythmusic/metaiomp4.cpp	2008-02-29 13:21:46.000000000 -0800
+++ mythplugins-0.21-16317-new/mythmusic/mythmusic/metaiomp4.cpp	2008-02-29 13:40:38.000000000 -0800
@@ -1,4 +1,5 @@
 #include <iostream>
+#include <cstdlib>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
