diff -Naur mythplugins-0.21-20453-old/mythzoneminder/mythzmserver/main.cpp mythplugins-0.21-20453-new/mythzoneminder/mythzmserver/main.cpp
--- mythplugins-0.21-20453-old/mythzoneminder/mythzmserver/main.cpp	2009-04-26 11:40:23.000000000 -0700
+++ mythplugins-0.21-20453-new/mythzoneminder/mythzmserver/main.cpp	2009-04-26 11:43:07.000000000 -0700
@@ -18,6 +18,7 @@
 #include <unistd.h>
 #include <cstdlib>
 #include <cstring>
+#include <cstdio>
 #include <map>
 #include <sys/types.h>
 #include <sys/socket.h>
diff -Naur mythplugins-0.21-20453-old/mythzoneminder/mythzmserver/zmserver.cpp mythplugins-0.21-20453-new/mythzoneminder/mythzmserver/zmserver.cpp
--- mythplugins-0.21-20453-old/mythzoneminder/mythzmserver/zmserver.cpp	2009-04-26 11:40:23.000000000 -0700
+++ mythplugins-0.21-20453-new/mythzoneminder/mythzmserver/zmserver.cpp	2009-04-26 11:43:07.000000000 -0700
@@ -17,6 +17,7 @@
 #include <iostream>
 #include <cstdlib>
 #include <cstring>
+#include <cstdio>
 #include <errno.h>
 #include <sys/socket.h>
 #include <fcntl.h>
