diff -Naur libcdio-0.79-old/include/cdio++/iso9660.hpp libcdio-0.79-new/include/cdio++/iso9660.hpp
--- libcdio-0.79-old/include/cdio++/iso9660.hpp	2006-03-07 13:04:01.000000000 -0800
+++ libcdio-0.79-new/include/cdio++/iso9660.hpp	2008-01-08 17:31:10.000000000 -0800
@@ -29,6 +29,7 @@
 
 #include <cdio/iso9660.h>
 #include <cdio++/cdio.hpp>
+#include <malloc.h>
 #include <string.h>
 #include <vector>		// vector class library
 using namespace std;
