diff -Naur libcdio-0.81-old/src/Makefile.am libcdio-0.81-new/src/Makefile.am
--- libcdio-0.81-old/src/Makefile.am	2008-08-31 13:47:07.000000000 -0700
+++ libcdio-0.81-new/src/Makefile.am	2008-10-28 06:21:12.000000000 -0700
@@ -90,6 +90,6 @@
              iso-info.help2man iso-read.help2man $(GETOPT_C) getopt.h \
              $(man_MANS)
 
-$(man_MANS): %.1: % %.help2man 
+$(man_MANS): %.1: %.help2man
 	-$(HELP2MAN) --opt-include=$<.help2man --no-info --output=$@ ./$<
 MOSTLYCLEANFILES = $(man_MANS)
