diff -Naur libcdio-0.80-old/src/Makefile.am libcdio-0.80-new/src/Makefile.am
--- libcdio-0.80-old/src/Makefile.am	2008-03-15 09:21:28.000000000 -0700
+++ libcdio-0.80-new/src/Makefile.am	2008-03-15 22:52:33.000000000 -0700
@@ -39,7 +39,7 @@
 CDDB_LIBS=@CDDB_LIBS@
 CDDA_PLAYER_LIBS=@CDDA_PLAYER_LIBS@
 
-$(man_MANS): %.1: % %.help2man
+$(man_MANS): %.1: %.help2man
 	-$(HELP2MAN) --opt-include=$<.help2man --no-info --output=$@ ./$<
 MOSTLYCLEANFILES = $(man_MANS)
 
diff -Naur libcdio-0.80-old/src/Makefile.in libcdio-0.80-new/src/Makefile.in
--- libcdio-0.80-old/src/Makefile.in	2008-03-15 10:45:35.000000000 -0700
+++ libcdio-0.80-new/src/Makefile.in	2008-03-15 22:53:27.000000000 -0700
@@ -831,7 +831,7 @@
 	uninstall-binPROGRAMS uninstall-man uninstall-man1
 
 
-$(man_MANS): %.1: % %.help2man
+$(man_MANS): %.1: %.help2man
 	-$(HELP2MAN) --opt-include=$<.help2man --no-info --output=$@ ./$<
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
