diff -Naur myththemes-addons-trunk-16831-old/configure myththemes-addons-trunk-16831-new/configure
--- myththemes-addons-trunk-16831-old/configure	2008-03-28 09:47:04.000000000 -0700
+++ myththemes-addons-trunk-16831-new/configure	2008-03-28 09:54:32.000000000 -0700
@@ -5,10 +5,14 @@
 
 prefix="/usr/local"
 
+qmake="qmake"
+
 for opt do
   case "$opt" in
   --prefix=*) prefix=`echo $opt | cut -d '=' -f 2`
   ;;
+  --qmake=*) qmake=`echo $opt | cut -d '=' -f 2`
+  ;;
   esac
 done
 
@@ -19,6 +23,7 @@
 Options: [defaults in brackets after descriptions]
 Generic options:
   --prefix=PREFIX          MythTV install location PREFIX [$prefix]"
+  --qmake=QMAKE            use specified qmake [$qmake]
   --help                   print this message
 EOF
 exit 1
@@ -36,4 +41,18 @@
   exit
 fi
 
-qmake themes.pro
+is_qmake4()
+{
+    $1 --version 2>&1 | grep -q "Qt version 4"
+}
+ 
+if ! is_qmake4 $qmake; then
+    if is_qmake4 qmake-qt4; then
+        qmake=qmake-qt4
+    else
+        echo "qmake for Qt4 not found. Please specify the correct qmake with --qmake="
+        exit 1
+    fi
+fi
+
+${qmake} QMAKE=$(which ${qmake}) themes.pro
