diff -Naur mythtv-0.19-old/programs/mythlcdserver/lcdprocclient.cpp mythtv-0.19-new/programs/mythlcdserver/lcdprocclient.cpp
--- mythtv-0.19-old/programs/mythlcdserver/lcdprocclient.cpp	2005-12-09 21:11:52.000000000 -0800
+++ mythtv-0.19-new/programs/mythlcdserver/lcdprocclient.cpp	2006-08-06 19:30:16.000000000 -0700
@@ -623,7 +623,7 @@
     // version LCDd is using:
 
     if (serverVersion.startsWith ("CVS-current") ||
-            serverVersion.startsWith ("0.5dev"))
+            serverVersion.startsWith ("0.5"))
     {
         // Latest CVS versions of LCDd has priorities switched
         pVersion = LCD_VERSION_5;
