diff -Naur mythdvd-0.17-old/mythdvd/settings.cpp mythdvd-0.17-new/mythdvd/settings.cpp
--- mythdvd-0.17-old/mythdvd/settings.cpp	2005-01-30 08:10:49.000000000 -0800
+++ mythdvd-0.17-new/mythdvd/settings.cpp	2005-03-17 08:29:56.314571802 -0800
@@ -86,7 +86,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("DVDPlayerCommand");
     gc->setLabel(QObject::tr("DVD Player Command"));
-    gc->setValue("mplayer dvd:// -dvd-device %d -fs -zoom -vo xv");
+    gc->setValue("xine -pfhq --no-splash dvd://");
     gc->setHelpText(QObject::tr("This can be any command to launch a DVD player "
                     "(e.g. MPlayer, ogle, etc.). If present, %d will "
                     "be substituted for the DVD device (e.g. /dev/dvd)."));
@@ -98,7 +98,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("VCDPlayerCommand");
     gc->setLabel(QObject::tr("VCD Player Command"));
-    gc->setValue("mplayer vcd:// -cdrom-device %d -fs -zoom -vo xv");
+    gc->setValue("xine -pfhq --no-splash cdrom://");
     gc->setHelpText(QObject::tr("This can be any command to launch a VCD player "
                     "(e.g. MPlayer, xine, etc.). If present, %d will "
                     "be substituted for the VCD device (e.g. /dev/cdrom)."));
