diff -Naur dhcp-4.0.0-old/common/options.c dhcp-4.0.0-new/common/options.c
--- dhcp-4.0.0-old/common/options.c	2007-11-30 13:51:43.000000000 -0800
+++ dhcp-4.0.0-new/common/options.c	2007-12-26 17:08:45.000000000 -0800
@@ -1265,8 +1265,8 @@
 						       cfg_options,
 						       scope, tmp, MDL);
 			} else if (vuname) {
-			    name.data = (unsigned char *)s;
-			    name.len = strlen (s);
+			    name.data = (unsigned char *)vuname;
+			    name.len = strlen (vuname);
 			}
 		    } else {
 			name.data = (unsigned char *)s;
