diff -Naur makedepend-1.0.1-old/main.c makedepend-1.0.1-new/main.c
--- makedepend-1.0.1-old/main.c	2006-08-01 18:38:38.000000000 -0700
+++ makedepend-1.0.1-new/main.c	2007-04-29 20:18:14.000000000 -0700
@@ -404,10 +404,12 @@
 		}
 	    }
 #else /* !__UNIXOS2__, does not use INCLUDEDIR at all */
+#ifdef INCLUDEDIR
 	    if (incp >= includedirs + MAXDIRS)
 		fatalerr("Too many -I flags.\n");
 	    *incp++ = INCLUDEDIR;
 #endif
+#endif
 
 #ifdef EXTRAINCDIR
 	    if (incp >= includedirs + MAXDIRS)
diff -Naur makedepend-1.0.1-old/Makefile.am makedepend-1.0.1-new/Makefile.am
--- makedepend-1.0.1-old/Makefile.am	2006-08-01 18:38:38.000000000 -0700
+++ makedepend-1.0.1-new/Makefile.am	2007-04-29 20:18:14.000000000 -0700
@@ -14,7 +14,7 @@
                      def.h \
                      imakemdep.h
 
-makedepend_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\" -DINCLUDEDIR=\"/usr/include\"
+makedepend_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\"
 
 EXTRA_DIST = autogen.sh makedepend.man
 
diff -Naur makedepend-1.0.1-old/Makefile.in makedepend-1.0.1-new/Makefile.in
--- makedepend-1.0.1-old/Makefile.in	2007-03-21 12:28:24.000000000 -0700
+++ makedepend-1.0.1-new/Makefile.in	2007-04-29 20:19:24.000000000 -0700
@@ -197,7 +197,7 @@
                      def.h \
                      imakemdep.h
 
-makedepend_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\" -DINCLUDEDIR=\"/usr/include\"
+makedepend_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\"
 EXTRA_DIST = autogen.sh makedepend.man
 
 # Man page
