diff -Naur stella-2.7-old/configure stella-2.7-new/configure
--- stella-2.7-old/configure	2008-03-30 17:59:30.000000000 -0700
+++ stella-2.7-new/configure	2009-01-19 10:10:33.000000000 -0800
@@ -30,10 +30,10 @@
 _build_speakjet=no
 
 # more defaults
-_ranlib=ranlib
+_ranlib="${RANLIB:-'ranlib'}"
 _install=install
-_ar="ar cru"
-_strip=strip
+_ar="${AR:-'ar'} cru"
+_strip="${STRIP:-'strip'}"
 _mkdir="mkdir -p"
 _echo=printf
 _cat=cat
@@ -100,7 +100,7 @@
 	# In cross-compiling mode, we cannot run the result
 	eval "$1 -o tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp 2> /dev/null" && rm -f tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp
 else
-	eval "$1 -o tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp 2> /dev/null" && eval "./tmp_cxx_compiler 2> /dev/null" && rm -f tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp
+	eval "$1 -o tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp 2> /dev/null" && rm -f tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp
 fi
 }
 
diff -Naur stella-2.7-old/Makefile stella-2.7-new/Makefile
--- stella-2.7-old/Makefile	2009-01-11 13:31:21.000000000 -0800
+++ stella-2.7-new/Makefile	2009-01-19 10:10:33.000000000 -0800
@@ -163,7 +163,7 @@
 
 install: all
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
-	$(INSTALL) -c -s -m 755 "$(srcdir)/stella$(EXEEXT)" "$(DESTDIR)$(BINDIR)/stella$(EXEEXT)"
+	$(INSTALL) -c -m 755 "$(srcdir)/stella$(EXEEXT)" "$(DESTDIR)$(BINDIR)/stella$(EXEEXT)"
 	$(INSTALL) -d "$(DESTDIR)$(DOCDIR)"
 	$(INSTALL) -c -m 644 "$(srcdir)/Announce.txt" "$(srcdir)/Changes.txt" "$(srcdir)/Copyright.txt" "$(srcdir)/License.txt" "$(srcdir)/README-SDL.txt" "$(srcdir)/Readme.txt" "$(srcdir)/Todo.txt" "$(srcdir)/docs/index.html" "$(srcdir)/docs/debugger.html" "$(DESTDIR)$(DOCDIR)/"
 	$(INSTALL) -d "$(DESTDIR)$(DOCDIR)/graphics"
