diff -Naur mythtv-master-20110124-b4f1039-45d6b63-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110124-b4f1039-45d6b63-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110124-b4f1039-45d6b63-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-01-25 10:36:32.000000000 -0800
+++ mythtv-master-20110124-b4f1039-45d6b63-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-01-25 10:38:46.000000000 -0800
@@ -1244,6 +1244,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
