diff -Naur mythtv-master-20110617-54d1f0e-8eeb80e-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110617-54d1f0e-8eeb80e-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110617-54d1f0e-8eeb80e-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-06-18 11:30:33.000000000 -0700
+++ mythtv-master-20110617-54d1f0e-8eeb80e-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-06-18 12:21:33.000000000 -0700
@@ -1302,6 +1302,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
