diff -Naur mythtv-master-20110302-624c23c-ecaeffc-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110302-624c23c-ecaeffc-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110302-624c23c-ecaeffc-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-03-03 07:20:26.000000000 -0800
+++ mythtv-master-20110302-624c23c-ecaeffc-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-03-03 07:46:41.000000000 -0800
@@ -1263,6 +1263,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
