diff -Naur mythtv-master-20141222-5b386a0-old/mythtv/external/zeromq/configure.in mythtv-master-20141222-5b386a0-new/mythtv/external/zeromq/configure.in
--- mythtv-master-20141222-5b386a0-old/mythtv/external/zeromq/configure.in	2015-06-11 00:52:28.000000000 -0700
+++ mythtv-master-20141222-5b386a0-new/mythtv/external/zeromq/configure.in	2015-06-11 01:01:12.605000002 -0700
@@ -10,7 +10,7 @@
 
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
-AM_CONFIG_HEADER(src/platform.hpp)
+AC_CONFIG_HEADERS(src/platform.hpp)
 AM_INIT_AUTOMAKE(tar-ustar dist-zip foreign)
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
