diff -Naur mythtv-master-20121113-5e1c14e-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-master-20121113-5e1c14e-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-master-20121113-5e1c14e-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-11-14 14:45:10.000000000 -0800
+++ mythtv-master-20121113-5e1c14e-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-11-14 14:45:11.000000000 -0800
@@ -2432,6 +2432,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
