diff -Naur mythtv-master-20110511-30bdb49-8eeb80e-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110511-30bdb49-8eeb80e-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110511-30bdb49-8eeb80e-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-05-12 10:58:57.000000000 -0700
+++ mythtv-master-20110511-30bdb49-8eeb80e-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-05-12 10:59:05.000000000 -0700
@@ -1280,6 +1280,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
