diff -Naur mythtv-master-20121214-021b2e5-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20121214-021b2e5-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20121214-021b2e5-old/mythtv/programs/mythfrontend/main.cpp	2012-12-14 22:45:46.000000000 -0800
+++ mythtv-master-20121214-021b2e5-new/mythtv/programs/mythfrontend/main.cpp	2012-12-14 22:45:48.000000000 -0800
@@ -1658,6 +1658,18 @@
                 QString("NetworkControl failed to bind to port %1.")
                    .arg(port));
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
 #ifdef __linux__
 #ifdef CONFIG_BINDINGS_PYTHON
