diff -Naur mythtv-master-20110603-feafcfc-8eeb80e-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110603-feafcfc-8eeb80e-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110603-feafcfc-8eeb80e-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-06-04 10:11:15.000000000 -0700
+++ mythtv-master-20110603-feafcfc-8eeb80e-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-06-04 10:27:30.000000000 -0700
@@ -1288,6 +1288,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
