diff -Naur mythtv-master-20110214-919163d-1f6d53a-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110214-919163d-1f6d53a-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110214-919163d-1f6d53a-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-02-14 21:36:13.000000000 -0800
+++ mythtv-master-20110214-919163d-1f6d53a-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-02-14 21:38:38.000000000 -0800
@@ -1270,6 +1270,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
