diff -Naur mythtv-master-20120406-83671d5-old/mythplugins/mythgallery/mythgallery/galleryfilterdlg.cpp mythtv-master-20120406-83671d5-new/mythplugins/mythgallery/mythgallery/galleryfilterdlg.cpp
--- mythtv-master-20120406-83671d5-old/mythplugins/mythgallery/mythgallery/galleryfilterdlg.cpp	2012-04-07 14:52:25.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythplugins/mythgallery/mythgallery/galleryfilterdlg.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -15,6 +15,8 @@
  *
  * ============================================================ */
 
+#include <unistd.h>
+
 using namespace std;
 
 // Qt headers
diff -Naur mythtv-master-20120406-83671d5-old/mythplugins/mythmusic/mythmusic/importmusic.cpp mythtv-master-20120406-83671d5-new/mythplugins/mythmusic/mythmusic/importmusic.cpp
--- mythtv-master-20120406-83671d5-old/mythplugins/mythmusic/mythmusic/importmusic.cpp	2012-04-07 14:52:25.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythplugins/mythmusic/mythmusic/importmusic.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // qt
 #include <QDir>
 #include <QFontMetrics>
diff -Naur mythtv-master-20120406-83671d5-old/mythplugins/mythmusic/mythmusic/metadata.cpp mythtv-master-20120406-83671d5-new/mythplugins/mythmusic/mythmusic/metadata.cpp
--- mythtv-master-20120406-83671d5-old/mythplugins/mythmusic/mythmusic/metadata.cpp	2012-04-07 14:52:25.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythplugins/mythmusic/mythmusic/metadata.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // qt
 #include <QApplication>
 #include <QRegExp>
diff -Naur mythtv-master-20120406-83671d5-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20120406-83671d5-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20120406-83671d5-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2012-04-07 14:52:25.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -14,6 +14,8 @@
  * ============================================================ */
 
 
+#include <unistd.h>
+
 #include <iostream>
 #include <cstdlib>
 #include <cstring>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmyth/mythcontext.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmyth/mythcontext.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmyth/mythcontext.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmyth/mythcontext.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QDir>
 #include <QFileInfo>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmyth/programinfoupdater.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmyth/programinfoupdater.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmyth/programinfoupdater.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmyth/programinfoupdater.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // MythTV headers
 #include "programinfoupdater.h"
 #include "mthreadpool.h"
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/httpcomms.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/httpcomms.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/httpcomms.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/httpcomms.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/lcddevice.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/lcddevice.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/lcddevice.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/lcddevice.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -7,6 +7,8 @@
     (c) 2002, 2003 Thor Sigvaldason, Dan Morphis and Isaac Richards
 */
 
+#include <unistd.h>
+
 // ANSI C headers
 #include <cstdlib>
 #include <cmath>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/mythdownloadmanager.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/mythdownloadmanager.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/mythdownloadmanager.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/mythdownloadmanager.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // qt
 #include <QCoreApplication>
 #include <QRunnable>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/mythsocket.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/mythsocket.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/mythsocket.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/mythsocket.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 #include <cassert>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/mythsocketthread.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/mythsocketthread.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/mythsocketthread.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/mythsocketthread.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythbase/remotefile.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythbase/remotefile.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythmetadata/metadatafactory.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythmetadata/metadatafactory.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythmetadata/metadatafactory.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythmetadata/metadatafactory.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QApplication>
 #include <QList>
 #include <QUrl>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,4 @@
+#include <unistd.h>
 
 using namespace std;
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/audioinputoss.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/audioinputoss.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/audioinputoss.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -20,6 +20,8 @@
  * 02110-1301, USA.
  */
 
+#include <unistd.h>
+
 #include "mythconfig.h"
 #if HAVE_SYS_SOUNDCARD_H
     #include <sys/soundcard.h>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/firewiredevice.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/firewiredevice.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/firewiredevice.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/firewiredevice.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -4,6 +4,8 @@
  *  Distributed as part of MythTV under GPL v2 and later.
  */
 
+#include <unistd.h>
+
 // C++ headers
 #include <algorithm>
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/httplivestream.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/httplivestream.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/httplivestream.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/httplivestream.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -19,6 +19,8 @@
  *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <unistd.h>
+
 #include <stdio.h>
 
 #include <QDir>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/mythbdplayer.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/mythbdplayer.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/mythbdplayer.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/mythbdplayer.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "bdringbuffer.h"
 #include "avformatdecoderbd.h"
 #include "mythbdplayer.h"
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/mythcommflagplayer.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/mythcommflagplayer.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/mythcommflagplayer.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/mythcommflagplayer.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/mythdvdplayer.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/mythdvdplayer.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythtv/mythdvdplayer.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythtv/mythdvdplayer.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "dvdringbuffer.h"
 #include "DetectLetterbox.h"
 #include "audiooutput.h"
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythui/mythrender_vdpau.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythui/mythrender_vdpau.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythui/mythrender_vdpau.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythui/mythrender_vdpau.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "math.h"
 
 #include <QSize>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythui/mythuiwebbrowser.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythui/mythuiwebbrowser.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythui/mythuiwebbrowser.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythui/mythuiwebbrowser.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -7,6 +7,8 @@
  *
  */
 
+#include <unistd.h>
+
 #include "mythuiwebbrowser.h"
 
 // qt
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/mmulticastsocketdevice.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/mmulticastsocketdevice.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/mmulticastsocketdevice.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/mmulticastsocketdevice.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -10,6 +10,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include <errno.h>
 #include "mythconfig.h"
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/upnpdevice.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/upnpdevice.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/upnpdevice.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/upnpdevice.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -10,6 +10,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include "upnp.h"
 #include "upnpdevice.h"
 #include "httpcomms.h"
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/upnptasknotify.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/upnptasknotify.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/upnptasknotify.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/upnptasknotify.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -10,6 +10,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 // ANSI C headers
 #include <cstdlib>
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/upnptasksearch.cpp mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/upnptasksearch.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/libs/libmythupnp/upnptasksearch.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/libs/libmythupnp/upnptasksearch.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -10,6 +10,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include <compat.h>
 #include <stdlib.h>
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/programs/mythbackend/main.cpp mythtv-master-20120406-83671d5-new/mythtv/programs/mythbackend/main.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/programs/mythbackend/main.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/programs/mythbackend/main.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #ifndef _WIN32
 #include <QCoreApplication>
 #else
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/backendconnectionmanager.cpp mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/backendconnectionmanager.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/backendconnectionmanager.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/backendconnectionmanager.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QRunnable>
 #include <QString>
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/services/frontend.cpp mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/services/frontend.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/services/frontend.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/services/frontend.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QKeyEvent>
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/upnpscanner.cpp mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/upnpscanner.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/upnpscanner.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/upnpscanner.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QTextCodec>
 
diff -Naur mythtv-master-20120406-83671d5-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-master-20120406-83671d5-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/programs/mythtranscode/transcode.cpp	2012-04-07 14:52:26.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/programs/mythtranscode/transcode.cpp	2012-04-07 14:55:48.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <fcntl.h>
 #include <math.h>
 #include <iostream>
