diff -Naur xf86-video-vmware-10.14.1-old/configure.ac xf86-video-vmware-10.14.1-new/configure.ac
--- xf86-video-vmware-10.14.1-old/configure.ac	2006-12-06 16:47:15.000000000 -0800
+++ xf86-video-vmware-10.14.1-new/configure.ac	2008-01-22 15:15:47.000000000 -0800
@@ -38,6 +38,7 @@
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+PKG_PROG_PKG_CONFIG
 
 AH_TOP([#include "xorg-server.h"])
 
@@ -55,7 +56,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.0.99.901 xproto fontsproto $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 
