diff -Naur dropbear-0.50-old/options.h dropbear-0.50-new/options.h
--- dropbear-0.50-old/options.h	2007-08-08 08:39:37.000000000 -0700
+++ dropbear-0.50-new/options.h	2007-10-02 22:43:42.000000000 -0700
@@ -51,7 +51,7 @@
 #define DROPBEAR_SMALL_CODE
 
 /* Enable X11 Forwarding - server only */
-#define ENABLE_X11FWD
+/*#define ENABLE_X11FWD*/
 
 /* Enable TCP Fowarding */
 /* 'Local' is "-L" style (client listening port forwarded via server)
