diff -Naur dropbear-0.50-old/options.h dropbear-0.50-new/options.h
--- dropbear-0.50-old/options.h	2007-08-08 08:39:37.000000000 -0700
+++ dropbear-0.50-new/options.h	2007-10-02 20:31:55.000000000 -0700
@@ -132,8 +132,8 @@
  * but there's an interface via a PAM module - don't bother using it otherwise.
  * You can't enable both PASSWORD and PAM. */
 
-#define ENABLE_SVR_PASSWORD_AUTH
-/*#define ENABLE_SVR_PAM_AUTH */ /* requires ./configure --enable-pam */
+/*#define ENABLE_SVR_PASSWORD_AUTH*/
+#define ENABLE_SVR_PAM_AUTH  /* requires ./configure --enable-pam */
 #define ENABLE_SVR_PUBKEY_AUTH
 
 #define ENABLE_CLI_PASSWORD_AUTH
