diff -Naur freetype-2.1.10-old/builds/unix/freetype-config.in freetype-2.1.10-new/builds/unix/freetype-config.in
--- freetype-2.1.10-old/builds/unix/freetype-config.in	2005-06-04 14:58:48.000000000 -0700
+++ freetype-2.1.10-new/builds/unix/freetype-config.in	2005-07-03 17:02:24.000000000 -0700
@@ -16,7 +16,6 @@
 libdir=@libdir@
 enable_shared=@enable_shared@
 wl=@wl@
-hardcode_libdir_flag_spec='@hardcode_libdir_flag_spec@'
 
 usage()
 {
@@ -128,9 +127,9 @@
 fi
 
 if test "$echo_cflags" = "yes" ; then
-  cflags="-I$includedir/freetype2"
+  cflags="-I${DESTDIR}/$includedir/freetype2"
   if test "$includedir" != "/usr/include" ; then
-    echo $cflags -I$includedir
+    echo $cflags -I${DESTDIR}/$includedir
   else
     echo $cflags
   fi
@@ -138,12 +137,9 @@
 
 if test "$echo_libs" = "yes" ; then
   rpath=
-  if test "$enable_shared" = "yes" ; then
-    eval "rpath=\"$hardcode_libdir_flag_spec\""
-  fi
   libs="-lfreetype @LIBZ@"
   if test "$libdir" != "/usr/lib" && test "$libdir" != "/usr/lib64"; then
-    echo -L$libdir $rpath $libs
+    echo -L${DESTDIR}/$libdir $libs
   else
     echo $libs
   fi
@@ -151,7 +147,7 @@
 
 if test "$echo_libtool" = "yes" ; then
   convlib="libfreetype.la"
-  echo $libdir/$convlib
+  echo ${DESTDIR}/$libdir/$convlib
 fi
 
 # EOF
