diff -Naur unionfs-1.5pre-cvs200701042308-old/Makefile unionfs-1.5pre-cvs200701042308-new/Makefile
--- unionfs-1.5pre-cvs200701042308-old/Makefile	2007-01-04 20:09:48.000000000 -0800
+++ unionfs-1.5pre-cvs200701042308-new/Makefile	2007-01-07 08:54:18.000000000 -0800
@@ -8,6 +8,7 @@
 # This is where the make install target will put stuff
 PREFIX   = /usr/local
 MANDIR   = ${PREFIX}/man
+SBINDIR  = ${PREFIX}/sbin
 MODPREFIX=
 VERSION  = 1.5pre-cvs200701042308
 SUP_MAJOR= 2
@@ -54,7 +55,7 @@
 	@echo ""
 
 unionfs.ko: FRC
-	make -C ${LINUXSRC} SUBDIRS=$(PWD) FISTDEVMK=$(PWD)/fistdev.mk modules
+	make -C ${LINUXSRC} SUBDIRS=`pwd` FISTDEVMK=`pwd`/fistdev.mk modules
 
 FRC:
 
@@ -71,7 +72,7 @@
 	ctags $^
 
 clean:
-	-make -C ${LINUXSRC} SUBDIRS=$(PWD) FISTDEVMK=$(PWD)/fistdev.mk clean
+	-make -C ${LINUXSRC} SUBDIRS=`pwd` FISTDEVMK=`pwd`/fistdev.mk clean
 	rm -f ${unionfs-objs} ${obj-m} ${obj-m:.o=.ko} *.d .*.flags \#* *~
 	rm -f TAGS tags
 	rm -f unionfs-${VERSION}.tar${COMPEXT}
@@ -101,13 +102,11 @@
 install-mod: unionfs.ko
 	mkdir -p ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs
 	cp unionfs.ko ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs
-	-/sbin/depmod -a
 
 install: install-utils install-mod
 
 uninstall:
 	rm ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs.ko
-	-/sbin/depmod -a
 	rm /usr/include/unionfs.h
 	rm ${MANDIR}/man8/unionimap.8
 	rm ${MANDIR}/man8/unionctl.8
diff -Naur unionfs-1.5pre-cvs200701042308-old/utils/Makefile unionfs-1.5pre-cvs200701042308-new/utils/Makefile
--- unionfs-1.5pre-cvs200701042308-old/utils/Makefile	2007-01-04 20:09:48.000000000 -0800
+++ unionfs-1.5pre-cvs200701042308-new/utils/Makefile	2007-01-07 08:54:18.000000000 -0800
@@ -24,12 +24,12 @@
 PHONY. utils: ${BINS}
 
 install: utils
-	mkdir -p ${PREFIX}/sbin
-	cp unionctl ${PREFIX}/sbin
-	cp uniondbg ${PREFIX}/sbin
-	cp unionimap ${PREFIX}/sbin
+	mkdir -p ${SBINDIR}
+	cp unionctl ${SBINDIR}
+	cp uniondbg ${SBINDIR}
+	cp unionimap ${SBINDIR}
 
 uninstall:
-	rm ${PREFIX}/sbin/unionctl
-	rm ${PREFIX}/sbin/uniondbg
-	rm ${PREFIX}/sbin/unionimap
+	rm ${SBINDIR}/unionctl
+	rm ${SBINDIR}/uniondbg
+	rm ${SBINDIR}/unionimap
