diff -Naur myththemes-15941-old/cpsvndir myththemes-15941-new/cpsvndir
--- myththemes-15941-old/cpsvndir	2008-02-12 09:34:20.000000000 -0800
+++ myththemes-15941-new/cpsvndir	2008-02-12 09:34:48.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
