diff -Naur myththemes-15142-old/cpsvndir myththemes-15142-new/cpsvndir
--- myththemes-15142-old/cpsvndir	2007-12-12 09:07:46.000000000 -0800
+++ myththemes-15142-new/cpsvndir	2007-12-12 09:08:17.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
