diff -Naur xorg-server-1.6.0-old/hw/xfree86/common/xf86Config.c xorg-server-1.6.0-new/hw/xfree86/common/xf86Config.c
--- xorg-server-1.6.0-old/hw/xfree86/common/xf86Config.c	2009-03-14 12:03:52.000000000 -0700
+++ xorg-server-1.6.0-new/hw/xfree86/common/xf86Config.c	2009-03-14 12:08:26.000000000 -0700
@@ -111,16 +111,6 @@
 #endif
 
 static ModuleDefault ModuleDefaults[] = {
-    {.name = "extmod",   .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "dbe",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "glx",      .toLoad = TRUE,    .load_opt=NULL},
-#ifdef XRECORD
-    {.name = "record",   .toLoad = TRUE,    .load_opt=NULL},
-#endif
-    {.name = "dri",      .toLoad = TRUE,    .load_opt=NULL},
-#ifdef DRI2
-    {.name = "dri2",     .toLoad = TRUE,    .load_opt=NULL},
-#endif
     {.name = NULL,       .toLoad = FALSE,   .load_opt=NULL}
 };
 
