diff -Naur xorg-server-1.6.1-old/hw/xfree86/common/Makefile.am xorg-server-1.6.1-new/hw/xfree86/common/Makefile.am
--- xorg-server-1.6.1-old/hw/xfree86/common/Makefile.am	2009-04-14 10:14:57.000000000 -0700
+++ xorg-server-1.6.1-new/hw/xfree86/common/Makefile.am	2009-06-24 09:31:55.000000000 -0700
@@ -7,7 +7,7 @@
 
 if XV
 XVSOURCES = xf86xv.c xf86xvmc.c
-XVSDKINCS = xf86xv.h xf86xvmc.h
+XVSDKINCS = xf86xv.h xf86xvmc.h xf86xvpriv.h
 endif
 
 XISOURCES = xf86Xinput.c xisb.c
@@ -47,7 +47,7 @@
               xf86PciInfo.h xf86Priv.h xf86Privstr.h xf86Resources.h \
               xf86cmap.h xf86fbman.h xf86str.h xf86RAC.h xf86Xinput.h xisb.h \
               $(XVSDKINCS) xorgVersion.h \
-              xf86sbusBus.h xf86xv.h xf86xvmc.h xf86xvpriv.h
+              xf86sbusBus.h
 
 DISTCLEANFILES = xf86Build.h
 CLEANFILES = $(BUILT_SOURCES)
diff -Naur xorg-server-1.6.1-old/hw/xfree86/parser/Makefile.am xorg-server-1.6.1-new/hw/xfree86/parser/Makefile.am
--- xorg-server-1.6.1-old/hw/xfree86/parser/Makefile.am	2009-04-14 10:14:57.000000000 -0700
+++ xorg-server-1.6.1-new/hw/xfree86/parser/Makefile.am	2009-06-24 09:39:23.000000000 -0700
@@ -35,6 +35,5 @@
 	xf86tokens.h
 
 sdk_HEADERS = \
-	$(LIBHEADERS) \
-	xf86Parser.h \
-	xf86Optrec.h
+	xf86Optrec.h \
+	xf86Parser.h
