diff -Naur xorg-server-1.17.2-old/hw/xfree86/common/xf86pciBus.c xorg-server-1.17.2-new/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.17.2-old/hw/xfree86/common/xf86pciBus.c	2015-07-14 11:22:43.962000081 -0700
+++ xorg-server-1.17.2-new/hw/xfree86/common/xf86pciBus.c	2015-07-14 11:26:29.253000407 -0700
@@ -1105,7 +1105,7 @@
         driverList[0] = "ast";
         break;
     case 0x1002:
-        driverList[0] = "ati";
+        driverList[0] = "radeon";
         break;
     case 0x102c:
         driverList[0] = "chips";
