diff -Naur xorg-server-1.0.1-old/hw/xfree86/common/xf86Init.c xorg-server-1.0.1-new/hw/xfree86/common/xf86Init.c
--- xorg-server-1.0.1-old/hw/xfree86/common/xf86Init.c	2005-12-14 12:11:16.000000000 -0800
+++ xorg-server-1.0.1-new/hw/xfree86/common/xf86Init.c	2006-04-17 11:40:22.000000000 -0700
@@ -1376,7 +1376,7 @@
     }
   
   /* First the options that are only allowed for root */
-  if (getuid() == 0 || geteuid != 0)
+  if (getuid() == 0 || geteuid() != 0)
   {
     if (!strcmp(argv[i], "-modulepath"))
     {
@@ -1679,7 +1679,7 @@
   }
   if (!strcmp(argv[i], "-configure"))
   {
-    if (getuid() != 0 && geteuid == 0) {
+    if (getuid() != 0 && geteuid() == 0) {
 	ErrorF("The '-configure' option can only be used by root.\n");
 	exit(1);
     }
