diff -Naur alsa-driver-1.0.14rc2-old/Makefile alsa-driver-1.0.14rc2-new/Makefile
--- alsa-driver-1.0.14rc2-old/Makefile	2007-01-15 05:25:13.000000000 -0800
+++ alsa-driver-1.0.14rc2-new/Makefile	2007-01-18 18:46:39.000000000 -0800
@@ -198,7 +198,7 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	-depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
 	@echo "*** This is a \`staged' install using the prefix"
 	@echo "***	\"$(DESTDIR)\"."
