diff -Naur unionfs-20050916-1007-old/Makefile unionfs-20050916-1007-new/Makefile
--- unionfs-20050916-1007-old/Makefile	2005-09-16 07:07:40.000000000 -0700
+++ unionfs-20050916-1007-new/Makefile	2005-09-16 19:48:27.000000000 -0700
@@ -1,5 +1,5 @@
 #
-# Makefile for the Linux 2.4 unionfs
+# Makefile for the Linux 2.4/2.6 unionfs
 #
 
 # this should point to where your kernel headers are
@@ -10,6 +10,7 @@
 # This is where the make install target will put stuff
 PREFIX   = /usr/local
 MANDIR   = ${PREFIX}/man
+SBINDIR  = ${PREFIX}/sbin
 MODPREFIX=
 VERSION=20050916-1007
 TAROPTS = czvf
@@ -57,7 +58,7 @@
 	@echo ""
 
 unionfs2.6:
-	make -C ${LINUXSRC} SUBDIRS=$(PWD) FISTDEVMK=$(PWD)/fistdev.mk modules
+	make -C ${LINUXSRC} SUBDIRS=`pwd` FISTDEVMK=`pwd`/fistdev.mk modules
 
 unionfs2.4: ${obj-m}
 
@@ -84,7 +85,7 @@
 
 clean:
 ifeq ($(KERNELVERSION),2.6)
-	-make -C ${LINUXSRC} SUBDIRS=$(PWD) FISTDEVMK=$(PWD)/fistdev.mk clean
+	-make -C ${LINUXSRC} SUBDIRS=`pwd` FISTDEVMK=`pwd`/fistdev.mk clean
 endif
 	rm -f ${unionfs-objs} ${obj-m} ${obj-m:.o=.ko} *.d .*.flags \#* *~
 	rm -f ${BINS} TAGS tags
@@ -96,10 +97,10 @@
 PHONY. utils: ${BINS}
 
 install-utils: utils
-	mkdir -p ${PREFIX}/sbin
-	cp unionctl ${PREFIX}/sbin
-	cp uniondbg ${PREFIX}/sbin
-	cp unionimap ${PREFIX}/sbin
+	mkdir -p ${SBINDIR}
+	cp unionctl ${SBINDIR}
+	cp uniondbg ${SBINDIR}
+	cp unionimap ${SBINDIR}
 	mkdir -p ${MANDIR}/man4
 	mkdir -p ${MANDIR}/man8
 	cp man/unionfs.4 ${MANDIR}/man4
@@ -109,18 +110,16 @@
 install-mod: unionfs$(KERNELVERSION)
 	mkdir -p ${MODPREFIX}/${MODDIR}/kernel/fs
 	if [ -f unionfs.ko ] ; then cp unionfs.ko ${MODPREFIX}/${MODDIR}/kernel/fs ; else cp unionfs.o ${MODPREFIX}/${MODDIR}/kernel/fs ; fi
-	-/sbin/depmod -a
 
 install: install-utils install-mod
 
 uninstall:
 	if [ -f unionfs.ko ] ; then rm ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs.ko ; else rm ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs.o ; fi
-	-/sbin/depmod -a
 	rm ${MANDIR}/man8/unionctl.8
 	rm ${MANDIR}/man8/uniondbg.8
 	rm ${MANDIR}/man4/unionfs.4
-	rm ${PREFIX}/sbin/unionctl
-	rm ${PREFIX}/sbin/uniondbg
+	rm ${SBINDIR}/unionctl
+	rm ${SBINDIR}/uniondbg
 
 release: unionfs-${VERSION}.tar${COMPEXT}
 
