diff -Naur alsa-driver-1.0.23-old/configure alsa-driver-1.0.23-new/configure
--- alsa-driver-1.0.23-old/configure	2010-04-16 04:17:26.000000000 -0700
+++ alsa-driver-1.0.23-new/configure	2010-08-27 09:47:24.000000000 -0700
@@ -6986,7 +6986,7 @@
 	KERNEL_INC="$KERNEL_INC -I$mach_dir/mach-$machine"
     fi
 
-if test "$processor" == "x86_64"; then
+if test "$processor" = "x86_64"; then
   if test -n "$mach_dir"; then
 { echo "$as_me:$LINENO: checking for i386 machine type" >&5
 echo $ECHO_N "checking for i386 machine type... $ECHO_C" >&6; }
diff -Naur alsa-driver-1.0.23-old/configure.in alsa-driver-1.0.23-new/configure.in
--- alsa-driver-1.0.23-old/configure.in	2010-04-16 04:11:04.000000000 -0700
+++ alsa-driver-1.0.23-new/configure.in	2010-08-27 09:45:53.000000000 -0700
@@ -1329,7 +1329,7 @@
 	KERNEL_INC="$KERNEL_INC -I$mach_dir/mach-$machine"
     fi
 
-if test "$processor" == "x86_64"; then
+if test "$processor" = "x86_64"; then
   if test -n "$mach_dir"; then
 AC_MSG_CHECKING(for i386 machine type)
 	machine="default"
