diff -Naur libwww-perl-5.808-old/lib/LWP/Simple.pm libwww-perl-5.808-new/lib/LWP/Simple.pm
--- libwww-perl-5.808-old/lib/LWP/Simple.pm	2004-05-21 02:11:55.000000000 -0700
+++ libwww-perl-5.808-new/lib/LWP/Simple.pm	2008-02-17 20:37:00.000000000 -0800
@@ -169,7 +169,7 @@
    print $sock join("\015\012" =>
                     "GET $path HTTP/1.0",
                     "Host: $netloc",
-                    "User-Agent: lwp-trivial/$VERSION",
+                    "User-Agent: Lynx (textmode)",
                     "", "");
 
    my $buf = "";
