diff -Naur mythtv-0.22-22547-old/configure mythtv-0.22-22547-new/configure
--- mythtv-0.22-22547-old/configure	2009-10-20 16:09:56.000000000 -0700
+++ mythtv-0.22-22547-new/configure	2009-10-20 16:09:56.000000000 -0700
@@ -3449,7 +3449,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
