diff -Naur mythtv-0.22-23473-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23473-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23473-old/libs/libmyth/mythcontext.cpp	2010-02-05 16:22:52.000000000 -0800
+++ mythtv-0.22-23473-new/libs/libmyth/mythcontext.cpp	2010-02-05 16:25:20.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
