diff -Naur mythtv-0.22-22659-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.22-22659-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.22-22659-old/libs/libmyth/mediamonitor-unix.cpp	2009-10-30 11:43:08.000000000 -0700
+++ mythtv-0.22-22659-new/libs/libmyth/mediamonitor-unix.cpp	2009-10-30 11:43:08.000000000 -0700
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
