diff -Naur mythtv-0.22-27412-old/mythtv/libs/libmyth/mythcontext.cpp mythtv-0.22-27412-new/mythtv/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-27412-old/mythtv/libs/libmyth/mythcontext.cpp	2010-12-02 14:12:01.000000000 -0800
+++ mythtv-0.22-27412-new/mythtv/libs/libmyth/mythcontext.cpp	2010-12-02 14:12:02.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
