diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmyth/mythcontext.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmyth/mythcontext.cpp	2011-01-15 08:08:06.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmyth/mythcontext.cpp	2011-01-15 08:08:07.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
