diff -Naur mythtv-0.22-25120-old/libs/libmyth/mythcontext.cpp mythtv-0.22-25120-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-25120-old/libs/libmyth/mythcontext.cpp	2010-06-20 07:22:36.000000000 -0700
+++ mythtv-0.22-25120-new/libs/libmyth/mythcontext.cpp	2010-06-20 07:25:10.000000000 -0700
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
