diff -Naur mythtv-0.22-24255-old/libs/libmythdvdnav/libmythdvdnav.pro mythtv-0.22-24255-new/libs/libmythdvdnav/libmythdvdnav.pro
--- mythtv-0.22-24255-old/libs/libmythdvdnav/libmythdvdnav.pro	2010-05-03 19:06:44.000000000 -0700
+++ mythtv-0.22-24255-new/libs/libmythdvdnav/libmythdvdnav.pro	2010-05-14 08:09:53.000000000 -0700
@@ -13,6 +13,9 @@
 #build position independent code since the library is linked into a shared library
 QMAKE_CFLAGS += -fPIC -DPIC
 
+# for -ldl
+LIBS += -ldl
+
 DEFINES += HAVE_AV_CONFIG_H _LARGEFILE_SOURCE
 
 # DEFINES += LOG_DEBUG TRACE
