diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2011-01-15 08:08:21.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2011-01-15 08:08:23.000000000 -0800
@@ -18,7 +18,7 @@
 #ifdef Q_WS_MACX
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(QObject::tr("This directory must exist and "
                        "MythGallery needs to have read permission."));
@@ -51,7 +51,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("GalleryMoviePlayerCmd");
     gc->setLabel(QObject::tr("Command run to display movie files"));
-    gc->setValue("mplayer -fs %s");
+    gc->setValue("mplayer -fs -zoom -quiet -loop 0 %s");
     gc->setHelpText(QObject::tr("This command is executed whenever a movie "
                     "file is selected"));
     return gc;
@@ -71,7 +71,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("GalleryImportDirs");
     gc->setLabel(QObject::tr("Paths to import images from"));
-    gc->setValue("/mnt/cdrom:/mnt/camera");
+    gc->setValue("");
     gc->setHelpText(QObject::tr("This is a colon separated list of paths. "
                     "If the path in the list is a directory, its contents will "
                     "be copied. If it is an executable, it will be run."));
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythmusic/mythmusic/globalsettings.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythmusic/mythmusic/globalsettings.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythmusic/mythmusic/globalsettings.cpp	2011-01-15 08:08:21.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythmusic/mythmusic/globalsettings.cpp	2011-01-15 08:08:23.000000000 -0800
@@ -40,7 +40,7 @@
 #ifdef Q_WS_MACX
     gc->setValue(QDir::homePath() + "/Music");
 #else
-    gc->setValue("/mnt/store/music/");
+    gc->setValue("/mnt/music/");
 #endif
     gc->setHelpText(QObject::tr("This directory must exist, and the user "
                     "running MythMusic needs to have write permission "
@@ -53,6 +53,10 @@
     HostComboBox *gc = new HostComboBox("MusicAudioDevice", true);
     gc->setLabel(QObject::tr("Audio device"));
     gc->addSelection(QObject::tr("default"), "default");
+
+    gc->addSelection(QObject::tr("ALSA:default"), "ALSA:default");
+    gc->addSelection(QObject::tr("ALSA:iec958"), "ALSA:iec958");
+
     QDir dev("/dev", "dsp*", QDir::Name, QDir::System);
     gc->fillSelectionsFromDir(dev);
     dev.setNameFilter("adsp*");
@@ -80,7 +84,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("TreeLevels", true);
     gc->setLabel(QObject::tr("Tree Sorting"));
-    gc->setValue("splitartist artist album title");
+    gc->setValue("directory");
     gc->setHelpText(QObject::tr("Order in which to sort the Music "
                     "Tree. Possible values are a space-separated list of "
                     "genre, splitartist, splitartist1, artist, album, and "
@@ -103,7 +107,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("NonID3FileNameFormat");
     gc->setLabel(QObject::tr("Filename Format"));
-    gc->setValue("GENRE/ARTIST/ALBUM/TRACK_TITLE");
+    gc->setValue("ARTIST/YEAR - ALBUM/TRACK - TITLE");
     gc->setHelpText(QObject::tr("Directory and filename Format used to grab "
                     "information if no ID3 information is found. Accepts "
                     "GENRE, ARTIST, ALBUM, TITLE, ARTIST_TITLE and "
@@ -208,7 +212,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("FilenameTemplate");
     gc->setLabel(QObject::tr("File storage location"));
-    gc->setValue("ARTIST/ALBUM/TRACK-TITLE"); // Don't translate
+    gc->setValue("ARTIST/YEAR-ALBUM/TRACK-TITLE"); // Don't translate
     gc->setHelpText(QObject::tr("Defines the location/name for new songs. "
                     "Valid tokens are: GENRE, ARTIST, ALBUM, "
                     "TRACK, TITLE, YEAR"));
@@ -332,7 +336,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("ListAsShuffled");
     gc->setLabel(QObject::tr("List as Shuffled"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("List songs on the playback screen "
                     "in the order they will be played."));
     return gc;
@@ -342,7 +346,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("ShowWholeTree");
     gc->setLabel(QObject::tr("Show entire music tree"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If selected, you can navigate your entire "
                     "music tree from the playing screen. N.B. Does not work "
                     "with accelerated buttons disabled"));
@@ -478,7 +482,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("CDWriterEnabled");
     gc->setLabel(QObject::tr("Enable CD Writing."));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("Requires a SCSI or an IDE-SCSI CD Writer."));
     return gc;
 };
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythvideo/mythvideo/globals.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythvideo/mythvideo/globals.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythvideo/mythvideo/globals.cpp	2011-01-15 08:08:21.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythvideo/mythvideo/globals.cpp	2011-01-15 08:08:23.000000000 -0800
@@ -32,5 +32,5 @@
 #ifdef Q_WS_MACX
 const QString DEFAULT_VIDEOSTARTUP_DIR = QDir::homePath() + "/Movies";
 #else
-const QString DEFAULT_VIDEOSTARTUP_DIR = "/share/Movies/dvd";
+const QString DEFAULT_VIDEOSTARTUP_DIR = "/mnt/video";
 #endif
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythvideo/mythvideo/globalsettings.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythvideo/mythvideo/globalsettings.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythplugins/mythvideo/mythvideo/globalsettings.cpp	2011-01-15 08:08:21.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythplugins/mythvideo/mythvideo/globalsettings.cpp	2011-01-15 08:08:23.000000000 -0800
@@ -38,7 +38,7 @@
     gc->addSelection(QObject::tr("Browser"),
                      QString::number(VideoDialog::DLG_BROWSER));
     gc->addSelection(QObject::tr("Listings"),
-                     QString::number(VideoDialog::DLG_TREE));
+                     QString::number(VideoDialog::DLG_TREE),true);
     gc->addSelection(QObject::tr("Manager"),
                      QString::number(VideoDialog::DLG_MANAGER));
     gc->setHelpText(QObject::tr("The default view for MythVideo. "
@@ -531,7 +531,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("VCDPlayerCommand");
     gc->setLabel(QObject::tr("VCD Player Command"));
-    gc->setValue("mplayer vcd:// -cdrom-device %d -fs -zoom -vo xv");
+    gc->setValue("mplayer vcd:// -cdrom-device %d -fs -zoom -quiet");
     gc->setHelpText(QObject::tr("This can be any command to launch a VCD "
                     "player (e.g. MPlayer, xine, etc.). If present, %d will "
                     "be substituted for the VCD device (e.g. /dev/cdrom)."));
@@ -559,7 +559,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("TitlePlayCommand");
     gc->setLabel(QObject::tr("Title Playing Command"));
-    gc->setValue("mplayer dvd://%t -dvd-device %d -fs -zoom -vo xv -aid %a "
+    gc->setValue("mplayer dvd://%t -dvd-device %d -fs -zoom -quiet -aid %a "
                  "-channels %c");
     gc->setHelpText(QObject::tr("This is a command used to preview a given "
                     "title on a DVD. If present %t will be set "
diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/programs/mythfrontend/globalsettings.cpp	2011-01-15 08:08:22.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/programs/mythfrontend/globalsettings.cpp	2011-01-15 08:08:23.000000000 -0800
@@ -44,7 +44,7 @@
     gc->setLabel(QObject::tr("Audio output device"));
 
 #ifdef USING_ALSA
-    gc->addSelection("ALSA:default",       "ALSA:default");
+    gc->addSelection("ALSA:default",       "ALSA:default", true);
     gc->addSelection("ALSA:spdif",         "ALSA:spdif");
     gc->addSelection("ALSA:surround51",    "ALSA:surround51");
     gc->addSelection("ALSA:analog",        "ALSA:analog");
@@ -140,7 +140,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("MythControlsVolume");
     gc->setLabel(QObject::tr("Use internal volume controls"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("MythTV can control the PCM and master "
                     "mixer volume.  If you prefer to use an external mixer "
                     "program, then disable this option."));
@@ -163,7 +163,7 @@
     }
 #endif
 #ifdef USING_ALSA
-    gc->addSelection("ALSA:default", "ALSA:default");
+    gc->addSelection("ALSA:default", "ALSA:default", true);
 #endif
 #ifdef USING_MINGW
     gc->addSelection("DirectX:", "DirectX:");
@@ -1480,7 +1480,7 @@
     gc->setLabel(QObject::tr("Sort Episodes"));
     gc->addSelection(QObject::tr("Record date"), "Date");
     gc->addSelection(QObject::tr("Original Air date"), "OrigAirDate");
-    gc->addSelection(QObject::tr("Program ID"), "Id");
+    gc->addSelection(QObject::tr("Program ID"), "Id", true);
     gc->setHelpText(QObject::tr("Selects how to sort a shows episodes"));
     return gc;
 }
@@ -1489,7 +1489,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("StickyKeys");
     gc->setLabel(QObject::tr("Sticky keys"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If enabled, fast forward and rewind "
                     "continue after the key is released.  Pressing the key "
                     "again increases the fast forward or rewind speed.  The "
@@ -2025,7 +2025,7 @@
     HostComboBox *gc = new HostComboBox("PlaybackExitPrompt");
     gc->setLabel(QObject::tr("Action on playback exit"));
     gc->addSelection(QObject::tr("Just exit"), "0");
-    gc->addSelection(QObject::tr("Save position and exit"), "2");
+    gc->addSelection(QObject::tr("Save position and exit"), "2", true);
     gc->addSelection(QObject::tr("Always prompt (excluding Live TV)"), "1");
     gc->addSelection(QObject::tr("Always prompt (including Live TV)"), "4");
     gc->addSelection(QObject::tr("Prompt for Live TV only"), "8");
@@ -2040,7 +2040,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("EndOfRecordingExitPrompt");
     gc->setLabel(QObject::tr("Prompt at end of recording"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If set, a menu will be displayed allowing "
                     "you to delete the recording when it has finished "
                     "playing."));
@@ -2129,7 +2129,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("PlaybackPreview");
     gc->setLabel(QObject::tr("Display live preview of recordings"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("When enabled, a preview of the recording "
                     "will play in a small window on the \"Watch a "
                     "Recording\" menu."));
@@ -2203,7 +2203,7 @@
     HostComboBox *gc = new HostComboBox("AllowQuitShutdown");
     gc->setLabel(QObject::tr("System Exit key"));
     gc->addSelection(QObject::tr("ESC"), "4");
-    gc->addSelection(QObject::tr("No exit key"), "0");
+    gc->addSelection(QObject::tr("No exit key"), "0", true);
     gc->addSelection(QObject::tr("Control-ESC"), "1");
     gc->addSelection(QObject::tr("Meta-ESC"), "2");
     gc->addSelection(QObject::tr("Alt-ESC"), "3");
@@ -2258,7 +2258,7 @@
 {
     HostLineEdit *ge = new HostLineEdit("HaltCommand");
     ge->setLabel(QObject::tr("Halt command"));
-    ge->setValue("halt");
+    ge->setValue("poweroff");
     ge->setHelpText(QObject::tr("Command or script to run if you select "
                     "the shutdown option from the exit menu, if the option "
                     "is displayed. You must configure an exit key to "
@@ -2969,7 +2969,7 @@
     GlobalCheckBox *bc = new GlobalCheckBox("LastFreeCard");
     bc->setLabel(QObject::tr("Avoid conflicts between live TV and "
                  "scheduled shows"));
-    bc->setValue(false);
+    bc->setValue(true);
     bc->setHelpText(QObject::tr("If enabled, live TV will choose a tuner card "
                     "that is less likely to have scheduled recordings "
                     "rather than the best card available."));
