diff -Naur mythtv-0.22-23953-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23953-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23953-old/libs/libmyth/mythcontext.cpp	2010-04-04 07:54:48.000000000 -0700
+++ mythtv-0.22-23953-new/libs/libmyth/mythcontext.cpp	2010-04-04 07:57:18.000000000 -0700
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
