diff -Naur mythtv-0.22-23604-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23604-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23604-old/libs/libmyth/mythcontext.cpp	2010-02-27 11:30:19.000000000 -0800
+++ mythtv-0.22-23604-new/libs/libmyth/mythcontext.cpp	2010-02-27 11:33:03.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
