diff -Naur mythtv-0.22-22994-old/libs/libmyth/mythcontext.cpp mythtv-0.22-22994-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-22994-old/libs/libmyth/mythcontext.cpp	2009-12-22 08:50:51.000000000 -0800
+++ mythtv-0.22-22994-new/libs/libmyth/mythcontext.cpp	2009-12-22 10:12:44.000000000 -0800
@@ -1654,7 +1654,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
