diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-01-15 08:05:07.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-01-15 08:08:03.000000000 -0800
@@ -1724,6 +1724,9 @@
                 if (!codec_is_std(video_codec_id))
                     thread_count = 1;
 
+                if (enc->codec_id == CODEC_ID_SVQ3)
+                    thread_count = 1;
+
                 VERBOSE(VB_PLAYBACK, QString("Using %1 CPUs for decoding")
                         .arg(HAVE_THREADS ? thread_count : 1));
 
