diff -Naur mythtv-0.22-23893-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23893-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23893-old/libs/libmyth/mythcontext.cpp	2010-04-01 20:57:07.000000000 -0700
+++ mythtv-0.22-23893-new/libs/libmyth/mythcontext.cpp	2010-04-01 21:00:30.000000000 -0700
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
