diff -Naur mythtv-0.22-22512-old/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh mythtv-0.22-22512-new/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh
--- mythtv-0.22-22512-old/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh	2009-10-16 16:32:10.000000000 -0700
+++ mythtv-0.22-22512-new/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh	2009-10-16 16:32:10.000000000 -0700
@@ -4,13 +4,13 @@
 # wishes to use it. It has been used by me for many months with the
 # VIP211 receivers. -- Daniel Kristjansson (March 20th, 2008)
 
-# If you have a separate IR reciever you will want to give it "/dev/lircd"
-# and use something like "/dev/lircd1" for the transmitter.
+# If you have a separate IR reciever you will want to give it "/var/run/lirc/lircd"
+# and use something like "/var/run/lirc/lircd1" for the transmitter.
 # If the tranmitter is on a remote LIRC server, use the ip/hostname:port
 # with -a to address it.
 
-DEVICE="-d /dev/lircd"
-#DEVICE="-d /dev/lircd1"
+DEVICE="-d /var/run/lirc/lircd"
+#DEVICE="-d /var/run/lirc/lircd1"
 #DEVICE="-a hostname:port"
 
 # Use dish1, dish2 .. dish16 depending on which id you are using
diff -Naur mythtv-0.22-22512-old/libs/libmythui/mythmainwindow.cpp mythtv-0.22-22512-new/libs/libmythui/mythmainwindow.cpp
--- mythtv-0.22-22512-old/libs/libmythui/mythmainwindow.cpp	2009-10-16 16:32:09.000000000 -0700
+++ mythtv-0.22-22512-new/libs/libmythui/mythmainwindow.cpp	2009-10-16 16:32:10.000000000 -0700
@@ -1861,7 +1861,7 @@
 
     d->lircThread = new LIRC(
         this,
-        GetMythDB()->GetSetting("LircSocket", "/dev/lircd"),
+        GetMythDB()->GetSetting("LircSocket", "/var/run/lirc/lircd"),
         "mythtv", config_file,
         GetMythDB()->GetSetting("LircKeyPressedApp", ""));
 
diff -Naur mythtv-0.22-22512-old/programs/mythfrontend/globalsettings.cpp mythtv-0.22-22512-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.22-22512-old/programs/mythfrontend/globalsettings.cpp	2009-10-16 16:32:10.000000000 -0700
+++ mythtv-0.22-22512-new/programs/mythfrontend/globalsettings.cpp	2009-10-16 16:32:10.000000000 -0700
@@ -2284,7 +2284,7 @@
 {
     HostLineEdit *ge = new HostLineEdit("LircSocket");
     ge->setLabel(QObject::tr("LIRC Daemon Socket"));
-    ge->setValue("/dev/lircd");
+    ge->setValue("/var/run/lirc/lircd");
     QString help = QObject::tr(
         "UNIX socket or IP address[:port] to connect in "
         "order to communicate with the LIRC Daemon.");
