diff -Naur mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.22-20100711-3c6ce60-697a6ac-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2011-01-15 08:08:18.000000000 -0800
+++ mythtv-0.22-20100711-3c6ce60-697a6ac-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2011-01-15 08:08:20.000000000 -0800
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
