diff -Naur mythtv-0.22-25328-old/configure mythtv-0.22-25328-new/configure
--- mythtv-0.22-25328-old/configure	2010-08-17 07:13:46.000000000 -0700
+++ mythtv-0.22-25328-new/configure	2010-08-27 10:56:57.000000000 -0700
@@ -2622,13 +2622,13 @@
         enabled shared || check_cflags -mdynamic-no-pic
     fi
     # Apple GCC doesn't support march=core2 yet:
-    if test $cpu == "core2"; then
+    if test $cpu = "core2"; then
         cpu="nocona"
     fi
     # Core Duo might be identified as march=pentium3, which errors:
     # ...MachineExceptions.h:... error: '__m128i' does not name a type
     # ...MachineExceptions.h:... error: '__m128i' does not name a type
-    if test $cpu == "pentium3"; then
+    if test $cpu = "pentium3"; then
         cpu=prescott
     fi
 fi
