diff -Naur mythtv-0.22-23110-old/libs/libmyth/mythcontext.cpp mythtv-0.22-23110-new/libs/libmyth/mythcontext.cpp
--- mythtv-0.22-23110-old/libs/libmyth/mythcontext.cpp	2010-01-11 20:04:19.000000000 -0800
+++ mythtv-0.22-23110-new/libs/libmyth/mythcontext.cpp	2010-01-11 20:06:58.000000000 -0800
@@ -1655,7 +1655,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
