diff -Naur mythtv-master-20110722-546af2f-f037f74-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110722-546af2f-f037f74-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110722-546af2f-f037f74-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-07-23 04:24:06.000000000 -0700
+++ mythtv-master-20110722-546af2f-f037f74-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-07-23 04:35:41.000000000 -0700
@@ -1314,6 +1314,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         enc->flags2 |= CODEC_FLAG2_FAST;
