diff -Naur mythtv-master-20120508-d6ed457-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-master-20120508-d6ed457-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-master-20120508-d6ed457-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-05-09 04:12:49.000000000 -0700
+++ mythtv-master-20120508-d6ed457-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-05-09 04:17:27.000000000 -0700
@@ -2418,6 +2418,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
