diff -Naur mythtv-master-20141222-5b386a0-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20141222-5b386a0-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20141222-5b386a0-old/mythtv/programs/mythfrontend/main.cpp	2015-06-11 01:01:17.929000003 -0700
+++ mythtv-master-20141222-5b386a0-new/mythtv/programs/mythfrontend/main.cpp	2015-06-11 01:01:19.608000002 -0700
@@ -1851,6 +1851,18 @@
                 QString("NetworkControl failed to bind to port %1.")
                    .arg(port));
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
 #if CONFIG_DARWIN
     GetMythMainWindow()->SetEffectsEnabled(false);
