diff -Naur mythtv-master-20121113-5e1c14e-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20121113-5e1c14e-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20121113-5e1c14e-old/mythtv/programs/mythfrontend/main.cpp	2012-11-14 14:45:16.000000000 -0800
+++ mythtv-master-20121113-5e1c14e-new/mythtv/programs/mythfrontend/main.cpp	2012-11-14 14:45:17.000000000 -0800
@@ -1675,6 +1675,18 @@
                 QString("NetworkControl failed to bind to port %1.")
                    .arg(port));
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
 #ifdef __linux__
 #ifdef CONFIG_BINDINGS_PYTHON
