diff -Naur mythtv-master-20120508-d6ed457-old/mythtv/programs/mythfrontend/videodlg.cpp mythtv-master-20120508-d6ed457-new/mythtv/programs/mythfrontend/videodlg.cpp
--- mythtv-master-20120508-d6ed457-old/mythtv/programs/mythfrontend/videodlg.cpp	2012-05-09 04:17:36.000000000 -0700
+++ mythtv-master-20120508-d6ed457-new/mythtv/programs/mythfrontend/videodlg.cpp	2012-05-09 04:17:45.000000000 -0700
@@ -1027,6 +1027,8 @@
                 SLOT(UpdateText(MythUIButtonListItem *)));
     }
 
+    connect(&m_d->m_parentalLevel, SIGNAL(SigLevelChanged()),
+            SLOT(reloadData()));
     return true;
 }
 
@@ -1035,8 +1037,6 @@
     m_d->m_parentalLevel.SetLevel(ParentalLevel(gCoreContext->
                     GetNumSetting("VideoDefaultParentalLevel",
                             ParentalLevel::plLowest)));
-    connect(&m_d->m_parentalLevel, SIGNAL(SigLevelChanged()),
-            SLOT(reloadData()));
 }
 
 void VideoDialog::Load()
