diff -Naur mythtv-master-20130118-5a465ba-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-master-20130118-5a465ba-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-master-20130118-5a465ba-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2013-01-19 05:21:26.000000000 -0800
+++ mythtv-master-20130118-5a465ba-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2013-01-19 05:21:28.000000000 -0800
@@ -2437,6 +2437,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
