diff -Naur mythtv-master-20121214-021b2e5-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-master-20121214-021b2e5-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-master-20121214-021b2e5-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-12-14 22:45:37.000000000 -0800
+++ mythtv-master-20121214-021b2e5-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2012-12-14 22:45:38.000000000 -0800
@@ -2435,6 +2435,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
