diff -Naur mythtv-master-20110728-42fbadb-f037f74-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110728-42fbadb-f037f74-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110728-42fbadb-f037f74-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-07-29 08:33:33.000000000 -0700
+++ mythtv-master-20110728-42fbadb-f037f74-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-07-29 08:35:51.000000000 -0700
@@ -1314,6 +1314,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         enc->flags2 |= CODEC_FLAG2_FAST;
