diff -Naur mythtv-master-20110331-de61d61-ecaeffc-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110331-de61d61-ecaeffc-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110331-de61d61-ecaeffc-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-04-01 09:48:50.000000000 -0700
+++ mythtv-master-20110331-de61d61-ecaeffc-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-04-01 09:48:57.000000000 -0700
@@ -1280,6 +1280,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
