diff -Naur mythtv-master-20110114-5707d26-45d6b63-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110114-5707d26-45d6b63-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110114-5707d26-45d6b63-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-01-15 08:44:20.000000000 -0800
+++ mythtv-master-20110114-5707d26-45d6b63-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-01-15 09:03:04.000000000 -0800
@@ -1234,6 +1234,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
