diff -Naur mythtv-master-20110829-837f15c-f037f74-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20110829-837f15c-f037f74-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20110829-837f15c-f037f74-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-08-30 09:12:11.000000000 -0700
+++ mythtv-master-20110829-837f15c-f037f74-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-08-30 09:14:35.000000000 -0700
@@ -1321,6 +1321,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         enc->flags2 |= CODEC_FLAG2_FAST;
