diff -Naur mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20120406-83671d5-old/mythtv/programs/mythfrontend/main.cpp	2012-04-07 14:55:55.000000000 -0700
+++ mythtv-master-20120406-83671d5-new/mythtv/programs/mythfrontend/main.cpp	2012-04-07 14:55:56.000000000 -0700
@@ -1687,6 +1687,18 @@
                 QString("NetworkControl failed to bind to port %1.")
                    .arg(port));
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
 #ifdef __linux__
 #ifdef CONFIG_BINDINGS_PYTHON
