diff -Naur mythstream-0.18_1-old/mythstream/mythstream.cpp mythstream-0.18_1-new/mythstream/mythstream.cpp
--- mythstream-0.18_1-old/mythstream/mythstream.cpp	2007-09-25 15:54:33.000000000 -0700
+++ mythstream-0.18_1-new/mythstream/mythstream.cpp	2007-11-22 12:13:11.000000000 -0800
@@ -460,8 +460,8 @@
 
         parameter = editBox->text();
 
-        delete popBox;        
-        popBox = NULL;
+        popBox->hide();
+        popBox->deleteLater();
 
         return parameter;
 }
@@ -549,8 +549,8 @@
 			  streamStorage->removeRecord(ident_maingui, values, errorMessage);
 		}
 
-		delete popBox;		
-		popBox = NULL;
+		popBox->hide();
+		popBox->deleteLater();
 
 		return;
 }
