diff -Naur Mesa-7.0.4-old/src/mesa/x86/Makefile Mesa-7.0.4-new/src/mesa/x86/Makefile
--- Mesa-7.0.4-old/src/mesa/x86/Makefile	2008-08-30 14:22:07.000000000 -0700
+++ Mesa-7.0.4-new/src/mesa/x86/Makefile	2008-08-30 14:23:00.000000000 -0700
@@ -21,7 +21,7 @@
 
 
 gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+	$(BUILD_CC) $(INCLUDE_DIRS) $(BUILD_CFLAGS) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
 matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
