diff -Naur Mesa-7.0.3-rc1-old/configs/linux-dri-x86 Mesa-7.0.3-rc1-new/configs/linux-dri-x86
--- Mesa-7.0.3-rc1-old/configs/linux-dri-x86	2008-02-11 19:57:33.000000000 -0800
+++ Mesa-7.0.3-rc1-new/configs/linux-dri-x86	2008-02-11 19:58:43.000000000 -0800
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86
 
diff -Naur Mesa-7.0.3-rc1-old/configs/linux-dri-x86-64 Mesa-7.0.3-rc1-new/configs/linux-dri-x86-64
--- Mesa-7.0.3-rc1-old/configs/linux-dri-x86-64	2008-02-11 19:57:33.000000000 -0800
+++ Mesa-7.0.3-rc1-new/configs/linux-dri-x86-64	2008-02-11 19:58:43.000000000 -0800
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86-64
 
