diff -Naur mythtv-10701-old/themes/cpsvndir mythtv-10701-new/themes/cpsvndir
--- mythtv-10701-old/themes/cpsvndir	2006-08-07 18:37:45.000000000 -0700
+++ mythtv-10701-new/themes/cpsvndir	2006-08-18 17:53:49.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
