diff -Naur mythtv-8839-old/programs/mythfrontend/globalsettings.cpp mythtv-8839-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-8839-old/programs/mythfrontend/globalsettings.cpp	2006-02-02 04:59:00.000000000 -0800
+++ mythtv-8839-new/programs/mythfrontend/globalsettings.cpp	2006-02-02 05:12:10.000000000 -0800
@@ -23,6 +23,10 @@
     HostComboBox *gc = new HostComboBox("AudioOutputDevice", true);
 
     gc->setLabel(QObject::tr("Audio output device"));
+
+    gc->addSelection(QObject::tr("ALSA:default"), "ALSA:default", true);
+    gc->addSelection(QObject::tr("ALSA:iec958"), "ALSA:iec958");
+
     QDir dev("/dev", "dsp*", QDir::Name, QDir::System);
     gc->fillSelectionsFromDir(dev);
     dev.setNameFilter("adsp*");
@@ -44,7 +48,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("MythControlsVolume");
     gc->setLabel(QObject::tr("Use internal volume controls"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("MythTV can control the PCM and master "
                     "mixer volume.  If you prefer to use an external mixer "
                     "program, then disable this option."));
@@ -56,6 +60,9 @@
     HostComboBox *gc = new HostComboBox("MixerDevice", true);
 
     gc->setLabel(QObject::tr("Mixer Device"));
+
+    gc->addSelection(QObject::tr("default"), "default", true);
+
     QDir dev("/dev", "mixer*", QDir::Name, QDir::System);
     gc->fillSelectionsFromDir(dev);
 
@@ -163,7 +170,7 @@
     gc->setLabel(QObject::tr("Algorithm"));
     gc->addSelection(QObject::tr("Linear blend"), "linearblend");
     gc->addSelection(QObject::tr("Kernel (less motion blur)"), "kerneldeint");
-    gc->addSelection(QObject::tr("Bob (2x framerate)"), "bobdeint");
+    gc->addSelection(QObject::tr("Bob (2x framerate)"), "bobdeint", true);
     gc->addSelection(QObject::tr("One field"), "onefield");
     gc->setHelpText(QObject::tr("Deinterlace algorithm.") + " " +
                     QObject::tr("'Kernel' requires SSE CPU support.") + " " +
@@ -203,7 +210,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("DecodeExtraAudio");
     gc->setLabel(QObject::tr("Extra audio buffering"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("Enable this setting if MythTV is playing "
                     "\"crackly\" audio and you are using hardware encoding. "
                     "This setting will have no effect "
@@ -650,7 +657,7 @@
     gc->setLabel(QObject::tr("Sort Episodes"));
     gc->addSelection(QObject::tr("Record date"), "Date");
     gc->addSelection(QObject::tr("Original Air date"), "OrigAirDate");
-    gc->addSelection(QObject::tr("Program ID"), "Id");
+    gc->addSelection(QObject::tr("Program ID"), "Id", true);
     gc->setHelpText(QObject::tr("Selects how to sort a shows episodes"));
     return gc;
 }
@@ -659,7 +666,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("StickyKeys");
     gc->setLabel(QObject::tr("Sticky keys"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If enabled, fast forward and rewind "
                     "continue after the key is released.  Pressing the key "
                     "again increases the fast forward or rewind speed.  The "
@@ -752,6 +759,8 @@
             gc->addSelection(theme->fileName());
     }
 
+    gc->setValue(gc->getValueIndex("G.A.N.T."));
+
     return gc;
 }
 
@@ -779,6 +788,8 @@
             gc->addSelection(theme->fileName());
     }
 
+    gc->setValue(gc->getValueIndex("Titivillus-OSD"));
+
     return gc;
 }
 
@@ -789,6 +800,10 @@
     QDir ttf(gContext->GetFontsDir(), gContext->GetFontsNameFilter());
     gc->fillSelectionsFromDir(ttf, false);
 
+    gc->setValue(gc->getValueIndex("FreeSans.ttf"));
+
+    gc->setValue(gc->getValueIndex("FreeSans.ttf"));
+
     return gc;
 }
 
@@ -875,7 +890,7 @@
     gc->addSelection(QObject::tr("Standard XvMC"), "xvmc");
 #endif // USING_XVMC
 #ifdef USING_XVMC_VLD
-    gc->addSelection(QObject::tr("VIA XvMC"), "xvmc-vld");
+    gc->addSelection(QObject::tr("VIA XvMC"), "xvmc-vld", true);
 #endif // USING_XVMC_VLD    
     gc->setHelpText(
         QObject::tr("Decoder to use to play back MPEG2 video.") + " " +
@@ -900,7 +915,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("UseVideoTimebase");
     gc->setLabel(QObject::tr("Use video as timebase"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("Use the video as the timebase and warp "
                     "the audio to keep it in sync. (Experimental)"));
     return gc;
@@ -1019,7 +1034,7 @@
     HostComboBox *gc = new HostComboBox("PlaybackExitPrompt");
     gc->setLabel(QObject::tr("Action on playback exit"));
     gc->addSelection(QObject::tr("Just exit"), "0");
-    gc->addSelection(QObject::tr("Save position and exit"), "2");
+    gc->addSelection(QObject::tr("Save position and exit"), "2", true);
     gc->addSelection(QObject::tr("Always prompt"), "1");
     gc->setHelpText(QObject::tr("If set to prompt, a menu will be displayed "
                     "when you exit playback mode.  The options available will "
@@ -1032,7 +1047,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("EndOfRecordingExitPrompt");
     gc->setLabel(QObject::tr("Prompt at end of recording"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If set, a menu will be displayed allowing "
                     "you to delete the recording when it has finished "
                     "playing."));
@@ -1080,7 +1095,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("PlaybackPreview");
     gc->setLabel(QObject::tr("Display live preview of recordings"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("When enabled, a preview of the recording "
                     "will play in a small window on the \"Watch a "
                     "Recording\" menu."));
@@ -1139,7 +1154,7 @@
     HostComboBox *gc = new HostComboBox("AllowQuitShutdown");
     gc->setLabel(QObject::tr("System Exit key"));
     gc->addSelection(QObject::tr("ESC"), "4");
-    gc->addSelection(QObject::tr("No exit key"), "0");
+    gc->addSelection(QObject::tr("No exit key"), "0", true);
     gc->addSelection(QObject::tr("Control-ESC"), "1");
     gc->addSelection(QObject::tr("Meta-ESC"), "2");
     gc->addSelection(QObject::tr("Alt-ESC"), "3");
@@ -1165,7 +1180,7 @@
 {
     HostLineEdit *ge = new HostLineEdit("HaltCommand");
     ge->setLabel(QObject::tr("Halt command"));
-    ge->setValue("halt");
+    ge->setValue("poweroff");
     ge->setHelpText(QObject::tr("If you have configured an exit key using the "
                     "System Shutdown option, you will be given the opportunity "
                     "to exit MythTV or halt the system completely. "
@@ -1812,7 +1827,7 @@
     GlobalCheckBox *bc = new GlobalCheckBox("LastFreeCard");
     bc->setLabel(QObject::tr("Avoid conflicts between live TV and "
                  "scheduled shows."));
-    bc->setValue(false);
+    bc->setValue(true);
     bc->setHelpText(QObject::tr("If enabled, live TV will choose a tuner card "
                     "that is less likely to have scheduled recordings "
                     "rather than the best card available."));
@@ -1867,7 +1882,8 @@
     gc->addSelection(QObject::tr("Blender - Transparent (CPU Usage - Middle)"),
                      QString::number((int)UIGuideType::Dense));
     gc->addSelection(QObject::tr("Eco - Transparent (CPU Usage - Low)"),
-                     QString::number((int)UIGuideType::Eco));
+                     QString::number((int)UIGuideType::Eco),
+                     true);
     gc->addSelection(QObject::tr("Solid (CPU Usage - Middle)"),
                      QString::number((int)UIGuideType::Solid));
     return gc;
