diff -Naur xf86-video-ati-6.12.1-old/configure.ac xf86-video-ati-6.12.1-new/configure.ac
--- xf86-video-ati-6.12.1-old/configure.ac	2009-03-18 17:44:39.000000000 -0700
+++ xf86-video-ati-6.12.1-new/configure.ac	2009-03-18 19:39:39.000000000 -0700
@@ -38,6 +38,7 @@
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+PKG_PROG_PKG_CONFIG
 
 if test "x$GCC" = "xyes"; then
 	CPPFLAGS="$CPPFLAGS -Wall"
@@ -76,7 +77,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.2 xproto fontsproto $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 
