diff -Naur mythstream-0.18_1-old/libs/harvester.cpp mythstream-0.18_1-new/libs/harvester.cpp
--- mythstream-0.18_1-old/libs/harvester.cpp	2007-09-25 15:58:20.000000000 -0700
+++ mythstream-0.18_1-new/libs/harvester.cpp	2008-01-16 12:52:41.000000000 -0800
@@ -46,6 +46,7 @@
 #include <qdatastream.h>
 #include <qdom.h>
 
+#include <stdlib.h>
 #include <sys/stat.h>
 
 #include "harvester.h"
diff -Naur mythstream-0.18_1-old/libs/recorder.cpp mythstream-0.18_1-new/libs/recorder.cpp
--- mythstream-0.18_1-old/libs/recorder.cpp	2005-05-14 06:22:51.000000000 -0700
+++ mythstream-0.18_1-new/libs/recorder.cpp	2008-01-16 12:53:57.000000000 -0800
@@ -23,6 +23,7 @@
 
 #include <iostream>
 
+#include <stdlib.h>
 #include <qnetwork.h>
 #include <qdatetime.h>
 #include <qpainter.h>
diff -Naur mythstream-0.18_1-old/libs/storageconfig.cpp mythstream-0.18_1-new/libs/storageconfig.cpp
--- mythstream-0.18_1-old/libs/storageconfig.cpp	2007-09-05 14:03:14.000000000 -0700
+++ mythstream-0.18_1-new/libs/storageconfig.cpp	2008-01-16 12:50:59.000000000 -0800
@@ -23,6 +23,7 @@
 // development snapshot, many todo's
 
 #include <iostream>
+#include <stdlib.h>
 #include <qlayout.h>
 #include <qcursor.h>
 #include <qregexp.h>
diff -Naur mythstream-0.18_1-old/libs/streamconfig.cpp mythstream-0.18_1-new/libs/streamconfig.cpp
--- mythstream-0.18_1-old/libs/streamconfig.cpp	2007-09-05 13:36:21.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamconfig.cpp	2008-01-16 12:49:56.000000000 -0800
@@ -23,6 +23,7 @@
 // development snapshot, many todo's
 
 #include <iostream>
+#include <stdlib.h>
 #include <qlayout.h>
 #include <qcursor.h>
 #include <qregexp.h>
diff -Naur mythstream-0.18_1-old/libs/streamstatus.cpp mythstream-0.18_1-new/libs/streamstatus.cpp
--- mythstream-0.18_1-old/libs/streamstatus.cpp	2007-09-18 17:13:45.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstatus.cpp	2008-01-16 12:51:51.000000000 -0800
@@ -23,6 +23,7 @@
 
 #include <iostream>
 
+#include <stdlib.h>
 #include <qapplication.h>
 #include <qnetwork.h>
 #include <qdatetime.h>
diff -Naur mythstream-0.18_1-old/libs/streamstorage.cpp mythstream-0.18_1-new/libs/streamstorage.cpp
--- mythstream-0.18_1-old/libs/streamstorage.cpp	2007-09-23 11:26:15.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstorage.cpp	2008-01-16 12:48:57.000000000 -0800
@@ -35,6 +35,7 @@
 #include "streamstorage.h"
 #include "configelements.h"
 
+#include <stdlib.h>
 #include <sys/stat.h>
 
 using namespace std;
