diff -Naur mythstream-0.17_2-old/mythstream.pro mythstream-0.17_2-new/mythstream.pro
--- mythstream-0.17_2-old/mythstream.pro	2006-11-30 14:01:57.000000000 -0800
+++ mythstream-0.17_2-new/mythstream.pro	2006-11-30 14:03:06.000000000 -0800
@@ -1,6 +1,6 @@
 TEMPLATE = subdirs
 
 # Directories
-SUBDIRS = mythstream
-#SUBDIRS = mythstream alt_parsers
+#SUBDIRS = mythstream
+SUBDIRS = mythstream alt_parsers
 
diff -Naur mythstream-0.17_2-old/settings.pro mythstream-0.17_2-new/settings.pro
--- mythstream-0.17_2-old/settings.pro	2006-11-30 14:02:22.000000000 -0800
+++ mythstream-0.17_2-new/settings.pro	2006-11-30 14:03:06.000000000 -0800
@@ -13,9 +13,9 @@
 DEFINES += SHELL=2
 DEFINES += EXEC=3
 
-DEFINES += PARSER=PERL
+#DEFINES += PARSER=PERL
 #DEFINES += PARSER=SHELL
-#DEFINES += PARSER=EXEC
+DEFINES += PARSER=EXEC
 
 DEFINES += _GNU_SOURCE
 DEFINES += PREFIX=\"$${PREFIX}\"
