diff -Naur gspcav1-20070508-old/Makefile gspcav1-20070508-new/Makefile
--- gspcav1-20070508-old/Makefile	2007-05-08 08:08:03.000000000 -0700
+++ gspcav1-20070508-new/Makefile	2007-05-12 10:41:09.000000000 -0700
@@ -48,11 +48,9 @@
 	rm -f $(MODULE_INSTALLDIR)spca5xx.ko
 	rm -f $(MODULE_INSTALLDIR2)gspca.ko
 	install -c -m 0644 gspca.ko $(MODULE_INSTALLDIR)
-	/sbin/depmod -ae
 
 uninstall:
 	rm -f $(MODULE_INSTALLDIR)gspca.ko
-	/sbin/depmod -aq
 
 endif
 
