diff -Naur gspcav1-20071220-old/Makefile gspcav1-20071220-new/Makefile
--- gspcav1-20071220-old/Makefile	2004-12-31 16:59:58.000000000 -0800
+++ gspcav1-20071220-new/Makefile	2007-12-21 00:59:53.000000000 -0800
@@ -48,11 +48,9 @@
 	rm -f $(MODULE_INSTALLDIR)spca5xx.ko
 	rm -f $(MODULE_INSTALLDIR2)gspca.ko
 	install -c -m 0644 gspca.ko $(MODULE_INSTALLDIR)
-	/sbin/depmod -ae
 
 uninstall:
 	rm -f $(MODULE_INSTALLDIR)gspca.ko
-	/sbin/depmod -aq
 
 endif
 
