diff -Naur curl-7.19.2-old/lib/ssluse.c curl-7.19.2-new/lib/ssluse.c
--- curl-7.19.2-old/lib/ssluse.c	2008-11-11 05:20:02.000000000 -0800
+++ curl-7.19.2-new/lib/ssluse.c	2008-11-13 17:40:38.000000000 -0800
@@ -923,10 +923,6 @@
   int i;
   int year=0,month=0,day=0,hour=0,minute=0,second=0;
 
-#ifdef CURL_DISABLE_VERBOSE_STRINGS
-  (void)prefix;
-#endif
-
   i=tm->length;
   asn1_string=(const char *)tm->data;
 
