diff -Naur curl-7.19.1-old/lib/ssluse.c curl-7.19.1-new/lib/ssluse.c
--- curl-7.19.1-old/lib/ssluse.c	2008-10-23 04:49:19.000000000 -0700
+++ curl-7.19.1-new/lib/ssluse.c	2008-11-05 21:29:36.000000000 -0800
@@ -923,10 +923,6 @@
   int i;
   int year=0,month=0,day=0,hour=0,minute=0,second=0;
 
-#ifdef CURL_DISABLE_VERBOSE_STRINGS
-  (void)prefix;
-#endif
-
   i=tm->length;
   asn1_string=(const char *)tm->data;
 
