diff -Naur myththemes-14921-old/cpsvndir myththemes-14921-new/cpsvndir
--- myththemes-14921-old/cpsvndir	2007-11-22 10:42:29.000000000 -0800
+++ myththemes-14921-new/cpsvndir	2007-11-22 10:43:02.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
