diff -Naur myththemes-15447-old/cpsvndir myththemes-15447-new/cpsvndir
--- myththemes-15447-old/cpsvndir	2008-01-16 20:09:28.000000000 -0800
+++ myththemes-15447-new/cpsvndir	2008-01-16 20:10:02.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
