diff -Naur mythplugins-0.19-old/mythdvd/mtd/jobthread.cpp mythplugins-0.19-new/mythdvd/mtd/jobthread.cpp
--- mythplugins-0.19-old/mythdvd/mtd/jobthread.cpp	2005-12-12 18:21:44.000000000 -0800
+++ mythplugins-0.19-new/mythdvd/mtd/jobthread.cpp	2006-02-19 19:11:45.000000000 -0800
@@ -11,6 +11,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include <iostream>
 using namespace std;
diff -Naur mythplugins-0.19-old/mythdvd/mtd/logging.cpp mythplugins-0.19-new/mythdvd/mtd/logging.cpp
--- mythplugins-0.19-old/mythdvd/mtd/logging.cpp	2003-07-28 08:34:51.000000000 -0700
+++ mythplugins-0.19-new/mythdvd/mtd/logging.cpp	2006-02-19 19:11:45.000000000 -0800
@@ -9,6 +9,7 @@
 */
 #include "logging.h"
 
+#include <unistd.h>
 #include <qdatetime.h>
 
 #include <mythtv/mythcontext.h>
diff -Naur mythplugins-0.19-old/mythdvd/mtd/mtd.cpp mythplugins-0.19-new/mythdvd/mtd/mtd.cpp
--- mythplugins-0.19-old/mythdvd/mtd/mtd.cpp	2005-08-18 18:59:51.000000000 -0700
+++ mythplugins-0.19-new/mythdvd/mtd/mtd.cpp	2006-02-19 19:11:45.000000000 -0800
@@ -8,6 +8,7 @@
 
 */
 
+#include <unistd.h>
 #include <qstringlist.h>
 #include <qregexp.h>
 #include <qdir.h>
diff -Naur mythplugins-0.19-old/mythweb/config/canned_searches.php mythplugins-0.19-new/mythweb/config/canned_searches.php
--- mythplugins-0.19-old/mythweb/config/canned_searches.php	2006-02-01 23:59:05.000000000 -0800
+++ mythplugins-0.19-new/mythweb/config/canned_searches.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * Each of these searches will show up as an advanced search
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 02:59:05 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-01 23:59:05 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8837 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/config/conf.php mythplugins-0.19-new/mythweb/config/conf.php
--- mythplugins-0.19-old/mythweb/config/conf.php	2006-02-02 00:21:54.000000000 -0800
+++ mythplugins-0.19-new/mythweb/config/conf.php	2006-02-19 19:11:45.000000000 -0800
@@ -7,7 +7,7 @@
  * a php interface to change it (someday...)
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 03:21:54 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-02 00:21:54 -0800 (Thu, 02 Feb 2006) $
  * @version     $Revision: 8838 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/config/theme_default.php mythplugins-0.19-new/mythweb/config/theme_default.php
--- mythplugins-0.19-old/mythweb/config/theme_default.php	2006-02-02 00:21:54.000000000 -0800
+++ mythplugins-0.19-new/mythweb/config/theme_default.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configuration options for the default MythWeb theme
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 03:21:54 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-02 00:21:54 -0800 (Thu, 02 Feb 2006) $
  * @version     $Revision: 8838 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/config/theme_wap.php mythplugins-0.19-new/mythweb/config/theme_wap.php
--- mythplugins-0.19-old/mythweb/config/theme_wap.php	2006-02-02 00:21:54.000000000 -0800
+++ mythplugins-0.19-new/mythweb/config/theme_wap.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configuration options for the MythWeb WAP theme
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 03:21:54 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-02 00:21:54 -0800 (Thu, 02 Feb 2006) $
  * @version     $Revision: 8838 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/config/theme_wml.php mythplugins-0.19-new/mythweb/config/theme_wml.php
--- mythplugins-0.19-old/mythweb/config/theme_wml.php	2006-02-02 00:21:54.000000000 -0800
+++ mythplugins-0.19-new/mythweb/config/theme_wml.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configuration options for the MythWeb WML theme
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 03:21:54 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-02 00:21:54 -0800 (Thu, 02 Feb 2006) $
  * @version     $Revision: 8838 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/.htaccess mythplugins-0.19-new/mythweb/.htaccess
--- mythplugins-0.19-old/mythweb/.htaccess	2006-01-16 15:32:30.000000000 -0800
+++ mythplugins-0.19-new/mythweb/.htaccess	2006-02-19 19:11:45.000000000 -0800
@@ -7,7 +7,7 @@
 # See http://www.mythtv.org/ for information about MythTV itself.
 #
 # @url       $URL$
-# @date      $Date: 2006-01-16 18:32:30 -0500 (Mon, 16 Jan 2006) $
+# @date      $Date: 2006-01-16 15:32:30 -0800 (Mon, 16 Jan 2006) $
 # @version   $Revision: 8627 $
 # @author    $Author: xris $
 #
diff -Naur mythplugins-0.19-old/mythweb/includes/channels.php mythplugins-0.19-new/mythweb/includes/channels.php
--- mythplugins-0.19-old/mythweb/includes/channels.php	2006-02-09 22:15:17.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/channels.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * The Channel object, and a couple of related subroutines.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 01:15:17 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 22:15:17 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8914 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/css.php mythplugins-0.19-new/mythweb/includes/css.php
--- mythplugins-0.19-old/mythweb/includes/css.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/css.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Various routines that deal with figuring out and/or displaying css.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/db.php mythplugins-0.19-new/mythweb/includes/db.php
--- mythplugins-0.19-old/mythweb/includes/db.php	2006-01-28 19:19:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/db.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * The Database class and associated statement libraries.
  *
  * @url       $URL: svn+ssh://xris@svn.siliconmechanics.com/var/svn/web/trunk/shared_code/includes/db.php $
- * @date      $Date: 2006-01-28 22:19:02 -0500 (Sat, 28 Jan 2006) $
+ * @date      $Date: 2006-01-28 19:19:02 -0800 (Sat, 28 Jan 2006) $
  * @version   $Revision: 8758 $
  * @author    $Author: xris $
  * @license   GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/db_update.php mythplugins-0.19-new/mythweb/includes/db_update.php
--- mythplugins-0.19-old/mythweb/includes/db_update.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/db_update.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Make sure the database is at the most recent version
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/errordisplay.php mythplugins-0.19-new/mythweb/includes/errordisplay.php
--- mythplugins-0.19-old/mythweb/includes/errordisplay.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/errordisplay.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * This file contains a number of error-display related routines.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/errors.php mythplugins-0.19-new/mythweb/includes/errors.php
--- mythplugins-0.19-old/mythweb/includes/errors.php	2006-02-01 20:03:59.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/errors.php	2006-02-19 19:11:45.000000000 -0800
@@ -5,7 +5,7 @@
  * User-defined error handler.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:03:59 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:03:59 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8829 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/init.php mythplugins-0.19-new/mythweb/includes/init.php
--- mythplugins-0.19-old/mythweb/includes/init.php	2006-02-01 21:36:08.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -7,7 +7,7 @@
  * shared files for the entire program.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 00:36:08 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-01 21:36:08 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8832 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/mobile.php mythplugins-0.19-new/mythweb/includes/mobile.php
--- mythplugins-0.19-old/mythweb/includes/mobile.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/mobile.php	2006-02-19 19:11:45.000000000 -0800
@@ -24,7 +24,7 @@
  * code is intented to be easily re-usable in different applications.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/mouseovers.php mythplugins-0.19-new/mythweb/includes/mouseovers.php
--- mythplugins-0.19-old/mythweb/includes/mouseovers.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/mouseovers.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * mouseover events.  It requires the mouseovers.js file.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/mythbackend.php mythplugins-0.19-new/mythweb/includes/mythbackend.php
--- mythplugins-0.19-old/mythweb/includes/mythbackend.php	2006-01-17 13:43:30.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/mythbackend.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Routines that allow mythweb to communicate with mythbackend
  *
  * @url         $URL$
- * @date        $Date: 2006-01-17 16:43:30 -0500 (Tue, 17 Jan 2006) $
+ * @date        $Date: 2006-01-17 13:43:30 -0800 (Tue, 17 Jan 2006) $
  * @version     $Revision: 8634 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/programs.php mythplugins-0.19-new/mythweb/includes/programs.php
--- mythplugins-0.19-old/mythweb/includes/programs.php	2006-02-09 23:09:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/programs.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * This contains the Program class
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 02:09:02 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 23:09:02 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8915 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/recording_schedules.php mythplugins-0.19-new/mythweb/includes/recording_schedules.php
--- mythplugins-0.19-old/mythweb/includes/recording_schedules.php	2006-02-06 18:57:11.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/recording_schedules.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * The Schedule object and a couple of related subroutines.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-06 21:57:11 -0500 (Mon, 06 Feb 2006) $
+ * @date        $Date: 2006-02-06 18:57:11 -0800 (Mon, 06 Feb 2006) $
  * @version     $Revision: 8887 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/recordings.php mythplugins-0.19-new/mythweb/includes/recordings.php
--- mythplugins-0.19-old/mythweb/includes/recordings.php	2006-02-09 22:15:17.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/recordings.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * The Recording object, and a couple of related subroutines.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 01:15:17 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 22:15:17 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8914 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/session.php mythplugins-0.19-new/mythweb/includes/session.php
--- mythplugins-0.19-old/mythweb/includes/session.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/session.php	2006-02-19 19:11:45.000000000 -0800
@@ -6,7 +6,7 @@
  * *after* the database connection has been established.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/sorting.php mythplugins-0.19-new/mythweb/includes/sorting.php
--- mythplugins-0.19-old/mythweb/includes/sorting.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/sorting.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Routines for sorting Program objects
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/translate.php mythplugins-0.19-new/mythweb/includes/translate.php
--- mythplugins-0.19-old/mythweb/includes/translate.php	2006-02-01 20:32:34.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/translate.php	2006-02-19 19:11:45.000000000 -0800
@@ -9,7 +9,7 @@
  * preferred language based on the browser's specifications.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:32:34 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:32:34 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8830 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/includes/utils.php mythplugins-0.19-new/mythweb/includes/utils.php
--- mythplugins-0.19-old/mythweb/includes/utils.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/includes/utils.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Utility routines used throughout mythweb
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/js/ajax.js mythplugins-0.19-new/mythweb/js/ajax.js
--- mythplugins-0.19-old/mythweb/js/ajax.js	2005-10-22 10:32:26.000000000 -0700
+++ mythplugins-0.19-new/mythweb/js/ajax.js	2006-02-19 19:11:45.000000000 -0800
@@ -2,7 +2,7 @@
  * A small but growing library of generic AJAX routines.
  *
  * @url         $URL$
- * @date        $Date: 2005-10-22 13:32:26 -0400 (Sat, 22 Oct 2005) $
+ * @date        $Date: 2005-10-22 10:32:26 -0700 (Sat, 22 Oct 2005) $
  * @version     $Revision: 7561 $
  * @author      $Author: xris $
  * @license     LGPL
diff -Naur mythplugins-0.19-old/mythweb/js/browser.js mythplugins-0.19-new/mythweb/js/browser.js
--- mythplugins-0.19-old/mythweb/js/browser.js	2006-02-01 19:34:43.000000000 -0800
+++ mythplugins-0.19-new/mythweb/js/browser.js	2006-02-19 19:11:45.000000000 -0800
@@ -15,8 +15,8 @@
  * support for anything earlier than version 4 browsers because of their lack
  * of the RegExp object.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/js/browser.js $
- * @date        $Date: 2006-02-01 22:34:43 -0500 (Wed, 01 Feb 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/js/browser.js $
+ * @date        $Date: 2006-02-01 19:34:43 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8826 $
  * @author      $Author: xris $
  * @copyright   Silicon Mechanics
diff -Naur mythplugins-0.19-old/mythweb/js/init.js mythplugins-0.19-new/mythweb/js/init.js
--- mythplugins-0.19-old/mythweb/js/init.js	2005-11-02 23:09:14.000000000 -0800
+++ mythplugins-0.19-new/mythweb/js/init.js	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * other javascript files.
  *
  * @url         $URL$
- * @date        $Date: 2005-11-03 02:09:14 -0500 (Thu, 03 Nov 2005) $
+ * @date        $Date: 2005-11-02 23:09:14 -0800 (Wed, 02 Nov 2005) $
  * @version     $Revision: 7680 $
  * @author      $Author: xris $
  * @license     LGPL
diff -Naur mythplugins-0.19-old/mythweb/js/mouseovers.js mythplugins-0.19-new/mythweb/js/mouseovers.js
--- mythplugins-0.19-old/mythweb/js/mouseovers.js	2006-01-24 19:27:24.000000000 -0800
+++ mythplugins-0.19-new/mythweb/js/mouseovers.js	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Primarily used for interactive menus
  *
  * @url         $URL: svn+ssh://xris@svn.siliconmechanics.com/var/svn/web/trunk/shared_code/js/mouseovers.js $
- * @date        $Date: 2006-01-24 22:27:24 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:27:24 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8708 $
  * @author      $Author: xris $
  * @license     LGPL
diff -Naur mythplugins-0.19-old/mythweb/js/utils.js mythplugins-0.19-new/mythweb/js/utils.js
--- mythplugins-0.19-old/mythweb/js/utils.js	2005-10-22 10:32:26.000000000 -0700
+++ mythplugins-0.19-new/mythweb/js/utils.js	2006-02-19 19:11:45.000000000 -0800
@@ -2,7 +2,7 @@
  * A random assortment of javascript utility routines
  *
  * @url         $URL$
- * @date        $Date: 2005-10-22 13:32:26 -0400 (Sat, 22 Oct 2005) $
+ * @date        $Date: 2005-10-22 10:32:26 -0700 (Sat, 22 Oct 2005) $
  * @version     $Revision: 7561 $
  * @author      $Author: xris $
  * @license     LGPL
diff -Naur mythplugins-0.19-old/mythweb/js/visibility.js mythplugins-0.19-new/mythweb/js/visibility.js
--- mythplugins-0.19-old/mythweb/js/visibility.js	2005-10-22 10:32:26.000000000 -0700
+++ mythplugins-0.19-new/mythweb/js/visibility.js	2006-02-19 19:11:45.000000000 -0800
@@ -2,7 +2,7 @@
  * Functions to show/hide html elements
  *
  * @url         $URL$
- * @date        $Date: 2005-10-22 13:32:26 -0400 (Sat, 22 Oct 2005) $
+ * @date        $Date: 2005-10-22 10:32:26 -0700 (Sat, 22 Oct 2005) $
  * @version     $Revision: 7561 $
  * @author      $Author: xris $
  * @license     LGPL
diff -Naur mythplugins-0.19-old/mythweb/languages/Finnish.php mythplugins-0.19-new/mythweb/languages/Finnish.php
--- mythplugins-0.19-old/mythweb/languages/Finnish.php	2006-02-10 18:48:14.000000000 -0800
+++ mythplugins-0.19-new/mythweb/languages/Finnish.php	2006-02-19 19:11:45.000000000 -0800
@@ -1,6 +1,6 @@
 <?php
 /*
- *  $Date: 2006-02-10 21:48:14 -0500 (Fri, 10 Feb 2006) $
+ *  $Date: 2006-02-10 18:48:14 -0800 (Fri, 10 Feb 2006) $
  *  $Revision: 8920 $
  *  $Author: xris $
  *
diff -Naur mythplugins-0.19-old/mythweb/modules/backend_log/handler.php mythplugins-0.19-new/mythweb/modules/backend_log/handler.php
--- mythplugins-0.19-old/mythweb/modules/backend_log/handler.php	2005-12-07 21:00:04.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/backend_log/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * View the logs
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/backend_log/handler.php $
- * @date        $Date: 2005-12-08 00:00:04 -0500 (Thu, 08 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/backend_log/handler.php $
+ * @date        $Date: 2005-12-07 21:00:04 -0800 (Wed, 07 Dec 2005) $
  * @version     $Revision: 8171 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/backend_log/init.php mythplugins-0.19-new/mythweb/modules/backend_log/init.php
--- mythplugins-0.19-old/mythweb/modules/backend_log/init.php	2005-12-15 22:55:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/backend_log/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb Backend Logs module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/backend_log/init.php $
- * @date        $Date: 2005-12-16 01:55:02 -0500 (Fri, 16 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/backend_log/init.php $
+ * @date        $Date: 2005-12-15 22:55:02 -0800 (Thu, 15 Dec 2005) $
  * @version     $Revision: 8285 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/movietimes/init.php mythplugins-0.19-new/mythweb/modules/movietimes/init.php
--- mythplugins-0.19-old/mythweb/modules/movietimes/init.php	2006-01-28 19:19:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/movietimes/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb Movie Times module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/movietimes/init.php $
- * @date        $Date: 2006-01-28 22:19:02 -0500 (Sat, 28 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/movietimes/init.php $
+ * @date        $Date: 2006-01-28 19:19:02 -0800 (Sat, 28 Jan 2006) $
  * @version     $Revision: 8758 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/music/handler.php mythplugins-0.19-new/mythweb/modules/music/handler.php
--- mythplugins-0.19-old/mythweb/modules/music/handler.php	2005-12-21 23:17:32.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/music/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * MythMusic browser
  *
  * @url         $URL: svn+ssh://xris@cvs.mythtv.org/var/lib/svn/trunk/mythplugins/mythweb/modules/music/init.php $
- * @date        $Date: 2005-12-22 02:17:32 -0500 (Thu, 22 Dec 2005) $
+ * @date        $Date: 2005-12-21 23:17:32 -0800 (Wed, 21 Dec 2005) $
  * @version     $Revision: 8347 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/music/init.php mythplugins-0.19-new/mythweb/modules/music/init.php
--- mythplugins-0.19-old/mythweb/modules/music/init.php	2006-01-28 19:19:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/music/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb Music module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/music/init.php $
- * @date        $Date: 2006-01-28 22:19:02 -0500 (Sat, 28 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/music/init.php $
+ * @date        $Date: 2006-01-28 19:19:02 -0800 (Sat, 28 Jan 2006) $
  * @version     $Revision: 8758 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/settings/channels.php mythplugins-0.19-new/mythweb/modules/settings/channels.php
--- mythplugins-0.19-old/mythweb/modules/settings/channels.php	2005-12-30 16:21:51.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/settings/channels.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configure MythTV Channels
  *
  * @url         $URL$
- * @date        $Date: 2005-12-30 19:21:51 -0500 (Fri, 30 Dec 2005) $
+ * @date        $Date: 2005-12-30 16:21:51 -0800 (Fri, 30 Dec 2005) $
  * @version     $Revision: 8441 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/settings/handler.php mythplugins-0.19-new/mythweb/modules/settings/handler.php
--- mythplugins-0.19-old/mythweb/modules/settings/handler.php	2005-12-14 23:33:52.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/settings/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Handler for all MythWeb Settings routines
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/settings/handler.php $
- * @date        $Date: 2005-12-15 02:33:52 -0500 (Thu, 15 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/settings/handler.php $
+ * @date        $Date: 2005-12-14 23:33:52 -0800 (Wed, 14 Dec 2005) $
  * @version     $Revision: 8273 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/settings/init.php mythplugins-0.19-new/mythweb/modules/settings/init.php
--- mythplugins-0.19-old/mythweb/modules/settings/init.php	2005-12-30 16:08:13.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/settings/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb settings module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/settings/init.php $
- * @date        $Date: 2005-12-30 19:08:13 -0500 (Fri, 30 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/settings/init.php $
+ * @date        $Date: 2005-12-30 16:08:13 -0800 (Fri, 30 Dec 2005) $
  * @version     $Revision: 8440 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/settings/keys.php mythplugins-0.19-new/mythweb/modules/settings/keys.php
--- mythplugins-0.19-old/mythweb/modules/settings/keys.php	2006-01-14 11:52:53.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/settings/keys.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configure MythTV Key Bindings
  *
  * @url         $URL$
- * @date        $Date: 2006-01-14 14:52:53 -0500 (Sat, 14 Jan 2006) $
+ * @date        $Date: 2006-01-14 11:52:53 -0800 (Sat, 14 Jan 2006) $
  * @version     $Revision: 8607 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/settings/session.php mythplugins-0.19-new/mythweb/modules/settings/session.php
--- mythplugins-0.19-old/mythweb/modules/settings/session.php	2006-01-28 11:36:32.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/settings/session.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Display/save mythweb session settings
  *
  * @url         $URL$
- * @date        $Date: 2006-01-28 14:36:32 -0500 (Sat, 28 Jan 2006) $
+ * @date        $Date: 2006-01-28 11:36:32 -0800 (Sat, 28 Jan 2006) $
  * @version     $Revision: 8744 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/status/handler.php mythplugins-0.19-new/mythweb/modules/status/handler.php
--- mythplugins-0.19-old/mythweb/modules/status/handler.php	2005-12-07 21:00:04.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/status/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * port 6544)
  *
  * @url         $URL$
- * @date        $Date: 2005-12-08 00:00:04 -0500 (Thu, 08 Dec 2005) $
+ * @date        $Date: 2005-12-07 21:00:04 -0800 (Wed, 07 Dec 2005) $
  * @version     $Revision: 8171 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/status/init.php mythplugins-0.19-new/mythweb/modules/status/init.php
--- mythplugins-0.19-old/mythweb/modules/status/init.php	2005-12-15 22:55:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/status/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb Status module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/status/init.php $
- * @date        $Date: 2005-12-16 01:55:02 -0500 (Fri, 16 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/status/init.php $
+ * @date        $Date: 2005-12-15 22:55:02 -0800 (Thu, 15 Dec 2005) $
  * @version     $Revision: 8285 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/stream/handler.php mythplugins-0.19-new/mythweb/modules/stream/handler.php
--- mythplugins-0.19-old/mythweb/modules/stream/handler.php	2005-12-16 21:27:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/stream/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,8 +3,8 @@
  * This module would work fine, except that PHP freaks out with files larger
  * than 2G, so it's disabled and left half-finished.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/stream/handler.php $
- * @date        $Date: 2005-12-17 00:27:36 -0500 (Sat, 17 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/stream/handler.php $
+ * @date        $Date: 2005-12-16 21:27:36 -0800 (Fri, 16 Dec 2005) $
  * @version     $Revision: 8297 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/stream/init.php mythplugins-0.19-new/mythweb/modules/stream/init.php
--- mythplugins-0.19-old/mythweb/modules/stream/init.php	2005-12-07 21:00:04.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/stream/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb streaming module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/stream/init.php $
- * @date        $Date: 2005-12-08 00:00:04 -0500 (Thu, 08 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/stream/init.php $
+ * @date        $Date: 2005-12-07 21:00:04 -0800 (Wed, 07 Dec 2005) $
  * @version     $Revision: 8171 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/channel.php mythplugins-0.19-new/mythweb/modules/tv/channel.php
--- mythplugins-0.19-old/mythweb/modules/tv/channel.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/channel.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Show the current lineup for a specific channel
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/detail.php mythplugins-0.19-new/mythweb/modules/tv/detail.php
--- mythplugins-0.19-old/mythweb/modules/tv/detail.php	2006-02-06 18:57:11.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/detail.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * commands.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-06 21:57:11 -0500 (Mon, 06 Feb 2006) $
+ * @date        $Date: 2006-02-06 18:57:11 -0800 (Mon, 06 Feb 2006) $
  * @version     $Revision: 8887 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/handler.php mythplugins-0.19-new/mythweb/modules/tv/handler.php
--- mythplugins-0.19-old/mythweb/modules/tv/handler.php	2005-12-31 16:00:31.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Handler for all MythWeb TV routines
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/tv/handler.php $
- * @date        $Date: 2005-12-31 19:00:31 -0500 (Sat, 31 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/tv/handler.php $
+ * @date        $Date: 2005-12-31 16:00:31 -0800 (Sat, 31 Dec 2005) $
  * @version     $Revision: 8460 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/init.php mythplugins-0.19-new/mythweb/modules/tv/init.php
--- mythplugins-0.19-old/mythweb/modules/tv/init.php	2005-12-31 16:00:31.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb TV module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/tv/init.php $
- * @date        $Date: 2005-12-31 19:00:31 -0500 (Sat, 31 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/tv/init.php $
+ * @date        $Date: 2005-12-31 16:00:31 -0800 (Sat, 31 Dec 2005) $
  * @version     $Revision: 8460 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/list.php mythplugins-0.19-new/mythweb/modules/tv/list.php
--- mythplugins-0.19-old/mythweb/modules/tv/list.php	2005-11-10 23:25:35.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/list.php	2006-02-19 19:11:45.000000000 -0800
@@ -6,7 +6,7 @@
  * This is the default viewing mode, and shows the current program listings.
  *
  * @url         $URL$
- * @date        $Date: 2005-11-11 02:25:35 -0500 (Fri, 11 Nov 2005) $
+ * @date        $Date: 2005-11-10 23:25:35 -0800 (Thu, 10 Nov 2005) $
  * @version     $Revision: 7840 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/recorded.php mythplugins-0.19-new/mythweb/modules/tv/recorded.php
--- mythplugins-0.19-old/mythweb/modules/tv/recorded.php	2006-02-09 22:15:17.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/recorded.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * view and manipulate recorded programs.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 01:15:17 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 22:15:17 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8914 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/schedules_custom.php mythplugins-0.19-new/mythweb/modules/tv/schedules_custom.php
--- mythplugins-0.19-old/mythweb/modules/tv/schedules_custom.php	2006-02-01 20:03:21.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/schedules_custom.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Schedule a custom recording by manually specifying various search options
  *
  * @url         $URL$
- * @date        $Date: 2006-02-01 23:03:21 -0500 (Wed, 01 Feb 2006) $
+ * @date        $Date: 2006-02-01 20:03:21 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8828 $
  * @author      $Author: xris $
  *
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/schedules_manual.php mythplugins-0.19-new/mythweb/modules/tv/schedules_manual.php
--- mythplugins-0.19-old/mythweb/modules/tv/schedules_manual.php	2006-02-09 22:15:17.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/schedules_manual.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Schedule a custom recording by manually specifying starttime and length
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 01:15:17 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 22:15:17 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8914 $
  * @author      $Author: xris $
  *
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/schedules.php mythplugins-0.19-new/mythweb/modules/tv/schedules.php
--- mythplugins-0.19-old/mythweb/modules/tv/schedules.php	2005-12-31 14:26:52.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/schedules.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * View all recording schedules
  *
  * @url         $URL$
- * @date        $Date: 2005-12-31 17:26:52 -0500 (Sat, 31 Dec 2005) $
+ * @date        $Date: 2005-12-31 14:26:52 -0800 (Sat, 31 Dec 2005) $
  * @version     $Revision: 8456 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/searches.php mythplugins-0.19-new/mythweb/modules/tv/searches.php
--- mythplugins-0.19-old/mythweb/modules/tv/searches.php	2005-12-07 20:35:19.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/searches.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * An index for handy SQL searches in the listings data
  *
  * @url         $URL$
- * @date        $Date: 2005-12-07 23:35:19 -0500 (Wed, 07 Dec 2005) $
+ * @date        $Date: 2005-12-07 20:35:19 -0800 (Wed, 07 Dec 2005) $
  * @version     $Revision: 8169 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/search.php mythplugins-0.19-new/mythweb/modules/tv/search.php
--- mythplugins-0.19-old/mythweb/modules/tv/search.php	2006-02-10 18:45:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/search.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Searches the database for programs matching a particular query.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 21:45:02 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-10 18:45:02 -0800 (Fri, 10 Feb 2006) $
  * @version     $Revision: 8919 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/tv/upcoming.php mythplugins-0.19-new/mythweb/modules/tv/upcoming.php
--- mythplugins-0.19-old/mythweb/modules/tv/upcoming.php	2006-02-06 18:57:11.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/tv/upcoming.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * View and fix scheduling conflicts.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-06 21:57:11 -0500 (Mon, 06 Feb 2006) $
+ * @date        $Date: 2006-02-06 18:57:11 -0800 (Mon, 06 Feb 2006) $
  * @version     $Revision: 8887 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/video/edit.php mythplugins-0.19-new/mythweb/modules/video/edit.php
--- mythplugins-0.19-old/mythweb/modules/video/edit.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/video/edit.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Edit MythVideo file information
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  *
diff -Naur mythplugins-0.19-old/mythweb/modules/video/handler.php mythplugins-0.19-new/mythweb/modules/video/handler.php
--- mythplugins-0.19-old/mythweb/modules/video/handler.php	2006-01-25 18:45:37.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/video/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * View MythVideo files
  *
  * @url         $URL$
- * @date        $Date: 2006-01-25 21:45:37 -0500 (Wed, 25 Jan 2006) $
+ * @date        $Date: 2006-01-25 18:45:37 -0800 (Wed, 25 Jan 2006) $
  * @version     $Revision: 8727 $
  * @author      $Author: xris $
  *
diff -Naur mythplugins-0.19-old/mythweb/modules/video/init.php mythplugins-0.19-new/mythweb/modules/video/init.php
--- mythplugins-0.19-old/mythweb/modules/video/init.php	2006-01-28 19:19:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/video/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb Video module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/video/init.php $
- * @date        $Date: 2006-01-28 22:19:02 -0500 (Sat, 28 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/video/init.php $
+ * @date        $Date: 2006-01-28 19:19:02 -0800 (Sat, 28 Jan 2006) $
  * @version     $Revision: 8758 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/weather/handler.php mythplugins-0.19-new/mythweb/modules/weather/handler.php
--- mythplugins-0.19-old/mythweb/modules/weather/handler.php	2005-12-15 22:55:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/weather/handler.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Handler for the Weather module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/weather/handler.php $
- * @date        $Date: 2005-12-16 01:55:02 -0500 (Fri, 16 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/weather/handler.php $
+ * @date        $Date: 2005-12-15 22:55:02 -0800 (Thu, 15 Dec 2005) $
  * @version     $Revision: 8285 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/weather/init.php mythplugins-0.19-new/mythweb/modules/weather/init.php
--- mythplugins-0.19-old/mythweb/modules/weather/init.php	2005-12-15 22:55:02.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/weather/init.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Initialization routines for the MythWeb Weather module
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/weather/init.php $
- * @date        $Date: 2005-12-16 01:55:02 -0500 (Fri, 16 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/weather/init.php $
+ * @date        $Date: 2005-12-15 22:55:02 -0800 (Thu, 15 Dec 2005) $
  * @version     $Revision: 8285 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/modules/welcome.php mythplugins-0.19-new/mythweb/modules/welcome.php
--- mythplugins-0.19-old/mythweb/modules/welcome.php	2005-11-06 23:56:40.000000000 -0800
+++ mythplugins-0.19-new/mythweb/modules/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page that lists the available mythweb sections
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/modules/welcome.php $
- * @date        $Date: 2005-11-07 02:56:40 -0500 (Mon, 07 Nov 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/modules/welcome.php $
+ * @date        $Date: 2005-11-06 23:56:40 -0800 (Sun, 06 Nov 2005) $
  * @version     $Revision: 7776 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/mythweb.php mythplugins-0.19-new/mythweb/mythweb.php
--- mythplugins-0.19-old/mythweb/mythweb.php	2006-01-30 16:05:51.000000000 -0800
+++ mythplugins-0.19-new/mythweb/mythweb.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * The main brain script for all of MythWeb.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/mythweb.php $
- * @date        $Date: 2006-01-30 19:05:51 -0500 (Mon, 30 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/mythweb.php $
+ * @date        $Date: 2006-01-30 16:05:51 -0800 (Mon, 30 Jan 2006) $
  * @version     $Revision: 8793 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/backend_log/backend_log.php mythplugins-0.19-new/mythweb/themes/default/backend_log/backend_log.php
--- mythplugins-0.19-old/mythweb/themes/default/backend_log/backend_log.php	2005-12-05 20:10:28.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/backend_log/backend_log.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,8 +3,8 @@
  * The display code for the main welcome page that lists the available mythweb
  * sections.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/backend_log/backend_log.php $
- * @date        $Date: 2005-12-05 23:10:28 -0500 (Mon, 05 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/backend_log/backend_log.php $
+ * @date        $Date: 2005-12-05 20:10:28 -0800 (Mon, 05 Dec 2005) $
  * @version     $Revision: 8132 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/backend_log/welcome.php mythplugins-0.19-new/mythweb/themes/default/backend_log/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/backend_log/welcome.php	2005-12-12 23:23:18.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/backend_log/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the status module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/backend_log/welcome.php $
- * @date        $Date: 2005-12-13 02:23:18 -0500 (Tue, 13 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/backend_log/welcome.php $
+ * @date        $Date: 2005-12-12 23:23:18 -0800 (Mon, 12 Dec 2005) $
  * @version     $Revision: 8252 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/footer.php mythplugins-0.19-new/mythweb/themes/default/footer.php
--- mythplugins-0.19-old/mythweb/themes/default/footer.php	2005-11-07 22:43:55.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/footer.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * This footer file is shared by all MythWeb modules.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/footer.php $
- * @date        $Date: 2005-11-08 01:43:55 -0500 (Tue, 08 Nov 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/footer.php $
+ * @date        $Date: 2005-11-07 22:43:55 -0800 (Mon, 07 Nov 2005) $
  * @version     $Revision: 7795 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/header.php mythplugins-0.19-new/mythweb/themes/default/header.php
--- mythplugins-0.19-old/mythweb/themes/default/header.php	2006-02-10 18:48:14.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/header.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * This header file is shared by all MythWeb modules.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/header.php $
- * @date        $Date: 2006-02-10 21:48:14 -0500 (Fri, 10 Feb 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/header.php $
+ * @date        $Date: 2006-02-10 18:48:14 -0800 (Fri, 10 Feb 2006) $
  * @version     $Revision: 8920 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/movietimes/welcome.php mythplugins-0.19-new/mythweb/themes/default/movietimes/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/movietimes/welcome.php	2005-12-12 23:23:18.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/movietimes/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the Movie Times module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/movietimes/welcome.php $
- * @date        $Date: 2005-12-13 02:23:18 -0500 (Tue, 13 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/movietimes/welcome.php $
+ * @date        $Date: 2005-12-12 23:23:18 -0800 (Mon, 12 Dec 2005) $
  * @version     $Revision: 8252 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/music/music.php mythplugins-0.19-new/mythweb/themes/default/music/music.php
--- mythplugins-0.19-old/mythweb/themes/default/music/music.php	2005-12-17 01:16:41.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/music/music.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * MythMusic browser.
  *
  * @url         $URL$
- * @date        $Date: 2005-12-17 04:16:41 -0500 (Sat, 17 Dec 2005) $
+ * @date        $Date: 2005-12-17 01:16:41 -0800 (Sat, 17 Dec 2005) $
  * @version     $Revision: 8298 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/music/welcome.php mythplugins-0.19-new/mythweb/themes/default/music/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/music/welcome.php	2005-12-12 23:23:18.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/music/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the Music module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/music/welcome.php $
- * @date        $Date: 2005-12-13 02:23:18 -0500 (Tue, 13 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/music/welcome.php $
+ * @date        $Date: 2005-12-12 23:23:18 -0800 (Mon, 12 Dec 2005) $
  * @version     $Revision: 8252 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/settings/channels.php mythplugins-0.19-new/mythweb/themes/default/settings/channels.php
--- mythplugins-0.19-old/mythweb/themes/default/settings/channels.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/settings/channels.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configure MythTV Channel info
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/settings/keys.php mythplugins-0.19-new/mythweb/themes/default/settings/keys.php
--- mythplugins-0.19-old/mythweb/themes/default/settings/keys.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/settings/keys.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configure MythTV Key Bindings
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/settings/session.php mythplugins-0.19-new/mythweb/themes/default/settings/session.php
--- mythplugins-0.19-old/mythweb/themes/default/settings/session.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/settings/session.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Configure MythWeb Session info
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/settings/settings.php mythplugins-0.19-new/mythweb/themes/default/settings/settings.php
--- mythplugins-0.19-old/mythweb/themes/default/settings/settings.php	2006-02-01 23:59:05.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/settings/settings.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Main settings index
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 02:59:05 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-01 23:59:05 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8837 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/settings/welcome.php mythplugins-0.19-new/mythweb/themes/default/settings/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/settings/welcome.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/settings/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the Settings module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/settings/welcome.php $
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/settings/welcome.php $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/status/status.php mythplugins-0.19-new/mythweb/themes/default/status/status.php
--- mythplugins-0.19-old/mythweb/themes/default/status/status.php	2005-11-07 19:37:20.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/status/status.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,8 +3,8 @@
  * The display code for the main welcome page that lists the available mythweb
  * sections.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/status/status.php $
- * @date        $Date: 2005-11-07 22:37:20 -0500 (Mon, 07 Nov 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/status/status.php $
+ * @date        $Date: 2005-11-07 19:37:20 -0800 (Mon, 07 Nov 2005) $
  * @version     $Revision: 7792 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/status/welcome.php mythplugins-0.19-new/mythweb/themes/default/status/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/status/welcome.php	2005-12-12 23:23:18.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/status/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the status module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/status/welcome.php $
- * @date        $Date: 2005-12-13 02:23:18 -0500 (Tue, 13 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/status/welcome.php $
+ * @date        $Date: 2005-12-12 23:23:18 -0800 (Mon, 12 Dec 2005) $
  * @version     $Revision: 8252 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/channel.php mythplugins-0.19-new/mythweb/themes/default/tv/channel.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/channel.php	2005-12-30 15:11:40.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/channel.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Show the current lineup for a specific channel
  *
  * @url         $URL$
- * @date        $Date: 2005-12-30 18:11:40 -0500 (Fri, 30 Dec 2005) $
+ * @date        $Date: 2005-12-30 15:11:40 -0800 (Fri, 30 Dec 2005) $
  * @version     $Revision: 8431 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/detail.php mythplugins-0.19-new/mythweb/themes/default/tv/detail.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/detail.php	2006-01-29 23:16:42.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/detail.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * commands.
  *
  * @url         $URL$
- * @date        $Date: 2006-01-30 02:16:42 -0500 (Mon, 30 Jan 2006) $
+ * @date        $Date: 2006-01-29 23:16:42 -0800 (Sun, 29 Jan 2006) $
  * @version     $Revision: 8776 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/list.php mythplugins-0.19-new/mythweb/themes/default/tv/list.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/list.php	2006-02-09 22:15:17.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/list.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Print the program list
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 01:15:17 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 22:15:17 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8914 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/recorded.php mythplugins-0.19-new/mythweb/themes/default/tv/recorded.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/recorded.php	2006-02-02 00:21:54.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/recorded.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Show recorded programs.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 03:21:54 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-02 00:21:54 -0800 (Thu, 02 Feb 2006) $
  * @version     $Revision: 8838 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/schedules_custom.php mythplugins-0.19-new/mythweb/themes/default/tv/schedules_custom.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/schedules_custom.php	2006-01-29 23:22:21.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/schedules_custom.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Schedule a custom recording by manually specifying various search options
  *
  * @url         $URL$
- * @date        $Date: 2006-01-30 02:22:21 -0500 (Mon, 30 Jan 2006) $
+ * @date        $Date: 2006-01-29 23:22:21 -0800 (Sun, 29 Jan 2006) $
  * @version     $Revision: 8778 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/schedules_manual.php mythplugins-0.19-new/mythweb/themes/default/tv/schedules_manual.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/schedules_manual.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/schedules_manual.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Schedule a custom recording by manually specifying starttime and length
  *
  * @url         $URL$
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/schedules.php mythplugins-0.19-new/mythweb/themes/default/tv/schedules.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/schedules.php	2005-12-31 14:26:52.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/schedules.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * It must define one method.   documentation will be added someday.
  *
  * @url         $URL$
- * @date        $Date: 2005-12-31 17:26:52 -0500 (Sat, 31 Dec 2005) $
+ * @date        $Date: 2005-12-31 14:26:52 -0800 (Sat, 31 Dec 2005) $
  * @version     $Revision: 8456 $
  * @author      $Author: xris $
  *
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/searches.php mythplugins-0.19-new/mythweb/themes/default/tv/searches.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/searches.php	2005-12-15 19:07:00.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/searches.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Handy predefined searches
  *
  * @url         $URL$
- * @date        $Date: 2005-12-15 22:07:00 -0500 (Thu, 15 Dec 2005) $
+ * @date        $Date: 2005-12-15 19:07:00 -0800 (Thu, 15 Dec 2005) $
  * @version     $Revision: 8278 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/search.php mythplugins-0.19-new/mythweb/themes/default/tv/search.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/search.php	2006-02-09 22:15:17.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/search.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Show search results, or how to perform a detailed search.
  *
  * @url         $URL$
- * @date        $Date: 2006-02-10 01:15:17 -0500 (Fri, 10 Feb 2006) $
+ * @date        $Date: 2006-02-09 22:15:17 -0800 (Thu, 09 Feb 2006) $
  * @version     $Revision: 8914 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/upcoming.php mythplugins-0.19-new/mythweb/themes/default/tv/upcoming.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/upcoming.php	2005-12-29 11:48:49.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/upcoming.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Show all upcoming recordings.
  *
  * @url         $URL$
- * @date        $Date: 2005-12-29 14:48:49 -0500 (Thu, 29 Dec 2005) $
+ * @date        $Date: 2005-12-29 11:48:49 -0800 (Thu, 29 Dec 2005) $
  * @version     $Revision: 8421 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/tv/welcome.php mythplugins-0.19-new/mythweb/themes/default/tv/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/tv/welcome.php	2006-01-24 19:41:36.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/tv/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the TV module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/tv/welcome.php $
- * @date        $Date: 2006-01-24 22:41:36 -0500 (Tue, 24 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/tv/welcome.php $
+ * @date        $Date: 2006-01-24 19:41:36 -0800 (Tue, 24 Jan 2006) $
  * @version     $Revision: 8709 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/video/video.php mythplugins-0.19-new/mythweb/themes/default/video/video.php
--- mythplugins-0.19-old/mythweb/themes/default/video/video.php	2006-02-01 23:59:05.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/video/video.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Rudimentary interface to MythVideo data
  *
  * @url         $URL$
- * @date        $Date: 2006-02-02 02:59:05 -0500 (Thu, 02 Feb 2006) $
+ * @date        $Date: 2006-02-01 23:59:05 -0800 (Wed, 01 Feb 2006) $
  * @version     $Revision: 8837 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/video/welcome.php mythplugins-0.19-new/mythweb/themes/default/video/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/video/welcome.php	2005-12-12 23:23:18.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/video/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the Video module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/video/welcome.php $
- * @date        $Date: 2005-12-13 02:23:18 -0500 (Tue, 13 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/video/welcome.php $
+ * @date        $Date: 2005-12-12 23:23:18 -0800 (Mon, 12 Dec 2005) $
  * @version     $Revision: 8252 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/weather/weather.php mythplugins-0.19-new/mythweb/themes/default/weather/weather.php
--- mythplugins-0.19-old/mythweb/themes/default/weather/weather.php	2005-11-07 22:43:55.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/weather/weather.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,7 +3,7 @@
  * Display template for the Weather module
  *
  * @url         $URL$
- * @date        $Date: 2005-11-08 01:43:55 -0500 (Tue, 08 Nov 2005) $
+ * @date        $Date: 2005-11-07 22:43:55 -0800 (Mon, 07 Nov 2005) $
  * @version     $Revision: 7795 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/weather/welcome.php mythplugins-0.19-new/mythweb/themes/default/weather/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/weather/welcome.php	2005-12-12 23:23:18.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/weather/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * Welcome page description of the Weather module.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/weather/welcome.php $
- * @date        $Date: 2005-12-13 02:23:18 -0500 (Tue, 13 Dec 2005) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/weather/welcome.php $
+ * @date        $Date: 2005-12-12 23:23:18 -0800 (Mon, 12 Dec 2005) $
  * @version     $Revision: 8252 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/default/welcome.php mythplugins-0.19-new/mythweb/themes/default/welcome.php
--- mythplugins-0.19-old/mythweb/themes/default/welcome.php	2006-02-04 13:20:25.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/default/welcome.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,8 +3,8 @@
  * The display code for the main welcome page that lists the available mythweb
  * sections.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/default/welcome.php $
- * @date        $Date: 2006-02-04 16:20:25 -0500 (Sat, 04 Feb 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/default/welcome.php $
+ * @date        $Date: 2006-02-04 13:20:25 -0800 (Sat, 04 Feb 2006) $
  * @version     $Revision: 8863 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/wap/footer.php mythplugins-0.19-new/mythweb/themes/wap/footer.php
--- mythplugins-0.19-old/mythweb/themes/wap/footer.php	2006-01-25 18:43:22.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/wap/footer.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * This footer file is shared by all MythWeb modules.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/wap/footer.php $
- * @date        $Date: 2006-01-25 21:43:22 -0500 (Wed, 25 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/wap/footer.php $
+ * @date        $Date: 2006-01-25 18:43:22 -0800 (Wed, 25 Jan 2006) $
  * @version     $Revision: 8726 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/wap/header.php mythplugins-0.19-new/mythweb/themes/wap/header.php
--- mythplugins-0.19-old/mythweb/themes/wap/header.php	2006-01-25 18:43:22.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/wap/header.php	2006-02-19 19:11:45.000000000 -0800
@@ -2,8 +2,8 @@
 /**
  * This header file is shared by all MythWeb modules.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/wap/header.php $
- * @date        $Date: 2006-01-25 21:43:22 -0500 (Wed, 25 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/wap/header.php $
+ * @date        $Date: 2006-01-25 18:43:22 -0800 (Wed, 25 Jan 2006) $
  * @version     $Revision: 8726 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/wap/status/status.php mythplugins-0.19-new/mythweb/themes/wap/status/status.php
--- mythplugins-0.19-old/mythweb/themes/wap/status/status.php	2006-01-25 18:43:22.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/wap/status/status.php	2006-02-19 19:11:45.000000000 -0800
@@ -3,8 +3,8 @@
  * The display code for the main welcome page that lists the available mythweb
  * sections.
  *
- * @url         $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/tags/release-0-19/mythplugins/mythweb/themes/wap/status/status.php $
- * @date        $Date: 2006-01-25 21:43:22 -0500 (Wed, 25 Jan 2006) $
+ * @url         $URL: http://svn.mythtv.org/svn/branches/release-0-19-fixes/mythplugins/mythweb/themes/wap/status/status.php $
+ * @date        $Date: 2006-01-25 18:43:22 -0800 (Wed, 25 Jan 2006) $
  * @version     $Revision: 8726 $
  * @author      $Author: xris $
  * @license     GPL
diff -Naur mythplugins-0.19-old/mythweb/themes/wap/tv/upcoming.php mythplugins-0.19-new/mythweb/themes/wap/tv/upcoming.php
--- mythplugins-0.19-old/mythweb/themes/wap/tv/upcoming.php	2006-01-25 18:43:22.000000000 -0800
+++ mythplugins-0.19-new/mythweb/themes/wap/tv/upcoming.php	2006-02-19 19:11:45.000000000 -0800
@@ -4,7 +4,7 @@
  * It must define one method.   documentation will be added someday.
  *
  * @url         $URL$
- * @date        $Date: 2006-01-25 21:43:22 -0500 (Wed, 25 Jan 2006) $
+ * @date        $Date: 2006-01-25 18:43:22 -0800 (Wed, 25 Jan 2006) $
  * @version     $Revision: 8726 $
  * @author      $Author: xris $
  * @license     GPL
