diff -Naur platform-1.0.10-old/CMakeLists.txt platform-1.0.10-new/CMakeLists.txt
--- platform-1.0.10-old/CMakeLists.txt	2015-06-30 01:54:58.000000000 -0700
+++ platform-1.0.10-new/CMakeLists.txt	2015-07-30 13:05:23.668006629 -0700
@@ -78,7 +78,7 @@
   configure_pc_file(platform platform.pc.in
                                  platform.pc
                                  ${CMAKE_INSTALL_PREFIX}
-                                 ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}
+                                 ${CMAKE_INSTALL_LIBDIR}
                                  ${CMAKE_INSTALL_PREFIX}/include)
 
   install(FILES ${CMAKE_BINARY_DIR}/platform.pc
