diff -Naur linux-2.6.31.12-old/drivers/net/via-rhine.c linux-2.6.31.12-new/drivers/net/via-rhine.c
--- linux-2.6.31.12-old/drivers/net/via-rhine.c	2010-01-18 10:30:45.000000000 -0800
+++ linux-2.6.31.12-new/drivers/net/via-rhine.c	2010-02-23 21:17:45.000000000 -0800
@@ -127,7 +127,7 @@
 module_param(max_interrupt_work, int, 0);
 module_param(debug, int, 0);
 module_param(rx_copybreak, int, 0);
-module_param(avoid_D3, bool, 0);
+module_param(avoid_D3, bool, 0644);
 MODULE_PARM_DESC(max_interrupt_work, "VIA Rhine maximum events handled per interrupt");
 MODULE_PARM_DESC(debug, "VIA Rhine debug level (0-7)");
 MODULE_PARM_DESC(rx_copybreak, "VIA Rhine copy breakpoint for copy-only-tiny-frames");
