diff -Naur utelnetd-0.1.9-old/Makefile utelnetd-0.1.9-new/Makefile
--- utelnetd-0.1.9-old/Makefile	2003-08-08 00:25:18.000000000 -0700
+++ utelnetd-0.1.9-new/Makefile	2005-01-06 21:55:35.750835614 -0800
@@ -27,8 +27,6 @@
 PROGS     = utelnetd
 
 INSTMODE  = 0755
-INSTOWNER = root
-INSTGROUP = root
 
 OBJS      = utelnetd.o
 
@@ -59,7 +57,7 @@
 .PHONY: install
 install: $(PROGS)
 	$(INSTALL) -d $(INSTDIR)
-	$(INSTALL) -m $(INSTMODE) -o $(INSTOWNER) -g $(INSTGROUP) $(PROGS) $(INSTDIR)
+	$(INSTALL) -m $(INSTMODE) $(PROGS) $(INSTDIR)
 
 .PHONY: clean
 clean:
