diff -Naur -x mytharchive mythtv-0.20-old/bindings/perl/MythTV/Channel.pm mythtv-0.20-new/bindings/perl/MythTV/Channel.pm
--- mythtv-0.20-old/bindings/perl/MythTV/Channel.pm	2006-04-16 01:05:27.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV/Channel.pm	2006-10-28 19:17:11.000000000 -0700
@@ -3,8 +3,8 @@
 #
 # Object containing info about a particular MythTV channel.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV/Channel.pm $
-# @date      $Date: 2006-04-16 04:05:27 -0400 (Sun, 16 Apr 2006) $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV/Channel.pm $
+# @date      $Date: 2006-04-16 01:05:27 -0700 (Sun, 16 Apr 2006) $
 # @version   $Revision: 9725 $
 # @author    $Author: xris $
 #
diff -Naur -x mytharchive mythtv-0.20-old/bindings/perl/MythTV/Program.pm mythtv-0.20-new/bindings/perl/MythTV/Program.pm
--- mythtv-0.20-old/bindings/perl/MythTV/Program.pm	2006-04-16 01:05:27.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV/Program.pm	2006-10-28 19:17:11.000000000 -0700
@@ -3,9 +3,9 @@
 #
 # Object containing info about a particular MythTV program.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV/Program.pm $
-# @date      $Date: 2006-04-16 04:05:27 -0400 (Sun, 16 Apr 2006) $
-# @version   $Revision: 9725 $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV/Program.pm $
+# @date      $Date: 2006-10-04 22:46:03 -0700 (Wed, 04 Oct 2006) $
+# @version   $Revision: 11444 $
 # @author    $Author: xris $
 #
 
@@ -84,6 +84,7 @@
         $self->{'hasairdate'}      = $rows->[38]; #
         $self->{'timestretch'}     = $rows->[39]; #
         $self->{'recpriority2'}    = $rows->[40]; #
+        $self->{'parentid'}        = $rows->[41]; #
 
     # Load the channel data
         if ($self->{'chanid'}) {
diff -Naur -x mytharchive mythtv-0.20-old/bindings/perl/MythTV/Recording.pm mythtv-0.20-new/bindings/perl/MythTV/Recording.pm
--- mythtv-0.20-old/bindings/perl/MythTV/Recording.pm	2006-07-19 19:20:17.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV/Recording.pm	2006-10-28 19:17:11.000000000 -0700
@@ -3,9 +3,9 @@
 #
 # Object containing info about a particular MythTV recording.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV/Recording.pm $
-# @date      $Date: 2006-07-19 22:20:17 -0400 (Wed, 19 Jul 2006) $
-# @version   $Revision: 10602 $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV/Recording.pm $
+# @date      $Date: 2006-10-07 23:48:26 -0700 (Sat, 07 Oct 2006) $
+# @version   $Revision: 11472 $
 # @author    $Author: xris $
 #
 
@@ -50,6 +50,10 @@
                 $self->{'file_port'} = ($2 or 6543);
                 $self->{'basename'}  = $3;
             }
+            else {
+                $self->{'basename'}  = $self->{'filename'};
+                $self->{'basename'} =~ s/^$self->{'video_dir'}\/+//sg;
+            }
         }
 
     # Pull the cutlist info from the database
@@ -231,6 +235,10 @@
     # Make sure some things are actually numbers
         $info{'width'}  += 0;
         $info{'height'} += 0;
+    # HD fix
+        if ($info{'height'} == 1080) {
+            $info{'height'} = 1088;
+        }
     # Make some corrections for myth bugs
         $info{'audio_sample_rate'} = 44100 if ($info{'audio_sample_rate'} == 42501 || $info{'audio_sample_rate'} =~ /^44\d\d\d$/);
         $info{'aspect'} = '4:3';
@@ -297,6 +305,19 @@
                ."    $program -v -v -v -v -nolirc -nojoystick -vo null -ao null \\\n"
                ."             -frames 0 -identify '$file'\n\n";
         }
+    # HD fix
+        if ($info{'height'} == 1080) {
+            $info{'height'} = 1088;
+        }
+    # mplayer is confused and we need to detect the aspect on our own
+        if ($info{'aspect'} =~ /^0.0/) {
+            if ($info{'height'} == 1088 || $info{'height'} == 720) {
+                $info{'aspect'} = 16 / 9;
+            }
+            else {
+                $info{'aspect'} = 4 / 3;
+            }
+        }
     # Cleanup
         $info{'aspect'}   = aspect_str($info{'aspect'});
         $info{'aspect_f'} = aspect_float($info{'aspect'});
@@ -330,7 +351,7 @@
             return $w / $h;
         }
     # Parse out decimal formats
-        if ($aspect == 1)          { return  1;     }
+        if ($aspect eq '1')        { return  1;     }
         elsif ($aspect =~ m/^1.3/) { return  4 / 3; }
         elsif ($aspect =~ m/^1.7/) { return 16 / 9; }
     # Unknown aspect
diff -Naur -x mytharchive mythtv-0.20-old/bindings/perl/MythTV.pm mythtv-0.20-new/bindings/perl/MythTV.pm
--- mythtv-0.20-old/bindings/perl/MythTV.pm	2006-07-04 13:30:37.000000000 -0700
+++ mythtv-0.20-new/bindings/perl/MythTV.pm	2006-10-28 19:17:11.000000000 -0700
@@ -1,14 +1,14 @@
 #
 # MythTV bindings for perl.
 #
-# @url       $URL: svn+ssh://ijr@cvs.mythtv.org/var/lib/svn/trunk/mythtv/bindings/perl/MythTV.pm $
-# @date      $Date: 2006-07-04 16:30:37 -0400 (Tue, 04 Jul 2006) $
-# @version   $Revision: 10390 $
+# @url       $URL: http://svn.mythtv.org/svn/branches/release-0-20-fixes/mythtv/bindings/perl/MythTV.pm $
+# @date      $Date: 2006-10-04 22:46:03 -0700 (Wed, 04 Oct 2006) $
+# @version   $Revision: 11444 $
 # @author    $Author: xris $
 #
 
 # Version
-    $VERSION = '.20svn';
+    $VERSION = '.20';
 
 # Load sub libraries
     use MythTV::Program;
@@ -48,12 +48,12 @@
 
 # MYTH_PROTO_VERSION is defined in libmyth in mythtv/libs/libmyth/mythcontext.h
 # and should be the current MythTV protocol version.
-    our $PROTO_VERSION = 29;
+    our $PROTO_VERSION = 31;
 
 # NUMPROGRAMLINES is defined in mythtv/libs/libmythtv/programinfo.h and is
 # the number of items in a ProgramInfo QStringList group used by
 # ProgramInfo::ToSringList and ProgramInfo::FromStringList.
-    our $NUMPROGRAMLINES = 41;
+    our $NUMPROGRAMLINES = 42;
 
 # Reasons a recording wouldn't be happening (from libs/libmythtv/programinfo.h)
     our %RecStatus_Types = (
diff -Naur -x mytharchive mythtv-0.20-old/configfiles/lircrc.native.example.mceusb2 mythtv-0.20-new/configfiles/lircrc.native.example.mceusb2
--- mythtv-0.20-old/configfiles/lircrc.native.example.mceusb2	2006-06-11 22:18:38.000000000 -0700
+++ mythtv-0.20-new/configfiles/lircrc.native.example.mceusb2	2006-10-28 19:16:42.000000000 -0700
@@ -2,7 +2,7 @@
 # lircrc config file for the Microsoft Media Center Edition Remote, model 1039
 #
 # @url       $URL$
-# @date      $Date: 2006-06-12 01:18:38 -0400 (Mon, 12 Jun 2006) $
+# @date      $Date: 2006-06-11 22:18:38 -0700 (Sun, 11 Jun 2006) $
 # @version   $Revision: 10187 $
 # @author    $Author: xris $
 #
diff -Naur -x mytharchive mythtv-0.20-old/contrib/myth_archive_job.pl mythtv-0.20-new/contrib/myth_archive_job.pl
--- mythtv-0.20-old/contrib/myth_archive_job.pl	2005-11-29 08:28:53.000000000 -0800
+++ mythtv-0.20-new/contrib/myth_archive_job.pl	2006-10-28 19:17:13.000000000 -0700
@@ -72,7 +72,7 @@
 			printf( "    Curr Free: %6d MB\n", $freeSpace );
 		}
 
-		if (( $freeSpace - ($size / 1024.0 / 1024.0)) > $keepFree ) {
+		if (( $freeSpace - $size) > $keepFree ) {
 			printf( "Attempting archive to %s\nStatus: ", $archiveDir );
 			if (MoveFileToArchiveDir( $file, $directory, $archiveDir )) {
 				printf( "Success.\n" );
@@ -97,7 +97,7 @@
 	if ( ! -r $dir ) {
 		return 0;
 	} else {
-		my( $freeSpace ) = `df -m $dir | grep -v Available | awk '{print \$4}'`;
+		my( $freeSpace ) = `df -Pm $dir | grep -v Available | awk '{print \$4}'`;
 
 		return $freeSpace;
 	}
diff -Naur -x mytharchive mythtv-0.20-old/contrib/mythrename.pl mythtv-0.20-new/contrib/mythrename.pl
--- mythtv-0.20-old/contrib/mythrename.pl	2006-02-27 23:40:08.000000000 -0800
+++ mythtv-0.20-new/contrib/mythrename.pl	2006-10-28 19:17:13.000000000 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/perl -w
 #
-# $Date: 2006-02-28 02:40:08 -0500 (Tue, 28 Feb 2006) $
+# $Date: 2006-02-27 23:40:08 -0800 (Mon, 27 Feb 2006) $
 # $Revision: 9198 $
 # $Author: xris $
 #
diff -Naur -x mytharchive mythtv-0.20-old/contrib/optimize_mythdb.pl mythtv-0.20-new/contrib/optimize_mythdb.pl
--- mythtv-0.20-old/contrib/optimize_mythdb.pl	2006-08-05 14:51:46.000000000 -0700
+++ mythtv-0.20-new/contrib/optimize_mythdb.pl	2006-10-28 19:17:13.000000000 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/perl -w
 #
-# $Date: 2006-08-05 17:51:46 -0400 (Sat, 05 Aug 2006) $
+# $Date: 2006-08-05 14:51:46 -0700 (Sat, 05 Aug 2006) $
 # $Revision: 10685 $
 # $Author: xris $
 #
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmyth/mythcontext.h mythtv-0.20-new/libs/libmyth/mythcontext.h
--- mythtv-0.20-old/libs/libmyth/mythcontext.h	2006-08-28 13:47:26.000000000 -0700
+++ mythtv-0.20-new/libs/libmyth/mythcontext.h	2006-10-28 19:16:58.000000000 -0700
@@ -229,7 +229,7 @@
  *   You must also update this value in
  *   mythplugins/mythweb/includes/mythbackend.php
  */
-#define MYTH_PROTO_VERSION "30"
+#define MYTH_PROTO_VERSION "31"
 
 /** \class MythContext
  *  \brief This class contains the runtime context for MythTV.
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythdvdnav/vm.c mythtv-0.20-new/libs/libmythdvdnav/vm.c
--- mythtv-0.20-old/libs/libmythdvdnav/vm.c	2006-03-24 05:17:02.000000000 -0800
+++ mythtv-0.20-new/libs/libmythdvdnav/vm.c	2006-10-28 19:17:08.000000000 -0700
@@ -757,11 +757,11 @@
     }
   }
 
-  if((vm->state).domain == VTS_DOMAIN && !((vm->state).SPST_REG & 0x40))
+  if((vm->state).SPST_REG & 0x40)
     /* Bit 7 set means hide, and only let Forced display show */
-    return (streamN | 0x80);
+    return (streamN & 0x1F);
   else
-    return streamN;
+    return -1;
 }
 
 void vm_get_angle_info(vm_t *vm, int *current, int *num_avail) {
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.cpp mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.cpp	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -12,7 +12,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: AAFilter.cpp 9996 2006-05-23 17:40:36Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.h mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.h
--- mythtv-0.20-old/libs/libmythsoundtouch/AAFilter.h	2004-11-13 14:29:45.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/AAFilter.h	2006-10-28 19:16:53.000000000 -0700
@@ -13,7 +13,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2004-11-13 17:29:45 -0500 (Sat, 13 Nov 2004) $
+// Last changed  : $Date: 2004-11-13 14:29:45 -0800 (Sat, 13 Nov 2004) $
 // File revision : $Revision: 4714 $
 //
 // $Id: AAFilter.h 4714 2004-11-13 22:29:45Z ijr $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/BPMDetect.h mythtv-0.20-new/libs/libmythsoundtouch/BPMDetect.h
--- mythtv-0.20-old/libs/libmythsoundtouch/BPMDetect.h	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/BPMDetect.h	2006-10-28 19:16:53.000000000 -0700
@@ -26,7 +26,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: BPMDetect.h 9996 2006-05-23 17:40:36Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect.h mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect.h
--- mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect.h	2006-03-16 02:00:28.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect.h	2006-10-28 19:16:53.000000000 -0700
@@ -12,7 +12,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-03-16 05:00:28 -0500 (Thu, 16 Mar 2006) $
+// Last changed  : $Date: 2006-03-16 02:00:28 -0800 (Thu, 16 Mar 2006) $
 // File revision : $Revision: 9376 $
 //
 // $Id: cpu_detect.h 9376 2006-03-16 10:00:28Z ijr $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/cpu_detect_x86_gcc.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -12,7 +12,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: cpu_detect_x86_gcc.cpp 9996 2006-05-23 17:40:36Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.cpp mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.cpp	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -15,7 +15,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: FIFOSampleBuffer.cpp 9996 2006-05-23 17:40:36Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.h mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.h
--- mythtv-0.20-old/libs/libmythsoundtouch/FIFOSampleBuffer.h	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIFOSampleBuffer.h	2006-10-28 19:16:53.000000000 -0700
@@ -15,7 +15,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: FIFOSampleBuffer.h 9996 2006-05-23 17:40:36Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/FIFOSamplePipe.h mythtv-0.20-new/libs/libmythsoundtouch/FIFOSamplePipe.h
--- mythtv-0.20-old/libs/libmythsoundtouch/FIFOSamplePipe.h	2005-05-23 13:39:12.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIFOSamplePipe.h	2006-10-28 19:16:53.000000000 -0700
@@ -17,7 +17,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2005-05-23 16:39:12 -0400 (Mon, 23 May 2005) $
+// Last changed  : $Date: 2005-05-23 13:39:12 -0700 (Mon, 23 May 2005) $
 // File revision : $Revision: 6468 $
 //
 // $Id: FIFOSamplePipe.h 6468 2005-05-23 20:39:12Z ijr $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.cpp mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.cpp	2006-03-31 22:19:12.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -11,7 +11,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-04-01 01:19:12 -0500 (Sat, 01 Apr 2006) $
+// Last changed  : $Date: 2006-03-31 22:19:12 -0800 (Fri, 31 Mar 2006) $
 // File revision : $Revision: 9588 $
 //
 // $Id: FIRFilter.cpp 9588 2006-04-01 06:19:12Z ijr $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.h mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.h
--- mythtv-0.20-old/libs/libmythsoundtouch/FIRFilter.h	2004-11-13 14:29:45.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/FIRFilter.h	2006-10-28 19:16:53.000000000 -0700
@@ -11,7 +11,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2004-11-13 17:29:45 -0500 (Sat, 13 Nov 2004) $
+// Last changed  : $Date: 2004-11-13 14:29:45 -0800 (Sat, 13 Nov 2004) $
 // File revision : $Revision: 4714 $
 //
 // $Id: FIRFilter.h 4714 2004-11-13 22:29:45Z ijr $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/mmx_gcc.cpp mythtv-0.20-new/libs/libmythsoundtouch/mmx_gcc.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/mmx_gcc.cpp	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/mmx_gcc.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -15,7 +15,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: mmx_gcc.cpp 9998 2006-05-23 19:14:24Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.cpp mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.cpp	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -10,7 +10,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: RateTransposer.cpp 9998 2006-05-23 19:14:24Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.h mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.h
--- mythtv-0.20-old/libs/libmythsoundtouch/RateTransposer.h	2004-11-13 14:29:45.000000000 -0800
+++ mythtv-0.20-new/libs/libmythsoundtouch/RateTransposer.h	2006-10-28 19:16:53.000000000 -0700
@@ -14,7 +14,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2004-11-13 17:29:45 -0500 (Sat, 13 Nov 2004) $
+// Last changed  : $Date: 2004-11-13 14:29:45 -0800 (Sat, 13 Nov 2004) $
 // File revision : $Revision: 4714 $
 //
 // $Id: RateTransposer.h 4714 2004-11-13 22:29:45Z ijr $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.cpp mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.cpp	2006-05-24 03:48:15.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -41,7 +41,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-24 06:48:15 -0400 (Wed, 24 May 2006) $
+// Last changed  : $Date: 2006-05-24 03:48:15 -0700 (Wed, 24 May 2006) $
 // File revision : $Revision: 10003 $
 //
 // $Id: SoundTouch.cpp 10003 2006-05-24 10:48:15Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.h mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.h
--- mythtv-0.20-old/libs/libmythsoundtouch/SoundTouch.h	2006-05-23 10:40:36.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/SoundTouch.h	2006-10-28 19:16:53.000000000 -0700
@@ -41,7 +41,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 13:40:36 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 10:40:36 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9996 $
 //
 // $Id: SoundTouch.h 9996 2006-05-23 17:40:36Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/STTypes.h mythtv-0.20-new/libs/libmythsoundtouch/STTypes.h
--- mythtv-0.20-old/libs/libmythsoundtouch/STTypes.h	2006-05-24 07:56:37.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/STTypes.h	2006-10-28 19:16:53.000000000 -0700
@@ -8,7 +8,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-24 10:56:37 -0400 (Wed, 24 May 2006) $
+// Last changed  : $Date: 2006-05-24 07:56:37 -0700 (Wed, 24 May 2006) $
 // File revision : $Revision: 10005 $
 //
 // $Id: STTypes.h 10005 2006-05-24 14:56:37Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.cpp mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.cpp
--- mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.cpp	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -13,7 +13,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: TDStretch.cpp 9998 2006-05-23 19:14:24Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.h mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.h
--- mythtv-0.20-old/libs/libmythsoundtouch/TDStretch.h	2006-05-23 12:14:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythsoundtouch/TDStretch.h	2006-10-28 19:16:53.000000000 -0700
@@ -13,7 +13,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////
 //
-// Last changed  : $Date: 2006-05-23 15:14:24 -0400 (Tue, 23 May 2006) $
+// Last changed  : $Date: 2006-05-23 12:14:24 -0700 (Tue, 23 May 2006) $
 // File revision : $Revision: 9998 $
 //
 // $Id: TDStretch.h 9998 2006-05-23 19:14:24Z danielk $
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/avformatdecoder.cpp mythtv-0.20-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.20-old/libs/libmythtv/avformatdecoder.cpp	2006-09-03 20:50:52.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/avformatdecoder.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -464,9 +464,13 @@
 void AvFormatDecoder::SeekReset(long long newKey, uint skipFrames,
                                 bool doflush, bool discardFrames)
 {
-    if (ringBuffer->InDVDMenuOrStillFrame())
-        return;
-
+    if (ringBuffer->isDVD())
+    {
+        if (ringBuffer->InDVDMenuOrStillFrame() ||
+            newKey == 0) 
+            return;
+    }
+            
     VERBOSE(VB_PLAYBACK, LOC +
             QString("SeekReset(%1, %2, %3 flush, %4 discard)")
             .arg(newKey).arg(skipFrames)
@@ -931,23 +935,28 @@
     return fps;
 }
 
-void AvFormatDecoder::InitVideoCodec(AVStream *stream, AVCodecContext *enc)
+void AvFormatDecoder::InitVideoCodec(AVStream *stream, AVCodecContext *enc,
+                                     bool selectedStream)
 {
-    fps = normalized_fps(stream, enc);
-
     float aspect_ratio;
-    if (enc->sample_aspect_ratio.num == 0)
-        aspect_ratio = 0.0f;
-    else
-        aspect_ratio = av_q2d(enc->sample_aspect_ratio) *
-            enc->width / enc->height;
 
-    if (aspect_ratio <= 0.0f || aspect_ratio > 6.0f)
-        aspect_ratio = (float)enc->width / (float)enc->height;
+    if (selectedStream)
+    {
+        fps = normalized_fps(stream, enc);
+
+        if (enc->sample_aspect_ratio.num == 0)
+            aspect_ratio = 0.0f;
+        else
+            aspect_ratio = av_q2d(enc->sample_aspect_ratio) *
+                enc->width / enc->height;
+
+        if (aspect_ratio <= 0.0f || aspect_ratio > 6.0f)
+            aspect_ratio = (float)enc->width / (float)enc->height;
 
-    current_width = enc->width;
-    current_height = enc->height;
-    current_aspect = aspect_ratio;
+        current_width = enc->width;
+        current_height = enc->height;
+        current_aspect = aspect_ratio;
+    }
 
     enc->opaque = (void *)this;
     enc->get_buffer = avcodec_default_get_buffer;
@@ -965,7 +974,8 @@
 
     AVCodec *codec = avcodec_find_decoder(enc->codec_id);    
 
-    if (!gContext->GetNumSetting("DecodeExtraAudio", 0) &&
+    if (selectedStream &&
+        !gContext->GetNumSetting("DecodeExtraAudio", 0) &&
         codec->id != CODEC_ID_MPEG2VIDEO_XVMC           &&
         codec->id != CODEC_ID_MPEG2VIDEO_XVMC_VLD)
     {
@@ -981,7 +991,8 @@
         enc->draw_horiz_band = render_slice_xvmc;
         enc->slice_flags = SLICE_FLAG_CODED_ORDER |
             SLICE_FLAG_ALLOW_FIELD;
-        directrendering = true;
+        if (selectedStream)
+            directrendering = true;
     }
     else if (codec && codec->capabilities & CODEC_CAP_DR1 &&
              !(enc->width % 16))
@@ -990,26 +1001,30 @@
         enc->get_buffer = get_avf_buffer;
         enc->release_buffer = release_avf_buffer;
         enc->draw_horiz_band = NULL;
-        directrendering = true;
+        if (selectedStream)
+            directrendering = true;
     }
 
-    uint align_width  = enc->width;
-    uint align_height = enc->height;
+    if (selectedStream)
+    {
+        uint align_width  = enc->width;
+        uint align_height = enc->height;
 
-    align_dimensions(enc, align_width, align_height);
+        align_dimensions(enc, align_width, align_height);
 
-    if (align_width == 0 && align_height == 0)
-    {
-        VERBOSE(VB_PLAYBACK, LOC + "InitVideoCodec "
-                "failed to align dimensions, resetting decoder.");
-        align_width = 640;
-        align_height = 480;
-        fps = 29.97;
-        aspect_ratio = 4.0 / 3;
-    }
+        if (align_width == 0 && align_height == 0)
+        {
+            VERBOSE(VB_PLAYBACK, LOC + "InitVideoCodec "
+                    "failed to align dimensions, resetting decoder.");
+            align_width = 640;
+            align_height = 480;
+            fps = 29.97;
+            aspect_ratio = 4.0 / 3;
+        }
 
-    GetNVP()->SetVideoParams(align_width, align_height, fps,
-                             keyframedist, aspect_ratio, kScan_Detect);
+        GetNVP()->SetVideoParams(align_width, align_height, fps,
+                                 keyframedist, aspect_ratio, kScan_Detect);
+    }
 }
 
 #ifdef USING_XVMC
@@ -1325,7 +1340,7 @@
                             <<") type ("<<codec_type_string(enc->codec_type)
                             <<") already open.");
                 }
-                InitVideoCodec(ic->streams[i], enc);
+
                 // Only use libmpeg2 when not using XvMC
                 if (CODEC_ID_MPEG1VIDEO == enc->codec_id ||
                     CODEC_ID_MPEG2VIDEO == enc->codec_id)
@@ -1336,14 +1351,14 @@
                 enc->decode_cc_dvd  = decode_cc_dvd;
 
                 // Set the default stream to the stream
-                // with the lowest component tag.
-                if (selectedVideoIndex < 0 ||
-                    ic->streams[i]->component_tag <
-                    ic->streams[selectedVideoIndex]->component_tag)
+                // that is found first in the PMT
+                if (selectedVideoIndex < 0)
                 {
                     selectedVideoIndex = i;
                 }
 
+                InitVideoCodec(ic->streams[i], enc, selectedVideoIndex == i);
+
                 ScanATSCCaptionStreams(i);
 
                 break;
@@ -1497,20 +1512,22 @@
                 trackNo = GetTrackCount(kTrackTypeAudio) - 1;
             SetTrack(kTrackTypeAudio, trackNo);
         }
-        if (tracks[kTrackTypeSubtitle].size() > 1)
+        if (tracks[kTrackTypeSubtitle].size() > 0)
         {
             qBubbleSort(tracks[kTrackTypeSubtitle]);
             int trackNo = ringBuffer->DVD()->GetTrack(kTrackTypeSubtitle);
             uint captionmode = GetNVP()->GetCaptionMode();
-            if (captionmode == kDisplayAVSubtitle) {
-                if (trackNo < 0 || trackNo >= (int)GetTrackCount(kTrackTypeSubtitle))
-                {
-                    GetNVP()->SetCaptionsEnabled(false, false);
-                }
-                else
+            if (captionmode == kDisplayAVSubtitle &&
+                (trackNo < 0 || trackNo >= (int)GetTrackCount(kTrackTypeSubtitle)))
+            {
+                GetNVP()->SetCaptionsEnabled(false, false);
+            }
+            else
+            {
+                if (!ringBuffer->InDVDMenuOrStillFrame() && trackNo >= 0)
                 {
-                    if (!ringBuffer->InDVDMenuOrStillFrame() && trackNo >= 0)
-                        GetNVP()->SetCaptionsEnabled(true, false);
+                    SetTrack(kTrackTypeSubtitle, trackNo);
+                    GetNVP()->SetCaptionsEnabled(true, false);
                 }
             }
         }
@@ -3406,3 +3423,5 @@
 
     return fsize;
 }
+
+/* vim: set expandtab tabstop=4 shiftwidth=4: */
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/avformatdecoder.h mythtv-0.20-new/libs/libmythtv/avformatdecoder.h
--- mythtv-0.20-old/libs/libmythtv/avformatdecoder.h	2006-08-23 12:13:05.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/avformatdecoder.h	2006-10-28 19:16:52.000000000 -0700
@@ -168,7 +168,8 @@
 
     void DecodeDTVCC(const uint8_t *buf);
     void InitByteContext(void);
-    void InitVideoCodec(AVStream *stream, AVCodecContext *enc);
+    void InitVideoCodec(AVStream *stream, AVCodecContext *enc,
+                        bool selectedStream = false);
 
     /// Preprocess a packet, setting the video parms if nessesary.
     void MpegPreProcessPkt(AVStream *stream, AVPacket *pkt);
@@ -256,3 +257,5 @@
 };
 
 #endif
+
+/* vim: set expandtab tabstop=4 shiftwidth=4: */
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/channeleditor.cpp mythtv-0.20-new/libs/libmythtv/channeleditor.cpp
--- mythtv-0.20-old/libs/libmythtv/channeleditor.cpp	2006-04-07 12:14:30.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/channeleditor.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -303,17 +303,46 @@
         return;
 
     MSqlQuery query(MSqlQuery::InitCon());
-    if ((currentSourceID == "") || (currentSourceID == "Unassigned"))
+    if (currentSourceID.isEmpty())
     {
         query.prepare("TRUNCATE TABLE channel");
     }
+    else if (currentSourceID == "Unassigned")
+    {
+        query.prepare("SELECT sourceid "
+                      "FROM videosource "
+                      "GROUP BY sourceid");
+
+        if (!query.exec() || !query.isActive())
+        {
+            MythContext::DBError("ChannelEditor Delete Channels", query);
+            return;
+        }
+
+        QString tmp = "";
+        while (query.next())
+            tmp += "'" + query.value(0).toString() + "',";
+
+        if (tmp.isEmpty())
+        {
+            query.prepare("TRUNCATE TABLE channel");
+        }
+        else
+        {
+            tmp = tmp.left(tmp.length() - 1);
+            query.prepare(QString("DELETE FROM channel "
+                                  "WHERE sourceid NOT IN (%1)").arg(tmp));
+        }
+    }
     else
     {
         query.prepare("DELETE FROM channel "
                       "WHERE sourceid = :SOURCEID");
         query.bindValue(":SOURCEID", currentSourceID);
     }
-    query.exec();
+
+    if (!query.exec())
+        MythContext::DBError("ChannelEditor Delete Channels", query);
 
     list->fillSelections();
 }
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/channelutil.cpp mythtv-0.20-new/libs/libmythtv/channelutil.cpp
--- mythtv-0.20-old/libs/libmythtv/channelutil.cpp	2006-08-23 17:52:42.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/channelutil.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -1435,64 +1435,43 @@
     if (it == sorted.end())
         return 0; // no channels..
 
+    DBChanList::const_iterator start = it;
     bool skip_non_visible = true; // TODO make DB selectable
-    if (skip_non_visible)
+
+    if (CHANNEL_DIRECTION_DOWN == direction)
     {
-        DBChanList::const_iterator start = it;
-        if (CHANNEL_DIRECTION_DOWN == direction)
-        {
-            do
-            {
-                if (it == sorted.begin())
-                {
-                    it = find(sorted.begin(), sorted.end(),
-                              sorted.rbegin()->chanid);
-                }
-                else
-                {
-                    it--;
-                }
-            } while ((it != start) && !it->visible);
-        }
-        else if (CHANNEL_DIRECTION_UP == direction)
+        do
         {
-            do
-            {
-                it++;
-                it = (it == sorted.end()) ? sorted.begin() : it;
-            } while ((it != start) && !it->visible);
-        }
-    }
-    else if (CHANNEL_DIRECTION_DOWN == direction)
-    {
-        if (it == sorted.begin())
-            return sorted.rbegin()->chanid;
-        it--;
+            if (it == sorted.begin())
+                it = find(sorted.begin(), sorted.end(),
+                          sorted.rbegin()->chanid);
+            else
+                it--;
+        } while ((it != start) && skip_non_visible && !it->visible);
+
     }
     else if (CHANNEL_DIRECTION_UP == direction)
     {
-        it++;
-        it = (it == sorted.end()) ? sorted.begin() : it;
+        do
+        {
+            it++;
+            if (it == sorted.end())
+                it = sorted.begin();
+        } while ((it != start) && skip_non_visible && !it->visible);
     }
     else if (CHANNEL_DIRECTION_FAVORITE == direction)
     {
-        DBChanList::const_iterator it_orig = it;
-        for (;;++it)
+        do
         {
-            it = (it == sorted.end()) ? sorted.begin() : it;
+            it++;
+            if (it == sorted.end())
+                it = sorted.begin();
 
-            if (it == it_orig)
-            {
-                if (!it->favorite)
-                    ++it;
-                it = (it == sorted.end()) ? sorted.begin() : it;
-                break; // no (other?) favorites
-            }
-
-            if (it->favorite)
-                break; // found next favorite
-        }
+        } while ((it != start) &&
+                 (!it->favorite || (skip_non_visible && !it->visible)));
     }
 
     return it->chanid;
 }
+
+/* vim: set expandtab tabstop=4 shiftwidth=4: */
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/dbcheck.cpp mythtv-0.20-new/libs/libmythtv/dbcheck.cpp
--- mythtv-0.20-old/libs/libmythtv/dbcheck.cpp	2006-09-10 21:35:19.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/dbcheck.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -10,7 +10,7 @@
 #include "mythdbcon.h"
 
 /// This is the DB schema version expected by the running MythTV instance.
-const QString currentDatabaseVersion = "1158";
+const QString currentDatabaseVersion = "1160";
 
 static bool UpdateDBVersionNumber(const QString &newnumber);
 static bool performActualUpdate(const QString updates[], QString version,
@@ -2510,6 +2510,41 @@
             return false;
     }
 
+    if (dbver == "1158")
+    {
+        const QString updates[] = {
+"ALTER TABLE recorded ADD COLUMN watched TINYINT NOT NULL DEFAULT '0';",
+""
+};
+
+        if (!performActualUpdate(updates, "1159", dbver))
+            return false;
+    }
+
+    if (dbver == "1159")
+    {
+        MSqlQuery query(MSqlQuery::InitCon());
+        query.prepare("SELECT DISTINCT chanid, starttime FROM recordedmarkup "
+                      "WHERE type = 1;");
+        if (query.exec() && query.isActive() && query.size() > 0)
+        {
+            MSqlQuery fixup(MSqlQuery::InitCon());
+            while (query.next())
+            {
+                fixup.prepare(
+                       "UPDATE recorded SET cutlist = 1 "
+                       "WHERE chanid = :CHANID AND starttime =  :STARTTIME;");
+                fixup.bindValue(":CHANID", query.value(0).toString());
+                fixup.bindValue(":STARTTIME", query.value(1).toDateTime());
+
+                fixup.exec();
+            }
+        }
+
+        const QString updates[] = { "" };
+        if (!performActualUpdate(updates, "1160", dbver))
+            return false;
+    }
 
 //"ALTER TABLE capturecard DROP COLUMN dvb_recordts;" in 0.21
 //"ALTER TABLE capturecard DROP COLUMN dvb_hw_decoder;" in 0.21
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/diseqc.cpp mythtv-0.20-new/libs/libmythtv/diseqc.cpp
--- mythtv-0.20-old/libs/libmythtv/diseqc.cpp	2006-08-08 08:48:57.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/diseqc.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -450,7 +450,7 @@
     ApplyVoltage(settings, tuning);
 
     // turn off tone burst first if commands need to be sent
-    if (m_root->IsCommandNeeded(settings))
+    if (m_root->IsCommandNeeded(settings, tuning))
     {
         SetTone(false);
         usleep(DISEQC_SHORT_WAIT);
@@ -969,41 +969,40 @@
     if (pos < 0)
         return false;
 
-    // determine if switch command needs to be sent based on last pos
-    if ((m_last_pos == (uint)pos) && m_children[pos])
-        return m_children[pos]->Execute(settings, tuning);
-
     // perform switching
-    switch (m_type)
+    if (ShouldSwitch(settings, tuning))
     {
-        case kTypeTone:
-            success = ExecuteTone(settings, tuning, pos);
-            break;
-        case kTypeDiSEqCCommitted:
-        case kTypeDiSEqCUncommitted:
-            success = ExecuteDiseqc(settings, tuning, pos);
-            break;
-        case kTypeLegacySW21:
-        case kTypeLegacySW42:
-        case kTypeLegacySW64:
-            success = ExecuteLegacy(settings, tuning, pos);
-            break;
-        default:
-            success = false;
-            VERBOSE(VB_IMPORTANT, LOC_ERR +
-                    QString("Unknown switch type (%1)")
-                    .arg((uint)m_type));
-            break;
-    }
+        switch (m_type)
+        {
+            case kTypeTone:
+                success = ExecuteTone(settings, tuning, pos);
+                break;
+            case kTypeDiSEqCCommitted:
+            case kTypeDiSEqCUncommitted:
+                success = ExecuteDiseqc(settings, tuning, pos);
+                break;
+            case kTypeLegacySW21:
+            case kTypeLegacySW42:
+            case kTypeLegacySW64:
+                success = ExecuteLegacy(settings, tuning, pos);
+                break;
+            default:
+                success = false;
+                VERBOSE(VB_IMPORTANT, LOC_ERR +
+                        QString("Unknown switch type (%1)")
+                        .arg((uint)m_type));
+                break;
+        }
 
-    // if a child device will be sending a diseqc command, wait 100ms
-    if (m_children[pos]->IsCommandNeeded(settings))
-    {
-        VERBOSE(VB_CHANNEL, LOC + "Waiting for switch");
-        usleep(DISEQC_LONG_WAIT);
-    }
+        // if a child device will be sending a diseqc command, wait 100ms
+        if (m_children[pos]->IsCommandNeeded(settings, tuning))
+        {
+            VERBOSE(VB_CHANNEL, LOC + "Waiting for switch");
+            usleep(DISEQC_LONG_WAIT);
+        }
 
-    m_last_pos = pos;
+        m_last_pos = pos;
+    }
 
     // chain to child if the switch was successful
     if (success)
@@ -1015,6 +1014,8 @@
 void DiSEqCDevSwitch::Reset(void)
 {
     m_last_pos = (uint) -1;
+    m_last_high_band = (uint) -1;
+    m_last_horizontal = (uint) -1;
     dvbdev_vec_t::iterator it = m_children.begin();
     for (; it != m_children.end(); ++it)
     {
@@ -1023,19 +1024,15 @@
     }
 }
 
-bool DiSEqCDevSwitch::IsCommandNeeded(const DiSEqCDevSettings &settings) const
+bool DiSEqCDevSwitch::IsCommandNeeded(const DiSEqCDevSettings &settings,
+                                      const DVBTuning         &tuning) const
 {
-    // sanity check switch position
     int pos = GetPosition(settings);
     if (pos < 0)
         return false;
 
-    // if position is changing, a command is definitely needed
-    if ((uint)pos != m_last_pos)
-        return true;
-
-    // otherwise, the child that will be selected may need a command
-    return m_children[pos]->IsCommandNeeded(settings);
+    return (ShouldSwitch(settings, tuning) ||
+            m_children[pos]->IsCommandNeeded(settings, tuning));
 }
 
 DiSEqCDevDevice *DiSEqCDevSwitch::GetSelectedChild(const DiSEqCDevSettings &settings) const
@@ -1348,6 +1345,46 @@
     return false;
 }
 
+bool DiSEqCDevSwitch::ShouldSwitch(const DiSEqCDevSettings &settings,
+                                   const DVBTuning &tuning) const
+{
+    int pos = GetPosition(settings);
+    if (pos < 0)
+        return false;
+
+    // committed switch should change for band and polarity as well
+    if (kTypeDiSEqCCommitted == m_type)
+    {
+        // retrieve LNB info
+        bool high_band  = false;
+        bool horizontal = false;
+        DiSEqCDevLNB *lnb  = m_tree.FindLNB(settings);
+        if (lnb)
+        {
+            high_band   = lnb->IsHighBand(tuning);
+            horizontal  = lnb->IsHorizontal(tuning);
+        }
+
+        if(high_band != m_last_high_band ||
+           horizontal != m_last_horizontal)
+            return true;
+    }
+    else if (kTypeLegacySW42 == m_type ||
+             kTypeLegacySW64 == m_type)
+    {
+        // retrieve LNB info
+        bool horizontal = false;
+        DiSEqCDevLNB *lnb  = m_tree.FindLNB(settings);
+        if (lnb)
+            horizontal  = lnb->IsHorizontal(tuning);
+
+        if (horizontal != m_last_horizontal)
+            return true;
+    }
+
+    return m_last_pos != (uint)pos;
+}
+
 bool DiSEqCDevSwitch::ExecuteDiseqc(const DiSEqCDevSettings &settings,
                                     const DVBTuning &tuning,
                                     uint pos)
@@ -1385,7 +1422,13 @@
     VERBOSE(VB_CHANNEL, LOC + "Changing to DiSEqC switch port " +
             QString("%1/%2").arg(pos + 1).arg(m_num_ports));
 
-    return m_tree.SendCommand(DISEQC_ADR_SW_ALL, cmd, m_repeat, 1, &data);
+    bool ret = m_tree.SendCommand(DISEQC_ADR_SW_ALL, cmd, m_repeat, 1, &data);
+    if(ret)
+    {
+        m_last_high_band = high_band;
+        m_last_horizontal = horizontal;
+    }
+    return ret;
 }
 
 int DiSEqCDevSwitch::GetPosition(const DiSEqCDevSettings &settings) const
@@ -1490,7 +1533,8 @@
         m_child->Reset();
 }
 
-bool DiSEqCDevRotor::IsCommandNeeded(const DiSEqCDevSettings &settings) const
+bool DiSEqCDevRotor::IsCommandNeeded(const DiSEqCDevSettings &settings,
+                                     const DVBTuning         &tuning) const
 {
     double position = settings.GetValue(GetDeviceID());
 
@@ -1498,7 +1542,7 @@
         return true;
 
     if (m_child)
-        return m_child->IsCommandNeeded(settings);
+        return m_child->IsCommandNeeded(settings, tuning);
 
     return false;
 }
@@ -1528,15 +1572,20 @@
     return true;
 }
 
-uint DiSEqCDevRotor::GetVoltage(const DiSEqCDevSettings &settings,
-                                const DVBTuning         &tuning) const
+bool DiSEqCDevRotor::IsMoving(const DiSEqCDevSettings &settings) const
 {
     double position = settings.GetValue(GetDeviceID());
     double completed = GetProgress();
     bool   moving   = (completed < 1.0) || (position != m_last_position);
 
+    return (m_last_pos_known && moving);
+}
+
+uint DiSEqCDevRotor::GetVoltage(const DiSEqCDevSettings &settings,
+                                const DVBTuning         &tuning) const
+{
     // override voltage if the last position is known and the rotor is moving
-    if (m_last_pos_known && moving)
+    if (IsMoving(settings))
     {
         VERBOSE(VB_CHANNEL, LOC +
                 "Overriding voltage to 18V for faster rotor movement");
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/diseqc.h mythtv-0.20-new/libs/libmythtv/diseqc.h
--- mythtv-0.20-old/libs/libmythtv/diseqc.h	2006-07-18 09:18:43.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/diseqc.h	2006-10-28 19:16:52.000000000 -0700
@@ -159,7 +159,8 @@
     QString       GetDescription(void) const { return m_desc;        }
     virtual uint  GetChildCount(void)  const { return 0;             }
     virtual bool  IsCommandNeeded(
-        const DiSEqCDevSettings&)      const { return false;         }
+        const DiSEqCDevSettings&, const DVBTuning&)
+                                       const { return false;         }
     virtual uint  GetVoltage(
         const DiSEqCDevSettings&, const DVBTuning&) const = 0;
 
@@ -230,8 +231,11 @@
     // Gets
     dvbdev_switch_t GetType(void)       const { return m_type;      }
     uint            GetNumPorts(void)   const { return m_num_ports; }
+    bool            ShouldSwitch(const DiSEqCDevSettings &settings,
+                                 const DVBTuning &tuning) const;
     virtual uint    GetChildCount(void) const;
-    virtual bool    IsCommandNeeded(const DiSEqCDevSettings&) const;
+    virtual bool    IsCommandNeeded(const DiSEqCDevSettings&,
+                                    const DVBTuning&) const;
     virtual uint    GetVoltage(const DiSEqCDevSettings&,
                                const DVBTuning&) const;
 
@@ -257,6 +261,8 @@
     dvbdev_switch_t m_type;
     uint            m_num_ports;
     uint            m_last_pos;
+    uint            m_last_high_band;
+    uint            m_last_horizontal;
     dvbdev_vec_t    m_children;
 
     static const TypeTable SwitchTypeTable[7];
@@ -291,7 +297,9 @@
     double         GetProgress(void)     const;
     bool           IsPositionKnown(void) const;
     virtual uint   GetChildCount(void)   const { return 1;           }
-    virtual bool   IsCommandNeeded(const DiSEqCDevSettings&) const;
+    virtual bool   IsCommandNeeded(const DiSEqCDevSettings&,
+                                   const DVBTuning&) const;
+    bool           IsMoving(const DiSEqCDevSettings&) const;
     virtual uint   GetVoltage(const DiSEqCDevSettings&,
                               const DVBTuning&) const;
 
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/dvbdev/dvbci.h mythtv-0.20-new/libs/libmythtv/dvbdev/dvbci.h
--- mythtv-0.20-old/libs/libmythtv/dvbdev/dvbci.h	2006-02-21 15:20:25.000000000 -0800
+++ mythtv-0.20-new/libs/libmythtv/dvbdev/dvbci.h	2006-10-28 19:16:44.000000000 -0700
@@ -142,6 +142,7 @@
 
 class cCiHandler {
 public:
+  virtual ~cCiHandler() {};
   static cCiHandler *CreateCiHandler(const char *FileName);
   virtual int NumSlots(void) = 0;
   virtual bool Process(void) = 0;
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/DVDRingBuffer.cpp mythtv-0.20-new/libs/libmythtv/DVDRingBuffer.cpp
--- mythtv-0.20-old/libs/libmythtv/DVDRingBuffer.cpp	2006-08-28 10:11:08.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/DVDRingBuffer.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -315,7 +315,7 @@
                 ClearSubtitlesOSD();
 
                 if (autoselectsubtitle)
-                    curSubtitleTrack = dvdnav_get_active_spu_stream(dvdnav) & 0x1F;
+                    curSubtitleTrack = dvdnav_get_active_spu_stream(dvdnav);
 
                 if (parent)
                 {
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/DVDRingBuffer.h mythtv-0.20-new/libs/libmythtv/DVDRingBuffer.h
--- mythtv-0.20-old/libs/libmythtv/DVDRingBuffer.h	2006-08-08 12:05:25.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/DVDRingBuffer.h	2006-10-28 19:16:52.000000000 -0700
@@ -147,7 +147,7 @@
     /// menu pkt pts is not reliable
     long long      menupktpts;
     int            curAudioTrack;
-    int            curSubtitleTrack;
+    int8_t         curSubtitleTrack;
     bool           autoselectaudio;
     bool           autoselectsubtitle;
     const char     *dvdname;
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/eithelper.cpp mythtv-0.20-new/libs/libmythtv/eithelper.cpp
--- mythtv-0.20-old/libs/libmythtv/eithelper.cpp	2006-07-18 03:25:23.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/eithelper.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -1,5 +1,8 @@
 // -*- Mode: c++ -*-
 
+// Std C headers
+#include <time.h>
+
 // Std C++ headers
 #include <algorithm>
 using namespace std;
@@ -215,6 +218,9 @@
     fix |= fixup[(((uint64_t)eit->TSID()) << 32) |
                  (eit->OriginalNetworkID() << 16)];
     fix |= fixup[(eit->OriginalNetworkID() << 16) | eit->ServiceID()];
+    fix |= fixup[(((uint64_t)eit->TSID()) << 32) |
+                 (uint64_t)(eit->OriginalNetworkID() << 16) |
+		 (uint64_t)eit->ServiceID()];
     fix |= EITFixUp::kFixGenericDVB;
 
     uint networkid = eit->OriginalNetworkID();
@@ -364,8 +370,22 @@
         return;
 
     QDateTime starttime;
-    int off = secs_Between_1Jan1970_6Jan1980 + gps_offset + utc_offset;
-    starttime.setTime_t(off + event.start_time, Qt::LocalTime);
+    time_t off = secs_Between_1Jan1970_6Jan1980 + gps_offset + utc_offset;
+    time_t tmp = event.start_time + off;
+    tm result;
+
+    if (gmtime_r(&tmp, &result))
+    {
+        starttime.setDate(QDate(result.tm_year + 1900,
+                                result.tm_mon + 1,
+                                result.tm_mday));
+        starttime.setTime(QTime(result.tm_hour, result.tm_min, result.tm_sec));
+    }
+    else
+    {
+        starttime.setTime_t(tmp - utc_offset, Qt::LocalTime);
+    }
+
     EITFixUp::TimeFix(starttime);
     QDateTime endtime = starttime.addSecs(event.length);
 
@@ -515,9 +535,36 @@
     fix[ 4096 << 16] = EITFixUp::kFixAUStar;
     fix[ 4096 << 16] = EITFixUp::kFixAUStar;
 
-    fix[ 769LL << 32 | 8468 << 16] = EITFixUp::kEFixPro7Sat; // DVB-T Berlin
-    fix[3075LL << 32 | 8468 << 16] = EITFixUp::kEFixPro7Sat; // DVB-T Bremen
-    fix[                133 << 16] = EITFixUp::kEFixPro7Sat; // Premiere and pro7/Sat.1 
+    fix[  769LL << 32 | 8468 << 16] = EITFixUp::kEFixPro7Sat; // DVB-T Berlin
+    fix[ 3075LL << 32 | 8468 << 16] = EITFixUp::kEFixPro7Sat; // DVB-T Bremen
+    fix[ 8705LL << 32 | 8468 << 16] = EITFixUp::kEFixPro7Sat; // DVB-T Hessen
+    fix[13057LL << 32 | 8468 << 16] = EITFixUp::kEFixPro7Sat; // DVB-T Munich
+
+    // DVB-C germany: Kabel Deutschland encoding fixes
+    fix[   112LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10000LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10001LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10002LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10003LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10004LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10005LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10006LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10008LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    fix[ 10009LL << 32 | 61441U << 16] = EITFixUp::kEFixPro7Sat;
+    // on the multiplex with RTL only following channels must be fixed
+    fix[ 10007LL << 32 | 61441U << 16 | 53605] = EITFixUp::kEFixPro7Sat; //terranova
+    fix[ 10007LL << 32 | 61441U << 16 | 53607] = EITFixUp::kEFixPro7Sat; //Eurosport
+    fix[ 10007LL << 32 | 61441U << 16 | 53608] = EITFixUp::kEFixPro7Sat; //Das Vierte
+    fix[ 10007LL << 32 | 61441U << 16 | 53609] = EITFixUp::kEFixPro7Sat; //Viva
+
+    fix[ 774LL << 32 | 8468 << 16 | 16392] = EITFixUp::kEFixPro7Sat; //DVB-T Berlin dsf
+    fix[1082LL << 32 |    1 << 16 | 20001] = EITFixUp::kEFixPro7Sat; //DVB-S Pro7 Swiss
+    fix[1082LL << 32 |    1 << 16 | 20002] = EITFixUp::kEFixPro7Sat; //DVB-S Pro7 Austria
+    fix[1082LL << 32 |    1 << 16 | 20003] = EITFixUp::kEFixPro7Sat; //DVB-S Kabel1 Swiss
+    fix[1082LL << 32 |    1 << 16 | 20004] = EITFixUp::kEFixPro7Sat; //DVB-S Kabel1 Austria
+    fix[1082LL << 32 |    1 << 16 | 20005] = EITFixUp::kEFixPro7Sat; //DVB-S Sat.1 Austria
+
+    fix[ 133 << 16] = EITFixUp::kEFixPro7Sat; // Premiere and Pro7/Sat.1
 }
 
 static int calc_eit_utc_offset(void)
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/firewirerecorder.cpp mythtv-0.20-new/libs/libmythtv/firewirerecorder.cpp
--- mythtv-0.20-old/libs/libmythtv/firewirerecorder.cpp	2006-06-07 16:29:38.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/firewirerecorder.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -25,7 +25,7 @@
 const int FirewireRecorder::kBroadcastChannel    = 63;
 const int FirewireRecorder::kConnectionP2P       = 0;
 const int FirewireRecorder::kConnectionBroadcast = 1;
-const uint FirewireRecorder::kMaxBufferedPackets = 8000;
+const uint FirewireRecorder::kMaxBufferedPackets = 2000;
 
 // callback function for libiec61883
 int fw_tspacket_handler(unsigned char *tspacket, int /*len*/,
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/hdhrchannel.cpp mythtv-0.20-new/libs/libmythtv/hdhrchannel.cpp
--- mythtv-0.20-old/libs/libmythtv/hdhrchannel.cpp	2006-07-20 12:54:28.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/hdhrchannel.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -301,12 +301,6 @@
     return TunerSet("target", "0.0.0.0:0");
 }
 
-bool HDHRChannel::TuneTo(uint freqid)
-{
-    VERBOSE(VB_CHANNEL, LOC + "TuneTo("<<freqid<<")");
-    return TunerSet("channel", QString::number(freqid));
-}
-
 bool HDHRChannel::SetChannelByString(const QString &channum)
 {
     QString loc = LOC + QString("SetChannelByString(%1)").arg(channum);
@@ -379,8 +373,10 @@
         }
         else
         {
-            if (!TuneTo(freqid.toInt()))
-                return false;
+            VERBOSE(VB_IMPORTANT, LOC_ERR +
+                    "dtv_multiplex data is required for tuning");
+
+            return false;
         }
     }
     else if (!ChangeExternalChannel(freqid))
@@ -392,6 +388,8 @@
     // Set the major and minor channel for any additional multiplex tuning
     if (atsc_major || atsc_minor)
         SetCachedATSCInfo(QString("%1_%2").arg(atsc_major).arg(atsc_minor));
+    else if (mpeg_prog_num >= 0)
+        SetCachedATSCInfo(QString("0-%1").arg(mpeg_prog_num));
     else
         SetCachedATSCInfo(QString("%1_0").arg(channum));
 
@@ -449,10 +447,18 @@
 
 bool HDHRChannel::Tune(uint frequency, QString /*input*/, QString modulation)
 {
-    int freqid = get_closest_freqid("atsc", modulation, "us", frequency);
-    if (freqid > 0)
-        return TuneTo(freqid);
-    return false;
+    bool ok = false;
+
+    VERBOSE(VB_CHANNEL, LOC + "TuneTo("<<frequency<<","<<modulation<<")");
+
+    if (modulation == "8vsb")
+        ok = TunerSet("channel", QString("8vsb:%1").arg(frequency));
+    else if (modulation == "qam_64")
+        ok = TunerSet("channel", QString("qam64:%1").arg(frequency));
+    else if (modulation == "qam_256")
+        ok = TunerSet("channel", QString("qam256:%1").arg(frequency));
+
+    return ok;
 }
 
 bool HDHRChannel::SwitchToInput(const QString &inputname,
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/hdhrchannel.h mythtv-0.20-new/libs/libmythtv/hdhrchannel.h
--- mythtv-0.20-old/libs/libmythtv/hdhrchannel.h	2006-07-20 12:54:28.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/hdhrchannel.h	2006-10-28 19:16:53.000000000 -0700
@@ -62,7 +62,6 @@
 
     bool DeviceSetTarget(unsigned short localPort);
     bool DeviceClearTarget(void);
-    bool TuneTo(uint freqid);
 
     QString DeviceGet(const QString &name);
     QString DeviceSet(const QString &name, const QString &value);
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/hdhrrecorder.cpp mythtv-0.20-new/libs/libmythtv/hdhrrecorder.cpp
--- mythtv-0.20-old/libs/libmythtv/hdhrrecorder.cpp	2006-07-03 12:12:30.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/hdhrrecorder.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -151,15 +151,12 @@
     }
 }
 
-void HDHRRecorder::SetStreamData(MPEGStreamData *xdata)
+void HDHRRecorder::SetStreamData(MPEGStreamData *data)
 {
-    ATSCStreamData *data = dynamic_cast<ATSCStreamData*>(xdata);
-    VERBOSE(VB_IMPORTANT, LOC + "SetStreamData(xdata: "<<xdata<<") "<<data);
-
     if (data == _stream_data)
         return;
 
-    ATSCStreamData *old_data = _stream_data;
+    MPEGStreamData *old_data = _stream_data;
     _stream_data = data;
     if (old_data)
         delete old_data;
@@ -168,14 +165,20 @@
     {
         data->AddMPEGSPListener(this);
         data->AddMPEGListener(this);
-        data->SetDesiredChannel(data->DesiredMajorChannel(),
-                                data->DesiredMinorChannel());
+
+        ATSCStreamData *atsc = dynamic_cast<ATSCStreamData*>(data);
+
+        if (atsc && atsc->DesiredMinorChannel())
+            atsc->SetDesiredChannel(atsc->DesiredMajorChannel(),
+                                    atsc->DesiredMinorChannel());
+        else if (data->DesiredProgram() >= 0)
+            data->SetDesiredProgram(data->DesiredProgram());
     }
 }
 
-MPEGStreamData *HDHRRecorder::GetStreamData(void)
+ATSCStreamData *HDHRRecorder::GetATSCStreamData(void)
 {
-    return _stream_data;
+    return dynamic_cast<ATSCStreamData*>(_stream_data);
 }
 
 void HDHRRecorder::HandlePAT(const ProgramAssociationTable *_pat)
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/hdhrrecorder.h mythtv-0.20-new/libs/libmythtv/hdhrrecorder.h
--- mythtv-0.20-old/libs/libmythtv/hdhrrecorder.h	2006-06-06 13:12:12.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/hdhrrecorder.h	2006-10-28 19:16:52.000000000 -0700
@@ -39,8 +39,8 @@
     void StartRecording(void);
 
     void SetStreamData(MPEGStreamData*);
-    MPEGStreamData *GetStreamData(void);
-    ATSCStreamData *GetATSCStreamData(void) { return _stream_data; }
+    MPEGStreamData *GetStreamData(void) { return _stream_data; }
+    ATSCStreamData *GetATSCStreamData(void);
 
     // MPEG Stream Listener
     void HandlePAT(const ProgramAssociationTable*);
@@ -69,7 +69,7 @@
   private:
     HDHRChannel                   *_channel;
     struct hdhomerun_video_sock_t *_video_socket;
-    ATSCStreamData                *_stream_data;
+    MPEGStreamData                *_stream_data;
 
     ProgramAssociationTable       *_input_pat;
     ProgramMapTable               *_input_pmt;
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/mpeg/atsctables.h mythtv-0.20-new/libs/libmythtv/mpeg/atsctables.h
--- mythtv-0.20-old/libs/libmythtv/mpeg/atsctables.h	2006-04-23 08:04:23.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/mpeg/atsctables.h	2006-10-28 19:16:49.000000000 -0700
@@ -44,7 +44,7 @@
  */
 
 /** Seconds between start of GPS time and the start of UNIX time. */
-#define secs_Between_1Jan1970_6Jan1980 315982800
+#define secs_Between_1Jan1970_6Jan1980 315964800
 
 /** Leap seconds as of Jan 1st, 2006. */
 #define GPS_LEAP_SECONDS 14
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/mpeg/dvbdescriptors.h mythtv-0.20-new/libs/libmythtv/mpeg/dvbdescriptors.h
--- mythtv-0.20-old/libs/libmythtv/mpeg/dvbdescriptors.h	2006-08-10 14:23:33.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/mpeg/dvbdescriptors.h	2006-10-28 19:16:49.000000000 -0700
@@ -618,7 +618,7 @@
     uint Modulation() const { return _data[8]&0x1f; }
     QString ModulationString() const
     {
-        static QString ms[] = { "qpsk", "qpsk", "qpsk_8", "qam_16" };
+        static QString ms[] = { "qpsk", "qpsk", "8psk", "qam_16" };
         return (Modulation() <= kModulationQAM16) ? ms[Modulation()] : "auto";
     }
     // symbol_rate             28   9.0
@@ -1160,6 +1160,20 @@
         kServiceTypeRCS_FLS                  = 0x0F,
         kServiceTypeDVB_MHP                  = 0x10,
         kServiceTypeHDTV                     = 0x19,
+        kServiceTypeEchoStarTV1              = 0x91,
+        kServiceTypeEchoStarTV2              = 0x9a,
+        kServiceTypeEchoStarTV3              = 0xa4,
+        kServiceTypeEchoStarTV4              = 0xa6,
+        kServiceTypeNimiqTV1                 = 0x81,
+        kServiceTypeNimiqTV2                 = 0x85,
+        kServiceTypeNimiqTV3                 = 0x86,
+        kServiceTypeNimiqTV4                 = 0x89,
+        kServiceTypeNimiqTV5                 = 0x8a,
+        kServiceTypeNimiqTV6                 = 0x8d, 
+        kServiceTypeNimiqTV7                 = 0x8f,
+        kServiceTypeNimiqTV8                 = 0x90,
+        kServiceTypeNimiqTV9                 = 0x96,
+
     };
     // service_type             8   2.0
     uint ServiceType(void) const { return _data[2]; }
@@ -1178,7 +1192,20 @@
                                ServiceNameLength());
     }
     bool IsDTV(void) const
-        { return ServiceType() ==  kServiceTypeDigitalTelevision; }
+        { return ((ServiceType() ==  kServiceTypeDigitalTelevision) ||
+                  (ServiceType() ==  kServiceTypeEchoStarTV1) ||
+                  (ServiceType() ==  kServiceTypeEchoStarTV2) ||
+                  (ServiceType() ==  kServiceTypeEchoStarTV3) ||
+                  (ServiceType() ==  kServiceTypeEchoStarTV4) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV1) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV2) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV3) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV4) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV5) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV6) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV7) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV8) ||
+                  (ServiceType() ==  kServiceTypeNimiqTV9)); }
     bool IsDigitalAudio(void) const
         { return ServiceType() ==  kServiceTypeDigitalRadioSound; }
     bool IsHDTV(void) const
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/NuppelVideoPlayer.cpp mythtv-0.20-new/libs/libmythtv/NuppelVideoPlayer.cpp
--- mythtv-0.20-old/libs/libmythtv/NuppelVideoPlayer.cpp	2006-09-04 19:37:19.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/NuppelVideoPlayer.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -573,7 +573,7 @@
 
 void NuppelVideoPlayer::ReinitOSD(void)
 {
-    if (videoOutput)
+    if (videoOutput && !using_null_videoout)
     {
         QRect visible, total;
         float aspect, scaling;
@@ -6314,11 +6314,10 @@
     int numbuttons = ringBuffer->DVD()->NumMenuButtons();
     bool osdshown = osd->IsSetDisplaying("subtitles");
     long long menupktpts = ringBuffer->DVD()->GetMenuPktPts();
-    bool instillframe = ringBuffer->DVD()->InStillFrame();
 
     if ((numbuttons == 0) || 
         (osdshown) ||
-        (instillframe && buffer->timecode > 0) ||
+        (indvdstillframe && buffer->timecode > 0) ||
         ((!osdshown) && 
             (!indvdstillframe) &&
             (hidedvdbutton) &&
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/osdtypes.cpp mythtv-0.20-new/libs/libmythtv/osdtypes.cpp
--- mythtv-0.20-old/libs/libmythtv/osdtypes.cpp	2006-09-04 11:33:22.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/osdtypes.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -2146,7 +2146,7 @@
         QPoint((int)round(pos.x() / wmult),
                (int)round(pos.y() / hmult)));
 
-    VERBOSE(VB_IMPORTANT,
+    VERBOSE(VB_OSD,
             "OSDTypePositionImage::AddPosition["<<m_numpositions<<"]("
             <<pos.x()<<"x"<<pos.y()
             <<"  "<<wmult<<", "<<hmult<<")");
@@ -2157,7 +2157,7 @@
 void OSDTypePositionImage::Draw(OSDSurface *surface, int fade, int maxfade,
                                 int xoff, int yoff)
 {
-    VERBOSE(VB_IMPORTANT,
+    VERBOSE(VB_OSD,
             "OSDTypePositionImage::Draw["<<m_curposition<<"]("
             <<m_wmult<<", "<<m_hmult<<")");
 
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/programinfo.cpp mythtv-0.20-new/libs/libmythtv/programinfo.cpp
--- mythtv-0.20-old/libs/libmythtv/programinfo.cpp	2006-09-10 21:35:19.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/programinfo.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -309,6 +309,7 @@
     INT_TO_LIST(hasAirDate)     
     STR_TO_LIST((playgroup != "") ? playgroup : "Default")
     INT_TO_LIST(recpriority2)
+    INT_TO_LIST(parentid)
 }
 
 /** \fn ProgramInfo::FromStringList(QStringList&,int)
@@ -407,6 +408,7 @@
     INT_FROM_LIST(hasAirDate);
     STR_FROM_LIST(playgroup)
     INT_FROM_LIST(recpriority2)
+    INT_FROM_LIST(parentid)
 
     return true;
 }
@@ -1039,7 +1041,11 @@
     query.prepare("UPDATE recorded "
                   "SET recordid = :RECID "
                   "WHERE chanid = :CHANID AND starttime = :START");
-    query.bindValue(":RECID",  getRecordID());
+
+    if (rectype == kOverrideRecord && parentid > 0)
+        query.bindValue(":RECID", parentid);
+    else
+        query.bindValue(":RECID",  getRecordID());
     query.bindValue(":CHANID", chanid);
     query.bindValue(":START",  recstartts);
 
@@ -1419,15 +1425,15 @@
     return true;
 }               
 
-/** \fn ProgramInfo::GetRecordBasename(void) const
+/** \fn ProgramInfo::GetRecordBasename() const
  *  \brief Returns a filename for a recording based on the
  *         recording channel and date.
  */
-QString ProgramInfo::GetRecordBasename(void) const
+QString ProgramInfo::GetRecordBasename(bool fromDB) const
 {
     QString retval = "";
 
-    if (!pathname.isEmpty())
+    if (!fromDB && !pathname.isEmpty())
         retval = pathname.section('/', -1);
     else
     {
@@ -1452,13 +1458,13 @@
     return retval;
 }               
 
-/** \fn ProgramInfo::GetRecordFilename(const QString&) const
+/** \fn ProgramInfo::GetRecordFilename() const
  *  \brief Returns prefix+"/"+GetRecordBasename()
  *  \param prefix Prefix to apply to GetRecordBasename().
  */
-QString ProgramInfo::GetRecordFilename(const QString &prefix) const
+QString ProgramInfo::GetRecordFilename(const QString &prefix, bool fromDB) const
 {
-    return QString("%1/%2").arg(prefix).arg(GetRecordBasename());
+    return QString("%1/%2").arg(prefix).arg(GetRecordBasename(fromDB));
 }               
 
 /** \fn ProgramInfo::GetPlaybackURL(QString) const
@@ -1566,9 +1572,11 @@
 
     query.prepare("REPLACE INTO recordedprogram"
                  " SELECT * from program"
-                 " WHERE chanid = :CHANID AND starttime = :START;");
+                 " WHERE chanid = :CHANID AND starttime = :START"
+                 " AND title = :TITLE;");
     query.bindValue(":CHANID", chanid);
     query.bindValue(":START", startts);
+    query.bindValue(":TITLE", title);
     if (!query.exec() || !query.isActive())
         MythContext::DBError("Copy program data on record", query);
 
@@ -1662,17 +1670,28 @@
     query.bindValue(":PROFILE",     schd->getProfileName());
 
     bool ok = query.exec() && (query.numRowsAffected() > 0);
-    if (!ok && !query.isActive())
+    bool active = query.isActive();
+
+    query.prepare("UNLOCK TABLES");
+    query.exec();
+
+    if (!ok && !active)
         MythContext::DBError("insert_program -- insert", query);
     else
     {
         query.prepare("UPDATE record SET last_record = NOW() "
                       "WHERE recordid = :RECORDID");
-        query.bindValue(":RECORDID",    pg->recordid);
+        query.bindValue(":RECORDID", pg->recordid);
         query.exec();
+
+        if (pg->rectype == kOverrideRecord && pg->parentid > 0)
+        {
+            query.prepare("UPDATE record SET last_record = NOW() "
+                          "WHERE recordid = :PARENTID");
+            query.bindValue(":PARENTID", pg->parentid);
+            query.exec();
+        }
     }
-    query.prepare("UNLOCK TABLES");
-    query.exec();
 
     return ok;
 }
@@ -1710,12 +1729,11 @@
 void ProgramInfo::UpdateRecordingEnd(void)
 {
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("UPDATE recorded SET endtime = :ENDTIME, "
-                  "    recordid = :RECORDID "
+    query.prepare("UPDATE recorded SET endtime = :ENDTIME "
                   "WHERE chanid = :CHANID AND "
                   "    starttime = :STARTTIME ");
     query.bindValue(":ENDTIME", recendts);
-    query.bindValue(":RECORDID", recordid);
+
     query.bindValue(":CHANID", chanid);
     query.bindValue(":STARTTIME", recstartts);
 
@@ -2823,6 +2841,8 @@
         return QObject::tr("K", "RecStatusChar rsLowDiskSpace");
     case rsTunerBusy:
         return QObject::tr("B", "RecStatusChar rsTunerBusy");
+    case rsFailed:
+        return QObject::tr("f", "RecStatusChar rsFailed");
     case rsNotListed:
         return QObject::tr("N", "RecStatusChar rsNotListed");
     case rsNeverRecord:
@@ -2879,6 +2899,8 @@
             return QObject::tr("Low Disk Space");
         case rsTunerBusy:
             return QObject::tr("Tuner Busy");
+        case rsFailed:
+            return QObject::tr("Recorder Failed");
         case rsNotListed:
             return QObject::tr("Not Listed");
         case rsNeverRecord:
@@ -2932,6 +2954,9 @@
         case rsTunerBusy:
             message += QObject::tr("the tuner card was already being used.");
             break;
+        case rsFailed:
+            message += QObject::tr("the recording failed.");
+            break;
         default:
             message = QObject::tr("The status of this showing is unknown.");
             break;
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/programinfo.h mythtv-0.20-new/libs/libmythtv/programinfo.h
--- mythtv-0.20-old/libs/libmythtv/programinfo.h	2006-07-14 12:31:24.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/programinfo.h	2006-10-28 19:16:52.000000000 -0700
@@ -13,7 +13,7 @@
 typedef QMap<long long, long long> frm_pos_map_t;
 typedef QMap<long long, int> frm_dir_map_t;
 
-#define NUMPROGRAMLINES 41
+#define NUMPROGRAMLINES 42
 
 typedef enum {
     MARK_UNSET = -10,
@@ -59,6 +59,7 @@
 };
 
 enum RecStatusType {
+    rsFailed = -9,
     rsTunerBusy = -8,
     rsLowDiskSpace = -7,
     rsCancelled = -6,
@@ -157,8 +158,8 @@
 
     // Quick gets
     bool SetRecordBasename(QString basename);
-    QString GetRecordBasename(void) const;
-    QString GetRecordFilename(const QString &prefix) const;
+    QString GetRecordBasename(bool fromDB = false) const;
+    QString GetRecordFilename(const QString &prefix, bool fromDB = false) const;
     QString GetPlaybackURL(QString playbackHost = "") const;
     QString MakeUniqueKey(void) const;
     int CalculateLength(void) const;
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/tv_play.cpp mythtv-0.20-new/libs/libmythtv/tv_play.cpp
--- mythtv-0.20-old/libs/libmythtv/tv_play.cpp	2006-09-07 09:52:11.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/tv_play.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -5445,6 +5445,7 @@
         return;
 
     browsechannum = chan;
+    browsechanid = QString::null;
     BrowseDispInfo(BROWSE_SAME);
 }
 
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/tv_rec.cpp mythtv-0.20-new/libs/libmythtv/tv_rec.cpp
--- mythtv-0.20-old/libs/libmythtv/tv_rec.cpp	2006-09-09 12:45:01.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/tv_rec.cpp	2006-10-28 19:16:53.000000000 -0700
@@ -554,6 +554,10 @@
 
     WaitForEventThreadSleep();
 
+    if ((curRecording) && (curRecording->recstatus == rsFailed) &&
+        (retval == rsRecording))
+        retval = rsFailed;
+
     return retval;
 }
 
@@ -675,20 +679,21 @@
     VERBOSE(VB_RECORD, LOC + QString("FinishedRecording(%1) in recgroup: %2")
                                      .arg(curRec->title).arg(pigrp));
 
-    curRec->recstatus = rsRecorded;
+    if (curRec->recstatus != rsFailed)
+        curRec->recstatus = rsRecorded;
     curRec->recendts = mythCurrentDateTime();
 
     if (tvchain)
         tvchain->FinishedRecording(curRec);
 
+    // Make sure really short recordings have positive run time.
+    if (curRec->recendts <= curRec->recstartts)
+        curRec->recendts = curRec->recstartts.addSecs(60);
+
     curRec->recendts.setTime(QTime(
         curRec->recendts.addSecs(30).time().hour(),
         curRec->recendts.addSecs(30).time().minute()));
 
-    // Make sure really short recordings have positive run time.
-    if (curRec->recendts <= curRec->recstartts)
-        curRec->recendts = mythCurrentDateTime().addSecs(1);
-
     if (pigrp != "LiveTV")
     {
         MythEvent me(QString("UPDATE_RECORDING_STATUS %1 %2 %3 %4 %5")
@@ -699,7 +704,8 @@
                      .arg(curRec->recendts.toString(Qt::ISODate)));
         gContext->dispatch(me);
     }
-    curRec->FinishedRecording(false);
+
+    curRec->FinishedRecording(curRec->recstatus != rsRecorded);
 }
 
 #define TRANSITION(ASTATE,BSTATE) \
@@ -3443,6 +3449,9 @@
     {
         if (!(request.flags & kFlagLiveTV))
         {
+            if (curRecording)
+                curRecording->recstatus = rsFailed;
+
             VERBOSE(VB_IMPORTANT, LOC_ERR +
                     QString("Failed to set channel to %1. "
                             "Reverting to kState_None")
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/videodev2_myth.h mythtv-0.20-new/libs/libmythtv/videodev2_myth.h
--- mythtv-0.20-old/libs/libmythtv/videodev2_myth.h	2006-07-18 08:55:57.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/videodev2_myth.h	2006-10-28 19:16:52.000000000 -0700
@@ -721,7 +721,7 @@
 		__s64 value64;
 		void *reserved;
 	};
-};
+}  __attribute__ ((packed));
 
 struct v4l2_ext_controls
 {
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/videosource.cpp mythtv-0.20-new/libs/libmythtv/videosource.cpp
--- mythtv-0.20-old/libs/libmythtv/videosource.cpp	2006-09-07 08:43:22.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/videosource.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -2079,6 +2079,30 @@
     if (val == 0)
     {
         MSqlQuery query(MSqlQuery::InitCon());
+
+        // Delete the channels associated with the source
+        query.prepare("DELETE FROM channel "
+                      "WHERE sourceid = :SOURCEID");
+        query.bindValue(":SOURCEID", getValue());
+
+        if (!query.exec() || !query.isActive())
+        {
+            MythContext::DBError("Deleting Channels", query);
+            return;
+        }
+
+        // Delete the inputs associated with the source
+        query.prepare("DELETE FROM cardinput "
+                      "WHERE sourceid = :SOURCEID");
+        query.bindValue(":SOURCEID", getValue());
+
+        if (!query.exec() || !query.isActive())
+        {
+            MythContext::DBError("Deleting cardinputs", query);
+            return;
+        }
+
+        // Delete the source itself
         query.prepare("DELETE FROM videosource "
                       "WHERE sourceid = :SOURCEID");
         query.bindValue(":SOURCEID", getValue());
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/vsync.cpp mythtv-0.20-new/libs/libmythtv/vsync.cpp
--- mythtv-0.20-old/libs/libmythtv/vsync.cpp	2006-05-24 07:56:37.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/vsync.cpp	2006-10-28 19:16:52.000000000 -0700
@@ -437,12 +437,64 @@
     UpdateNexttrigger();
 }
 
+#ifdef USING_OPENGL_VSYNC
+class OpenGLVideoSyncPrivate
+{
+  public:
+    OpenGLVideoSyncPrivate()
+    {
+        m_glXGetVideoSyncSGI = (PFNGLXGETVIDEOSYNCSGIPROC)
+                glXGetProcAddress("glXGetVideoSyncSGI");
+        m_glXWaitVideoSyncSGI = (PFNGLXWAITVIDEOSYNCSGIPROC)
+                glXGetProcAddress("glXWaitVideoSyncSGI");
+    }
+
+    bool funcsLoaded()
+    {
+        return m_glXGetVideoSyncSGI && m_glXWaitVideoSyncSGI;
+    }
+
+  public:
+    int glXGetVideoSyncSGI(unsigned int *count)
+    {
+        return m_glXGetVideoSyncSGI(count);
+    }
+
+    int glXWaitVideoSyncSGI(int divisor, int remainder, unsigned int *count)
+    {
+        return m_glXWaitVideoSyncSGI(divisor, remainder, count);
+    }
+
+  private:
+    __GLXextFuncPtr glXGetProcAddress(const char * const procName)
+    {
+        __GLXextFuncPtr ret = glXGetProcAddressARB((const GLubyte *) procName);
+
+        if (!ret)
+        {
+            VERBOSE(VB_PLAYBACK,
+                    QString("Error: glXGetProcAddressARB unable to find %1")
+                    .arg(procName));
+        }
+
+        return ret;
+    }
+
+  private:
+    PFNGLXGETVIDEOSYNCSGIPROC m_glXGetVideoSyncSGI;
+    PFNGLXWAITVIDEOSYNCSGIPROC m_glXWaitVideoSyncSGI;
+};
+#endif // USING_OPENGL_VSYNC
+
 OpenGLVideoSync::OpenGLVideoSync(VideoOutput *video_output,
                                  int frame_interval, int refresh_interval,
                                  bool interlaced)
     : VideoSync(video_output, frame_interval, refresh_interval, interlaced),
-      m_drawable(0), m_context(0)
+      m_drawable(0), m_context(0), m_imp(0)
 {
+#ifdef USING_OPENGL_VSYNC
+    m_imp = new OpenGLVideoSyncPrivate;
+#endif // USING_OPENGL_VSYNC
 }
 
 OpenGLVideoSync::~OpenGLVideoSync()
@@ -457,6 +509,7 @@
         if (m_drawable)
             X11S(XDestroyWindow(vo->XJ_disp, m_drawable));
     }
+    delete m_imp;
 #endif /* USING_OPENGL_VSYNC */
 }
 
@@ -503,6 +556,12 @@
         return false;
     }
 
+    if (!m_imp->funcsLoaded())
+    {
+       VERBOSE(VB_PLAYBACK, QString("GL sync functions not found"));
+       return false;
+    }
+
     int attribList[] = {GLX_RGBA, 
                         GLX_RED_SIZE, 1,
                         GLX_GREEN_SIZE, 1,
@@ -552,7 +611,7 @@
     if (ret != False)
     {
         unsigned int count;
-        X11S(ret = glXGetVideoSyncSGI(&count));
+        X11S(ret = m_imp->glXGetVideoSyncSGI(&count));
         if (ret == 0)
         {
             VERBOSE(VB_PLAYBACK, "Using OpenGLVideoSync");
@@ -631,9 +690,9 @@
 
     // Wait for a refresh so we start out synched
     unsigned int count;
-    err = glXGetVideoSyncSGI(&count);
+    err = m_imp->glXGetVideoSyncSGI(&count);
     checkGLSyncError("OpenGLVideoSync::Start(): Frame Number Query", err);
-    err = glXWaitVideoSyncSGI(2, (count+1)%2 ,&count);
+    err = m_imp->glXWaitVideoSyncSGI(2, (count+1)%2 ,&count);
     checkGLSyncError("OpenGLVideoSync::Start(): A/V Sync", err);
 
     // Initialize next trigger 
@@ -649,13 +708,13 @@
     OffsetTimeval(m_nexttrigger, sync_delay);
 
     unsigned int frameNum = 0;
-    int err = glXGetVideoSyncSGI(&frameNum);
+    int err = m_imp->glXGetVideoSyncSGI(&frameNum);
     checkGLSyncError("Frame Number Query", err);
 
     // Always sync to the next retrace execpt when we are very late.
     if ((m_delay = CalcDelay()) > -(m_refresh_interval/2)) 
     {
-        err = glXWaitVideoSyncSGI(2, (frameNum+1)%2 ,&frameNum);
+        err = m_imp->glXWaitVideoSyncSGI(2, (frameNum+1)%2 ,&frameNum);
         checkGLSyncError(msg1, err);
         m_delay = CalcDelay();
     }
@@ -664,7 +723,7 @@
     if (m_delay > 0)
     {
         uint n = m_delay / m_refresh_interval + 1;
-        err = glXWaitVideoSyncSGI((n+1), (frameNum+n)%(n+1), &frameNum);
+        err = m_imp->glXWaitVideoSyncSGI((n+1), (frameNum+n)%(n+1), &frameNum);
         checkGLSyncError(msg2, err);
         m_delay = CalcDelay();
     }
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythtv/vsync.h mythtv-0.20-new/libs/libmythtv/vsync.h
--- mythtv-0.20-old/libs/libmythtv/vsync.h	2006-05-11 07:50:17.000000000 -0700
+++ mythtv-0.20-new/libs/libmythtv/vsync.h	2006-10-28 19:16:52.000000000 -0700
@@ -214,6 +214,8 @@
   private:
     GLXDrawable m_drawable;
     GLXContext  m_context;
+  private:
+    class OpenGLVideoSyncPrivate *m_imp;
 };
 
 #ifdef __linux__
diff -Naur -x mytharchive mythtv-0.20-old/libs/libmythupnp/httprequest.cpp mythtv-0.20-new/libs/libmythupnp/httprequest.cpp
--- mythtv-0.20-old/libs/libmythupnp/httprequest.cpp	2006-06-07 08:41:00.000000000 -0700
+++ mythtv-0.20-new/libs/libmythupnp/httprequest.cpp	2006-10-28 19:17:08.000000000 -0700
@@ -257,12 +257,16 @@
     else
         m_nResponseStatus = 404;
 
+    QString sDate = QDateTime::currentDateTime().toString( "d MMM yyyy hh:mm:ss" );  
+
     sHeader   = QString("HTTP/%1.%2 %3\r\n"
-                        "Content-Type: %4\r\n"
-                        "Content-Length: %5\r\n" )
+                        "Date: %4\r\n"
+                        "Content-Type: %5\r\n"
+                        "Content-Length: %6\r\n" )
                         .arg( m_nMajor )
                         .arg( m_nMinor )
                         .arg( GetResponseStatus())
+                        .arg( sDate )
                         .arg( sContentType )
                         .arg( llSize       ).utf8();
 
@@ -281,7 +285,19 @@
     {
         __off64_t offset = llStart;
         int       file   = open( sFileName.ascii(), O_RDONLY | O_LARGEFILE );
-        sendfile64( getSocketHandle(), file, &offset, llSize );
+        ssize_t   sent   = 0;  
+
+        do 
+        {  
+            // SSIZE_MAX should work in kernels 2.6.16 and later.  
+            // The loop is needed in any case.  
+
+            sent = sendfile64( getSocketHandle(), file, &offset, 
+                               (size_t)(llSize > INT_MAX ? INT_MAX : llSize));  
+
+            llSize -= sent;  
+        } 
+        while (( sent >= 0 ) && ( llSize > 0 ));  
 
         close( file );
     }
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythbackend/housekeeper.cpp mythtv-0.20-new/programs/mythbackend/housekeeper.cpp
--- mythtv-0.20-old/programs/mythbackend/housekeeper.cpp	2006-07-16 13:56:57.000000000 -0700
+++ mythtv-0.20-new/programs/mythbackend/housekeeper.cpp	2006-10-28 19:17:11.000000000 -0700
@@ -221,16 +221,11 @@
                 }
             }
 
-            if (wantToRun("JobQueueCleanup", 1, 0, 24))
-            {
+            if (wantToRun("DailyCleanup", 1, 0, 24)) {
                 JobQueue::CleanupOldJobsInQueue();
-                updateLastrun("JobQueueCleanup");
-            }
-
-            if (wantToRun("InUseProgramsCleanup", 1, 0, 24))
-            {
                 CleanupAllOldInUsePrograms();
-                updateLastrun("InUseProgramsCleanup");
+                CleanupRecordedTables();
+                updateLastrun("DailyCleanup");
             }
         }
 
@@ -331,6 +326,48 @@
     query.exec();
 }
 
+void HouseKeeper::CleanupRecordedTables(void)
+{
+    MSqlQuery query(MSqlQuery::InitCon());
+    MSqlQuery deleteQuery(MSqlQuery::InitCon());
+    int tableIndex = 0;
+    QString tables[] = {
+        "recordedprogram",
+        "recordedrating",
+        "recordedcredits",
+        "" }; // This blank entry must exist, do not remove.
+    QString table = tables[tableIndex];
+   
+    while (table != "")
+    {
+        query.prepare(QString("SELECT DISTINCT p.chanid, p.starttime "
+                              "FROM %1 p LEFT JOIN recorded r "
+                              "ON p.chanid = r.chanid "
+                              "AND p.starttime = r.progstart "
+                              "WHERE r.chanid IS NULL;")
+                              .arg(table));
+        if (!query.exec() || !query.isActive())
+        {
+            MythContext::DBError("HouseKeeper Cleaning Recorded Tables", query);
+            return;
+        }
+
+        deleteQuery.prepare(QString("DELETE FROM %1 "
+                                    "WHERE chanid = :CHANID "
+                                    "AND starttime = :STARTTIME;")
+                                    .arg(table));
+        while (query.next())
+        {
+            deleteQuery.bindValue(":CHANID", query.value(0).toString());
+            deleteQuery.bindValue(":STARTTIME", query.value(1).toString());
+            deleteQuery.exec();
+        }
+
+        tableIndex++;
+        table = tables[tableIndex];
+    }
+}
+
 void *HouseKeeper::doHouseKeepingThread(void *param)
 {
     HouseKeeper *hkeeper = (HouseKeeper*)param;
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythbackend/housekeeper.h mythtv-0.20-new/programs/mythbackend/housekeeper.h
--- mythtv-0.20-old/programs/mythbackend/housekeeper.h	2006-05-27 16:03:56.000000000 -0700
+++ mythtv-0.20-new/programs/mythbackend/housekeeper.h	2006-10-28 19:17:11.000000000 -0700
@@ -34,6 +34,7 @@
     void runFillDatabase();
     void CleanupMyOldRecordings(void);
     void CleanupAllOldInUsePrograms(void);
+    void CleanupRecordedTables(void);
     bool threadrunning;
     bool filldbRunning;
     bool isMaster;
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythbackend/main.cpp mythtv-0.20-new/programs/mythbackend/main.cpp
--- mythtv-0.20-old/programs/mythbackend/main.cpp	2006-06-08 21:41:00.000000000 -0700
+++ mythtv-0.20-new/programs/mythbackend/main.cpp	2006-10-28 19:17:11.000000000 -0700
@@ -268,6 +268,7 @@
     bool testsched = false;
     bool resched = false;
     bool nosched = false;
+    bool noupnp = false;
     bool nojobqueue = false;
     bool nohousekeeper = false;
     bool noexpirer = false;
@@ -378,6 +379,10 @@
         {
             nosched = true;
         } 
+        else if (!strcmp(a.argv()[argpos],"--noupnp"))
+        {
+            noupnp = true;
+        } 
         else if (!strcmp(a.argv()[argpos],"--nojobqueue"))
         {
             nojobqueue = true;
@@ -422,6 +427,7 @@
                     "--testsched                    Test run scheduler (ignore existing schedule)" << endl <<
                     "--resched                      Force the scheduler to update" << endl <<
                     "--nosched                      Do not perform any scheduling" << endl <<
+                    "--noupnp                       Do not enable the UPNP server" << endl <<
                     "--nojobqueue                   Do not start the JobQueue" << endl <<
                     "--nohousekeeper                Do not start the Housekeeper" << endl <<
                     "--noautoexpire                 Do not start the AutoExpire thread" << endl <<
@@ -626,7 +632,11 @@
     g_pHttpServer->RegisterExtension(new HttpStatus(&tvList, sched, expirer, ismaster ));
 
     // Start UPnP Services For Master Backends Only
-    if (ismaster)
+    if (ismaster && noupnp)
+        cerr << "********* The UPNP service has been DISABLED with the "
+                "--noupnp option *********\n";
+
+    if (ismaster && !noupnp)
     {
         g_pUPnp = new UPnp(ismaster, g_pHttpServer);
 
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythbackend/mainserver.cpp mythtv-0.20-new/programs/mythbackend/mainserver.cpp
--- mythtv-0.20-old/programs/mythbackend/mainserver.cpp	2006-09-10 21:35:19.000000000 -0700
+++ mythtv-0.20-new/programs/mythbackend/mainserver.cpp	2006-10-28 19:17:11.000000000 -0700
@@ -674,6 +674,36 @@
             return;
         }
 
+        if ((me->Message().left(16) == "DELETE_RECORDING") ||
+            (me->Message().left(22) == "FORCE_DELETE_RECORDING"))
+        {
+            QStringList tokens = QStringList::split(" ", me->Message());
+            if (tokens.size() != 3)
+            {
+                VERBOSE(VB_IMPORTANT, QString("Bad %1 message").arg(tokens[0]));
+                return;
+            }
+
+            QDateTime startts = QDateTime::fromString(tokens[2], Qt::ISODate);
+            ProgramInfo *pinfo = ProgramInfo::GetProgramFromRecorded(tokens[1],
+                                                                     startts);
+            if (pinfo)
+            {
+                if (tokens[0] == "FORCE_DELETE_RECORDING")
+                    DoHandleDeleteRecording(pinfo, NULL, true);
+                else
+                    DoHandleDeleteRecording(pinfo, NULL, false);
+            }
+            else
+            {
+                VERBOSE(VB_IMPORTANT,
+                    QString("Cannot find program info for '%1' while "
+                            "attempting to delete.").arg(me->Message()));
+            }
+
+            return;
+        }
+
         if (me->Message().left(21) == "RESCHEDULE_RECORDINGS" && m_sched)
         {
             QStringList tokens = QStringList::split(" ", me->Message());
@@ -1393,11 +1423,17 @@
     bool followLinks = gContext->GetNumSetting("DeletesFollowLinks", 0);
     bool slowDeletes = gContext->GetNumSetting("TruncateDeletesSlowly", 0);
     int fd = -1;
+    off_t size = 0;
     bool errmsg = false;
 
     /* Delete recording. */
     if (slowDeletes)
     {
+        // Since stat fails after unlinking on some filesystems,
+        // get the filesize first
+        struct stat st;
+        if (stat(ds->filename.ascii(), &st) == 0)
+            size = st.st_size;
         fd = DeleteFile(ds->filename, followLinks);
 
         if ((fd < 0) && checkFile.exists())
@@ -1445,7 +1481,7 @@
     if (slowDeletes && fd != -1)
     {
         m_expirer->TruncatePending();
-        TruncateAndClose(m_expirer, fd, ds->filename);
+        TruncateAndClose(m_expirer, fd, ds->filename, size);
         m_expirer->TruncateFinished();
     }
 }
@@ -1600,7 +1636,8 @@
  *         is running at a time.
  */
 bool MainServer::TruncateAndClose(const AutoExpire *expirer,
-                                  int fd, const QString &filename)
+                                  int fd, const QString &filename,
+                                  off_t fsize)
 {
     QMutexLocker locker(&truncate_and_close_lock);
 
@@ -1621,17 +1658,6 @@
             .arg(increment / (1024.0 * 1024.0), 0, 'f', 2)
             .arg(sleep_time));
 
-    // Get the on disk file size and preferred I/O block size.
-    struct stat buf;
-    fstat(fd, &buf);
-    // Estimate the file size.  Don't use buf.st_blksize * buf.st_blocks
-    // The unit for st_blocks is undefined.  See section "RATIONALE" at
-    // http://www.opengroup.org/onlinepubs/000095399/basedefs/sys/stat.h.html
-    off_t fsize = ((buf.st_size / buf.st_blksize) + 1) * buf.st_blksize;
-
-    // Round truncate increment up to a blocksize, w/min of 1 block.
-    increment = ((increment / buf.st_blksize) + 1) * buf.st_blksize;
-
     while (fsize > 0)
     {
         //VERBOSE(VB_FILE, QString("Truncating '%1' to %2 MB")
@@ -1810,7 +1836,7 @@
         pbssock = pbs->getSocket();
 
     QString fileprefix = gContext->GetFilePrefix();
-    QString filename = pginfo->GetRecordFilename(fileprefix);
+    QString filename = pginfo->GetRecordFilename(fileprefix, true);
 
     // If this recording was made by a another recorder, and that
     // recorder is available, tell it to do the deletion.
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythbackend/mainserver.h mythtv-0.20-new/programs/mythbackend/mainserver.h
--- mythtv-0.20-old/programs/mythbackend/mainserver.h	2006-07-03 11:31:35.000000000 -0700
+++ mythtv-0.20-new/programs/mythbackend/mainserver.h	2006-10-28 19:17:11.000000000 -0700
@@ -152,7 +152,8 @@
     static int  DeleteFile(const QString &filename, bool followLinks);
     static int  OpenAndUnlink(const QString &filename);
     static bool TruncateAndClose(const AutoExpire *expirer,
-                                 int fd, const QString &filename);
+                                 int fd, const QString &filename,
+                                 off_t fsize);
 
     QPtrList<LiveTVChain> liveTVChains;
     QMutex liveTVChainsLock;
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythbackend/scheduler.cpp mythtv-0.20-new/programs/mythbackend/scheduler.cpp
--- mythtv-0.20-old/programs/mythbackend/scheduler.cpp	2006-09-08 21:20:38.000000000 -0700
+++ mythtv-0.20-new/programs/mythbackend/scheduler.cpp	2006-10-28 19:17:11.000000000 -0700
@@ -1384,6 +1384,14 @@
 
             VERBOSE(VB_GENERAL, msg << ": " << details);
             gContext->LogEntry("scheduler", LP_NOTICE, msg, details);
+
+            if (nextRecording->recstatus == rsFailed)
+            {
+                MythEvent me(QString("FORCE_DELETE_RECORDING %1 %2")
+                         .arg(nextRecording->chanid)
+                         .arg(nextRecording->recstartts.toString(Qt::ISODate)));
+                gContext->dispatch(me);
+            }
         }
 
         if (statuschanged)
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythfilldatabase/filldata.cpp mythtv-0.20-new/programs/mythfilldatabase/filldata.cpp
--- mythtv-0.20-old/programs/mythfilldatabase/filldata.cpp	2006-09-04 19:54:51.000000000 -0700
+++ mythtv-0.20-new/programs/mythfilldatabase/filldata.cpp	2006-10-28 19:17:11.000000000 -0700
@@ -2782,24 +2782,6 @@
     query.bindValue(":OFFSET", offset);
     query.exec();
 
-    query.prepare("DELETE p FROM recordedprogram p "
-                  "LEFT JOIN recorded r ON "
-                  "  p.chanid = r.chanid AND p.starttime = r.progstart "
-                  "WHERE r.chanid IS NULL;");
-    query.exec();
-
-    query.prepare("DELETE p FROM recordedrating p "
-                  "LEFT JOIN recorded r ON "
-                  "  p.chanid = r.chanid AND p.starttime = r.progstart "
-                  "WHERE r.chanid IS NULL;");
-    query.exec();
-
-    query.prepare("DELETE p FROM recordedcredits p "
-                  "LEFT JOIN recorded r ON "
-                  "  p.chanid = r.chanid AND p.starttime = r.progstart "
-                  "WHERE r.chanid IS NULL;");
-    query.exec();
-
     query.prepare("DELETE FROM record WHERE (type = :SINGLE "
                   "OR type = :OVERRIDE OR type = :DONTRECORD) "
                   "AND enddate < NOW();");
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythtranscode/mpeg2fix.cpp mythtv-0.20-new/programs/mythtranscode/mpeg2fix.cpp
--- mythtv-0.20-old/programs/mythtranscode/mpeg2fix.cpp	2006-07-18 17:24:00.000000000 -0700
+++ mythtv-0.20-new/programs/mythtranscode/mpeg2fix.cpp	2006-10-28 19:17:10.000000000 -0700
@@ -726,7 +726,8 @@
         {
 
             case CODEC_TYPE_VIDEO:
-                vid_id = i;
+                if(vid_id == -1)
+                    vid_id = i;
                 break;
 
             case CODEC_TYPE_AUDIO:
@@ -1134,6 +1135,12 @@
             if (ret < 0)
             {
                 //insert a bogus frame (this won't be written out)
+                if(vFrame.isEmpty())
+                {
+                    VERBOSE(MPF_IMPORTANT, "Found end of file without finding "
+                                           " any frames");
+                    return TRANSCODE_EXIT_UNKNOWN_ERROR;
+                }
                 MPEG2frame *tmpFrame = GetPoolFrame(&vFrame.last()->pkt);
                 if (tmpFrame == NULL)
                     return TRANSCODE_EXIT_UNKNOWN_ERROR;
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythtranscode/replex/element.c mythtv-0.20-new/programs/mythtranscode/replex/element.c
--- mythtv-0.20-old/programs/mythtranscode/replex/element.c	2006-08-28 10:09:05.000000000 -0700
+++ mythtv-0.20-new/programs/mythtranscode/replex/element.c	2006-10-28 19:17:09.000000000 -0700
@@ -480,7 +480,7 @@
 {
 	int c = 0;
 	int fr =0;
-	uint8_t headr[4];
+	uint8_t headr[7];
 
 	af->set=0;
 
@@ -529,7 +529,7 @@
 int get_ac3_info(ringbuffer *rbuf, audio_frame_t *af, int off, int le)
 {
 	int c=0;
-	uint8_t headr[6];
+	uint8_t headr[7];
 	uint8_t frame;
 	int half = 0;
 	int fr;
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythtranscode/replex/replex.c mythtv-0.20-new/programs/mythtranscode/replex/replex.c
--- mythtv-0.20-old/programs/mythtranscode/replex/replex.c	2006-08-28 10:09:05.000000000 -0700
+++ mythtv-0.20-new/programs/mythtranscode/replex/replex.c	2006-10-28 19:17:09.000000000 -0700
@@ -294,10 +294,13 @@
 						// add each extra frame required direct to the output file
 						int x;
 						for (x = 0; x < framesdiff; x++){
-							if (type == AC3)
-								write(rx->dmx_out[num+1+rx->apidn], framebuf, aframe->framesize);
-							else
-								write(rx->dmx_out[num+1], framebuf, aframe->framesize);
+							if (type == AC3){
+								if (rx->dmx_out[num+1+rx->apidn])	
+									write(rx->dmx_out[num+1+rx->apidn], framebuf, aframe->framesize);
+							}else{
+								if (rx->dmx_out[num+1])
+									write(rx->dmx_out[num+1], framebuf, aframe->framesize);
+							}
 							*acount += 1;
 						}
 						
diff -Naur -x mytharchive mythtv-0.20-old/programs/mythwelcome/welcomedialog.cpp mythtv-0.20-new/programs/mythwelcome/welcomedialog.cpp
--- mythtv-0.20-old/programs/mythwelcome/welcomedialog.cpp	2006-07-25 01:40:11.000000000 -0700
+++ mythtv-0.20-new/programs/mythwelcome/welcomedialog.cpp	2006-10-28 19:17:11.000000000 -0700
@@ -465,7 +465,7 @@
     QStringList strlist;
     
     // get list of current recordings
-    QString querytext = QString("SELECT cardid FROM capturecard;");
+    QString querytext = QString("SELECT cardid FROM capturecard WHERE parentid = 0;");
     MSqlQuery query(MSqlQuery::InitCon());
     query.exec(querytext);
     QString Status = "";
