diff -Naur NVIDIA-Linux-x86-173.08-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-173.08-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-173.08-pkg1-old/usr/src/nv/conftest.sh	2008-04-02 01:17:56.000000000 -0700
+++ NVIDIA-Linux-x86-173.08-pkg1-new/usr/src/nv/conftest.sh	2008-04-26 16:05:08.000000000 -0700
@@ -962,6 +962,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "$CONFTEST_PREAMBLE
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
