diff -Naur mythtv-0.20.2-old/libs/libavcodec/libavcodec.pro mythtv-0.20.2-new/libs/libavcodec/libavcodec.pro
--- mythtv-0.20.2-old/libs/libavcodec/libavcodec.pro	2008-01-18 10:11:22.000000000 -0800
+++ mythtv-0.20.2-new/libs/libavcodec/libavcodec.pro	2008-01-18 10:12:09.000000000 -0800
@@ -366,11 +366,13 @@
 using_xvmc {
     SOURCES += xvmcvideo.c
     DEFINES += HAVE_XVMC
+    LIBS += $$CONFIG_XVMC_LIBS
 }
 
 using_xvmc_vld {
     SOURCES += xvmcvldvideo.c
     DEFINES += HAVE_XVMC_VLD
+    LIBS += $$CONFIG_XVMC_LIBS
 }
 
 contains( AMR_NB, yes) {
