diff -Naur mythtv-0.20-16082-old/libs/libavcodec/libavcodec.pro mythtv-0.20-16082-new/libs/libavcodec/libavcodec.pro
--- mythtv-0.20-16082-old/libs/libavcodec/libavcodec.pro	2008-09-15 14:08:07.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libavcodec/libavcodec.pro	2008-09-15 14:08:08.000000000 -0700
@@ -366,11 +366,13 @@
 using_xvmc {
     SOURCES += xvmcvideo.c
     DEFINES += HAVE_XVMC
+    LIBS += $$CONFIG_XVMC_LIBS
 }
 
 using_xvmc_vld {
     SOURCES += xvmcvldvideo.c
     DEFINES += HAVE_XVMC_VLD
+    LIBS += $$CONFIG_XVMC_LIBS
 }
 
 contains( AMR_NB, yes) {
