diff -Naur mythtv-0.20-16082-old/libs/libmyth/mythmediamonitor.cpp mythtv-0.20-16082-new/libs/libmyth/mythmediamonitor.cpp
--- mythtv-0.20-16082-old/libs/libmyth/mythmediamonitor.cpp	2008-09-15 14:05:01.000000000 -0700
+++ mythtv-0.20-16082-new/libs/libmyth/mythmediamonitor.cpp	2008-09-16 19:33:10.000000000 -0700
@@ -320,7 +320,7 @@
 {
 #ifdef linux
     QProcess udevinfo(this);
-    udevinfo.addArgument("udevinfo");
+    udevinfo.addArgument("udevadm info");
     udevinfo.addArgument("-q");
     udevinfo.addArgument("name");
     udevinfo.addArgument("-rp");
