diff -Naur mythtv-trunk-20209-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-20209-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-20209-old/libs/libmyth/mediamonitor-unix.cpp	2009-03-14 11:10:27.000000000 -0700
+++ mythtv-trunk-20209-new/libs/libmyth/mediamonitor-unix.cpp	2009-03-14 11:10:44.000000000 -0700
@@ -221,11 +221,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
