diff -Naur mythtv-trunk-17721-old/libs/libmythtv/videoout_xv.cpp mythtv-trunk-17721-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-trunk-17721-old/libs/libmythtv/videoout_xv.cpp	2008-07-06 07:41:10.000000000 -0700
+++ mythtv-trunk-17721-new/libs/libmythtv/videoout_xv.cpp	2008-07-06 07:44:19.000000000 -0700
@@ -1549,7 +1549,12 @@
         for (; it != renderers.end(); ++it)
             tmp += *it + ",";
 
-        renderer = renderers[0];
+        if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-blit") && (renderers.contains("xv-blit")))
+            renderer = "xv-blit";
+        else if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-opengl") && (renderers.contains("opengl")))
+            renderer = "opengl";
+        else
+            renderer = renderers[0];
         VERBOSE(VB_IMPORTANT, LOC + QString(
                     "Desired video renderer '%1' not available.\n\t\t\t"
                     "codec '%2' makes '%3' available, using '%4' instead.")
