diff -Naur mythtv-trunk-20728-old/configure mythtv-trunk-20728-new/configure
--- mythtv-trunk-20728-old/configure	2009-06-20 09:21:11.000000000 -0700
+++ mythtv-trunk-20728-new/configure	2009-06-20 09:21:14.000000000 -0700
@@ -3009,7 +3009,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
