diff -Naur mythtv-trunk-18515-old/configure mythtv-trunk-18515-new/configure
--- mythtv-trunk-18515-old/configure	2008-09-29 16:32:20.000000000 -0700
+++ mythtv-trunk-18515-new/configure	2008-09-29 16:37:08.000000000 -0700
@@ -2710,7 +2710,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
