diff -Naur mythtv-trunk-20768-old/libs/libmythtv/vsync.cpp mythtv-trunk-20768-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20768-old/libs/libmythtv/vsync.cpp	2009-06-30 16:19:16.000000000 -0700
+++ mythtv-trunk-20768-new/libs/libmythtv/vsync.cpp	2009-06-30 16:19:16.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
