diff -Naur mythtv-trunk-18943-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-18943-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-18943-old/libs/libmyth/mediamonitor-unix.cpp	2008-10-30 06:49:36.000000000 -0700
+++ mythtv-trunk-18943-new/libs/libmyth/mediamonitor-unix.cpp	2008-10-30 06:49:55.000000000 -0700
@@ -204,11 +204,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
