diff -Naur mythtv-trunk-19442-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-19442-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-19442-old/libs/libmyth/mediamonitor-unix.cpp	2008-12-23 20:10:58.000000000 -0800
+++ mythtv-trunk-19442-new/libs/libmyth/mediamonitor-unix.cpp	2008-12-23 20:10:59.000000000 -0800
@@ -204,11 +204,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
