diff -Naur mythtv-trunk-22301-old/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh mythtv-trunk-22301-new/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh
--- mythtv-trunk-22301-old/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh	2009-10-07 18:10:31.000000000 -0700
+++ mythtv-trunk-22301-new/contrib/channel_changers/change-channel-lirc/dish-change-channel-lirc.sh	2009-10-08 08:46:09.000000000 -0700
@@ -4,13 +4,13 @@
 # wishes to use it. It has been used by me for many months with the
 # VIP211 receivers. -- Daniel Kristjansson (March 20th, 2008)
 
-# If you have a separate IR reciever you will want to give it "/dev/lircd"
-# and use something like "/dev/lircd1" for the transmitter.
+# If you have a separate IR reciever you will want to give it "/var/run/lirc/lircd"
+# and use something like "/var/run/lirc/lircd1" for the transmitter.
 # If the tranmitter is on a remote LIRC server, use the ip/hostname:port
 # with -a to address it.
 
-DEVICE="-d /dev/lircd"
-#DEVICE="-d /dev/lircd1"
+DEVICE="-d /var/run/lirc/lircd"
+#DEVICE="-d /var/run/lirc/lircd1"
 #DEVICE="-a hostname:port"
 
 # Use dish1, dish2 .. dish16 depending on which id you are using
diff -Naur mythtv-trunk-22301-old/docs/mythtv-HOWTO-8.html mythtv-trunk-22301-new/docs/mythtv-HOWTO-8.html
--- mythtv-trunk-22301-old/docs/mythtv-HOWTO-8.html	2009-10-07 18:10:18.000000000 -0700
+++ mythtv-trunk-22301-new/docs/mythtv-HOWTO-8.html	2009-10-08 08:47:33.000000000 -0700
@@ -85,7 +85,7 @@
 $ make
 $ su
 # make install
-# chmod 666 /dev/lircd
+# chmod 666 /var/run/lirc/lircd
 </PRE>
 </CODE></BLOCKQUOTE>
 </P>
@@ -95,8 +95,6 @@
 <PRE>
 $ ls -l /dev/li*
 lr-xr-xr-x    1 root     root            5 Jan 27 09:00 /dev/lirc -> ttyS0
-srw-rw-rw-    1 root     root            0 Jan 27 15:01 /dev/lircd=
-prw-r--r--    1 root     root            0 Jan 27 09:00 /dev/lircm|
 </PRE>
 </CODE></BLOCKQUOTE>
 </P>
diff -Naur mythtv-trunk-22301-old/docs/mythtv-HOWTO.sgml mythtv-trunk-22301-new/docs/mythtv-HOWTO.sgml
--- mythtv-trunk-22301-old/docs/mythtv-HOWTO.sgml	2009-10-07 18:10:18.000000000 -0700
+++ mythtv-trunk-22301-new/docs/mythtv-HOWTO.sgml	2009-10-08 08:47:11.000000000 -0700
@@ -1572,7 +1572,7 @@
 $ make
 $ su
 # make install
-# chmod 666 /dev/lircd
+# chmod 666 /var/run/lirc/lircd
 </verb></tscreen>
 
 At this point, if you're using a serial receiver, check that there's a
@@ -1580,8 +1580,6 @@
 <tscreen><verb>
 $ ls -l /dev/li*
 lr-xr-xr-x    1 root     root            5 Jan 27 09:00 /dev/lirc -> ttyS0
-srw-rw-rw-    1 root     root            0 Jan 27 15:01 /dev/lircd=
-prw-r--r--    1 root     root            0 Jan 27 09:00 /dev/lircm|
 </verb></tscreen>
 
 As you can see, there's a link from /dev/lirc to ttyS0, a.k.a. "COM1", which is
diff -Naur mythtv-trunk-22301-old/docs/mythtv-HOWTO-singlehtml.html mythtv-trunk-22301-new/docs/mythtv-HOWTO-singlehtml.html
--- mythtv-trunk-22301-old/docs/mythtv-HOWTO-singlehtml.html	2009-10-07 18:10:18.000000000 -0700
+++ mythtv-trunk-22301-new/docs/mythtv-HOWTO-singlehtml.html	2009-10-08 08:47:21.000000000 -0700
@@ -2130,7 +2130,7 @@
 $ make
 $ su
 # make install
-# chmod 666 /dev/lircd
+# chmod 666 /var/run/lirc/lircd
 </PRE>
 </CODE></BLOCKQUOTE>
 </P>
@@ -2140,8 +2140,6 @@
 <PRE>
 $ ls -l /dev/li*
 lr-xr-xr-x    1 root     root            5 Jan 27 09:00 /dev/lirc -> ttyS0
-srw-rw-rw-    1 root     root            0 Jan 27 15:01 /dev/lircd=
-prw-r--r--    1 root     root            0 Jan 27 09:00 /dev/lircm|
 </PRE>
 </CODE></BLOCKQUOTE>
 </P>
diff -Naur mythtv-trunk-22301-old/docs/mythtv-HOWTO.txt mythtv-trunk-22301-new/docs/mythtv-HOWTO.txt
--- mythtv-trunk-22301-old/docs/mythtv-HOWTO.txt	2009-10-07 18:10:18.000000000 -0700
+++ mythtv-trunk-22301-new/docs/mythtv-HOWTO.txt	2009-10-08 08:48:48.000000000 -0700
@@ -2336,7 +2336,7 @@
        $ make
        $ su
        # make install
-       # chmod 666 /dev/lircd
+       # chmod 666 /var/run/lirc/lircd
 
 
 
@@ -2346,8 +2346,6 @@
 
        $ ls -l /dev/li*
        lr-xr-xr-x    1 root     root            5 Jan 27 09:00 /dev/lirc -> ttyS0
-       srw-rw-rw-    1 root     root            0 Jan 27 15:01 /dev/lircd=
-       prw-r--r--    1 root     root            0 Jan 27 09:00 /dev/lircm|
 
 
 
diff -Naur mythtv-trunk-22301-old/libs/libmythui/mythmainwindow.cpp mythtv-trunk-22301-new/libs/libmythui/mythmainwindow.cpp
--- mythtv-trunk-22301-old/libs/libmythui/mythmainwindow.cpp	2009-10-07 18:10:27.000000000 -0700
+++ mythtv-trunk-22301-new/libs/libmythui/mythmainwindow.cpp	2009-10-08 08:47:54.000000000 -0700
@@ -1855,7 +1855,7 @@
 
     d->lircThread = new LIRC(
         this,
-        GetMythDB()->GetSetting("LircSocket", "/dev/lircd"),
+        GetMythDB()->GetSetting("LircSocket", "/var/run/lircd"),
         "mythtv", config_file,
         GetMythDB()->GetSetting("LircKeyPressedApp", ""));
 
diff -Naur mythtv-trunk-22301-old/programs/mythfrontend/globalsettings.cpp mythtv-trunk-22301-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-trunk-22301-old/programs/mythfrontend/globalsettings.cpp	2009-10-08 08:49:44.000000000 -0700
+++ mythtv-trunk-22301-new/programs/mythfrontend/globalsettings.cpp	2009-10-08 08:48:04.000000000 -0700
@@ -2281,7 +2281,7 @@
 {
     HostLineEdit *ge = new HostLineEdit("LircSocket");
     ge->setLabel(QObject::tr("LIRC Daemon Socket"));
-    ge->setValue("/dev/lircd");
+    ge->setValue("/var/run/lirc/lircd");
     QString help = QObject::tr(
         "UNIX socket or IP address[:port] to connect in "
         "order to communicate with the LIRC Daemon.");
