diff -Naur mythtv-trunk-18885-old/configure mythtv-trunk-18885-new/configure
--- mythtv-trunk-18885-old/configure	2008-10-26 17:03:36.000000000 -0700
+++ mythtv-trunk-18885-new/configure	2008-10-26 17:06:35.000000000 -0700
@@ -2736,7 +2736,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
