diff -Naur mythtv-trunk-19310-old/programs/mythfrontend/globalsettings.cpp mythtv-trunk-19310-new/programs/mythfrontend/globalsettings.cpp
--- mythtv-trunk-19310-old/programs/mythfrontend/globalsettings.cpp	2008-12-09 21:09:34.000000000 -0800
+++ mythtv-trunk-19310-new/programs/mythfrontend/globalsettings.cpp	2008-12-09 21:09:35.000000000 -0800
@@ -46,7 +46,7 @@
     gc->setLabel(QObject::tr("Audio output device"));
 
 #ifdef USING_ALSA
-    gc->addSelection("ALSA:default",       "ALSA:default");
+    gc->addSelection("ALSA:default",       "ALSA:default", true);
     gc->addSelection("ALSA:spdif",         "ALSA:spdif");
     gc->addSelection("ALSA:surround51",    "ALSA:surround51");
     gc->addSelection("ALSA:analog",        "ALSA:analog");
@@ -140,7 +140,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("MythControlsVolume");
     gc->setLabel(QObject::tr("Use internal volume controls"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("MythTV can control the PCM and master "
                     "mixer volume.  If you prefer to use an external mixer "
                     "program, then disable this option."));
@@ -163,7 +163,7 @@
     }
 #endif
 #ifdef USING_ALSA
-    gc->addSelection("ALSA:default", "ALSA:default");
+    gc->addSelection("ALSA:default", "ALSA:default", true);
 #endif
 #ifdef USING_DIRECTX
     gc->addSelection("DirectX:", "DirectX:");
@@ -1523,7 +1523,7 @@
     gc->setLabel(QObject::tr("Sort Episodes"));
     gc->addSelection(QObject::tr("Record date"), "Date");
     gc->addSelection(QObject::tr("Original Air date"), "OrigAirDate");
-    gc->addSelection(QObject::tr("Program ID"), "Id");
+    gc->addSelection(QObject::tr("Program ID"), "Id", true);
     gc->setHelpText(QObject::tr("Selects how to sort a shows episodes"));
     return gc;
 }
@@ -1532,7 +1532,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("StickyKeys");
     gc->setLabel(QObject::tr("Sticky keys"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If enabled, fast forward and rewind "
                     "continue after the key is released.  Pressing the key "
                     "again increases the fast forward or rewind speed.  The "
@@ -2059,7 +2059,7 @@
     HostComboBox *gc = new HostComboBox("PlaybackExitPrompt");
     gc->setLabel(QObject::tr("Action on playback exit"));
     gc->addSelection(QObject::tr("Just exit"), "0");
-    gc->addSelection(QObject::tr("Save position and exit"), "2");
+    gc->addSelection(QObject::tr("Save position and exit"), "2", true);
     gc->addSelection(QObject::tr("Always prompt (excluding Live TV)"), "1");
     gc->addSelection(QObject::tr("Always prompt (including Live TV)"), "4");
     gc->addSelection(QObject::tr("Prompt for Live TV only"), "8");
@@ -2074,7 +2074,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("EndOfRecordingExitPrompt");
     gc->setLabel(QObject::tr("Prompt at end of recording"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If set, a menu will be displayed allowing "
                     "you to delete the recording when it has finished "
                     "playing."));
@@ -2162,7 +2162,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("PlaybackPreview");
     gc->setLabel(QObject::tr("Display live preview of recordings"));
-    gc->setValue(true);
+    gc->setValue(false);
     gc->setHelpText(QObject::tr("When enabled, a preview of the recording "
                     "will play in a small window on the \"Watch a "
                     "Recording\" menu."));
@@ -2211,7 +2211,7 @@
     HostComboBox *gc = new HostComboBox("AllowQuitShutdown");
     gc->setLabel(QObject::tr("System Exit key"));
     gc->addSelection(QObject::tr("ESC"), "4");
-    gc->addSelection(QObject::tr("No exit key"), "0");
+    gc->addSelection(QObject::tr("No exit key"), "0", true);
     gc->addSelection(QObject::tr("Control-ESC"), "1");
     gc->addSelection(QObject::tr("Meta-ESC"), "2");
     gc->addSelection(QObject::tr("Alt-ESC"), "3");
@@ -2266,7 +2266,7 @@
 {
     HostLineEdit *ge = new HostLineEdit("HaltCommand");
     ge->setLabel(QObject::tr("Halt command"));
-    ge->setValue("halt");
+    ge->setValue("poweroff");
     ge->setHelpText(QObject::tr("Command or script to run if you select "
                     "the shutdown option from the exit menu, if the option "
                     "is displayed. You must configure an exit key to "
@@ -2979,7 +2979,7 @@
     GlobalCheckBox *bc = new GlobalCheckBox("LastFreeCard");
     bc->setLabel(QObject::tr("Avoid conflicts between live TV and "
                  "scheduled shows"));
-    bc->setValue(false);
+    bc->setValue(true);
     bc->setHelpText(QObject::tr("If enabled, live TV will choose a tuner card "
                     "that is less likely to have scheduled recordings "
                     "rather than the best card available."));
@@ -3056,7 +3056,8 @@
     gc->addSelection(QObject::tr("Blender - Transparent (CPU Usage - Middle)"),
                      QString::number((int)UIGuideType::Dense));
     gc->addSelection(QObject::tr("Eco - Transparent (CPU Usage - Low)"),
-                     QString::number((int)UIGuideType::Eco));
+                     QString::number((int)UIGuideType::Eco),
+                     true);
     gc->addSelection(QObject::tr("Solid (CPU Usage - Middle)"),
                      QString::number((int)UIGuideType::Solid));
     return gc;
