diff -Naur mythtv-trunk-20743-old/configure mythtv-trunk-20743-new/configure
--- mythtv-trunk-20743-old/configure	2009-06-24 08:55:13.000000000 -0700
+++ mythtv-trunk-20743-new/configure	2009-06-24 08:55:18.000000000 -0700
@@ -1635,7 +1635,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
