diff -Naur mythtv-trunk-20007-old/configure mythtv-trunk-20007-new/configure
--- mythtv-trunk-20007-old/configure	2009-02-16 11:01:25.000000000 -0800
+++ mythtv-trunk-20007-new/configure	2009-02-16 11:01:27.000000000 -0800
@@ -1633,7 +1633,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
