diff -Naur mythtv-trunk-22939-old/configure mythtv-trunk-22939-new/configure
--- mythtv-trunk-22939-old/configure	2009-11-30 21:47:24.000000000 -0800
+++ mythtv-trunk-22939-new/configure	2009-11-30 21:47:24.000000000 -0800
@@ -3474,7 +3474,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && check_lib X11/Xlib.h XQueryExtension -lX11 || disable x11
