diff -Naur mythtv-trunk-21859-old/libs/libmythtv/vsync.cpp mythtv-trunk-21859-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-21859-old/libs/libmythtv/vsync.cpp	2009-09-14 16:42:10.000000000 -0700
+++ mythtv-trunk-21859-new/libs/libmythtv/vsync.cpp	2009-09-14 16:42:11.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
