diff -Naur mythtv-trunk-22939-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-22939-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-22939-old/libs/libmyth/mediamonitor-unix.cpp	2009-11-30 21:47:26.000000000 -0800
+++ mythtv-trunk-22939-new/libs/libmyth/mediamonitor-unix.cpp	2009-11-30 21:47:27.000000000 -0800
@@ -222,11 +222,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
