diff -Naur mythtv-trunk-16813-old/libs/libmythtv/tv_play.cpp mythtv-trunk-16813-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-16813-old/libs/libmythtv/tv_play.cpp	2008-03-26 20:07:13.000000000 -0700
+++ mythtv-trunk-16813-new/libs/libmythtv/tv_play.cpp	2008-03-26 20:07:18.000000000 -0700
@@ -1818,7 +1818,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
