diff -Naur mythtv-trunk-20107-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-20107-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-20107-old/libs/libmyth/mediamonitor-unix.cpp	2009-03-04 19:06:28.000000000 -0800
+++ mythtv-trunk-20107-new/libs/libmyth/mediamonitor-unix.cpp	2009-03-04 19:06:30.000000000 -0800
@@ -221,11 +221,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
