diff -Naur mythtv-trunk-16487-old/libs/libmythtv/videoout_xv.cpp mythtv-trunk-16487-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-trunk-16487-old/libs/libmythtv/videoout_xv.cpp	2008-03-09 23:18:25.000000000 -0700
+++ mythtv-trunk-16487-new/libs/libmythtv/videoout_xv.cpp	2008-03-09 23:24:14.000000000 -0700
@@ -1521,7 +1521,10 @@
         for (; it != renderers.end(); ++it)
             tmp += *it + ",";
 
-        renderer = renderers[0];
+    	if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-blit") && (renderers.contains("xv-blit")))
+            renderer = "xv-blit" ;
+        else
+            renderer = renderers[0];
         VERBOSE(VB_IMPORTANT, LOC + QString(
                     "Desired video renderer '%1' not available.\n\t\t\t"
                     "codec '%2' makes '%3' available, using '%4' instead.")
