diff -Naur mythtv-trunk-25233-old/bindings/python/python.pro mythtv-trunk-25233-new/bindings/python/python.pro
--- mythtv-trunk-25233-old/bindings/python/python.pro	2010-06-30 19:20:34.000000000 -0700
+++ mythtv-trunk-25233-new/bindings/python/python.pro	2010-06-30 19:22:20.000000000 -0700
@@ -2,14 +2,14 @@
 
 python_build.target = python_build
 python_build.depends = setup.py FORCE
-python_build.commands = $$PYTHON setup.py build
+python_build.commands = $$PYTHON setup.py build --cross-compile
 
 QMAKE_EXTRA_TARGETS += python_build
 PRE_TARGETDEPS += python_build
 
 python_install.target = python_install
 python_install.CONFIG = no_path
-python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\"
+python_install.commands = $$PYTHON setup.py install --skip-build --prefix=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)$${PREFIX}\"
 
 INSTALLS += python_install
 
diff -Naur mythtv-trunk-25233-old/bindings/python/setup.py mythtv-trunk-25233-new/bindings/python/setup.py
--- mythtv-trunk-25233-old/bindings/python/setup.py	2010-06-30 19:20:34.000000000 -0700
+++ mythtv-trunk-25233-new/bindings/python/setup.py	2010-06-30 19:22:20.000000000 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
+from setuptools import setup
 from distutils.cmd import Command
 from distutils.sysconfig import get_python_lib
 from distutils.command.install import INSTALL_SCHEMES
diff -Naur mythtv-trunk-25233-old/configure mythtv-trunk-25233-new/configure
--- mythtv-trunk-25233-old/configure	2010-06-30 19:20:34.000000000 -0700
+++ mythtv-trunk-25233-new/configure	2010-06-30 19:22:20.000000000 -0700
@@ -3949,10 +3949,10 @@
 fi
 
 # Check for python dependencies
-if enabled bindings_python; then
-    check_py_lib MySQLdb || disable_bindings_python "MySQLdb"
-    check_py_lib lxml    || disable_bindings_python "lxml"
-fi
+#if enabled bindings_python; then
+#    check_py_lib MySQLdb || disable_bindings_python "MySQLdb"
+#    check_py_lib lxml    || disable_bindings_python "lxml"
+#fi
 
 # Check for perl dependencies
 # Modules distributed in Perl core
