diff -Naur mythtv-trunk-16318-old/themes/cpsvndir mythtv-trunk-16318-new/themes/cpsvndir
--- mythtv-trunk-16318-old/themes/cpsvndir	2008-02-29 21:44:00.000000000 -0800
+++ mythtv-trunk-16318-new/themes/cpsvndir	2008-02-29 21:44:44.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
