diff -Naur mythtv-trunk-27219-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-27219-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-27219-old/libs/libmythtv/avformatdecoder.cpp	2010-11-14 15:09:50.000000000 -0800
+++ mythtv-trunk-27219-new/libs/libmythtv/avformatdecoder.cpp	2010-11-14 15:11:59.000000000 -0800
@@ -1257,6 +1257,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
