diff -Naur mythtv-trunk-20445-old/libs/libavformat/mpeg.c mythtv-trunk-20445-new/libs/libavformat/mpeg.c
--- mythtv-trunk-20445-old/libs/libavformat/mpeg.c	2009-04-23 22:12:42.000000000 -0700
+++ mythtv-trunk-20445-new/libs/libavformat/mpeg.c	2009-04-25 07:58:48.000000000 -0700
@@ -95,7 +95,7 @@
         return AVPROBE_SCORE_MAX/2+2; // +1 for .mpg
     if(priv1 + vid + audio > invalid && (priv1+vid+audio)*9 <= pspack*10)
         return AVPROBE_SCORE_MAX/2+2; // +1 for .mpg
-    if((!!vid ^ !!audio) && (audio+vid > 1) && !sys && !pspack && p->buf_size>2048) /* PES stream */
+    if((!!vid ^ !!audio) && (audio > 4 || vid > 1) && !sys && !pspack && p->buf_size>2048) /* PES stream */
         return AVPROBE_SCORE_MAX/2+2;
 
     //02-Penguin.flac has sys:0 priv1:0 pspack:0 vid:0 audio:1
