diff -Naur mythtv-trunk-26683-old/configure mythtv-trunk-26683-new/configure
--- mythtv-trunk-26683-old/configure	2010-10-06 16:51:17.000000000 -0700
+++ mythtv-trunk-26683-new/configure	2010-10-06 16:51:18.000000000 -0700
@@ -4739,8 +4739,8 @@
 #       -L from early in the link line, to ignore any old installed libmyth*
 # Note: Qt uses QMAKE_INCDIR_QT to build an absolute path for include
 #       directories, it cannot be modified.
-QMAKE_LIBDIR_QT-=$libdir
-LATE_LIBS+=-L$libdir
+QMAKE_LIBDIR_QT-=${sysroot}$libdir
+LATE_LIBS+=-L${sysroot}$libdir
 EOF
 
 #echo "endif # FFMPEG_CONFIG_MAK" >> config.mak
