diff -Naur mythtv-trunk-20127-old/libs/libmythdvdnav/libmythdvdnav.pro mythtv-trunk-20127-new/libs/libmythdvdnav/libmythdvdnav.pro
--- mythtv-trunk-20127-old/libs/libmythdvdnav/libmythdvdnav.pro	2009-03-06 18:05:38.000000000 -0800
+++ mythtv-trunk-20127-new/libs/libmythdvdnav/libmythdvdnav.pro	2009-03-06 18:09:18.000000000 -0800
@@ -3,7 +3,7 @@
 
 TEMPLATE = lib
 TARGET = mythdvdnav-$$LIBVERSION
-CONFIG += thread staticlib warn_off
+CONFIG += thread dll warn_off
 CONFIG -= qt
 target.path = $${LIBDIR}
 
diff -Naur mythtv-trunk-20127-old/libs/libmythtv/libmythtv.pro mythtv-trunk-20127-new/libs/libmythtv/libmythtv.pro
--- mythtv-trunk-20127-old/libs/libmythtv/libmythtv.pro	2009-03-06 18:04:41.000000000 -0800
+++ mythtv-trunk-20127-new/libs/libmythtv/libmythtv.pro	2009-03-06 18:09:18.000000000 -0800
@@ -63,7 +63,7 @@
 TARGETDEPS += ../libavcodec/libmythavcodec-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../libavformat/libmythavformat-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../libmythmpeg2/libmythmpeg2-$${MYTH_LIB_EXT}
-TARGETDEPS += ../libmythdvdnav/libmythdvdnav-$${MYTH_LIB_EXT}
+TARGETDEPS += ../libmythdvdnav/libmythdvdnav-$${MYTH_SHLIB_EXT}
 using_mheg: TARGETDEPS += ../libmythfreemheg/libmythfreemheg-$${MYTH_SHLIB_EXT}
 using_live: TARGETDEPS += ../libmythlivemedia/libmythlivemedia-$${MYTH_SHLIB_EXT}
 using_hdhomerun: TARGETDEPS += ../libmythhdhomerun/libmythhdhomerun-$${MYTH_SHLIB_EXT}
diff -Naur mythtv-trunk-20127-old/programs/programs-libs.pro mythtv-trunk-20127-new/programs/programs-libs.pro
--- mythtv-trunk-20127-old/programs/programs-libs.pro	2009-03-06 18:05:47.000000000 -0800
+++ mythtv-trunk-20127-new/programs/programs-libs.pro	2009-03-06 18:09:18.000000000 -0800
@@ -9,10 +9,12 @@
 LIBS += -L../../libs/libmythdb
 LIBS += -L../../libs/libmythui
 LIBS += -L../../libs/libmythupnp
+LIBS += -L../../libs/libmythdvdnav
 
 LIBS += -lmythtv-$$LIBVERSION -lmythavformat-$$LIBVERSION
 LIBS += -lmythavutil-$$LIBVERSION -lmythavcodec-$$LIBVERSION
 LIBS += -lmythupnp-$$LIBVERSION
+LIBS += -lmythdvdnav-$$LIBVERSION
 LIBS += -lmyth-$$LIBVERSION -lmythui-$$LIBVERSION
 LIBS += -lmythdb-$$LIBVERSION
 
@@ -35,6 +37,7 @@
 TARGETDEPS += ../../libs/libavcodec/libmythavcodec-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../../libs/libavformat/libmythavformat-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../../libs/libmythupnp/libmythupnp-$${MYTH_SHLIB_EXT}
+TARGETDEPS += ../../libs/libmythdvdnav/libmythdvdnav-$${MYTH_SHLIB_EXT}
 TARGETDEPS += ../../libs/libmythdb/libmythdb-$${MYTH_SHLIB_EXT}
 using_live: TARGETDEPS += ../../libs/libmythlivemedia/libmythlivemedia-$${MYTH_SHLIB_EXT}
 using_hdhomerun: TARGETDEPS += ../../libs/libmythhdhomerun/libmythhdhomerun-$${MYTH_SHLIB_EXT}
@@ -46,6 +49,7 @@
 DEPENDPATH += ../../libs/libavutil ../../libs/libavformat ../../libs/libavcodec
 DEPENDPATH += ../../libs/libswscale
 DEPENDPATH += ../../libs/libmythupnp ../../libs/libmythui
+DEPENDPATH += ../../libs/libmythdvdnav
 DEPENDPATH += ../../libs/libmythlivemedia ../../libmythdb ../../libmythhdhomerun
 
 CONFIG += opengl
