diff -Naur mythtv-trunk-26140-old/libs/libmythtv/util-xvmc.h mythtv-trunk-26140-new/libs/libmythtv/util-xvmc.h
--- mythtv-trunk-26140-old/libs/libmythtv/util-xvmc.h	2010-09-06 10:38:20.000000000 -0700
+++ mythtv-trunk-26140-new/libs/libmythtv/util-xvmc.h	2010-09-06 10:38:21.000000000 -0700
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(14),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
