diff -Naur mythtv-trunk-22825-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-22825-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-22825-old/libs/libmyth/mediamonitor-unix.cpp	2009-11-13 17:15:13.000000000 -0800
+++ mythtv-trunk-22825-new/libs/libmyth/mediamonitor-unix.cpp	2009-11-13 17:15:14.000000000 -0800
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
