diff -Naur mythtv-trunk-19356-old/configure mythtv-trunk-19356-new/configure
--- mythtv-trunk-19356-old/configure	2008-12-13 08:02:36.000000000 -0800
+++ mythtv-trunk-19356-new/configure	2008-12-13 08:02:40.000000000 -0800
@@ -1632,7 +1632,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
