diff -Naur mythtv-trunk-19357-old/configure mythtv-trunk-19357-new/configure
--- mythtv-trunk-19357-old/configure	2008-12-13 13:50:16.000000000 -0800
+++ mythtv-trunk-19357-new/configure	2008-12-13 13:50:17.000000000 -0800
@@ -2972,7 +2972,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
