diff -Naur mythtv-trunk-26882-old/libs/libmythdb/mythversion.h mythtv-trunk-26882-new/libs/libmythdb/mythversion.h
--- mythtv-trunk-26882-old/libs/libmythdb/mythversion.h	2010-10-18 10:06:23.000000000 -0700
+++ mythtv-trunk-26882-new/libs/libmythdb/mythversion.h	2010-10-18 10:11:03.000000000 -0700
@@ -11,7 +11,7 @@
 /// Update this whenever the plug-in API changes.
 /// Including changes in the libmythdb, libmyth, libmythtv, libmythav* and
 /// libmythui class methods used by plug-ins.
-#define MYTH_BINARY_VERSION "0.23.20101017-1"
+#define MYTH_BINARY_VERSION "0.24.20101017-1"
 
 /** \brief Increment this whenever the MythTV network protocol changes.
  *
