diff -Naur mythtv-trunk-26683-old/libs/libmythdb/mythversion.h mythtv-trunk-26683-new/libs/libmythdb/mythversion.h
--- mythtv-trunk-26683-old/libs/libmythdb/mythversion.h	2010-10-06 16:49:10.000000000 -0700
+++ mythtv-trunk-26683-new/libs/libmythdb/mythversion.h	2010-10-06 16:51:14.000000000 -0700
@@ -11,7 +11,7 @@
 /// Update this whenever the plug-in API changes.
 /// Including changes in the libmythdb, libmyth, libmythtv, libmythav* and
 /// libmythui class methods used by plug-ins.
-#define MYTH_BINARY_VERSION "0.23.20100917-1"
+#define MYTH_BINARY_VERSION "0.24.20100917-1"
 
 /** \brief Increment this whenever the MythTV network protocol changes.
  *
