diff -Naur mythtv-trunk-25255-old/configure mythtv-trunk-25255-new/configure
--- mythtv-trunk-25255-old/configure	2010-07-02 12:40:33.000000000 -0700
+++ mythtv-trunk-25255-new/configure	2010-07-02 12:40:34.000000000 -0700
@@ -4584,8 +4584,8 @@
 #       -L from early in the link line, to ignore any old installed libmyth*
 # Note: Qt uses QMAKE_INCDIR_QT to build an absolute path for include
 #       directories, it cannot be modified.
-QMAKE_LIBDIR_QT -= $libdir
-LATE_LIBS            += -L$libdir
+QMAKE_LIBDIR_QT -= ${sysroot}${libdir}
+LATE_LIBS            += -L${sysroot}${libdir}
 EOF
 
 diff $TMPH $MYTH_CONFIG_H >/dev/null 2>&1
