diff -Naur mythtv-trunk-18885-old/libs/libmythtv/tv_play.cpp mythtv-trunk-18885-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-18885-old/libs/libmythtv/tv_play.cpp	2008-10-26 17:06:51.000000000 -0700
+++ mythtv-trunk-18885-new/libs/libmythtv/tv_play.cpp	2008-10-26 17:06:54.000000000 -0700
@@ -1816,7 +1816,7 @@
     if (pthread_create(&decode, NULL, SpawnDecode, nvp))
         return false;
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
