diff -Naur mythtv-trunk-20209-old/programs/mythfrontend/main.cpp mythtv-trunk-20209-new/programs/mythfrontend/main.cpp
--- mythtv-trunk-20209-old/programs/mythfrontend/main.cpp	2009-03-14 11:10:07.000000000 -0700
+++ mythtv-trunk-20209-new/programs/mythfrontend/main.cpp	2009-03-14 11:10:15.000000000 -0700
@@ -1,10 +1,13 @@
 
 #include <unistd.h>
+#include <sys/socket.h>
+#include <netdb.h>
 #include <fcntl.h>
 #include <signal.h>
 #include <cerrno>
 #include <pthread.h>
 
+#include <cstring>
 #include <iostream>
 using namespace std;
 
@@ -1420,15 +1423,71 @@
         mainWindow->installEventFilter(mon);
     }
 
+    // Network control.
+    // Create a separate network control on each address enabling simultaneous local and remote control.
+    QList<NetworkControl *> networkControls;
     NetworkControl *networkControl = NULL;
+    QHostAddress networkAddress;
+    int networkPort = gContext->GetNumSetting("NetworkControlPort", 6545);
+    // Loopback address.
+    networkAddress.setAddress("127.0.0.1");
+    networkControl = new NetworkControl();
+    if (!networkControl->listen(networkAddress, networkPort))
+        VERBOSE(VB_IMPORTANT,
+                QString("NetworkControl failed to bind to port %1 of address %2.")
+                .arg(networkPort).arg(networkAddress.toString()));
+    networkControls.append(networkControl);
+    // Non-loopback address(es).
     if (gContext->GetNumSetting("NetworkControlEnabled", 0))
     {
-        int networkPort = gContext->GetNumSetting("NetworkControlPort", 6545);
-        networkControl = new NetworkControl();
-        if (!networkControl->listen(QHostAddress::Any,networkPort))
+        char hostname[256];
+        if (!gethostname(hostname,256))
+        {
+            struct addrinfo req;
+            struct addrinfo *res = NULL;
+            memset(&req, 0, sizeof(req));
+            req.ai_family = PF_UNSPEC;     // Request both IPv4 and IPv6 addresses.
+            req.ai_socktype = SOCK_STREAM; // Request only one socket type so that we get only one record per address.
+            if (!getaddrinfo(hostname, NULL, &req, &res))
+            {
+                struct addrinfo *cur = res;
+                while (cur != NULL)
+                {
+                    char address[INET_ADDRSTRLEN + INET6_ADDRSTRLEN];
+                    if (!getnameinfo(cur->ai_addr, cur->ai_addrlen,
+                                     address, INET_ADDRSTRLEN + INET6_ADDRSTRLEN,
+                                     NULL, 0,
+                                     NI_NUMERICHOST))
+                    {
+                        networkAddress.setAddress(address);
+                        if ( !(  networkAddress.isNull()                                                               ) &&
+                             !( (networkAddress.protocol() == QAbstractSocket::IPv4Protocol) && (networkAddress.toString() == "127.0.0.1") ) &&
+                             !( (networkAddress.protocol() == QAbstractSocket::IPv6Protocol) && (networkAddress.toString() == "::1"      ) ) )
+                        {
+                            networkControl = new NetworkControl();
+                            if (!networkControl->listen(networkAddress, networkPort))
+                                VERBOSE(VB_IMPORTANT,
+                                        QString("NetworkControl failed to bind to port %1 of address %2.")
+                                        .arg(networkPort).arg(networkAddress.toString()));
+                            networkControls.append(networkControl);
+                        }
+                    }
+                    cur = cur->ai_next;
+                }
+            }
+            else
+            {
+                VERBOSE(VB_IMPORTANT,
+                        QString("NetworkControl failed to determine network address(es)."));
+            }
+            if (res != NULL)
+                freeaddrinfo(res);
+        }
+        else
+        {
             VERBOSE(VB_IMPORTANT,
-                    QString("NetworkControl failed to bind to port %1.")
-                    .arg(networkPort));
+                   QString("NetworkControl failed to determine host name."));
+        }
     }
 
     gContext->addCurrentLocation("MainMenu");
@@ -1467,8 +1526,8 @@
         pthread_join(priv_thread, NULL);
     }
 
-    if (networkControl)
-        delete networkControl;
+    while (!networkControls.isEmpty())
+        delete networkControls.takeFirst();
 
     DestroyMythMainWindow();
 
