diff -Naur mythtv-trunk-21509-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-21509-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-21509-old/libs/libmyth/mediamonitor-unix.cpp	2009-08-24 23:33:40.000000000 -0700
+++ mythtv-trunk-21509-new/libs/libmyth/mediamonitor-unix.cpp	2009-08-24 23:33:40.000000000 -0700
@@ -223,11 +223,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
