diff -Naur mythtv-trunk-20493-old/configure mythtv-trunk-20493-new/configure
--- mythtv-trunk-20493-old/configure	2009-05-06 08:54:40.000000000 -0700
+++ mythtv-trunk-20493-new/configure	2009-05-06 08:54:40.000000000 -0700
@@ -3008,7 +3008,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
