diff -Naur mythtv-trunk-17081-old/libs/libmythtv/tv_play.cpp mythtv-trunk-17081-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-17081-old/libs/libmythtv/tv_play.cpp	2008-04-17 18:29:58.000000000 -0700
+++ mythtv-trunk-17081-new/libs/libmythtv/tv_play.cpp	2008-04-17 18:29:59.000000000 -0700
@@ -1830,7 +1830,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
