diff -Naur mythtv-trunk-26142-old/bindings/python/Makefile mythtv-trunk-26142-new/bindings/python/Makefile
--- mythtv-trunk-26142-old/bindings/python/Makefile	2010-09-06 06:29:44.000000000 -0700
+++ mythtv-trunk-26142-new/bindings/python/Makefile	2010-09-06 06:29:45.000000000 -0700
@@ -1,14 +1,14 @@
 include ../../config.mak
 
-ifdef INSTALL_ROOT
+#ifdef INSTALL_ROOT
 	ROOT_FLAGS = --root="$(INSTALL_ROOT)"
-endif
+#endif
 
-ifneq ($($(PREFIX):/=), /usr)
-	ifneq ($($(PREFIX):/=), /usr/local)
+#ifneq ($($(PREFIX):/=), /usr)
+#	ifneq ($($(PREFIX):/=), /usr/local)
 		PREFIX_FLAGS=--prefix="$(PREFIX)"
-	endif
-endif
+#	endif
+#endif
 
 all: python_build
 
@@ -18,7 +18,7 @@
 distclean: clean
 
 python_build: setup.py
-	$(PYTHON) setup.py build
+	$(PYTHON) setup.py build --cross-compile
 
 install: setup.py
 	$(PYTHON) setup.py install --skip-build $(ROOT_FLAGS) $(PREFIX_FLAGS)
@@ -31,4 +31,4 @@
 	$(PYTHON) setup.py uninstall $(PREFIX_FLAGS)
 endif
 
-.PHONY: all clean distclean install python_build uninstall
\ No newline at end of file
+.PHONY: all clean distclean install python_build uninstall
diff -Naur mythtv-trunk-26142-old/bindings/python/setup.py mythtv-trunk-26142-new/bindings/python/setup.py
--- mythtv-trunk-26142-old/bindings/python/setup.py	2010-09-06 06:29:44.000000000 -0700
+++ mythtv-trunk-26142-new/bindings/python/setup.py	2010-09-06 06:29:45.000000000 -0700
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
-from distutils.cmd import Command
+from setuptools import setup
+from setuptools import Command
 from distutils.sysconfig import get_python_lib
 from distutils.command.install import INSTALL_SCHEMES
 import os
