diff -Naur mythtv-trunk-16813-old/libs/libmyth/mythcdrom-linux.cpp mythtv-trunk-16813-new/libs/libmyth/mythcdrom-linux.cpp
--- mythtv-trunk-16813-old/libs/libmyth/mythcdrom-linux.cpp	2008-03-26 20:04:07.000000000 -0700
+++ mythtv-trunk-16813-new/libs/libmyth/mythcdrom-linux.cpp	2008-03-26 20:07:06.000000000 -0700
@@ -1,6 +1,7 @@
 #include "mythcdrom.h"
 #include "mythcdrom-linux.h"
 #include <sys/ioctl.h>       // ioctls
+#include <limits.h>
 #include <linux/cdrom.h>     // old ioctls for cdrom
 #include <scsi/sg.h>
 #include <fcntl.h>
diff -Naur mythtv-trunk-16813-old/libs/libmythtv/textsubtitleparser.cpp mythtv-trunk-16813-new/libs/libmythtv/textsubtitleparser.cpp
--- mythtv-trunk-16813-old/libs/libmythtv/textsubtitleparser.cpp	2008-03-26 20:03:57.000000000 -0700
+++ mythtv-trunk-16813-new/libs/libmythtv/textsubtitleparser.cpp	2008-03-26 20:07:06.000000000 -0700
@@ -5,6 +5,7 @@
  */
 
 #include <stdio.h>
+#include <limits.h>
 #include <cstring>
 #include <algorithm>
 #include <qtextcodec.h>
diff -Naur mythtv-trunk-16813-old/programs/mythcommflag/pgm.cpp mythtv-trunk-16813-new/programs/mythcommflag/pgm.cpp
--- mythtv-trunk-16813-old/programs/mythcommflag/pgm.cpp	2008-03-26 20:04:08.000000000 -0700
+++ mythtv-trunk-16813-new/programs/mythcommflag/pgm.cpp	2008-03-26 20:07:06.000000000 -0700
@@ -1,6 +1,7 @@
 extern "C" {
 #include "avcodec.h"
 }
+#include <limits.h>
 #include "frame.h"
 #include "mythcontext.h"
 #include "pgm.h"
