diff -Naur mythtv-trunk-25535-old/bindings/python/python.pro mythtv-trunk-25535-new/bindings/python/python.pro
--- mythtv-trunk-25535-old/bindings/python/python.pro	2010-08-01 21:49:51.000000000 -0700
+++ mythtv-trunk-25535-new/bindings/python/python.pro	2010-08-06 05:08:49.000000000 -0700
@@ -2,20 +2,20 @@
 
 python_build.target = python_build
 python_build.depends = setup.py FORCE
-python_build.commands = $$PYTHON setup.py build
+python_build.commands = $$PYTHON setup.py build --cross-compile
 
 QMAKE_EXTRA_TARGETS += python_build
 PRE_TARGETDEPS += python_build
 
 python_install.target = python_install
 python_install.CONFIG = no_path
-python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\"
+python_install.commands = $$PYTHON setup.py install --skip-build --root=\"$(if $(INSTALL_ROOT),$(INSTALL_ROOT),/)\" --prefix=\"$${PREFIX}\"
 
 INSTALLS += python_install
 
 python_uninstall.target = python_uninstall
 python_uninstall.depends = setup.py FORCE
-python_uninstall.commands = $$PYTHON setup.py uninstall
+python_uninstall.commands = $$PYTHON setup.py uninstall --prefix=\"$${PREFIX}\"
 
 QMAKE_EXTRA_TARGETS += python_uninstall
 
diff -Naur mythtv-trunk-25535-old/bindings/python/setup.py mythtv-trunk-25535-new/bindings/python/setup.py
--- mythtv-trunk-25535-old/bindings/python/setup.py	2010-08-01 21:49:51.000000000 -0700
+++ mythtv-trunk-25535-new/bindings/python/setup.py	2010-08-06 05:10:02.000000000 -0700
@@ -1,9 +1,9 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
-from distutils.cmd import Command
-from distutils.sysconfig import get_python_lib
-from distutils.command.install import INSTALL_SCHEMES
+from setuptools import setup
+from setuptools import Command
+from setuptools import get_python_lib
+from setuptools import INSTALL_SCHEMES
 import os
 import glob
 
