diff -Naur mythtv-trunk-17008-old/libs/libmyth/util.cpp mythtv-trunk-17008-new/libs/libmyth/util.cpp
--- mythtv-trunk-17008-old/libs/libmyth/util.cpp	2008-04-09 20:04:04.000000000 -0700
+++ mythtv-trunk-17008-new/libs/libmyth/util.cpp	2008-04-09 20:04:05.000000000 -0700
@@ -296,10 +296,11 @@
                     return GENERIC_EXIT_NOT_OK;
                 }
 
+                qApp->processEvents();
+
                 if (res > 0)
                     return WEXITSTATUS(status);
 
-                qApp->processEvents();
                 usleep(100000);
             }
         }
