diff -Naur mythtv-trunk-20667-old/libs/libmythtv/vsync.cpp mythtv-trunk-20667-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20667-old/libs/libmythtv/vsync.cpp	2009-06-02 18:49:30.000000000 -0700
+++ mythtv-trunk-20667-new/libs/libmythtv/vsync.cpp	2009-06-02 18:49:33.000000000 -0700
@@ -96,7 +96,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
