diff -Naur mythtv-trunk-20616-old/libs/libmythtv/vsync.cpp mythtv-trunk-20616-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20616-old/libs/libmythtv/vsync.cpp	2009-05-24 09:33:10.000000000 -0700
+++ mythtv-trunk-20616-new/libs/libmythtv/vsync.cpp	2009-05-24 09:33:12.000000000 -0700
@@ -96,7 +96,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
