diff -Naur mythtv-16133-old/themes/cpsvndir mythtv-16133-new/themes/cpsvndir
--- mythtv-16133-old/themes/cpsvndir	2008-02-18 08:01:38.000000000 -0800
+++ mythtv-16133-new/themes/cpsvndir	2008-02-18 08:02:13.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
