diff -Naur mythtv-trunk-24011-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-24011-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-24011-old/libs/libmyth/mediamonitor-unix.cpp	2010-04-06 07:03:38.000000000 -0700
+++ mythtv-trunk-24011-new/libs/libmyth/mediamonitor-unix.cpp	2010-04-06 07:03:38.000000000 -0700
@@ -222,11 +222,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
