diff -Naur mythtv-trunk-16318-old/libs/libmythtv/tv_play.cpp mythtv-trunk-16318-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-16318-old/libs/libmythtv/tv_play.cpp	2008-02-29 21:43:37.000000000 -0800
+++ mythtv-trunk-16318-new/libs/libmythtv/tv_play.cpp	2008-02-29 21:44:12.000000000 -0800
@@ -1815,7 +1815,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
