diff -Naur mythtv-trunk-17941-old/libs/libmythtv/tv_play.cpp mythtv-trunk-17941-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-17941-old/libs/libmythtv/tv_play.cpp	2008-07-25 17:54:57.000000000 -0700
+++ mythtv-trunk-17941-new/libs/libmythtv/tv_play.cpp	2008-07-25 17:54:58.000000000 -0700
@@ -1794,7 +1794,7 @@
     if (pthread_create(&decode, NULL, SpawnDecode, nvp))
         return false;
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
