diff -Naur mythtv-trunk-19555-old/configure mythtv-trunk-19555-new/configure
--- mythtv-trunk-19555-old/configure	2009-01-03 19:06:42.000000000 -0800
+++ mythtv-trunk-19555-new/configure	2009-01-03 19:06:44.000000000 -0800
@@ -1633,7 +1633,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
