diff -Naur mythtv-trunk-20963-old/libs/libmythtv/vsync.cpp mythtv-trunk-20963-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20963-old/libs/libmythtv/vsync.cpp	2009-07-19 17:15:14.000000000 -0700
+++ mythtv-trunk-20963-new/libs/libmythtv/vsync.cpp	2009-07-19 17:15:15.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
