diff -Naur mythtv-trunk-19169-old/libs/libmythtv/videoout_xv.cpp mythtv-trunk-19169-new/libs/libmythtv/videoout_xv.cpp
--- mythtv-trunk-19169-old/libs/libmythtv/videoout_xv.cpp	2008-11-28 22:25:37.000000000 -0800
+++ mythtv-trunk-19169-new/libs/libmythtv/videoout_xv.cpp	2008-11-28 22:25:38.000000000 -0800
@@ -1642,7 +1642,12 @@
         for (; it != renderers.end(); ++it)
             tmp += *it + ",";
 
-        renderer = renderers[0];
+        if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-blit") && (renderers.contains("xv-blit")))
+            renderer = "xv-blit";
+        else if ((db_vdisp_profile->GetVideoRenderer() == "xvmc-opengl") && (renderers.contains("opengl")))
+            renderer = "opengl";
+        else
+            renderer = renderers[0];
         VERBOSE(VB_IMPORTANT, LOC + QString(
                     "Desired video renderer '%1' not available.\n\t\t\t"
                     "codec '%2' makes '%3' available, using '%4' instead.")
