diff -Naur mythtv-trunk-26199-old/libs/libmythtv/avformatdecoder.cpp mythtv-trunk-26199-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-trunk-26199-old/libs/libmythtv/avformatdecoder.cpp	2010-09-10 09:01:47.000000000 -0700
+++ mythtv-trunk-26199-new/libs/libmythtv/avformatdecoder.cpp	2010-09-10 09:03:57.000000000 -0700
@@ -1270,6 +1270,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
