diff -Naur mythtv-trunk-17014-old/libs/libmyth/util.cpp mythtv-trunk-17014-new/libs/libmyth/util.cpp
--- mythtv-trunk-17014-old/libs/libmyth/util.cpp	2008-04-10 19:43:27.000000000 -0700
+++ mythtv-trunk-17014-new/libs/libmyth/util.cpp	2008-04-10 19:43:30.000000000 -0700
@@ -296,10 +296,11 @@
                     return GENERIC_EXIT_NOT_OK;
                 }
 
+                qApp->processEvents();
+
                 if (res > 0)
                     return WEXITSTATUS(status);
 
-                qApp->processEvents();
                 usleep(100000);
             }
         }
