diff -Naur mythtv-trunk-23304-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-23304-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-23304-old/libs/libmyth/mediamonitor-unix.cpp	2010-01-25 17:43:10.000000000 -0800
+++ mythtv-trunk-23304-new/libs/libmyth/mediamonitor-unix.cpp	2010-01-25 17:43:11.000000000 -0800
@@ -222,11 +222,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
