diff -Naur mythtv-trunk-20301-old/configure mythtv-trunk-20301-new/configure
--- mythtv-trunk-20301-old/configure	2009-04-02 21:25:17.000000000 -0700
+++ mythtv-trunk-20301-new/configure	2009-04-02 21:25:18.000000000 -0700
@@ -1633,7 +1633,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
