diff -Naur mythtv-trunk-20127-old/configure mythtv-trunk-20127-new/configure
--- mythtv-trunk-20127-old/configure	2009-03-06 18:09:18.000000000 -0800
+++ mythtv-trunk-20127-new/configure	2009-03-06 18:09:27.000000000 -0800
@@ -1633,7 +1633,7 @@
 set >> $logfile
 
 cc="${cross_prefix}${cc}"
-yasmexe="${cross_prefix}${yasmexe}"
+#yasmexe="${cross_prefix}${yasmexe}"
 ar="${cross_prefix}${ar}"
 nm="${cross_prefix}${nm}"
 ranlib="${cross_prefix}${ranlib}"
