diff -Naur mythtv-trunk-17598-old/libs/libmyth/mythcontext.cpp mythtv-trunk-17598-new/libs/libmyth/mythcontext.cpp
--- mythtv-trunk-17598-old/libs/libmyth/mythcontext.cpp	2008-06-23 16:21:03.000000000 -0700
+++ mythtv-trunk-17598-new/libs/libmyth/mythcontext.cpp	2008-06-23 16:21:04.000000000 -0700
@@ -1809,7 +1809,7 @@
 #if defined(CONFIG_DARWIN) || (__FreeBSD__) || defined(__OpenBSD__)
     const char *command = "ps -ax | grep -i mythbackend | grep -v grep > /dev/null";
 #else
-    const char *command = "ps -ae | grep mythbackend > /dev/null";
+    const char *command = "ps | grep mythbackend | grep -v grep > /dev/null";
 #endif
     bool res = myth_system(command,
                            MYTH_SYSTEM_DONT_BLOCK_LIRC |
