diff -Naur mythtv-trunk-23001-old/libs/libmyth/mythcontext.cpp mythtv-trunk-23001-new/libs/libmyth/mythcontext.cpp
--- mythtv-trunk-23001-old/libs/libmyth/mythcontext.cpp	2009-12-22 19:43:44.000000000 -0800
+++ mythtv-trunk-23001-new/libs/libmyth/mythcontext.cpp	2009-12-22 19:46:18.000000000 -0800
@@ -1659,7 +1659,7 @@
                     continue;
                 }
 
-                d->WOLInProgress = we_attempted_wol = true;
+//              d->WOLInProgress = we_attempted_wol = true;
             }
 
             myth_system(WOLcmd);
