diff -Naur mythtv-trunk-16465-old/libs/libmythtv/tv_play.cpp mythtv-trunk-16465-new/libs/libmythtv/tv_play.cpp
--- mythtv-trunk-16465-old/libs/libmythtv/tv_play.cpp	2008-03-08 16:10:49.000000000 -0800
+++ mythtv-trunk-16465-new/libs/libmythtv/tv_play.cpp	2008-03-08 16:10:52.000000000 -0800
@@ -1816,7 +1816,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
