diff -Naur mythtv-trunk-24947-old/libs/libmyth/mediamonitor-unix.cpp mythtv-trunk-24947-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-trunk-24947-old/libs/libmyth/mediamonitor-unix.cpp	2010-06-02 20:01:59.000000000 -0700
+++ mythtv-trunk-24947-new/libs/libmyth/mediamonitor-unix.cpp	2010-06-02 20:02:00.000000000 -0700
@@ -221,11 +221,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
