diff -Naur mythtv-trunk-20411-old/configure mythtv-trunk-20411-new/configure
--- mythtv-trunk-20411-old/configure	2009-04-17 20:44:48.000000000 -0700
+++ mythtv-trunk-20411-new/configure	2009-04-17 20:44:48.000000000 -0700
@@ -2986,7 +2986,7 @@
     fi
 fi
 
-enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3_threads -lfftw3 ||
+enabled libfftw3 && check_lib2 fftw3.h fftw_init_threads -lfftw3f_threads -lfftw3f ||
     disable libfftw3
 
 enabled x11 && has_library libX11 || disable x11
