diff -Naur mythtv-trunk-18806-old/programs/mythfrontend/playbackbox.cpp mythtv-trunk-18806-new/programs/mythfrontend/playbackbox.cpp
--- mythtv-trunk-18806-old/programs/mythfrontend/playbackbox.cpp	2008-10-20 20:39:53.000000000 -0700
+++ mythtv-trunk-18806-new/programs/mythfrontend/playbackbox.cpp	2008-10-20 20:39:53.000000000 -0700
@@ -4276,7 +4276,7 @@
     // This should be an impossible keypress we've simulated
     if ((e->key() == Qt::Key_LaunchMedia) &&
         (e->state() == (Qt::ButtonState)
-         (Qt::MouseButtonMask | Qt::KeyboardModifierMask)))
+         ((int)Qt::MouseButtonMask | (int)Qt::KeyboardModifierMask)))
     {
         e->accept();
         ncLock.lock();
@@ -4439,7 +4439,7 @@
                 // This should be an impossible keypress we're simulating
                 QKeyEvent *event;
                 int buttons =
-                        (Qt::MouseButtonMask | Qt::KeyboardModifierMask);
+                        ((int)Qt::MouseButtonMask | (int)Qt::KeyboardModifierMask);
 
                 event = new QKeyEvent(QEvent::KeyPress, Qt::Key_LaunchMedia,
                                       0, buttons);
