diff -Naur mythtv-trunk-16989-old/libs/libmyth/util.cpp mythtv-trunk-16989-new/libs/libmyth/util.cpp
--- mythtv-trunk-16989-old/libs/libmyth/util.cpp	2008-04-08 14:20:41.000000000 -0700
+++ mythtv-trunk-16989-new/libs/libmyth/util.cpp	2008-04-08 14:20:42.000000000 -0700
@@ -296,10 +296,11 @@
                     return GENERIC_EXIT_NOT_OK;
                 }
 
+                qApp->processEvents();
+
                 if (res > 0)
                     return WEXITSTATUS(status);
 
-                qApp->processEvents();
                 usleep(100000);
             }
         }
