diff -Naur mythtv-trunk-20842-old/libs/libmythtv/vsync.cpp mythtv-trunk-20842-new/libs/libmythtv/vsync.cpp
--- mythtv-trunk-20842-old/libs/libmythtv/vsync.cpp	2009-07-09 05:12:43.000000000 -0700
+++ mythtv-trunk-20842-new/libs/libmythtv/vsync.cpp	2009-07-09 05:12:44.000000000 -0700
@@ -100,7 +100,7 @@
 //    TESTVIDEOSYNC(VDPAUVideoSync);
 #endif    
 #ifndef _WIN32
-    TESTVIDEOSYNC(DRMVideoSync);
+//  TESTVIDEOSYNC(DRMVideoSync);
     if (tryOpenGL)
         TESTVIDEOSYNC(OpenGLVideoSync);
 #endif // _WIN32
