diff -Naur binutils-2.18.50.0.8-old/bfd/configure binutils-2.18.50.0.8-new/bfd/configure
--- binutils-2.18.50.0.8-old/bfd/configure	2008-07-10 08:32:57.000000000 -0700
+++ binutils-2.18.50.0.8-new/bfd/configure	2008-07-12 10:57:42.000000000 -0700
@@ -8749,7 +8749,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.50.0.8-old/binutils/configure binutils-2.18.50.0.8-new/binutils/configure
--- binutils-2.18.50.0.8-old/binutils/configure	2008-07-10 08:32:57.000000000 -0700
+++ binutils-2.18.50.0.8-new/binutils/configure	2008-07-12 10:57:42.000000000 -0700
@@ -8579,7 +8579,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.50.0.8-old/gas/configure binutils-2.18.50.0.8-new/gas/configure
--- binutils-2.18.50.0.8-old/gas/configure	2008-07-10 08:32:58.000000000 -0700
+++ binutils-2.18.50.0.8-new/gas/configure	2008-07-12 10:57:42.000000000 -0700
@@ -8577,7 +8577,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.50.0.8-old/gprof/configure binutils-2.18.50.0.8-new/gprof/configure
--- binutils-2.18.50.0.8-old/gprof/configure	2008-07-10 08:32:58.000000000 -0700
+++ binutils-2.18.50.0.8-new/gprof/configure	2008-07-12 10:57:42.000000000 -0700
@@ -8669,7 +8669,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.50.0.8-old/ld/configure binutils-2.18.50.0.8-new/ld/configure
--- binutils-2.18.50.0.8-old/ld/configure	2008-07-10 08:32:58.000000000 -0700
+++ binutils-2.18.50.0.8-new/ld/configure	2008-07-12 10:57:42.000000000 -0700
@@ -8868,7 +8868,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.50.0.8-old/libiberty/Makefile.in binutils-2.18.50.0.8-new/libiberty/Makefile.in
--- binutils-2.18.50.0.8-old/libiberty/Makefile.in	2008-04-03 09:54:05.000000000 -0700
+++ binutils-2.18.50.0.8-new/libiberty/Makefile.in	2008-07-12 10:57:42.000000000 -0700
@@ -352,7 +352,7 @@
 # multilib-specific flags, it's overridden by FLAGS_TO_PASS from the
 # default multilib, so we have to take LIBCFLAGS into account as well,
 # since it will be passed the multilib flags.
-MULTIOSDIR = `$(CC) $(LIBCFLAGS) -print-multi-os-directory`
+MULTIOSDIR = '../lib'
 install_to_libdir: all
 	${mkinstalldirs} $(DESTDIR)$(libdir)/$(MULTIOSDIR)
 	$(INSTALL_DATA) $(TARGETLIB) $(DESTDIR)$(libdir)/$(MULTIOSDIR)/$(TARGETLIB)n
diff -Naur binutils-2.18.50.0.8-old/libtool.m4 binutils-2.18.50.0.8-new/libtool.m4
--- binutils-2.18.50.0.8-old/libtool.m4	2008-04-03 09:54:04.000000000 -0700
+++ binutils-2.18.50.0.8-new/libtool.m4	2008-07-12 10:57:42.000000000 -0700
@@ -1929,7 +1929,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.50.0.8-old/opcodes/configure binutils-2.18.50.0.8-new/opcodes/configure
--- binutils-2.18.50.0.8-old/opcodes/configure	2008-07-10 08:32:59.000000000 -0700
+++ binutils-2.18.50.0.8-new/opcodes/configure	2008-07-12 10:57:42.000000000 -0700
@@ -8745,7 +8745,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
