diff -Naur binutils-2.18.93-old/bfd/configure binutils-2.18.93-new/bfd/configure
--- binutils-2.18.93-old/bfd/configure	2008-10-09 05:28:34.000000000 -0700
+++ binutils-2.18.93-new/bfd/configure	2008-10-09 13:40:00.000000000 -0700
@@ -8991,7 +8991,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.93-old/binutils/configure binutils-2.18.93-new/binutils/configure
--- binutils-2.18.93-old/binutils/configure	2008-08-23 20:13:05.000000000 -0700
+++ binutils-2.18.93-new/binutils/configure	2008-10-09 13:40:00.000000000 -0700
@@ -8822,7 +8822,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.93-old/gas/configure binutils-2.18.93-new/gas/configure
--- binutils-2.18.93-old/gas/configure	2008-08-23 20:13:05.000000000 -0700
+++ binutils-2.18.93-new/gas/configure	2008-10-09 13:40:00.000000000 -0700
@@ -8820,7 +8820,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.93-old/gprof/configure binutils-2.18.93-new/gprof/configure
--- binutils-2.18.93-old/gprof/configure	2008-08-15 01:31:50.000000000 -0700
+++ binutils-2.18.93-new/gprof/configure	2008-10-09 13:40:00.000000000 -0700
@@ -8911,7 +8911,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.93-old/ld/configure binutils-2.18.93-new/ld/configure
--- binutils-2.18.93-old/ld/configure	2008-08-23 20:13:05.000000000 -0700
+++ binutils-2.18.93-new/ld/configure	2008-10-09 13:40:00.000000000 -0700
@@ -9111,7 +9111,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.93-old/libiberty/Makefile.in binutils-2.18.93-new/libiberty/Makefile.in
--- binutils-2.18.93-old/libiberty/Makefile.in	2008-07-24 06:51:49.000000000 -0700
+++ binutils-2.18.93-new/libiberty/Makefile.in	2008-10-09 13:40:00.000000000 -0700
@@ -352,7 +352,7 @@
 # multilib-specific flags, it's overridden by FLAGS_TO_PASS from the
 # default multilib, so we have to take LIBCFLAGS into account as well,
 # since it will be passed the multilib flags.
-MULTIOSDIR = `$(CC) $(LIBCFLAGS) -print-multi-os-directory`
+MULTIOSDIR = '../lib'
 install_to_libdir: all
 	${mkinstalldirs} $(DESTDIR)$(libdir)/$(MULTIOSDIR)
 	$(INSTALL_DATA) $(TARGETLIB) $(DESTDIR)$(libdir)/$(MULTIOSDIR)/$(TARGETLIB)n
diff -Naur binutils-2.18.93-old/libtool.m4 binutils-2.18.93-new/libtool.m4
--- binutils-2.18.93-old/libtool.m4	2008-03-15 23:49:28.000000000 -0700
+++ binutils-2.18.93-new/libtool.m4	2008-10-09 13:40:00.000000000 -0700
@@ -1929,7 +1929,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.18.93-old/opcodes/configure binutils-2.18.93-new/opcodes/configure
--- binutils-2.18.93-old/opcodes/configure	2008-08-15 01:31:51.000000000 -0700
+++ binutils-2.18.93-new/opcodes/configure	2008-10-09 13:40:00.000000000 -0700
@@ -8987,7 +8987,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
