diff -Naur alsa-driver-1.0.16-old/Makefile alsa-driver-1.0.16-new/Makefile
--- alsa-driver-1.0.16-old/Makefile	2008-02-05 01:23:23.000000000 -0800
+++ alsa-driver-1.0.16-new/Makefile	2008-03-30 19:11:49.000000000 -0700
@@ -204,7 +204,7 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	-depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
 	@echo "*** This is a \`staged' install using the prefix"
 	@echo "***	\"$(DESTDIR)\"."
