diff -Naur mythstream-0.18_qt4-old/mythstream/main.cpp mythstream-0.18_qt4-new/mythstream/main.cpp
--- mythstream-0.18_qt4-old/mythstream/main.cpp	2008-09-26 06:35:05.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/main.cpp	2008-09-26 06:35:25.000000000 -0700
@@ -32,10 +32,10 @@
 #include "../libs/streamconfig.h"
 #include "../libs/storageconfig.h"
 
-#include <mythtv/mythcontext.h>
 #include <mythtv/mythdialogs.h>
 #include <mythtv/mythmedia.h>
 #include <mythtv/mythplugin.h>
+#include <mythtv/libmythdb/mythversion.h>
 #include <mythtv/libmythui/myththemedmenu.h>
 #include <mythtv/libmythui/mythuihelper.h>
 
