diff -Naur mythstream-0.18_qt4-old/libs/configelements.cpp mythstream-0.18_qt4-new/libs/configelements.cpp
--- mythstream-0.18_qt4-old/libs/configelements.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/configelements.cpp	2009-01-23 08:31:52.000000000 -0800
@@ -110,7 +110,7 @@
       lab->setMaximumHeight(maxHeight);
       tmp = new MyLineEdit(this, "edit" + QString::number(i));
 #ifdef MYTHTV      
-      tmp->setPopupPosition(VK_POSTOPDIALOG);
+      tmp->setPopupPosition(VKQT_POSTOPDIALOG);
 #endif
     }
     
diff -Naur mythstream-0.18_qt4-old/libs/configelements.h mythstream-0.18_qt4-new/libs/configelements.h
--- mythstream-0.18_qt4-old/libs/configelements.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/configelements.h	2009-01-23 08:31:17.000000000 -0800
@@ -30,7 +30,7 @@
 
 #ifdef MYTHTV
 #include <mythtv/mythwidgets.h>
-#include <mythtv/virtualkeyboard.h>
+#include <mythtv/virtualkeyboard_qt.h>
 #else
 #include <qlineedit.h>
 #endif
