diff -Naur mythstream-0.18_qt4-old/mythstream/main.cpp mythstream-0.18_qt4-new/mythstream/main.cpp
--- mythstream-0.18_qt4-old/mythstream/main.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/main.cpp	2008-07-01 11:03:43.000000000 -0700
@@ -37,6 +37,7 @@
 #include <mythtv/mythmedia.h>
 #include <mythtv/mythplugin.h>
 #include <mythtv/libmythui/myththemedmenu.h>
+#include <mythtv/libmythui/mythuihelper.h>
 
 using namespace std;
 
@@ -162,7 +163,7 @@
 
 int mythplugin_config(void)
 {
-    QString themedir = gContext->GetThemeDir( );
+    QString themedir = GetMythUI()->GetThemeDir( );
     runMenu( themedir, "streamconfigmenu.xml");
     return 0;
 }
