diff -Naur mythstream-0.18_qt4-old/libs/configelements.cpp mythstream-0.18_qt4-new/libs/configelements.cpp
--- mythstream-0.18_qt4-old/libs/configelements.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/configelements.cpp	2009-08-15 01:23:16.000000000 -0700
@@ -80,7 +80,7 @@
 
 // ------------------- Edit group box --------------------
 
-EditGroup::EditGroup(char* title, QWidget* parent) : Q3VGroupBox(title, parent)
+EditGroup::EditGroup(const char* title, QWidget* parent) : Q3VGroupBox(title, parent)
 {
     myParent = parent;
     setPalette(parent->palette());
diff -Naur mythstream-0.18_qt4-old/libs/configelements.h mythstream-0.18_qt4-new/libs/configelements.h
--- mythstream-0.18_qt4-old/libs/configelements.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/configelements.h	2009-08-15 01:31:47.000000000 -0700
@@ -80,9 +80,9 @@
     static const int r_web_pass  = 5;
 
 // storage value count
-    static const unsigned int r_count_file = 4;
-    static const unsigned int r_count_db   = 9;
-    static const unsigned int r_count_web  = 6;
+    static const int r_count_file = 4;
+    static const int r_count_db   = 9;
+    static const int r_count_web  = 6;
 
 // stream value indexes
     static const unsigned int s_folder  = 0;
@@ -194,7 +194,7 @@
         idle
     };
 
-    EditGroup(char* title, QWidget* parent);
+    EditGroup(const char* title, QWidget* parent);
     ~EditGroup();
 
     void buildGroup(int boxes);
diff -Naur mythstream-0.18_qt4-old/libs/harvester.cpp mythstream-0.18_qt4-new/libs/harvester.cpp
--- mythstream-0.18_qt4-old/libs/harvester.cpp	2008-04-07 15:23:50.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/harvester.cpp	2009-08-15 01:38:55.000000000 -0700
@@ -269,7 +269,7 @@
    QString currentDir = directory;
    bool pathCreated = true;
    
-   for ( uint i = 0; i < pathElements.size() - 1; i++ ) // assume last entry is file
+   for ( int i = 0; i < pathElements.size() - 1; i++ ) // assume last entry is file
    {
       currentDir += "/" + *pathElements.at(i);
       //cout << "create: " << currentDir << endl;
@@ -682,7 +682,7 @@
       
     QStringList cookieParts = QStringList::split( "; ", cookieMap[host] );
 
-    uint i = 0;
+    int i = 0;
     bool processed = false;
     
     while ( i < cookieParts.size() && !processed )
@@ -870,7 +870,7 @@
       QStringList cookieRawList = QStringList::split("&&COOKIESEP&&", cookieData);
 
       // feed every cookie to cookieBin
-      for (uint i = 0; i < cookieRawList.size(); i++)
+      for (int i = 0; i < cookieRawList.size(); i++)
          cookieBin.updateCookie( host, cookieRawList[i] );
     }
       
@@ -1564,6 +1564,8 @@
     eParser->addArgument( "\"" + sourceUrl + "\"" );  // change
     eParser->addArgument( "\"" + srcName + "\"" );  // change
 	break;
+	case NONE:
+	break;
     }
     
 
@@ -1795,7 +1797,7 @@
                   eParserUrl->url     = url;
                   //cout << "found name: " << name << ", handler: " << handl << endl;
                   // process meta information
-                  for (uint i = 0; i<lMeta.count(); i++)
+                  for (int i = 0; i<lMeta.count(); i++)
                   {
                      nMeta = lMeta.item(i);
                      metaName = "";
@@ -1967,6 +1969,8 @@
 			eParser->addArgument( "\"" + identString + "\"" );  // change
 			//cout << "PERL PARSER" << endl;
 			break;
+      case NONE:
+			break;
     }
 
     connect( eParser, SIGNAL(readyReadStdout()),
diff -Naur mythstream-0.18_qt4-old/libs/playerencap.cpp mythstream-0.18_qt4-new/libs/playerencap.cpp
--- mythstream-0.18_qt4-old/libs/playerencap.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/playerencap.cpp	2009-08-15 01:38:40.000000000 -0700
@@ -347,7 +347,7 @@
     QDomNodeList proplist;
 
     QDomNodeList list = parentNode.childNodes();
-    for (unsigned int i = 0; i < list.count(); i++)
+    for (int i = 0; i < list.count(); i++)
     {
         item    = list.item(i);
         label   = item.namedItem(QString("label")).toElement().text();
@@ -383,7 +383,7 @@
         if ( !props.isNull() )
         {
            proplist = props.childNodes();
-           for (unsigned int j = 0; j < proplist.count(); j++)
+           for (int j = 0; j < proplist.count(); j++)
            {
               item    = proplist.item(j);
               name    = item.namedItem(QString("name")).toElement().text();
@@ -421,7 +421,7 @@
     QString name, value;
 
     QDomNodeList list = parentNode.childNodes();
-    for (unsigned int i = 0; i < list.count(); i++)
+    for (int i = 0; i < list.count(); i++)
     {
       item  = list.item(i);
       name  = item.namedItem(QString("name")).toElement().text();
diff -Naur mythstream-0.18_qt4-old/libs/storageconfig.cpp mythstream-0.18_qt4-new/libs/storageconfig.cpp
--- mythstream-0.18_qt4-old/libs/storageconfig.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/storageconfig.cpp	2009-08-15 01:33:45.000000000 -0700
@@ -44,7 +44,7 @@
 {
 
   public:
-    StorageEditGroup(char* title, StorageConfig* parent) : EditGroup(title, parent)
+    StorageEditGroup(const char* title, StorageConfig* parent) : EditGroup(title, parent)
     {
         boxAddTitle  = "Add storage";
         boxEditTitle = "Edit storage";
@@ -383,7 +383,7 @@
    item->setText(1, rec->values[1]);
    item->setText(2, rec->values[0]);
 
-   for ( unsigned int i = 0; i < rec->values.count(); i++ )
+   for ( int i = 0; i < rec->values.count(); i++ )
      item->values.append(rec->values[i]);
 
    if (item->values.size() < 9 ) item->values.resize( 9, "" );
@@ -417,7 +417,7 @@
    repos->setText(1, rec->values[1]);
    repos->setText(2, rec->values[0]);
 
-   for ( unsigned int i = 0; i < rec->values.count(); i++ )
+   for ( int i = 0; i < rec->values.count(); i++ )
      repos->values[i] = rec->values[i];
 
    m_storageList->setSelected( repos, true );
diff -Naur mythstream-0.18_qt4-old/libs/storagehandlers.cpp mythstream-0.18_qt4-new/libs/storagehandlers.cpp
--- mythstream-0.18_qt4-old/libs/storagehandlers.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/storagehandlers.cpp	2009-08-15 01:28:05.000000000 -0700
@@ -58,7 +58,7 @@
 {
    error = false;
 
-   for (unsigned int i=0; i<values.count(); i++)
+   for (int i=0; i<values.count(); i++)
    {
      values[i]    = "";
      oldValues[i] = "";
diff -Naur mythstream-0.18_qt4-old/libs/streambrowser.cpp mythstream-0.18_qt4-new/libs/streambrowser.cpp
--- mythstream-0.18_qt4-old/libs/streambrowser.cpp	2008-04-07 15:09:09.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streambrowser.cpp	2009-08-15 01:34:49.000000000 -0700
@@ -1353,7 +1353,7 @@
         item->setPrefix(prefix);
         item->setAction(none);
         
-        for (uint i = 0; i < meta.size(); i++)
+        for (int i = 0; i < meta.size(); i++)
         {
             metaparts = QStringList::split("(stsep)", 
 	    meta.at(i), true);
@@ -2201,7 +2201,7 @@
         {
           cerr << "stream properties mismatch (received " << values.count()
                << " out of 5):" << endl;
-          for (uint i = 0; i<values.count(); i++)
+          for (int i = 0; i<values.count(); i++)
             cerr << "prop" << i << ": " << values[i].latin1() << endl;
           error = true;
         }
@@ -2265,10 +2265,12 @@
   checkAndSetIcon(item);
   // give *-prefix when item is inserted in the recording folder by recorder
   if ( folder->getName() == "recordings" )
+  {
     if ( rec->ident == ident_recorder )
       item->setPrefix("R");
     else
       item->setPrefix("#");
+  }
 
   emit eventValuesUpdated( folderlist );
 
diff -Naur mythstream-0.18_qt4-old/libs/streamconfig.cpp mythstream-0.18_qt4-new/libs/streamconfig.cpp
--- mythstream-0.18_qt4-old/libs/streamconfig.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamconfig.cpp	2009-08-15 01:23:16.000000000 -0700
@@ -59,7 +59,7 @@
 
 // ------------------- Storage Select group box --------------------
 
-MStorageGroup::MStorageGroup(char* title, QWidget* parent, ReposStorage *storageStorage,
+MStorageGroup::MStorageGroup(const char* title, QWidget* parent, ReposStorage *storageStorage,
                            StreamStorage *streamStorage) : Q3VGroupBox(title, parent)
 {
     myParent = parent;
@@ -354,35 +354,39 @@
       newStorage = storages->getStorageValuesByName( values, streams->getStorageName() );
 
       if ( activeStorage != streams->getStorageName() )
-      if ( oldStorage )
       {
-          //cout << "found old: " << activeStorage << endl;
-
-          newValues = oldStorageValues;
-          newValues[r_selected] = "-";
-
-          // condition should not fail
-          if ( oldStorageValues[r_selected] != "-" )
-            if ( !storages->updateRecord(ident_storagegroup, oldStorageValues, newValues, err) )
-              reportMessage(err, true);
+          if ( oldStorage )
+          {
+              //cout << "found old: " << activeStorage << endl;
+
+              newValues = oldStorageValues;
+              newValues[r_selected] = "-";
+
+              // condition should not fail
+              if ( oldStorageValues[r_selected] != "-" )
+                if ( !storages->updateRecord(ident_storagegroup, oldStorageValues, newValues, err) )
+                  reportMessage(err, true);
+          }
+          else
+              cerr << "warning: unknown previous connected storage" << endl;
       }
-        else
-          cerr << "warning: unknown previous connected storage" << endl;
 
       
       if ( activeStorage != streams->getStorageName() )
-      if ( newStorage )
       {
-          // cout << "found new: " << streams->getStorageName() << endl;
-
-          newValues = values;
-          newValues[r_selected] = "*";
-          if ( values[r_selected] != "*" )
-            if ( !storages->updateRecord(ident_storagegroup, values, newValues, err) )
-              reportMessage(err, true);
+          if ( newStorage )
+          {
+              // cout << "found new: " << streams->getStorageName() << endl;
+
+              newValues = values;
+              newValues[r_selected] = "*";
+              if ( values[r_selected] != "*" )
+                if ( !storages->updateRecord(ident_storagegroup, values, newValues, err) )
+                  reportMessage(err, true);
+          }
+          else
+              cerr << "error: unknown previous connected storage" << endl;
       }
-        else
-          cerr << "error: unknown previous connected storage" << endl;
 
       synchronized(false);
 
@@ -424,7 +428,7 @@
 {
 
   public:
-    StreamEditGroup(char* title, StreamConfig* parent) : EditGroup(title, parent)
+    StreamEditGroup(const char* title, StreamConfig* parent) : EditGroup(title, parent)
     {
         boxAddTitle  = "Add stream";
         boxEditTitle = "Edit stream";
@@ -500,7 +504,7 @@
       edit4 = getLineEdit("edit3");
       edit5 = getLineEdit("edit4");
 
-      if ( status == add | status == idle )
+      if ( (status == add) | (status == idle) )
       {
         streamConfig->createStation(edit1->text(), edit2->text(),
                                 edit3->text(), edit4->text(), edit5->text());
diff -Naur mythstream-0.18_qt4-old/libs/streamconfig.h mythstream-0.18_qt4-new/libs/streamconfig.h
--- mythstream-0.18_qt4-old/libs/streamconfig.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamconfig.h	2009-08-15 01:23:16.000000000 -0700
@@ -62,7 +62,7 @@
     Q_OBJECT
 
   public:
-    MStorageGroup(char* title, QWidget* parent, ReposStorage *storageStorage, StreamStorage *streamStorage);
+    MStorageGroup(const char* title, QWidget* parent, ReposStorage *storageStorage, StreamStorage *streamStorage);
     ~MStorageGroup();
 
   public slots:
diff -Naur mythstream-0.18_qt4-old/libs/streamstatus.cpp mythstream-0.18_qt4-new/libs/streamstatus.cpp
--- mythstream-0.18_qt4-old/libs/streamstatus.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamstatus.cpp	2009-08-15 01:23:16.000000000 -0700
@@ -911,7 +911,7 @@
    delete proc;
    proc = NULL;
 
-   if ( getStatus() == starting | getStatus() == buffering )
+   if ( (getStatus() == starting) | (getStatus() == buffering) )
      setStatus(nostream);  // nothing to play,pause or videoinit = no stream
    else
      setStatus(stopped);
diff -Naur mythstream-0.18_qt4-old/mythstream/mythstream.cpp mythstream-0.18_qt4-new/mythstream/mythstream.cpp
--- mythstream-0.18_qt4-old/mythstream/mythstream.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/mythstream.cpp	2009-08-15 01:41:53.000000000 -0700
@@ -227,11 +227,13 @@
    if (displayResetDelay != -1 && --displayResetDelay == 0)
    {
    if (streamStatus == "playing" || streamStatus == "buffering" )
+   {
       if (playingVideo)
          displayStatus = video;
       else
          displayStatus = audio;
 
+   }
    if (streamStatus == "idle")
    {
       displayStatus = browse;
@@ -984,11 +986,13 @@
    }
       else if ( name.left(6) == "button" )
    {
-      if(   (layerSet = m_Theme->GetSet("audio_panel"))
-         && (button = (UITextButtonType *)layerSet->GetType(name)) )
+      if ( (layerSet = m_Theme->GetSet("audio_panel")) )
       {
-         button->setText(value);
-         if (!button->isShown() ) button->show();
+         if ( (button = (UITextButtonType *)layerSet->GetType(name)) )
+         {
+            button->setText(value);
+            if (!button->isShown() ) button->show();
+         }
       }
    }
       else if (type == "command")
@@ -1256,7 +1260,7 @@
 
    gContext->GetMainWindow()->TranslateKeyPress("Stream", e, actions);
 
-   for (unsigned int i = 0; i < actions.size() && !handled; i++)
+   for (int i = 0; i < actions.size() && !handled; i++)
    {
       QString action = actions[i];
       QString browserAction = "";
