diff -Naur mythstream-0.18_qt4-old/libs/recorder.cpp mythstream-0.18_qt4-new/libs/recorder.cpp
--- mythstream-0.18_qt4-old/libs/recorder.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/recorder.cpp	2008-10-11 07:59:05.000000000 -0700
@@ -112,7 +112,7 @@
    if (recList.count() != 0) // no storage event should occur during recording
      cerr << "TARGET warning: storage manipulation during recording" << endl;
 
-   if (eventType == Storage::loaded)
+   if (eventType == MStorage::loaded)
    {
       if (recList.count() != 0) // no storage event should occur during recording
         stopAllRecordings();
diff -Naur mythstream-0.18_qt4-old/libs/storage.cpp mythstream-0.18_qt4-new/libs/storage.cpp
--- mythstream-0.18_qt4-old/libs/storage.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/storage.cpp	2008-10-11 07:57:46.000000000 -0700
@@ -37,9 +37,9 @@
 using namespace std;
 
 
-//----------------------- Storage ---------------------------
+//----------------------- MStorage ---------------------------
 
-Storage::Storage(QString table, QString name,
+MStorage::MStorage(QString table, QString name,
                  int key_pri, int key_mul1, int key_mul2) : QObject()
 {
     currentStorage = 0;
@@ -53,7 +53,7 @@
     this->key_mul2 = key_mul2;
 }
 
-Storage::~Storage()
+MStorage::~MStorage()
 {
   if (currentStorage)
     delete currentStorage;
@@ -62,7 +62,7 @@
 }
 
 
-bool Storage::storageReadOnly()
+bool MStorage::storageReadOnly()
 {
   if (currentStorage)
     return currentStorage->storageReadOnly();
@@ -70,7 +70,7 @@
     return false;
 }
 
-QString Storage::getStorageDescription()
+QString MStorage::getStorageDescription()
 {
   if (currentStorage)
     return currentStorage->getStorageDescription();
@@ -78,7 +78,7 @@
     return "TARGET storage warning: no storage selected";
 }
 
-Storage::AccessType Storage::getAccessType()
+MStorage::AccessType MStorage::getAccessType()
 {
   if (currentStorage)
     return (AccessType)currentStorage->getAccessType();
@@ -86,7 +86,7 @@
     return none;
 }
 
-QString Storage::getStorageName()
+QString MStorage::getStorageName()
 {
   if (currentStorage)
     return currentStorage->getStorageName();
@@ -94,7 +94,7 @@
     return "warning: no storage selected";
 }
 
-bool Storage::getSynchronized()
+bool MStorage::getSynchronized()
 {
   if (currentStorage)
     return currentStorage->getSynchronized();
@@ -102,7 +102,7 @@
     return false;
 }
 
-QString Storage::getLastError()
+QString MStorage::getLastError()
 {
   if (currentStorage)
     return currentStorage->getLastError();
@@ -110,13 +110,13 @@
     return "";
 }
 
-void Storage::resetRecordList()
+void MStorage::resetRecordList()
 {
   if (currentStorage)
     currentStorage->resetRecordList();
 }
 
-bool Storage::getNextRecord(ValueList& values)
+bool MStorage::getNextRecord(ValueList& values)
 {
   if (currentStorage)
     return currentStorage->getNextRecord(currentStorage->getListHandler(), values);
@@ -124,7 +124,7 @@
     return false;
 }
 
-void Storage::linkNewStorage(GenStorage *storage)
+void MStorage::linkNewStorage(GenStorage *storage)
 {
     currentStorage = storage;
 
@@ -158,7 +158,7 @@
 
 }
 
-bool Storage::selectDbStorage(int ident, QString name, QString dbDriver,
+bool MStorage::selectDbStorage(int ident, QString name, QString dbDriver,
                               QString hostName, unsigned int port, QString dbName, QString table,
                               QString login, QString password)
 {
@@ -175,7 +175,7 @@
 }
 
 #ifdef MYTHTV
-bool Storage::selectDefaultDb(int ident)
+bool MStorage::selectDefaultDb(int ident)
 {
     if (oldStorage) delete oldStorage;
     oldStorage = currentStorage;
@@ -189,7 +189,7 @@
 }
 #endif
 
-bool Storage::selectFileStorage(int ident, QString name, QString fileName)
+bool MStorage::selectFileStorage(int ident, QString name, QString fileName)
 {
     if (oldStorage) delete oldStorage;
     oldStorage = currentStorage;
@@ -202,7 +202,7 @@
     return fileStorage->openFileStorage(ident, fileName);
 }
 
-void Storage::selectWebStorage(int ident, QString name, QString url, QString login, QString password)
+void MStorage::selectWebStorage(int ident, QString name, QString url, QString login, QString password)
 {
 
     if (oldStorage) delete oldStorage;
@@ -217,7 +217,7 @@
 }
 
 
-bool Storage::loadList(int ident, QString& error)
+bool MStorage::loadList(int ident, QString& error)
 {
     if (currentStorage)
     {
@@ -238,7 +238,7 @@
 
 // - copy from resource1 to resource2
 // - update file resource after removal
-bool Storage::storeList(int ident, QString& error)
+bool MStorage::storeList(int ident, QString& error)
 {
    if (!currentStorage)
    {
@@ -255,7 +255,7 @@
    return currentStorage->storeList(ident, oldStorage->getListHandler(), error);
 }
 
-bool Storage::insertRecord(int ident, ValueList& values, QString& error)
+bool MStorage::insertRecord(int ident, ValueList& values, QString& error)
 {
    if (currentStorage)
      return currentStorage->insertRecord(ident, values, error);
@@ -266,7 +266,7 @@
    }
 }
 
-bool Storage::updateRecord(int ident, ValueList& oldValues, ValueList& values, QString& error)
+bool MStorage::updateRecord(int ident, ValueList& oldValues, ValueList& values, QString& error)
 {
 
    if (currentStorage)
@@ -278,7 +278,7 @@
    }
 }
 
-bool Storage::removeRecord(int ident, ValueList values, QString& error)
+bool MStorage::removeRecord(int ident, ValueList values, QString& error)
 {
    if (currentStorage)
      return currentStorage->removeRecord(ident, values, error);
diff -Naur mythstream-0.18_qt4-old/libs/storage.h mythstream-0.18_qt4-new/libs/storage.h
--- mythstream-0.18_qt4-old/libs/storage.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/storage.h	2008-10-11 07:56:10.000000000 -0700
@@ -37,7 +37,9 @@
 /*
   Absolutely NOT thread safe
 
-  Storage constraints:
+  Storage has been renamed to MStorage: it conflicted with new mythtv class Storage
+
+  MStorage constraints:
     1) url is unique
     2) (folder, name) is unique
 
@@ -53,7 +55,7 @@
       -> In case of file storage: selectFileStorage + removeRecord + storeList compacts file
 */
 
-class Storage : public QObject
+class MStorage : public QObject
 {
     Q_OBJECT
 
@@ -74,9 +76,9 @@
       saved
     };
 
-    Storage(QString table = "", QString name = "",
+    MStorage(QString table = "", QString name = "",
             int key_pri = 2, int key_mul1 = 0, int key_mul2 = 1);
-    ~Storage();
+    ~MStorage();
 
     // select storage resource (without reading records)
 #ifdef MYTHTV
diff -Naur mythstream-0.18_qt4-old/libs/streamstorage.cpp mythstream-0.18_qt4-new/libs/streamstorage.cpp
--- mythstream-0.18_qt4-old/libs/streamstorage.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamstorage.cpp	2008-10-11 08:03:39.000000000 -0700
@@ -326,7 +326,7 @@
     return (values.count() >= r_count_file && values[r_name] == storageName);
 }
 
-ReposStorage::ReposStorage() : Storage("", "", 2, 2, 2)
+ReposStorage::ReposStorage() : MStorage("", "", 2, 2, 2)
 {
     QString error;
     QString source;
diff -Naur mythstream-0.18_qt4-old/libs/streamstorage.h mythstream-0.18_qt4-new/libs/streamstorage.h
--- mythstream-0.18_qt4-old/libs/streamstorage.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamstorage.h	2008-10-11 07:58:45.000000000 -0700
@@ -36,15 +36,15 @@
 
 // Open reposstorage first: it creates default files on first run
 
-class StreamStorage : public Storage
+class StreamStorage : public MStorage
 {
   public:
-    StreamStorage(QString table = "", QString name = "") : Storage(table, name) {}
+    StreamStorage(QString table = "", QString name = "") : MStorage(table, name) {}
 
     bool selectStorage(int ident, ValueList& values);
 };
 
-class ReposStorage : public Storage
+class ReposStorage : public MStorage
 {
   public:
     ReposStorage();
