diff -Naur xf86-video-openchrome-737-old/configure.ac xf86-video-openchrome-737-new/configure.ac
--- xf86-video-openchrome-737-old/configure.ac	2009-03-17 06:03:18.000000000 -0700
+++ xf86-video-openchrome-737-new/configure.ac	2009-03-17 06:04:51.000000000 -0700
@@ -38,6 +38,7 @@
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+PKG_PROG_PKG_CONFIG
 
 AH_TOP([#include "xorg-server.h"])
 
@@ -71,7 +72,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server xproto fontsproto libdrm $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 
