diff -Naur linux-2.6.32.9-old/drivers/staging/Kconfig linux-2.6.32.9-new/drivers/staging/Kconfig
--- linux-2.6.32.9-old/drivers/staging/Kconfig	2010-02-23 20:48:29.000000000 -0800
+++ linux-2.6.32.9-new/drivers/staging/Kconfig	2010-02-23 20:52:10.000000000 -0800
@@ -103,6 +103,8 @@
 
 source "drivers/gpu/drm/radeon/Kconfig"
 
+source "drivers/gpu/drm/nouveau/Kconfig"
+
 source "drivers/staging/octeon/Kconfig"
 
 source "drivers/staging/serqt_usb2/Kconfig"
