diff -Naur linux-2.6.32.10-old/drivers/staging/Kconfig linux-2.6.32.10-new/drivers/staging/Kconfig
--- linux-2.6.32.10-old/drivers/staging/Kconfig	2010-03-16 15:17:01.000000000 -0700
+++ linux-2.6.32.10-new/drivers/staging/Kconfig	2010-03-16 15:20:21.000000000 -0700
@@ -101,6 +101,8 @@
 
 source "drivers/gpu/drm/radeon/Kconfig"
 
+source "drivers/gpu/drm/nouveau/Kconfig"
+
 source "drivers/staging/octeon/Kconfig"
 
 source "drivers/staging/serqt_usb2/Kconfig"
