diff -Naur mythplugins-0.18.1-old/mythmusic/mythmusic/globalsettings.cpp mythplugins-0.18.1-new/mythmusic/mythmusic/globalsettings.cpp
--- mythplugins-0.18.1-old/mythmusic/mythmusic/globalsettings.cpp	2005-01-30 08:13:13.000000000 -0800
+++ mythplugins-0.18.1-new/mythmusic/mythmusic/globalsettings.cpp	2005-07-03 13:32:36.000000000 -0700
@@ -297,6 +297,16 @@
     return gc;
 };
 
+static HostCheckBox *MapFFWDAndRWND()
+{
+    HostCheckBox *gc = new HostCheckBox("MusicMapFFWDAndRWND");
+    gc->setLabel(QObject::tr("Map fast forward and rewind"));
+    gc->setValue(false);
+    gc->setHelpText(QObject::tr("If selected, then PageUp and PageDown "
+                    "will be mapped to fast forward and rewind."));
+    return gc;
+};
+
 //Player Settings
 
 static HostComboBox *PlayMode()
@@ -523,6 +533,7 @@
     playersettings->addChild(SetRandomWeight());
     playersettings->addChild(UseShowRatings());
     playersettings->addChild(UseShowWholeTree());
+    playersettings->addChild(MapFFWDAndRWND());
     playersettings->addChild(UseListShuffled());
     addChild(playersettings);
 
diff -Naur mythplugins-0.18.1-old/mythmusic/mythmusic/playbackbox.cpp mythplugins-0.18.1-new/mythmusic/mythmusic/playbackbox.cpp
--- mythplugins-0.18.1-old/mythmusic/mythmusic/playbackbox.cpp	2005-04-10 18:49:27.000000000 -0700
+++ mythplugins-0.18.1-new/mythmusic/mythmusic/playbackbox.cpp	2005-07-03 13:35:48.000000000 -0700
@@ -235,14 +235,14 @@
             else
                 previous();
         }
-        else if (action == "FFWD")
+        else if ((action =="FFWD") && gContext->GetNumSetting("MusicMapFFWDAndRWND", 1))
         { 
             if (ff_button)
                 ff_button->push();
             else
                 seekforward();
         }
-        else if (action == "RWND")
+        else if ((action =="RWND") && gContext->GetNumSetting("MusicMapFFWDAndRWND", 1))
         { 
             if (rew_button)
                 rew_button->push();
