diff -Naur libXvMCW-0.9.3-old/configure libXvMCW-0.9.3-new/configure
--- libXvMCW-0.9.3-old/configure	2004-11-18 12:26:04.000000000 -0800
+++ libXvMCW-0.9.3-new/configure	2005-04-25 17:22:24.069315870 -0700
@@ -8320,10 +8320,6 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "$x_libraries/libXv.so.1"; then
   eval "$as_ac_File=yes"
 else
@@ -8345,10 +8341,6 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "$x_libraries/libXv.so.1.0"; then
   eval "$as_ac_File=yes"
 else
@@ -8381,10 +8373,6 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "$x_libraries/libXvMC.so.1"; then
   eval "$as_ac_File=yes"
 else
@@ -8406,10 +8394,6 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "$x_libraries/libXvMC.so.1.0"; then
   eval "$as_ac_File=yes"
 else
