diff -Naur alsa-driver-1.0.15-old/Makefile alsa-driver-1.0.15-new/Makefile
--- alsa-driver-1.0.15-old/Makefile	2007-10-15 01:45:05.000000000 -0700
+++ alsa-driver-1.0.15-new/Makefile	2007-10-17 18:19:20.000000000 -0700
@@ -204,7 +204,7 @@
 endif
 	@for d in $(SUBDIRS); do if ! $(MAKE) -C $$d modules_install; then exit 1; fi; done
 ifeq ($(DESTDIR),)
-	-/sbin/depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
+	-depmod -a $(kaversion) $(SYSTEM_MAP_OPT)
 else
 	@echo "*** This is a \`staged' install using the prefix"
 	@echo "***	\"$(DESTDIR)\"."
