diff -Naur mythstream-0.18-old/mythstream.pro mythstream-0.18-new/mythstream.pro
--- mythstream-0.18-old/mythstream.pro	2007-09-23 16:52:07.000000000 -0700
+++ mythstream-0.18-new/mythstream.pro	2007-09-23 16:53:10.000000000 -0700
@@ -1,6 +1,6 @@
 TEMPLATE = subdirs
 
 # Directories
-SUBDIRS = mythstream
-#SUBDIRS = mythstream alt_parsers
+#SUBDIRS = mythstream
+SUBDIRS = mythstream alt_parsers
 
diff -Naur mythstream-0.18-old/settings.pro mythstream-0.18-new/settings.pro
--- mythstream-0.18-old/settings.pro	2007-09-23 16:52:28.000000000 -0700
+++ mythstream-0.18-new/settings.pro	2007-09-23 16:53:10.000000000 -0700
@@ -13,9 +13,9 @@
 DEFINES += SHELL=2
 DEFINES += EXEC=3
 
-DEFINES += PARSER=PERL
+#DEFINES += PARSER=PERL
 #DEFINES += PARSER=SHELL
-#DEFINES += PARSER=EXEC
+DEFINES += PARSER=EXEC
 
 DEFINES += _GNU_SOURCE
 DEFINES += PREFIX=\"$${PREFIX}\"
