diff -Naur mythstream-0.18_1-old/libs/streamstorage.cpp mythstream-0.18_1-new/libs/streamstorage.cpp
--- mythstream-0.18_1-old/libs/streamstorage.cpp	2007-09-23 11:26:15.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstorage.cpp	2007-09-26 11:27:58.000000000 -0700
@@ -146,7 +146,7 @@
           }
         }
         else
-          if ( fi->fileName() != "list.xml" ) // shouldn't be there
+          if ( fi->fileName() != "list.xml" && fi->fileName().right(3) == ".pl" ) // PERL only, leave binaries 
             copyFile(sourceDir + "/" + fi->fileName(), destDir + "/" + fi->fileName() );
         ++it;
     }
