diff -Naur unionfs-1.1.4-old/Makefile unionfs-1.1.4-new/Makefile
--- unionfs-1.1.4-old/Makefile	2006-04-11 13:24:21.000000000 -0700
+++ unionfs-1.1.4-new/Makefile	2006-04-11 17:30:45.000000000 -0700
@@ -8,6 +8,7 @@
 # This is where the make install target will put stuff
 PREFIX   = /usr/local
 MANDIR   = ${PREFIX}/man
+SBINDIR  = ${PREFIX}/sbin
 MODPREFIX=
 VERSION  = 1.1.4
 TAROPTS = czvf
@@ -49,7 +50,7 @@
 	@echo ""
 
 unionfs.ko: FRC
-	make -C ${LINUXSRC} SUBDIRS=$(PWD) FISTDEVMK=$(PWD)/fistdev.mk modules
+	make -C ${LINUXSRC} SUBDIRS=`pwd` FISTDEVMK=`pwd`/fistdev.mk modules
 
 FRC:
 
@@ -75,7 +76,7 @@
 	ctags $^
 
 clean:
-	-make -C ${LINUXSRC} SUBDIRS=$(PWD) FISTDEVMK=$(PWD)/fistdev.mk clean
+	-make -C ${LINUXSRC} SUBDIRS=`pwd` FISTDEVMK=`pwd`/fistdev.mk clean
 	rm -f ${unionfs-objs} ${obj-m} ${obj-m:.o=.ko} *.d .*.flags \#* *~
 	rm -f ${BINS} TAGS tags
 	rm -f unionfs-${VERSION}.tar${COMPEXT}
@@ -88,10 +89,10 @@
 PHONY. utils: ${BINS}
 
 install-utils: utils
-	mkdir -p ${PREFIX}/sbin
-	cp unionctl ${PREFIX}/sbin
-	cp uniondbg ${PREFIX}/sbin
-	cp unionimap ${PREFIX}/sbin
+	mkdir -p ${SBINDIR}
+	cp unionctl ${SBINDIR}
+	cp uniondbg ${SBINDIR}
+	cp unionimap ${SBINDIR}
 	mkdir -p ${MANDIR}/man4
 	mkdir -p ${MANDIR}/man8
 	cp man/unionfs.4 ${MANDIR}/man4
@@ -101,18 +102,16 @@
 install-mod: unionfs.ko
 	mkdir -p ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs
 	cp unionfs.ko ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs
-	-/sbin/depmod -a
 
 install: install-utils install-mod
 
 uninstall:
 	rm ${MODPREFIX}/${MODDIR}/kernel/fs/unionfs.ko
-	-/sbin/depmod -a
 	rm ${MANDIR}/man8/unionctl.8
 	rm ${MANDIR}/man8/uniondbg.8
 	rm ${MANDIR}/man4/unionfs.4
-	rm ${PREFIX}/sbin/unionctl
-	rm ${PREFIX}/sbin/uniondbg
+	rm ${SBINDIR}/unionctl
+	rm ${SBINDIR}/uniondbg
 
 release: unionfs-${VERSION}.tar${COMPEXT}
 
