diff -Naur unichrome-20050815-old/libxvmc/Imakefile unichrome-20050815-new/libxvmc/Imakefile
--- unichrome-20050815-old/libxvmc/Imakefile	2005-04-18 15:41:30.000000000 -0700
+++ unichrome-20050815-new/libxvmc/Imakefile	2005-12-30 16:37:08.000000000 -0800
@@ -40,6 +40,8 @@
 
 #include <Library.tmpl>
 
+LibraryObjectRule()
+
 SharedLibraryTarget(viaXvMC,$(SoRev),$(OBJS) $(EXTRASHAREDOBJS),.,.)
 SharedLibraryTarget(viaXvMCPro,$(SoRev),$(OBJSPRO) $(EXTRASHAREDOBJS),.,.)
 InstallSharedLibrary(viaXvMC,$(SoRev),$(SHLIBDIR))
