diff -Naur xscreensaver-5.09-old/hacks/glx/Makefile.in xscreensaver-5.09-new/hacks/glx/Makefile.in
--- xscreensaver-5.09-old/hacks/glx/Makefile.in	2009-09-04 08:27:40.000000000 -0700
+++ xscreensaver-5.09-new/hacks/glx/Makefile.in	2009-09-04 10:59:33.000000000 -0700
@@ -781,3 +781,4 @@
 #
 # DO NOT DELETE: updated by make distdepend
 
+molecule.o: molecules.h
diff -Naur xscreensaver-5.09-old/hacks/Makefile.in xscreensaver-5.09-new/hacks/Makefile.in
--- xscreensaver-5.09-old/hacks/Makefile.in	2009-09-04 08:27:46.000000000 -0700
+++ xscreensaver-5.09-new/hacks/Makefile.in	2009-09-04 08:30:16.000000000 -0700
@@ -968,3 +968,4 @@
 #
 # DO NOT DELETE: updated by make distdepend
 
+m6502.o: m6502.h
