diff -Naur xscreensaver-5.11-old/configure.in xscreensaver-5.11-new/configure.in
--- xscreensaver-5.11-old/configure.in	2010-01-02 03:28:15.000000000 -0800
+++ xscreensaver-5.11-new/configure.in	2010-06-22 20:19:23.000000000 -0700
@@ -3780,7 +3780,6 @@
           driver/Makefile
           hacks/Makefile
           hacks/glx/Makefile
-          po/Makefile.in
           driver/XScreenSaver.ad)
 
 ###############################################################################
