diff -Naur mythplugins-7426-old/mythbrowser/i18n/i18n.pro mythplugins-7426-new/mythbrowser/i18n/i18n.pro
--- mythplugins-7426-old/mythbrowser/i18n/i18n.pro	2005-10-25 09:20:11.000000000 -0700
+++ mythplugins-7426-new/mythbrowser/i18n/i18n.pro	2005-10-25 13:17:38.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythbrowser/mythbookmarkmanager/mythbookmarkmanager.pro mythplugins-7426-new/mythbrowser/mythbookmarkmanager/mythbookmarkmanager.pro
--- mythplugins-7426-old/mythbrowser/mythbookmarkmanager/mythbookmarkmanager.pro	2005-10-25 09:20:11.000000000 -0700
+++ mythplugins-7426-new/mythbrowser/mythbookmarkmanager/mythbookmarkmanager.pro	2005-10-25 13:17:48.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 INCLUDEPATH += /usr/include/kde
diff -Naur mythplugins-7426-old/mythbrowser/mythbrowser/mythbrowser.pro mythplugins-7426-new/mythbrowser/mythbrowser/mythbrowser.pro
--- mythplugins-7426-old/mythbrowser/mythbrowser/mythbrowser.pro	2005-10-25 09:20:11.000000000 -0700
+++ mythplugins-7426-new/mythbrowser/mythbrowser/mythbrowser.pro	2005-10-25 13:06:49.000000000 -0700
@@ -1,3 +1,4 @@
+include (../../mythconfig.mak )
 include (../../settings.pro )
 
 INCLUDEPATH += /usr/include/kde
diff -Naur mythplugins-7426-old/mythcontrols/mythcontrols/mythcontrols.pro mythplugins-7426-new/mythcontrols/mythcontrols/mythcontrols.pro
--- mythplugins-7426-old/mythcontrols/mythcontrols/mythcontrols.pro	2005-10-25 09:20:15.000000000 -0700
+++ mythplugins-7426-new/mythcontrols/mythcontrols/mythcontrols.pro	2005-10-25 13:07:52.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = lib
diff -Naur mythplugins-7426-old/mythdvd/i18n/i18n.pro mythplugins-7426-new/mythdvd/i18n/i18n.pro
--- mythplugins-7426-old/mythdvd/i18n/i18n.pro	2005-10-25 09:20:09.000000000 -0700
+++ mythplugins-7426-new/mythdvd/i18n/i18n.pro	2005-10-25 13:18:17.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythdvd/mtd/mtd.pro mythplugins-7426-new/mythdvd/mtd/mtd.pro
--- mythplugins-7426-old/mythdvd/mtd/mtd.pro	2005-10-25 09:20:10.000000000 -0700
+++ mythplugins-7426-new/mythdvd/mtd/mtd.pro	2005-10-25 13:18:46.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 include ( config.pro )
 
diff -Naur mythplugins-7426-old/mythdvd/mythdvd/mythdvd.pro mythplugins-7426-new/mythdvd/mythdvd/mythdvd.pro
--- mythplugins-7426-old/mythdvd/mythdvd/mythdvd.pro	2005-10-25 09:20:10.000000000 -0700
+++ mythplugins-7426-new/mythdvd/mythdvd/mythdvd.pro	2005-10-25 13:08:14.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 include ( config.pro )
 
diff -Naur mythplugins-7426-old/mythgallery/i18n/i18n.pro mythplugins-7426-new/mythgallery/i18n/i18n.pro
--- mythplugins-7426-old/mythgallery/i18n/i18n.pro	2005-10-25 09:20:15.000000000 -0700
+++ mythplugins-7426-new/mythgallery/i18n/i18n.pro	2005-10-25 13:18:26.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythgallery/mythgallery/mythgallery.pro mythplugins-7426-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-7426-old/mythgallery/mythgallery/mythgallery.pro	2005-10-25 09:20:15.000000000 -0700
+++ mythplugins-7426-new/mythgallery/mythgallery/mythgallery.pro	2005-10-25 13:08:33.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 include (config.pro)
diff -Naur mythplugins-7426-old/mythgame/i18n/i18n.pro mythplugins-7426-new/mythgame/i18n/i18n.pro
--- mythplugins-7426-old/mythgame/i18n/i18n.pro	2005-10-25 09:20:17.000000000 -0700
+++ mythplugins-7426-new/mythgame/i18n/i18n.pro	2005-10-25 13:19:18.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythgame/mythgame/mythgame.pro mythplugins-7426-new/mythgame/mythgame/mythgame.pro
--- mythplugins-7426-old/mythgame/mythgame/mythgame.pro	2005-10-25 09:20:17.000000000 -0700
+++ mythplugins-7426-new/mythgame/mythgame/mythgame.pro	2005-10-25 13:08:46.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = lib
diff -Naur mythplugins-7426-old/mythmusic/i18n/i18n.pro mythplugins-7426-new/mythmusic/i18n/i18n.pro
--- mythplugins-7426-old/mythmusic/i18n/i18n.pro	2005-10-25 09:20:10.000000000 -0700
+++ mythplugins-7426-new/mythmusic/i18n/i18n.pro	2005-10-25 13:19:41.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythmusic/mythmusic/mythmusic.pro mythplugins-7426-new/mythmusic/mythmusic/mythmusic.pro
--- mythplugins-7426-old/mythmusic/mythmusic/mythmusic.pro	2005-10-25 09:20:11.000000000 -0700
+++ mythplugins-7426-new/mythmusic/mythmusic/mythmusic.pro	2005-10-25 13:09:00.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 include (config.pro)
 
diff -Naur mythplugins-7426-old/mythnews/i18n/i18n.pro mythplugins-7426-new/mythnews/i18n/i18n.pro
--- mythplugins-7426-old/mythnews/i18n/i18n.pro	2005-10-25 09:20:18.000000000 -0700
+++ mythplugins-7426-new/mythnews/i18n/i18n.pro	2005-10-25 13:19:55.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythnews/mythnews/mythnews.pro mythplugins-7426-new/mythnews/mythnews/mythnews.pro
--- mythplugins-7426-old/mythnews/mythnews/mythnews.pro	2005-10-25 09:20:18.000000000 -0700
+++ mythplugins-7426-new/mythnews/mythnews/mythnews.pro	2005-10-25 13:09:15.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = lib
diff -Naur mythplugins-7426-old/mythphone/i18n/i18n.pro mythplugins-7426-new/mythphone/i18n/i18n.pro
--- mythplugins-7426-old/mythphone/i18n/i18n.pro	2005-10-25 09:20:16.000000000 -0700
+++ mythplugins-7426-new/mythphone/i18n/i18n.pro	2005-10-25 13:20:00.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythphone/mythphone/mythphone.pro mythplugins-7426-new/mythphone/mythphone/mythphone.pro
--- mythplugins-7426-old/mythphone/mythphone/mythphone.pro	2005-10-25 09:20:17.000000000 -0700
+++ mythplugins-7426-new/mythphone/mythphone/mythphone.pro	2005-10-25 13:09:24.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 include ( config.pro )
 
diff -Naur mythplugins-7426-old/mythvideo/i18n/i18n.pro mythplugins-7426-new/mythvideo/i18n/i18n.pro
--- mythplugins-7426-old/mythvideo/i18n/i18n.pro	2005-10-25 09:20:15.000000000 -0700
+++ mythplugins-7426-new/mythvideo/i18n/i18n.pro	2005-10-25 13:20:19.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythvideo/mythvideo/mythvideo.pro mythplugins-7426-new/mythvideo/mythvideo/mythvideo.pro
--- mythplugins-7426-old/mythvideo/mythvideo/mythvideo.pro	2005-10-25 09:20:16.000000000 -0700
+++ mythplugins-7426-new/mythvideo/mythvideo/mythvideo.pro	2005-10-25 13:09:33.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = lib
diff -Naur mythplugins-7426-old/mythweather/i18n/i18n.pro mythplugins-7426-new/mythweather/i18n/i18n.pro
--- mythplugins-7426-old/mythweather/i18n/i18n.pro	2005-10-25 09:20:12.000000000 -0700
+++ mythplugins-7426-new/mythweather/i18n/i18n.pro	2005-10-25 13:20:25.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = app
diff -Naur mythplugins-7426-old/mythweather/mythweather/mythweather.pro mythplugins-7426-new/mythweather/mythweather/mythweather.pro
--- mythplugins-7426-old/mythweather/mythweather/mythweather.pro	2005-10-25 09:20:14.000000000 -0700
+++ mythplugins-7426-new/mythweather/mythweather/mythweather.pro	2005-10-25 13:09:43.000000000 -0700
@@ -1,3 +1,4 @@
+include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
 
 TEMPLATE = lib
diff -Naur mythplugins-7426-old/settings.pro mythplugins-7426-new/settings.pro
--- mythplugins-7426-old/settings.pro	2005-10-25 09:20:18.000000000 -0700
+++ mythplugins-7426-new/settings.pro	2005-10-25 13:13:48.000000000 -0700
@@ -1,4 +1,3 @@
-include ( mythconfig.mak )
 CONFIG += $$CCONFIG
 
 LIBVERSION = 0.18
