diff -Naur vlc-2.0.5-old/modules/codec/avcodec/vaapi.c vlc-2.0.5-new/modules/codec/avcodec/vaapi.c
--- vlc-2.0.5-old/modules/codec/avcodec/vaapi.c	2012-10-14 13:38:05.000000000 -0700
+++ vlc-2.0.5-new/modules/codec/avcodec/vaapi.c	2015-08-02 13:59:48.084993249 -0700
@@ -46,6 +46,10 @@
 #include <libavcodec/vaapi.h>
 
 #include <X11/Xlib.h>
+#include "va/va_version.h"
+#if VA_CHECK_VERSION(0,34,0)
+#include "va/va_compat.h"
+#endif
 #include <va/va_x11.h>
 
 typedef struct
