diff -Naur vlc-1.1.6-old/modules/control/lirc.c vlc-1.1.6-new/modules/control/lirc.c
--- vlc-1.1.6-old/modules/control/lirc.c	2011-01-23 08:05:21.000000000 -0800
+++ vlc-1.1.6-new/modules/control/lirc.c	2011-01-24 14:20:01.000000000 -0800
@@ -25,6 +25,7 @@
  * Preamble
  *****************************************************************************/
 
+#include <errno.h>
 #include <fcntl.h>
 
 #ifdef HAVE_CONFIG_H
