diff -Naur kdelibs-3.5.10-old/kabc/Makefile.am kdelibs-3.5.10-new/kabc/Makefile.am
--- kdelibs-3.5.10-old/kabc/Makefile.am	2006-05-22 11:14:18.000000000 -0700
+++ kdelibs-3.5.10-new/kabc/Makefile.am	2009-05-13 06:31:45.000000000 -0700
@@ -55,6 +55,7 @@
 kab2kabc_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 kab2kabc_LDADD = libkabc.la ../kab/libkab.la
 kab2kabc_SOURCES = kab2kabc.cpp
+kab2kabc_DEPENDENCIES = addressee.h libkabc.la ../kab/libkab.la
 
 autostart_DATA = kab2kabc.desktop
 autostartdir = $(datadir)/autostart
diff -Naur kdelibs-3.5.10-old/kabc/Makefile.in kdelibs-3.5.10-new/kabc/Makefile.in
--- kdelibs-3.5.10-old/kabc/Makefile.in	2008-08-20 09:05:55.000000000 -0700
+++ kdelibs-3.5.10-new/kabc/Makefile.in	2009-05-13 06:37:25.000000000 -0700
@@ -109,7 +109,7 @@
 #>- kab2kabc_OBJECTS = $(am_kab2kabc_OBJECTS)
 #>+ 1
 kab2kabc_OBJECTS = kab2kabc.$(OBJEXT)
-kab2kabc_DEPENDENCIES = libkabc.la ../kab/libkab.la
+kab2kabc_DEPENDENCIES = addressee.h libkabc.la ../kab/libkab.la
 #>- kab2kabc_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 #>- 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
 #>- 	$(CXXFLAGS) $(kab2kabc_LDFLAGS) $(LDFLAGS) -o $@
