diff -Naur zoneinfo-2008h-old/Makefile zoneinfo-2008h-new/Makefile
--- zoneinfo-2008h-old/Makefile	2008-09-07 12:52:19.000000000 -0700
+++ zoneinfo-2008h-new/Makefile	2008-10-15 10:43:17.000000000 -0700
@@ -344,9 +344,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
