diff -Naur zoneinfo-2010e-old/Makefile zoneinfo-2010e-new/Makefile
--- zoneinfo-2010e-old/Makefile	2010-01-06 07:21:32.000000000 -0800
+++ zoneinfo-2010e-new/Makefile	2010-03-08 15:23:56.000000000 -0800
@@ -347,9 +347,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
