diff -Naur zoneinfo-2010j-old/Makefile zoneinfo-2010j-new/Makefile
--- zoneinfo-2010j-old/Makefile	2010-05-10 06:07:25.000000000 -0700
+++ zoneinfo-2010j-new/Makefile	2010-05-11 15:18:23.000000000 -0700
@@ -347,9 +347,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
