diff -Naur zoneinfo-2009a-old/Makefile zoneinfo-2009a-new/Makefile
--- zoneinfo-2009a-old/Makefile	2009-01-14 14:35:15.000000000 -0800
+++ zoneinfo-2009a-new/Makefile	2009-01-21 19:26:31.000000000 -0800
@@ -344,9 +344,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
