diff -Naur zoneinfo-2009c-old/Makefile zoneinfo-2009c-new/Makefile
--- zoneinfo-2009c-old/Makefile	2009-01-27 12:46:47.000000000 -0800
+++ zoneinfo-2009c-new/Makefile	2009-03-16 21:35:20.000000000 -0700
@@ -344,9 +344,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib -o -x /usr/bin/ranlib ] ; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
