diff -Naur Mesa-7.0.3-old/configs/linux-dri-x86 Mesa-7.0.3-new/configs/linux-dri-x86
--- Mesa-7.0.3-old/configs/linux-dri-x86	2008-04-06 21:50:59.000000000 -0700
+++ Mesa-7.0.3-new/configs/linux-dri-x86	2008-04-06 21:53:50.000000000 -0700
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86
 
diff -Naur Mesa-7.0.3-old/configs/linux-dri-x86-64 Mesa-7.0.3-new/configs/linux-dri-x86-64
--- Mesa-7.0.3-old/configs/linux-dri-x86-64	2008-04-06 21:50:59.000000000 -0700
+++ Mesa-7.0.3-new/configs/linux-dri-x86-64	2008-04-06 21:53:50.000000000 -0700
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86-64
 
