diff -Naur Mesa-7.0.1-old/configs/linux-dri-x86 Mesa-7.0.1-new/configs/linux-dri-x86
--- Mesa-7.0.1-old/configs/linux-dri-x86	2007-06-21 15:10:53.000000000 -0700
+++ Mesa-7.0.1-new/configs/linux-dri-x86	2008-02-08 23:32:35.000000000 -0800
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86
 
diff -Naur Mesa-7.0.1-old/configs/linux-dri-x86-64 Mesa-7.0.1-new/configs/linux-dri-x86-64
--- Mesa-7.0.1-old/configs/linux-dri-x86-64	2007-06-21 15:10:53.000000000 -0700
+++ Mesa-7.0.1-new/configs/linux-dri-x86-64	2008-02-08 23:32:35.000000000 -0800
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 # Configuration for linux-dri: Linux DRI hardware drivers for XFree86 & others
 
-include $(TOP)/configs/linux-dri
+include $(TOP)/configs/linux-dri-xcb
 
 CONFIG_NAME = linux-dri-x86-64
 
diff -Naur Mesa-7.0.1-old/configs/linux-dri-xcb Mesa-7.0.1-new/configs/linux-dri-xcb
--- Mesa-7.0.1-old/configs/linux-dri-xcb	2008-02-08 23:32:13.000000000 -0800
+++ Mesa-7.0.1-new/configs/linux-dri-xcb	2008-02-08 23:33:11.000000000 -0800
@@ -41,7 +41,7 @@
 LIBDRM_CFLAGS = `$(PKG_CONFIG) --cflags libdrm`
 LIBDRM_LIB = `$(PKG_CONFIG) --libs libdrm`
 DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
-GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lm -lpthread -ldl \
+GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes -lm -lpthread -ldl \
                 $(LIBDRM_LIB) `$(PKG_CONFIG) --libs xcb` `$(PKG_CONFIG) --libs x11-xcb` `$(PKG_CONFIG) --libs xcb-glx`
 
 
