diff -Naur XML-LibXML-Common-0.13-old/Makefile.PL XML-LibXML-Common-0.13-new/Makefile.PL
--- XML-LibXML-Common-0.13-old/Makefile.PL	2002-09-14 13:05:11.000000000 -0700
+++ XML-LibXML-Common-0.13-new/Makefile.PL	2008-02-16 22:14:19.000000000 -0800
@@ -83,7 +83,7 @@
     warn "win32 compile\n" if $is_Win32;
 }
 
-unless (have_library("xml2") or have_library("libxml2")) {
+unless (1 or have_library("xml2") or have_library("libxml2")) {
     die <<DEATH;
 libxml2 not found
 Try setting LIBS and INC values on the command line
