diff -Naur linux-2.6.27.19-old/arch/x86/boot/Makefile linux-2.6.27.19-new/arch/x86/boot/Makefile
--- linux-2.6.27.19-old/arch/x86/boot/Makefile	2009-03-14 14:35:25.000000000 -0700
+++ linux-2.6.27.19-new/arch/x86/boot/Makefile	2009-03-14 14:50:08.000000000 -0700
@@ -19,7 +19,7 @@
 # Set it to -DSVGA_MODE=NORMAL_VGA if you just want the EGA/VGA mode.
 # The number is the same as you would ordinarily press at bootup.
 
-SVGA_MODE := -DSVGA_MODE=NORMAL_VGA
+SVGA_MODE := -DSVGA_MODE=786
 
 # If you want the RAM disk device, define this to be the size in blocks.
 
diff -Naur linux-2.6.27.19-old/kernel/printk.c linux-2.6.27.19-new/kernel/printk.c
--- linux-2.6.27.19-old/kernel/printk.c	2009-03-14 14:35:25.000000000 -0700
+++ linux-2.6.27.19-new/kernel/printk.c	2009-03-14 14:50:08.000000000 -0700
@@ -49,7 +49,7 @@
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 3 /* anything MORE serious than KERN_ERR */
 
 DECLARE_WAIT_QUEUE_HEAD(log_wait);
 
