diff -Naur NVIDIA-Linux-x86-180.60-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-180.60-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-180.60-pkg1-old/usr/src/nv/conftest.sh	2009-05-12 12:51:13.000000000 -0700
+++ NVIDIA-Linux-x86-180.60-pkg1-new/usr/src/nv/conftest.sh	2009-06-01 16:53:18.000000000 -0700
@@ -1190,7 +1190,6 @@
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
