diff -Naur mythtv-15481-old/programs/mythfrontend/main.cpp mythtv-15481-new/programs/mythfrontend/main.cpp
--- mythtv-15481-old/programs/mythfrontend/main.cpp	2008-01-19 07:09:58.000000000 -0800
+++ mythtv-15481-new/programs/mythfrontend/main.cpp	2008-01-28 07:37:00.000000000 -0800
@@ -1395,13 +1395,24 @@
     if (gContext->GetNumSetting("NetworkControlEnabled", 0))
     {
         int networkPort = gContext->GetNumSetting("NetworkControlPort", 6545);
-        networkControl = new NetworkControl(networkPort);
+        networkControl = new NetworkControl(QHostAddress((Q_UINT32)0x00000000L), networkPort);
         if (!networkControl->ok())
             VERBOSE(VB_IMPORTANT,
                     QString("NetworkControl failed to bind to port %1.")
                     .arg(networkPort));
     }
 
+    NetworkControl *localControl = NULL;
+    if (1)
+    {
+        int networkPort = gContext->GetNumSetting("NetworkControlPort", 6545) + 1;
+        localControl = new NetworkControl(QHostAddress((Q_UINT32)0x7f000001L), networkPort);
+        if (!localControl->ok())
+            VERBOSE(VB_IMPORTANT,
+                    QString("NetworkControl failed to bind to port %1.")
+                    .arg(networkPort));
+    }
+
     gContext->addCurrentLocation("MainMenu");
 
     int exitstatus = NO_EXIT;
@@ -1452,6 +1463,9 @@
     if (networkControl)
         delete networkControl;
 
+    if (localControl)
+        delete localControl;
+
     DestroyMythMainWindow();
     delete themeBase;
     delete gContext;
diff -Naur mythtv-15481-old/programs/mythfrontend/networkcontrol.cpp mythtv-15481-new/programs/mythfrontend/networkcontrol.cpp
--- mythtv-15481-old/programs/mythfrontend/networkcontrol.cpp	2008-01-19 07:09:58.000000000 -0800
+++ mythtv-15481-new/programs/mythfrontend/networkcontrol.cpp	2008-01-28 07:37:00.000000000 -0800
@@ -37,8 +37,8 @@
         return test.lower() == command.left(test.length()).lower();
 }
 
-NetworkControl::NetworkControl(int port)
-          : QServerSocket(port, 1),
+NetworkControl::NetworkControl(const QHostAddress & address, int port)
+          : QServerSocket(address, port, 1),
             prompt("# "),
             gotAnswer(false), answer(""),
             client(NULL), cs(NULL)
diff -Naur mythtv-15481-old/programs/mythfrontend/networkcontrol.h mythtv-15481-new/programs/mythfrontend/networkcontrol.h
--- mythtv-15481-old/programs/mythfrontend/networkcontrol.h	2008-01-19 07:09:58.000000000 -0800
+++ mythtv-15481-new/programs/mythfrontend/networkcontrol.h	2008-01-28 07:37:00.000000000 -0800
@@ -19,7 +19,7 @@
 {
     Q_OBJECT
   public:
-    NetworkControl(int port);
+    NetworkControl(const QHostAddress & address, int port);
     ~NetworkControl();
 
     void newConnection(int socket);
