diff -Naur mythtv-15447-old/libs/libmythtv/tv_play.cpp mythtv-15447-new/libs/libmythtv/tv_play.cpp
--- mythtv-15447-old/libs/libmythtv/tv_play.cpp	2008-01-16 20:00:30.000000000 -0800
+++ mythtv-15447-new/libs/libmythtv/tv_play.cpp	2008-01-16 20:01:16.000000000 -0800
@@ -1730,7 +1730,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
