diff -Naur mythtv-15941-old/libs/libmythfreesurround/el_processor.cpp mythtv-15941-new/libs/libmythfreesurround/el_processor.cpp
--- mythtv-15941-old/libs/libmythfreesurround/el_processor.cpp	2008-02-12 08:48:02.000000000 -0800
+++ mythtv-15941-new/libs/libmythfreesurround/el_processor.cpp	2008-02-12 08:52:38.000000000 -0800
@@ -17,6 +17,7 @@
 */
 
 #include "el_processor.h"
+#include <cstdlib>
 #include <complex>
 #include <cmath>
 #include <vector>
diff -Naur mythtv-15941-old/libs/libmythtv/cardutil.cpp mythtv-15941-new/libs/libmythtv/cardutil.cpp
--- mythtv-15941-old/libs/libmythtv/cardutil.cpp	2008-02-12 08:47:52.000000000 -0800
+++ mythtv-15941-new/libs/libmythtv/cardutil.cpp	2008-02-12 08:51:57.000000000 -0800
@@ -1,3 +1,5 @@
+#include <algorithm>
+
 // Standard UNIX C headers
 #include <fcntl.h>
 #include <unistd.h>
diff -Naur mythtv-15941-old/programs/mythbackend/upnpmedia.cpp mythtv-15941-new/programs/mythbackend/upnpmedia.cpp
--- mythtv-15941-old/programs/mythbackend/upnpmedia.cpp	2008-02-12 08:48:03.000000000 -0800
+++ mythtv-15941-new/programs/mythbackend/upnpmedia.cpp	2008-02-12 08:51:57.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include "httprequest.h"
 #include <qfileinfo.h>
 #include <qregexp.h>
