diff -Naur mythtv-13699-old/configure mythtv-13699-new/configure
--- mythtv-13699-old/configure	2007-06-15 06:48:16.000000000 -0700
+++ mythtv-13699-new/configure	2007-06-15 06:50:22.000000000 -0700
@@ -2333,7 +2333,7 @@
 enabled amr_nb     && require libamrnb amrnb/interf_dec.h Speech_Decode_Frame_init -lamrnb -lm
 enabled amr_wb     && require libamrwb amrwb/dec_if.h D_IF_init -lamrwb -lm
 enabled liba52     && require liba52 a52dec/a52.h a52_init -la52
-enabled libdts     && require libdts dts.h dts_init $dtslib
+enabled libdts     && require libdts dts.h dca_init $dtslib
 enabled libgsm     && require libgsm gsm.h gsm_create -lgsm
 enabled libmp3lame && require LAME lame/lame.h lame_init -lmp3lame -lm
 enabled libtheora  && require libtheora theora/theora.h theora_info_init -ltheora -logg
