diff -Naur mythtv-15941-old/libs/libmythtv/tv_play.cpp mythtv-15941-new/libs/libmythtv/tv_play.cpp
--- mythtv-15941-old/libs/libmythtv/tv_play.cpp	2008-02-12 09:03:38.000000000 -0800
+++ mythtv-15941-new/libs/libmythtv/tv_play.cpp	2008-02-12 09:04:22.000000000 -0800
@@ -1745,7 +1745,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
