diff -Naur mythtv-15728-old/libs/libmyth/audiooutputdigitalencoder.cpp mythtv-15728-new/libs/libmyth/audiooutputdigitalencoder.cpp
--- mythtv-15728-old/libs/libmyth/audiooutputdigitalencoder.cpp	2008-02-02 09:18:43.000000000 -0800
+++ mythtv-15728-new/libs/libmyth/audiooutputdigitalencoder.cpp	2008-02-02 20:20:16.000000000 -0800
@@ -1,6 +1,12 @@
 // Std C headers
 #include <cstdio>
 
+// MythTV headers
+#include "config.h"
+#include "mythcontext.h"
+#include "audiooutputdigitalencoder.h"
+#include "compat.h"
+
 // libav headers
 extern "C" {
 #include "libavcodec/avcodec.h"
@@ -11,12 +17,6 @@
 #endif
 }
 
-// MythTV headers
-#include "config.h"
-#include "mythcontext.h"
-#include "audiooutputdigitalencoder.h"
-#include "compat.h"
-
 #define LOC QString("DEnc: ")
 #define LOC_ERR QString("DEnc, Error: ")
 
