diff -Naur mednafen-old/src/pce/adpcm.cpp mednafen-new/src/pce/adpcm.cpp
--- mednafen-old/src/pce/adpcm.cpp	2009-05-14 19:18:41.000000000 -0700
+++ mednafen-new/src/pce/adpcm.cpp	2010-04-19 17:39:28.000000000 -0700
@@ -244,12 +244,12 @@
           if(f >= 14)
           {
            int rolloff = (int)((((double)32087.5 / (16 - f)) / 2) * 0.70);
-           ADPCMSynth.treble_eq( blip_eq_t::blip_eq_t(-1000, rolloff, FSettings.SndRate));
+           ADPCMSynth.treble_eq( blip_eq_t(-1000, rolloff, FSettings.SndRate));
           }
           else
           {
            int rolloff = (int)((((double)32087.5 / (16 - f)) / 2) * 0.80);
-           ADPCMSynth.treble_eq( blip_eq_t::blip_eq_t(-1000, rolloff, FSettings.SndRate));
+           ADPCMSynth.treble_eq( blip_eq_t(-1000, rolloff, FSettings.SndRate));
           }
          }
 }
diff -Naur mednafen-old/src/pcfx/soundbox.cpp mednafen-new/src/pcfx/soundbox.cpp
--- mednafen-old/src/pcfx/soundbox.cpp	2009-05-14 18:31:25.000000000 -0700
+++ mednafen-new/src/pcfx/soundbox.cpp	2010-04-19 18:16:13.000000000 -0700
@@ -416,7 +416,7 @@
   int rolloff = (int)((double)0.90 * 21477272 * 2 / 1365 / (1 << SampleFreq) ) / 2;
   for(int ch = 0; ch < 2; ch++)
    for(int lr = 0; lr < 2; lr++)
-    ADPCMSynth[ch][lr].treble_eq(blip_eq_t::blip_eq_t(-1000, rolloff, FSettings.SndRate));
+    ADPCMSynth[ch][lr].treble_eq(blip_eq_t(-1000, rolloff, FSettings.SndRate));
   LSampleFreq = SampleFreq;
  }
 }
