diff -Naur mednafen-old/src/file.cpp mednafen-new/src/file.cpp
--- mednafen-old/src/file.cpp	2007-11-17 15:09:49.000000000 -0800
+++ mednafen-new/src/file.cpp	2009-06-23 09:23:45.000000000 -0700
@@ -418,8 +418,10 @@
    if(!(fceufp = MakeMemWrap(t, 0)))
     return(0);
 
-   char *ld = strrchr(path, '.');
+   char *tmp_path = strdup(path);
+   char *ld = strrchr(tmp_path, '.');
    fceufp->ext = strdup(ld ? ld + 1 : "");
+   free(tmp_path);
   }
   else                  /* Probably gzip */
   {
