diff -Naur libirman-0.4.4-old/configure.in libirman-0.4.4-new/configure.in
--- libirman-0.4.4-old/configure.in	2006-02-25 10:57:20.000000000 -0800
+++ libirman-0.4.4-new/configure.in	2006-11-20 22:24:59.000000000 -0800
@@ -7,6 +7,7 @@
 AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_RANLIB
+AC_CHECK_PROG(AR, ar, ar)
 
 dnl Checks for header files.
 AC_HEADER_STDC
diff -Naur libirman-0.4.4-old/Makefile.in libirman-0.4.4-new/Makefile.in
--- libirman-0.4.4-old/Makefile.in	2006-02-25 10:57:20.000000000 -0800
+++ libirman-0.4.4-new/Makefile.in	2006-11-20 22:26:46.000000000 -0800
@@ -10,7 +10,7 @@
 
 .PHONY: clean install-dirs install-normal install-swtest dist
 
-AR=ar
+AR = @AR@
 ARFLAGS = rc
 RANLIB = @RANLIB@
 CC = @CC@
@@ -96,7 +96,7 @@
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 
 ircmd.o: ircmd.c irman.h config.h
-	$(CC) $(CPPFLAGS) -DIR_SYSCONF_DIR=\"$(sysconfdir)\" -c $(srcdir)/ircmd.c -o ircmd.o
+	$(CC) $(CPPFLAGS) -DIR_SYSCONF_DIR=\"$(sysconfdir)\" $(CFLAGS) -c $(srcdir)/ircmd.c -o ircmd.o
 irfunc_sw.o: irfunc.c irman.h
 	$(CC) $(CPPFLAGS) -DIR_SOFTWARE_TEST $(CFLAGS) -c $(srcdir)/irfunc.c -o irfunc_sw.o
 
@@ -106,28 +106,28 @@
 	@echo notice: the test programs are only installed in the current directory \(`pwd`\)
 
 install-dirs:
-	$(top_srcdir)/mkinstalldirs $(libdir)
-	$(top_srcdir)/mkinstalldirs $(bindir)
-	$(top_srcdir)/mkinstalldirs $(sbindir)
-	$(top_srcdir)/mkinstalldirs $(includedir)
-	$(top_srcdir)/mkinstalldirs $(sysconfdir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(libdir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(bindir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sbindir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(includedir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sysconfdir)
 
 install-normal:
-	$(INSTALL) -m 755 workmanir $(bindir)/workmanir
-	$(INSTALL) -m 755 libirman.a $(libdir)/libirman.a
-	$(INSTALL) -m 644 irman.conf $(sysconfdir)/irman.conf
-	$(INSTALL) -m 644 irman.h $(includedir)/irman.h
+	$(INSTALL) -m 755 workmanir $(DESTDIR)$(bindir)/workmanir
+	$(INSTALL) -m 755 libirman.a $(DESTDIR)$(libdir)/libirman.a
+	$(INSTALL) -m 644 irman.conf $(DESTDIR)$(sysconfdir)/irman.conf
+	$(INSTALL) -m 644 irman.h $(DESTDIR)$(includedir)/irman.h
 
 install-swtest:
-	$(INSTALL) -m 755 workmanir $(bindir)/workmanir
-	$(INSTALL) -m 755 libirman_sw.a $(libdir)/libirman_sw.a
+	$(INSTALL) -m 755 workmanir $(DESTDIR)$(bindir)/workmanir
+	$(INSTALL) -m 755 libirman_sw.a $(DESTDIR)$(libdir)/libirman_sw.a
 
 uninstall:
-	$(RM) $(bindir)/workmanir
-	$(RM) $(bindir)/workmanir_sw
-	$(RM) $(libdir)/libirman.a
-	$(RM) $(libdir)/libirman_sw.a
-	$(RM) $(sysconfdir)/irman.conf
+	$(RM) $(DESTDIR)$(bindir)/workmanir
+	$(RM) $(DESTDIR)$(bindir)/workmanir_sw
+	$(RM) $(DESTDIR)$(libdir)/libirman.a
+	$(RM) $(DESTDIR)$(libdir)/libirman_sw.a
+	$(RM) $(DESTDIR)$(sysconfdir)/irman.conf
 
 dist:
 	rm -rf $(distdir)
