diff -Naur xf86-video-intel-2.4.0-old/man/intel.man xf86-video-intel-2.4.0-new/man/intel.man
--- xf86-video-intel-2.4.0-old/man/intel.man	2008-07-22 23:14:09.000000000 -0700
+++ xf86-video-intel-2.4.0-new/man/intel.man	2008-07-23 07:35:38.000000000 -0700
@@ -124,6 +124,11 @@
 Disable or enable XVideo support.
 Default: XVideo is enabled for configurations where it is supported.
 .TP
+.BI "Option \*qXvPreferOverlay\*q \*q" boolean \*q
+Make hardware overlay be first XV adapter.
+Some applications assume only one XV adapter present thus use first one.
+Default: Textured video adapter is first and overlay second.
+.TP
 .BI "Option \*qLegacy3D\*q \*q" boolean \*q
 Enable support for the legacy i915_dri.so 3D driver.
 This will, among other things, make the 2D driver tell libGL to
diff -Naur xf86-video-intel-2.4.0-old/src/i830_driver.c xf86-video-intel-2.4.0-new/src/i830_driver.c
--- xf86-video-intel-2.4.0-old/src/i830_driver.c	2008-07-22 23:14:09.000000000 -0700
+++ xf86-video-intel-2.4.0-new/src/i830_driver.c	2008-07-23 07:35:38.000000000 -0700
@@ -317,6 +317,7 @@
 #ifdef INTEL_XVMC
    OPTION_XVMC,
 #endif
+   OPTION_PREFER_OVERLAY
 } I830Opts;
 
 static OptionInfoRec I830Options[] = {
@@ -345,6 +346,7 @@
 #ifdef INTEL_XVMC
    {OPTION_XVMC,	"XvMC",		OPTV_BOOLEAN,	{0},	TRUE},
 #endif
+   {OPTION_PREFER_OVERLAY, "XvPreferOverlay", OPTV_BOOLEAN, {0}, TRUE},
    {-1,			NULL,		OPTV_NONE,	{0},	FALSE}
 };
 /* *INDENT-ON* */
@@ -1680,6 +1682,8 @@
    pI830->XvDisabled =
 	!xf86ReturnOptValBool(pI830->Options, OPTION_XVIDEO, TRUE);
 
+   pI830->XvPreferOverlay = xf86ReturnOptValBool(pI830->Options, OPTION_PREFER_OVERLAY, FALSE);
+
 #ifdef I830_XV
    if (xf86GetOptValInteger(pI830->Options, OPTION_VIDEO_KEY,
 			    &(pI830->colorKey))) {
diff -Naur xf86-video-intel-2.4.0-old/src/i830.h xf86-video-intel-2.4.0-new/src/i830.h
--- xf86-video-intel-2.4.0-old/src/i830.h	2008-07-22 23:14:09.000000000 -0700
+++ xf86-video-intel-2.4.0-new/src/i830.h	2008-07-23 07:35:38.000000000 -0700
@@ -515,6 +515,7 @@
 
    Bool XvDisabled;			/* Xv disabled in PreInit. */
    Bool XvEnabled;			/* Xv enabled for this generation. */
+   Bool XvPreferOverlay;
 
 #ifdef I830_XV
    int colorKey;
diff -Naur xf86-video-intel-2.4.0-old/src/i830_video.c xf86-video-intel-2.4.0-new/src/i830_video.c
--- xf86-video-intel-2.4.0-old/src/i830_video.c	2008-07-22 23:14:09.000000000 -0700
+++ xf86-video-intel-2.4.0-new/src/i830_video.c	2008-07-23 07:35:38.000000000 -0700
@@ -611,7 +611,6 @@
     {
 	texturedAdaptor = I830SetupImageVideoTextured(pScreen);
 	if (texturedAdaptor != NULL) {
-	    adaptors[num_adaptors++] = texturedAdaptor;
 	    xf86DrvMsg(pScrn->scrnIndex, X_INFO, "Set up textured video\n");
 	} else {
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
@@ -625,7 +624,6 @@
     {
 	overlayAdaptor = I830SetupImageVideoOverlay(pScreen);
 	if (overlayAdaptor != NULL) {
-	    adaptors[num_adaptors++] = overlayAdaptor;
 	    xf86DrvMsg(pScrn->scrnIndex, X_INFO, "Set up overlay video\n");
 	} else {
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
@@ -633,6 +631,16 @@
 	}
 	I830InitOffscreenImages(pScreen);
     }
+
+    if (overlayAdaptor && pI830->XvPreferOverlay)
+       adaptors[num_adaptors++] = overlayAdaptor;
+
+    if (texturedAdaptor)
+       adaptors[num_adaptors++] = texturedAdaptor;
+
+    if (overlayAdaptor && !pI830->XvPreferOverlay)
+       adaptors[num_adaptors++] = overlayAdaptor;
+
 #ifdef INTEL_XVMC
     if (intel_xvmc_probe(pScrn)) {
 	if (texturedAdaptor)
