diff -Naur xf86-video-intel-2.3.2-old/configure.ac xf86-video-intel-2.3.2-new/configure.ac
--- xf86-video-intel-2.3.2-old/configure.ac	2008-06-17 19:25:52.000000000 -0700
+++ xf86-video-intel-2.3.2-new/configure.ac	2008-06-17 20:38:52.000000000 -0700
@@ -39,6 +39,7 @@
 AC_PROG_LIBTOOL
 AC_PROG_CC
 AM_PROG_CC_C_O
+PKG_PROG_PKG_CONFIG
 
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
 AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
@@ -83,7 +84,8 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server xproto fontsproto $REQUIRED_MODULES])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 
