diff -Naur xorg-server-1.0.1-old/render/mitri.c xorg-server-1.0.1-new/render/mitri.c
--- xorg-server-1.0.1-old/render/mitri.c	2006-11-12 14:13:09.000000000 -0800
+++ xorg-server-1.0.1-new/render/mitri.c	2006-11-12 14:14:25.000000000 -0800
@@ -145,7 +145,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     for (tri = tris; npoint >= 3; npoint--, points++, tri++)
@@ -177,7 +177,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     first = points++;
