diff -Naur cifs-1.34a-old/fs/cifs/Makefile cifs-1.34a-new/fs/cifs/Makefile
--- cifs-1.34a-old/fs/cifs/Makefile	2005-05-03 18:13:53.000000000 -0700
+++ cifs-1.34a-new/fs/cifs/Makefile	2005-06-10 10:00:54.340524374 -0700
@@ -1,6 +1,6 @@
 #
 # Makefile for Linux CIFS VFS client 
 #
-obj-$(CONFIG_CIFS) += cifs.o
+obj-m += cifs.o
 
 cifs-objs := cifsfs.o cifssmb.o cifs_debug.o connect.o dir.o file.o inode.o link.o misc.o netmisc.o smbdes.o smbencrypt.o transport.o asn1.o md4.o md5.o cifs_unicode.o nterr.o xattr.o cifsencrypt.o fcntl.o readdir.o ioctl.o
