diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_3d.h xf86-video-openchrome-20121211-0db1c3c-new/src/via_3d.h
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_3d.h	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_3d.h	2015-07-08 16:14:56.112000018 -0700
@@ -24,6 +24,7 @@
 #ifndef VIA_3D_H
 #define VIA_3D_H
 
+#include "xorg-server.h"
 #include "xf86.h"
 #include "via_dmabuffer.h"
 
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_dri.c xf86-video-openchrome-20121211-0db1c3c-new/src/via_dri.c
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_dri.c	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_dri.c	2015-07-08 16:15:08.163000019 -0700
@@ -27,6 +27,7 @@
 #include "config.h"
 #endif
 
+#include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
 #include "xf86Priv.h"
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_driver.h xf86-video-openchrome-20121211-0db1c3c-new/src/via_driver.h
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_driver.h	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_driver.h	2015-07-08 16:15:02.307000019 -0700
@@ -36,6 +36,7 @@
 #endif
 
 #include "vgaHW.h"
+#include "xorg-server.h"
 #include "xf86.h"
 
 #if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 6
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_memmgr.c xf86-video-openchrome-20121211-0db1c3c-new/src/via_memmgr.c
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_memmgr.c	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_memmgr.c	2015-07-08 16:14:41.015000019 -0700
@@ -26,6 +26,7 @@
 #endif
 #include <sys/mman.h>
 
+#include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
 #include "xf86fbman.h"
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_memmgr.h xf86-video-openchrome-20121211-0db1c3c-new/src/via_memmgr.h
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_memmgr.h	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_memmgr.h	2015-07-08 16:14:53.138000019 -0700
@@ -22,6 +22,7 @@
  */
 #ifndef _VIA_MEMMGR_H_
 #define _VIA_MEMMGR_H_
+#include "xorg-server.h"
 #include "xf86.h"
 
 #define TTM_PL_FLAG_SYSTEM	1
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_vgahw.c xf86-video-openchrome-20121211-0db1c3c-new/src/via_vgahw.c
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_vgahw.c	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_vgahw.c	2015-07-08 16:14:45.353000019 -0700
@@ -31,6 +31,7 @@
 #endif
 
 #include "compiler.h"
+#include "xorg-server.h"
 #include "xf86.h"
 #include "via_driver.h" /* for HAVE_DEBUG */
 
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_xv.c xf86-video-openchrome-20121211-0db1c3c-new/src/via_xv.c
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_xv.c	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_xv.c	2015-07-08 16:14:48.932000019 -0700
@@ -30,6 +30,7 @@
 #include "config.h"
 #endif
 
+#include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
 
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_xvmc.c xf86-video-openchrome-20121211-0db1c3c-new/src/via_xvmc.c
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_xvmc.c	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_xvmc.c	2015-07-08 16:14:59.130000019 -0700
@@ -25,6 +25,7 @@
 #include "config.h"
 #endif
 
+#include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
 
diff -Naur xf86-video-openchrome-20121211-0db1c3c-old/src/via_xv_overlay.c xf86-video-openchrome-20121211-0db1c3c-new/src/via_xv_overlay.c
--- xf86-video-openchrome-20121211-0db1c3c-old/src/via_xv_overlay.c	2015-06-14 12:56:50.000000000 -0700
+++ xf86-video-openchrome-20121211-0db1c3c-new/src/via_xv_overlay.c	2015-07-08 16:14:34.244000019 -0700
@@ -27,6 +27,7 @@
 #include "config.h"
 #endif
 
+#include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
 
