diff -Naur XML-LibXML-1.66-old/Makefile.PL XML-LibXML-1.66-new/Makefile.PL
--- XML-LibXML-1.66-old/Makefile.PL	2008-01-29 13:05:08.000000000 -0800
+++ XML-LibXML-1.66-new/Makefile.PL	2008-02-16 22:09:59.000000000 -0800
@@ -259,7 +259,7 @@
 
 # -------------------------------------------------------------------------- #
 # test if the libraries are really installed!
-unless (have_library("xml2") or have_library("libxml2")) {
+unless (1 or have_library("xml2") or have_library("libxml2")) {
     print STDERR <<DEATH;
 libxml2 not found
 Try setting LIBS and INC values on the command line
