diff -Naur gtk+-2.14.4-old/Makefile.am gtk+-2.14.4-new/Makefile.am
--- gtk+-2.14.4-old/Makefile.am	2008-10-16 21:06:38.000000000 -0700
+++ gtk+-2.14.4-new/Makefile.am	2008-10-19 20:14:12.000000000 -0700
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
diff -Naur gtk+-2.14.4-old/Makefile.in gtk+-2.14.4-new/Makefile.in
--- gtk+-2.14.4-old/Makefile.in	2008-10-16 21:07:13.000000000 -0700
+++ gtk+-2.14.4-new/Makefile.in	2008-10-19 20:14:44.000000000 -0700
@@ -516,7 +516,7 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
