diff -Naur gtk+-2.16.2-old/Makefile.am gtk+-2.16.2-new/Makefile.am
--- gtk+-2.16.2-old/Makefile.am	2009-05-30 20:39:52.000000000 -0700
+++ gtk+-2.16.2-new/Makefile.am	2009-05-31 16:08:09.000000000 -0700
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
diff -Naur gtk+-2.16.2-old/Makefile.in gtk+-2.16.2-new/Makefile.in
--- gtk+-2.16.2-old/Makefile.in	2009-05-30 21:11:01.000000000 -0700
+++ gtk+-2.16.2-new/Makefile.in	2009-05-31 16:08:09.000000000 -0700
@@ -527,7 +527,7 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
