diff -Naur mythplugins-trunk-18866-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-18866-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-18866-old/mythvideo/mtd/dvdprobe.cpp	2008-10-24 17:01:23.000000000 -0700
+++ mythplugins-trunk-18866-new/mythvideo/mtd/dvdprobe.cpp	2008-10-24 17:01:24.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
