diff -Naur mythplugins-trunk-27024-old/configure mythplugins-trunk-27024-new/configure
--- mythplugins-trunk-27024-old/configure	2010-10-28 21:10:55.000000000 -0700
+++ mythplugins-trunk-27024-new/configure	2010-10-28 21:12:07.000000000 -0700
@@ -343,10 +343,15 @@
 
 check_py_lib(){
     log check_py_lib "$@"
+if test "x${PYTHONXCPREFIX}" = "x" ; then
+   py_prefix=${prefix}
+else
+   py_prefix=${PYTHONXCPREFIX}
+fi
     check_cmd $python <<EOF
 from distutils.sysconfig import get_python_lib
 import sys
-sys.path.append(get_python_lib(prefix='${prefix}'))
+sys.path.append(get_python_lib(prefix='${py_prefix}'))
 try:
     import ${1}
 except:
