diff -Naur mythplugins-trunk-20411-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-20411-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-20411-old/mythmusic/mythmusic/aacdecoder.cpp	2009-04-17 20:42:47.000000000 -0700
+++ mythplugins-trunk-20411-new/mythmusic/mythmusic/aacdecoder.cpp	2009-04-17 20:44:58.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
