diff -Naur mythplugins-trunk-19122-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19122-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19122-old/mythmusic/mythmusic/aacdecoder.cpp	2008-11-18 05:36:44.000000000 -0800
+++ mythplugins-trunk-19122-new/mythmusic/mythmusic/aacdecoder.cpp	2008-11-18 05:39:15.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
