diff -Naur mythplugins-trunk-19169-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19169-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19169-old/mythmusic/mythmusic/aacdecoder.cpp	2008-11-28 22:23:24.000000000 -0800
+++ mythplugins-trunk-19169-new/mythmusic/mythmusic/aacdecoder.cpp	2008-11-28 22:25:58.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
