diff -Naur mythplugins-trunk-18670-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-18670-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-18670-old/mythvideo/mtd/dvdprobe.cpp	2008-10-10 22:35:29.000000000 -0700
+++ mythplugins-trunk-18670-new/mythvideo/mtd/dvdprobe.cpp	2008-10-10 22:35:30.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
