diff -Naur mythplugins-trunk-23114-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-23114-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-23114-old/mythvideo/mtd/dvdprobe.cpp	2010-01-11 20:03:00.000000000 -0800
+++ mythplugins-trunk-23114-new/mythvideo/mtd/dvdprobe.cpp	2010-01-11 20:03:00.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
