diff -Naur mythplugins-trunk-22413-old/configure mythplugins-trunk-22413-new/configure
--- mythplugins-trunk-22413-old/configure	2009-10-12 20:42:28.000000000 -0700
+++ mythplugins-trunk-22413-new/configure	2009-10-12 20:42:29.000000000 -0700
@@ -671,6 +671,7 @@
     if test "$paranoia" = "yes" ; then
       echo "#define HAVE_PARANOIA 1" >> ./mythmusic/mythmusic/config.h
       echo "CONFIG += paranoia" >> ./mythmusic/mythmusic/config.pro
+      echo "INCLUDEPATH += $sysroot/usr/include/cdda" >> ./mythmusic/mythmusic/config.pro
     else
       echo "#undef  HAVE_PARANOIA" >> ./mythmusic/mythmusic/config.h
     fi
diff -Naur mythplugins-trunk-22413-old/mythmusic/mythmusic/mythmusic.pro mythplugins-trunk-22413-new/mythmusic/mythmusic/mythmusic.pro
--- mythplugins-trunk-22413-old/mythmusic/mythmusic/mythmusic.pro	2009-10-12 20:42:28.000000000 -0700
+++ mythplugins-trunk-22413-new/mythmusic/mythmusic/mythmusic.pro	2009-10-12 20:42:29.000000000 -0700
@@ -9,7 +9,6 @@
    error(Missing config.pro: please run the configure script)
 }
 
-INCLUDEPATH *= $${SYSROOT}/usr/include/cdda
 TEMPLATE = lib
 CONFIG += plugin thread
 TARGET = mythmusic
