diff -Naur mythplugins-trunk-18986-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-18986-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-18986-old/mythmusic/mythmusic/aacdecoder.cpp	2008-11-05 22:25:52.000000000 -0800
+++ mythplugins-trunk-18986-new/mythmusic/mythmusic/aacdecoder.cpp	2008-11-05 22:29:29.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
