diff -Naur mythplugins-trunk-20787-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-20787-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-20787-old/mythmusic/mythmusic/aacdecoder.cpp	2009-07-03 09:57:54.000000000 -0700
+++ mythplugins-trunk-20787-new/mythmusic/mythmusic/aacdecoder.cpp	2009-07-03 10:01:18.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
