diff -Naur mythplugins-trunk-17177-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17177-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17177-old/mythgallery/mythgallery/mythgallery.pro	2008-04-29 20:25:14.000000000 -0700
+++ mythplugins-trunk-17177-new/mythgallery/mythgallery/mythgallery.pro	2008-04-29 20:25:15.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
