diff -Naur mythplugins-trunk-19555-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19555-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19555-old/mythmusic/mythmusic/aacdecoder.cpp	2009-01-03 19:04:37.000000000 -0800
+++ mythplugins-trunk-19555-new/mythmusic/mythmusic/aacdecoder.cpp	2009-01-03 19:07:03.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
