diff -Naur mythplugins-trunk-20616-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-20616-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-20616-old/mythmusic/mythmusic/aacdecoder.cpp	2009-05-24 09:31:08.000000000 -0700
+++ mythplugins-trunk-20616-new/mythmusic/mythmusic/aacdecoder.cpp	2009-05-24 09:33:30.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
