diff -Naur mythplugins-trunk-23485-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-23485-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-23485-old/mythvideo/mtd/dvdprobe.cpp	2010-02-05 16:21:38.000000000 -0800
+++ mythplugins-trunk-23485-new/mythvideo/mtd/dvdprobe.cpp	2010-02-05 16:21:39.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
