diff -Naur mythplugins-trunk-20488-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-20488-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-20488-old/mythmusic/mythmusic/aacdecoder.cpp	2009-05-04 18:38:32.000000000 -0700
+++ mythplugins-trunk-20488-new/mythmusic/mythmusic/aacdecoder.cpp	2009-05-04 18:41:35.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
