diff -Naur mythplugins-trunk-17014-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17014-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17014-old/mythgallery/mythgallery/mythgallery.pro	2008-04-10 19:43:43.000000000 -0700
+++ mythplugins-trunk-17014-new/mythgallery/mythgallery/mythgallery.pro	2008-04-10 19:43:44.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
