diff -Naur mythplugins-trunk-19357-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19357-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19357-old/mythmusic/mythmusic/aacdecoder.cpp	2008-12-13 13:48:08.000000000 -0800
+++ mythplugins-trunk-19357-new/mythmusic/mythmusic/aacdecoder.cpp	2008-12-13 13:50:46.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
