diff -Naur mythplugins-trunk-19640-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19640-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19640-old/mythmusic/mythmusic/aacdecoder.cpp	2009-01-11 06:48:40.000000000 -0800
+++ mythplugins-trunk-19640-new/mythmusic/mythmusic/aacdecoder.cpp	2009-01-11 06:51:10.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
