diff -Naur mythplugins-trunk-19951-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-19951-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-19951-old/mythmusic/mythmusic/aacdecoder.cpp	2009-02-07 09:09:51.000000000 -0800
+++ mythplugins-trunk-19951-new/mythmusic/mythmusic/aacdecoder.cpp	2009-02-07 09:12:20.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
