diff -Naur mythplugins-trunk-21385-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-21385-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-21385-old/mythvideo/mtd/dvdprobe.cpp	2009-08-20 20:21:40.000000000 -0700
+++ mythplugins-trunk-21385-new/mythvideo/mtd/dvdprobe.cpp	2009-08-20 20:23:26.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
