diff -Naur mythplugins-trunk-17721-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-17721-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-17721-old/mythvideo/mtd/dvdprobe.cpp	2008-07-06 07:42:08.000000000 -0700
+++ mythplugins-trunk-17721-new/mythvideo/mtd/dvdprobe.cpp	2008-07-06 07:44:37.000000000 -0700
@@ -18,6 +18,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
