diff -Naur mythplugins-trunk-17941-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-17941-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-17941-old/mythmusic/mythmusic/aacdecoder.cpp	2008-07-25 17:52:09.000000000 -0700
+++ mythplugins-trunk-17941-new/mythmusic/mythmusic/aacdecoder.cpp	2008-07-25 17:55:05.000000000 -0700
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
