diff -Naur mythplugins-trunk-17482-old/mythgallery/mythgallery/mythgallery.pro mythplugins-trunk-17482-new/mythgallery/mythgallery/mythgallery.pro
--- mythplugins-trunk-17482-old/mythgallery/mythgallery/mythgallery.pro	2008-06-16 21:46:27.000000000 -0700
+++ mythplugins-trunk-17482-new/mythgallery/mythgallery/mythgallery.pro	2008-06-16 21:46:27.000000000 -0700
@@ -1,5 +1,6 @@
 include ( ../../mythconfig.mak )
 include ( ../../settings.pro )
+include (../../programs-libs.pro )
 
 include (config.pro)
 
