diff -Naur mythplugins-trunk-20007-old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-trunk-20007-new/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-trunk-20007-old/mythmusic/mythmusic/aacdecoder.cpp	2009-02-16 11:02:33.000000000 -0800
+++ mythplugins-trunk-20007-new/mythmusic/mythmusic/aacdecoder.cpp	2009-02-16 11:02:34.000000000 -0800
@@ -13,9 +13,6 @@
 #ifdef __STDC_LIMIT_MACROS
 #define FAAD_MODIFIED
 #endif
-#ifdef __NEAACDEC_H__
-#define FAAD_MODIFIED
-#endif
 
 // ANSI C headers
 #include <cstdlib>
