diff -Naur mythplugins-trunk-17272-old/mythvideo/mtd/dvdprobe.cpp mythplugins-trunk-17272-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-trunk-17272-old/mythvideo/mtd/dvdprobe.cpp	2008-05-05 19:44:53.000000000 -0700
+++ mythplugins-trunk-17272-new/mythvideo/mtd/dvdprobe.cpp	2008-05-05 19:47:19.000000000 -0700
@@ -18,6 +18,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 
