diff -Naur mythplugins-0.22-23003-old/mythvideo/mtd/dvdprobe.cpp mythplugins-0.22-23003-new/mythvideo/mtd/dvdprobe.cpp
--- mythplugins-0.22-23003-old/mythvideo/mtd/dvdprobe.cpp	2009-12-22 19:37:18.000000000 -0800
+++ mythplugins-0.22-23003-new/mythvideo/mtd/dvdprobe.cpp	2009-12-22 19:37:19.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
