diff -Naur alsa-driver-1.0.20-old/alsa-kernel/pci/hda/patch_analog.c alsa-driver-1.0.20-new/alsa-kernel/pci/hda/patch_analog.c
--- alsa-driver-1.0.20-old/alsa-kernel/pci/hda/patch_analog.c	2009-05-06 00:06:04.000000000 -0700
+++ alsa-driver-1.0.20-new/alsa-kernel/pci/hda/patch_analog.c	2009-05-06 21:00:30.000000000 -0700
@@ -2950,6 +2950,7 @@
 	SND_PCI_QUIRK(0x1043, 0x81f6, "Asus M2N-SLI", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8277, "Asus P5K-E/WIFI-AP", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8311, "Asus P5Q-Premium/Pro", AD1988_6STACK_DIG),
+	SND_PCI_QUIRK(0x1043, 0x8345, "Asus M3N78-VM", AD1988_6STACK_DIG),
 	{}
 };
 
