diff -Naur linux-3.3.4-old/drivers/media/rc/mceusb.c linux-3.3.4-new/drivers/media/rc/mceusb.c
--- linux-3.3.4-old/drivers/media/rc/mceusb.c	2012-04-27 15:11:48.000000000 -0700
+++ linux-3.3.4-new/drivers/media/rc/mceusb.c	2012-04-27 15:12:19.000000000 -0700
@@ -361,6 +361,8 @@
 	{ USB_DEVICE(VENDOR_FORMOSA, 0xe03c) },
 	/* Formosa Industrial Computing */
 	{ USB_DEVICE(VENDOR_FORMOSA, 0xe03e) },
+	/* Formosa Industrial Computing AIM IR606 */
+	{ USB_DEVICE(VENDOR_FORMOSA, 0xe042) },
 	/* Fintek eHome Infrared Transceiver (HP branded) */
 	{ USB_DEVICE(VENDOR_FINTEK, 0x5168) },
 	/* Fintek eHome Infrared Transceiver */
