diff -Naur xorg-server-1.6.5-old/Xext/panoramiX.h xorg-server-1.6.5-new/Xext/panoramiX.h
--- xorg-server-1.6.5-old/Xext/panoramiX.h	2009-10-11 19:52:40.000000000 -0700
+++ xorg-server-1.6.5-new/Xext/panoramiX.h	2010-10-28 18:54:23.000000000 -0700
@@ -41,7 +41,7 @@
 #define _PANORAMIX_H_
 
 #define _PANORAMIX_SERVER
-#include <X11/extensions/panoramiXext.h>
+#include <X11/extensions/panoramiXproto.h>
 #undef _PANORAMIX_SERVER
 
 #include "gcstruct.h"
