diff -Naur xorg-server-1.5.2-old/hw/xfree86/common/xf86Config.c xorg-server-1.5.2-new/hw/xfree86/common/xf86Config.c
--- xorg-server-1.5.2-old/hw/xfree86/common/xf86Config.c	2008-10-11 21:34:44.000000000 -0700
+++ xorg-server-1.5.2-new/hw/xfree86/common/xf86Config.c	2008-10-11 21:35:26.000000000 -0700
@@ -115,17 +115,6 @@
 static char *fontPath = NULL;
 
 static ModuleDefault ModuleDefaults[] = {
-    {.name = "extmod",   .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "dbe",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "glx",      .toLoad = TRUE,    .load_opt=NULL},
-    {.name = "freetype", .toLoad = TRUE,    .load_opt=NULL},
-#ifdef XRECORD
-    {.name = "record",   .toLoad = TRUE,    .load_opt=NULL},
-#endif
-    {.name = "dri",      .toLoad = TRUE,    .load_opt=NULL},
-#ifdef DRI2
-    {.name = "dri2",     .toLoad = TRUE,    .load_opt=NULL},
-#endif
     {.name = NULL,       .toLoad = FALSE,   .load_opt=NULL}
 };
 
