diff -Naur mame-0.145-old/makefile mame-0.145-new/makefile
--- mame-0.145-old/makefile	2012-02-09 06:16:39.000000000 -0800
+++ mame-0.145-new/makefile	2012-02-09 06:17:37.000000000 -0800
@@ -687,7 +687,7 @@
 # $(SRC)/emu, as well as all the OSD objects and anything in the $(OBJ) tree
 depend: maketree $(MAKEDEP_TARGET)
 	@echo Rebuilding depend.mak...
-	$(MAKEDEP) -I. $(INCPATH) -X$(SRC)/emu -X$(SRC)/osd/... -X$(OBJ)/... src/$(TARGET) > depend.mak
+	mame-makedep -I. $(INCPATH) -X$(SRC)/emu -X$(SRC)/osd/... -X$(OBJ)/... src/$(TARGET) > depend.mak
 
 tools: maketree $(TOOLS)
 
@@ -765,12 +765,12 @@
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
+	@mame-file2str  $< $@ layout_$(basename $(notdir $<))
 
 $(OBJ)/%.fh: $(SRC)/%.png $(PNG2BDC_TARGET) $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(PNG2BDC) $< $(OBJ)/temp.bdc
-	@$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
+	@mame-png2bdc $< $(OBJ)/temp.bdc
+	@mame-file2str $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
 
 $(DRIVLISTOBJ): $(DRIVLISTSRC)
 	@echo Compiling $<...
@@ -782,11 +782,11 @@
 
 $(DRIVLISTSRC): $(SRC)/$(TARGET)/$(SUBTARGET).lst $(MAKELIST_TARGET)
 	@echo Building driver list $<...
-	@$(MAKELIST) $< >$@
+	@mame-makelist $< >$@
 
 $(DEVLISTSRC): $(SRC)/$(TARGET)/$(SUBTARGET)_dev.lst $(MAKEDEV_TARGET)
 	@echo Building device list $<...
-	@$(MAKEDEV) $< >$@
+	@mame-makedev $< >$@
 
 $(OBJ)/%.a:
 	@echo Archiving $@...
diff -Naur mame-0.145-old/src/emu/cpu/cpu.mak mame-0.145-new/src/emu/cpu/cpu.mak
--- mame-0.145-old/src/emu/cpu/cpu.mak	2012-02-09 06:16:39.000000000 -0800
+++ mame-0.145-new/src/emu/cpu/cpu.mak	2012-02-09 06:17:37.000000000 -0800
@@ -1124,7 +1124,7 @@
 # rule to generate the C files
 $(CPUOBJ)/m68000/m68kops.c: $(M68KMAKE) $(CPUSRC)/m68000/m68k_in.c
 	@echo Generating M68K source files...
-	$(M68KMAKE) $(CPUOBJ)/m68000 $(CPUSRC)/m68000/m68k_in.c
+	mame-m68kmake$(EXE) $(CPUOBJ)/m68000 $(CPUSRC)/m68000/m68k_in.c
 
 # rule to build the generator
 ifneq ($(CROSS_BUILD),1)
@@ -1724,7 +1724,7 @@
 # rule to generate the C file
 $(CPUOBJ)/tms57002/tms57002.inc: $(TMSMAKE) $(CPUSRC)/tms57002/tmsinstr.lst
 	@echo Generating TMS57002 source file...
-	$(TMSMAKE) $(CPUSRC)/tms57002/tmsinstr.lst $@
+	mame-tmsmake$(EXE) $(CPUSRC)/tms57002/tmsinstr.lst $@
 
 # rule to build the generator
 ifneq ($(CROSS_BUILD),1)
