diff -Naur NVIDIA-Linux-x86-169.07-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-169.07-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-169.07-pkg1-old/usr/src/nv/conftest.sh	2007-12-13 19:35:09.000000000 -0800
+++ NVIDIA-Linux-x86-169.07-pkg1-new/usr/src/nv/conftest.sh	2007-12-20 16:47:06.000000000 -0800
@@ -852,6 +852,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
