diff -Naur xine-lib-1.1.0-old/misc/xine-config.in xine-lib-1.1.0-new/misc/xine-config.in
--- xine-lib-1.1.0-old/misc/xine-config.in	2004-11-03 10:58:29.000000000 -0800
+++ xine-lib-1.1.0-new/misc/xine-config.in	2005-07-22 14:09:08.000000000 -0700
@@ -100,11 +100,11 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@ @THREAD_CFLAGS_CONFIG@
+      echo -I${DESTDIR}@includedir@ @THREAD_CFLAGS_CONFIG@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -lxine @ZLIB_LIBS_CONFIG@ @NET_LIBS@ @THREAD_LIBS_CONFIG@ @LIBICONV@ @RT_LIBS@
+      echo -L${DESTDIR}@libdir@ -lxine @ZLIB_LIBS_CONFIG@ @NET_LIBS@ @THREAD_LIBS_CONFIG@ @LIBICONV@ @RT_LIBS@
 fi      
 if test "$echo_plugindir" = "yes"; then
       echo "@XINE_PLUGINPATH@"
