diff -Naur xine-lib-1.0.1-old/misc/xine-config.in xine-lib-1.0.1-new/misc/xine-config.in
--- xine-lib-1.0.1-old/misc/xine-config.in	2005-04-26 05:49:41.876873615 -0700
+++ xine-lib-1.0.1-new/misc/xine-config.in	2005-04-26 05:50:43.946818827 -0700
@@ -100,11 +100,11 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@ @THREAD_CFLAGS_CONFIG@
+      echo -I${DESTDIR}@includedir@ @THREAD_CFLAGS_CONFIG@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -lxine @ZLIB_LIBS_CONFIG@ @NET_LIBS@ @THREAD_LIBS_CONFIG@ @LIBICONV@ @RT_LIBS@
+      echo -L${DESTDIR}@libdir@ -lxine @ZLIB_LIBS_CONFIG@ @NET_LIBS@ @THREAD_LIBS_CONFIG@ @LIBICONV@ @RT_LIBS@
 fi      
 if test "$echo_plugindir" = "yes"; then
       echo "@XINE_PLUGINPATH@"
