diff -Naur pixman-0.21.4-old/configure.ac pixman-0.21.4-new/configure.ac
--- pixman-0.21.4-old/configure.ac	2011-01-19 04:29:18.000000000 -0800
+++ pixman-0.21.4-new/configure.ac	2011-01-19 13:57:09.000000000 -0800
@@ -150,7 +150,7 @@
 # if we're using Sun Studio and neither the user nor a config.site
 # has set CFLAGS.
 if test $SUNCC = yes &&			\
-   test "$test_CFLAGS" == "" &&		\
+   test "$test_CFLAGS" = "" &&		\
    test "$CFLAGS" = "-g"
 then
   CFLAGS="-O -g"
