diff -Naur pixman-0.19.4-old/configure.ac pixman-0.19.4-new/configure.ac
--- pixman-0.19.4-old/configure.ac	2010-09-21 07:11:23.000000000 -0700
+++ pixman-0.19.4-new/configure.ac	2010-09-21 09:55:54.000000000 -0700
@@ -150,7 +150,7 @@
 # if we're using Sun Studio and neither the user nor a config.site
 # has set CFLAGS.
 if test $SUNCC = yes &&			\
-   test "$test_CFLAGS" == "" &&		\
+   test "$test_CFLAGS" = "" &&		\
    test "$CFLAGS" = "-g"
 then
   CFLAGS="-O -g"
