diff -Naur NVIDIA-Linux-x86_64-169.04-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-169.04-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-169.04-pkg2-old/usr/src/nv/conftest.sh	2007-11-14 16:43:30.000000000 -0800
+++ NVIDIA-Linux-x86_64-169.04-pkg2-new/usr/src/nv/conftest.sh	2007-12-19 13:36:55.000000000 -0800
@@ -849,6 +849,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
