diff -Naur linux-2.6.23.1-old/arch/i386/kernel/cpu/cpufreq/longhaul.c linux-2.6.23.1-new/arch/i386/kernel/cpu/cpufreq/longhaul.c
--- linux-2.6.23.1-old/arch/i386/kernel/cpu/cpufreq/longhaul.c	2007-10-17 17:34:06.000000000 -0700
+++ linux-2.6.23.1-new/arch/i386/kernel/cpu/cpufreq/longhaul.c	2007-10-17 17:36:32.000000000 -0700
@@ -29,6 +29,7 @@
 #include <linux/pci.h>
 #include <linux/slab.h>
 #include <linux/string.h>
+#include <linux/ide.h>
 #include <linux/delay.h>
 
 #include <asm/msr.h>
@@ -108,6 +109,24 @@
 }
 #endif
 
+static void ide_idle(void)
+{
+	int i;
+	ide_hwif_t *hwif = ide_hwifs;
+	ide_drive_t *drive;
+
+	i = 0;
+	do {
+		drive = &hwif->drives[i];
+		i++;
+		if (strncmp(drive->name,"hd",2) == 0) {
+			while (drive->waiting_for_dma)
+			udelay(10);
+		} else {
+			i = 0;
+		}
+	} while (i != 0);
+}
 
 static unsigned int calc_speed(int mult)
 {
@@ -278,6 +297,7 @@
 			fsb, mult/10, mult%10, print_speed(speed/1000));
 retry_loop:
 	preempt_disable();
+	ide_idle(); /* avoid ide timeouts when bus master off */
 	local_irq_save(flags);
 
 	pic2_mask = inb(0xA1);
