diff -Naur zsnes-1.51-old/src/parsegen.cpp zsnes-1.51-new/src/parsegen.cpp
--- zsnes-1.51-old/src/parsegen.cpp	2007-10-30 21:30:26.000000000 -0700
+++ zsnes-1.51-new/src/parsegen.cpp	2008-01-16 09:08:57.000000000 -0800
@@ -33,6 +33,7 @@
 #include <stack>
 using namespace std;
 
+#include <string.h>
 #include <errno.h>
 #include <zlib.h>
 
@@ -1822,7 +1823,7 @@
   }
 }
 
-int main(size_t argc, const char *const *const argv)
+int main(int argc, const char *const *const argv)
 {
   const char *cheader_file = 0;
   bool compile = false;
diff -Naur zsnes-1.51-old/src/tools/depbuild.cpp zsnes-1.51-new/src/tools/depbuild.cpp
--- zsnes-1.51-old/src/tools/depbuild.cpp	2006-12-27 03:04:05.000000000 -0800
+++ zsnes-1.51-new/src/tools/depbuild.cpp	2008-01-16 09:07:05.000000000 -0800
@@ -21,6 +21,7 @@
 This program generates dependencies for all C/C++/Assembly files
 */
 
+#include <stdlib.h>
 #include <iostream>
 #include <string>
 #include <cstdio>
@@ -183,7 +184,7 @@
   }
 }
 
-int main(size_t argc, const char *const *const argv)
+int main(int argc, const char *const *const argv)
 {
   if (argc < 5)
   {
diff -Naur zsnes-1.51-old/src/tools/strutil.cpp zsnes-1.51-new/src/tools/strutil.cpp
--- zsnes-1.51-old/src/tools/strutil.cpp	2006-12-27 03:04:05.000000000 -0800
+++ zsnes-1.51-new/src/tools/strutil.cpp	2008-01-16 09:04:51.000000000 -0800
@@ -19,6 +19,7 @@
 This is part of a toolkit used to assist in ZSNES development
 */
 
+#include <string.h>
 #include "strutil.h"
 using namespace std;
 
