diff -Naur NVIDIA-Linux-x86_64-96.43.07-pkg2-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-96.43.07-pkg2-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-96.43.07-pkg2-old/usr/src/nv/conftest.sh	2008-07-02 12:45:45.000000000 -0700
+++ NVIDIA-Linux-x86_64-96.43.07-pkg2-new/usr/src/nv/conftest.sh	2008-07-27 10:29:14.000000000 -0700
@@ -883,7 +883,6 @@
             #
             echo "#include <linux/autoconf.h>
             #include <linux/version.h>
-            #include <linux/utsname.h>
             int main() {
               if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) {
                 return 0;
