diff -Naur linux-uvc-113-old/Makefile linux-uvc-113-new/Makefile
--- linux-uvc-113-old/Makefile	2007-07-10 12:17:18.000000000 -0700
+++ linux-uvc-113-new/Makefile	2007-07-10 12:33:28.000000000 -0700
@@ -16,7 +16,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Modules.symvers
