diff -Naur mythplugins-0.18-old/mythmusic/mythmusic/main.cpp mythplugins-0.18-new/mythmusic/mythmusic/main.cpp
--- mythplugins-0.18-old/mythmusic/mythmusic/main.cpp	2005-04-24 10:31:38.761047147 -0700
+++ mythplugins-0.18-new/mythmusic/mythmusic/main.cpp	2005-04-24 11:07:52.312256606 -0700
@@ -379,8 +379,6 @@
     REG_KEY("Music", "DELETE", "Delete track from playlist", "D");
     REG_KEY("Music", "NEXTTRACK", "Move to the next track", ">,.,Z,End");
     REG_KEY("Music", "PREVTRACK", "Move to the previous track", ",,<,Q,Home");
-    REG_KEY("Music", "FFWD", "Fast forward", "PgDown");
-    REG_KEY("Music", "RWND", "Rewind", "PgUp");
     REG_KEY("Music", "PAUSE", "Pause/Start playback", "P");
     REG_KEY("Music", "STOP", "Stop playback", "O");
     REG_KEY("Music", "VOLUMEDOWN", "Volume down", "[,{,F10");
