diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/ChangeLog qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/ChangeLog
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/ChangeLog	2010-08-22 10:51:36.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/ChangeLog	2010-08-22 10:55:58.000000000 -0700
@@ -1,3 +1,20 @@
+2010-06-30  Andreas Kling  <andreas.kling@nokia.com>
+
+        Reviewed by Antti Koivisto.
+
+        [Qt, Gtk, Symbian] Mind the semantics of NPRect for windowed plugin clip rects
+        https://bugs.webkit.org/show_bug.cgi?id=41406
+
+        NPRect has right/bottom instead of width/height.
+        This was fixed in PluginViewMac with http://trac.webkit.org/changeset/45815
+
+        * plugins/gtk/PluginViewGtk.cpp:
+        (WebCore::PluginView::setNPWindowIfNeeded):
+        * plugins/qt/PluginViewQt.cpp:
+        (WebCore::PluginView::setNPWindowIfNeeded):
+        * plugins/symbian/PluginViewSymbian.cpp:
+        (WebCore::PluginView::setNPWindowIfNeeded):
+
 2010-06-20  Robert Hogan  <robert@webkit.org>
 
         Reviewed by Simon Hausmann.
diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp	2010-08-22 10:51:36.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/qt/PluginViewQt.cpp	2010-08-22 10:54:56.000000000 -0700
@@ -501,10 +501,10 @@
         m_npWindow.x = m_windowRect.x();
         m_npWindow.y = m_windowRect.y();
 
-        m_npWindow.clipRect.left = m_clipRect.x();
-        m_npWindow.clipRect.top = m_clipRect.y();
-        m_npWindow.clipRect.right = m_clipRect.width();
-        m_npWindow.clipRect.bottom = m_clipRect.height();
+        m_npWindow.clipRect.left = max(0, m_clipRect.x());
+        m_npWindow.clipRect.top = max(0, m_clipRect.y());
+        m_npWindow.clipRect.right = m_clipRect.x() + m_clipRect.width();
+        m_npWindow.clipRect.bottom = m_clipRect.y() + m_clipRect.height();
     } else {
         m_npWindow.x = 0;
         m_npWindow.y = 0;
diff -Naur qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/symbian/PluginViewSymbian.cpp qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/symbian/PluginViewSymbian.cpp
--- qt-everywhere-opensource-src-4.6.3-old/src/3rdparty/webkit/WebCore/plugins/symbian/PluginViewSymbian.cpp	2010-08-22 10:50:43.000000000 -0700
+++ qt-everywhere-opensource-src-4.6.3-new/src/3rdparty/webkit/WebCore/plugins/symbian/PluginViewSymbian.cpp	2010-08-22 10:54:56.000000000 -0700
@@ -264,10 +264,10 @@
         m_npWindow.x = m_windowRect.x();
         m_npWindow.y = m_windowRect.y();
 
-        m_npWindow.clipRect.left = m_clipRect.x();
-        m_npWindow.clipRect.top = m_clipRect.y();
-        m_npWindow.clipRect.right = m_clipRect.width();
-        m_npWindow.clipRect.bottom = m_clipRect.height();
+        m_npWindow.clipRect.left = max(0, m_clipRect.x());
+        m_npWindow.clipRect.top = max(0, m_clipRect.y());
+        m_npWindow.clipRect.right = m_clipRect.x() + m_clipRect.width();
+        m_npWindow.clipRect.bottom = m_clipRect.y() + m_clipRect.height();
     
     } else {
         // always call this method before painting.
