diff -Naur libXvMC-1.0.5-old/src/XvMCWrapper.c libXvMC-1.0.5-new/src/XvMCWrapper.c
--- libXvMC-1.0.5-old/src/XvMCWrapper.c	2009-02-02 20:46:43.000000000 -0800
+++ libXvMC-1.0.5-new/src/XvMCWrapper.c	2010-08-10 14:17:04.000000000 -0700
@@ -44,7 +44,9 @@
   #include "config.h"
 #endif
 
+#include <X11/Xlib.h>
 #include <X11/extensions/XvMC.h>
+#include <X11/extensions/XvMClib.h>
 #include <X11/extensions/vldXvMC.h>
 #include <dlfcn.h>
 #include <stdio.h>
