diff -Naur mythstream_mythtv-r21640-old/libs/harvester.h mythstream_mythtv-r21640-new/libs/harvester.h
--- mythstream_mythtv-r21640-old/libs/harvester.h	2009-10-13 14:49:54.000000000 -0700
+++ mythstream_mythtv-r21640-new/libs/harvester.h	2009-10-13 14:50:38.000000000 -0700
@@ -23,7 +23,7 @@
 #include <q3process.h>
 #include <qthread.h>
 #include <q3process.h>
-#include <qhttp.h>
+#include <QtNetwork/qhttp.h>
 #include <q3dict.h>
 #include <qmap.h>
 //Added by qt3to4:
