diff -Naur pkg-config-0.23-old/pkg.m4 pkg-config-0.23-new/pkg.m4
--- pkg-config-0.23-old/pkg.m4	2008-01-22 15:00:31.000000000 -0800
+++ pkg-config-0.23-new/pkg.m4	2008-01-22 15:06:35.000000000 -0800
@@ -25,7 +25,15 @@
 # ----------------------------------
 AC_DEFUN([PKG_PROG_PKG_CONFIG],
 [m4_pattern_forbid([^_?PKG_[A-Z_]+$])
-m4_pattern_allow([^PKG_CONFIG(_PATH)?$])
+m4_pattern_allow([^PKG_CONFIG$])
+m4_pattern_allow([^PKG_CONFIG_ALLOW_SYSTEM_CFLAGS$])
+m4_pattern_allow([^PKG_CONFIG_ALLOW_SYSTEM_LIBS$])
+m4_pattern_allow([^PKG_CONFIG_DEBUG_SPEW$])
+m4_pattern_allow([^PKG_CONFIG_DISABLE_UNINSTALLED$])
+m4_pattern_allow([^PKG_CONFIG_LIBDIR$])
+m4_pattern_allow([^PKG_CONFIG_PATH$])
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+m4_pattern_allow([^PKG_CONFIG_TOP_BUILD_DIR$])
 AC_ARG_VAR([PKG_CONFIG], [path to pkg-config utility])dnl
 if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
 	AC_PATH_TOOL([PKG_CONFIG], [pkg-config])
