diff -Naur openldap-2.3.39-old/clients/tools/Makefile.in openldap-2.3.39-new/clients/tools/Makefile.in
--- openldap-2.3.39-old/clients/tools/Makefile.in	2007-01-02 13:43:41.000000000 -0800
+++ openldap-2.3.39-new/clients/tools/Makefile.in	2007-10-27 15:12:35.000000000 -0700
@@ -98,7 +98,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
diff -Naur openldap-2.3.39-old/servers/slapd/Makefile.in openldap-2.3.39-new/servers/slapd/Makefile.in
--- openldap-2.3.39-old/servers/slapd/Makefile.in	2007-02-14 07:59:43.000000000 -0800
+++ openldap-2.3.39-new/servers/slapd/Makefile.in	2007-10-27 15:12:35.000000000 -0700
@@ -375,7 +375,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i && test -f $$i/Makefile ; then \
diff -Naur openldap-2.3.39-old/servers/slurpd/Makefile.in openldap-2.3.39-new/servers/slurpd/Makefile.in
--- openldap-2.3.39-old/servers/slurpd/Makefile.in	2007-01-02 13:44:11.000000000 -0800
+++ openldap-2.3.39-new/servers/slurpd/Makefile.in	2007-10-27 15:12:35.000000000 -0700
@@ -60,7 +60,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
