diff -Naur libcdaudio-0.99.12-old/libcdaudio-config.in libcdaudio-0.99.12-new/libcdaudio-config.in
--- libcdaudio-0.99.12-old/libcdaudio-config.in	2003-02-10 12:46:14.000000000 -0800
+++ libcdaudio-0.99.12-new/libcdaudio-config.in	2004-12-30 08:51:07.283984157 -0800
@@ -39,7 +39,7 @@
 		echo @LIBCDAUDIO_VERSION@
 	;;
 	--cflags)
-		includedir=${prefix}/include
+		includedir=${DESTDIR}${prefix}/include
 		if test $includedir != /usr/include ; then
 			includes=-I$includedir
 		fi
@@ -49,7 +49,7 @@
 		echo @LIB_LDADD@
 	;;
 	--libs)
-		libdir=-L${exec_prefix}/lib
+		libdir=-L${DESTDIR}${exec_prefix}/lib
 		echo $libdir -lcdaudio @LIBS@
 	;;
 	*)
