diff -Naur gdk-pixbuf-2.24.0-old/configure.ac gdk-pixbuf-2.24.0-new/configure.ac
--- gdk-pixbuf-2.24.0-old/configure.ac	2011-08-26 20:27:52.000000000 -0700
+++ gdk-pixbuf-2.24.0-new/configure.ac	2012-01-10 16:24:54.000000000 -0800
@@ -147,7 +147,7 @@
 AC_SYS_LARGEFILE
 
 AM_PROG_AS
-AC_PATH_PROG(NM, nm, nm)
+AC_PATH_TOOL(NM, nm, nm)
 
 AC_MSG_CHECKING([for some Win32 platform])
 AS_CASE(["$host"],
