diff -Naur gdk-pixbuf-2.26.1-old/configure.ac gdk-pixbuf-2.26.1-new/configure.ac
--- gdk-pixbuf-2.26.1-old/configure.ac	2012-04-14 11:28:48.000000000 -0700
+++ gdk-pixbuf-2.26.1-new/configure.ac	2012-05-02 18:44:13.000000000 -0700
@@ -147,7 +147,7 @@
 AC_SYS_LARGEFILE
 
 AM_PROG_AS
-AC_PATH_PROG(NM, nm, nm)
+AC_PATH_TOOL(NM, nm, nm)
 
 AC_MSG_CHECKING([for some Win32 platform])
 AS_CASE(["$host"],
