diff -Naur e2fsprogs-1.37-old/lib/Makefile.elf-lib e2fsprogs-1.37-new/lib/Makefile.elf-lib
--- e2fsprogs-1.37-old/lib/Makefile.elf-lib	2005-05-03 16:18:49.095099976 -0700
+++ e2fsprogs-1.37-new/lib/Makefile.elf-lib	2005-05-03 16:19:12.228491134 -0700
@@ -47,7 +47,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
