diff -Naur e2fsprogs-1.40.9-old/lib/Makefile.elf-lib e2fsprogs-1.40.9-new/lib/Makefile.elf-lib
--- e2fsprogs-1.40.9-old/lib/Makefile.elf-lib	2008-01-01 13:47:01.000000000 -0800
+++ e2fsprogs-1.40.9-new/lib/Makefile.elf-lib	2008-04-27 07:08:23.000000000 -0700
@@ -45,7 +45,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
