diff -Naur e2fsprogs-1.38-old/configure.in e2fsprogs-1.38-new/configure.in
--- e2fsprogs-1.38-old/configure.in	2005-04-06 11:44:15.000000000 -0700
+++ e2fsprogs-1.38-new/configure.in	2006-01-04 10:46:36.000000000 -0800
@@ -571,14 +571,6 @@
 dnl
 dnl Word sizes...
 dnl
-if test "$cross_compiling" = yes -a "$ac_cv_sizeof_long" = ""; then
-  # if cross-compiling, with no cached values, just assume something common. 
-  ac_cv_sizeof_short=2
-  ac_cv_sizeof_int=4
-  ac_cv_sizeof_long=4
-  ac_cv_sizeof_long_long=8
-  AC_MSG_WARN([Cross-compiling, so cannot check type sizes; assuming short=2, int=4, long=4, long long=8])
-fi
 AC_CHECK_SIZEOF(short)
 AC_CHECK_SIZEOF(int)
 AC_CHECK_SIZEOF(long)
