diff -Naur glibc-2.9-old/sysdeps/i386/dl-tlsdesc.S glibc-2.9-new/sysdeps/i386/dl-tlsdesc.S
--- glibc-2.9-old/sysdeps/i386/dl-tlsdesc.S	2008-05-12 22:33:06.000000000 -0700
+++ glibc-2.9-new/sysdeps/i386/dl-tlsdesc.S	2009-05-02 06:19:22.000000000 -0700
@@ -128,8 +128,7 @@
 .Lslow:
 	cfi_adjust_cfa_offset (28)
 	movl	%ebx, 16(%esp)
-	call	__i686.get_pc_thunk.bx
-	addl	$_GLOBAL_OFFSET_TABLE_, %ebx
+	LOAD_PIC_REG(bx)
 	call	___tls_get_addr@PLT
 	movl	16(%esp), %ebx
 	jmp	.Lret
