diff -Naur glibc-2.6-old/sysdeps/unix/sysv/linux/i386/sync_file_range.S glibc-2.6-new/sysdeps/unix/sysv/linux/i386/sync_file_range.S
--- glibc-2.6-old/sysdeps/unix/sysv/linux/i386/sync_file_range.S	2006-03-31 13:47:34.000000000 -0800
+++ glibc-2.6-new/sysdeps/unix/sysv/linux/i386/sync_file_range.S	2007-06-28 18:39:02.000000000 -0700
@@ -63,10 +63,10 @@
 
 	cmpl	$-4095, %eax
 	jae	SYSCALL_ERROR_LABEL
-L(pseudo_end):
-	ret
 #else
 	movl	$-ENOSYS, %eax
 	jmp	SYSCALL_ERROR_LABEL
 #endif
+L(pseudo_end):
+	ret
 PSEUDO_END (sync_file_range)
