diff -Naur glibc-2.4-old/configure glibc-2.4-new/configure
--- glibc-2.4-old/configure	2006-03-06 03:18:56.000000000 -0800
+++ glibc-2.4-new/configure	2006-03-29 07:17:07.000000000 -0800
@@ -3856,15 +3856,15 @@
     path_binutils=`(cd $path_binutils; pwd) | sed 's%/*$%/%'`
     CC="$CC -B$path_binutils"
 fi
-AS=`$CC -print-prog-name=as`
-LD=`$CC -print-prog-name=ld`
-AR=`$CC -print-prog-name=ar`
+AS=${AS}
+LD=${LD}
+AR=${AR}
 
-OBJDUMP=`$CC -print-prog-name=objdump`
+OBJDUMP=${OBJDUMP}
 
 
 # ranlib has to be treated a bit differently since it might not exist at all.
-ac_ranlib=`$CC -print-prog-name=ranlib`
+ac_ranlib=${RANLIB}
 if test "x$ac_ranlib" = xranlib; then
 # This extra check has to happen since gcc simply echos the parameter in
 # case it cannot find the value in its own directories.
