diff -Naur glibc-2.21-old/sunrpc/Makefile glibc-2.21-new/sunrpc/Makefile
--- glibc-2.21-old/sunrpc/Makefile	2015-02-05 22:40:18.000000000 -0800
+++ glibc-2.21-new/sunrpc/Makefile	2015-07-01 12:45:15.579000413 -0700
@@ -54,7 +54,7 @@
 				    des_crypt.h)
 headers-not-in-tirpc = $(addprefix rpc/,key_prot.h rpc_des.h) \
 		       $(rpcsvc:%=rpcsvc/%) rpcsvc/bootparam.h
-headers = rpc/netdb.h
+headers = rpc/key_prot.h rpc/netdb.h
 install-others = $(inst_sysconfdir)/rpc
 generated += $(rpcsvc:%.x=rpcsvc/%.h) $(rpcsvc:%.x=x%.c) $(rpcsvc:%.x=x%.stmp) \
 	     $(rpcsvc:%.x=rpcsvc/%.stmp) rpcgen
