diff -Naur glibc-2.9-old/resolv/Versions glibc-2.9-new/resolv/Versions
--- glibc-2.9-old/resolv/Versions	2008-08-01 10:15:34.000000000 -0700
+++ glibc-2.9-new/resolv/Versions	2009-04-27 10:41:36.000000000 -0700
@@ -102,7 +102,7 @@
     _nss_dns_gethostbyname_r; _nss_dns_getnetbyaddr_r;
     _nss_dns_getnetbyname_r; _nss_dns_getcanonname_r;
     _nss_dns_gethostbyaddr2_r;
-    _nss_dns_gethostbyname4_r;
+#    _nss_dns_gethostbyname4_r;
   }
 }
 
