diff -Naur mythtv-0.24-27235-old/libs/libmythtv/util-xvmc.h mythtv-0.24-27235-new/libs/libmythtv/util-xvmc.h
--- mythtv-0.24-27235-old/libs/libmythtv/util-xvmc.h	2010-11-16 11:42:25.000000000 -0800
+++ mythtv-0.24-27235-new/libs/libmythtv/util-xvmc.h	2010-11-16 11:42:26.000000000 -0800
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(14),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
