diff -Naur mythtv-0.24-27113-old/configure mythtv-0.24-27113-new/configure
--- mythtv-0.24-27113-old/configure	2010-11-07 06:02:45.000000000 -0800
+++ mythtv-0.24-27113-new/configure	2010-11-07 06:02:46.000000000 -0800
@@ -4740,8 +4740,8 @@
 #       -L from early in the link line, to ignore any old installed libmyth*
 # Note: Qt uses QMAKE_INCDIR_QT to build an absolute path for include
 #       directories, it cannot be modified.
-QMAKE_LIBDIR_QT-=$libdir
-LATE_LIBS+=-L$libdir
+QMAKE_LIBDIR_QT-=${sysroot}$libdir
+LATE_LIBS+=-L${sysroot}$libdir
 EOF
 
 #echo "endif # FFMPEG_CONFIG_MAK" >> config.mak
