diff -Naur mythtv-0.24-27393-old/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-27393-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-27393-old/libs/libmythtv/avformatdecoder.cpp	2010-11-30 17:53:57.000000000 -0800
+++ mythtv-0.24-27393-new/libs/libmythtv/avformatdecoder.cpp	2010-11-30 17:56:05.000000000 -0800
@@ -1257,6 +1257,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
