diff -Naur mythtv-0.24-20110829-1de0431-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-20110829-1de0431-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-20110829-1de0431-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-08-30 09:05:09.000000000 -0700
+++ mythtv-0.24-20110829-1de0431-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-08-30 09:07:33.000000000 -0700
@@ -1305,6 +1305,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
