diff -Naur mythtv-0.24-20110302-5367795-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-20110302-5367795-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-20110302-5367795-931028f-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-03-03 07:08:57.000000000 -0800
+++ mythtv-0.24-20110302-5367795-931028f-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-03-03 07:11:28.000000000 -0800
@@ -1305,6 +1305,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
