diff -Naur mythtv-0.24-20110214-316718a-b6f0080-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-20110214-316718a-b6f0080-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-20110214-316718a-b6f0080-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-02-14 21:27:20.000000000 -0800
+++ mythtv-0.24-20110214-316718a-b6f0080-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2011-02-14 21:27:29.000000000 -0800
@@ -1305,6 +1305,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
