diff -Naur mythtv-0.24-20120522-fe9bd6d-931028f-old/mythtv/libs/libmythtv/util-xvmc.h mythtv-0.24-20120522-fe9bd6d-931028f-new/mythtv/libs/libmythtv/util-xvmc.h
--- mythtv-0.24-20120522-fe9bd6d-931028f-old/mythtv/libs/libmythtv/util-xvmc.h	2012-05-24 09:27:14.000000000 -0700
+++ mythtv-0.24-20120522-fe9bd6d-931028f-new/mythtv/libs/libmythtv/util-xvmc.h	2012-05-24 09:27:16.000000000 -0700
@@ -53,7 +53,7 @@
         num_xvmc_surf(1),
         needed_for_display(1),
         min_num_xvmc_surfaces(8),
-        max_num_xvmc_surfaces(16),
+        max_num_xvmc_surfaces(14),
         num_xvmc_surfaces(min_num_xvmc_surfaces),
         aggressive(false) {}
 
