diff -Naur mythtv-0.24-26906-old/libs/libmythtv/avformatdecoder.cpp mythtv-0.24-26906-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.24-26906-old/libs/libmythtv/avformatdecoder.cpp	2010-10-19 20:08:25.000000000 -0700
+++ mythtv-0.24-26906-new/libs/libmythtv/avformatdecoder.cpp	2010-10-19 20:10:36.000000000 -0700
@@ -1256,6 +1256,9 @@
                 .arg(ff_codec_id_string(enc->codec_id)));
     }
 
+    if (enc->codec_id == CODEC_ID_SVQ3)
+        enc->thread_count = 1;
+
     if (special_decode)
     {
         if (special_decode & kAVSpecialDecode_SingleThreaded)
