diff -Naur bdremote-ng-0.5-old/src/bdrcfg.c bdremote-ng-0.5-new/src/bdrcfg.c
--- bdremote-ng-0.5-old/src/bdrcfg.c	2009-11-19 15:00:04.000000000 -0800
+++ bdremote-ng-0.5-new/src/bdrcfg.c	2009-11-19 15:13:55.000000000 -0800
@@ -54,6 +54,7 @@
   _config->remote_addr = NULL;
   _config->detach      = 1;
   FREEVAL(_config->release);
+  _config->lirc_namespace = 0;
   _config->release     = NULL;
   FREEVAL(_config->user);
   _config->user        = NULL;
@@ -138,6 +139,7 @@
     {
       fprintf(printStream, " - release     : %s.\n", _config->release);
     }
+  fprintf(printStream, " - LIRC names  : %d.\n", _config->lirc_namespace);
   fprintf(printStream, " - debug       : %d.\n", _config->debug);
   if (_config->debug)
     {
diff -Naur bdremote-ng-0.5-old/src/bdrcfg.h bdremote-ng-0.5-new/src/bdrcfg.h
--- bdremote-ng-0.5-old/src/bdrcfg.h	2009-11-19 14:54:29.000000000 -0800
+++ bdremote-ng-0.5-new/src/bdrcfg.h	2009-11-19 15:05:24.000000000 -0800
@@ -50,6 +50,8 @@
   int repeat_delay;
   /** Generate key release appended with the following string */
   char* release;
+  /** Enable/disable use of LIRC namespace. */
+  int lirc_namespace;
   /** Enable/disable printing of debug messages. */
   int debug;
   /** Indicates if the BT address of the interface to use was set. */
diff -Naur bdremote-ng-0.5-old/src/bdremoteng.c bdremote-ng-0.5-new/src/bdremoteng.c
--- bdremote-ng-0.5-old/src/bdremoteng.c	2009-11-19 14:54:29.000000000 -0800
+++ bdremote-ng-0.5-new/src/bdremoteng.c	2009-11-19 15:14:40.000000000 -0800
@@ -96,7 +96,7 @@
   memset(&config, 0, sizeof(config));
   setDefaults(&config);
 
-  while ((opt=getopt(argc,argv,"+p:t:a:i:r:e:R:u:g:f:ndh"))!=-1)
+  while ((opt=getopt(argc,argv,"+p:t:a:i:r:e:R:u:g:f:ndhl"))!=-1)
     {
       switch(opt)
         {
@@ -133,6 +133,9 @@
         case 'R':
           setRelease(&config, optarg);
           break;
+        case 'l':
+          config.lirc_namespace = 1;
+          break;
         case 'f':
           setLogFilename(&config, optarg);
           break;
@@ -290,6 +293,7 @@
   printf("\t-r <rate>            Key repeat rate. Generate <rate> repeats per second.\n"
          "\t-e <num>             Wait <num> ms before repeating a key.\n"
          "\t-R <suffix>          Auto-generate release events with appended <suffix>.\n"
+         "\t-l                   Follow LIRC namespace for the key names.\n"
          "\t-u <username>        Change UID to the UID of this user.\n"
          "\t-g <group>           Change GID to the GID of this group.\n"
          "\t-f <filename>        Write log to <filename>.\n"
diff -Naur bdremote-ng-0.5-old/src/keydef.h bdremote-ng-0.5-new/src/keydef.h
--- bdremote-ng-0.5-old/src/keydef.h	2009-11-19 14:53:46.000000000 -0800
+++ bdremote-ng-0.5-new/src/keydef.h	2009-11-19 15:05:24.000000000 -0800
@@ -41,8 +41,10 @@
 /** Key information. */
 struct key_info
 {
-  /** The name. */
-  char* name;
+  /** The original name. */
+  char* name_orig;
+  /** The LIRC namespace name. */
+  char* name_lirc;
   /** The code. */
   unsigned int code;
   /** The mask. */
@@ -61,57 +63,57 @@
 /** Array of keys. */
 struct key_info ps3remote_keys[] =
   {
-    {"enter\0"   ,0x0b, 0x080000},
-    {"ps\0"      ,0x43, 0x010000},
-    {"select\0"  ,0x50, 0x000001},
-    {"l3\0"          ,0x51, 0x000002},
-    {"r3\0"          ,0x52, 0x000004},
-    {"start\0"   ,0x53, 0x000008},
-    {"up\0"          ,0x54, 0x000010},
-    {"right\0"   ,0x55, 0x000020},
-    {"down\0"    ,0x56, 0x000040},
-    {"left\0"    ,0x57, 0x000080},
-    {"l2\0"          ,0x58, 0x000100},
-    {"r2\0"          ,0x59, 0x000200},
-    {"l1\0"          ,0x5a, 0x000400},
-    {"r1\0"          ,0x5b, 0x000800},
-    {"triangle\0",0x5c, 0x001000},
-    {"circle\0"  ,0x5d, 0x002000},
-    {"cross\0"   ,0x5e, 0x004000},
-    {"square\0"  ,0x5f, 0x008000},
-    {"num1\0"    ,0x00, 0x000000},
-    {"num2\0"    ,0x01, 0x000000},
-    {"num3\0"    ,0x02, 0x000000},
-    {"num4\0"    ,0x03, 0x000000},
-    {"num5\0"    ,0x04, 0x000000},
-    {"num6\0"    ,0x05, 0x000000},
-    {"num7\0"    ,0x06, 0x000000},
-    {"num8\0"    ,0x07, 0x000000},
-    {"num9\0"    ,0x08, 0x000000},
-    {"num0\0"    ,0x09, 0x000000},
-    {"return\0"  ,0x0e, 0x000000},
-    {"clear\0"   ,0x0f, 0x000000},
-    {"eject\0"   ,0x16, 0x000000},
-    {"topmenu\0" ,0x1a, 0x000000},
-    {"time\0"    ,0x28, 0x000000},
-    {"prev\0"    ,0x30, 0x000000},
-    {"next\0"    ,0x31, 0x000000},
-    {"play\0"    ,0x32, 0x000000},
-    {"scanrev\0" ,0x33, 0x000000},
-    {"scanfwd\0" ,0x34, 0x000000},
-    {"stop\0"    ,0x38, 0x000000},
-    {"pause\0"   ,0x39, 0x000000},
-    {"popup\0"   ,0x40, 0x000000},
-    {"steprev\0" ,0x60, 0x000000},
-    {"stepfwd\0" ,0x61, 0x000000},
-    {"subtitle\0",0x63, 0x000000},
-    {"audio\0"   ,0x64, 0x000000},
-    {"angle\0"   ,0x65, 0x000000},
-    {"display\0" ,0x70, 0x000000},
-    {"blue\0"    ,0x80, 0x000000},
-    {"red\0"     ,0x81, 0x000000},
-    {"green\0"   ,0x82, 0x000000},
-    {"yellow\0"  ,0x83, 0x000000},
+    {"enter\0"   , "KEY_ENTER\0"       , 0x0b, 0x080000},
+    {"ps\0"      , "KEY_POWER\0"       , 0x43, 0x010000},
+    {"select\0"  , "KEY_SELECT\0"      , 0x50, 0x000001},
+    {"l3\0"      , "BTN_C\0"           , 0x51, 0x000002},
+    {"r3\0"      , "BTN_Z\0"           , 0x52, 0x000004},
+    {"start\0"   , "BTN_START\0"       , 0x53, 0x000008},
+    {"up\0"      , "KEY_UP\0"          , 0x54, 0x000010},
+    {"right\0"   , "KEY_RIGHT\0"       , 0x55, 0x000020},
+    {"down\0"    , "KEY_DOWN\0"        , 0x56, 0x000040},
+    {"left\0"    , "KEY_LEFT\0"        , 0x57, 0x000080},
+    {"l2\0"      , "BTN_B\0"           , 0x58, 0x000100},
+    {"r2\0"      , "BTN_Y\0"           , 0x59, 0x000200},
+    {"l1\0"      , "BTN_A\0"           , 0x5a, 0x000400},
+    {"r1\0"      , "BTN_X\0"           , 0x5b, 0x000800},
+    {"triangle\0", "KEY_OPTION\0"      , 0x5c, 0x001000},
+    {"circle\0"  , "KEY_BACK\0"        , 0x5d, 0x002000},
+    {"cross\0"   , "KEY_CLEAR\0"       , 0x5e, 0x004000},
+    {"square\0"  , "KEY_ZOOM\0"        , 0x5f, 0x008000},
+    {"num1\0"    , "KEY_1\0"           , 0x00, 0x000000},
+    {"num2\0"    , "KEY_2\0"           , 0x01, 0x000000},
+    {"num3\0"    , "KEY_3\0"           , 0x02, 0x000000},
+    {"num4\0"    , "KEY_4\0"           , 0x03, 0x000000},
+    {"num5\0"    , "KEY_5\0"           , 0x04, 0x000000},
+    {"num6\0"    , "KEY_6\0"           , 0x05, 0x000000},
+    {"num7\0"    , "KEY_7\0"           , 0x06, 0x000000},
+    {"num8\0"    , "KEY_8\0"           , 0x07, 0x000000},
+    {"num9\0"    , "KEY_9\0"           , 0x08, 0x000000},
+    {"num0\0"    , "KEY_0\0"           , 0x09, 0x000000},
+    {"return\0"  , "KEY_MEDIA_REPEAT\0", 0x0e, 0x000000},
+    {"clear\0"   , "KEY_CLEAR\0"       , 0x0f, 0x000000},
+    {"eject\0"   , "KEY_EJECTCD\0"     , 0x16, 0x000000},
+    {"topmenu\0" , "KEY_MENU\0"        , 0x1a, 0x000000},
+    {"time\0"    , "KEY_TIME\0"        , 0x28, 0x000000},
+    {"prev\0"    , "KEY_PREVIOUS\0"    , 0x30, 0x000000},
+    {"next\0"    , "KEY_NEXT\0"        , 0x31, 0x000000},
+    {"play\0"    , "KEY_PLAY\0"        , 0x32, 0x000000},
+    {"scanrev\0" , "KEY_REWIND\0"      , 0x33, 0x000000},
+    {"scanfwd\0" , "KEY_FORWARD\0"     , 0x34, 0x000000},
+    {"stop\0"    , "KEY_STOP\0"        , 0x38, 0x000000},
+    {"pause\0"   , "KEY_PAUSE\0"       , 0x39, 0x000000},
+    {"popup\0"   , "KEY_CONTEXT_MENU\0", 0x40, 0x000000},
+    {"steprev\0" , "KEY_FRAMEBACK\0"   , 0x60, 0x000000},
+    {"stepfwd\0" , "KEY_FRAMEFORWARD\0", 0x61, 0x000000},
+    {"subtitle\0", "KEY_SUBTITLE\0"    , 0x63, 0x000000},
+    {"audio\0"   , "KEY_AUDIO\0"       , 0x64, 0x000000},
+    {"angle\0"   , "KEY_ANGLE\0"       , 0x65, 0x000000},
+    {"display\0" , "KEY_INFO\0"        , 0x70, 0x000000},
+    {"blue\0"    , "KEY_BLUE\0"        , 0x80, 0x000000},
+    {"red\0"     , "KEY_RED\0"         , 0x81, 0x000000},
+    {"green\0"   , "KEY_GREEN\0"       , 0x82, 0x000000},
+    {"yellow\0"  , "KEY_YELLOW\0"      , 0x83, 0x000000},
   };
 
 #endif /* BD_KEYDEF_H */
diff -Naur bdremote-ng-0.5-old/src/lirc_thr.c bdremote-ng-0.5-new/src/lirc_thr.c
--- bdremote-ng-0.5-old/src/lirc_thr.c	2009-11-19 15:04:08.000000000 -0800
+++ bdremote-ng-0.5-new/src/lirc_thr.c	2009-11-19 15:06:45.000000000 -0800
@@ -97,6 +97,7 @@
   queueData* qd  = NULL;
   int res        = Q_ERR;
   keyState ks;
+  char* name     = NULL;
 
 #if ENABLE_REPEAT
   unsigned long sendAfterMs = 0;
@@ -183,7 +184,15 @@
 		  BDREMOTE_LOG(ld->config->debug,
 			       fprintf(printStream, "Key is down (repeat #%d): %lu\n", ks.repeat_count, ks.elapsed);
 			       );
-		  broadcastToLirc(ld, ps3remote_keys[ks.lastKey].name, 0 /*ks.repeat_sent*/, ps3remote_keys[ks.lastKey].code);
+                  if (ld->config->lirc_namespace)
+                  {
+                      name = ps3remote_keys[ks.lastKey].name_lirc;
+                  }
+                  else
+                  {
+                      name = ps3remote_keys[ks.lastKey].name_orig;
+                  }
+		  broadcastToLirc(ld, name, 0 /*ks.repeat_sent*/, ps3remote_keys[ks.lastKey].code);
 		  /* Reset elapsed time.*/
 		  ks.elapsed = 0;
 		  
@@ -263,6 +272,7 @@
                      keyState* _ks)
 {
   int num = ps3remote_undef;
+  char* name = NULL;
   if (_code != ps3remote_keyup)
     {
       /* Key pressed. */
@@ -276,7 +286,15 @@
     }
   if (num != ps3remote_undef)
     {
-      broadcastToLirc(_ld, ps3remote_keys[num].name, 0, ps3remote_keys[num].code);
+      if (_ld->config->lirc_namespace)
+      {
+          name = ps3remote_keys[num].name_lirc;
+      }
+      else
+      {
+          name = ps3remote_keys[num].name_orig;
+      }
+      broadcastToLirc(_ld, name, 0, ps3remote_keys[num].code);
     }
 }
 
@@ -286,6 +304,7 @@
                    keyState* _ks)
 {
   char release_name[100];
+  char* name = NULL;
 
   if (_code == ps3remote_keyup)
     {
@@ -297,9 +316,17 @@
         {
           if (_ld->config->release != NULL)
           {
-              if ((strlen(ps3remote_keys[_ks->lastKey].name) + strlen(_ld->config->release)) < 100)
+              if (_ld->config->lirc_namespace)
+              {
+                  name = ps3remote_keys[_ks->lastKey].name_lirc;
+              }
+              else
+              {
+                  name = ps3remote_keys[_ks->lastKey].name_orig;
+              }
+              if ((strlen(name) + strlen(_ld->config->release)) < 100)
               {
-                  sprintf(release_name, "%s%s", ps3remote_keys[_ks->lastKey].name, _ld->config->release);
+                  sprintf(release_name, "%s%s", name, _ld->config->release);
                   broadcastToLirc(_ld, release_name, 0, _code);
               }
           }
