diff -Naur libmng-1.0.9-old/libmng_conf.h libmng-1.0.9-new/libmng_conf.h
--- libmng-1.0.9-old/libmng_conf.h	2005-01-30 02:45:00.000000000 -0800
+++ libmng-1.0.9-new/libmng_conf.h	2005-06-11 17:14:48.000000000 -0700
@@ -96,12 +96,12 @@
 
 #ifndef MNG_NO_INCLUDE_JNG
 #ifdef MNG_SUPPORT_FULL                /* full support includes JNG */
-#define MNG_SUPPORT_IJG6B
+/* #define MNG_SUPPORT_IJG6B */
 #endif
 
 #ifndef MNG_SUPPORT_IJG6B
 #if defined(MNG_BUILD_SO) || defined(MNG_USE_SO) || defined(MNG_BUILD_DLL) || defined(MNG_USE_DLL)
-#define MNG_SUPPORT_IJG6B
+/* #define MNG_SUPPORT_IJG6B */
 #endif
 #endif
 
