diff -Naur MPlayer-20249.6701-old/configure MPlayer-20249.6701-new/configure
--- MPlayer-20249.6701-old/configure	2006-10-21 06:12:15.000000000 -0700
+++ MPlayer-20249.6701-new/configure	2006-10-21 06:16:38.000000000 -0700
@@ -634,8 +634,7 @@
     morphos) system_name=MorphOS ;;
     mingw32msvc) system_name=MINGW32 ;;
   esac
-  # We need to convert underscores so that values like k6-2 and pentium-mmx can be passed
-  host_arch=`echo $_target | cut -d '-' -f 1 | tr '_' '-'`
+  host_arch=`echo $_target | cut -d '-' -f 1`
 fi
 
 echo "Detected operating system: $system_name"
@@ -2846,12 +2845,7 @@
 
 echocheck "word size"
 _mp_wordsize="#undef MP_WORDSIZE"
-cat > $TMPC << EOF
-#include <stdio.h>
-#include <sys/types.h>
-int main(void) { printf("%d\n", sizeof(size_t)*8); return 0; }
-EOF
-cc_check && _wordsize=`$TMPO` && _mp_wordsize="#define MP_WORDSIZE $_wordsize"
+_wordsize=64 &&  _mp_wordsize="#define MP_WORDSIZE $_wordsize"
 echores "$_wordsize"
 
 
