diff -Naur linux-2.6.36-old/include/linux/posix_types.h linux-2.6.36-new/include/linux/posix_types.h
--- linux-2.6.36-old/include/linux/posix_types.h	2010-10-21 08:45:40.000000000 -0700
+++ linux-2.6.36-new/include/linux/posix_types.h	2010-10-21 08:46:37.000000000 -0700
@@ -28,10 +28,10 @@
 #define __FDSET_LONGS	(__FD_SETSIZE/__NFDBITS)
 
 #undef __FDELT
-#define	__FDELT(d)	((d) / __NFDBITS)
+#define	__FDELT(d)	(((unsigned long)d) / __NFDBITS)
 
 #undef __FDMASK
-#define	__FDMASK(d)	(1UL << ((d) % __NFDBITS))
+#define	__FDMASK(d)	(1UL << (((unsigned long)d) % __NFDBITS))
 
 typedef struct {
 	unsigned long fds_bits [__FDSET_LONGS];
