diff -Naur mythstream-0.18_1-old/mythstream/mythstream.cpp mythstream-0.18_1-new/mythstream/mythstream.cpp
--- mythstream-0.18_1-old/mythstream/mythstream.cpp	2007-09-25 15:54:33.000000000 -0700
+++ mythstream-0.18_1-new/mythstream/mythstream.cpp	2007-11-24 10:08:41.000000000 -0800
@@ -460,7 +460,7 @@
 
         parameter = editBox->text();
 
-        delete popBox;        
+        popBox->deleteLater();
         popBox = NULL;
 
         return parameter;
@@ -549,7 +549,7 @@
 			  streamStorage->removeRecord(ident_maingui, values, errorMessage);
 		}
 
-		delete popBox;		
+		popBox->deleteLater();
 		popBox = NULL;
 
 		return;
