diff -Naur mythstream-0.18_1-old/libs/harvester.cpp mythstream-0.18_1-new/libs/harvester.cpp
--- mythstream-0.18_1-old/libs/harvester.cpp	2007-09-25 15:58:20.000000000 -0700
+++ mythstream-0.18_1-new/libs/harvester.cpp	2008-01-20 08:21:05.000000000 -0800
@@ -33,6 +33,7 @@
    http://lists.trolltech.com/qt-interest/2004-05/thread00280-0.html
 */
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qnetwork.h>
diff -Naur mythstream-0.18_1-old/libs/recorder.cpp mythstream-0.18_1-new/libs/recorder.cpp
--- mythstream-0.18_1-old/libs/recorder.cpp	2005-05-14 06:22:51.000000000 -0700
+++ mythstream-0.18_1-new/libs/recorder.cpp	2008-01-20 08:20:53.000000000 -0800
@@ -21,6 +21,7 @@
 
 // recorder experimental
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qnetwork.h>
diff -Naur mythstream-0.18_1-old/libs/storageconfig.cpp mythstream-0.18_1-new/libs/storageconfig.cpp
--- mythstream-0.18_1-old/libs/storageconfig.cpp	2007-09-05 14:03:14.000000000 -0700
+++ mythstream-0.18_1-new/libs/storageconfig.cpp	2008-01-20 08:21:33.000000000 -0800
@@ -22,6 +22,7 @@
 
 // development snapshot, many todo's
 
+#include <cstdlib>
 #include <iostream>
 #include <qlayout.h>
 #include <qcursor.h>
diff -Naur mythstream-0.18_1-old/libs/streamconfig.cpp mythstream-0.18_1-new/libs/streamconfig.cpp
--- mythstream-0.18_1-old/libs/streamconfig.cpp	2007-09-05 13:36:21.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamconfig.cpp	2008-01-20 08:21:55.000000000 -0800
@@ -22,6 +22,7 @@
 
 // development snapshot, many todo's
 
+#include <cstdlib>
 #include <iostream>
 #include <qlayout.h>
 #include <qcursor.h>
diff -Naur mythstream-0.18_1-old/libs/streamstatus.cpp mythstream-0.18_1-new/libs/streamstatus.cpp
--- mythstream-0.18_1-old/libs/streamstatus.cpp	2007-09-18 17:13:45.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstatus.cpp	2008-01-20 08:22:10.000000000 -0800
@@ -21,6 +21,7 @@
 
 // development snapshot, many todo's
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qapplication.h>
diff -Naur mythstream-0.18_1-old/libs/streamstorage.cpp mythstream-0.18_1-new/libs/streamstorage.cpp
--- mythstream-0.18_1-old/libs/streamstorage.cpp	2007-09-23 11:26:15.000000000 -0700
+++ mythstream-0.18_1-new/libs/streamstorage.cpp	2008-01-20 08:22:21.000000000 -0800
@@ -19,6 +19,7 @@
  *
  * ============================================================ */
 
+#include <cstdlib>
 #include <iostream>
 
 #include <qnetwork.h>
