diff -Naur mythstream-0.18_1-old/libs/harvester.cpp mythstream-0.18_1-new/libs/harvester.cpp
--- mythstream-0.18_1-old/libs/harvester.cpp	2008-01-16 12:52:41.000000000 -0800
+++ mythstream-0.18_1-new/libs/harvester.cpp	2008-01-16 16:26:06.000000000 -0800
@@ -1674,7 +1674,7 @@
     
     eParser->addArgument( "sh" );
     eParser->addArgument( "-c" );
-    eParser->addArgument( parser + " list.xml \"" + sourceUrl + "\"" + srcName + "\"" );
+    eParser->addArgument( parser + " list.xml \"" + sourceUrl + "\" \"" + srcName + "\"" );
     //cout << "SHELL PARSER" << endl;
     
 #endif
