diff -Naur libart_lgpl-2.3.19-old/art_misc.h libart_lgpl-2.3.19-new/art_misc.h
--- libart_lgpl-2.3.19-old/art_misc.h	2007-01-01 15:59:22.000000000 -0800
+++ libart_lgpl-2.3.19-new/art_misc.h	2007-04-03 19:05:29.000000000 -0700
@@ -34,9 +34,15 @@
 #include <libart_lgpl/art_config.h>
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
 void *art_alloc(size_t size);
 void art_free(void *ptr);
 void *art_realloc(void *ptr, size_t size);
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
 
 /* These aren't, strictly speaking, configuration macros, but they're
    damn handy to have around, and may be worth playing with for
