diff -Naur mednafen-0.8.7-old/src/cdrom/scsicd.cpp mednafen-0.8.7-new/src/cdrom/scsicd.cpp
--- mednafen-0.8.7-old/src/cdrom/scsicd.cpp	2007-11-18 16:20:03.000000000 -0800
+++ mednafen-0.8.7-new/src/cdrom/scsicd.cpp	2008-01-16 07:34:04.000000000 -0800
@@ -19,6 +19,7 @@
 #include <trio/trio.h>
 #include "scsicd.h"
 #include "cdromif.h"
+#include <string.h>
 
 static uint32 CD_DATA_TRANSFER_RATE;
 static uint32 System_Clock;
diff -Naur mednafen-0.8.7-old/src/gb/sound.cpp mednafen-0.8.7-new/src/gb/sound.cpp
--- mednafen-0.8.7-old/src/gb/sound.cpp	2007-08-25 13:12:57.000000000 -0700
+++ mednafen-0.8.7-new/src/gb/sound.cpp	2008-01-16 07:09:45.000000000 -0800
@@ -21,6 +21,7 @@
 #include "sound.h"
 #include "sound/Gb_Apu.h"
 #include <blip/Stereo_Buffer.h>
+#include <string.h>
 
 static Gb_Apu gb_apu;
 static Stereo_Buffer gb_buf;
diff -Naur mednafen-0.8.7-old/src/pce/adpcm.cpp mednafen-0.8.7-new/src/pce/adpcm.cpp
--- mednafen-0.8.7-old/src/pce/adpcm.cpp	2007-10-08 19:05:47.000000000 -0700
+++ mednafen-0.8.7-new/src/pce/adpcm.cpp	2008-01-16 07:19:16.000000000 -0800
@@ -19,6 +19,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "pce.h"
 #include "psg.h"
 #include "cdrom.h"
diff -Naur mednafen-0.8.7-old/src/pce/hes.cpp mednafen-0.8.7-new/src/pce/hes.cpp
--- mednafen-0.8.7-old/src/pce/hes.cpp	2007-10-16 22:01:28.000000000 -0700
+++ mednafen-0.8.7-new/src/pce/hes.cpp	2008-01-16 07:18:20.000000000 -0800
@@ -15,6 +15,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "pce.h"
 #include "hes.h"
 #include "huc.h"
diff -Naur mednafen-0.8.7-old/src/pce/input.cpp mednafen-0.8.7-new/src/pce/input.cpp
--- mednafen-0.8.7-old/src/pce/input.cpp	2007-12-21 21:29:47.000000000 -0800
+++ mednafen-0.8.7-new/src/pce/input.cpp	2008-01-16 07:17:02.000000000 -0800
@@ -15,6 +15,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "pce.h"
 #include "input.h"
 #include "huc.h"
diff -Naur mednafen-0.8.7-old/src/pce/pce.cpp mednafen-0.8.7-new/src/pce/pce.cpp
--- mednafen-0.8.7-old/src/pce/pce.cpp	2007-10-16 19:38:01.000000000 -0700
+++ mednafen-0.8.7-new/src/pce/pce.cpp	2008-01-16 07:15:30.000000000 -0800
@@ -15,6 +15,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include <zlib.h>
 #include "pce.h"
 #include "vdc.h"
diff -Naur mednafen-0.8.7-old/src/pce/vdc.cpp mednafen-0.8.7-new/src/pce/vdc.cpp
--- mednafen-0.8.7-old/src/pce/vdc.cpp	2007-10-22 03:20:24.000000000 -0700
+++ mednafen-0.8.7-new/src/pce/vdc.cpp	2008-01-16 07:15:41.000000000 -0800
@@ -31,6 +31,7 @@
 //{ printf("%s: %d\n", x, vdc->display_counter); }
 
 
+#include <string.h>
 #include <math.h>
 #include "pce.h"
 #include "../video.h"
diff -Naur mednafen-0.8.7-old/src/pcfx/input.cpp mednafen-0.8.7-new/src/pcfx/input.cpp
--- mednafen-0.8.7-old/src/pcfx/input.cpp	2007-07-29 19:53:47.000000000 -0700
+++ mednafen-0.8.7-new/src/pcfx/input.cpp	2008-01-16 07:22:56.000000000 -0800
@@ -15,6 +15,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "pcfx.h"
 #include "interrupt.h"
 #include "input.h"
diff -Naur mednafen-0.8.7-old/src/pcfx/king.cpp mednafen-0.8.7-new/src/pcfx/king.cpp
--- mednafen-0.8.7-old/src/pcfx/king.cpp	2007-11-14 12:30:14.000000000 -0800
+++ mednafen-0.8.7-new/src/pcfx/king.cpp	2008-01-16 07:22:15.000000000 -0800
@@ -29,6 +29,7 @@
 #include "timer.h"
 #include "debug.h"
 #include <trio/trio.h>
+#include <string.h>
 
 #include "../video.h"
 #include "../clamp.h"
diff -Naur mednafen-0.8.7-old/src/pcfx/rainbow.cpp mednafen-0.8.7-new/src/pcfx/rainbow.cpp
--- mednafen-0.8.7-old/src/pcfx/rainbow.cpp	2007-11-11 01:11:03.000000000 -0800
+++ mednafen-0.8.7-new/src/pcfx/rainbow.cpp	2008-01-16 07:23:29.000000000 -0800
@@ -17,6 +17,7 @@
 
 /* MJPEG-ish decoder based on the algorithm and huffman data tables provided by David Michel of MagicEngine and MagicEngine-FX */
 
+#include <string.h>
 #include "pcfx.h"
 #include "rainbow.h"
 #include "king.h"
diff -Naur mednafen-0.8.7-old/src/pcfx/v810_cpuD.cpp mednafen-0.8.7-new/src/pcfx/v810_cpuD.cpp
--- mednafen-0.8.7-old/src/pcfx/v810_cpuD.cpp	2007-09-05 23:38:01.000000000 -0700
+++ mednafen-0.8.7-new/src/pcfx/v810_cpuD.cpp	2008-01-16 07:21:07.000000000 -0800
@@ -23,7 +23,7 @@
 //what do we realy need?
 //#include <stdio.h>
 //#include <stdlib.h> 
-//#include <string.h>
+#include <string.h>
 //#include <ctype.h>
 
 #include "pcfx.h"
diff -Naur mednafen-0.8.7-old/src/sms/cart.cpp mednafen-0.8.7-new/src/sms/cart.cpp
--- mednafen-0.8.7-old/src/sms/cart.cpp	2007-11-14 13:03:43.000000000 -0800
+++ mednafen-0.8.7-new/src/sms/cart.cpp	2008-01-16 07:24:10.000000000 -0800
@@ -16,6 +16,7 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <string.h>
 #include "shared.h"
 #include "romdb.h"
 #include "cart.h"
diff -Naur mednafen-0.8.7-old/src/sms/render.cpp mednafen-0.8.7-new/src/sms/render.cpp
--- mednafen-0.8.7-old/src/sms/render.cpp	2007-05-03 08:26:18.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/render.cpp	2008-01-16 07:24:37.000000000 -0800
@@ -3,6 +3,7 @@
     Display rendering.
 */
 
+#include <string.h>
 #include "shared.h"
 
 static void remap_8_to_32(int line);
diff -Naur mednafen-0.8.7-old/src/sms/Sms_Apu.cpp mednafen-0.8.7-new/src/sms/Sms_Apu.cpp
--- mednafen-0.8.7-old/src/sms/Sms_Apu.cpp	2007-06-07 06:48:09.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/Sms_Apu.cpp	2008-01-16 07:28:35.000000000 -0800
@@ -1,5 +1,6 @@
 // Sms_Snd_Emu 0.1.4. http://www.slack.net/~ant/
 
+#include <string.h>
 #include "../mednafen.h"
 #include "Sms_Apu.h"
 
diff -Naur mednafen-0.8.7-old/src/sms/sms.cpp mednafen-0.8.7-new/src/sms/sms.cpp
--- mednafen-0.8.7-old/src/sms/sms.cpp	2007-05-03 05:42:30.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/sms.cpp	2008-01-16 07:25:02.000000000 -0800
@@ -2,6 +2,7 @@
     sms.c --
     Sega Master System console emulation.
 */
+#include <string.h>
 #include "shared.h"
 
 /* SMS context */
diff -Naur mednafen-0.8.7-old/src/sms/sound.cpp mednafen-0.8.7-new/src/sms/sound.cpp
--- mednafen-0.8.7-old/src/sms/sound.cpp	2007-06-07 06:47:53.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/sound.cpp	2008-01-16 07:25:32.000000000 -0800
@@ -16,6 +16,7 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <string.h>
 #include "shared.h"
 #include <blip/Blip_Buffer.h>
 #include "Sms_Apu.h"
diff -Naur mednafen-0.8.7-old/src/sms/system.cpp mednafen-0.8.7-new/src/sms/system.cpp
--- mednafen-0.8.7-old/src/sms/system.cpp	2007-08-07 02:47:52.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/system.cpp	2008-01-16 07:25:57.000000000 -0800
@@ -16,6 +16,7 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <string.h>
 #include "shared.h"
 #include "../memory.h"
 #include "../netplay.h"
diff -Naur mednafen-0.8.7-old/src/sms/tms.cpp mednafen-0.8.7-new/src/sms/tms.cpp
--- mednafen-0.8.7-old/src/sms/tms.cpp	2007-05-02 05:14:12.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/tms.cpp	2008-01-16 07:27:31.000000000 -0800
@@ -2,6 +2,7 @@
     tms.c --
     TMS9918 and legacy video mode support.
 */
+#include <string.h>
 #include "shared.h"
 
 int text_counter;               /* Text offset counter */
diff -Naur mednafen-0.8.7-old/src/sms/vdp.cpp mednafen-0.8.7-new/src/sms/vdp.cpp
--- mednafen-0.8.7-old/src/sms/vdp.cpp	2007-05-24 01:05:58.000000000 -0700
+++ mednafen-0.8.7-new/src/sms/vdp.cpp	2008-01-16 07:28:04.000000000 -0800
@@ -2,6 +2,7 @@
     vdp.c --
     Video Display Processor (VDP) emulation.
 */
+#include <string.h>
 #include "shared.h"
 #include "hvc.h"
 
diff -Naur mednafen-0.8.7-old/src/string/escape.cpp mednafen-0.8.7-new/src/string/escape.cpp
--- mednafen-0.8.7-old/src/string/escape.cpp	2007-10-07 14:31:57.000000000 -0700
+++ mednafen-0.8.7-new/src/string/escape.cpp	2008-01-16 07:34:47.000000000 -0800
@@ -17,6 +17,7 @@
 
 #include "../mednafen.h"
 #include "escape.h"
+#include <string.h>
 
 static unsigned int hex_nibble_to_val(char nibble)
 {
diff -Naur mednafen-0.8.7-old/src/video/text.cpp mednafen-0.8.7-new/src/video/text.cpp
--- mednafen-0.8.7-old/src/video/text.cpp	2007-10-07 14:32:16.000000000 -0700
+++ mednafen-0.8.7-new/src/video/text.cpp	2008-01-16 07:35:30.000000000 -0800
@@ -18,6 +18,7 @@
 #include "video-common.h"
 #include "../string/ConvertUTF.h"
 #include "font-data.h"
+#include <string.h>
 
 typedef struct
 {
diff -Naur mednafen-0.8.7-old/src/wswan/eeprom.cpp mednafen-0.8.7-new/src/wswan/eeprom.cpp
--- mednafen-0.8.7-old/src/wswan/eeprom.cpp	2007-02-02 12:08:43.000000000 -0800
+++ mednafen-0.8.7-new/src/wswan/eeprom.cpp	2008-01-16 07:32:10.000000000 -0800
@@ -22,6 +22,7 @@
 #include "eeprom.h"
 #include "memory.h"
 #include <ctype.h>
+#include <string.h>
 
 uint8 wsEEPROM[2048];
 
diff -Naur mednafen-0.8.7-old/src/wswan/gfx.cpp mednafen-0.8.7-new/src/wswan/gfx.cpp
--- mednafen-0.8.7-old/src/wswan/gfx.cpp	2007-10-05 12:18:18.000000000 -0700
+++ mednafen-0.8.7-new/src/wswan/gfx.cpp	2008-01-16 07:29:13.000000000 -0800
@@ -18,6 +18,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "wswan.h"
 #include "gfx.h"
 #include "memory.h"
diff -Naur mednafen-0.8.7-old/src/wswan/main.cpp mednafen-0.8.7-new/src/wswan/main.cpp
--- mednafen-0.8.7-old/src/wswan/main.cpp	2007-10-05 12:36:47.000000000 -0700
+++ mednafen-0.8.7-new/src/wswan/main.cpp	2008-01-16 07:29:25.000000000 -0800
@@ -24,6 +24,7 @@
 #include "../mempatcher.h"
 
 #include <fcntl.h>
+#include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <math.h>
diff -Naur mednafen-0.8.7-old/src/wswan/memory.cpp mednafen-0.8.7-new/src/wswan/memory.cpp
--- mednafen-0.8.7-old/src/wswan/memory.cpp	2007-11-14 13:00:36.000000000 -0800
+++ mednafen-0.8.7-new/src/wswan/memory.cpp	2008-01-16 07:30:14.000000000 -0800
@@ -26,6 +26,7 @@
 #include "rtc.h"
 #include "v30mz.h"
 #include "../mempatcher.h"
+#include <string.h>
 #include <time.h>
 #include <math.h>
 
diff -Naur mednafen-0.8.7-old/src/wswan/sound.cpp mednafen-0.8.7-new/src/wswan/sound.cpp
--- mednafen-0.8.7-old/src/wswan/sound.cpp	2007-05-23 00:04:01.000000000 -0700
+++ mednafen-0.8.7-new/src/wswan/sound.cpp	2008-01-16 07:31:07.000000000 -0800
@@ -15,6 +15,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "wswan.h"
 #include "sound.h"
 #include "v30mz.h"
diff -Naur mednafen-0.8.7-old/src/wswan/tcache.cpp mednafen-0.8.7-new/src/wswan/tcache.cpp
--- mednafen-0.8.7-old/src/wswan/tcache.cpp	2007-02-02 12:23:25.000000000 -0800
+++ mednafen-0.8.7-new/src/wswan/tcache.cpp	2008-01-16 07:31:38.000000000 -0800
@@ -18,6 +18,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <string.h>
 #include "wswan.h"
 #include "gfx.h"
 #include "memory.h"
