diff -Naur alsa-driver-1.0.16-old/alsa-kernel/pci/hda/patch_atihdmi.c alsa-driver-1.0.16-new/alsa-kernel/pci/hda/patch_atihdmi.c
--- alsa-driver-1.0.16-old/alsa-kernel/pci/hda/patch_atihdmi.c	2008-02-05 01:23:23.000000000 -0800
+++ alsa-driver-1.0.16-new/alsa-kernel/pci/hda/patch_atihdmi.c	2008-03-30 19:51:42.000000000 -0700
@@ -163,5 +163,6 @@
 	{ .id = 0x1002791a, .name = "ATI RS690/780 HDMI", .patch = patch_atihdmi },
 	{ .id = 0x1002aa01, .name = "ATI R6xx HDMI", .patch = patch_atihdmi },
 	{ .id = 0x10951392, .name = "SI HDMI", .patch = patch_atihdmi },
+	{ .id = 0x17e80047, .name = "Unknown HDMI",  .patch = patch_atihdmi },
 	{} /* terminator */
 };
