diff -Naur myththemes-12854-old/cpsvndir myththemes-12854-new/cpsvndir
--- myththemes-12854-old/cpsvndir	2007-02-23 21:59:55.000000000 -0800
+++ myththemes-12854-new/cpsvndir	2007-02-23 22:00:36.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
