diff -Naur myththemes-11467-old/cpsvndir myththemes-11467-new/cpsvndir
--- myththemes-11467-old/cpsvndir	2006-10-07 15:52:39.000000000 -0700
+++ myththemes-11467-new/cpsvndir	2006-10-07 15:53:43.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
