commit 23cdeb95950a60f51f8c7c93a5bd25bb92f02aa5
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jul 2 09:19:49 2009 +0000

    Drop duplicate LIBDRM_DEPS pkg-config check.

diff --git a/configure.ac b/configure.ac
index b6e82d0..63d08aa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -95,7 +95,6 @@ AC_SYS_LARGEFILE
 PKG_CHECK_MODULES([X11], [x11])
 PKG_CHECK_MODULES([XEXT],[xext])
 PKG_CHECK_MODULES([DRM], [libdrm])
-PKG_CHECK_MODULES(LIBDRM_DEPS, [libdrm])
 
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
 AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
@@ -100,7 +99,7 @@ PKG_CHECK_MODULES([DRM], [libdrm])
 AM_CONDITIONAL(BUILD_I965_DRIVER, test x$enable_i965_driver = xyes)
 
 # We only need the headers, we don't link against the DRM libraries
-LIBVA_CFLAGS="$LIBDRM_DEPS_CFLAGS"
+LIBVA_CFLAGS="$DRM_CFLAGS"
 AC_SUBST(LIBVA_CFLAGS)
 AC_SUBST(LIBVA_LIBS)
 
