commit 2dc7b48446ca0b7b5d32420f798dbfed355ba832
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Sep 9 11:09:59 2009 +0000

    Drop assert() in isDRI1Connected().

diff --git a/src/x11/dri1_util.c b/src/x11/dri1_util.c
index 3f0da50..ac5f843 100644
--- a/src/x11/dri1_util.c
+++ b/src/x11/dri1_util.c
@@ -168,7 +168,6 @@ isDRI1Connected(VADriverContextP ctx, char **driver_name)
     
     dri_state->fd = drmOpenOnce(NULL, BusID, &newlyopened);
     XFree(BusID);
-    assert(dri_state->fd >= 0);
 
     if (dri_state->fd < 0)
         goto err_out1;
