commit 09194310ee4bd7447d51cda393e5bf7949ee21da
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jul 2 13:26:49 2009 +0000

    Remove obsolete fields.

diff --git a/src/x11/va_x11.c b/dri/va_x11.c
index 667b5ae..b3d36ee 100644
--- a/src/x11/va_x11.c
+++ b/src/x11/va_x11.c
@@ -234,7 +234,6 @@ VADisplay vaGetDisplay (
       pDriverContext  = calloc(1, sizeof(*pDriverContext));
       if (pDisplayContext && pDriverContext)
       {
-	  pDriverContext->old_pNext        = (void *)(unsigned long)0xdeadbeef;
 	  pDriverContext->x11_dpy          = native_dpy;
 	  pDisplayContext->pNext           = pDisplayContexts;
 	  pDisplayContext->pDriverContext  = pDriverContext;
diff --git a/src/va_backend.h b/src/va_backend.h
index 2b12e1e..b185afa 100644
--- a/src/va_backend.h
+++ b/src/va_backend.h
@@ -407,8 +407,6 @@ struct VADriverVTable
 
 struct VADriverContext
 {
-    void *old_pNext;			/* preserved for binary compatibility */
-
     void *pDriverData;
     struct VADriverVTable vtable;
 
@@ -414,7 +414,6 @@ struct VADriverContext
 
     Display *x11_dpy;
     int x11_screen;
-    int old_dri2; /* obsolete */
     int version_major;
     int version_minor;
     int max_profiles;
