commit feffa09de925ebcccd5e6a4024b2a0a04f2f4f7d
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Aug 13 14:43:30 2009 +0000

    Fix libva-x11-VERSION.so.* dependencies.

diff --git a/src/Makefile.am b/src/Makefile.am
index 74fbdb5..2fcd03b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -40,7 +40,7 @@ libva_x11_la_SOURCES =
 libva_x11_la_LIBADD  = $(libvacorelib) x11/libva_x11.la $(LIBVA_LIBS) $(X11_LIBS) $(XEXT_LIBS) $(DRM_LIBS) $(XFIXES_LIBS)
 
 libva_x11_la_LDFLAGS = $(LDADD)
-libva_x11_la_DEPENDENCIES = $(libvacorelib)
+libva_x11_la_DEPENDENCIES = $(libvacorelib) x11/libva_x11.la
 
 SUBDIRS = x11
 
