commit 85d14e0b8abddd129a5d9f52d19648fc36a25b24
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Mon Mar 2 15:01:46 2009 +0000

    Add new H.264 fields for VDPAU backend.

diff --git a/src/va.h b/src/va.h
index 0b3c9fa..50a7bf6 100755
--- a/src/va.h
+++ b/src/va.h
@@ -848,6 +848,7 @@ This is simplely a buffer containing raw bit-stream bytes
 typedef struct _VAPictureH264
 {
     VASurfaceID picture_id;
+    unsigned int frame_idx;
     unsigned int flags;
     unsigned int TopFieldOrderCnt;
     unsigned int BottomFieldOrderCnt;
@@ -881,6 +882,10 @@ typedef struct _VAPictureParameterBufferH264
             unsigned int mb_adaptive_frame_field_flag		: 1; 
             unsigned int direct_8x8_inference_flag		: 1; 
             unsigned int MinLumaBiPredSize8x8			: 1; /* see A.3.3.2 */
+            unsigned int log2_max_frame_num_minus4		: 4;
+            unsigned int pic_order_cnt_type			: 2;
+            unsigned int log2_max_pic_order_cnt_lsb_minus4	: 4;
+            unsigned int delta_pic_order_always_zero_flag	: 1;
         } bits;
         unsigned int value;
     } seq_fields;
@@ -897,6 +902,9 @@ typedef struct _VAPictureParameterBufferH264
             unsigned int transform_8x8_mode_flag	: 1;
             unsigned int field_pic_flag			: 1;
             unsigned int constrained_intra_pred_flag	: 1;
+            unsigned int pic_order_present_flag			: 1;
+            unsigned int deblocking_filter_control_present_flag : 1;
+            unsigned int redundant_pic_cnt_present_flag		: 1;
         } bits;
         unsigned int value;
     } pic_fields;
