commit 521dfb651648a28784336edbe3dcea702f5695ca
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Mon Mar 2 15:09:08 2009 +0000

    Add new H.264 fields for XvBA backend.

diff --git a/src/va.h b/src/va.h
index b88bcfd..4c63b3e 100755
--- a/src/va.h
+++ b/src/va.h
@@ -891,6 +891,7 @@ typedef struct _VAPictureParameterBufferH264
         struct {
             unsigned int chroma_format_idc			: 2; 
             unsigned int residual_colour_transform_flag		: 1; 
+            unsigned int gaps_in_frame_num_value_allowed_flag	: 1; 
             unsigned int frame_mbs_only_flag			: 1; 
             unsigned int mb_adaptive_frame_field_flag		: 1; 
             unsigned int direct_8x8_inference_flag		: 1; 
@@ -904,7 +905,9 @@ typedef struct _VAPictureParameterBufferH264
     } seq_fields;
     unsigned char num_slice_groups_minus1;
     unsigned char slice_group_map_type;
+    unsigned short slice_group_change_rate_minus1;
     signed char pic_init_qp_minus26;
+    signed char pic_init_qs_minus26;
     signed char chroma_qp_index_offset;
     signed char second_chroma_qp_index_offset;
     union {
@@ -918,6 +921,7 @@ typedef struct _VAPictureParameterBufferH264
             unsigned int pic_order_present_flag			: 1;
             unsigned int deblocking_filter_control_present_flag : 1;
             unsigned int redundant_pic_cnt_present_flag		: 1;
+            unsigned int reference_pic_flag			: 1; /* nal_ref_idc != 0 */
         } bits;
         unsigned int value;
     } pic_fields;
