diff -Naur libva-0.30.4-old/i965_drv_video/Makefile.am libva-0.30.4-new/i965_drv_video/Makefile.am
--- libva-0.30.4-old/i965_drv_video/Makefile.am	2009-07-03 05:12:55.000000000 -0700
+++ libva-0.30.4-new/i965_drv_video/Makefile.am	2009-07-03 05:40:35.000000000 -0700
@@ -27,7 +27,7 @@
 i965_drv_video_la_LTLIBRARIES = i965_drv_video.la
 i965_drv_video_ladir = @LIBVA_DRIVERS_PATH@
 i965_drv_video_la_LDFLAGS = -module -avoid-version -no-undefined -Wl,--no-undefined @DRM_LIBS@ -ldrm_intel
-i965_drv_video_la_LIBADD = ../src/libva.la -lpthread
+i965_drv_video_la_LIBADD = ../src/libva-x11.la -lpthread
 
 i965_drv_video_la_SOURCES =	\
 	object_heap.c		\
