diff -Naur libva-0.30.4-old/i965_drv_video/Makefile.am libva-0.30.4-new/i965_drv_video/Makefile.am
--- libva-0.30.4-old/i965_drv_video/Makefile.am	2009-07-02 05:39:19.000000000 -0700
+++ libva-0.30.4-new/i965_drv_video/Makefile.am	2009-07-03 05:12:55.000000000 -0700
@@ -38,3 +38,15 @@
 	i965_media_mpeg2.c	\
 	i965_render.c		\
 	i965_drv_video.c
+
+noinst_HEADERS =                \
+	object_heap.h           \
+	intel_batchbuffer.h     \
+	intel_memman.h          \
+	intel_driver.h          \
+	i965_media.h            \
+	i965_media_mpeg2.h      \
+	i965_render.h           \
+	i965_drv_video.h        \
+	i965_defines.h          \
+	i965_structs.h
