diff -Naur xorg-server-1.4.1-old/hw/xfree86/modes/xf86EdidModes.c xorg-server-1.4.1-new/hw/xfree86/modes/xf86EdidModes.c
--- xorg-server-1.4.1-old/hw/xfree86/modes/xf86EdidModes.c	2008-06-10 09:10:13.000000000 -0700
+++ xorg-server-1.4.1-new/hw/xfree86/modes/xf86EdidModes.c	2008-06-10 09:10:50.000000000 -0700
@@ -407,8 +407,11 @@
 
     Monitor->DDC = DDC;
 
-    Monitor->widthmm = 10 * DDC->features.hsize;
-    Monitor->heightmm = 10 * DDC->features.vsize;
+    /* Set display size if it was not specified in the config file */
+    if (Monitor->widthmm <= 0)
+        Monitor->widthmm = 10 * DDC->features.hsize;
+    if (Monitor->heightmm <= 0)
+        Monitor->heightmm = 10 * DDC->features.vsize;
 
     /* If this is a digital display, then we can use reduced blanking */
     if (DDC->features.input_type)
