diff -Naur xorg-server-1.4.2-old/hw/xfree86/ddc/edid.h xorg-server-1.4.2-new/hw/xfree86/ddc/edid.h
--- xorg-server-1.4.2-old/hw/xfree86/ddc/edid.h	2008-06-10 10:57:19.000000000 -0700
+++ xorg-server-1.4.2-new/hw/xfree86/ddc/edid.h	2009-01-15 07:03:38.000000000 -0800
@@ -407,8 +407,13 @@
 #define DS_RANGES 0xFD
 #define DS_WHITE_P 0xFB
 #define DS_STD_TIMINGS 0xFA
+#define DS_CMD 0xF9
+#define DS_CVT 0xF8
+#define DS_EST_III 0xF7
 #define DS_DUMMY 0x10
 #define DS_UNKOWN 0x100 /* type is an int */
+#define DS_VENDOR 0x101
+#define DS_VENDOR_MAX 0x110
 
 struct monitor_ranges {
   int min_v;
