diff -Naur xorg-server-1.4-old/configure.ac xorg-server-1.4-new/configure.ac
--- xorg-server-1.4-old/configure.ac	2007-09-05 22:59:00.000000000 -0700
+++ xorg-server-1.4-new/configure.ac	2008-01-22 22:03:12.000000000 -0800
@@ -1921,7 +1921,8 @@
 	AC_SUBST(XORGCFG_DEP_CFLAGS)
 	AC_SUBST(XORGCFG_DEP_LIBS)
 
-	APPDEFAULTDIR=$(pkg-config --variable=appdefaultdir xt)
+        m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+	APPDEFAULTDIR=$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=appdefaultdir xt)
 	AC_SUBST(APPDEFAULTDIR)
 fi
 AM_CONDITIONAL(BUILD_XORGCFG, [test x$XORGCFG = xyes])
diff -Naur xorg-server-1.4-old/xorg-server.m4 xorg-server-1.4-new/xorg-server.m4
--- xorg-server-1.4-old/xorg-server.m4	2007-08-23 12:04:55.000000000 -0700
+++ xorg-server-1.4-new/xorg-server.m4	2008-01-22 22:03:12.000000000 -0800
@@ -29,8 +29,10 @@
 # is defined, then add $1 to $REQUIRED_MODULES.
 
 AC_DEFUN([XORG_DRIVER_CHECK_EXT],[
+	AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
 	SAVE_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS -I`pkg-config --variable=sdkdir xorg-server`"
+	m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+	CFLAGS="$CFLAGS -I$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=sdkdir xorg-server`"
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "xorg-server.h"
 #if !defined $1
