diff -Naur xorg-server-ddcca23a81abf5215f906a7ad097f1ed088ed92b-old/hw/xfree86/modes/xf86EdidModes.c xorg-server-ddcca23a81abf5215f906a7ad097f1ed088ed92b-new/hw/xfree86/modes/xf86EdidModes.c
--- xorg-server-ddcca23a81abf5215f906a7ad097f1ed088ed92b-old/hw/xfree86/modes/xf86EdidModes.c	2008-05-28 20:02:13.000000000 -0700
+++ xorg-server-ddcca23a81abf5215f906a7ad097f1ed088ed92b-new/hw/xfree86/modes/xf86EdidModes.c	2008-05-28 20:03:12.000000000 -0700
@@ -407,8 +407,11 @@
 
     Monitor->DDC = DDC;
 
-    Monitor->widthmm = 10 * DDC->features.hsize;
-    Monitor->heightmm = 10 * DDC->features.vsize;
+    /* Set display size if it was not specified in the config file */
+    if (Monitor->widthmm <= 0)
+        Monitor->widthmm = 10 * DDC->features.hsize;
+    if (Monitor->heightmm <= 0)
+        Monitor->heightmm = 10 * DDC->features.vsize;
 
     /* If this is a digital display, then we can use reduced blanking */
     if (DDC->features.input_type)
