diff -Naur NVIDIA-Linux-x86-71.86.01-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-71.86.01-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-71.86.01-pkg1-old/usr/src/nv/conftest.sh	2007-09-05 20:40:58.000000000 -0700
+++ NVIDIA-Linux-x86-71.86.01-pkg1-new/usr/src/nv/conftest.sh	2007-12-19 13:43:51.000000000 -0800
@@ -849,6 +849,11 @@
             # Determine the kernel version using a compile check.
             #
             echo "#include <linux/autoconf.h>
+            #ifdef CONFIG_X86_32
+            #define __i386__
+            #else
+            #define __x86_64__
+            #endif
             #include <linux/version.h>
             #include <linux/utsname.h>
             int main() {
