diff -Naur xf86-video-nouveau-b0d7f4d-old/configure.ac xf86-video-nouveau-b0d7f4d-new/configure.ac
--- xf86-video-nouveau-b0d7f4d-old/configure.ac	2012-01-12 20:42:33.000000000 -0800
+++ xf86-video-nouveau-b0d7f4d-new/configure.ac	2012-01-12 20:43:03.000000000 -0800
@@ -76,7 +76,8 @@
 		  HAVE_XEXTPROTO_71="yes"; AC_DEFINE(HAVE_XEXTPROTO_71, 1, [xextproto 7.1 available]),
 		  HAVE_XEXTPROTO_71="no")
 AM_CONDITIONAL(HAVE_XEXTPROTO_71, [ test "$HAVE_XEXTPROTO_71" = "yes" ])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 PKG_CHECK_MODULES([PCIACCESS], [pciaccess >= 0.10])
