diff -Naur lm_sensors-2.10.4-old/kernel/include/Module.mk lm_sensors-2.10.4-new/kernel/include/Module.mk
--- lm_sensors-2.10.4-old/kernel/include/Module.mk	2007-07-16 21:52:51.000000000 -0700
+++ lm_sensors-2.10.4-new/kernel/include/Module.mk	2007-09-26 21:25:21.000000000 -0700
@@ -22,7 +22,7 @@
 KERNELINCLUDEDIR := $(MODULE_DIR)
 KERNELCHIPSDIR := kernel/chips
 
-KERNELINCLUDEFILES := $(MODULE_DIR)/i2c-dev.h $(MODULE_DIR)/sensors.h
+KERNELINCLUDEFILES := $(MODULE_DIR)/sensors.h
 
 $(KERNELINCLUDEDIR)/sensors.h: $(KERNELINCLUDEDIR)/sensors.h.template
 	cat $@.template > $@
