diff -Naur lm_sensors-3.1.2-old/prog/detect/sensors-detect lm_sensors-3.1.2-new/prog/detect/sensors-detect
--- lm_sensors-3.1.2-old/prog/detect/sensors-detect	2010-01-18 08:22:07.000000000 -0800
+++ lm_sensors-3.1.2-new/prog/detect/sensors-detect	2010-03-31 14:10:10.000000000 -0700
@@ -558,7 +558,7 @@
 		i2c_detect => sub { adt7467_detect(@_, 3); },
 	}, {
 		name => "Analog Devices ADT7467 or ADT7468",
-		driver => "to-be-written",
+		driver => "lm85",
 		i2c_addrs => [0x2e],
 		i2c_detect => sub { adt7467_detect(@_, 0); },
 	}, {
