diff -Naur lm_sensors-3.0.1-old/CHANGES lm_sensors-3.0.1-new/CHANGES
--- lm_sensors-3.0.1-old/CHANGES	2008-05-12 13:42:08.000000000 -0700
+++ lm_sensors-3.0.1-new/CHANGES	2008-05-12 13:42:27.000000000 -0700
@@ -25,6 +25,7 @@
                   Add SMSC SCH5514D-NS detection (no sensors)
                   Lower the confidence of MAX6650/MAX6651
                   Add Fintek F71858DG detection
+                  Don't probe I2C addresses 0x40-0x47
                   Add Fintek F81216D detection (no sensors)
   unhide_ICH_SMBus: Add support for the 82801AA (ICH)
 
diff -Naur lm_sensors-3.0.1-old/prog/detect/sensors-detect lm_sensors-3.0.1-new/prog/detect/sensors-detect
--- lm_sensors-3.0.1-old/prog/detect/sensors-detect	2008-01-22 04:22:47.000000000 -0800
+++ lm_sensors-3.0.1-new/prog/detect/sensors-detect	2008-05-12 13:42:27.000000000 -0700
@@ -1269,7 +1269,9 @@
      {
        name => "Maxim MAX6633/MAX6634/MAX6635",
        driver => "lm92",
-       i2c_addrs => [0x40..0x4f],
+       i2c_addrs => [0x48..0x4f], # The MAX6633 can also use 0x40-0x47 but we
+                                  # don't want to probe these addresses, it's
+                                  # dangerous.
        i2c_detect => sub { lm92_detect(2, @_); },
      },
      {
