diff -Naur myththemes-0.21-18831-old/cpsvndir myththemes-0.21-18831-new/cpsvndir
--- myththemes-0.21-18831-old/cpsvndir	2008-10-23 15:52:55.000000000 -0700
+++ myththemes-0.21-18831-new/cpsvndir	2008-10-23 15:52:56.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
