diff -Naur myththemes-0.21-19358-old/cpsvndir myththemes-0.21-19358-new/cpsvndir
--- myththemes-0.21-19358-old/cpsvndir	2008-12-13 13:16:08.000000000 -0800
+++ myththemes-0.21-19358-new/cpsvndir	2008-12-13 13:16:08.000000000 -0800
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
