diff -Naur libX11-1.1.5-old/configure.ac libX11-1.1.5-new/configure.ac
--- libX11-1.1.5-old/configure.ac	2008-09-04 14:38:59.000000000 -0700
+++ libX11-1.1.5-new/configure.ac	2008-09-05 04:00:14.000000000 -0700
@@ -38,8 +38,18 @@
 case "$ac_cv_use_xcb" in
 no)
 	X11_REQUIRES="xau xcmiscproto bigreqsproto"
-	PKG_CHECK_MODULES(XDMCP, xdmcp)
-	AC_CHECK_LIB(Xdmcp, XdmcpWrap, [xdmauth="yes"], [xdmauth="no"], [$XDMCP_LIBS])
+	X11_EXTRA_DEPS="xau"
+	PKG_CHECK_MODULES(XDMCP, xdmcp,
+		AC_CHECK_LIB(Xdmcp, XdmcpWrap,
+			[
+	        	AC_CHECK_LIB(Xdmcp, XdmcpWrap, [xdmauth="yes"], [xdmauth="no"], [$XDMCP_LIBS])
+	        	X11_EXTRA_DEPS="$X11_EXTRA_DEPS xdmcp"
+			],
+			[
+			XDMCP_CFLAGS=
+			XDMCP_LIBS=
+			], [$XDMCP_LIBS]),
+		[AC_MSG_RESULT(no)])
 	AC_DEFINE(USE_XCB, 0, [Use XCB for low-level protocol implementation])
 	;;
 *)
diff -Naur libX11-1.1.5-old/src/ConnDis.c libX11-1.1.5-new/src/ConnDis.c
--- libX11-1.1.5-old/src/ConnDis.c	2008-09-04 14:35:13.000000000 -0700
+++ libX11-1.1.5-new/src/ConnDis.c	2008-09-05 04:00:14.000000000 -0700
@@ -39,7 +39,9 @@
 #include <X11/Xlibint.h>
 #include <X11/Xtrans/Xtrans.h>
 #include <X11/Xauth.h>
+#ifdef HASXDMAUTH
 #include <X11/Xdmcp.h>
+#endif
 #include <stdio.h>
 #include <ctype.h>
 #include <unistd.h>
diff -Naur libX11-1.1.5-old/x11.pc.in libX11-1.1.5-new/x11.pc.in
--- libX11-1.1.5-old/x11.pc.in	2007-10-13 06:24:34.000000000 -0700
+++ libX11-1.1.5-new/x11.pc.in	2008-09-05 04:00:14.000000000 -0700
@@ -9,7 +9,7 @@
 Description: X Library
 Version: @PACKAGE_VERSION@
 Requires: xproto @XKBPROTO_REQUIRES@
-Requires.private: xau xdmcp @X11_EXTRA_DEPS@
+Requires.private: @X11_EXTRA_DEPS@
 Cflags: -I${includedir} @XTHREAD_CFLAGS@
 Libs: -L${libdir} -lX11
 Libs.private: @XTHREADLIB@
