diff -Naur xf86-video-geode-2.11.5-old/configure.ac xf86-video-geode-2.11.5-new/configure.ac
--- xf86-video-geode-2.11.5-old/configure.ac	2009-09-22 06:46:03.000000000 -0700
+++ xf86-video-geode-2.11.5-new/configure.ac	2009-09-22 16:41:41.000000000 -0700
@@ -43,6 +43,7 @@
 AC_PROG_LIBTOOL
 AC_PROG_CC
 AM_PROG_AS
+PKG_PROG_PKG_CONFIG
 
 AH_TOP([#include "xorg-server.h"])
 
@@ -91,7 +92,8 @@
                   HAVE_XEXTPROTO_71="yes"; AC_DEFINE(HAVE_XEXTPROTO_71, 1, [xextproto 7.1 available]),
                   HAVE_XEXTPROTO_71="no")
 AM_CONDITIONAL(HAVE_XEXTPROTO_71, [ test "$HAVE_XEXTPROTO_71" = "yes" ])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 SAVE_CPPFLAGS="$CPPFLAGS"
