diff -Naur myththemes-addons-0.20-14914-old/configure myththemes-addons-0.20-14914-new/configure
--- myththemes-addons-0.20-14914-old/configure	2008-02-29 19:44:50.000000000 -0800
+++ myththemes-addons-0.20-14914-new/configure	2008-03-26 08:46:24.000000000 -0700
@@ -36,4 +36,8 @@
   exit
 fi
 
-qmake themes.pro
+if test -x $QTDIR/bin/qmake; then
+  $QTDIR/bin/qmake QMAKE=${QTDIR}/bin/qmake themes.pro
+else
+  qmake myththemes.pro
+fi
