diff -Naur mythtv-16133-old/programs/mythbackend/upnpmedia.cpp mythtv-16133-new/programs/mythbackend/upnpmedia.cpp
--- mythtv-16133-old/programs/mythbackend/upnpmedia.cpp	2008-02-18 07:46:55.000000000 -0800
+++ mythtv-16133-new/programs/mythbackend/upnpmedia.cpp	2008-02-18 07:47:52.000000000 -0800
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include "httprequest.h"
 #include <qfileinfo.h>
 #include <qregexp.h>
