diff -Naur mythtv-16133-old/libs/libmythtv/tv_play.cpp mythtv-16133-new/libs/libmythtv/tv_play.cpp
--- mythtv-16133-old/libs/libmythtv/tv_play.cpp	2008-02-18 08:00:55.000000000 -0800
+++ mythtv-16133-new/libs/libmythtv/tv_play.cpp	2008-02-18 08:01:45.000000000 -0800
@@ -1745,7 +1745,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
