diff -Naur mythtv-15728-old/libs/libavcodec/libavcodec.pro mythtv-15728-new/libs/libavcodec/libavcodec.pro
--- mythtv-15728-old/libs/libavcodec/libavcodec.pro	2008-02-02 09:19:39.000000000 -0800
+++ mythtv-15728-new/libs/libavcodec/libavcodec.pro	2008-02-02 09:20:45.000000000 -0800
@@ -361,11 +361,13 @@
 using_xvmc {
     SOURCES *= xvmcvideo.c
     DEFINES += HAVE_XVMC
+    LIBS += $$CONFIG_XVMC_LIBS
 }
 
 using_xvmc_vld {
     SOURCES += xvmcvldvideo.c
     DEFINES += HAVE_XVMC_VLD
+    LIBS += $$CONFIG_XVMC_LIBS
 }
 
 using_dvdv {
