diff -Naur mythtv-14921-old/libs/libmythtv/tv_play.cpp mythtv-14921-new/libs/libmythtv/tv_play.cpp
--- mythtv-14921-old/libs/libmythtv/tv_play.cpp	2007-11-22 10:24:45.000000000 -0800
+++ mythtv-14921-new/libs/libmythtv/tv_play.cpp	2007-11-29 06:32:44.000000000 -0800
@@ -1389,7 +1389,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
