diff -Naur mythtv-13228-old/themes/cpsvndir mythtv-13228-new/themes/cpsvndir
--- mythtv-13228-old/themes/cpsvndir	2007-04-15 12:01:21.000000000 -0700
+++ mythtv-13228-new/themes/cpsvndir	2007-04-15 12:02:12.000000000 -0700
@@ -8,7 +8,7 @@
 mkdir -p $DESTDIR/$SOURCE || exit 1
 
 find $SOURCE -path '*/.svn' -prune -or -type d -exec mkdir -p $DESTDIR/{} \;
-find $SOURCE -path '*/.svn' -prune -or -type f -exec cp {} $DESTDIR/{} \;
+find $SOURCE -path '*/.svn' -prune -or -type f -exec cp --preserve=timestamps {} $DESTDIR/{} \;
 
 
 exit 0
