diff -Naur mythtv-15728-old/libs/libmythtv/tv_play.cpp mythtv-15728-new/libs/libmythtv/tv_play.cpp
--- mythtv-15728-old/libs/libmythtv/tv_play.cpp	2008-02-02 09:27:53.000000000 -0800
+++ mythtv-15728-new/libs/libmythtv/tv_play.cpp	2008-02-02 09:28:38.000000000 -0800
@@ -1738,7 +1738,7 @@
     SetupPlayer(isWatchingRecording);
     pthread_create(&decode, NULL, SpawnDecode, nvp);
 
-    maxWait = (maxWait <= 0) ? 20000 : maxWait;
+    maxWait = (maxWait <= 0) ? 40000 : maxWait;
 #ifdef USING_VALGRIND
     maxWait = (1<<30);
 #endif // USING_VALGRIND
