diff -Naur module-init-tools-3.2.2-old/Makefile.in module-init-tools-3.2.2-new/Makefile.in
--- module-init-tools-3.2.2-old/Makefile.in	2005-12-06 20:43:46.000000000 -0800
+++ module-init-tools-3.2.2-new/Makefile.in	2005-12-07 20:29:25.000000000 -0800
@@ -103,8 +103,8 @@
 EXTRA_depmod_SOURCES = moduleops_core.c
 EXTRA_modinfo_SOURCES = backwards_compat.c
 
-MAN5 = modprobe.conf.5 modules.dep.5
-MAN8 = depmod.8 insmod.8 lsmod.8 rmmod.8 modprobe.8 modinfo.8
+MAN5 =
+MAN8 =
 SGML = $(addprefix doc/,  $(MAN5:%.5=%.sgml) $(MAN8:%.8=%.sgml))
 man_MANS = $(MAN5) $(MAN8)
 
