diff -Naur libusb-0.1.8-old/libusb-config.in libusb-0.1.8-new/libusb-config.in
--- libusb-0.1.8-old/libusb-config.in	2004-01-27 14:36:37.000000000 -0800
+++ libusb-0.1.8-new/libusb-config.in	2004-12-30 08:48:43.538311119 -0800
@@ -50,8 +50,8 @@
       exit 0
       ;;
     --cflags)
-      if test "@includedir@" != /usr/include ; then
-        includes="-I@includedir@"
+      if test "${DESTDIR}@includedir@" != /usr/include ; then
+        includes="-I${DESTDIR}@includedir@"
       fi
       echo_cflags=yes
       ;;
@@ -75,5 +75,5 @@
 	echo $includes
 fi
 if test "$echo_libs" = "yes"; then
-	echo -L@libdir@ -lusb @OSLIBS@
+	echo -L${DESTDIR}@libdir@ -lusb @OSLIBS@
 fi
