diff -Naur libusb-0.1.11-old/linux.c libusb-0.1.11-new/linux.c
--- libusb-0.1.11-old/linux.c	2006-01-19 15:25:05.000000000 -0800
+++ libusb-0.1.11-new/linux.c	2006-01-29 23:20:30.000000000 -0800
@@ -565,7 +565,7 @@
     if (!dev->children) {
       if (usb_debug > 1)
         fprintf(stderr, "error allocating %d bytes memory for dev->children\n",
-                sizeof(struct usb_device *) * dev->num_children);
+                (int)(sizeof(struct usb_device *) * dev->num_children));
 
       dev->num_children = 0;
       close(fd);
