diff -Naur v4l-utils-0.8.4-old/utils/keytable/keytable.c v4l-utils-0.8.4-new/utils/keytable/keytable.c
--- v4l-utils-0.8.4-old/utils/keytable/keytable.c	2011-06-01 04:23:03.000000000 -0700
+++ v4l-utils-0.8.4-new/utils/keytable/keytable.c	2011-07-10 07:22:14.000000000 -0700
@@ -108,7 +108,7 @@
 	"\nOn the options bellow, the arguments are:\n"
 	"  DEV      - the /dev/input/event* device to control\n"
 	"  SYSDEV   - the ir class as found at /sys/class/rc\n"
-	"  TABLE    - a file wit a set of scancode=keycode value pairs\n"
+	"  TABLE    - a file with a set of scancode=keycode value pairs\n"
 	"  SCANKEY  - a set of scancode1=keycode1,scancode2=keycode2.. value pairs\n"
 	"  PROTOCOL - protocol name (nec, rc-5, rc-6, other) to be enabled\n"
 	"  DELAY    - Delay before repeating a keystroke\n"
@@ -267,7 +267,7 @@
 				perror("value");
 		}
 
-		nextkey->codes[0] = (unsigned) strtol(scancode, NULL, 0);
+		nextkey->codes[0] = (unsigned) strtoul(scancode, NULL, 0);
 		nextkey->codes[1] = (unsigned) value;
 		nextkey->next = calloc(1, sizeof(*nextkey));
 		if (!nextkey->next) {
@@ -410,7 +410,7 @@
 		do {
 			if (!p)
 				goto err_inval;
-			nextkey->codes[0] = strtol(p, NULL, 0);
+			nextkey->codes[0] = strtoul(p, NULL, 0);
 			if (errno)
 				goto err_inval;
 
@@ -426,7 +426,7 @@
 			nextkey->codes[1] = key;
 
 			if (debug)
-				fprintf(stderr, "scancode %i=%i\n",
+				fprintf(stderr, "scancode %u=%u\n",
 					nextkey->codes[0], nextkey->codes[1]);
 
 			nextkey->next = calloc(1, sizeof(keys));
@@ -1427,9 +1427,9 @@
 		int rc;
 
 		for (cur = &cfg; cur->next; cur = cur->next) {
-			if (strcasecmp(cur->driver, rc_dev.drv_name) && strcasecmp(cur->driver, "*"))
+			if ((!rc_dev.drv_name || strcasecmp(cur->driver, rc_dev.drv_name)) && strcasecmp(cur->driver, "*"))
 				continue;
-			if (strcasecmp(cur->table, rc_dev.keytable_name) && strcasecmp(cur->table, "*"))
+			if ((!rc_dev.keytable_name || strcasecmp(cur->table, rc_dev.keytable_name)) && strcasecmp(cur->table, "*"))
 				continue;
 			break;
 		}
