diff -Naur patch-2.5.4-old/common.h patch-2.5.4-new/common.h
--- patch-2.5.4-old/common.h	1999-08-29 23:20:08.000000000 -0700
+++ patch-2.5.4-new/common.h	2006-10-18 14:56:00.000000000 -0700
@@ -118,6 +118,10 @@
 #define ISSPACE(c) (CTYPE_DOMAIN (c) && isspace (c))
 #endif
 
+#ifndef ISTAB
+#define ISTAB(c) (CTYPE_DOMAIN (c) && (c) == 0011)
+#endif
+
 #ifndef ISDIGIT
 #define ISDIGIT(c) ((unsigned) (c) - '0' <= 9)
 #endif
diff -Naur patch-2.5.4-old/util.c patch-2.5.4-new/util.c
--- patch-2.5.4-old/util.c	1999-08-29 23:20:08.000000000 -0700
+++ patch-2.5.4-new/util.c	2006-10-18 14:58:24.000000000 -0700
@@ -937,7 +937,7 @@
 	    if (strip_leading < 0 || --sleading >= 0)
 		name = t+1;
 	  }
-	else if (ISSPACE ((unsigned char) *t))
+	else if (ISTAB ((unsigned char) *t))
 	  {
 	    char const *u = t;
 
