diff -Naur mythtv-0.18.1-old/configure mythtv-0.18.1-new/configure
--- mythtv-0.18.1-old/configure	2006-05-03 03:03:31.000000000 -0700
+++ mythtv-0.18.1-new/configure	2006-05-03 03:05:46.000000000 -0700
@@ -1716,18 +1716,21 @@
     fi
 fi
 
-if has_library libviaXvMC ; then
-    echo > /dev/null
-else
-    xvmc_vld="no"
-fi
 if test x"$xvmc_vld" = x"yes" ; then
+    xvmc_vld="no"
     if has_library libXvMCW ; then
         VENDOR_XVMC_LIBS="-lXvMCW"
-    else
+        xvmc_vld="yes"
+        xvmc="yes"
+    elif has_library libviaXvMC ; then
         VENDOR_XVMC_LIBS="-lviaXvMC"
+        xvmc_vld="yes"
+        xvmc="yes"
+    elif has_library libviaXvMCPro ; then
+        VENDOR_XVMC_LIBS="-lviaXvMCPro"
+        xvmc_vld="yes"
+        xvmc="yes"
     fi
-    xvmc="yes"
 fi
 
 if test x"$VENDOR_XVMC_LIBS" = x"-lXvMCW" ; then
