diff -Naur uvc-182-old/Makefile uvc-182-new/Makefile
--- uvc-182-old/Makefile	2008-02-14 19:47:19.000000000 -0800
+++ uvc-182-new/Makefile	2008-02-14 19:48:27.000000000 -0800
@@ -17,7 +17,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Module.symvers version.h
