diff -Naur uvc-168-old/Makefile uvc-168-new/Makefile
--- uvc-168-old/Makefile	2008-01-16 22:16:21.000000000 -0800
+++ uvc-168-new/Makefile	2008-01-16 22:16:52.000000000 -0800
@@ -16,7 +16,6 @@
 install:
 	@echo "Installing USB Video Class driver..."
 	@(cd $(KERNEL_DIR) && make -C $(KERNEL_DIR) SUBDIRS=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install)
-	depmod -ae
 
 clean:
 	-rm -f *.o *.ko .*.cmd .*.flags *.mod.c Module.symvers
