diff -Naur file-5.11-old/configure.ac file-5.11-new/configure.ac
--- file-5.11-old/configure.ac	2012-02-21 11:16:29.000000000 -0800
+++ file-5.11-new/configure.ac	2012-12-29 00:27:43.000000000 -0800
@@ -3,7 +3,7 @@
 AM_INIT_AUTOMAKE()
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_MSG_CHECKING(for builtin ELF support)
