diff -Naur fontconfig-2.3.2-old/fc-case/Makefile.am fontconfig-2.3.2-new/fc-case/Makefile.am
--- fontconfig-2.3.2-old/fc-case/Makefile.am	2005-01-13 10:16:20.000000000 -0800
+++ fontconfig-2.3.2-new/fc-case/Makefile.am	2006-04-23 09:46:42.000000000 -0700
@@ -23,6 +23,9 @@
 #
 
 CC = @CC_FOR_BUILD@
+CPPFLAGS =
+CFLAGS =
+LDFLAGS =
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.3.2-old/fc-case/Makefile.in fontconfig-2.3.2-new/fc-case/Makefile.in
--- fontconfig-2.3.2-old/fc-case/Makefile.in	2005-04-27 12:08:48.000000000 -0700
+++ fontconfig-2.3.2-new/fc-case/Makefile.in	2006-04-23 09:47:29.000000000 -0700
@@ -74,10 +74,10 @@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
-CFLAGS = @CFLAGS@
+CFLAGS =
 CONFDIR = @CONFDIR@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS =
 CROSS_COMPILING_FALSE = @CROSS_COMPILING_FALSE@
 CROSS_COMPILING_TRUE = @CROSS_COMPILING_TRUE@
 CXX = @CXX@
@@ -117,7 +117,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
-LDFLAGS = @LDFLAGS@
+LDFLAGS =
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
diff -Naur fontconfig-2.3.2-old/fc-glyphname/Makefile.am fontconfig-2.3.2-new/fc-glyphname/Makefile.am
--- fontconfig-2.3.2-old/fc-glyphname/Makefile.am	2005-01-13 10:16:20.000000000 -0800
+++ fontconfig-2.3.2-new/fc-glyphname/Makefile.am	2006-04-23 09:47:47.000000000 -0700
@@ -23,6 +23,9 @@
 #
 
 CC = @CC_FOR_BUILD@
+CPPFLAGS =
+CFLAGS =
+LDFLAGS =
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.3.2-old/fc-glyphname/Makefile.in fontconfig-2.3.2-new/fc-glyphname/Makefile.in
--- fontconfig-2.3.2-old/fc-glyphname/Makefile.in	2005-04-27 12:08:48.000000000 -0700
+++ fontconfig-2.3.2-new/fc-glyphname/Makefile.in	2006-04-23 09:48:06.000000000 -0700
@@ -74,10 +74,10 @@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
-CFLAGS = @CFLAGS@
+CFLAGS =
 CONFDIR = @CONFDIR@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS =
 CROSS_COMPILING_FALSE = @CROSS_COMPILING_FALSE@
 CROSS_COMPILING_TRUE = @CROSS_COMPILING_TRUE@
 CXX = @CXX@
@@ -117,7 +117,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
-LDFLAGS = @LDFLAGS@
+LDFLAGS =
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
diff -Naur fontconfig-2.3.2-old/fc-lang/Makefile.am fontconfig-2.3.2-new/fc-lang/Makefile.am
--- fontconfig-2.3.2-old/fc-lang/Makefile.am	2005-01-13 10:16:20.000000000 -0800
+++ fontconfig-2.3.2-new/fc-lang/Makefile.am	2006-04-23 09:48:26.000000000 -0700
@@ -23,6 +23,9 @@
 #
 
 CC = @CC_FOR_BUILD@
+CPPFLAGS =
+CFLAGS =
+LDFLAGS =
 EXEEXT = @EXEEXT_FOR_BUILD@
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
diff -Naur fontconfig-2.3.2-old/fc-lang/Makefile.in fontconfig-2.3.2-new/fc-lang/Makefile.in
--- fontconfig-2.3.2-old/fc-lang/Makefile.in	2005-04-27 12:08:48.000000000 -0700
+++ fontconfig-2.3.2-new/fc-lang/Makefile.in	2006-04-23 09:48:44.000000000 -0700
@@ -74,10 +74,10 @@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
-CFLAGS = @CFLAGS@
+CFLAGS =
 CONFDIR = @CONFDIR@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS =
 CROSS_COMPILING_FALSE = @CROSS_COMPILING_FALSE@
 CROSS_COMPILING_TRUE = @CROSS_COMPILING_TRUE@
 CXX = @CXX@
@@ -117,7 +117,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
-LDFLAGS = @LDFLAGS@
+LDFLAGS =
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
