diff -Naur xf86-video-openchrome-307-old/unichrome/via_bandwidth.c xf86-video-openchrome-307-new/unichrome/via_bandwidth.c
--- xf86-video-openchrome-307-old/unichrome/via_bandwidth.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_bandwidth.c	2007-03-24 09:51:19.000000000 -0700
@@ -226,6 +226,16 @@
 	else
 	    ViaSeqMask(hwp, 0x22, 0x00, 0x1F);
 	break;
+    case VIA_CX700:
+        hwp->writeSeq(hwp, 0x17, 0x2F);
+        ViaSeqMask(hwp, 0x16, 0x14, 0xBF);
+        ViaSeqMask(hwp, 0x18, 0x08, 0xBF);
+
+        if ((mode->HDisplay >= 1400) && (pScrn->bitsPerPixel == 32))
+            ViaSeqMask(hwp, 0x22, 0x10, 0x1F);
+        else
+            ViaSeqMask(hwp, 0x22, 0x00, 0x1F);
+        break;
 
     default:
 	xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "ViaSetPrimaryFIFO:"
@@ -343,6 +353,22 @@
 	else
 	    ViaCrtcMask(hwp, 0x94, 0x20, 0x7F);
 	break;
+    case VIA_CX700:
+        ViaCrtcMask(hwp, 0x68, 0xA0, 0xF0);
+        ViaCrtcMask(hwp, 0x94, 0x00, 0x80);
+        ViaCrtcMask(hwp, 0x95, 0x00, 0x80);
+
+        ViaCrtcMask(hwp, 0x68, 0x04, 0x0F);
+        ViaCrtcMask(hwp, 0x95, 0x10, 0x70);
+
+        ViaCrtcMask(hwp, 0x92, 0x08, 0x0F);
+        ViaCrtcMask(hwp, 0x95, 0x00, 0x07);
+
+        if ((mode->HDisplay >= 1400) && (pScrn->bitsPerPixel == 32))
+            ViaCrtcMask(hwp, 0x94, 0x10, 0x7F);
+        else
+            ViaCrtcMask(hwp, 0x94, 0x20, 0x7F);
+        break;
     default:
 	xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "ViaSetSecondaryFIFO:"
 		   " Chipset %d not implemented\n", pVia->Chipset);
diff -Naur xf86-video-openchrome-307-old/unichrome/via_dri.c xf86-video-openchrome-307-new/unichrome/via_dri.c
--- xf86-video-openchrome-307-old/unichrome/via_dri.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_dri.c	2007-03-24 09:51:19.000000000 -0700
@@ -185,6 +185,7 @@
 	switch (pVia->ChipId) {
 	case PCI_CHIP_VT3314:
 	case PCI_CHIP_VT3259:
+	case PCI_CHIP_VT3157:
 	    pVIADRI->reg_pause_addr = 0x40c;
 	    break;
 	default:
diff -Naur xf86-video-openchrome-307-old/unichrome/via_driver.c xf86-video-openchrome-307-new/unichrome/via_driver.c
--- xf86-video-openchrome-307-old/unichrome/via_driver.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_driver.c	2007-03-24 09:51:19.000000000 -0700
@@ -105,6 +105,7 @@
     {VIA_K8M800,   "K8M800"},
     {VIA_PM800,    "PM800/PM880/CN400"},
     {VIA_VM800,    "VM800/CN700/P4M800Pro"},
+    {VIA_CX700,    "CX700"},
     {-1,            NULL }
 };
 
@@ -116,6 +117,7 @@
     {VIA_K8M800,   PCI_CHIP_VT3204,    RES_SHARED_VGA},
     {VIA_PM800,    PCI_CHIP_VT3259,    RES_SHARED_VGA},
     {VIA_VM800,    PCI_CHIP_VT3314,    RES_SHARED_VGA},
+    {VIA_CX700,    PCI_CHIP_VT3157,    RES_SHARED_VGA},
     {-1,            -1,                RES_UNDEFINED}
 };
 
diff -Naur xf86-video-openchrome-307-old/unichrome/via_id.c xf86-video-openchrome-307-new/unichrome/via_id.c
--- xf86-video-openchrome-307-old/unichrome/via_id.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_id.c	2007-03-24 09:51:19.000000000 -0700
@@ -185,6 +185,8 @@
     {"Fujitsu/Siemens Amilo L7320",           VIA_VM800,   0x1734, 0x10CD, VIA_DEVICE_CRT | VIA_DEVICE_LCD},
     {"ASRock P4VM800",                        VIA_VM800,   0x1849, 0x3344, VIA_DEVICE_CRT},
     {"Asustek P5V800-MX",                     VIA_VM800,   0x3344, 0x1122, VIA_DEVICE_CRT},
+    /* CX700 */
+    {"VIA VT3157 (CX700)",                    VIA_CX700,   0x1106, 0x3157, VIA_DEVICE_CRT},
     /* keep this */
     {NULL,                                    VIA_UNKNOWN, 0x0000, 0x0000, VIA_DEVICE_NONE}
 };
diff -Naur xf86-video-openchrome-307-old/unichrome/via_id.h xf86-video-openchrome-307-new/unichrome/via_id.h
--- xf86-video-openchrome-307-old/unichrome/via_id.h	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_id.h	2007-03-24 09:51:19.000000000 -0700
@@ -34,6 +34,7 @@
     VIA_K8M800,
     VIA_PM800,
     VIA_VM800,
+    VIA_CX700,
     VIA_LAST
 };
 
@@ -56,6 +57,7 @@
 #define PCI_CHIP_CLE3122        0x3122 /* CLE266 */
 #define PCI_CHIP_VT3205         0x7205 /* KM400 */
 #define PCI_CHIP_VT3314         0x3344 /* VM800 */
+#define PCI_CHIP_VT3157         0x3157 /* CX700 */
 
 /*
  * There is also quite some conflicting information on the
diff -Naur xf86-video-openchrome-307-old/unichrome/via_mode.c xf86-video-openchrome-307-new/unichrome/via_mode.c
--- xf86-video-openchrome-307-old/unichrome/via_mode.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_mode.c	2007-03-24 09:51:19.000000000 -0700
@@ -794,6 +794,8 @@
 	return ViaBandwidthTable[VIA_BW_PM800].Bandwidth[pVia->MemClk];
     case VIA_VM800:
 	return ViaBandwidthTable[VIA_BW_VM800].Bandwidth[pVia->MemClk];
+    case VIA_CX700:
+	return ViaBandwidthTable[VIA_BW_CX700].Bandwidth[pVia->MemClk];
     default:
 	xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "ViaBandwidthAllowed: Unknown Chipset.\n");
 	return VIA_BW_MIN;
@@ -1713,7 +1715,8 @@
 	   is removed -- copy from clock handling code below */
 	if ((pVia->Chipset == VIA_CLE266) && CLE266_REV_IS_AX(pVia->ChipRev))
 	    ViaSetPrimaryDotclock(pScrn, 0x471C); /* CLE266Ax use 2x XCLK */
-	else if ((pVia->Chipset == VIA_K8M800) || (pVia->Chipset == VIA_PM800) || (pVia->Chipset == VIA_VM800))
+	else if ((pVia->Chipset == VIA_K8M800) || (pVia->Chipset == VIA_PM800) ||
+	         (pVia->Chipset == VIA_VM800) || (pVia->Chipset == VIA_CX700))
 	    ViaSetPrimaryDotclock(pScrn, 0x529001);
 	else
 	    ViaSetPrimaryDotclock(pScrn, 0x871C);
@@ -1728,11 +1731,13 @@
     if (pBIOSInfo->ClockExternal) {
 	if ((pVia->Chipset == VIA_CLE266) && CLE266_REV_IS_AX(pVia->ChipRev))
 	    ViaSetPrimaryDotclock(pScrn, 0x471C); /* CLE266Ax use 2x XCLK */
-	else if ((pVia->Chipset == VIA_K8M800) || (pVia->Chipset == VIA_PM800) ||(pVia->Chipset == VIA_VM800))
+	else if ((pVia->Chipset == VIA_K8M800) || (pVia->Chipset == VIA_PM800) ||
+	         (pVia->Chipset == VIA_VM800) || (pVia->Chipset == VIA_CX700))
 	    ViaSetPrimaryDotclock(pScrn, 0x529001);
 	else
 	    ViaSetPrimaryDotclock(pScrn, 0x871C);
-	if ((pVia->Chipset != VIA_K8M800) && (pVia->Chipset != VIA_PM800) && (pVia->Chipset != VIA_VM800))
+	if ((pVia->Chipset != VIA_K8M800) && (pVia->Chipset != VIA_PM800) &&
+	    (pVia->Chipset != VIA_VM800) && (pVia->Chipset != VIA_CX700))
 	  ViaCrtcMask(hwp, 0x6B, 0x01, 0x01);
     } else {
 	ViaSetPrimaryDotclock(pScrn, pBIOSInfo->Clock);
diff -Naur xf86-video-openchrome-307-old/unichrome/via_mode.h xf86-video-openchrome-307-new/unichrome/via_mode.h
--- xf86-video-openchrome-307-old/unichrome/via_mode.h	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_mode.h	2007-03-24 09:51:19.000000000 -0700
@@ -41,7 +41,8 @@
 #define VIA_BW_K8M800   4
 #define VIA_BW_PM800    5
 #define VIA_BW_VM800    6
-#define VIA_BW_ALL      7
+#define VIA_BW_CX700    7
+#define VIA_BW_ALL      8
 
 /*
  * 393216000 is for SDR133 in via_refresh.h
@@ -57,7 +58,8 @@
     { VIA_BW_KM400A,  { VIA_BW_MIN, VIA_BW_MIN, VIA_BW_MIN,  394000000,  461000000,  461000000,  461000000, VIA_BW_MIN } },
     { VIA_BW_K8M800,  { VIA_BW_MIN, VIA_BW_MIN, VIA_BW_MIN,  394000000,  461000000,  461000000,  461000000, VIA_BW_MIN } },
     { VIA_BW_PM800,   { VIA_BW_MIN, VIA_BW_MIN, VIA_BW_MIN,  394000000,  461000000,  461000000,  461000000,  922000000 } },
-    { VIA_BW_VM800,   { VIA_BW_MIN, VIA_BW_MIN, VIA_BW_MIN,  394000000,  461000000,  461000000,  461000000,  922000000 } }
+    { VIA_BW_VM800,   { VIA_BW_MIN, VIA_BW_MIN, VIA_BW_MIN,  394000000,  461000000,  461000000,  461000000,  922000000 } },
+    { VIA_BW_CX700,   { VIA_BW_MIN, VIA_BW_MIN, VIA_BW_MIN,  394000000,  461000000,  461000000,  461000000,  922000000 } }
 };
     
 /*
diff -Naur xf86-video-openchrome-307-old/unichrome/via_swov.c xf86-video-openchrome-307-new/unichrome/via_swov.c
--- xf86-video-openchrome-307-old/unichrome/via_swov.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_swov.c	2007-03-24 09:51:19.000000000 -0700
@@ -250,6 +250,13 @@
 	HWDiff->dwHQVInitPatch = VID_HWDIFF_FALSE;
 	HWDiff->dwHQVDisablePatch = VID_HWDIFF_TRUE;
 	break;
+    case VIA_CX700:
+        HWDiff->dwThreeHQVBuffer = VID_HWDIFF_TRUE;
+        HWDiff->dwHQVFetchByteUnit = VID_HWDIFF_TRUE;
+        HWDiff->dwSupportTwoColorKey = VID_HWDIFF_FALSE;
+        HWDiff->dwHQVInitPatch = VID_HWDIFF_FALSE;
+        HWDiff->dwHQVDisablePatch = VID_HWDIFF_TRUE;
+        break;
     default:
 	xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
 	    "VIAVidHWDiffInit: Unhandled ChipSet.\n");
@@ -739,6 +746,7 @@
     case PCI_CHIP_VT3204:
     case PCI_CHIP_VT3259:
     case PCI_CHIP_VT3314:
+    case PCI_CHIP_VT3157:
 	model = 0;
 	break;
     case PCI_CHIP_CLE3122:
@@ -867,6 +875,7 @@
     case PCI_CHIP_VT3204:
     case PCI_CHIP_VT3259:
     case PCI_CHIP_VT3314:
+    case PCI_CHIP_VT3157:
 	VIDOutD(V3_ColorSpaceReg_1, col1);
 	VIDOutD(V3_ColorSpaceReg_2, col2);
 	DBG_DD(ErrorF("000002C4 %08lx\n", col1));
@@ -895,6 +904,7 @@
     case PCI_CHIP_VT3204:
     case PCI_CHIP_VT3259:
     case PCI_CHIP_VT3314:
+    case PCI_CHIP_VT3157:
 	return VIDEO_HQV_INUSE | SW_USE_HQV | VIDEO_3_INUSE;
     case PCI_CHIP_CLE3122:
 	return VIDEO_HQV_INUSE | SW_USE_HQV | VIDEO_1_INUSE;
@@ -926,6 +936,7 @@
 	case PCI_CHIP_VT3204:
 	case PCI_CHIP_VT3259:
 	case PCI_CHIP_VT3314:
+	case PCI_CHIP_VT3157:
 	    return V3_ENABLE | V3_EXPIRE_NUM_3205;
 
 	case PCI_CHIP_CLE3122:
@@ -1205,7 +1216,7 @@
 static void
 SetFIFO_V3(VIAPtr pVia, CARD8 depth, CARD8 prethreshold, CARD8 threshold)
 {
-    if (pVia->ChipId == PCI_CHIP_VT3314) {
+    if ((pVia->ChipId == PCI_CHIP_VT3314) || (pVia->ChipId == PCI_CHIP_VT3157)){
 	SaveVideoRegister(pVia, ALPHA_V3_FIFO_CONTROL,
 	    (VIDInD(ALPHA_V3_FIFO_CONTROL) & ALPHA_FIFO_MASK) |
 	    ((depth - 1) & 0xff) | ((threshold & 0xff) << 8));
@@ -1267,6 +1278,7 @@
 	SetFIFO_V3(pVia, 100, 89, 89);
 	break;
     case PCI_CHIP_VT3314:
+    case PCI_CHIP_VT3157:
 	SetFIFO_V3(pVia, 64, 61, 61);
 	break;
     case PCI_CHIP_VT3205:
diff -Naur xf86-video-openchrome-307-old/unichrome/via_video.c xf86-video-openchrome-307-new/unichrome/via_video.c
--- xf86-video-openchrome-307-old/unichrome/via_video.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_video.c	2007-03-24 09:51:19.000000000 -0700
@@ -278,7 +278,8 @@
     if (pVia->ChipId != PCI_CHIP_VT3205 &&
 	pVia->ChipId != PCI_CHIP_VT3204 &&
 	pVia->ChipId != PCI_CHIP_VT3259 &&
-	pVia->ChipId != PCI_CHIP_VT3314) {
+	pVia->ChipId != PCI_CHIP_VT3314 &&
+	pVia->ChipId != PCI_CHIP_VT3157) {
 	CARD32 bandwidth = (mode->HDisplay >> 4) * (mode->VDisplay >> 5) *
 	    pScrn->bitsPerPixel * mode->VRefresh;
 
@@ -555,7 +556,9 @@
 	((pVia->Chipset == VIA_CLE266) ||
 	(pVia->Chipset == VIA_KM400) ||
 	(pVia->Chipset == VIA_K8M800) ||
-	(pVia->Chipset == VIA_PM800) || (pVia->Chipset == VIA_VM800));
+	(pVia->Chipset == VIA_PM800) ||
+	(pVia->Chipset == VIA_VM800) ||
+        (pVia->Chipset == VIA_CX700));
     if ((pVia->drmVerMajor < 2) ||
 	((pVia->drmVerMajor == 2) && (pVia->drmVerMinor < 9)))
 	pVia->useDmaBlit = FALSE;
@@ -571,7 +574,7 @@
 
     if ((pVia->Chipset == VIA_CLE266) || (pVia->Chipset == VIA_KM400) ||
 	(pVia->Chipset == VIA_K8M800) || (pVia->Chipset == VIA_PM800) ||
-	(pVia->Chipset == VIA_VM800)) {
+	(pVia->Chipset == VIA_VM800) || (pVia->Chipset == VIA_CX700)) {
 	num_new = viaSetupAdaptors(pScreen, &newAdaptors);
 	num_adaptors = xf86XVListGenericAdaptors(pScrn, &adaptors);
     } else {
diff -Naur xf86-video-openchrome-307-old/unichrome/via_xvmc.c xf86-video-openchrome-307-new/unichrome/via_xvmc.c
--- xf86-video-openchrome-307-old/unichrome/via_xvmc.c	2007-03-24 09:50:41.000000000 -0700
+++ xf86-video-openchrome-307-new/unichrome/via_xvmc.c	2007-03-24 09:51:19.000000000 -0700
@@ -330,7 +330,8 @@
     pVia->XvMCEnabled = 0;
 
     if (!(pVia->Chipset == VIA_CLE266) && !(pVia->Chipset == VIA_K8M800) &&
-	!(pVia->Chipset == VIA_PM800) && !(pVia->Chipset == VIA_VM800)) {
+	!(pVia->Chipset == VIA_PM800) && !(pVia->Chipset == VIA_VM800) &&
+        !(pVia->Chipset == VIA_CX700)) {
 	xf86DrvMsg(pScrn->scrnIndex, X_WARNING,
 	    "[XvMC] Not supported on this chipset.\n");
 	return;
@@ -503,6 +504,7 @@
 	((pVia->Chipset == VIA_CLE266) ||
 	 (pVia->Chipset == VIA_KM400) ||
 	 (pVia->Chipset == VIA_PM800));
+	 /* No VM800 nor CX700? */
     contextRec->chipId = pVia->ChipId;
     contextRec->screen = pScrn->pScreen->myNum;
     contextRec->depth = pScrn->bitsPerPixel;
