diff -Naur xf86-video-intel-2.14.0-old/configure.ac xf86-video-intel-2.14.0-new/configure.ac
--- xf86-video-intel-2.14.0-old/configure.ac	2011-01-07 15:46:04.000000000 -0800
+++ xf86-video-intel-2.14.0-new/configure.ac	2011-01-07 17:52:59.000000000 -0800
@@ -52,6 +52,7 @@
 # Initialize libtool
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
+PKG_PROG_PKG_CONFIG
 
 PKG_CHECK_MODULES(GEN4ASM, [intel-gen4asm >= 1.1], [gen4asm=yes], [gen4asm=no])
 AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
@@ -96,7 +97,8 @@
 PKG_CHECK_MODULES(DRM, [libdrm >= 2.4.23])
 PKG_CHECK_MODULES(PCIACCESS, [pciaccess >= 0.10])
 
-sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 save_CFLAGS="$CFLAGS"
 CFLAGS="$XORG_CFLAGS $DRI_CFLAGS $DRM_CFLAGS"
