diff -Naur mythstream-0.17_2-old/libs/streamstorage.cpp mythstream-0.17_2-new/libs/streamstorage.cpp
--- mythstream-0.17_2-old/libs/streamstorage.cpp	2006-10-08 12:52:22.000000000 -0700
+++ mythstream-0.17_2-new/libs/streamstorage.cpp	2006-10-13 16:49:40.000000000 -0700
@@ -134,7 +134,7 @@
           }
         }
         else
-          if ( fi->fileName() != "list.xml" ) // but it shouldn't be there
+          if ( fi->fileName() != "list.xml" && fi->fileName().right(3) == ".pl" ) // PERL only, leave binaries 
             copyFile(sourceDir + "/" + fi->fileName(), destDir + "/" + fi->fileName() );
         ++it;
     }
