diff -Naur MPlayer-34027-old/configure MPlayer-34027-new/configure
--- MPlayer-34027-old/configure	2011-08-29 08:24:16.000000000 -0700
+++ MPlayer-34027-new/configure	2011-08-29 08:24:37.000000000 -0700
@@ -5734,7 +5734,7 @@
 echocheck "Blu-ray support"
 if test "$_bluray" = auto ; then
   _bluray=no
-  statement_check libbluray/bluray.h 'bd_get_title_info(0, 0)' -lbluray && _bluray=yes
+  statement_check libbluray/bluray.h 'bd_get_title_info(0, 0, 0)' -lbluray && _bluray=yes
 fi
 if test "$_bluray" = yes ; then
   def_bluray='#define CONFIG_LIBBLURAY 1'
diff -Naur MPlayer-34027-old/stream/stream_bluray.c MPlayer-34027-new/stream/stream_bluray.c
--- MPlayer-34027-old/stream/stream_bluray.c	2011-08-29 08:24:11.000000000 -0700
+++ MPlayer-34027-new/stream/stream_bluray.c	2011-08-29 08:24:37.000000000 -0700
@@ -116,7 +116,7 @@
     case STREAM_CTRL_GET_NUM_CHAPTERS: {
         BLURAY_TITLE_INFO *ti;
 
-        ti = bd_get_title_info(b->bd, b->current_title);
+        ti = bd_get_title_info(b->bd, b->current_title, 0);
         if (!ti)
             return STREAM_UNSUPPORTED;
 
@@ -137,7 +137,7 @@
         int64_t pos;
         int r;
 
-        ti = bd_get_title_info(b->bd, b->current_title);
+        ti = bd_get_title_info(b->bd, b->current_title, 0);
         if (!ti)
             return STREAM_UNSUPPORTED;
 
@@ -156,7 +156,7 @@
     case STREAM_CTRL_GET_NUM_ANGLES: {
         BLURAY_TITLE_INFO *ti;
 
-        ti = bd_get_title_info(b->bd, b->current_title);
+        ti = bd_get_title_info(b->bd, b->current_title, 0);
         if (!ti)
             return STREAM_UNSUPPORTED;
 
@@ -175,7 +175,7 @@
         BLURAY_TITLE_INFO *ti;
         int angle = *((int *) arg);
 
-        ti = bd_get_title_info(b->bd, b->current_title);
+        ti = bd_get_title_info(b->bd, b->current_title, 0);
         if (!ti)
             return STREAM_UNSUPPORTED;
 
@@ -236,7 +236,7 @@
     }
 
     /* check for available titles on disc */
-    title_count = bd_get_titles(bd, TITLES_RELEVANT);
+    title_count = bd_get_titles(bd, TITLES_RELEVANT, 0);
     mp_msg(MSGT_IDENTIFY, MSGL_INFO, "ID_BLURAY_TITLES=%d\n", title_count);
     if (!title_count) {
         mp_msg(MSGT_OPEN, MSGL_ERR, MSGTR_BlurayNoTitles);
@@ -250,7 +250,7 @@
         BLURAY_TITLE_INFO *ti;
         int sec, msec;
 
-        ti = bd_get_title_info(bd, i);
+        ti = bd_get_title_info(bd, i, 0);
         if (!ti)
             continue;
 
@@ -284,7 +284,7 @@
            "ID_BLURAY_CURRENT_TITLE=%d\n", title + 1);
 
     /* Get current title information */
-    info = bd_get_title_info(bd, title);
+    info = bd_get_title_info(bd, title, 0);
     if (!info)
         goto err_no_info;
 
