diff -Naur MPlayer-31555-old/configure MPlayer-31555-new/configure
--- MPlayer-31555-old/configure	2010-06-24 13:32:20.000000000 -0700
+++ MPlayer-31555-new/configure	2010-06-24 13:38:42.000000000 -0700
@@ -7592,6 +7592,7 @@
 if test "$_libvpx_lavc" = yes ; then
   def_libvpx_lavc='#define CONFIG_LIBVPX 1'
   _libavdecoders="$_libavdecoders LIBVPX_DECODER"
+  _libavencoders="$_libavencoders LIBVPX_ENCODER"
   codecmodules="libvpx $codecmodules"
 else
   def_libvpx_lavc='#define CONFIG_LIBVPX 0'
