diff -Naur MPlayer-30135-old/libmpcodecs/vd_ffmpeg.c MPlayer-30135-new/libmpcodecs/vd_ffmpeg.c
--- MPlayer-30135-old/libmpcodecs/vd_ffmpeg.c	2009-12-29 08:13:18.000000000 -0800
+++ MPlayer-30135-new/libmpcodecs/vd_ffmpeg.c	2009-12-29 08:15:00.000000000 -0800
@@ -149,7 +149,12 @@
         avcodec_flush_buffers(avctx);
         return CONTROL_TRUE;
     case VDCTRL_QUERY_UNSEEN_FRAMES:
-        return avctx->has_b_frames + 10;
+    {
+        int thread_delay = 0;
+        if(avctx->active_thread_type&FF_THREAD_FRAME)
+            thread_delay = avctx->thread_count - 1;
+        return avctx->has_b_frames + thread_delay + 10;
+    }
     }
     return CONTROL_UNKNOWN;
 }
diff -Naur MPlayer-30135-old/libvo/vo_corevideo.m MPlayer-30135-new/libvo/vo_corevideo.m
--- MPlayer-30135-old/libvo/vo_corevideo.m	2009-12-29 08:13:30.000000000 -0800
+++ MPlayer-30135-new/libvo/vo_corevideo.m	2009-12-29 08:15:00.000000000 -0800
@@ -500,6 +500,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -538,6 +540,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
