diff -Naur MPlayer-31044-old/libvo/vo_corevideo.m MPlayer-31044-new/libvo/vo_corevideo.m
--- MPlayer-31044-old/libvo/vo_corevideo.m	2010-04-19 07:04:15.000000000 -0700
+++ MPlayer-31044-new/libvo/vo_corevideo.m	2010-04-19 07:14:22.000000000 -0700
@@ -495,6 +495,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -533,6 +535,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
