diff -Naur MPlayer-30969-old/libvo/vo_corevideo.m MPlayer-30969-new/libvo/vo_corevideo.m
--- MPlayer-30969-old/libvo/vo_corevideo.m	2010-03-27 08:51:39.000000000 -0700
+++ MPlayer-30969-new/libvo/vo_corevideo.m	2010-03-27 08:57:13.000000000 -0700
@@ -484,6 +484,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -522,6 +524,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
