diff -Naur MPlayer-31618-old/libvo/vo_corevideo.m MPlayer-31618-new/libvo/vo_corevideo.m
--- MPlayer-31618-old/libvo/vo_corevideo.m	2010-07-02 09:13:25.000000000 -0700
+++ MPlayer-31618-new/libvo/vo_corevideo.m	2010-07-02 09:14:16.000000000 -0700
@@ -494,6 +494,8 @@
 	NSRect visibleFrame;
 	CVReturn error = kCVReturnSuccess;
 
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+
 	//config window
 	[window setContentSize:NSMakeSize(vo_dwidth, vo_dheight)];
 
@@ -532,6 +534,8 @@
 
 	if(vo_ontop)
 		[mpGLView ontop];
+
+        [pool release];
 }
 
 /*
