diff -Naur mythweather-0.17-old/mythweather/weather.cpp mythweather-0.17-new/mythweather/weather.cpp
--- mythweather-0.17-old/mythweather/weather.cpp	2005-02-09 11:35:24.000000000 -0800
+++ mythweather-0.17-new/mythweather/weather.cpp	2005-03-14 03:33:00.638291169 -0800
@@ -2483,9 +2483,9 @@
 
 bool Weather::GetStaticRadarMap()
 {
-    QUrl weatherMapLink1URL(QString("http://w3.weather.com/weather/map/%1"
+    QUrl weatherMapLink1URL(QString("http://www.weather.com/weather/map/%1"
                                     "?from=LAPmaps&setcookie=1 HTTP/1.1\n"
-                                    "Connection: close\nHost: w3.weather.com\n\n\n")
+                                    "Connection: close\nHost: www.weather.com\n\n\n")
                                     .arg(locale));
 
     INETComms *weatherMapLink1 = new INETComms(weatherMapLink1URL);
@@ -2507,7 +2507,7 @@
     if (mapLoc == "<NULL>")
         return true;
      
-    QUrl weatherMapLink2URL("http://w3.weather.com/" + mapLoc); 
+    QUrl weatherMapLink2URL("http://www.weather.com/" + mapLoc); 
      
     if (debug)
         cerr << "MythWeather: Grabbing Weather Map Link (part 2) From: " 
@@ -2589,7 +2589,7 @@
          return false;
 
      // find radar maps url's
-     QString sURL = "http://w3.weather.com/weather/map/" + locale +
+     QString sURL = "http://www.weather.com/weather/map/" + locale +
         "?name=index_large_animated&day=1";
      QString tempData = "";
 
@@ -2602,7 +2602,7 @@
      if (mapLoc == "<NULL>")
          return false;
 
-     mapLoc = "http://w3.weather.com/" + mapLoc;
+     mapLoc = "http://www.weather.com/" + mapLoc;
      if (debug)
         cerr << "MythWeather: Grabbing Weather Map Link (part 2) From: "
            << mapLoc << endl;
