diff -Naur libdrm-2.4.62-old/include/drm/drm.h libdrm-2.4.62-new/include/drm/drm.h
--- libdrm-2.4.62-old/include/drm/drm.h	2015-06-29 11:19:22.000000000 -0700
+++ libdrm-2.4.62-new/include/drm/drm.h	2015-07-20 15:02:21.694987253 -0700
@@ -648,7 +648,11 @@
 	__u64 value;
 };
 
+#ifdef O_CLOEXEC
 #define DRM_CLOEXEC O_CLOEXEC
+#else
+#define DRM_CLOEXEC 0
+#endif
 struct drm_prime_handle {
 	__u32 handle;
 
