diff -Naur mythtv-0.23-20110720-226ba71-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp mythtv-0.23-20110720-226ba71-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp
--- mythtv-0.23-20110720-226ba71-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-07-23 04:19:47.000000000 -0700
+++ mythtv-0.23-20110720-226ba71-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-07-23 04:19:48.000000000 -0700
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
