diff -Naur mythtv-0.23-20110222-4fe0b09-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp mythtv-0.23-20110222-4fe0b09-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp
--- mythtv-0.23-20110222-4fe0b09-de0d6b3-old/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-03-03 06:57:33.000000000 -0800
+++ mythtv-0.23-20110222-4fe0b09-de0d6b3-new/mythplugins/mythvideo/mtd/dvdprobe.cpp	2011-03-03 06:57:52.000000000 -0800
@@ -15,6 +15,7 @@
 #endif
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <cmath>
 #include <climits> // for CDSL_CURRENT which is currently INT_MAX
