diff -Naur mythtv-0.23-27052-old/libs/libmythtv/avformatdecoder.cpp mythtv-0.23-27052-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.23-27052-old/libs/libmythtv/avformatdecoder.cpp	2010-11-01 19:19:57.000000000 -0700
+++ mythtv-0.23-27052-new/libs/libmythtv/avformatdecoder.cpp	2010-11-01 19:22:06.000000000 -0700
@@ -1974,6 +1974,9 @@
                 if (!codec_is_std(video_codec_id))
                     thread_count = 1;
 
+                if (enc->codec_id == CODEC_ID_SVQ3)
+                    thread_count = 1;
+
                 VERBOSE(VB_PLAYBACK, QString("Using %1 CPUs for decoding")
                         .arg(HAVE_THREADS ? thread_count : 1));
 
