diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmyth/mythcontext.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmyth/mythcontext.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmyth/mythcontext.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmyth/mythcontext.cpp	2012-03-24 09:20:39.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QPixmap>
 #include <QImage>
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmyth/programinfoupdater.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmyth/programinfoupdater.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmyth/programinfoupdater.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmyth/programinfoupdater.cpp	2012-03-24 09:31:44.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "programinfoupdater.h"
 #include "remoteutil.h"
 #include "compat.h"
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/httpcomms.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/httpcomms.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/httpcomms.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/httpcomms.cpp	2012-03-24 08:33:00.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/lcddevice.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/lcddevice.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/lcddevice.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/lcddevice.cpp	2012-03-24 08:36:23.000000000 -0700
@@ -7,6 +7,8 @@
     (c) 2002, 2003 Thor Sigvaldason, Dan Morphis and Isaac Richards
 */
 
+#include <unistd.h>
+
 // ANSI C headers
 #include <cstdlib>
 #include <cmath>
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/mythdbcon.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/mythdbcon.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/mythdbcon.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/mythdbcon.cpp	2012-03-24 08:29:06.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/mythsocket.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/mythsocket.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/mythsocket.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/mythsocket.cpp	2012-03-24 08:08:32.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 #include <cassert>
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/mythsocketthread.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/mythsocketthread.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/mythsocketthread.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/mythsocketthread.cpp	2012-03-24 08:25:31.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // ANSI C
 #include <cstdlib>
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/remotefile.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/remotefile.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythdb/remotefile.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythdb/remotefile.cpp	2012-03-24 08:35:46.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <iostream>
 using namespace std;
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythfreemheg/BaseClasses.h mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythfreemheg/BaseClasses.h
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythfreemheg/BaseClasses.h	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythfreemheg/BaseClasses.h	2012-03-24 08:39:27.000000000 -0700
@@ -89,7 +89,7 @@
             return MHSequence<BASE>::m_Values[--MHSequence<BASE>::m_VecSize];
         }
         // Push an element on the stack.
-        void Push(BASE b) { Append(b); }
+        void Push(BASE b) { this->Append(b); }
         // Return the top of the stack.
         BASE Top() { 
             MHASSERT(MHSequence<BASE>::m_VecSize > 0);
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/audioinputoss.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/audioinputoss.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/audioinputoss.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/audioinputoss.cpp	2012-03-24 10:22:46.000000000 -0700
@@ -20,6 +20,8 @@
  * 02110-1301, USA.
  */
 
+#include <unistd.h>
+
 #include "mythconfig.h"
 #if HAVE_SYS_SOUNDCARD_H
     #include <sys/soundcard.h>
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/firewiredevice.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/firewiredevice.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/firewiredevice.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/firewiredevice.cpp	2012-03-24 10:49:45.000000000 -0700
@@ -4,6 +4,8 @@
  *  Distributed as part of MythTV under GPL v2 and later.
  */
 
+#include <unistd.h>
+
 // C++ headers
 #include <algorithm>
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/firewirerecorder.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/firewirerecorder.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/firewirerecorder.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/firewirerecorder.cpp	2012-03-24 10:55:02.000000000 -0700
@@ -4,6 +4,8 @@
  *  Distributed as part of MythTV under GPL v2 and later.
  */
 
+#include <unistd.h>
+
 // MythTV includes
 #include "firewirerecorder.h"
 #include "firewirechannel.h"
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/previewgenerator.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/previewgenerator.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythtv/previewgenerator.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythtv/previewgenerator.cpp	2012-03-24 09:53:42.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 // C headers
 #include <cmath>
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythui/mythrender_vdpau.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythui/mythrender_vdpau.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythui/mythrender_vdpau.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythui/mythrender_vdpau.cpp	2012-03-24 08:49:06.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "math.h"
 
 #include <QSize>
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythupnp/upnpdevice.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythupnp/upnpdevice.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythupnp/upnpdevice.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythupnp/upnpdevice.cpp	2012-03-24 08:47:57.000000000 -0700
@@ -21,6 +21,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include "upnp.h"
 #include "upnpdevice.h"
 #include "httpcomms.h"
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythupnp/upnptasknotify.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythupnp/upnptasknotify.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythupnp/upnptasknotify.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythupnp/upnptasknotify.cpp	2012-03-24 08:45:32.000000000 -0700
@@ -21,6 +21,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 // ANSI C headers
 #include <cstdlib>
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythupnp/upnptasksearch.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythupnp/upnptasksearch.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/libs/libmythupnp/upnptasksearch.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/libs/libmythupnp/upnptasksearch.cpp	2012-03-24 08:52:04.000000000 -0700
@@ -21,6 +21,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <unistd.h>
+
 #include <compat.h>
 #include <stdlib.h>
 
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/programs/mythfrontend/backendconnectionmanager.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/programs/mythfrontend/backendconnectionmanager.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/programs/mythfrontend/backendconnectionmanager.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/programs/mythfrontend/backendconnectionmanager.cpp	2012-03-24 11:06:10.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <QCoreApplication>
 #include <QThreadPool>
 #include <QRunnable>
diff -Naur mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-0.23-20111005-bcd91d5-de0d6b3-old/mythtv/programs/mythtranscode/transcode.cpp	2011-11-10 16:10:16.000000000 -0800
+++ mythtv-0.23-20111005-bcd91d5-de0d6b3-new/mythtv/programs/mythtranscode/transcode.cpp	2012-03-24 11:04:41.000000000 -0700
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include <fcntl.h>
 #include <math.h>
 
