diff -Naur mythtv-0.23-26231-old/libs/libmythtv/avformatdecoder.cpp mythtv-0.23-26231-new/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.23-26231-old/libs/libmythtv/avformatdecoder.cpp	2010-09-11 09:33:30.000000000 -0700
+++ mythtv-0.23-26231-new/libs/libmythtv/avformatdecoder.cpp	2010-09-11 09:33:31.000000000 -0700
@@ -4717,6 +4717,14 @@
 
     GetNVP()->ReinitAudio();
 
+    if (using_passthru != DoPassThrough(codec_ctx))
+    {
+        // detected audio capabilities changed half-way
+        // recheck audio stream
+        VERBOSE(VB_AUDIO, LOC + "Audio config changed. Rescanning");
+        return SetupAudioStream();
+    }
+
     if (LCD *lcd = LCD::Get())
     {
         LCDAudioFormatSet audio_format;
