diff -Naur mythtv-0.23-26732-old/libs/libmyth/mediamonitor-unix.cpp mythtv-0.23-26732-new/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-0.23-26732-old/libs/libmyth/mediamonitor-unix.cpp	2010-10-10 07:52:04.000000000 -0700
+++ mythtv-0.23-26732-new/libs/libmyth/mediamonitor-unix.cpp	2010-10-10 07:52:05.000000000 -0700
@@ -222,11 +222,12 @@
     QTextStream  stream(udevinfo);
     QStringList  args;
 
+    args << "info";
     args << "-q";
     args << "name";
     args << "-rp";
     args << sysfs;
-    udevinfo->start("udevinfo", args);
+    udevinfo->start("udevadm", args);
 
     if (!udevinfo->waitForStarted(2000 /*ms*/))
     {
